Index: FreeImage/Source/FreeImage/PSDParser.cpp
===================================================================
--- FreeImage.orig/Source/FreeImage/PSDParser.cpp	2018-09-11 14:55:28.411603539 +0200
+++ FreeImage/Source/FreeImage/PSDParser.cpp	2018-09-11 14:56:03.955786070 +0200
@@ -97,7 +97,7 @@ enum {
 template <int N>
 class PSDGetValue {
 public:
-	static inline int get(const BYTE * iprBuffer) {} // error
+	static inline int get(const BYTE * iprBuffer) { return 0; } // error
 };
 
 template <>
