From 9180e172d2ea644b75bd081fabb3b95a1236d281 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <git@refi64.dev>
Date: Wed, 20 Mar 2024 10:27:14 -0500
Subject: [PATCH] Don't fail cross-compilatino on unsupported architectures

---
 chromium/scripts/build_ffmpeg.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py b/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
index 63a59a3cd4..0fc4ff809a 100755
--- a/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
+++ b/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
@@ -602,10 +602,10 @@ def main(argv):
   host_arch = ROBO_CONFIGURATION.host_architecture()
   parallel_jobs = 8
 
-  if target_os.split('-', 1)[0] != host_os and (host_os != 'linux' or
-                                                host_arch != 'x64'):
-    print('Cross compilation can only be done from a linux x64 host.')
-    return 1
+  # if target_os.split('-', 1)[0] != host_os and (host_os != 'linux' or
+  #                                               host_arch != 'x64'):
+  #   print('Cross compilation can only be done from a linux x64 host.')
+  #   return 1
 
   for os in ARCH_MAP.keys():
     if os != target_os and target_os != 'all':
-- 
2.43.2

