diff --git a/src/libANGLE/validationEGL.cpp b/src/libANGLE/validationEGL.cpp
--- a/src/libANGLE/validationEGL.cpp
+++ b/src/libANGLE/validationEGL.cpp
@@ -7054,9 +7054,8 @@ bool ValidateQuerySurface64KHR(const ValidationContext *val,
             break;
         default:
         {
-            EGLint querySurfaceValue;
             ANGLE_VALIDATION_TRY(
-                ValidateQuerySurface(val, dpy, surfaceID, attribute, &querySurfaceValue));
+                ValidateQuerySurface(val, dpy, surfaceID, attribute, nullptr));
         }
         break;
     }
