From 558fba70670e52f20c2941e5f4279e4ff86d0191 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 6 May 2021 15:52:34 +0000
Subject: [PATCH] Update metainfo file to follow the latest specification

---
 data/brasero.appdata.xml.in | 30 ++++++++++++++++++++++--------
 1 file changed, 22 insertions(+), 8 deletions(-)

diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index e7caf98fe..7b1a1cbed 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -1,7 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<application>
-  <id type="desktop">brasero.desktop</id>
-  <licence>CC0</licence>
+<component type="desktop-application">
+  <id>brasero.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
+  <_name>Brasero</_name>
+  <_summary>Create and copy CDs and DVDs</_summary>
   <description>
     <_p>
       Brasero is a application to burn CD/DVD for the GNOME Desktop.
@@ -16,12 +19,23 @@
     </_p>
   </description>
   <screenshots>
-    <screenshot type="default">https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=copy-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=audio-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=data-project.png</screenshot>
+    <screenshot type="default">
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=copy-disc.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=audio-disc.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=data-project.png</image>
+    </screenshot>
   </screenshots>
   <url type="homepage">https://wiki.gnome.org/Apps/Brasero</url>
-  <updatecontact>brasero-list@gnome.org</updatecontact>
+  <url type="bugtracker">https://gitlab.gnome.org/GNOME/brasero/issues</url>
+  <url type="donation">https://www.gnome.org/donate/</url>
+  <url type="help">https://help.gnome.org/users/brasero/stable/</url>
+  <url type="translate">https://wiki.gnome.org/TranslationProject</url>
+  <update_contact>brasero-list@gnome.org</update_contact>
+  <_developer_name>The GNOME Project</_developer_name>
   <project_group>GNOME</project_group>
   <translation type="gettext">brasero</translation>
-</application>
+</component>
--

Change id to match flatpak. Don't upstream, requires other changes.

---
diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index 7b1a1cbe..1a0f40c4 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop-application">
-  <id>brasero.desktop</id>
+  <id>org.gnome.Brasero</id>
+  <launchable type="desktop-id">org.gnome.Brasero.desktop</launchable>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-2.0+</project_license>
   <_name>Brasero</_name>
--

https://gitlab.gnome.org/GNOME/brasero/-/merge_requests/11#note_1870597

---
diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index 1a0f40c4..a652a0cc 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -36,7 +36,7 @@
   <url type="help">https://help.gnome.org/users/brasero/stable/</url>
   <url type="translate">https://wiki.gnome.org/TranslationProject</url>
   <update_contact>brasero-list@gnome.org</update_contact>
-  <_developer_name>The GNOME Project</_developer_name>
+  <_developer_name>Brasero Contributors</_developer_name>
   <project_group>GNOME</project_group>
   <translation type="gettext">brasero</translation>
 </component>
--

https://gitlab.gnome.org/GNOME/brasero/-/merge_requests/11#note_1870596

---
diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index a652a0cc..d76e9fe5 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -35,6 +35,7 @@
   <url type="donation">https://www.gnome.org/donate/</url>
   <url type="help">https://help.gnome.org/users/brasero/stable/</url>
   <url type="translate">https://wiki.gnome.org/TranslationProject</url>
+  <url type="vcs-browser">https://gitlab.gnome.org/GNOME/brasero</url>
   <update_contact>brasero-list@gnome.org</update_contact>
   <_developer_name>Brasero Contributors</_developer_name>
   <project_group>GNOME</project_group>
--

https://gitlab.gnome.org/GNOME/brasero/-/merge_requests/11#note_1870599

---
diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index d76e9fe5..85cdf8bd 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -40,4 +40,5 @@
   <_developer_name>Brasero Contributors</_developer_name>
   <project_group>GNOME</project_group>
   <translation type="gettext">brasero</translation>
+  <content_rating type="oars-1.1" />
 </component>
