diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index e7caf98f..81e8369f 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <application>
-  <id type="desktop">brasero.desktop</id>
-  <licence>CC0</licence>
+  <id type="desktop">org.gnome.Brasero.desktop</id>
+  <metadata_licence>CC0-1.0</metadata_licence>
+  <project_license> GPL-2.0-or-later</project_license>
   <description>
     <_p>
       Brasero is a application to burn CD/DVD for the GNOME Desktop.
@@ -16,12 +17,29 @@
     </_p>
   </description>
   <screenshots>
-    <screenshot type="default">https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=copy-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=audio-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=data-project.png</screenshot>
+    <screenshot type="default">
+      <caption>Disc copy</caption>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=copy-disc.png</image>
+    </screenshot>
+    <screenshot>
+      <caption>Burning an audio disc</caption>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=audio-disc.png</image>
+    </screenshot>
+    <screenshot>
+      <caption>Data project</caption>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=data-project.png</image>
+    </screenshot>
   </screenshots>
   <url type="homepage">https://wiki.gnome.org/Apps/Brasero</url>
+  <url type="bugtracker">https://gitlab.gnome.org/GNOME/brasero/-/issues</url>
+  <url type="help">https://help.gnome.org/users/brasero/stable/</url>
+  <url type="translate">https://l10n.gnome.org/module/brasero/</url>
+  <url type="vcs-browser">https://gitlab.gnome.org/GNOME/brasero</url>
   <updatecontact>brasero-list@gnome.org</updatecontact>
   <project_group>GNOME</project_group>
   <translation type="gettext">brasero</translation>
+  <content_rating type="oars-1.1" />
+  <releases>
+    <release version="3.12.3" date="2021-09-20" />
+  </releases>
 </application>
