https://gitlab.gnome.org/GNOME/brasero/-/merge_requests/23

---
diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index a63bdd32..ded15024 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -41,4 +41,347 @@
   <project_group>GNOME</project_group>
   <translation type="gettext">brasero</translation>
   <content_rating type="oars-1.1" />
+  <releases>
+    <release version="3.12.3" date="2021-09-20">
+      <description>
+        <ul>
+          <li>Blu-ray fixes (Bastien Nocera)</li>
+          <li>Support Tracker 3.0 (Carlos Garnacho)</li>
+          <li>Add CI (Bastien Nocera)</li>
+          <li>Transcode plugin gstreamer plugin improvement (Robert McQueen)</li>
+          <li>Fix crash with malformed URI (Geyslan)</li>
+          <li>Fix default project label (Pimm "de Chinchilla" Hogeling)</li>
+          <li>Man page fixes (Ville Skyttä)</li>
+          <li>Header guard fix (Ting-Wei Lan)</li>
+          <li>Help improvements (scootergrisen, Andre Klapper)</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.12.2" date="2017-07-31">
+      <description>
+        <ul>
+          <li>Build fixes</li>
+          <li>Support Tracker 2.0</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.12.1" date="2015-04-28">
+      <description>
+        <ul>
+          <li>Add a symbolic app icon</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.12.0" date="2014-11-13">
+      <description>
+        <ul>
+          <li>Fix crash in plugin dialog</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.11.4" date="2014-09-10">
+      <description>
+        <ul>
+          <li>Make desktop file validate</li>
+          <li>Make appdata file validate</li>
+          <li>Allow building with tracker 1.0</li>
+          <li>Fix problem with constructors</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.11.3" date="2013-12-16">
+      <description>
+        <ul>
+          <li>Indonesian translation of help</li>
+          <li>Support Tracker 0.18</li>
+          <li>Translation updates</li>
+          <li>Remove X session support (make Brasero work under Wayland)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.11.0" date="2013-11-22">
+      <description>
+        <ul>
+          <li>Translation updates</li>
+          <li>Include an appdata file</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.10.0" date="2013-11-22">
+      <description>
+        <ul>
+          <li>Translation updates</li>
+          <li>Fixed drawing issues in media selection combo boxes</li>
+          <li>Fix several crashers</li>
+          <li>Fixes for new autotools</li>
+          <li>Disable audio normalisation - it's broken since GStreamer 1.0</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.8.0" date="2013-03-26">
+      <description>
+        <ul>
+          <li>Documentation fixes</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.7.91" type="development" date="2013-03-07">
+      <description>
+        <ul>
+          <li>Add a question mark to a question</li>
+          <li>Support notification filtering</li>
+          <li>Remove nonexisting options from the man page</li>
+          <li>Fix while loop in `brasero_libisofs_write_image_to_fd_thread</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.6.0" date="2012-11-12">
+      <description>
+        <ul>
+          <li>Bump version to 3.6.0 (Tomas Bzatek)</li>
+          <li>Fix build, link libbrasero-burn against libm (for ceil(3)) (Frédéric Péters)</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.5.92" type="development" date="2012-09-18">
+      <description>
+        <ul>
+          <li>Bump version to 3.5.92 (Tomas Bzatek)</li>
+          <li>Fix build against gstreamer-plugins-base-1.0 (Tomas Bzatek)</li>
+          <li>Fix build without preview (Tomas Bzatek)</li>
+          <li>build: Require gstpbutils-1.0 instead of 0.10 (Tomas Bzatek)</li>
+          <li>Port to new documentation infrastructure (Javier Jardón)</li>
+          <li>Port brasero to GStreamer 1.0 (Tim-Philipp Müller)</li>
+          <li>brasero-project: include libxml/xmlerror.h for xmlError (Tim-Philipp Müller)</li>
+          <li>Put settings in the correct path (Robert Ancell)</li>
+          <li>Add Desktop Actions for Unity (Jeremy Bicha)</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.4.1" date="2012-04-16">
+      <description>
+        <ul>
+          <li>Bump version to 3.4.1 (David King)</li>
+          <li>[l10n] Fixes on Catalan translation (Jordi Serratosa)</li>
+          <li>Updated Vietnamese translation (Nguyễn Thái Ngọc Duy)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.4.0" date="2012-03-26">
+      <description>
+        <ul>
+          <li>Bump version to 3.4.0 (David King)</li>
+          <li>Added Telugu to DOC_LINGUAS (Sasi Bhushan)</li>
+          <li>Added Telugu help Translations (Sasi Bhushan)</li>
+          <li>Updated Japanese translation. (Jiro Matsuzawa)</li>
+          <li>Add initial Khmer translation (Khoem Sokhem)</li>
+          <li>hindi translation by Chandan Kumar (Rajesh Ranjan)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.3.92" type="development" date="2012-03-19">
+      <description>
+        <ul>
+          <li>Bump version to 3.3.92 (David King)</li>
+          <li>Malayalam translation updated by Sadiq and me (Praveen Arimbrathodiyil)</li>
+          <li>Updated Kannada Translation (Shankar Prasad)</li>
+          <li>Updated Kannada Translation (Shankar Prasad)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Fix formatting error in man page (Michael Biebl)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Updated Hungarian help translation (Gabor Kelemen)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.3.91" type="development" date="2012-03-05">
+      <description>
+        <ul>
+          <li>Bump version to 3.3.91 (David King)</li>
+          <li>Updated Persian Translation (Arash Mousavi)</li>
+          <li>Updated Belarusian translation. (Ihar Hrachyshka)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Update Czech help translation by Stepan Simsa (Petr Kovar)</li>
+          <li>git.mk: update from 'canonical source' (Ryan Lortie)</li>
+          <li>po: Added en_CA.po and updated LINGUAS file (Tiffany Antopolski)</li>
+          <li>Call gettext() where translated string array items are used. (Gabor Kelemen)</li>
+          <li>build: Support build against tracker 0.13/0.14 (Vincent Untz)</li>
+          <li>Updated Galician doc translations (Fran Diéguez)</li>
+          <li>Updated Dutch translation (Hannie Dumoleyn)</li>
+          <li>Added lv for Latvian in Makefile.am in help. (Peteris Krisjanis)</li>
+          <li>Added figure picture for Latvian help. (Peteris Krisjanis)</li>
+          <li>Added Latvian translation for help by Matīss Jānis Āboltiņš. (Peteris Krisjanis)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Fixed typo in Arabic translation (Abderrahim Kitouni)</li>
+          <li>Updated Slovenian translation (Andrej Žnidaršič)</li>
+          <li>Updated French translation (Bruno Brouard)</li>
+          <li>Updated French doc translation (Bruno Brouard)</li>
+          <li>Updated Macedonian translation. (Jovan Naumovski)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Updated Slovenian translation (Matej Urbančič)</li>
+          <li>Fix broken tags in Hungarian help translation. bgo #666258 (Gabor Kelemen)</li>
+          <li>[l10n]Added Brumese translation (Thura Hlaing)</li>
+          <li>Updated Russian translation (Yuri Myasoedov)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>[l10n] Updated German doc translation (Mario Blättermann)</li>
+          <li>Updated Spanish translation (Daniel Mustieles)</li>
+          <li>Updated Spanish translation (Daniel Mustieles)</li>
+          <li>Created page for Integrity Checking tool (Paulina Gonzalez)</li>
+          <li>Updated Spanish translation (Daniel Mustieles)</li>
+          <li>[l10n] Updated German doc translation (Gabor Karsay)</li>
+          <li>Updated Polish translation of help (Aleksander Łukasiewicz)</li>
+          <li>Fixed typo in split-track.page (Ekaterina Gerasimova)</li>
+          <li>Add a fix for optical drive not selected by default. (Luis Medinas)</li>
+          <li>Updated Belarusian translation. (Ihar Hrachyshka)</li>
+          <li>Updated Hungarian translation (Gabor Kelemen)</li>
+          <li>Updated Hungarian translation (Gabor Kelemen)</li>
+          <li>Updated Hungarian help translation (Gabor Kelemen)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Updated French doc translation (Bruno Brouard)</li>
+          <li>Honor aclocal flags, bug 641650 (Craig Keogh)</li>
+          <li>Correct CD cover dimensions, bug 660918 (Erling A. Jacobsen)</li>
+          <li>Update categories in brasero.desktop (Vincent Untz)</li>
+          <li>Use g_mkdtemp() rather than mkdtemp(), bug 623890 (David King)</li>
+          <li>Link with uninstalled libraries, not system ones (Mike Auty)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.2.0" date="2011-09-27">
+      <description>
+        <ul>
+          <li>Bump to 3.2.0 (Luis Medinas)</li>
+          <li>Updated Traditional Chinese translation(Hong Kong and Taiwan) (Chao-Hsiung Liao)</li>
+          <li>Updated Galician translations (Fran Dieguez)</li>
+          <li>[l10n] Updated German doc translation (Mario Blättermann)</li>
+          <li>[l10n] Updated German doc translation (Mario Blättermann)</li>
+          <li>[l10n] Changed screenshot for German doc translation (Mario Blättermann)</li>
+          <li>Added Korean doc translation (Seong-ho Cho)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.1.92" type="development" date="2011-09-20">
+      <description>
+        <ul>
+          <li>Bump to 3.1.92 (Luis Medinas)</li>
+          <li>build: Support tracker 0.11/0.12 (Vincent Untz)</li>
+          <li>Updated Belarusian translation (some fixes after manual testing). (Ihar Hrachyshka)</li>
+          <li>Updated Japanese translation (Jiro Matsuzawa)</li>
+          <li>Updated Spanish translation (dmustieles)</li>
+          <li>Updated Spanish translation (Daniel Mustieles)</li>
+          <li>Updated galician translations (Fran Dieguez)</li>
+          <li>Updated Persian translation (Arash Mousavi)</li>
+          <li>Updated Persian translation (Arash Mousavi)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Fixed one string (Jorge González)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>String was not necessary (Jorge González)</li>
+          <li>Fixed two strings (Jorge González)</li>
+          <li>Make maintiner mode enabled by default (Javier Jardón)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>[l10n] Updated German doc translation (Mario Blättermann)</li>
+          <li>Updated Spanish translation (dmustieles)</li>
+          <li>configure.in: Remove gconf dependency (Javier Jardón)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Updated Spanish translation (dmustieles)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Updated Spanish translation (Jorge González)</li>
+          <li>Rewrote help in Mallard (Ekaterina Gerasimova)</li>
+          <li>Updated Lithuanian translation (Aurimas Černius)</li>
+          <li>Updated Belarusian translation. (Ihar Hrachyshka)</li>
+          <li>build: fix a build warning (Cosimo Cecchi)</li>
+          <li>all: switch to g_format_size() (Cosimo Cecchi)</li>
+          <li>Updated POTFILES.in (Piotr Drąg)</li>
+          <li>Adapt configure for tracker-sparql. (Luis Medinas)</li>
+          <li>Use libtracker-sparql instead of libtracker-client since it's deprecated. Fixes bgo #651630. (Luis Medinas)</li>
+          <li>Updated Belarusian translation. (Ihar Hrachyshka)</li>
+          <li>Remove beagle from Brasero. Fixes bgo #657313 and #639257. (Luis Medinas)</li>
+          <li>Updated Japanese translation (OKANO Takayoshi)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Revert "build: add checks for beagle libraryies to configure as it is used in the code" (Stefan Sauer)</li>
+          <li>build: add checks for beagle libraryies to configure as it is used in the code (Stefan Sauer)</li>
+          <li>Make Brasero build with tracker 0.10. 0.8 is older enough and shouldn't be available in any system with GNOME 3.2. This is a tempory fix.</li>
+          <li>Fix bgo #649178 - Wrong semi-colon in brasero-nautilus.desktop. (Luis Medinas)</li>
+          <li>Fix Documenters list in the About Dialog. Fixes bgo #646112. Added myself to Application authors (Luis Medinas)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.1.90" type="development" date="2011-08-30">
+      <description>
+        <ul>
+          <li>Updated POTFILES.in (Piotr Drąg)</li>
+          <li>Adapt configure for tracker-sparql. (Luis Medinas)</li>
+          <li>Use libtracker-sparql instead of libtracker-client since it's deprecated. Fixes bgo #651630. (Luis Medinas)</li>
+          <li>Updated Belarusian translation. (Ihar Hrachyshka)</li>
+          <li>Remove beagle from Brasero. Fixes bgo #657313 and #639257. (Luis Medinas)</li>
+          <li>Updated Japanese translation (OKANO Takayoshi)</li>
+          <li>Updated Traditional Chinese translation(Hong Kong and Taiwan) (Cheng-Chia Tseng)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Revert "build: add checks for beagle libraryies to configure as it is used in the code" (Stefan Sauer)</li>
+          <li>build: add checks for beagle libraryies to configure as it is used in the code (Stefan Sauer)</li>
+          <li>Make Brasero build with tracker 0.10. 0.8 is older enough and shouldn't be available in any system with GNOME 3.2. This is a temporary fix until we move to tracker-sparql. (Luis Medinas)</li>
+          <li>Fix bgo #649178 - Wrong semi-colon in brasero-nautilus.desktop. (Luis Medinas)</li>
+          <li>Fix Documenters list in the About Dialog. Fixes bgo #646112. Added myself to Application authors (Luis Medinas)</li>
+          <li>Bump to 3.1.5. (Luis Medinas)</li>
+          <li>nautilus: wait for the track to load file info before adding to session (Cosimo Cecchi)</li>
+          <li>Updated Thai translation. (Theppitak Karoonboonyanan)</li>
+          <li>Added eu (Basque) language to DOC_LINGUAS (Inaki Larranaga Murgoitio)</li>
+          <li>Added Basque language (Inaki Larranaga Murgoitio)</li>
+          <li>Updated Finnish translation (Tommi Vainikainen)</li>
+          <li>all: don't use deprecated H/V variants of GTK widgets (Kjartan Maraas)</li>
+          <li>Revert accidental commit (Kjartan Maraas)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Remove GtkHBox/GtkVBox and friends (Kjartan Maraas)</li>
+          <li>Fix two typos. (Qijiang Fan)</li>
+          <li>Updated Turkish translation file header (Muhammet Kara)</li>
+          <li>Updated Turkish translation (Muhammet Kara)</li>
+          <li>Updated Russian translation (Yuri Kozlov)</li>
+          <li>Updated Swedish translation (Daniel Nylander)</li>
+          <li>Updated Norwegian bokmål translation (Kjartan Maraas)</li>
+          <li>Added Persian translation (Arash Mousavi)</li>
+          <li>Added Persian translation (Arash Mousavi)</li>
+          <li>[l10n]Updated Catalan (Valencian) translation (Carles Ferrando)</li>
+          <li>Use the primary-toolbar style class where applicable. (Cosimo Cecchi)</li>
+          <li>Added UG translation (Abduxukur Abdurixit)</li>
+          <li>Updated Turkish translation (Muhammet Kara)</li>
+          <li>Added UG translation (Abduxukur Abdurixit)</li>
+          <li>Updated Vietnamese translation (Nguyễn Thái Ngọc Duy)</li>
+          <li>po/vi.po: import from Damned Lies (Nguyễn Thái Ngọc Duy)</li>
+          <li>Updated Traditional Chinese translation(Hong Kong and Taiwan) (Chao-Hsiung Liao)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="3.0.0" date="2011-04-04">
+      <description>
+        <ul>
+          <li>Bump to 3.0.0 (Luis Medinas)</li>
+          <li>Updated Japanese translation. (Jiro Matsuzawa)</li>
+          <li>Updated Esperanto translation (Kristjan SCHMIDT)</li>
+          <li>[l10n]Fixes on Catalan translation (Jordi Serratosa)</li>
+          <li>Updated Japanese translation. (Takayuki KUSANO)</li>
+          <li>Updated kn translations (Shankar Prasad)</li>
+          <li>Updated Gujarati Translations (Sweta Kothari)</li>
+          <li>hindi updated (Rajesh Ranjan)</li>
+          <li>Updated Danish translation (Kris Thomsen)</li>
+          <li>Added UG translation (Abduxukur Abdurixit)</li>
+          <li>Updated Tamil translation (Dr.T.Vasudevan)</li>
+          <li>Updated Tamil translation (Dr.T.Vasudevan)</li>
+          <li>update tranlation for Punjabi by A S Alam (A S Alam)</li>
+          <li>Updated Hebrew translation. (Yaron Shahrabani)</li>
+          <li>Updated Swedish translation (Daniel Nylander)</li>
+        </ul>
+      </description>
+    </release>
+  </releases>
 </component>
