diff --git a/tools/gn_args.py b/tools/gn_args.py
index 0359f2feb..3a1c029f9 100644
--- a/tools/gn_args.py
+++ b/tools/gn_args.py
@@ -368,14 +368,6 @@ def ValidateArgs(args):
         'x86', 'x64', 'arm',
         'arm64'), 'target_cpu must be "x86", "x64", "arm" or "arm64"'
 
-  if platform == 'linux':
-    if target_cpu == 'x86':
-      assert use_sysroot, 'target_cpu="x86" requires use_sysroot=true'
-    elif target_cpu == 'arm':
-      assert use_sysroot, 'target_cpu="arm" requires use_sysroot=true'
-    elif target_cpu == 'arm64':
-      assert use_sysroot, 'target_cpu="arm64" requires use_sysroot=true'
-
   # ASAN requires Release builds.
   if is_asan:
     assert not is_debug, "is_asan=true requires is_debug=false"
