diff --git a/cmake/cef_variables.cmake.in b/cmake/cef_variables.cmake.in
index f3f64ea3a..e656568c3 100644
--- a/cmake/cef_variables.cmake.in
+++ b/cmake/cef_variables.cmake.in
@@ -25,15 +25,10 @@ elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
 endif()
 
 # Determine the project architecture.
-if(NOT DEFINED PROJECT_ARCH)
-  if(("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "arm64") OR
-     ("${CMAKE_CXX_COMPILER_ARCHITECTURE_ID}" STREQUAL "ARM64"))
-    set(PROJECT_ARCH "arm64")
-  elseif(CMAKE_SIZEOF_VOID_P MATCHES 8)
-    set(PROJECT_ARCH "x86_64")
-  else()
-    set(PROJECT_ARCH "x86")
-  endif()
+if("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
+  set(PROJECT_ARCH "x86_64")
+else()
+  set(PROJECT_ARCH "arm64")
 endif()
 
 if(${CMAKE_GENERATOR} STREQUAL "Ninja")
