diff --git a/build/rust/allocator/lib.rs b/build/rust/allocator/lib.rs
index 142c6e6084..15474f1b05 100644
--- a/build/rust/allocator/lib.rs
+++ b/build/rust/allocator/lib.rs
@@ -90,6 +90,12 @@ mod both_allocators {
     #[linkage = "weak"]
     fn __rust_no_alloc_shim_is_unstable_v2() {}
 
+    #[rustc_std_internal_symbol]
+    #[linkage = "weak"]
+    fn __rust_alloc_error_handler_should_panic_v2() -> u8 {
+        0
+    }
+
     // TODO(crbug.com/422538133) Remove after rolling past
     // https://github.com/rust-lang/rust/pull/141061
     #[no_mangle]
