From 54d94c684d87152e98ebbf589b3dfc4e2b852034 Mon Sep 17 00:00:00 2001
From: Felipe Borges <felipeborges@gnome.org>
Date: Wed, 3 Jun 2020 16:31:31 +0200
Subject: [PATCH] WIP

---
 genisoimage/genisoimage.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/genisoimage/genisoimage.h b/genisoimage/genisoimage.h
index bbedfb0..6f0f7d1 100644
--- a/genisoimage/genisoimage.h
+++ b/genisoimage/genisoimage.h
@@ -376,7 +376,7 @@ extern int	use_fileversion;
 extern int	split_SL_component;
 extern int	split_SL_field;
 extern char	*trans_tbl;
-char		*outfile;
+extern char		*outfile;

 #define	JMAX		64	/* maximum Joliet file name length (spec) */
 #define	JLONGMAX	103	/* out of spec Joliet file name length */
--
2.26.2
