commit 805b073ae0952134b3226d0ad3e5710bb429b438
Author: Sam Thursfield <sam@afuera.me.uk>
Date:   Wed Nov 13 11:21:24 2019 +0100

    Makefile: Don't use `rsync`

diff --git a/pd/src/makefile.in b/pd/src/makefile.in
index 4f30df23..e220c98c 100644
--- a/pd/src/makefile.in
+++ b/pd/src/makefile.in
@@ -180,7 +180,6 @@ install: all
 	install -d $(DESTDIR)$(libpdbindir)/css
 	install -d $(DESTDIR)$(libpdbindir)/locales
 	install -m644 ../nw/*.* $(DESTDIR)$(libpdbindir)/
-	rsync -ax --exclude=.git --exclude=.svn ../nw/nw $(DESTDIR)$(libpdbindir)
 	install -m644 ../nw/css/*.* $(DESTDIR)$(libpdbindir)/css/
 	for dir in $(shell ls -1 ../nw/locales); do \
 		echo "installing locale $$dir"; \
@@ -195,12 +194,13 @@ install: all
 	ln -sf $(bindir)/pd-l2ork $(DESTDIR)$(libpdbindir)/pd-l2ork
 	install -m755 $(BIN_DIR)/pdsend $(DESTDIR)$(bindir)/pdsend
 	install -m755 $(BIN_DIR)/pdreceive $(DESTDIR)$(bindir)/pdreceive 
-	rsync -ax --exclude=.git --exclude=.svn ../doc/* $(DESTDIR)$(pddocdir)
+	install -d $(DESTDIR)$(pddocdir)
+	cp -a ../doc/* $(DESTDIR)$(pddocdir)/
 	mv $(ABOUT_FILE) $(ABOUT_FILE).tmp
 	cat $(ABOUT_FILE).tmp | sed 's|PD_VERSION|Pd-l2ork version $(PD_L2ORK_VERSION) ($(PD_BUILD_VERSION))|' \
 		> $(ABOUT_FILE)
 	rm $(ABOUT_FILE).tmp
-	rsync -ax --exclude=.git --exclude=.svn ../extra $(DESTDIR)$(libpddir)/
+	cp -a ../extra $(DESTDIR)$(libpddir)/
 	install -d $(DESTDIR)$(includedir)/pd-l2ork
 	install -m644 m_pd.h $(DESTDIR)$(includedir)/pd-l2ork/m_pd.h
 	install -m644 m_imp.h $(DESTDIR)$(includedir)/pd-l2ork/m_imp.h
