diff --git a/cmake/SIPMacros.cmake b/cmake/SIPMacros.cmake
index ce2096c..0305eb5 100644
--- a/cmake/SIPMacros.cmake
+++ b/cmake/SIPMacros.cmake
@@ -124,6 +124,6 @@ macro(ADD_SIP_PYTHON_MODULE MODULE_NAME MODULE_SIP)
         set_target_properties(${_logical_name} PROPERTIES SUFFIX ".pyd" IMPORT_PREFIX "_")
     endif(WIN32)
 
-    install(TARGETS ${_logical_name} DESTINATION "${Python3_SITEARCH}/${_parent_module_path}")
+    install(TARGETS ${_logical_name} DESTINATION "/app/lib/python3.9/site-packages/${_parent_module_path}")
 
-endmacro(ADD_SIP_PYTHON_MODULE)
\ No newline at end of file
+endmacro(ADD_SIP_PYTHON_MODULE)
