diff --git a/src/core/config/linux.pri b/src/core/config/linux.pri
index 85b948db..a08b03f7 100644
--- a/src/core/config/linux.pri
+++ b/src/core/config/linux.pri
@@ -6,7 +6,7 @@ gn_args += \
     use_gio=false \
     use_gnome_keyring=false \
     linux_use_bundled_binutils=false \
-    use_udev=true \
+    use_udev=false \
     use_bundled_fontconfig=false \
     use_sysroot=false \
     enable_session_service=false \
