From fecb912308a716e3e83bfbf2c2b3a07f9dface7b Mon Sep 17 00:00:00 2001
From: nmlynch94 <nmlynch94@gmail.com>
Date: Sun, 26 Oct 2025 12:21:35 -0400
Subject: [PATCH] remove speech service dependency

---
 chrome/test/BUILD.gn | 1 -
 1 file changed, 1 deletion(-)

diff --git a/chrome/test/BUILD.gn b/chrome/test/BUILD.gn
index e0440fef57..710073a082 100644
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -8492,7 +8492,6 @@ test("unit_tests") {
       "//chrome/browser/webauthn:test_support",
       "//chrome/browser/webauthn/proto",
       "//chrome/common/read_anything:mojo_bindings",
-      "//chrome/services/speech:unit_tests",
       "//components/app_constants",
       "//components/application_locale_storage",
       "//components/color",
-- 
2.51.1.dirty

