From ed6e6f982b2a5a24190e2b5c821863a1d98b9a20 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tobias=20Sch=C3=B6nberg?= <tobias47n9e@gmail.com>
Date: Sun, 29 Aug 2021 12:55:15 +0200
Subject: [PATCH] Remove launcher self-update

---
 src/launcher_wizard.js | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/src/launcher_wizard.js b/src/launcher_wizard.js
index 144481a..a16e25c 100644
--- a/src/launcher_wizard.js
+++ b/src/launcher_wizard.js
@@ -32,22 +32,6 @@ class Wizard extends EventEmitter {
 	generateSteps() {
 		var steps = [];
 		if (!config.no_downloads) {
-			steps.push({
-				name: 'launcher_update',
-				action: () => {
-					const isDev = require('electron-is-dev');
-					log.info('Checking for launcher update');
-					if (!isDev) {
-						updater.checkForUpdates();
-					} else {
-						console.log('Development version: no self-update required');
-						setTimeout(() => {
-							this.nextStep();
-						}, 300);
-					}
-				}
-			});
-
 			config.downloads.engines.forEach((engine) => {
 				steps.push({
 					name: 'engine',
-- 
2.30.2

