From c71d9a8800e39f2b03e9c4478b82fd723e19f06f Mon Sep 17 00:00:00 2001
From: nmlynch94 <nmlynch94@gmail.com>
Date: Mon, 27 Oct 2025 11:21:04 -0400
Subject: [PATCH] ungoogled chromium remove extra locales

---
 patches/series | 1 -
 1 file changed, 1 deletion(-)

diff --git a/patches/series b/patches/series
index a5bbd9b..34535b0 100644
--- a/patches/series
+++ b/patches/series
@@ -105,7 +105,6 @@ extra/ungoogled-chromium/first-run-page.patch
 extra/ungoogled-chromium/add-flag-to-reduce-system-info.patch
 extra/ungoogled-chromium/add-flag-to-remove-client-hints.patch
 extra/ungoogled-chromium/disable-downloads-page-referrer-url.patch
-extra/ungoogled-chromium/enable-extra-locales.patch
 extra/ungoogled-chromium/disable-chromelabs.patch
 extra/ungoogled-chromium/remove-pac-size-limit.patch
 extra/ungoogled-chromium/enable-certificate-transparency-and-add-flag.patch
-- 
2.51.1.dirty

