diff --git a/utils/prune_binaries.py b/utils/prune_binaries.py
index 30803305..1f5a8ac1 100755
--- a/utils/prune_binaries.py
+++ b/utils/prune_binaries.py
@@ -215,12 +215,7 @@ def _callback(args):
     unremovable_files = prune_files(args.directory, prune_list)
     if unremovable_files:
         file_list = '\n'.join(f for f in itertools.islice(unremovable_files, 5))
-        if len(unremovable_files) > 5:
-            file_list += '\n... and ' + str(len(unremovable_files) - 5) + ' more'
-            get_logger().debug('files that could not be pruned:\n%s',
-                               '\n'.join(f for f in unremovable_files))
-        get_logger().error('%d files could not be pruned:\n%s', len(unremovable_files), file_list)
-        sys.exit(1)
+        get_logger().debug('%d files could not be pruned:\n%s', len(unremovable_files), file_list)
 
 
 def main():
