diff --git a/Makefile b/Makefile
index 119110d..22b1f31 100644
--- a/Makefile
+++ b/Makefile
@@ -46,7 +46,7 @@ TO_MAN= lua.1 luac.1
 
 # Lua version and release.
 V= 5.3
-R= $V.4
+R= $V.5
 
 # Targets start here.
 all:	$(PLAT)
diff --git a/src/luaconf.h b/src/luaconf.h
index 9eeeea6..95250da 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -11,6 +11,9 @@
 #include <limits.h>
 #include <stddef.h>
 
+#define LUA_COMPAT_5_1
+#define LUA_COMPAT_5_2
+
 
 /*
 ** ===================================================================
@@ -200,7 +203,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"$FLATPAK_DEST/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
@@ -745,7 +748,7 @@
 @@ of a function in debug information.
 ** CHANGE it if you want a different size.
 */
-#define LUA_IDSIZE	60
+#define LUA_IDSIZE	512
 
 
 /*
