From ca921c88f17c154d27583edc0240c172a66da445 Mon Sep 17 00:00:00 2001
From: gasinvein <gasinvein@gmail.com>
Date: Thu, 1 Oct 2020 12:59:43 +0300
Subject: [PATCH] Fix lib install

---
 src/meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index a91bc12..808e1e5 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -44,4 +44,5 @@ shared_library(meson.project_name().to_lower(),
     vkBasalt_src, shader_include,
     include_directories : vkBasalt_include_path,
     dependencies : [x11_dep, reshade_dep],
-    install : lib_dir)
+    install : true,
+    install_dir : lib_dir)
-- 
2.26.2

