
diff --git a/cmake/templates/appdata.xml.in b/cmake/templates/appdata.xml.in
index f130671..ee774b7 100644
--- a/cmake/templates/appdata.xml.in
+++ b/cmake/templates/appdata.xml.in
@@ -9,7 +9,7 @@
   <url type="homepage">@META_APP_URL@</url>
   <url type="bugtracker">@META_APP_BUGTRACKER_URL@</url>
   <launchable type="desktop-id">@META_ID@.desktop</launchable>
-  <developer><name>@META_APP_AUTHOR@</name></developer>
+  <developer id="io.github.martchus"><name>@META_APP_AUTHOR@</name></developer>
   <provides>
     <binary>@META_TARGET_NAME@</binary>
   </provides>
