From 9267ae4f88249bb88fc73ea380c1575fdc0fde5b Mon Sep 17 00:00:00 2001
From: Luna <advaith.madhukar@gmail.com>
Date: Sun, 10 Oct 2021 07:41:33 +0530
Subject: [PATCH 1/2] change default install to /app

---
 ykman-gui/deployment.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ykman-gui/deployment.pri b/ykman-gui/deployment.pri
index 779f74e..d2435c4 100644
--- a/ykman-gui/deployment.pri
+++ b/ykman-gui/deployment.pri
@@ -3,7 +3,7 @@ unix:!android {
         qnx {
             target.path = /tmp/$${TARGET}/bin
         } else {
-            target.path = /usr/bin
+            target.path = /app/bin
         }
         export(target.path)
     }
-- 
2.32.0

