diff --git a/data/simple-scan.appdata.xml.in b/data/simple-scan.appdata.xml.in
index ce92f9c4..9e3b40bf 100644
--- a/data/simple-scan.appdata.xml.in
+++ b/data/simple-scan.appdata.xml.in
@@ -88,27 +88,6 @@
         </p>
       </description>
     </release>
-    <release date="2021-02-25" version="40.beta" type="development">
-      <description>
-        <p>
-          This is an unstable release in the 40 development series,
-          with the following changes:
-        </p>
-        <ul>
-          <li>Don't repeat vendor name in the device name</li>
-          <li>GNOME 40 consistency updates</li>
-          <li>Fix crop size after decreasing page size</li>
-          <li>UI fixes: change crop icon, improve layout of buttons</li>
-          <li>Add redetect button to device list</li>
-          <li>Add new resolution 200 DPI</li>
-          <li>Add zero second delay for Multiple Pages from FlatBed</li>
-          <li>Display an error, when no documents found in the feeder</li>
-          <li>Rename 'Start again' button to 'New document'</li>
-          <li>Fix crash on lexmark backend</li>
-        </ul>
-        <p>This release also updates translations.</p>
-      </description>
-    </release>
     <release date="2020-10-02" version="3.38.1">
       <description>
         <p>
