From 1e9b167a6dc13dcb8a82ac1297e372807bdc1a8d Mon Sep 17 00:00:00 2001
From: Carl Schwan <carl@carlschwan.eu>
Date: Wed, 28 Aug 2024 15:52:12 +0200
Subject: [PATCH] Fix desktop file of launcher

---
 launcher/org.kde.calligra.desktop | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/launcher/org.kde.calligra.desktop b/launcher/org.kde.calligra.desktop
index d198222a8f0..7b8ad179bb6 100644
--- a/launcher/org.kde.calligra.desktop
+++ b/launcher/org.kde.calligra.desktop
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Type=Application
-Name=Calligra Launcher
+Name=Calligra
 Exec=calligralauncher
-Comment=Office Suite Launcher
-Icon=calligrawords
+Comment=Office Suite
+Icon=org.kde.calligra.words
 X-KDE-StartupNotify=true
 Categories=Qt;KDE;Office;WordProcessor;
-NoDisplay=true
-- 
2.45.2

