From 9c46f3d2b2c2109d6ad65690543b14de5aae5b18 Mon Sep 17 00:00:00 2001
From: Carl Schwan <carl@carlschwan.eu>
Date: Thu, 29 Aug 2024 18:12:30 +0200
Subject: [PATCH] Disable gemini completely

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fefcf3cbb76..c8c9afdb7cf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -974,9 +974,9 @@ if(SHOULD_BUILD_PART_COMPONENTS)
     add_subdirectory(components)
 endif()
 
-if(SHOULD_BUILD_GEMINI)
-    add_subdirectory(gemini)
-endif()
+#if(SHOULD_BUILD_GEMINI)
+#    add_subdirectory(gemini)
+#endif()
 
 # non-app directories are moved here because they can depend on SHOULD_BUILD_{appname} variables set above
 add_subdirectory(libs)
-- 
2.45.2

