diff --git a/Telegram/ThirdParty/dispatch/cmake/modules/DispatchCompilerWarnings.cmake b/Telegram/ThirdParty/dispatch/cmake/modules/DispatchCompilerWarnings.cmake
index 35b80f3..cc554ed 100644
--- a/Telegram/ThirdParty/dispatch/cmake/modules/DispatchCompilerWarnings.cmake
+++ b/Telegram/ThirdParty/dispatch/cmake/modules/DispatchCompilerWarnings.cmake
@@ -2,7 +2,6 @@
 if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")
   # TODO: someone needs to provide the msvc equivalent warning flags
 else()
-  add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Werror>)
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Wall>)
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Wextra>)
