diff --git a/src/controlador/Configuer.java b/src/controlador/Configuer.java
index ebe71aa..4546349 100644
--- a/src/controlador/Configuer.java
+++ b/src/controlador/Configuer.java
@@ -37,7 +37,7 @@ public class Configuer {
     }
 
     private void LoadCfg() {
-        String tmp = System.getProperty("user.dir") + File.separator + "config.chc";
+        String tmp = System.getenv("XDG_CONFIG_HOME") + File.separator + "config.chc";
         File f = new File(tmp);
 
         ResourceBundle resourceMap = getResourceMap();
@@ -104,7 +104,7 @@ public class Configuer {
     }
 
     public void SaveCfg() {
-        String tmp = System.getProperty("user.dir") + File.separator + "config.chc";
+        String tmp = System.getenv("XDG_CONFIG_HOME") + File.separator + "config.chc";
         try {
             try (FileOutputStream cfgFile = new FileOutputStream(tmp)) {
                 Properties prop = new Properties();
@@ -219,7 +219,7 @@ public class Configuer {
     }
     
     public String getAutoSaveFile() {
-        String tmp = System.getProperty("user.dir") + File.separator + "autosave.chc";
+        String tmp = System.getenv("XDG_DATA_HOME") + File.separator + "autosave.chc";
         return tmp;
     }
 }
diff --git a/src/helper/AjudaManager.java b/src/helper/AjudaManager.java
index 7106f95..3000d08 100644
--- a/src/helper/AjudaManager.java
+++ b/src/helper/AjudaManager.java
@@ -73,7 +73,7 @@ public class AjudaManager extends ParteAjuda {
     }
 
     public static boolean SaveDataHelp(AjudaManager obj) {
-        String tmp = System.getProperty("user.dir") + File.separator + "Ajuda.brMh";
+        String tmp = "/app/br-modelo/Ajuda.brMh";
         obj.setMudou(false);
         if (!SaveData(obj, tmp)) {
             obj.setMudou(true);
diff --git a/src/partepronta/GerenciadorPartes.java b/src/partepronta/GerenciadorPartes.java
index f37d2e9..8c1c27e 100644
--- a/src/partepronta/GerenciadorPartes.java
+++ b/src/partepronta/GerenciadorPartes.java
@@ -115,7 +115,7 @@ public class GerenciadorPartes implements Serializable{
     }
     
     public static GerenciadorPartes LoadDataTemplate() {
-        String tmp = System.getProperty("user.dir") + File.separator + "Template.brMt";
+        String tmp = System.getenv("XDG_DATA_HOME") + File.separator + "Template.brMt";
         GerenciadorPartes gp = LoadData(tmp);
         if (gp == null) {
             gp = new GerenciadorPartes();
@@ -139,7 +139,7 @@ public class GerenciadorPartes implements Serializable{
     }
     
     public static boolean SaveDataTemplate(GerenciadorPartes obj) {
-        String tmp = System.getProperty("user.dir") + File.separator + "Template.brMt";
+        String tmp = System.getenv("XDG_DATA_HOME") + File.separator + "Template.brMt";
         return SaveData(obj, tmp);
     }
     //</editor-fold>
