--- ./resources/CMakeLists.txt.old	2021-04-16 10:17:35.663338223 -0300
+++ ./resources/CMakeLists.txt		2021-04-16 10:18:09.786338573 -0300
@@ -24,7 +24,7 @@
    INSTALL(FILES "${RESOURCEDIR}/GrandOrgue.icns" DESTINATION "${RESOURCEINSTDIR}")
    LIST(APPEND DEPLIST "${RESOURCEDIR}/GrandOrgue.icns")
 elseif(UNIX)
-  install(FILES GrandOrgue.png DESTINATION share/pixmaps)
+  install(FILES GrandOrgue.png DESTINATION share/icons/hicolor/32x32/apps)
   install(FILES grandorgue.xml DESTINATION share/mime/packages)
 
   if (GETTEXT_FOUND)
