From 3edfb493cde3d45a810cf3b14c53bfc6b8fd1d86 Mon Sep 17 00:00:00 2001
From: Luna <advaith.madhukar@gmail.com>
Date: Sun, 10 Oct 2021 07:42:09 +0530
Subject: [PATCH 2/2] don't pip install dep, we already install it

---
 ykman-cli/ykman-cli.pro | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/ykman-cli/ykman-cli.pro b/ykman-cli/ykman-cli.pro
index bd941d7..0039f8a 100644
--- a/ykman-cli/ykman-cli.pro
+++ b/ykman-cli/ykman-cli.pro
@@ -27,9 +27,6 @@ mac|win32 {
 macx {
     pip.commands = python3 -m venv pymodules && source pymodules/bin/activate && pip3 install -r ../requirements.txt && deactivate
 }
-!macx {
-    pip.commands = pip3 install -r ../requirements.txt --target pymodules
-}
 
 # On mac, embedd a Info.plist file in the binary, needed for codesign
 macx{
-- 
2.32.0

