From 4d39f9e7b19e8e411bc3720726121d1a936e1255 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Thu, 1 Oct 2020 16:39:00 -0500
Subject: [PATCH 2/2] Ensure libffmpeg.so is in the rpath of binaries

---
 third_party/ffmpeg/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/third_party/ffmpeg/BUILD.gn b/third_party/ffmpeg/BUILD.gn
index 387216d9b..cf5d38099 100755
--- a/third_party/ffmpeg/BUILD.gn
+++ b/third_party/ffmpeg/BUILD.gn
@@ -458,6 +458,12 @@ if (is_component_ffmpeg) {
     if (is_mac && !is_component_build) {
       ldflags += [ "-Wl,-install_name,@rpath/libffmpeg.dylib" ]
     }
+
+    if (is_linux && !is_component_build) {
+      public_configs += [
+        "//build/config/gcc:rpath_for_built_shared_libraries",
+      ]
+    }
   }
 } else {
   source_set("ffmpeg") {
-- 
2.26.2

