From 46251955728ebfa92e2f1a435ba10e64fbcf9768 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dami=C3=A1n=20Nohales?= <damiannohales@gmail.com>
Date: Sun, 8 Nov 2020 12:23:39 -0300
Subject: [PATCH 2/2] flatpak: Set @electron/get cacheRoot to flatpak-node dir

---
 script/build.ts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/script/build.ts b/script/build.ts
index 8ffa764c9..00fd24dea 100755
--- a/script/build.ts
+++ b/script/build.ts
@@ -179,6 +179,9 @@ function packageApp() {
     dir: outRoot,
     overwrite: true,
     tmpdir: false,
+    download: {
+      cacheRoot: path.join(projectRoot, 'flatpak-node', 'electron-cache'),
+    },
     derefSymlinks: false,
     prune: false, // We'll prune them ourselves below.
     ignore: [
-- 
2.26.2

