From 4c2a2c69546db32b9282dd7093370f8022610b59 Mon Sep 17 00:00:00 2001
From: nullrequest <30698906+Lunarequest@users.noreply.github.com>
Date: Sat, 4 Sep 2021 21:26:47 +0530
Subject: [PATCH 3/3] disable autoupdate if running in flatpak

---
 main.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/main.js b/main.js
index 979580a..4625fcc 100644
--- a/main.js
+++ b/main.js
@@ -38,6 +38,8 @@ const mprisProvider = require('./src/providers/mprisProvider')
 /* Variables =========================================================================== */
 const defaultUrl = 'https://music.youtube.com'
 
+const isFlatpak = (process.platform === 'linux' && process.env.FLATPAK_HOST==='1')
+
 let mainWindow,
     view,
     miniplayer,
@@ -1739,7 +1741,7 @@ if (!gotTheLock) {
                 tray.updateImage(payload)
         })
 
-        if (!isDev) {
+        if (!isDev && isFlatpak===false) {
             updater.checkUpdate(mainWindow, view)
 
             setInterval(function () {
-- 
2.32.0

