From 1dc9c0f4beb166649b804ff1cfc4692dc729d183 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Thu, 1 Oct 2020 16:39:00 -0500
Subject: [PATCH 5/5] Ensure libffmpeg.so is in the rpath of binaries

---
 third_party/ffmpeg/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/third_party/ffmpeg/BUILD.gn b/third_party/ffmpeg/BUILD.gn
index 337e45cb3..305f10d48 100755
--- a/third_party/ffmpeg/BUILD.gn
+++ b/third_party/ffmpeg/BUILD.gn
@@ -463,6 +463,12 @@ if (is_component_ffmpeg) {
     if (is_mac && !is_component_build) {
       ldflags += [ "-Wl,-install_name,@rpath/libffmpeg.dylib" ]
     }
+
+    if (is_linux && !is_component_build) {
+      all_dependent_configs = [
+        "//build/config/gcc:rpath_for_built_shared_libraries",
+      ]
+    }
   }
 } else {
   source_set("ffmpeg") {
-- 
2.26.2

