From cfc044a457e476a97adace1c5f3d113e985bae71 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Wed, 26 Aug 2020 20:30:35 -0500
Subject: [PATCH 05/11] Enable Chromecast by default

---
 extensions/common/feature_switch.cc | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/extensions/common/feature_switch.cc b/extensions/common/feature_switch.cc
index ae90db29eae44..95d72d5e8f422 100644
--- a/extensions/common/feature_switch.cc
+++ b/extensions/common/feature_switch.cc
@@ -39,12 +39,7 @@ class CommonSwitches {
                          FeatureSwitch::DEFAULT_ENABLED),
         load_media_router_component_extension(
             kLoadMediaRouterComponentExtensionFlag,
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
-            FeatureSwitch::DEFAULT_ENABLED)
-#else
-            FeatureSwitch::DEFAULT_DISABLED)
-#endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING)
-  {
+            FeatureSwitch::DEFAULT_ENABLED) {
   }
 
   FeatureSwitch force_dev_mode_highlighting;
@@ -81,10 +76,9 @@ FeatureSwitch* FeatureSwitch::load_media_router_component_extension() {
 
 FeatureSwitch::ScopedOverride::ScopedOverride(FeatureSwitch* feature,
                                               bool override_value)
-    : feature_(feature),
-      previous_value_(feature->GetOverrideValue()) {
-  feature_->SetOverrideValue(
-      override_value ? OVERRIDE_ENABLED : OVERRIDE_DISABLED);
+    : feature_(feature), previous_value_(feature->GetOverrideValue()) {
+  feature_->SetOverrideValue(override_value ? OVERRIDE_ENABLED
+                                            : OVERRIDE_DISABLED);
 }
 
 FeatureSwitch::ScopedOverride::~ScopedOverride() {
-- 
2.26.2

