From 3dcea20792dbbd835ad85998ea2c301ae5e8ced1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb.prv@gmx.de>
Date: Tue, 11 Dec 2018 18:45:20 +0100
Subject: [PATCH 7/9] Use QStandardPaths to locate resources

---
 src/frameworks/UBPlatformUtils_linux.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/frameworks/UBPlatformUtils_linux.cpp b/src/frameworks/UBPlatformUtils_linux.cpp
index de5bdb9f..52c6e30c 100644
--- a/src/frameworks/UBPlatformUtils_linux.cpp
+++ b/src/frameworks/UBPlatformUtils_linux.cpp
@@ -31,6 +31,7 @@
 
 #include <QtGui>
 #include <QApplication>
+#include <QStandardPaths>
 
 #include <unistd.h>
 #include <X11/keysym.h>
@@ -45,7 +46,13 @@ void UBPlatformUtils::init()
 
 QString UBPlatformUtils::applicationResourcesDirectory()
 {
-    return QApplication::applicationDirPath();
+    QString appDataPath = QStandardPaths::locate(QStandardPaths::GenericDataLocation,
+                                                 QApplication::applicationName() + "/library/applications/",
+                                                 QStandardPaths::LocateDirectory);
+    if (appDataPath.isEmpty())
+        return QApplication::applicationDirPath();
+    else
+        return appDataPath + "../../";
 }
 
 void UBPlatformUtils::hideFile(const QString &filePath)
-- 
2.19.2

