From e4412e70335c7373c653c98262e58dc579f0cdda Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb.prv@gmx.de>
Date: Wed, 12 Dec 2018 11:02:11 +0100
Subject: [PATCH 8/8] Flatpak: Workaround: Don't check for webkit as Qt module

---
 OpenBoard.pro | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/OpenBoard.pro b/OpenBoard.pro
index 48a30230..e314c707 100644
--- a/OpenBoard.pro
+++ b/OpenBoard.pro
@@ -28,7 +28,7 @@ VERSION_RC = $$replace(VERSION_RC, "b", "176") # 0xB0
 VERSION_RC = $$replace(VERSION_RC, "rc", "192" ) # 0xC0
 VERSION_RC = $$replace(VERSION_RC, "r", "240") # 0xF0
 
-QT += webkit
+#QT += webkit
 QT += svg
 QT += network
 QT += xml
@@ -36,7 +36,7 @@ QT += script
 QT += xmlpatterns
 QT += uitools
 QT += multimedia
-QT += webkitwidgets
+#QT += webkitwidgets
 QT += multimediawidgets
 QT += printsupport
 QT += core
-- 
2.19.2

