From b4760f68fee92e5edc4c717a47aa8724640bed70 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Wed, 26 Aug 2020 20:23:56 -0500
Subject: [PATCH 08/18] Add support for Flatpak's Widevine location

---
 chrome/common/chrome_paths.cc | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/chrome/common/chrome_paths.cc b/chrome/common/chrome_paths.cc
index 8f26e588b4e63..8aad4f39dcf27 100644
--- a/chrome/common/chrome_paths.cc
+++ b/chrome/common/chrome_paths.cc
@@ -382,9 +382,12 @@ bool PathProvider(int key, base::FilePath* result) {
       break;
 
 #if (defined(OS_LINUX) || defined(OS_CHROMEOS)) && \
-    BUILDFLAG(BUNDLE_WIDEVINE_CDM)
+    BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
     case chrome::DIR_BUNDLED_WIDEVINE_CDM:
-      if (!GetComponentDirectory(&cur))
+      if (sandbox::FlatpakSandbox::GetInstance()->GetSandboxLevel() >
+          sandbox::FlatpakSandbox::SandboxLevel::kNone)
+        cur = base::FilePath(FILE_PATH_LITERAL("/app/widevine"));
+      else if (!GetComponentDirectory(&cur))
         return false;
 #if !defined(OS_CHROMEOS)
       // TODO(crbug.com/971433): Move Widevine CDM to a separate folder on
@@ -393,7 +396,7 @@ bool PathProvider(int key, base::FilePath* result) {
 #endif  // !defined(OS_CHROMEOS)
       break;
 #endif  // (defined(OS_LINUX) || defined(OS_CHROMEOS)) &&
-        // BUILDFLAG(BUNDLE_WIDEVINE_CDM)
+        // BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS) && \
     BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
-- 
2.26.2

