diff --git a/friza/gui/ui.cxx b/friza/gui/ui.cxx
index 0768fb7..9e5763f 100644
--- a/friza/gui/ui.cxx
+++ b/friza/gui/ui.cxx
@@ -46,7 +46,7 @@ typedef struct {
 	LV2UI_Controller controller;
 } GUI;
 
-LV2UI_Handle friza_instantiate(const struct _LV2UI_Descriptor * descriptor,
+LV2UI_Handle friza_instantiate(const struct LV2UI_Descriptor * descriptor,
                                const char * plugin_uri,
                                const char * bundle_path,
                                LV2UI_Write_Function write_function,
diff --git a/friza/gui/ui.hxx b/friza/gui/ui.hxx
index 9db0f5a..6c15dcc 100644
--- a/friza/gui/ui.hxx
+++ b/friza/gui/ui.hxx
@@ -18,7 +18,7 @@
  * MA 02110-1301, USA.
  */
 
-extern LV2UI_Handle friza_instantiate(const struct _LV2UI_Descriptor * descriptor,
+extern LV2UI_Handle friza_instantiate(const struct LV2UI_Descriptor * descriptor,
                                       const char * plugin_uri,
                                       const char * bundle_path,
                                       LV2UI_Write_Function write_function,
diff --git a/src/avtk/lv2/testUi.cxx b/src/avtk/lv2/testUi.cxx
index 0917856..efd35e7 100644
--- a/src/avtk/lv2/testUi.cxx
+++ b/src/avtk/lv2/testUi.cxx
@@ -10,7 +10,7 @@
 #include "../test_ui.hxx"
 
 
-static LV2UI_Handle avtk_instantiate(const struct _LV2UI_Descriptor * descriptor,
+static LV2UI_Handle avtk_instantiate(const struct LV2UI_Descriptor * descriptor,
                                      const char * plugin_uri,
                                      const char * bundle_path,
                                      LV2UI_Write_Function write_function,
diff --git a/src/ui/lv2_ui.cxx b/src/ui/lv2_ui.cxx
index 8d74b4c..d8c8869 100644
--- a/src/ui/lv2_ui.cxx
+++ b/src/ui/lv2_ui.cxx
@@ -33,7 +33,7 @@
 #include "whaaa.hxx"
 
 
-static LV2UI_Handle artyfx_instantiate(const struct _LV2UI_Descriptor * descriptor,
+static LV2UI_Handle artyfx_instantiate(const struct LV2UI_Descriptor * descriptor,
                                        const char * plugin_uri,
                                        const char * bundle_path,
                                        LV2UI_Write_Function write_function,
