diff --git a/lib/smmorphplan.cc b/lib/smmorphplan.cc
index df39ee20..6a88aa66 100644
--- a/lib/smmorphplan.cc
+++ b/lib/smmorphplan.cc
@@ -247,7 +247,7 @@ MorphPlan::load_internal (GenericIn *in, ExtraParameters *params)
 
                   delete blob_in; // close blob file handle
 
-                  GenericIn *in = MMapIn::open_mem (&blob_data[0], &blob_data[blob_data.size()]);
+                  GenericIn *in = MMapIn::open_mem (&blob_data[0], &blob_data[0] + blob_data.size());
                   InFile blob_infile (in);
                   load_op->load (blob_infile);
 
@@ -271,7 +271,7 @@ MorphPlan::load_internal (GenericIn *in, ExtraParameters *params)
 
                   vector<unsigned char>& blob_data = blob_data_map[ifile.event_blob_sum()];
 
-                  GenericIn *in = MMapIn::open_mem (&blob_data[0], &blob_data[blob_data.size()]);
+                  GenericIn *in = MMapIn::open_mem (&blob_data[0], &blob_data[0] + blob_data.size());
                   InFile blob_infile (in);
                   load_op->load (blob_infile);
 
