diff --git a/src/ui/lv2_ui.cxx b/src/ui/lv2_ui.cxx
index d8c8869..907f513 100644
--- a/src/ui/lv2_ui.cxx
+++ b/src/ui/lv2_ui.cxx
@@ -33,7 +33,7 @@
 #include "whaaa.hxx"
 
 
-static LV2UI_Handle artyfx_instantiate(const struct LV2UI_Descriptor * descriptor,
+static LV2UI_Handle artyfx_instantiate(const LV2UI_Descriptor * descriptor,
                                        const char * plugin_uri,
                                        const char * bundle_path,
                                        LV2UI_Write_Function write_function,
