--- artyfx-5/CMakeLists.txt	2016-12-24 14:36:45.000000000 -0500
+++ artyfx-5/CMakeLists.txt-new	2020-06-20 23:39:39.943932728 -0400
@@ -82,7 +82,7 @@
   add_dependencies( artyfx_ui avtk )
   target_link_libraries( artyfx_ui "-lavtk -lcairo -lX11" )
   set_target_properties( artyfx_ui PROPERTIES PREFIX "")
-  install (TARGETS artyfx_ui DESTINATION lib/lv2/artyfx.lv2/)
+  install (TARGETS artyfx_ui DESTINATION lv2/artyfx.lv2/)
 ENDIF()
 
 IF(BUILD_BENCH)
@@ -94,10 +94,10 @@
 set_target_properties(artyfx PROPERTIES PREFIX "")
 
 # add the install targets
-install (TARGETS artyfx    DESTINATION lib/lv2/artyfx.lv2/)
+install (TARGETS artyfx    DESTINATION lv2/artyfx.lv2/)
 
 FILE(GLOB modF "artyfx.lv2/modgui/*")
 FILE(GLOB ttls "artyfx.lv2/*.ttl"   )
 
-install(FILES     ${modF} DESTINATION lib/lv2/artyfx.lv2/modgui/)
-install(FILES     ${ttls} DESTINATION lib/lv2/artyfx.lv2/)
+install(FILES     ${modF} DESTINATION lv2/artyfx.lv2/modgui/)
+install(FILES     ${ttls} DESTINATION lv2/artyfx.lv2/)
