Author: Julien LEMOINE <speedblue@debian.org>
Bug-Debian: http://bugs.debian.org/166378
Description: Fixes divide-by-zero on alphas.
--- a/src/kugeltabellen.cpp
+++ b/src/kugeltabellen.cpp
@@ -7,6 +7,7 @@
 #ifndef M_PI
 #define M_PI 3.14159265358979323846
 #endif
+#define EPSILON 0.00001
 
 GLint calc_ind2(GLint b, GLint c, GLint Aufloesung) {
   return ((c*(2*Aufloesung+3-c))/2)+b;
@@ -28,9 +29,9 @@
       GLfloat temp_y=ay+((b+0.0)/(Aufloesung+0.0))*(by-ay)+((c+0.0)/(Aufloesung+0.0))*(cy-ay);
       GLfloat temp_z=az+((b+0.0)/(Aufloesung+0.0))*(bz-az)+((c+0.0)/(Aufloesung+0.0))*(cz-az);
       GLfloat Abstand=sqrt(temp_x*temp_x+temp_y*temp_y+temp_z*temp_z);
-      temp_x=temp_x/Abstand;
-      temp_y=temp_y/Abstand;
-      temp_z=temp_z/Abstand;
+      temp_x= (temp_x/Abstand == 0) ? EPSILON : temp_x/Abstand;
+      temp_y= (temp_y/Abstand == 0) ? EPSILON : temp_y/Abstand;
+      temp_z= (temp_z/Abstand == 0) ? EPSILON : temp_z/Abstand;
       vertices[iv++]=temp_x; 
       vertices[iv++]=temp_y; 
       vertices[iv++]=temp_z; 
