From dd92364be67a14170bd1ea3575ed1569745e8883 Mon Sep 17 00:00:00 2001
From: nik gaffney <nik@fo.am>
Date: Sun, 25 Sep 2022 22:30:39 +0200
Subject: [PATCH] added homebrew arm64 paths

---
 samplebrain.pro | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/samplebrain.pro b/samplebrain.pro
index 58ec2bc..f64772b 100644
--- a/samplebrain.pro
+++ b/samplebrain.pro
@@ -43,7 +43,8 @@ SOURCES += app/MainWindow.cpp \
 
 INCLUDEPATH += brain/src
 INCLUDEPATH += /usr/local/include
-LIBS += -L.. -L/usr/local/lib -lportaudio -lfftw3 -lsndfile -llo -ldl -lpthread -lm
+INCLUDEPATH += /opt/homebrew/include
+LIBS += -L.. -L/usr/local/lib -L/opt/homebrew/lib -lportaudio -lfftw3 -lsndfile -llo -ldl -lpthread -lm
 
 QMAKE_CXXFLAGS += -O3 -Wall -Wno-unused -std=c++11
 
@@ -57,5 +58,3 @@ unix:iconfile.files = desktop/samplebrain.svg
 
 target.path = /usr/bin
 INSTALLS += target desktopfile iconfile
-
-
-- 
GitLab

