Author: Thierry Reding <thierry@doppeltgemoppelt.de>
Description: By default, start in windowed mode.
Bug-Debian: http://bugs.debian.org/173197
Bug-Debian: http://bugs.debian.org/284855
--- a/src/KommandoZeilenParameter.cpp
+++ b/src/KommandoZeilenParameter.cpp
@@ -154,7 +154,7 @@
   TischLampen          = 2;
   GrueneLampe          = 0;
 
-  FullScreen           = 1;
+  FullScreen           = 0;
 
   EffektLautstaerke    = .5;
   MusikLautstaerke     = .5;
