From defee25db77a20383f0fab4638c6cd7ed1b77fdf Mon Sep 17 00:00:00 2001
From: nik gaffney <nik@fo.am>
Date: Mon, 26 Sep 2022 18:32:40 +0200
Subject: [PATCH] added ICON

---
 samplebrain.pro | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/samplebrain.pro b/samplebrain.pro
index 19d082a..c7df563 100644
--- a/samplebrain.pro
+++ b/samplebrain.pro
@@ -49,7 +49,8 @@ LIBS += -L.. -L/usr/local/lib -L/opt/homebrew/lib -lportaudio -lfftw3 -lsndfile
 QMAKE_CXXFLAGS += -O3 -Wall -Wno-unused -std=c++11
 
 # assets
-RESOURCES     = app/samplebrain.qrc
+RESOURCES = app/samplebrain.qrc
+ICON      = desktop/samplebrain.icns
 
 PREFIX = $$(PREFIX)
 isEmpty(PREFIX) {
-- 
GitLab

