--- a/pm_test/sysex.c
+++ b/pm_test/sysex.c
@@ -171,7 +171,7 @@
 	    if (seconds == 0) seconds = 1;
             printf("Correctly received %d byte sysex message.\n", i);
 	    total_bytes += i;
-	    printf("Cummulative bytes/sec: %d\n", total_bytes / seconds);
+	    printf("Cummulative bytes/sec: %ld\n", total_bytes / seconds);
         }
     }
 cleanup:
@@ -231,6 +231,7 @@
         Pm_WriteSysEx(midi_out, 0, msg);
     }
     stop_time = Pt_Time();
+    printf("Delay was: %x\n", start_time - stop_time);
     Pm_Close(midi_out);
     return;
 }
