We don't actually need iomanip.h
--- a/configure.in
+++ b/configure.in
@@ -227,7 +227,6 @@
 )
 AC_CHECK_HEADERS(               \
   string                        \
-  iomanip.h                     \
   ,,AC_MSG_ERROR([Missing a vital header file for id3lib])
 )
 
--- a/configure
+++ b/configure
@@ -10296,7 +10296,6 @@
 
 for ac_header in \
   string                        \
-  iomanip.h                     \
 
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
