diff --git a/pm_linux/pmlinuxalsa.c b/pm_linux/pmlinuxalsa.c
index 8e85cfe..1ec3e56 100755
--- a/pm_linux/pmlinuxalsa.c
+++ b/pm_linux/pmlinuxalsa.c
@@ -494,7 +494,7 @@ static void handle_event(snd_seq_event_t *ev)
     PmTimestamp timestamp;
 
     /* time stamp should be in ticks, using our queue where 1 tick = 1ms */
-    assert((ev->flags & SND_SEQ_TIME_STAMP_MASK) == SND_SEQ_TIME_STAMP_TICK);
+    if ((ev->flags & SND_SEQ_TIME_STAMP_MASK) != SND_SEQ_TIME_STAMP_TICK) return;
 
     /* if no time_proc, just return "native" ticks (ms) */
     if (time_proc == NULL) {
