diff --git a/BUILD.gn b/BUILD.gn
index 9cd65d2f..f780310e 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -20,7 +20,7 @@ config("absl_component_build") {
 assert(!is_nacl || is_nacl_saigo,
        "base must not be built in most nacl toolchains")
 
-component("absl") {
+static_library("absl") {
   public_deps = [ ":absl_component_deps" ]
   if (is_component_build) {
     public_configs = [ ":absl_component_build" ]
