diff --git a/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma b/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
index 95f46ed9a..66a28119a 100755
--- a/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
+++ b/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
@@ -1,2 +1,11 @@
 #!/bin/bash
-java -Djdk.tls.maxHandshakeMessageSize=65536 -jar /usr/lib/Autofirma/autofirma.jar "$@"
+_autofirma_dir="${HOME}/.afirma/Autofirma"
+
+if [ ! -r "${_autofirma_dir}/Autofirma_ROOT.cer" ] || [ ! -r "${_autofirma_dir}/autofirma.pfx" ]; then
+  mkdir -p "${_autofirma_dir}"
+  /app/lib/postinst || echo "WARNING: Certificate initialization failed, the application may not work correctly."
+fi
+
+unset _autofirma_dir
+
+java -Djdk.tls.maxHandshakeMessageSize=65536 -jar /app/lib/autofirma.jar "$@"
