diff --git a/afirma-simple-installer/linux/instalador_deb/src/DEBIAN/postinst b/afirma-simple-installer/linux/instalador_deb/src/DEBIAN/postinst
index f8a34f0f8..88a445a99 100755
--- a/afirma-simple-installer/linux/instalador_deb/src/DEBIAN/postinst
+++ b/afirma-simple-installer/linux/instalador_deb/src/DEBIAN/postinst
@@ -1,53 +1,22 @@
 #!/bin/sh
 # generamos certificado CA y SSL
-java -jar /usr/lib/Autofirma/autofirmaConfigurador.jar
+java -jar /app/lib/autofirmaConfigurador.jar -jnlp -headless
 # damos permiso al script de instalacion del certificado para firefox y lo ejecutamos
 
 echo "Generacion de certificados"
 
-if [ -f "/usr/lib/Autofirma/script.sh" ]; then
-chmod +x /usr/lib/Autofirma/script.sh
-/usr/lib/Autofirma/script.sh
+_autofirma_dir="${HOME}/.afirma/Autofirma"
 
+# Introduced because Flatpak's do not have uninstall scripts, so... just in case
+if [ -f "${_autofirma_dir}/uninstall.sh" ]; then
+. "${_autofirma_dir}/uninstall.sh" 2>/dev/null || true
 fi
 
-echo "Instalacion del certificado CA en el almacenamiento de Firefox y Chrome"
-
-#Creacion de certificado .pem y .crt para su importacion en el almacenamiento del sistema
-openssl x509 -inform der -in /usr/lib/Autofirma/Autofirma_ROOT.cer -out /usr/lib/Autofirma/Autofirma_ROOT.pem
-
-mv /usr/lib/Autofirma/Autofirma_ROOT.pem /usr/lib/Autofirma/Autofirma_ROOT.crt
-
-if [ -d "/usr/share/ca-certificates/Autofirma" ]; then
-rm -R /usr/share/ca-certificates/Autofirma/
-fi
-
-mkdir /usr/share/ca-certificates/Autofirma/
-cp /usr/lib/Autofirma/Autofirma_ROOT.crt /usr/share/ca-certificates/Autofirma/Autofirma_ROOT.crt
-cp /usr/lib/Autofirma/Autofirma_ROOT.crt /usr/local/share/ca-certificates/Autofirma_ROOT.crt
-update-ca-certificates
-echo "Instalacion del certificado CA en el almacenamiento del sistema"
-
-# Enlazamos el fichero de asociacion del protocolo "afirma" en Firefox
-# para que aplique a otros navegadores
-if [ -d /etc/icecat ] ; then
-   if [ ! -d /etc/icecat/pref ] ; then
-      mkdir /etc/icecat/pref
-   fi
-   ln -s /etc/firefox/pref/Autofirma.js /etc/icecat/pref/Autofirma.js
-fi
-if [ -d /etc/firefox-esr ] ; then
-   if [ ! -d /etc/firefox-esr/pref ] ; then
-      mkdir /etc/firefox-esr/pref
-   fi
-   ln -s /etc/firefox/pref/Autofirma.js /etc/firefox-esr/pref/Autofirma.js
+if [ -f "${_autofirma_dir}/script.sh" ]; then
+chmod +x "${_autofirma_dir}/script.sh"
+. "${_autofirma_dir}/script.sh"
 fi
 
-# Definimos la aplicacion autofirma como la de por defecto para abrir los x/handler (no es necesario porque se define en el fichero de control)
-#xdg-mime default /usr/share/applications/afirma.desktop x-scheme-handler/afirma
+echo "Instalacion del certificado CA en el almacenamiento de Firefox y Chrome"
 
-# borramos los ficheros de instalacion
-rm /usr/lib/Autofirma/script.sh
-rm /usr/lib/Autofirma/Autofirma_ROOT.crt
 exit 0
-
