diff --git a/config.pri b/config.pri
index 6f46619c..c453a39b 100644
--- a/config.pri
+++ b/config.pri
@@ -31,7 +31,7 @@ unix {                                                       # Linux / MacOS X
     plugins.path          = "$$INSTALLBASE/plugins"
   } else {
     ### this is what is called "prefix" when using GNU autotools
-    isEmpty(INSTALLBASE): INSTALLBASE = /usr
+    isEmpty(INSTALLBASE): INSTALLBASE = /app
     target.path           = "$$INSTALLBASE/bin"
     documentation.path    = "$$INSTALLBASE/share/doc/alphaplot"
     manual.path = "$$INSTALLBASE/share/doc/alphaplot/manual"
@@ -80,8 +80,8 @@ contains(PRESET, linux_all_dynamic) {
   ### dynamically link against installed system-wide
   message(Build configuration: Linux all dynamic)
 
-  INCLUDEPATH   = "$(HOME)/usr/include" $$INCLUDEPATH
-  QMAKE_LIBDIR  = "$(HOME)/usr/lib" $$QMAKE_LIBDIR
+  INCLUDEPATH   = "/app/include" $$INCLUDEPATH
+  QMAKE_LIBDIR  = "/app/lib" $$QMAKE_LIBDIR
 
   LIBS         += -lGLU -lgsl -lgslcblas
 }
diff --git a/fitPlugins/exp_saturation/exp_saturation.pro b/fitPlugins/exp_saturation/exp_saturation.pro
index 37babcc5..c185c1aa 100644
--- a/fitPlugins/exp_saturation/exp_saturation.pro
+++ b/fitPlugins/exp_saturation/exp_saturation.pro
@@ -10,7 +10,7 @@ CONFIG += release
 DESTDIR = ../
 
 INSTALLS += target
-unix:  target.path = /usr/lib$${libsuff}/AlphaPlot/plugins
+unix:  target.path = /app/lib$${libsuff}/AlphaPlot/plugins
 win32: target.path = ../../output/plugins
 
 win32:INCLUDEPATH += ../../3rdparty/gsl-1.8/include
diff --git a/fitPlugins/explin/explin.pro b/fitPlugins/explin/explin.pro
index ef5b8062..338d3c64 100644
--- a/fitPlugins/explin/explin.pro
+++ b/fitPlugins/explin/explin.pro
@@ -10,7 +10,7 @@ CONFIG += release
 DESTDIR = ../
 
 INSTALLS += target
-unix:  target.path = /usr/lib$${libsuff}/AlphaPlot/plugins
+unix:  target.path = /app/lib$${libsuff}/AlphaPlot/plugins
 win32: target.path = ../../output/plugins
 
 win32:INCLUDEPATH += ../../3rdparty/gsl-1.8/include
diff --git a/fitPlugins/fitRational0/fitRational0.pro b/fitPlugins/fitRational0/fitRational0.pro
index 20778bc6..9b7e08c5 100644
--- a/fitPlugins/fitRational0/fitRational0.pro
+++ b/fitPlugins/fitRational0/fitRational0.pro
@@ -21,7 +21,7 @@ unix:LIBS += -L/usr/lib$${libsuff} -lgsl -lgslcblas
 #unix:LIBS         += /usr/lib/libgsl.a /usr/lib/libgslcblas.a
 
 # where to install the plugins
-unix:target.path=/usr/lib$${libsuff}/AlphaPlot/plugins
+unix:target.path=/app/lib$${libsuff}/AlphaPlot/plugins
 win32: target.path = ../../output/plugins
 
 SOURCES += fitRational0.cpp
diff --git a/fitPlugins/fitRational1/fitRational1.pro b/fitPlugins/fitRational1/fitRational1.pro
index a92e83fb..9865e1fd 100644
--- a/fitPlugins/fitRational1/fitRational1.pro
+++ b/fitPlugins/fitRational1/fitRational1.pro
@@ -21,7 +21,7 @@ unix:LIBS += -L/usr/lib$${libsuff} -lgsl -lgslcblas
 #unix:LIBS         += /usr/lib/libgsl.a /usr/lib/libgslcblas.a
 
 # where to install the plugins
-unix:target.path=/usr/lib$${libsuff}/AlphaPlot/plugins
+unix:target.path=/app/lib$${libsuff}/AlphaPlot/plugins
 win32: target.path = ../../output/plugins
 
 SOURCES += fitRational1.cpp
diff --git a/fitPlugins/planck_wavelength/planck_wavelength.pro b/fitPlugins/planck_wavelength/planck_wavelength.pro
index f26a048f..1aa7af3b 100644
--- a/fitPlugins/planck_wavelength/planck_wavelength.pro
+++ b/fitPlugins/planck_wavelength/planck_wavelength.pro
@@ -10,7 +10,7 @@ CONFIG += release
 DESTDIR = ../
 
 INSTALLS += target
-unix:  target.path = /usr/lib$${libsuff}/AlphaPlot/plugins
+unix:  target.path = /app/lib$${libsuff}/AlphaPlot/plugins
 win32: target.path = ../../output/plugins
 
 win32:INCLUDEPATH       += ../../3rdparty/gsl-1.8/include
