From 1ba3a77a6095fff45b73413ef9b4319c56921d48 Mon Sep 17 00:00:00 2001
From: Philippe Faist <phfaist@gmail.com>
Date: Wed, 6 Jan 2021 18:44:25 +0100
Subject: [PATCH] applied suggested patch, fixes #53

---
 src/klftools/klfflowlistwidget_p.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/klftools/klfflowlistwidget_p.h b/src/klftools/klfflowlistwidget_p.h
index fc14fce..ee3af0c 100644
--- a/src/klftools/klfflowlistwidget_p.h
+++ b/src/klftools/klfflowlistwidget_p.h
@@ -39,6 +39,7 @@
 #include <QHoverEvent>
 #include <QTimer>
 #include <QPixmap>
+#include <QPainterPath>
 
 #include <klfdefs.h>
 
-- 
2.51.0

