diff --git a/src/as-validator.c b/src/as-validator.c
index 849338c7..c4da24af 100644
--- a/src/as-validator.c
+++ b/src/as-validator.c
@@ -653,6 +653,15 @@ as_validator_add_override (AsValidator *validator,
 		"developer-id-missing",
 		/* allow in case a component really doesn't have an active homepage */
 		"url-homepage-missing",
+		/* flathub apps often instruct users to follow some additional steps for sandbox-breaking apps */
+		"description-has-plaintext-url",
+		/* we have our own validation for these fields */
+		"component-name-too-long",
+		"summary-too-long",
+		/* this is a breaking change as 1.0.0 allowed this field to be anything */
+		"developer-id-invalid",
+		/* this has not been the case with appstream-glib */
+		"cid-domain-not-lowercase",
 		NULL
 	};
 
