--- ardour/gtk2_ardour/ardour.appdata.xml.in.in	2019-02-07 06:42:07.636417275 -0500
+++ ardour/gtk2_ardour/ardour.appdata.xml.in.in.new	2019-02-07 06:43:21.635902231 -0500
@@ -18,6 +18,8 @@
       composers.
     </p>
   </description>
+  <content_rating type="oars-1.1" />
+  <launchable type="desktop-id">org.ardour.Ardour.desktop</launchable>
   <url type="homepage">http://ardour.org</url>
   <screenshots>
       <screenshot type="default">
