diff -up Ardour-5.12.0/libs/ardour/revision.cc.appdata-revision Ardour-5.12.0/libs/ardour/revision.cc
--- Ardour-5.12.0/libs/ardour/revision.cc.appdata-revision	2017-09-15 17:10:17.000000000 -0400
+++ Ardour-5.12.0/libs/ardour/revision.cc	2017-09-19 14:56:36.928102435 -0400
@@ -1,2 +1,2 @@
 #include "ardour/revision.h"
-namespace ARDOUR { const char* revision = "5.12"; }
+namespace ARDOUR { const char* revision = "5.12"; const char* date = "2017-09-14"; }
