--- argyllcms-2/gamut/gamut.h	2020-01-18 20:38:19.000000000 -0500
+++ argyllcms-2/gamut/gamut.h-new	2020-12-18 10:58:28.379440542 -0500
@@ -36,7 +36,7 @@
 #define MAXGAMN 10				/* Maximum gamut point neighbors returned */
 #define NSLOTS 6				/* Number of maximum direction slots */
 
-struct _vrml *wrl;				/* Declared in vrml.h, which may be #included after this */
+extern struct _vrml *wrl;				/* Declared in vrml.h, which may be #included after this */
 
 /* ------------------------------------ */
 #define NODE_STRUCT							\
--- argyllcms/spectro/vinflate.c	2020-12-18 11:03:02.613660622 -0500
+++ argyllcms/spectro/vinflate.c-new	2020-12-18 11:04:23.824206479 -0500
@@ -92,7 +92,7 @@
 */
 
 #define WSIZE 0x8000
-unsigned int wp;             /* current position in slide */
+static unsigned int wp;             /* current position in slide */
 uch slide[32768];
 
 static int vflush_output(unsigned int w) {
@@ -160,8 +160,8 @@
    the stream.
  */
 
-ulg bb;                         /* bit buffer */
-unsigned bk;                    /* bits in bit buffer */
+static ulg bb;                         /* bit buffer */
+static unsigned bk;                    /* bits in bit buffer */
 
 ush vmask_bits[] = {
     0x0000,
@@ -239,7 +239,7 @@
 #define N_MAX 288       /* maximum number of codes in any set */
 
 
-unsigned hufts;         /* track memory usage */
+static unsigned hufts;         /* track memory usage */
 
 /* Given a list of code lengths and a maximum table size, make a set of
    tables to decode that set of codes.  Return zero on success, one if
--- argyllcms/spectro/inflate.c	2020-01-18 20:38:27.000000000 -0500
+++ argyllcms/spectro/inflate.c-new	2020-12-18 11:04:55.992818816 -0500
@@ -87,7 +87,7 @@
 /* unsigned wp;             current position in slide */
 
 #define WSIZE 0x8000
-unsigned int wp;             /* current position in slide */
+static unsigned int wp;             /* current position in slide */
 uch slide[32768];
 
 static int flush_output(unsigned int w) {
@@ -155,8 +155,8 @@
    the stream.
  */
 
-ulg bb;                         /* bit buffer */
-unsigned bk;                    /* bits in bit buffer */
+static ulg bb;                         /* bit buffer */
+static unsigned bk;                    /* bits in bit buffer */
 
 ush mask_bits[] = {
     0x0000,
@@ -210,7 +210,7 @@
 #define N_MAX 288       /* maximum number of codes in any set */
 
 
-unsigned hufts;         /* track memory usage */
+static unsigned hufts;         /* track memory usage */
 
 
 static int huft_build(b, n, s, d, e, t, m)
--- Argyll_V2.1.2/Jamtop	2020-01-18 20:38:05.000000000 -0500
+++ Argyll_V2.1.2/Jamtop-new	2020-12-18 22:23:42.606220829 -0500
@@ -110,8 +110,8 @@
 		} else {
 			if [ GLOB /usr/X11R6/lib : libX11.so libX11.a ] {
 				LibWinD = /usr/X11R6/lib ;
-			} else if [ GLOB /usr/lib/i386-linux-gnu : libX11.so libX11.a ] {
-				LibWinD = /usr/lib/i386-linux-gnu ;
+			} else if [ GLOB /usr/lib/aarch64-linux-gnu : libX11.so libX11.a ] {
+				LibWinD = /usr/lib/aarch64-linux-gnu ;
 			} else if [ GLOB /usr/lib : libX11.so libX11.a ] {
 				LibWinD = /usr/lib ;
 			} else if [ GLOB /usr/local/lib : libX11.so libX11.a ] {
@@ -156,14 +156,14 @@
 		if  [ GLOB /usr/include$(subd) : $(lcase).h $(lcase)lib.h ]
 	     || [ GLOB /usr/local/include$(subd) : $(lcase).h $(lcase)lib.h ]
 		 || [ GLOB /usr/include/x86_64-linux-gnu$(subd) : $(lcase).h $(lcase)lib.h ]
-	     || [ GLOB /usr/include/i386-linux-gnu$(subd) : $(lcase).h $(lcase)lib.h ] {
+	     || [ GLOB /usr/include/aarch64-linux-gnu$(subd) : $(lcase).h $(lcase)lib.h ] {
 			if  [ GLOB /usr/lib : lib$(lcase).so ] || [ GLOB /usr/lib : lib$(lcase).a ]
 			 || [ GLOB /usr/local/lib : lib$(lcase).so ] || [ GLOB /usr/local/lib : lib$(lcase).a ]
 			 || [ GLOB /usr/lib64 : lib$(lcase).so ] || [ GLOB /usr/lib64 : lib$(lcase).a ]
 			 || [ GLOB /usr/lib/x86_64-linux-gnu : lib$(lcase).so ]
 			 || [ GLOB /usr/lib/x86_64-linux-gnu : lib$(lcase).a ]
-	         || [ GLOB /usr/lib/i386-linux-gnu : lib$(lcase).so ]
-	         || [ GLOB /usr/lib/i386-linux-gnu : lib$(lcase).a ] {
+	         || [ GLOB /usr/lib/aarch64-linux-gnu : lib$(lcase).so ]
+	         || [ GLOB /usr/lib/aarch64-linux-gnu : lib$(lcase).a ] {
 				echo "Using system $(UCASE) library" ;
 				$(UCASE)LIB = ;
 				$(UCASE)INC = ;
