From e5bf215274640cf9b0f6c5564aaf27fdc7771e1a Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Sun, 7 May 2023 00:46:34 +0200
Subject: [PATCH] Add release information to metainfo file

---
 data/meson.build | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 9163799..005793c 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -3,16 +3,24 @@
 install_data('asgen-hints.json', install_dir: 'share/appstream')
 install_data('hicolor-theme-index.theme', install_dir: 'share/appstream')
 
-metainfo_file = files('org.freedesktop.appstream.generator.metainfo.xml')
-install_data(metainfo_file, install_dir: join_paths (get_option ('datadir'), 'metainfo'))
-appstreamcli_exe = find_program('appstreamcli', required: false)
-if appstreamcli_exe.found()
+ascli_exe = find_program('appstreamcli', required: true)
+
+metainfo_filename = 'org.freedesktop.appstream.generator.metainfo.xml'
+metainfo_with_relinfo = custom_target('add-metainfo-releases',
+    input : ['../NEWS', files(metainfo_filename)],
+    output : [metainfo_filename],
+    command : [ascli_exe, 'news-to-metainfo', '--limit=6', '@INPUT0@', '@INPUT1@', '@OUTPUT@'],
+    install : true,
+    install_dir : join_paths (get_option ('datadir'), 'metainfo')
+)
+
+if ascli_exe.found()
   # Validate our MetaInfo file
   test('asgen-validate_metainfo',
-    appstreamcli_exe,
+    ascli_exe,
     args: ['validate',
            '--no-net', '--pedantic',
-           metainfo_file]
+           files(metainfo_filename)]
   )
 endif
 
