diff --git a/modules/juce_audio_processors/format_types/juce_LV2PluginFormat.cpp b/modules/juce_audio_processors/format_types/juce_LV2PluginFormat.cpp
index b4f35881..a3e73c36 100644
--- a/modules/juce_audio_processors/format_types/juce_LV2PluginFormat.cpp
+++ b/modules/juce_audio_processors/format_types/juce_LV2PluginFormat.cpp
@@ -5269,7 +5269,7 @@ public:
         return { "%APPDATA%\\LV2;"
                  "%COMMONPROGRAMFILES%\\LV2" };
       #else
-       #if JUCE_64BIT
+       #if 0 //JUCE_64BIT
         if (File ("/usr/lib64/lv2").exists() || File ("/usr/local/lib64/lv2").exists())
             return { "~/.lv2;"
                      "/usr/lib64/lv2;"
@@ -5277,8 +5277,8 @@ public:
        #endif
 
         return { "~/.lv2;"
-                 "/usr/lib/lv2;"
-                 "/usr/local/lib/lv2" };
+                 "/app/lib/lv2;"
+                 "/app/extensions/Plugins/lv2" };
       #endif
     }
 
