diff --git a/Source/ModularSynth.cpp b/Source/ModularSynth.cpp
index 7cdbb98..f281ac5 100644
--- a/Source/ModularSynth.cpp
+++ b/Source/ModularSynth.cpp
@@ -2103,7 +2103,7 @@ float* ModularSynth::GetInputBuffer(int channel)
 
 float* ModularSynth::GetOutputBuffer(int channel)
 {
-   assert(channel >= 0 && channel < mOutputBuffers.size());
+   assert(channel >= 0 && !mOutputBuffers.empty() && channel < mOutputBuffers.size());
    return mOutputBuffers[channel];
 }
 
diff --git a/Source/VSTPlayhead.cpp b/Source/VSTPlayhead.cpp
index c022da5..cffe539 100644
--- a/Source/VSTPlayhead.cpp
+++ b/Source/VSTPlayhead.cpp
@@ -34,7 +34,13 @@ juce::Optional<juce::AudioPlayHead::PositionInfo> VSTPlayhead::getPosition() con
    timeSignature.numerator = TheTransport->GetTimeSigTop();
    timeSignature.denominator = TheTransport->GetTimeSigBottom();
    loopPoint.ppqStart = 0;
-   loopPoint.ppqEnd = 480 * pos.getTimeSignature()->denominator;
+
+   juce::Optional<juce::AudioPlayHead::TimeSignature> posTM = pos.getTimeSignature();
+
+   loopPoint.ppqEnd = 480;
+   if (posTM) {
+     loopPoint.ppqEnd *= pos.getTimeSignature()->denominator;
+   }
 
    pos.setBpm(TheTransport->GetTempo());
    pos.setTimeSignature(timeSignature);
