diff --git a/Source/UserPrefs.h b/Source/UserPrefs.h
index 6f838b1..83b515e 100644
--- a/Source/UserPrefs.h
+++ b/Source/UserPrefs.h
@@ -204,8 +204,8 @@ namespace
    const char* kDefaultYoutubeDlPath = "/opt/local/bin/youtube-dl";
    const char* kDefaultFfmpegPath = "/opt/local/bin/ffmpeg";
 #elif BESPOKE_LINUX
-   const char* kDefaultYoutubeDlPath = "/usr/bin/youtube-dl";
-   const char* kDefaultFfmpegPath = "/usr/bin/ffmpeg";
+   const char* kDefaultYoutubeDlPath = "/app/bin/youtube-dl";
+   const char* kDefaultFfmpegPath = "/app/bin/ffmpeg";
 #else
    const char* kDefaultYoutubeDlPath = "c:/youtube-dl/bin/youtube-dl.exe";
    const char* kDefaultFfmpegPath = "c:/ffmpeg/bin/ffmpeg.exe";
