
diff -uNrw -X .diff_ignore --no-dereference xtrkcad-fork-xtrkcad/app/bin/CMakeLists.txt new-fork-xtrkcad/app/bin/CMakeLists.txt
--- xtrkcad-fork-xtrkcad/app/bin/CMakeLists.txt	2025-12-20 09:20:00.551188215 -0500
+++ new-fork-xtrkcad/app/bin/CMakeLists.txt	2025-12-22 09:37:51.891897792 -0500
@@ -418,6 +418,17 @@
 	RUNTIME DESTINATION ${XTRKCAD_BIN_INSTALL_DIR}
 )
 
+install(
+	FILES ${CMAKE_SOURCE_DIR}/distribution/flatpak/common_source.sh
+	DESTINATION ${XTRKCAD_BIN_INSTALL_DIR}
+)
+
+install(
+    FILES ${CMAKE_SOURCE_DIR}/distribution/flatpak/run-xtrkcad.sh
+    DESTINATION ${XTRKCAD_BIN_INSTALL_DIR}
+    PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_EXECUTE GROUP_READ WORLD_EXECUTE WORLD_READ
+)
+
 if(XTRKCAD_TESTING AND CMOCKA_FOUND)
 	add_subdirectory( unittest )
 endif()
