diff --git a/libc/libc.c b/libc/libc.c
index 6a62dbe..6750ca9 100644
--- a/libc/libc.c
+++ b/libc/libc.c
@@ -20,6 +20,9 @@
 #include <sys/types.h>
 #include <sys/user.h> // PAGE_SIZE, PAGE_SHIFT
 
+#ifndef PAGE_SIZE
+#define PAGE_SIZE PAGESIZE
+#endif
 
 struct bionic_dirent {
 	uint64_t d_ino;
diff --git a/linker/linker.c b/linker/linker.c
index f180efb..0491b20 100644
--- a/linker/linker.c
+++ b/linker/linker.c
@@ -95,6 +95,15 @@
 #define ELF_ST_TYPE ELF32_ST_TYPE
 #endif
 
+#if defined(__aarch64__)
+#ifndef R_AARCH64_TLS_TPREL64
+#define R_AARCH64_TLS_TPREL64 R_AARCH64_TLS_TPREL
+#endif
+#ifndef R_AARCH64_TLS_DTPREL32
+#define R_AARCH64_TLS_DTPREL32 R_AARCH64_TLS_DTPREL
+#endif
+#endif
+
 /* >>> IMPORTANT NOTE - READ ME BEFORE MODIFYING <<<
  *
  * Do NOT use malloc() and friends or pthread_*() code here.
