diff --git a/pthread_wrapper/libpthread.c b/pthread_wrapper/libpthread.c
index a1eccb6..295ad69 100644
--- a/pthread_wrapper/libpthread.c
+++ b/pthread_wrapper/libpthread.c
@@ -8,6 +8,10 @@
 #include <sys/mman.h>
 #include <setjmp.h>
 
+extern void __pthread_unwind_next(__pthread_unwind_buf_t *__buf) __cleanup_fct_attribute __attribute ((__noreturn__));
+extern void __pthread_register_cancel(__pthread_unwind_buf_t *__buf) __cleanup_fct_attribute;
+extern void __pthread_unregister_cancel (__pthread_unwind_buf_t *__buf) __cleanup_fct_attribute;
+
 // when __GLIBC__ (or some glibc specific symbol) is not defined, we're assuming musl; can't check to be sure because 🤡
 
 #ifndef PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
