
diff -uNrb -X .diff_ignore --no-dereference xtrkcad-fork-xtrkcad/app/bin/bitmaps/CMakeLists.txt new-fork-xtrkcad/app/bin/bitmaps/CMakeLists.txt
--- xtrkcad-fork-xtrkcad/app/bin/bitmaps/CMakeLists.txt	2025-12-20 09:20:00.554521567 -0500
+++ new-fork-xtrkcad/app/bin/bitmaps/CMakeLists.txt	2025-12-22 13:30:43.784030621 -0500
@@ -115,7 +115,7 @@
 				OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/png/${img3}${pix}.png
 				DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/svg/${img3}.svg
 				COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_CURRENT_BINARY_DIR}/png/
-				COMMAND ${Inkscape_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/svg/${img3}.svg -h ${pix} ${Inkscape_EXPORT}${CMAKE_CURRENT_BINARY_DIR}/png/${img3}${pix}.png
+				COMMAND ${Inkscape_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/svg/${img3}.svg -h ${pix} -o ${CMAKE_CURRENT_BINARY_DIR}/png/${img3}${pix}.png
 			)
 		endif()
 		if(UNIX)
