diff -ur CalculiX/ccx_2.22/src/Makefile_MT CalculiX_new/ccx_2.22/src/Makefile_MT
--- CalculiX/ccx_2.22/src/Makefile_MT	2020-07-23 21:41:38.507761972 +0200
+++ CalculiX_new/ccx_2.22/src/Makefile_MT	2020-10-12 04:00:16.010748092 +0200
@@ -1,6 +1,6 @@
 
-CFLAGS = -Wall -O2 -fopenmp -I ../SPOOLES.2.2 -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE -DUSE_MT=1 
-FFLAGS = -Wall -O2 -fopenmp 
+CFLAGS = -Wall -O2 -fopenmp -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE -DUSE_MT=1
+FFLAGS = -Wall -O2 -fopenmp -fallow-argument-mismatch
 
 CC=cc
 FC=gfortran
@@ -18,12 +18,11 @@ OCCXF = $(SCCXF:.f=.o)
 OCCXC = $(SCCXC:.c=.o)
 OCCXMAIN = $(SCCXMAIN:.c=.o)
 
-DIR=../SPOOLES.2.2
-
 LIBS = \
-       $(DIR)/MT/src/spoolesMT.a \
-       $(DIR)/spooles.a \
-       ../ARPACK/libarpack_INTEL.a \
+       /app/lib/spoolesMT.a \
+       /app/lib/spooles.a \
+       /app/lib/libarpack.a \
+       /app/lib/libopenblas.a \
        -lpthread -lm
 
 ccx_2.22_MT: $(OCCXMAIN) ccx_2.22_MT.a  $(LIBS)
