--- /home/adardano3//Developer/zed-industries/zed/Cargo.lock	2024-05-18 01:56:22.155262204 -0400
+++ /dev/fd/63	2024-05-18 01:56:55.915460802 -0400
@@ -638,7 +638,8 @@
 [[package]]
 name = "async-pipe"
 version = "0.1.3"
-source = "git+https://github.com/zed-industries/async-pipe-rs?rev=82d00a04211cf4e1236029aa03e6b6ce2a74c553#82d00a04211cf4e1236029aa03e6b6ce2a74c553"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "3205e9ef663755ece59cd2e57ac2518d8dc7a2b7e86cc0aca04f641cc1566215"
 dependencies = [
  "futures 0.3.28",
  "log",
@@ -1490,7 +1491,8 @@
 [[package]]
 name = "blade-graphics"
 version = "0.4.0"
-source = "git+https://github.com/kvark/blade?rev=f5766863de9dcc092e90fdbbc5e0007a99e7f9bf#f5766863de9dcc092e90fdbbc5e0007a99e7f9bf"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "1e5d7faa9dd49dbe69759f2997f6dc181e6d0d5fb0c73626572a9fa366f58bc3"
 dependencies = [
  "ash",
  "ash-window",
@@ -1520,7 +1522,8 @@
 [[package]]
 name = "blade-macros"
 version = "0.2.1"
-source = "git+https://github.com/kvark/blade?rev=f5766863de9dcc092e90fdbbc5e0007a99e7f9bf#f5766863de9dcc092e90fdbbc5e0007a99e7f9bf"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "1e5d7faa9dd49dbe69759f2997f6dc181e6d0d5fb0c73626572a9fa366f58bc3"
 dependencies = [
  "proc-macro2",
  "quote",
@@ -3947,7 +3950,8 @@
 [[package]]
 name = "font-kit"
 version = "0.11.0"
-source = "git+https://github.com/zed-industries/font-kit?rev=5a5c4d4#5a5c4d4ca395c74eb0abde38508e170ce0fd761a"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "21fe28504d371085fae9ac7a3450f0b289ab71e07c8e57baa3fb68b9e57d6ce5"
 dependencies = [
  "bitflags 1.3.2",
  "byteorder",
@@ -4800,7 +4804,8 @@
 [[package]]
 name = "heed"
 version = "0.20.0-alpha.9"
-source = "git+https://github.com/meilisearch/heed?rev=036ac23f73a021894974b9adc815bc95b3e0482a#036ac23f73a021894974b9adc815bc95b3e0482a"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9648a50991c86df7d00c56c268c27754fcf4c80be2ba57fc4a00dc928c6fe934"
 dependencies = [
  "bitflags 2.4.2",
  "byteorder",
@@ -4818,12 +4823,14 @@
 [[package]]
 name = "heed-traits"
 version = "0.20.0-alpha.9"
-source = "git+https://github.com/meilisearch/heed?rev=036ac23f73a021894974b9adc815bc95b3e0482a#036ac23f73a021894974b9adc815bc95b3e0482a"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9648a50991c86df7d00c56c268c27754fcf4c80be2ba57fc4a00dc928c6fe934"
 
 [[package]]
 name = "heed-types"
 version = "0.20.0-alpha.9"
-source = "git+https://github.com/meilisearch/heed?rev=036ac23f73a021894974b9adc815bc95b3e0482a#036ac23f73a021894974b9adc815bc95b3e0482a"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9648a50991c86df7d00c56c268c27754fcf4c80be2ba57fc4a00dc928c6fe934"
 dependencies = [
  "bincode",
  "byteorder",
@@ -5858,7 +5865,8 @@
 [[package]]
 name = "lmdb-master-sys"
 version = "0.1.0"
-source = "git+https://github.com/meilisearch/heed?rev=036ac23f73a021894974b9adc815bc95b3e0482a#036ac23f73a021894974b9adc815bc95b3e0482a"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9648a50991c86df7d00c56c268c27754fcf4c80be2ba57fc4a00dc928c6fe934"
 dependencies = [
  "cc",
  "doxygen-rs",
@@ -5911,7 +5919,8 @@
 [[package]]
 name = "lsp-types"
 version = "0.94.1"
-source = "git+https://github.com/zed-industries/lsp-types?branch=updated-completion-list-item-defaults#90a040a1d195687bd19e1df47463320a44e93d7a"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "c66bfd44a06ae10647fe3f8214762e9369fd4248df1350924b4ef9e770a85ea1"
 dependencies = [
  "bitflags 1.3.2",
  "serde",
@@ -6583,7 +6592,8 @@
 [[package]]
 name = "nvim-rs"
 version = "0.6.0-pre"
-source = "git+https://github.com/KillTheMule/nvim-rs?branch=master#0d2b1c884f3c39a76b5b7aac0b429f4624843954"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for nvim-rs@0.6.0-pre"
 dependencies = [
  "async-trait",
  "futures 0.3.28",
@@ -7001,7 +7011,8 @@
 [[package]]
 name = "pathfinder_simd"
 version = "0.5.3"
-source = "git+https://github.com/servo/pathfinder.git?rev=30419d07660dc11a21e42ef4a7fa329600cff152#30419d07660dc11a21e42ef4a7fa329600cff152"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "ebf45976c56919841273f2a0fc684c28437e2f304e264557d9c72be5d5a718be"
 dependencies = [
  "rustc_version",
 ]
@@ -9874,7 +9885,8 @@
 [[package]]
 name = "taffy"
 version = "0.3.11"
-source = "git+https://github.com/DioxusLabs/taffy?rev=1876f72bee5e376023eaa518aa7b8a34c769bd1b#1876f72bee5e376023eaa518aa7b8a34c769bd1b"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "fab62c50c3d17993e7f0c72932e51ceeac5ec2b51c225fda8529d606159c99d8"
 dependencies = [
  "arrayvec",
  "grid",
@@ -10623,7 +10635,8 @@
 [[package]]
 name = "tree-sitter"
 version = "0.20.100"
-source = "git+https://github.com/tree-sitter/tree-sitter?rev=7b4894ba2ae81b988846676f54c0988d4027ef4f#7b4894ba2ae81b988846676f54c0988d4027ef4f"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter@0.20.100"
 dependencies = [
  "cc",
  "regex",
@@ -10633,7 +10646,8 @@
 [[package]]
 name = "tree-sitter-bash"
 version = "0.20.4"
-source = "git+https://github.com/tree-sitter/tree-sitter-bash?rev=7331995b19b8f8aba2d5e26deb51d2195c18bc94#7331995b19b8f8aba2d5e26deb51d2195c18bc94"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-bash@0.20.4"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10652,7 +10666,8 @@
 [[package]]
 name = "tree-sitter-cpp"
 version = "0.20.0"
-source = "git+https://github.com/tree-sitter/tree-sitter-cpp?rev=f44509141e7e483323d2ec178f2d2e6c0fc041c1#f44509141e7e483323d2ec178f2d2e6c0fc041c1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "8a869e3c5cef4e5db4e9ab16a8dc84d73010e60ada14cdc60d2f6d8aed17779d"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10661,7 +10676,8 @@
 [[package]]
 name = "tree-sitter-css"
 version = "0.19.0"
-source = "git+https://github.com/tree-sitter/tree-sitter-css?rev=769203d0f9abe1a9a691ac2b9fe4bb4397a73c51#769203d0f9abe1a9a691ac2b9fe4bb4397a73c51"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "c44fce8f9b603fef51e6384e2771ec5448bca1b71f1aa4ee2717a1803f9b279b"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10670,7 +10686,8 @@
 [[package]]
 name = "tree-sitter-elixir"
 version = "0.1.0"
-source = "git+https://github.com/elixir-lang/tree-sitter-elixir?rev=a2861e88a730287a60c11ea9299c033c7d076e30#a2861e88a730287a60c11ea9299c033c7d076e30"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9a9916f3e1c80b3c8aab8582604e97e8720cb9b893489b347cf999f80f9d469e"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10689,7 +10706,8 @@
 [[package]]
 name = "tree-sitter-go"
 version = "0.19.1"
-source = "git+https://github.com/tree-sitter/tree-sitter-go?rev=aeb2f33b366fd78d5789ff104956ce23508b85db#aeb2f33b366fd78d5789ff104956ce23508b85db"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "71967701c8214be4aa77e0260e98361e6fd71ceec1d9d03abb37a22c9f60d0ff"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10698,7 +10716,8 @@
 [[package]]
 name = "tree-sitter-gomod"
 version = "1.0.2"
-source = "git+https://github.com/camdencheek/tree-sitter-go-mod#bbe2fe3be4b87e06a613e685250f473d2267f430"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-gomod@1.0.2"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10707,7 +10726,8 @@
 [[package]]
 name = "tree-sitter-gowork"
 version = "0.0.1"
-source = "git+https://github.com/d1y/tree-sitter-go-work#a2a4b99b53b3740855ff33f0b54cab0bb4ce6f45"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-gowork@0.0.1"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10716,7 +10736,8 @@
 [[package]]
 name = "tree-sitter-heex"
 version = "0.0.1"
-source = "git+https://github.com/phoenixframework/tree-sitter-heex?rev=2e1348c3cf2c9323e87c2744796cf3f3868aa82a#2e1348c3cf2c9323e87c2744796cf3f3868aa82a"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-heex@0.0.1"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10735,7 +10756,8 @@
 [[package]]
 name = "tree-sitter-jsdoc"
 version = "0.20.0"
-source = "git+https://github.com/tree-sitter/tree-sitter-jsdoc?rev=6a6cf9e7341af32d8e2b2e24a37fbfebefc3dc55#6a6cf9e7341af32d8e2b2e24a37fbfebefc3dc55"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-jsdoc@0.20.0"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10754,7 +10776,8 @@
 [[package]]
 name = "tree-sitter-json"
 version = "0.20.0"
-source = "git+https://github.com/tree-sitter/tree-sitter-json?rev=40a81c01a40ac48744e0c8ccabbaba1920441199#40a81c01a40ac48744e0c8ccabbaba1920441199"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-json@0.20.0"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10763,7 +10786,8 @@
 [[package]]
 name = "tree-sitter-markdown"
 version = "0.0.1"
-source = "git+https://github.com/MDeiml/tree-sitter-markdown?rev=330ecab87a3e3a7211ac69bbadc19eabecdb1cca#330ecab87a3e3a7211ac69bbadc19eabecdb1cca"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-markdown@0.0.1"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10772,7 +10796,8 @@
 [[package]]
 name = "tree-sitter-proto"
 version = "0.0.2"
-source = "git+https://github.com/rewinfrey/tree-sitter-proto?rev=36d54f288aee112f13a67b550ad32634d0c2cb52#36d54f288aee112f13a67b550ad32634d0c2cb52"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "Error fetching checksum for tree-sitter-proto@0.0.2"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10821,7 +10846,8 @@
 [[package]]
 name = "tree-sitter-typescript"
 version = "0.20.2"
-source = "git+https://github.com/tree-sitter/tree-sitter-typescript?rev=5d20856f34315b068c41edaee2ac8a100081d259#5d20856f34315b068c41edaee2ac8a100081d259"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "079c695c32d39ad089101c66393aeaca30e967fba3486a91f573d2f0e12d290a"
 dependencies = [
  "cc",
  "tree-sitter",
@@ -10830,7 +10856,8 @@
 [[package]]
 name = "tree-sitter-yaml"
 version = "0.0.1"
-source = "git+https://github.com/zed-industries/tree-sitter-yaml?rev=f545a41f57502e1b5ddf2a6668896c1b0620f930#f545a41f57502e1b5ddf2a6668896c1b0620f930"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "324767d0ad6bc588467aa4b98f6f5cd6eda64ece1eae568f8fcf5b899bcf0fe9"
 dependencies = [
  "cc",
  "tree-sitter",
