diff --git a/Cargo.toml b/Cargo.toml
index fa7b39986..053fe0695 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -256,8 +256,8 @@ async-recursion = "1.0.0"
 async-tar = "0.4.2"
 async-trait = "0.1"
 bitflags = "2.4.2"
-blade-graphics = { git = "https://github.com/kvark/blade", rev = "f5766863de9dcc092e90fdbbc5e0007a99e7f9bf" }
-blade-macros = { git = "https://github.com/kvark/blade", rev = "f5766863de9dcc092e90fdbbc5e0007a99e7f9bf" }
+blade-graphics = "Retrying to extract version for blade-graphics using an alternative method"
+blade-macros = "Retrying to extract version for blade-macros using an alternative method"
 cap-std = "3.0"
 chrono = { version = "0.4", features = ["serde"] }
 clap = { version = "4.4", features = ["derive"] }
@@ -337,28 +337,28 @@ toml = "0.8"
 tokio = { version = "1", features = ["full"] }
 tower-http = "0.4.4"
 tree-sitter = { version = "0.20", features = ["wasm"] }
-tree-sitter-bash = { git = "https://github.com/tree-sitter/tree-sitter-bash", rev = "7331995b19b8f8aba2d5e26deb51d2195c18bc94" }
+tree-sitter-bash = "Retrying to extract version for tree-sitter-bash using an alternative method"
 tree-sitter-c = "0.20.1"
-tree-sitter-cpp = { git = "https://github.com/tree-sitter/tree-sitter-cpp", rev = "f44509141e7e483323d2ec178f2d2e6c0fc041c1" }
-tree-sitter-css = { git = "https://github.com/tree-sitter/tree-sitter-css", rev = "769203d0f9abe1a9a691ac2b9fe4bb4397a73c51" }
-tree-sitter-elixir = { git = "https://github.com/elixir-lang/tree-sitter-elixir", rev = "a2861e88a730287a60c11ea9299c033c7d076e30" }
+tree-sitter-cpp = "Retrying to extract version for tree-sitter-cpp using an alternative method"
+tree-sitter-css = "Retrying to extract version for tree-sitter-css using an alternative method"
+tree-sitter-elixir = "Retrying to extract version for tree-sitter-elixir using an alternative method"
 tree-sitter-embedded-template = "0.20.0"
-tree-sitter-go = { git = "https://github.com/tree-sitter/tree-sitter-go", rev = "aeb2f33b366fd78d5789ff104956ce23508b85db" }
-tree-sitter-gomod = { git = "https://github.com/camdencheek/tree-sitter-go-mod" }
-tree-sitter-gowork = { git = "https://github.com/d1y/tree-sitter-go-work" }
+tree-sitter-go = "Retrying to extract version for tree-sitter-go using an alternative method"
+tree-sitter-gomod = "Retrying to extract version for tree-sitter-gomod using an alternative method"
+tree-sitter-gowork = "Retrying to extract version for tree-sitter-gowork using an alternative method"
 rustc-demangle = "0.1.23"
-tree-sitter-heex = { git = "https://github.com/phoenixframework/tree-sitter-heex", rev = "2e1348c3cf2c9323e87c2744796cf3f3868aa82a" }
+tree-sitter-heex = "Retrying to extract version for tree-sitter-heex using an alternative method"
 tree-sitter-html = "0.19.0"
-tree-sitter-jsdoc = { git = "https://github.com/tree-sitter/tree-sitter-jsdoc", rev = "6a6cf9e7341af32d8e2b2e24a37fbfebefc3dc55" }
-tree-sitter-json = { git = "https://github.com/tree-sitter/tree-sitter-json", rev = "40a81c01a40ac48744e0c8ccabbaba1920441199" }
-tree-sitter-markdown = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "330ecab87a3e3a7211ac69bbadc19eabecdb1cca" }
-tree-sitter-proto = { git = "https://github.com/rewinfrey/tree-sitter-proto", rev = "36d54f288aee112f13a67b550ad32634d0c2cb52" }
+tree-sitter-jsdoc = "Retrying to extract version for tree-sitter-jsdoc using an alternative method"
+tree-sitter-json = "Retrying to extract version for tree-sitter-json using an alternative method"
+tree-sitter-markdown = "Retrying to extract version for tree-sitter-markdown using an alternative method"
+tree-sitter-proto = "Retrying to extract version for tree-sitter-proto using an alternative method"
 tree-sitter-python = "0.20.2"
 tree-sitter-regex = "0.20.0"
 tree-sitter-ruby = "0.20.0"
 tree-sitter-rust = "0.20.3"
-tree-sitter-typescript = { git = "https://github.com/tree-sitter/tree-sitter-typescript", rev = "5d20856f34315b068c41edaee2ac8a100081d259" }
-tree-sitter-yaml = { git = "https://github.com/zed-industries/tree-sitter-yaml", rev = "f545a41f57502e1b5ddf2a6668896c1b0620f930" }
+tree-sitter-typescript = "Retrying to extract version for tree-sitter-typescript using an alternative method"
+tree-sitter-yaml = "Retrying to extract version for tree-sitter-yaml using an alternative method"
 unindent = "0.1.7"
 unicase = "2.6"
 unicode-segmentation = "1.10"
@@ -415,9 +415,9 @@ features = [
 ]
 
 [patch.crates-io]
-tree-sitter = { git = "https://github.com/tree-sitter/tree-sitter", rev = "7b4894ba2ae81b988846676f54c0988d4027ef4f" }
+tree-sitter = "Retrying to extract version for tree-sitter using an alternative method"
 # Workaround for a broken nightly build of gpui: See #7644 and revisit once 0.5.3 is released.
-pathfinder_simd = { git = "https://github.com/servo/pathfinder.git", rev = "30419d07660dc11a21e42ef4a7fa329600cff152" }
+pathfinder_simd = "Retrying to extract version for pathfinder_simd using an alternative method"
 
 [profile.dev]
 split-debuginfo = "unpacked"
