diff --git a/CMakeLists.txt b/CMakeLists.txt
index e807704..533c69b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(CMAKE_CXX_STANDARD 17)
 add_subdirectory(modules)
 include_directories(modules)
 
-juce_set_aax_sdk_path(C:/SDKs/AAX_SDK/)
+#juce_set_aax_sdk_path(C:/SDKs/AAX_SDK/)
 
 set(JUCE_FORMATS AU VST3)
 
@@ -37,7 +37,7 @@ juce_add_plugin(Chameleon
     PLUGIN_CODE Chm3 
     FORMATS ${JUCE_FORMATS}
     ProductName "Chameleon"
-    LV2_URI https://github.com/GuitarML/Chameleon
+    LV2URI https://github.com/GuitarML/Chameleon
     LV2_SHARED_LIBRARY_NAME Chameleon
     ICON_BIG resources/logo.png
 
