diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 98ae25c76..165089350 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -1518,7 +1518,7 @@ config("compiler_codegen") {
     # Disable outlining everywhere on arm64 except Win. For more information see
     # crbug.com/931297 for Android and crbug.com/1410297 for iOS.
     # TODO(crbug.com/40890229): Enable this on Windows if possible.
-    cflags += [ "-mno-outline" ]
+    cflags += [ "-mno-outline-atomics" ]
 
     # This can be removed once https://bugs.llvm.org/show_bug.cgi?id=40348
     # has been resolved, and -mno-outline is obeyed by the linker during
