diff --git a/src/core/utilities.cpp b/src/core/utilities.cpp
index fe106e2b7..da163caa1 100644
--- a/src/core/utilities.cpp
+++ b/src/core/utilities.cpp
@@ -346,6 +346,14 @@ QString GetConfigPath(ConfigPath config) {
 #ifdef Q_OS_DARWIN
       return mac::GetApplicationSupportPath() + "/" +
              QCoreApplication::organizationName();
+#elif defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN)
+     char* xdg = getenv("XDG_CONFIG_HOME");
+     if (!xdg || !*xdg) {
+       return QString("%1/.config/%2")
+           .arg(QDir::homePath(), QCoreApplication::organizationName());
+     } else {
+       return QString("%1/%2").arg(xdg, QCoreApplication::organizationName());
+     }
 #else
       return QString("%1/.config/%2")
           .arg(QDir::homePath(), QCoreApplication::organizationName());
