--- a/usr/share/applications/codium-insiders.desktop
+++ b/usr/share/applications/codium-insiders.desktop
@@ -2,8 +2,8 @@
 Name=VSCodium - Insiders
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=/usr/share/codium-insiders/codium-insiders --unity-launch %F
-Icon=vscodium-insiders
+Exec=codium-insiders --unity-launch %F
+Icon=com.vscodium.codium-insiders
 Type=Application
 StartupNotify=false
 StartupWMClass=VSCodium - Insiders
@@ -14,5 +14,5 @@

 [Desktop Action new-empty-window]
 Name=New Empty Window
-Exec=/usr/share/codium-insiders/codium-insiders --new-window %F
-Icon=vscodium-insiders
+Exec=codium-insiders --new-window %F
+Icon=com.vscodium.codium-insiders
