
diff -uNr -X .diff_ignore xtrkcad-fork-xtrkcad/distribution/flatpak/common_source.sh new-fork-xtrkcad/distribution/flatpak/common_source.sh
--- xtrkcad-fork-xtrkcad/distribution/flatpak/common_source.sh	1969-12-31 19:00:00.000000000 -0500
+++ new-fork-xtrkcad/distribution/flatpak/common_source.sh	2025-12-21 06:38:32.966312196 -0500
@@ -0,0 +1,7 @@
+if [ -n "$XDG_CACHE_HOME" ]; then
+    cp -pr /app/share/xtrkcad/* ${XDG_CACHE_HOME}
+fi
+export XTRKCADLIB=${XDG_CACHE_HOME}
+# there is no access to host gtk3 modules
+#unset GTK3_MODULES
+#unset GTK_MODULES
