
diff -uNrw -X .diff_ignore --no-dereference xtrkcad-fork-xtrkcad/distribution/flatpak/common_source.sh new-fork-xtrkcad/distribution/flatpak/common_source.sh
--- xtrkcad-fork-xtrkcad/distribution/flatpak/common_source.sh	1969-12-31 19:00:00.000000000 -0500
+++ new-fork-xtrkcad/distribution/flatpak/common_source.sh	2025-12-21 11:32:31.877640192 -0500
@@ -0,0 +1,4 @@
+if [ -n "$XDG_CACHE_HOME" ]; then
+    cp -pr /app/share/xtrkcad/* ${XDG_CACHE_HOME}
+fi
+export XTRKCADLIB=${XDG_CACHE_HOME}
