--- xtrkcad-fork-xtrkcad/distribution/flatpak/common_source.sh	2025-12-05 05:43:18.422084216 -0500
+++ new-fork-xtrkcad/distribution/flatpak/common_source.sh	2025-12-05 09:11:17.075174682 -0500
@@ -0,0 +1,21 @@
+if [ -n "$XDG_CACHE_HOME" ]; then
+    rm -rf ${XDG_CACHE_HOME}/*
+    cp -pr /app/share/xtrkcad/* ${XDG_CACHE_HOME}
+fi
+export XTRKCADLIB=${XDG_CACHE_HOME}
+# there is no access to host gtk3 modules
+unset GTK3_MODULES
+unset GTK_MODULES
+BOOKMARK="file://$XTRKCADLIB xtrkcad-lib"
+# gtk2 bookmarks
+FILE="$HOME/.gtk-bookmarks"
+touch $FILE
+sed "/xtrkcad-lib$/d" $FILE > ${FILE}_
+cp ${FILE}_ $FILE
+rm ${FILE}_
+echo "$BOOKMARK" >> $FILE
+if [ "$1" = "-d" ]; then
+    echo "ARCH=$(uname -m)"
+    env | sort | grep -E "DISPLAY|XTRK|XDG"
+    echo
+fi
