From ff8e7fd3b8e21eaac77bfc8e6ff90b48ecc3bc1c Mon Sep 17 00:00:00 2001
From: Henri Menke <henri@henrimenke.de>
Date: Thu, 11 Sep 2025 10:05:58 +0200
Subject: [PATCH] Compatibility with CMake 4

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d41e605..ec3c15d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@
 # #################################### #
 # $Id$
 # #################################### #
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.1...3.10)
 
 PROJECT(klatexformula)
 
-- 
2.51.0

