From 01e8b4ffad3cf12c17fbfd7a07a6d4ecd36afea4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Claudio=20Andr=C3=A9?= <claudioandre.br@gmail.com>
Date: Sat, 15 Dec 2018 23:35:51 -0200
Subject: [PATCH] Handle self confined system wide build.

---
 src/listconf.c |  3 +++
 src/params.h   | 12 ++++++++++--
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/src/listconf.c b/src/listconf.c
index d974cf1ad..9e7557e3e 100644
--- a/src/listconf.c
+++ b/src/listconf.c
@@ -143,6 +143,9 @@ static void listconf_list_build_info(void)
 	       SIMD_PARA_SHA256, SIMD_PARA_SHA512);
 #endif
 #if JOHN_SYSTEMWIDE
+#if defined(_BOXED)
+	puts("Deploy: " "sandbox ready");
+#endif
 	puts("System-wide exec: " JOHN_SYSTEMWIDE_EXEC);
 	puts("System-wide home: " JOHN_SYSTEMWIDE_HOME);
 	puts("Private home: " JOHN_PRIVATE_HOME);
diff --git a/src/params.h b/src/params.h
index ee3314e5d..ad349735f 100644
--- a/src/params.h
+++ b/src/params.h
@@ -100,10 +100,18 @@
 
 #if JOHN_SYSTEMWIDE
 #ifndef JOHN_SYSTEMWIDE_EXEC /* please refer to the notes above */
-#define JOHN_SYSTEMWIDE_EXEC		"/usr/libexec/john"
+#if defined(_SNAP)
+#define JOHN_SYSTEMWIDE_EXEC		"/snap/john-the-ripper/current/run"
+#else
+#define JOHN_SYSTEMWIDE_EXEC		"/app/bin"
+#endif
 #endif
 #ifndef JOHN_SYSTEMWIDE_HOME
-#define JOHN_SYSTEMWIDE_HOME		"/usr/share/john"
+#if defined(_SNAP)
+#define JOHN_SYSTEMWIDE_HOME		"/snap/john-the-ripper/current/run"
+#else
+#define JOHN_SYSTEMWIDE_HOME		"/app/bin"
+#endif
 #endif
 #define JOHN_PRIVATE_HOME		"~/.john"
 #endif
-- 
2.19.1

