diff -ruN a/python/lib/Variables.py b/python/lib/Variables.py
--- a/python/lib/Variables.py	2020-05-17 16:05:33.000000000 +0200
+++ b/python/lib/Variables.py	2023-07-26 10:13:31.348152118 +0200
@@ -26,7 +26,7 @@
 # Debian packagers should switch this to TRUE
 # It will disable update alerts, bug reports, statistics
 # It will set the good locale directory, and it will use the good msttcorefonts
-os.environ["DEBIAN_PACKAGE"] = "FALSE"
+os.environ["DEBIAN_PACKAGE"] = "TRUE"
 
 # Variables PlayOnMac
 if os.environ["POL_OS"] == "Mac":
