From 495ac0d98cd4d70048af849d7be78c7cccdc7672 Mon Sep 17 00:00:00 2001
From: Federico Bruni <fede@inventati.org>
Date: Sun, 15 Aug 2021 00:08:01 +0200
Subject: [PATCH] Add tags to pass Flathub appstream validation

---
 denemo.appdata.xml.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/denemo.appdata.xml.in b/denemo.appdata.xml.in
index fdbcb427e..2fd7a929a 100644
--- a/denemo.appdata.xml.in
+++ b/denemo.appdata.xml.in
@@ -46,4 +46,8 @@
     <keyword>MIDI</keyword>
     <keyword>Music</keyword>
  </keywords>
+ <releases>
+  <release version="2.5.0" date="2021-03-03" />
+ </releases>
+ <content_rating type="oars-1.1" />
 </application>
-- 
2.31.1

