From 6824e221dbdff3126fe50e0717084a9654018a41 Mon Sep 17 00:00:00 2001
From: Richard Shann <richard@rshann.plus.com>
Date: Sat, 14 Aug 2021 10:16:31 +0100
Subject: [PATCH] Fix missing screenshots

---
 denemo.appdata.xml.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/denemo.appdata.xml.in b/denemo.appdata.xml.in
index c28380aef..fdbcb427e 100644
--- a/denemo.appdata.xml.in
+++ b/denemo.appdata.xml.in
@@ -29,9 +29,9 @@
  </description>
  <url type="homepage">http://www.denemo.org/</url>
  <screenshots> 
- <screenshot type="default" width="1584" height="891">http://www.denemo.org/~rshann/DenemoScreenshot.png</screenshot>
- <screenshot width="1584" height="891">http://www.denemo.org/~rshann/DenemoScreenshot2.png</screenshot>
- <screenshot width="1056" height="594">http://www.denemo.org/~rshann/DenemoScreenshot3.png</screenshot>
+ <screenshot type="default" width="826" height="1169">http://www.denemo.org/wp-content/uploads/2013/10/display68.png</screenshot>
+ <screenshot width="1920" height="1080">http://www.denemo.org/wp-content/uploads/2013/10/display4.png</screenshot>
+ <screenshot width="1920" height="1080">http://www.denemo.org/wp-content/uploads/2013/10/display7.png</screenshot>
  </screenshots>
  <updatecontact>denemo-devel@gnu.org</updatecontact>
  <project_group>Denemo</project_group>
-- 
2.31.1

