diff -ruN a/lib/scripts.lib b/lib/scripts.lib
--- a/lib/scripts.lib	2020-05-17 16:05:33.000000000 +0200
+++ b/lib/scripts.lib	2023-07-05 14:18:40.509038313 +0200
@@ -391,7 +391,7 @@
 			# Do nothing on Mac OS
 			if [ -n "$Categories" -a "$POL_OS" = "Linux" ] || [ -n "$Categories" -a "$POL_OS" = "BSD" ]; then
 				LOCALAPPS="$HOME/.local/share/applications"
-				make_desktop_shortcut "$iconPath" "$ICON_FILENAME" "$LOCALAPPS" "$PLAYONLINUX/playonlinux --run \"$ICON_FILENAME\"" "$binary_name" "$Categories" "playonlinux-"
+				make_desktop_shortcut "$iconPath" "$ICON_FILENAME" "$LOCALAPPS" "flatpak run com.playonlinux.PlayOnLinux4 --run \"$ICON_FILENAME\"" "$binary_name" "$Categories" "playonlinux-"
 			fi
 		fi
 
@@ -401,7 +401,7 @@
 			if [ "$POL_OS" = "Mac" ]; then
 				POL_SetupWindow_AutoApp "$ICON_FILENAME"
 			else
-				make_desktop_shortcut "$iconPath" "$ICON_FILENAME" "$DESKTOP" "$PLAYONLINUX/playonlinux --run \"$ICON_FILENAME\"" "$binary_name" "$Categories"
+				make_desktop_shortcut "$iconPath" "$ICON_FILENAME" "$DESKTOP" "flatpak run com.playonlinux.PlayOnLinux4 --run \"$ICON_FILENAME\"" "$binary_name" "$Categories"
 			fi
 		fi
 	fi
@@ -1046,7 +1046,7 @@
 		# http://stackoverflow.com/questions/8939580/bash-split-a-string-exactly-like-readline-would-split-it
 		eval set -- $(grep '^POL_Wine ' "$POL_USER_ROOT/shortcuts/$PACKAGE"|tail -n 1)
 		local binary="$2"
-		make_desktop_shortcut "$iconPath" "$PACKAGE" "$DESKTOP" "$PLAYONLINUX/playonlinux --run \"$PACKAGE\"" "$binary" ""
+		make_desktop_shortcut "$iconPath" "$PACKAGE" "$DESKTOP" "flatpak run com.playonlinux.PlayOnLinux4 --run \"$PACKAGE\"" "$binary" ""
 		;;
 
 	    *)
