diff --git a/sources/documentview.cpp b/sources/documentview.cpp
index 8f986aa..3ecabb9 100644
--- a/sources/documentview.cpp
+++ b/sources/documentview.cpp
@@ -1050,7 +1050,8 @@ QStringList DocumentView::saveFilter() const
 
 bool DocumentView::canSave() const
 {
-    return m_document->canSave();
+    // 'Save as' is not available because of filesystem sanboxing provided by Flatpak.
+    return false;
 }
 
 void DocumentView::setContinuousMode(bool continuousMode)
