--- roccat-tools-5.9.0/roccateventhandler/CMakeLists.txt	2019-04-13 21:09:00.000000000 +0200
+++ roccat-tools-5.9.0/roccateventhandler/CMakeLists.txt	2022-05-23 16:29:23.192271828 +0200
@@ -37,7 +37,3 @@
   DESTINATION bin
   PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
 )
-
-INSTALL(FILES roccateventhandler.desktop
-  DESTINATION /etc/xdg/autostart
-)
