diff --git a/src/plugins/CMakeLists.txt b/src/plugins/CMakeLists.txt
index 9d713ff3..716de541 100644
--- a/src/plugins/CMakeLists.txt
+++ b/src/plugins/CMakeLists.txt
@@ -33,11 +33,7 @@ macro(add_opt_subdirectory PLUGIN_NAME)
 	endif()
 endmacro()
 
-add_opt_subdirectory(chocolate-doom)
-add_opt_subdirectory(odamex)
-add_opt_subdirectory(zandronum)
 add_opt_subdirectory(zandronumq)
-add_opt_subdirectory(turok2ex)
 
 # For unmaintained source ports
 option(BUILD_LEGACY_PLUGINS "Build discontinued plugins." OFF)
