
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/src/dynamic_loader/CMakeLists.txt	Wed Sep 30 18:32:20 2015 +0200
@@ -0,0 +1,20 @@
+set(DYNAMIC_LOADER_FILES
+    dynamic_loader.h dynamic_loader.cc
+    dynamic_loader_implementation.h dynamic_loader_implementation.cc
+    win32_implementation.h sunny_implementation.h gmodule_implementation.h
+)
+
+if(WIN32)
+  list(APPEND DYNAMIC_LOADER_FILES win32_implementation.cc)
+else()
+  if(HAVE_DL)
+    list(APPEND DYNAMIC_LOADER_FILES sunny_implementation.cc)
+  else()
+    if(HAVE_GMODULE)
+      list(APPEND DYNAMIC_LOADER_FILES gmodule_implementation.cc)
+    else()
+      list(APPEND DYNAMIC_LOADER_FILES null_implementation.h)
+    endif()
+  endif()
+endif()
+add_library(dynamic_loader STATIC ${DYNAMIC_LOADER_FILES})
diff -r 5704f1d63687 -r a10a2e299fd1 src/layer_selection/CMakeLists.txt
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/src/layer_selection/CMakeLists.txt	Wed Sep 30 18:32:20 2015 +0200
@@ -0,0 +1,5 @@
+set(LAYER_SELECTION_FILES
+    info.h info.cc layer_selection.h layer_selection.cc
+    selector.h selector.cc
+)
+add_library(layer_selection STATIC ${LAYER_SELECTION_FILES})
