diff -r 8fe051721be3 -r 50be9e24c1b3 src/dynamic_loader/sunny_implementation.cc
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/src/dynamic_loader/sunny_implementation.cc	Wed Sep 30 16:21:45 2015 +0200
@@ -0,0 +1,65 @@
+#include "sunny_implementation.h"
+
+
+#ifdef HAVE_DL
+
+
+SunnyDynamicLoaderImplementation::SunnyDynamicLoaderImplementation(const std::string& a_library_name) :
+    super(a_library_name), handle_(nullptr)
+{}
+
+
+void
+SunnyDynamicLoaderImplementation::open()
+{
+    if (handle_)
+    {
+        throw super::error("already open");
+    }
+    handle_ = dlopen(library_name().c_str(), RTLD_LAZY);
+    if (!handle_)
+    {
+        throw super::error(dlerror());
+    }
+}
+
+
+void
+SunnyDynamicLoaderImplementation::close()
+{
+    if (!handle_)
+    {
+        throw super::error("not open");
+    }
+    if (dlclose(handle_) != 0)
+    {
+        throw super::error(dlerror());
+    }
+}
+
+
+void*
+SunnyDynamicLoaderImplementation::resolve(const std::string& symbol_name) const
+{
+    if (!handle_)
+    {
+        throw super::error("not open");
+    }
+
+    dlerror();
+    void* symbol = dlsym(handle_, symbol_name.c_str());
+    char* message = dlerror();
+    if (message != nullptr)
+    {
+        throw super::error(message);
+    }
+    if (symbol == nullptr)
+    {
+        throw super::error("symbol points nowhere");
+    }
+
+    return symbol;
+}
+
+
+#endif // HAVE_DL
diff -r 8fe051721be3 -r 50be9e24c1b3 src/dynamic_loader/sunny_implementation.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/src/dynamic_loader/sunny_implementation.h	Wed Sep 30 16:21:45 2015 +0200
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2013, 2015 Dr. Christoph L. Spiel
+ *
+ * This file is part of Enblend.
+ *
+ * Enblend is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Enblend is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Enblend; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef SUNNY_IMPLEMENTATION_H_INCLUDED
+#define SUNNY_IMPLEMENTATION_H_INCLUDED
+
+
+#include "dynamic_loader_implementation.h"
+
+#include "config.h"
+
+
+#ifdef HAVE_DL
+
+#define HAVE_DYNAMICLOADER_IMPL
+
+#include <dlfcn.h>
+
+class SunnyDynamicLoaderImplementation : public DynamicLoaderImplementation
+{
+    typedef DynamicLoaderImplementation super;
+
+public:
+    explicit SunnyDynamicLoaderImplementation(const std::string& a_library_name);
+    void open();
+    void close();
+    void* resolve(const std::string& symbol_name) const;
+
+private:
+    void* handle_;
+}; // class SunnyDynamicLoaderImplementation
+
+typedef SunnyDynamicLoaderImplementation ActualDynamicLoaderImplementation;
+
+#endif // HAVE_DL
+
+#endif // SUNNY_IMPLEMENTATION_H_INCLUDED
+
+
+// Local Variables:
+// mode: c++
+// End:
