--- fabla-3/CMakeLists.txt	2016-12-05 07:35:00.000000000 -0500
+++ fabla-3/CMakeLists.txt-new	2020-06-21 00:08:48.220680002 -0400
@@ -26,7 +26,7 @@
 include_directories( ${SNDFILE_INCLUDE_DIRS}  )
 link_directories   ( ${SNDFILE_LIBRARY_DIRS}  )
 
-SET(CMAKE_SHARED_LINKER_FLAGS "-fPIC -shared -Wl,-z,nodelete -Wl,--no-undefined")
+SET(CMAKE_SHARED_LINKER_FLAGS "-fPIC -shared -Wl,-z,nodelete -Wl,--no-undefined -lXft -lX11 -lpthread -lfontconfig -lfreetype -ldl")
 
 IF(RELEASE_BUILD)
   SET(CMAKE_CXX_FLAGS "-g -Wall -Wno-unused-variable -msse2 -mfpmath=sse -ffast-math")
@@ -70,16 +70,16 @@
 set_target_properties(fabla PROPERTIES PREFIX "")
 
 # add the install targets
-install (TARGETS fabla DESTINATION lib/lv2/fabla.lv2/)
+install (TARGETS fabla DESTINATION lv2/fabla.lv2/)
 
 # install .ttl files
-install(FILES "dsp/fabla.ttl"     DESTINATION lib/lv2/fabla.lv2/)
-install(FILES "dsp/manifest.ttl"  DESTINATION lib/lv2/fabla.lv2/)
+install(FILES "dsp/fabla.ttl"     DESTINATION lv2/fabla.lv2/)
+install(FILES "dsp/manifest.ttl"  DESTINATION lv2/fabla.lv2/)
 
-install( DIRECTORY "presets/fabla808.lv2"           DESTINATION lib/lv2 )
-install( DIRECTORY "presets/fablaHardElectro.lv2"   DESTINATION lib/lv2 )
-install( DIRECTORY "presets/fablaSavageDrums.lv2"   DESTINATION lib/lv2 )
-install( DIRECTORY "presets/fablaEasternHop.lv2"    DESTINATION lib/lv2 )
+install( DIRECTORY "presets/fabla808.lv2"           DESTINATION lv2 )
+install( DIRECTORY "presets/fablaHardElectro.lv2"   DESTINATION lv2 )
+install( DIRECTORY "presets/fablaSavageDrums.lv2"   DESTINATION lv2 )
+install( DIRECTORY "presets/fablaEasternHop.lv2"    DESTINATION lv2 )
 
 # install presets
-#install(COPY "presets/fabla808.lv2"  DESTINATION lib/lv2/ )
+#install(COPY "presets/fabla808.lv2"  DESTINATION lv2/ )
