diff -ur FAudio-19.06.07.old/CMakeLists.txt FAudio-19.06.07/CMakeLists.txt
--- FAudio-19.06.07.old/CMakeLists.txt	2019-06-28 22:44:10.585331478 +0200
+++ FAudio-19.06.07/CMakeLists.txt	2019-06-28 22:50:41.445436174 +0200
@@ -156,7 +156,7 @@
 
 # SDL2 Dependency
 if (NOT DEFINED SDL2_INCLUDE_DIRS AND NOT DEFINED SDL2_LIBRARIES)
-	find_package(SDL2 CONFIG)
+        pkg_check_modules(SDL2 sdl2)
 endif()
 if (TARGET SDL2::SDL2)
 	message(STATUS "using TARGET SDL2::SDL2")
Only in FAudio-19.06.07: CMakeLists.txt~
