diff --git a/filmulator-gui/filmulator-gui.desktop.in b/filmulator-gui/filmulator-gui.desktop.in
index 96ee5c7..9c686c1 100644
--- a/filmulator-gui/filmulator-gui.desktop.in
+++ b/filmulator-gui/filmulator-gui.desktop.in
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=Filmulator
 Comment=Streamlined photo editor
-Exec=cd/usr/lib/filmulator-gui && /usr/lib/filmulator-gui/filmulator-gui
+Exec=filmulator-gui
 Icon=filmulator-gui
 Terminal=false
 Categories=Graphics;Photography;Qt;
diff --git a/filmulator-gui/filmulator-gui.pro b/filmulator-gui/filmulator-gui.pro
index 8c5b407..cd5eb5a 100644
--- a/filmulator-gui/filmulator-gui.pro
+++ b/filmulator-gui/filmulator-gui.pro
@@ -62,8 +62,8 @@ SOURCES += qml/filmulator-gui/*.qml \
 TRANSLATIONS = translations/filmulator-gui_de.ts
 
 # Installation path
-unix:target.path = /usr/lib/filmulator-gui
-unix:desktop.path = /usr/share/applications
+unix:target.path = /app/lib/filmulator-gui
+unix:desktop.path = /app/share/applications
 
 unix:desktop.files += ./filmulator_gui.desktop
 
@@ -76,8 +76,8 @@ win32:LIBS += -L/usr/lib
 
 unix {
 script.extra = move_script; install -m 755 -p filmulator
-extra.path = /usr/bin
-LIBS += -L/usr/local/lib
+extra.path = /app/bin
+LIBS += -L/app/lib
 }
 
 # Please do not modify the following two lines. Required for deployment.
