diff --git a/CMakeLists.txt b/CMakeLists.txt
index 84c66a7..74c831d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,22 +28,9 @@ project(Fire VERSION 0.9.9)
 
 # find_package(JUCE CONFIG REQUIRED)        # If you've installed JUCE to your system
 # or
-# add_subdirectory(JUCE)                    # If you've put JUCE in a subdirectory called JUCE
-
-include(FetchContent)
-FetchContent_Declare(
-        JUCE
-        GIT_REPOSITORY https://github.com/juce-framework/JUCE.git
-        GIT_TAG 7.0.1
-)
-FetchContent_MakeAvailable(JUCE)
-
-FetchContent_Declare(
-        readerwriterqueue
-        GIT_REPOSITORY https://github.com/cameron314/readerwriterqueue
-        GIT_TAG v1.0.6
-)
-FetchContent_MakeAvailable(readerwriterqueue)
+add_subdirectory(JUCE)                    # If you've put JUCE in a subdirectory called JUCE
+
+add_subdirectory(readerwriterqueue)
 
 # If you are building a VST2 or AAX plugin, CMake needs to be told where to find these SDKs on your
 # system. This setup should be done before calling `juce_add_plugin`.
@@ -173,12 +160,7 @@ set(TestFiles
         test/PluginTest.cpp)
 
 # Download the tagged version of Catch2
-Include(FetchContent)
-FetchContent_Declare(
-        Catch2
-        GIT_REPOSITORY https://github.com/catchorg/Catch2.git
-        GIT_TAG v2.13.7)
-FetchContent_MakeAvailable(Catch2)
+add_subdirectory(Catch2)
 
 # Setup the test executable, again C++ 20 please
 add_executable(Tests ${TestFiles})
@@ -199,4 +181,4 @@ source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR}/test PREFIX "" FILES ${TestFiles})
 # We have to manually provide the source directory here for now
 # https://github.com/catchorg/Catch2/issues/2026
 include(${Catch2_SOURCE_DIR}/contrib/Catch.cmake)
-catch_discover_tests(Tests)
\ No newline at end of file
+catch_discover_tests(Tests)
