diff --git a/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop b/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
index 18a6b3197..37fc5535a 100755
--- a/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
+++ b/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
@@ -1,14 +1,12 @@
 [Desktop Entry]
-Encoding=UTF-8
-Version=1.9
 Name=Autofirma
 Type=Application
 Terminal=false
-Categories=Office;Utilities;Signature;Java
-Exec=/usr/bin/autofirma %u
-Icon=/usr/lib/Autofirma/Autofirma.png
+Categories=Office;Utility;Java
+Exec=autofirma %u
+Icon=es.gob.afirma
 GenericName=Herramienta de firma
-Comment=Herramienta de firma
+Comment=Aplicación de firma electrónica del Gobierno de España
 MimeType=x-scheme-handler/afirma;
 StartupNotify=true
 StartupWMClass=autofirma
