diff --git a/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma b/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
index 95f46ed9a..fc4e034da 100755
--- a/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
+++ b/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
@@ -2 +2 @@
-java -Djdk.tls.maxHandshakeMessageSize=65536 -jar /usr/lib/Autofirma/autofirma.jar "$@"
+java -Djdk.tls.maxHandshakeMessageSize=65536 -jar /app/lib/autofirma.jar "$@"
diff --git a/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop b/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
index 18a6b3197..3bea4688e 100755
--- a/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
+++ b/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
@@ -8,2 +8,2 @@ Categories=Office;Utilities;Signature;Java
-Exec=/usr/bin/autofirma %u
-Icon=/usr/lib/Autofirma/Autofirma.png
+Exec=autofirma %u
+Icon=es.gob.afirma
