diff --git a/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop b/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
index 3bea4688e..29ceebdb4 100755
--- a/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
+++ b/afirma-simple-installer/linux/instalador_deb/src/usr/share/applications/afirma.desktop
@@ -2,2 +1,0 @@
-Encoding=UTF-8
-Version=1.9
@@ -7 +5 @@ Terminal=false
-Categories=Office;Utilities;Signature;Java
+Categories=Office;Utility;Java
@@ -11 +9 @@ GenericName=Herramienta de firma
-Comment=Herramienta de firma
+Comment=Aplicación de firma electrónica del Gobierno de España
