diff --git a/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma b/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
index 95f46ed9a..fc4e034da 100755
--- a/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
+++ b/afirma-simple-installer/linux/instalador_deb/src/usr/bin/autofirma
@@ -2 +2 @@
-java -Djdk.tls.maxHandshakeMessageSize=65536 -jar /usr/lib/Autofirma/autofirma.jar "$@"
+java -Djdk.tls.maxHandshakeMessageSize=65536 -jar /app/lib/autofirma.jar "$@"
