diff --git a/src/kclock/qml/AnalogClock.qml b/src/kclock/qml/AnalogClock.qml
index a983d4f..62077c5 100644
--- a/src/kclock/qml/AnalogClock.qml
+++ b/src/kclock/qml/AnalogClock.qml
@@ -32,20 +32,17 @@ Item {
     property int minutes
     property int seconds
     
-    PlasmaCore.DataSource {
-        id: dataSource
-        engine: "time"
-        connectedSources: "Local"
+    Timer {
+        running: true
+        repeat: true
+        triggeredOnStart: true
         interval: 1000
-        onDataChanged: {
-            var date = new Date(data["Local"]["DateTime"]);
+        onTriggered: {
+            let date = new Date();
             hours = date.getHours();
             minutes = date.getMinutes();
             seconds = date.getSeconds();
         }
-        Component.onCompleted: {
-            onDataChanged();
-        }
     }
 
     PlasmaCore.Svg {
diff --git a/src/kclock/qml/TimerForm.qml b/src/kclock/qml/TimerForm.qml
index daea471..5470a6d 100644
--- a/src/kclock/qml/TimerForm.qml
+++ b/src/kclock/qml/TimerForm.qml
@@ -164,6 +164,5 @@ Kirigami.FormLayout {
         id: label
         Kirigami.FormData.label: i18n("<b>Label (optional):</b>")
         text: i18n("New Timer") // default
-        onTextChanged: timer.label = text
     }
 } 
diff --git a/src/kclockd/alarm.cpp b/src/kclockd/alarm.cpp
index 83e99ab..91e8036 100644
--- a/src/kclockd/alarm.cpp
+++ b/src/kclockd/alarm.cpp
@@ -229,7 +229,7 @@ void Alarm::calculateNextRingTime()
 quint64 Alarm::nextRingTime()
 {
     // day changed, re-calculate
-    if (this->m_nextRingTime < static_cast<unsigned long long>(QDateTime::currentSecsSinceEpoch())) {
+    if (this->m_nextRingTime <= static_cast<unsigned long long>(QDateTime::currentSecsSinceEpoch())) {
         calculateNextRingTime();
     }
     return m_nextRingTime;
