diff -u b/linux/org.frescobaldi.Frescobaldi.metainfo.xml.in b/linux/org.frescobaldi.Frescobaldi.metainfo.xml.in
--- b/linux/org.frescobaldi.Frescobaldi.metainfo.xml.in
+++ b/linux/org.frescobaldi.Frescobaldi.metainfo.xml.in
@@ -1,11 +1,10 @@
-<?xml version="1.0" encoding="UTF-8"?>
+<?xml version="1.0" encoding="utf-8"?>
 <component type="desktop-application">
   <id>org.frescobaldi.Frescobaldi</id>
-  <metadata_license>CC0-1.0</metadata_license>
-  <project_license>GPL-2.0+</project_license>
   <name>Frescobaldi</name>
   <summary>LilyPond Music Editor</summary>
-
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
   <description>
     <p>
       Frescobaldi is an advanced text editor to edit LilyPond sheet music files.
@@ -34,139 +33,51 @@
   </description>
-
+  <url type="homepage">http://frescobaldi.org/</url>
+  <launchable type="desktop-id">org.frescobaldi.Frescobaldi.desktop</launchable>
+  <translation type="gettext">frescobaldi</translation>
   <screenshots>
     <screenshot type="default">
       <caption>Text editor and music view</caption>
-      <image>https://user-images.githubusercontent.com/1644647/69150986-10d89880-0ada-11ea-8e40-ed3f7a1ea00b.png</image>
+      <image type="source">https://user-images.githubusercontent.com/1644647/69150986-10d89880-0ada-11ea-8e40-ed3f7a1ea00b.png</image>
     </screenshot>
     <screenshot>
       <caption>Change the input without leaving the music view</caption>
-      <image>https://user-images.githubusercontent.com/1644647/69151435-efc47780-0ada-11ea-917b-580abf4e8a99.png</image>
+      <image type="source">https://user-images.githubusercontent.com/1644647/69151435-efc47780-0ada-11ea-917b-580abf4e8a99.png</image>
     </screenshot>
     <screenshot>
       <caption>Zoom with the magnifier glass</caption>
-      <image>https://user-images.githubusercontent.com/1644647/69188369-b280d980-0b1c-11ea-8f0e-e20b0f9f463d.png</image>
+      <image type="source">https://user-images.githubusercontent.com/1644647/69188369-b280d980-0b1c-11ea-8f0e-e20b0f9f463d.png</image>
     </screenshot>
     <screenshot>
       <caption>Syntax highlighting and automatic completion</caption>
-      <image>https://user-images.githubusercontent.com/1644647/69151460-f7841c00-0ada-11ea-9577-1d2d358770ef.png</image>
+      <image type="source">https://user-images.githubusercontent.com/1644647/69151460-f7841c00-0ada-11ea-9577-1d2d358770ef.png</image>
     </screenshot>
     <screenshot>
       <caption>Snippet manager and editor</caption>
-      <image>https://user-images.githubusercontent.com/1644647/69151482-023eb100-0adb-11ea-9221-191052d5c92d.png</image>
+      <image type="source">https://user-images.githubusercontent.com/1644647/69151482-023eb100-0adb-11ea-9221-191052d5c92d.png</image>
     </screenshot>
     <screenshot>
       <caption>Score Wizard to quickly setup a music score</caption>
-      <image>https://user-images.githubusercontent.com/1644647/69151502-08cd2880-0adb-11ea-9c17-92886ef0a846.png</image>
+      <image type="source">https://user-images.githubusercontent.com/1644647/69151502-08cd2880-0adb-11ea-9c17-92886ef0a846.png</image>
     </screenshot>
   </screenshots>
-
-  <url type="homepage">http://frescobaldi.org/</url>
-
-  <launchable type="desktop-id">org.frescobaldi.Frescobaldi.desktop</launchable>
-
-  <translation type="gettext">frescobaldi</translation>
-
   <releases>
-    <release version="3.1.2" date="2020-04-13">
-      <description>
-        <p>Bug fixes:</p>
-        <ul>
-          <li>Fixed the error "AttributeError: 'NoneType' object has no attribute 'cursor'"</li>
-          <li>Fixed the "Two Pages (first page right)" mode</li>
-          <li>Fixed pinch gesture zooming in Music View on Mac OS X</li>
-          <li>Fixed printing the music score on Mac OS X</li>
-          <li>Fixed cursor nagivation keyboard shortcuts on Mac OS X; new shortcuts:</li>
-          <ul>
-            <li>next document: ctrl+tab</li>
-            <li>previous document: ctrl+shift+tab</li>
-            <li>start of line: cmd+left</li>
-            <li>end of line: cmd+right</li>
-          </ul>
-          <li>Fixed the display of filenames in document tabs on Mac OS X</li>
-          <li>Fixed the global menu with no windows open on Mac OS X</li>
-          <li>Fixed the error with convert-ly on in the Mac app bundle</li>
-        </ul>
-        <p>Improvements:</p>
-        <ul>
-          <li>German hyphenation patterns were added</li>
-        </ul>
-        <p>Translations:</p>
-        <ul>
-          <li>Dutch translation updated by Wilbert Berendsen</li>
-          <li>Italian translation updated by Davide Liessi</li>
-        </ul>
-      </description>
-    </release>
-    <release version="3.1.1" date="2020-01-04">
+    <release type="stable" version="3.1.2" date="2020-04-13T00:00:00Z">
       <description>
-        <p>Bugfixes:</p>
-        <ul>
-          <li>Fixed the error AttributeError: 'PopplerDocument' has no attribute 'ispresent'.</li>
-          <li>Fixed the error NameError: 'QPinchGesture' is not defined (pinch in music view).</li>
-          <li>Fixed the error NameError: 'doc' is not defined (when printing on Mac OSX).</li>
-        </ul>
-      </description>
-    </release>
-    <release version="3.1.0" date="2019-12-27">
-      <description>
-        <p>Requirement changes:</p>
-        <ul>
-          <li>Frescobaldi now requires Python3.3+</li>
-        </ul>
-        <p>New features:</p>
-        <ul>
-          <li>New "Document Fonts" dialog supporting text and music fonts and providing a font sample previewer.</li>
-          <li>Possibility to load external extensions.</li>
-          <li>New "First System Only" option in Custom Engrave.</li>
-          <li>New "Goto Line" and "Rename file" commands.</li>
-          <li>Music View: view settings are remembered per-document.</li>
-          <li>Music View: copy to Image can copy/export to SVG, PDF and EPS in addition to PNG/JPG.</li>
-          <li>Music View: new commands to rotate left/right; new toolbar button to show/hide the magnifier.</li>
-          <li>Music View: new preferences Horizontal/Vertical and Continuous/non-continuous.</li>
-          <li>Music View: new raster layout mode (displays as many pages as fit in a View).</li>
-          <li>Manuscript viewer: new toolbar buttons to rotate the pages left/right and to show/hide the magnifier.</li>
-        </ul>
-        <p>Bug fixes:</p>
-        <ul>
-          <li>Fixed seeking in MIDI player during playing stops sound.</li>
-          <li>Paper orientation is now properly handled in New Score Wizard.</li>
-          <li>Fixed discrepancy of LilyPond vs. system version of GhostScript
-    on Linux.</li>
-          <li>Fixed includepaths on Windows.</li>
-          <li>Fixed the error NameError: name 'widgets' is not defined.</li>
-        </ul>
-        <p>Improvements:</p>
-        <ul>
-          <li>Score Wizard: Titles and Headers are now shown in preview; new checkbox to write/omit pitches after \relative command; allow "none" for instrument names on first system.</li>
-          <li>Smarter behaviour of the autocompletion popup (#918, #922).</li>
-          <li>New command File->Rename/Move.</li>
-          <li>Sessions can be grouped in the Sessions menu.</li>
-          <li>Show absolute path of include files in tooltip.</li>
-          <li>Restructure Tools Menu.</li>
-          <li>File Open toolbar button shows recent files menu on long click.</li>
-          <li>Added "Blank Sheet Music" template snippet.</li>
-        </ul>
-      </description>
-    </release>
-    <release version="3.0.0" date="2017-02-17">
-      <description>
-        <p>Requirement changes:</p>
-        <ul>
-          <li>Frescobaldi now requires Python3.2+, Qt5, PyQt5, python-poppler-qt5.</li>
-        </ul>
-        <p>New features:</p>
+        <p>This release adds the following feature:</p>
         <ul>
-          <li>Zoom with pinch gesture in Music View.</li>
-          <li>An option (enabled by default) to move the cursor to the end of the line when PageDown is pressed on the last line, and to move the cursor to the start of the first line if PageUp is pressed on the first line.</li>
+          <li>Added a german hyphenation dictionary.</li>
         </ul>
-        <p>Improvements:</p>
+        <p>This release fixes the following bugs:</p>
         <ul>
-          <li>Retina display support in Music View.</li>
+          <li>Improved Python selection on Mac, required to run without errors the LilyPond bundled scripts, such as convert-ly.</li>
+          <li>Fixed printing score on Mac.</li>
+          <li>Fixed pinch gesture zoom in music view on Mac.</li>
+          <li>Changed some keyboard shortcuts on Mac: Cmd+left and Cmd+right will now move the cursor to the beginning and to the end of the line, respectively; new shortcuts to navigate between documents are Ctrl+tab (next document) and Ctrl+Shift+tab</li>
+          <li>File names in document tabs are now shown correctly on Mac.</li>
         </ul>
+        <p>This release updates translations.</p>
       </description>
     </release>
   </releases>
-
-  <content_rating type="oars-1.1" />
-
+  <content_rating type="oars-1.1"/>
 </component>
