diff --git a/scripts/packaging/generate_projects_files.py b/scripts/packaging/generate_projects_files.py
index e065463..738ef81 100755
--- a/scripts/packaging/generate_projects_files.py
+++ b/scripts/packaging/generate_projects_files.py
@@ -95,12 +95,6 @@ class ProjectFilesGenerator:
 
         with open(os.path.join(script_dir, "exist_in_projects.txt")) as f:
             self.exist_in_projects = f.read().splitlines()
-        self.exist_in_projects_platform_path = "exist_in_projects"
-        if sys.platform == 'linux':
-            self.exist_in_projects_platform_path += "_" + self.platform + '_' + distro.version()
-        self.exist_in_projects_platform_path += ".txt"
-        with open(os.path.join(script_dir, self.exist_in_projects_platform_path)) as f:
-            self.exist_in_projects += f.read().splitlines()
         self.exist_in_projects = [os.path.join(WEBOTS_HOME, line) for line in self.exist_in_projects]
         valid_environment = self.check_exist_in_projects(self.exist_in_projects) & valid_environment
 
