--- Mnemosyne-2.10.1/mnemosyne/libmnemosyne/gui_translators/gettext_gui_translator.py	2022-12-02 14:00:58.000000000 +0100
+++ mnemosyne/libmnemosyne/gui_translators/gettext_gui_translator.py	2023-05-27 17:04:27.950235592 +0200
@@ -17,7 +17,7 @@
         if os.path.exists("mo"):
             self.lang_path = "mo"
         else:  # pragma: no cover
-            self.lang_path = os.path.join(sys.exec_prefix, "share", "locale")
+            self.lang_path = os.path.join("/app", "share", "locale")
 
     def supported_languages(self):
         try:
