diff -ru gnome-inform7-6M62/com.inform7.GUI.gschema.xml gnome-inform7-6M62-patched/com.inform7.GUI.gschema.xml
--- gnome-inform7-6M62/com.inform7.GUI.gschema.xml	2013-09-11 02:19:32.000000000 -0400
+++ gnome-inform7-6M62-patched/com.inform7.GUI.gschema.xml	2019-01-14 18:29:10.202971332 -0500
@@ -1,30 +1,30 @@
 <schemalist>
 
-  <enum id="com.inform7.GUI.preferences.font-set-enum">
+  <enum id="com.inform7.IDE.preferences.font-set-enum">
     <value nick="Standard" value="0" />
     <value nick="Monospace" value="1" />
     <value nick="Custom" value="2" />
   </enum>
 
-  <enum id="com.inform7.GUI.preferences.font-size-enum">
+  <enum id="com.inform7.IDE.preferences.font-size-enum">
     <value nick="Standard" value="0" />
     <value nick="Medium" value="1" />
     <value nick="Large" value="2" />
     <value nick="Huge" value="3" />
   </enum>
 
-  <enum id="com.inform7.GUI.preferences.interpreter-enum">
+  <enum id="com.inform7.IDE.preferences.interpreter-enum">
     <value nick="Glulxe (default)" value="0" />
     <value nick="Git" value="1" />
   </enum>
 
-  <schema id="com.inform7.GUI" path="/com/inform7/GUI/">
-    <child name="preferences" schema="com.inform7.GUI.preferences"/>
-    <child name="state" schema="com.inform7.GUI.state"/>
+  <schema id="com.inform7.IDE" path="/com/inform7/IDE/">
+    <child name="preferences" schema="com.inform7.IDE.preferences"/>
+    <child name="state" schema="com.inform7.IDE.state"/>
   </schema>
 
-  <schema id="com.inform7.GUI.preferences" path="/com/inform7/GUI/preferences/">
-    <child name="skein" schema="com.inform7.GUI.preferences.skein"/>
+  <schema id="com.inform7.IDE.preferences" path="/com/inform7/IDE/preferences/">
+    <child name="skein" schema="com.inform7.IDE.preferences.skein"/>
 
     <key name="author-name" type="s">
       <default>''</default>
@@ -35,7 +35,7 @@
       created the project.</description>
     </key>
 
-    <key name="font-set" enum="com.inform7.GUI.preferences.font-set-enum">
+    <key name="font-set" enum="com.inform7.IDE.preferences.font-set-enum">
       <default>'Standard'</default>
       <summary>Font set to use in the Source panel</summary>
       <description>Font set to use in the Source panel. Allowed values are
@@ -52,7 +52,7 @@
       "Custom".</description>
     </key>
 
-    <key name="font-size" enum="com.inform7.GUI.preferences.font-size-enum">
+    <key name="font-size" enum="com.inform7.IDE.preferences.font-size-enum">
       <default>'Standard'</default>
       <summary>Font size to use in the application</summary>
       <description>Font size to use in the application. Allowed values are
@@ -126,7 +126,7 @@
       and renumber the previous Part 2 to Part 3 and so on.</description>
     </key>
 
-    <key name="interpreter" enum="com.inform7.GUI.preferences.interpreter-enum">
+    <key name="interpreter" enum="com.inform7.IDE.preferences.interpreter-enum">
       <default>'Glulxe (default)'</default>
       <summary>Use the Git interpreter for Glulx story files</summary>
       <description>This setting determines the interpreter to use in the Game
@@ -167,7 +167,7 @@
 
   </schema>
 
-  <schema id="com.inform7.GUI.state" path="/com/inform7/GUI/state/">
+  <schema id="com.inform7.IDE.state" path="/com/inform7/IDE/state/">
 
     <key name="spell-check" type="b">
       <default>true</default>
@@ -243,8 +243,8 @@
 
   </schema>
 
-  <schema id="com.inform7.GUI.preferences.skein"
-    path="/com/inform7/GUI/preferences/skein/">
+  <schema id="com.inform7.IDE.preferences.skein"
+    path="/com/inform7/IDE/preferences/skein/">
 
     <key name="horizontal-spacing" type="d">
       <default>40</default>
diff -ru gnome-inform7-6M62/data/ui/document.ui gnome-inform7-6M62-patched/data/ui/document.ui
--- gnome-inform7-6M62/data/ui/document.ui	2015-09-21 01:12:27.000000000 -0400
+++ gnome-inform7-6M62-patched/data/ui/document.ui	2019-01-14 18:30:19.155980667 -0500
@@ -398,7 +398,7 @@
     <property name="border_width">12</property>
     <property name="title" translatable="yes" comments="Edit Menu">Find and Replace</property>
     <property name="window_position">center-on-parent</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="type_hint">dialog</property>
     <signal name="close" handler="gtk_widget_hide" swapped="no"/>
     <signal name="delete-event" handler="gtk_widget_hide_on_delete" swapped="no"/>
@@ -677,7 +677,7 @@
     <property name="modal">True</property>
     <property name="default_width">250</property>
     <property name="destroy_with_parent">True</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="type_hint">dialog</property>
     <property name="skip_taskbar_hint">True</property>
     <property name="deletable">False</property>
@@ -764,7 +764,7 @@
     <property name="border_width">12</property>
     <property name="title" translatable="yes" comments="Search Files dialog">Find and Replace</property>
     <property name="window_position">center-on-parent</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="type_hint">dialog</property>
     <signal name="close" handler="gtk_widget_hide" swapped="no"/>
     <signal name="delete-event" handler="gtk_widget_hide_on_delete" swapped="no"/>
diff -ru gnome-inform7-6M62/data/ui/gnome-inform7.ui gnome-inform7-6M62-patched/data/ui/gnome-inform7.ui
--- gnome-inform7-6M62/data/ui/gnome-inform7.ui	2014-08-31 17:10:34.000000000 -0400
+++ gnome-inform7-6M62-patched/data/ui/gnome-inform7.ui	2019-01-14 18:31:45.329998516 -0500
@@ -205,7 +205,7 @@
     <property name="border_width">5</property>
     <property name="title" translatable="yes" comments="Settings panel">Inform Preferences</property>
     <property name="window_position">center</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="type_hint">dialog</property>
     <signal name="close" handler="gtk_widget_hide" swapped="no"/>
     <signal name="delete-event" handler="gtk_widget_hide_on_delete" swapped="no"/>
diff -ru gnome-inform7-6M62/data/ui/newdialog.ui gnome-inform7-6M62-patched/data/ui/newdialog.ui
--- gnome-inform7-6M62/data/ui/newdialog.ui	2014-04-28 02:00:44.000000000 -0400
+++ gnome-inform7-6M62-patched/data/ui/newdialog.ui	2019-01-14 18:30:37.592983161 -0500
@@ -18,7 +18,7 @@
     <property name="border_width">12</property>
     <property name="title" translatable="yes" comments="New Project window">Create New Project</property>
     <property name="window_position">center</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <signal name="close" handler="on_newdialog_close" swapped="no"/>
     <signal name="delete-event" handler="on_newdialog_delete_event" swapped="no"/>
     <signal name="prepare" handler="on_newdialog_prepare" swapped="no"/>
diff -ru gnome-inform7-6M62/data/ui/story.ui gnome-inform7-6M62-patched/data/ui/story.ui
--- gnome-inform7-6M62/data/ui/story.ui	2014-08-31 17:10:34.000000000 -0400
+++ gnome-inform7-6M62-patched/data/ui/story.ui	2019-01-14 18:31:07.483987206 -0500
@@ -421,7 +421,7 @@
     <property name="title" translatable="yes" comments="Notepad window">Notepad</property>
     <property name="default_width">300</property>
     <property name="default_height">300</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="type_hint">utility</property>
     <property name="skip_taskbar_hint">True</property>
     <property name="skip_pager_hint">True</property>
@@ -446,7 +446,7 @@
     <property name="can_focus">False</property>
     <property name="border_width">12</property>
     <property name="title" translatable="yes" comments="Skein spacing options dialog">Skein spacing options</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="type_hint">dialog</property>
     <child internal-child="vbox">
       <object class="GtkVBox" id="dialog-vbox1">
@@ -616,7 +616,7 @@
     <property name="can_focus">False</property>
     <property name="border_width">12</property>
     <property name="title" translatable="yes" comments="Skein trim dialog">Trim the skein</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="type_hint">dialog</property>
     <child internal-child="vbox">
       <object class="GtkVBox" id="dialog-vbox2">
diff -ru gnome-inform7-6M62/data/ui/welcomedialog.ui gnome-inform7-6M62-patched/data/ui/welcomedialog.ui
--- gnome-inform7-6M62/data/ui/welcomedialog.ui	2014-08-31 17:10:34.000000000 -0400
+++ gnome-inform7-6M62-patched/data/ui/welcomedialog.ui	2019-01-14 18:31:16.100989752 -0500
@@ -24,7 +24,7 @@
     <property name="title" translatable="yes" comments="Splash window">Welcome to Inform 7</property>
     <property name="resizable">False</property>
     <property name="window_position">center-always</property>
-    <property name="icon_name">inform7</property>
+    <property name="icon_name">com.inform7.IDE</property>
     <property name="urgency_hint">True</property>
     <signal name="delete-event" handler="gtk_main_quit" swapped="no"/>
     <child>
diff -ru gnome-inform7-6M62/gnome-inform7.desktop.in gnome-inform7-6M62-patched/gnome-inform7.desktop.in
--- gnome-inform7-6M62/gnome-inform7.desktop.in	2015-10-18 00:30:18.000000000 -0400
+++ gnome-inform7-6M62-patched/gnome-inform7.desktop.in	2019-01-14 18:42:08.849185412 -0500
@@ -4,6 +4,6 @@
 Name=Inform 7
 Exec=gnome-inform7 %F
 Comment=Write interactive fiction
-Icon=inform7
+Icon=com.inform7.IDE
 Terminal=false
 Categories=Development;IDE;GNOME;GTK;
diff -ru gnome-inform7-6M62/Makefile.am gnome-inform7-6M62-patched/Makefile.am
--- gnome-inform7-6M62/Makefile.am	2015-10-18 00:33:45.000000000 -0400
+++ gnome-inform7-6M62-patched/Makefile.am	2019-01-14 18:41:08.799167415 -0500
@@ -11,7 +11,7 @@
 	$(AM_V_GEN)$(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
 CLEANFILES = $(desktop_DATA)
 
-gsettings_SCHEMAS = com.inform7.GUI.gschema.xml
+gsettings_SCHEMAS = com.inform7.IDE.gschema.xml
 
 @GSETTINGS_RULES@
 
diff -ru gnome-inform7-6M62/Makefile.in gnome-inform7-6M62-patched/Makefile.in
--- gnome-inform7-6M62/Makefile.in	2016-01-10 14:12:40.000000000 -0500
+++ gnome-inform7-6M62-patched/Makefile.in	2019-01-14 18:41:08.800167415 -0500
@@ -120,11 +120,11 @@
 AM_V_GEN = $(am__v_GEN_@AM_V@)
 am__v_GEN_ = $(am__v_GEN_@AM_DEFAULT_V@)
 am__v_GEN_0 = @echo "  GEN     " $@;
-am__v_GEN_1 = 
+am__v_GEN_1 =
 AM_V_at = $(am__v_at_@AM_V@)
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
-am__v_at_1 = 
+am__v_at_1 =
 SOURCES =
 DIST_SOURCES =
 RECURSIVE_TARGETS = all-recursive check-recursive cscopelist-recursive \
@@ -425,7 +425,7 @@
 desktopdir = $(datadir)/applications
 desktop_DATA = gnome-inform7.desktop
 CLEANFILES = $(desktop_DATA)
-gsettings_SCHEMAS = com.inform7.GUI.gschema.xml
+gsettings_SCHEMAS = com.inform7.IDE.gschema.xml
 dist_doc_DATA = AUTHORS.md ChangeLog COPYING README.md
 EXTRA_DIST = autogen.sh $(gsettings_SCHEMAS) gnome-inform7.desktop.in \
 	$(NULL) $(am__append_1)
@@ -490,7 +490,7 @@
 stamp-h1: $(srcdir)/config.h.in $(top_builddir)/config.status
 	@rm -f stamp-h1
 	cd $(top_builddir) && $(SHELL) ./config.status config.h
-$(srcdir)/config.h.in:  $(am__configure_deps) 
+$(srcdir)/config.h.in:  $(am__configure_deps)
 	($(am__cd) $(top_srcdir) && $(AUTOHEADER))
 	rm -f stamp-h1
 	touch $@
diff -ru gnome-inform7-6M62/pixmaps/Makefile.am gnome-inform7-6M62-patched/pixmaps/Makefile.am
--- gnome-inform7-6M62/pixmaps/Makefile.am	2014-08-31 17:10:34.000000000 -0400
+++ gnome-inform7-6M62-patched/pixmaps/Makefile.am	2019-01-14 18:32:24.870010369 -0500
@@ -2,7 +2,7 @@
 
 # Named Icons
 # -----------
-# inform7.png: named application icon (inform.iconset)
+# com.inform7.IDE.png: named application icon (inform.iconset)
 # inform7-builtin.png: named icon for built-in extension
 # inform7-materials.png: named emblem for materials folder
 #
@@ -17,12 +17,12 @@
 # text/x-blurb: Blurb file (blurbfile.iconset)
 
 nobase_dist_icons_DATA = \
-	16x16/apps/inform7.png \
-	32x32/apps/inform7.png \
-	48x48/apps/inform7.png \
-	64x64/apps/inform7.png \
-	128x128/apps/inform7.png \
-	256x256/apps/inform7.png \
+	16x16/apps/com.inform7.IDE.png \
+	32x32/apps/com.inform7.IDE.png \
+	48x48/apps/com.inform7.IDE.png \
+	64x64/apps/com.inform7.IDE.png \
+	128x128/apps/com.inform7.IDE.png \
+	256x256/apps/com.inform7.IDE.png \
 	16x16/actions/inform7-builtin.png \
 	32x32/actions/inform7-builtin.png \
 	48x48/actions/inform7-builtin.png \
diff -ru gnome-inform7-6M62/pixmaps/Makefile.in gnome-inform7-6M62-patched/pixmaps/Makefile.in
--- gnome-inform7-6M62/pixmaps/Makefile.in	2016-01-10 14:12:40.000000000 -0500
+++ gnome-inform7-6M62-patched/pixmaps/Makefile.in	2019-01-14 18:32:25.181010462 -0500
@@ -337,7 +337,7 @@
 
 # Named Icons
 # -----------
-# inform7.png: named application icon (inform.iconset)
+# com.inform7.IDE.png: named application icon (inform.iconset)
 # inform7-builtin.png: named icon for built-in extension
 # inform7-materials.png: named emblem for materials folder
 #
@@ -351,12 +351,12 @@
 # text/x-natural-inform-extension: Extension file (i7xfile.iconset)
 # text/x-blurb: Blurb file (blurbfile.iconset)
 nobase_dist_icons_DATA = \
-	16x16/apps/inform7.png \
-	32x32/apps/inform7.png \
-	48x48/apps/inform7.png \
-	64x64/apps/inform7.png \
-	128x128/apps/inform7.png \
-	256x256/apps/inform7.png \
+	16x16/apps/com.inform7.IDE.png \
+	32x32/apps/com.inform7.IDE.png \
+	48x48/apps/com.inform7.IDE.png \
+	64x64/apps/com.inform7.IDE.png \
+	128x128/apps/com.inform7.IDE.png \
+	256x256/apps/com.inform7.IDE.png \
 	16x16/actions/inform7-builtin.png \
 	32x32/actions/inform7-builtin.png \
 	48x48/actions/inform7-builtin.png \
diff -ru gnome-inform7-6M62/README.md gnome-inform7-6M62-patched/README.md
--- gnome-inform7-6M62/README.md	2015-10-18 00:30:18.000000000 -0400
+++ gnome-inform7-6M62-patched/README.md	2019-01-14 18:32:41.655015402 -0500
@@ -1,4 +1,4 @@
-[![Build Status](https://travis-ci.org/ptomato/gnome-inform7.png?branch=master)](https://travis-ci.org/ptomato/gnome-inform7)
+[![Build Status](https://travis-ci.org/ptomato/inform7.png?branch=master)](https://travis-ci.org/ptomato/gnome-inform7)
 
 This version of Inform 7 for the Gnome platform was created by Philip
 Chimento, based on a design by Graham Nelson and Andrew Hunter.
diff -ru gnome-inform7-6M62/src/actions.c gnome-inform7-6M62-patched/src/actions.c
--- gnome-inform7-6M62/src/actions.c	2016-01-10 14:27:31.000000000 -0500
+++ gnome-inform7-6M62-patched/src/actions.c	2019-01-14 19:32:50.205706958 -0500
@@ -134,7 +134,7 @@
 action_import_into_skein(GtkAction *action, I7Story *story)
 {
 	GError *err = NULL;
-	
+
 	/* Ask the user for a file to import */
 	/* TRANSLATORS: File->Import Into Skein... */
 	GtkWidget *dialog = gtk_file_chooser_dialog_new(
@@ -157,7 +157,7 @@
 		gtk_widget_destroy(dialog);
 		return;
 	}
-	
+
 	GFile *file = gtk_file_chooser_get_file(GTK_FILE_CHOOSER(dialog));
 	gtk_widget_destroy(dialog);
 	if(!file)
@@ -168,7 +168,7 @@
 		error_dialog_file_operation(GTK_WINDOW(story), file, err, I7_FILE_ERROR_OPEN, NULL);
 	else
 		i7_story_show_pane(story, I7_PANE_SKEIN);
-	
+
 	g_object_unref(file);
 }
 
@@ -1194,7 +1194,7 @@
 void
 action_help_recipe_book(GtkAction *action, I7Story *story)
 {
-	GFile *file = i7_app_get_data_file_va(i7_app_get(), "Documentation", "Rindex.html", NULL);
+	GFile *file = i7_app_get_data_file_va(i7_app_get(), "Documentation", "Rallegs.html", NULL);
 	i7_story_show_docpage(story, file);
 	g_object_unref(file);
 }
@@ -1326,7 +1326,7 @@
 		"license", _("See Help\u2192License for licensing information."),
 		"authors", authors,
 		"translator-credits", translator_credits,
-		"logo-icon-name", "inform7",
+		"logo-icon-name", "com.inform7.IDE",
 		"title", _("About Inform"),
 		NULL);
 	g_free(copyright);
diff -ru gnome-inform7-6M62/src/configfile.h gnome-inform7-6M62-patched/src/configfile.h
--- gnome-inform7-6M62/src/configfile.h	2013-09-11 02:19:32.000000000 -0400
+++ gnome-inform7-6M62-patched/src/configfile.h	2019-01-14 18:36:06.729076882 -0500
@@ -55,9 +55,9 @@
 #define DEFAULT_TAB_WIDTH 8
 
 /* Schemas */
-#define SCHEMA_PREFERENCES "com.inform7.GUI.preferences"
-#define SCHEMA_SKEIN "com.inform7.GUI.preferences.skein"
-#define SCHEMA_STATE "com.inform7.GUI.state"
+#define SCHEMA_PREFERENCES "com.inform7.IDE.preferences"
+#define SCHEMA_SKEIN "com.inform7.IDE.preferences.skein"
+#define SCHEMA_STATE "com.inform7.IDE.state"
 
 /* Keys */
 #define PREFS_AUTHOR_NAME          "author-name"
diff -ru gnome-inform7-6M62/src/document.c gnome-inform7-6M62-patched/src/document.c
--- gnome-inform7-6M62/src/document.c	2015-10-18 00:30:18.000000000 -0400
+++ gnome-inform7-6M62-patched/src/document.c	2019-01-14 18:33:47.165035045 -0500
@@ -90,7 +90,7 @@
 	GSettings *state = i7_app_get_state(theapp);
 
 	/* Set the icon */
-	gtk_window_set_icon_name(GTK_WINDOW(self), "inform7");
+	gtk_window_set_icon_name(GTK_WINDOW(self), "com.inform7.IDE");
 
 	/* Set the minimum size so that the window can be sized smaller than the
 	 widgets inside it */
@@ -126,7 +126,7 @@
 	priv->document_action_group = GTK_ACTION_GROUP(load_object(builder, "document_actions"));
 	priv->selection_action_group = GTK_ACTION_GROUP(load_object(builder, "selection_actions"));
 	priv->copy_action_group = GTK_ACTION_GROUP(load_object(builder, "copy_actions"));
-	
+
 	self->ui_manager = gtk_ui_manager_new();
 	i7_app_insert_action_groups(theapp, self->ui_manager);
 	gtk_ui_manager_insert_action_group(self->ui_manager, priv->document_action_group, 0);
diff -ru gnome-inform7-6M62/src/newdialog.c gnome-inform7-6M62-patched/src/newdialog.c
--- gnome-inform7-6M62/src/newdialog.c	2013-09-10 02:14:06.000000000 -0400
+++ gnome-inform7-6M62-patched/src/newdialog.c	2019-01-14 18:37:51.325108240 -0500
@@ -210,7 +210,7 @@
 			break;
 	}
 	GdkPixbuf *icon = gtk_icon_theme_load_icon(gtk_icon_theme_get_default(),
-		"inform7", 128, 0, NULL); /* just set NULL on failure */
+		"com.inform7.IDE", 128, 0, NULL); /* just set NULL on failure */
 	gtk_assistant_set_page_side_image(assistant, page, icon);
 	if(icon)
 		g_object_unref(icon);
diff -ru gnome-inform7-6M62/src/searchwindow.c gnome-inform7-6M62-patched/src/searchwindow.c
--- gnome-inform7-6M62/src/searchwindow.c	2015-11-15 20:15:39.000000000 -0500
+++ gnome-inform7-6M62-patched/src/searchwindow.c	2019-01-14 18:37:06.159094698 -0500
@@ -294,7 +294,7 @@
 	priv->algorithm = I7_SEARCH_CONTAINS;
 
 	gtk_window_set_destroy_with_parent(GTK_WINDOW(self), TRUE);
-	gtk_window_set_icon_name(GTK_WINDOW(self), "inform7");
+	gtk_window_set_icon_name(GTK_WINDOW(self), "com.inform7.IDE");
 	gtk_window_set_skip_taskbar_hint(GTK_WINDOW(self), TRUE);
 	gtk_window_set_title(GTK_WINDOW(self), _("Search Results"));
 	gtk_window_set_type_hint(GTK_WINDOW(self), GDK_WINDOW_TYPE_HINT_UTILITY);
@@ -823,4 +823,4 @@
 	}
 	g_list_free(doc_index);
 	doc_index = NULL;
-}
\ No newline at end of file
+}
diff -ru gnome-inform7-6M62/src/skein.c gnome-inform7-6M62-patched/src/skein.c
--- gnome-inform7-6M62/src/skein.c	2015-12-12 03:29:11.000000000 -0500
+++ gnome-inform7-6M62-patched/src/skein.c	2019-01-14 18:36:33.367084867 -0500
@@ -151,7 +151,7 @@
 	gdk_color_parse("black", &priv->locked);
 	gdk_color_parse("black", &priv->unlocked);
 
-	priv->settings = g_settings_new("com.inform7.GUI.preferences.skein");
+	priv->settings = g_settings_new("com.inform7.IDE.preferences.skein");
 	g_settings_bind(priv->settings, "horizontal-spacing", self, "horizontal-spacing", G_SETTINGS_BIND_DEFAULT);
 	g_settings_bind(priv->settings, "vertical-spacing", self, "vertical-spacing", G_SETTINGS_BIND_DEFAULT);
 
@@ -359,12 +359,12 @@
 	I7Node *last = i7_skein_get_thread_bottom(self, priv->current);
 	int thread_depth = g_node_depth(last->gnode) - 1;
 	/* g_node_depth returns 0 for NULL and 1 for the root node */
-	
+
 	if(i > thread_depth)
 		return FALSE;
 
 	iter->stamp = priv->stamp;
-	
+
 	if(i == 0) {
 		iter->user_data = priv->root;
 		return TRUE;
@@ -373,13 +373,13 @@
 		iter->user_data = last;
 		return TRUE;
 	}
-	
+
 	/* Make the uninitialized iterator point to the i'th node between the root
 	 node and the bottom of the current node's thread */
 	GNode *gnode = last->gnode;
 	for( ; thread_depth > i; thread_depth--)
 		gnode = gnode->parent;
-	
+
 	iter->user_data = gnode->data;
 	return TRUE;
 }
@@ -389,7 +389,7 @@
 {
 	I7Skein *self = I7_SKEIN(model);
 	I7_SKEIN_USE_PRIVATE;
-	
+
 	g_return_val_if_fail(VALID_ITER(iter, priv), NULL);
 
 	GtkTreePath *path = gtk_tree_path_new();
@@ -403,7 +403,7 @@
 {
 	I7Skein *self = I7_SKEIN(model);
 	I7_SKEIN_USE_PRIVATE;
-	
+
 	g_return_if_fail(VALID_ITER(iter, priv));
 	g_return_if_fail(column >= 0 && column < I7_SKEIN_NUM_COLUMNS);
 
@@ -458,7 +458,7 @@
 {
 	I7Skein *self = I7_SKEIN(model);
 	I7_SKEIN_USE_PRIVATE;
-	
+
 	g_return_val_if_fail(VALID_ITER(iter, priv), FALSE);
 
 	/* Don't go beyond the bottom of "current" node's thread (end of the list) */
@@ -467,7 +467,7 @@
 		invalidate_iter(iter);
 		return FALSE;
 	}
-	
+
 	GNode *gnode = ((I7Node *)iter->user_data)->gnode;
 
 	for(gnode = gnode->children; gnode; gnode = gnode->next) {
@@ -509,7 +509,7 @@
 {
 	I7Skein *self = I7_SKEIN(model);
 	I7_SKEIN_USE_PRIVATE;
-	
+
 	/* If iter is NULL, return the number of toplevel nodes, i.e. the length of
 	 the list */
 	if(!iter) {
@@ -536,14 +536,14 @@
 	I7Node *last = i7_skein_get_thread_bottom(self, priv->current);
 	int thread_depth = g_node_depth(last->gnode) - 1;
 	/* g_node_depth returns 0 for NULL and 1 for the root node */
-	
+
 	if(n > thread_depth) {
 		invalidate_iter(iter);
 		return FALSE;
 	}
 
 	iter->stamp = priv->stamp;
-	
+
 	if(n == 0) {
 		iter->user_data = priv->root;
 		return TRUE;
@@ -552,13 +552,13 @@
 		iter->user_data = last;
 		return TRUE;
 	}
-	
+
 	/* Make the uninitialized iterator point to the i'th node between the root
 	 node and the bottom of the current node's thread */
 	GNode *gnode = last->gnode;
 	for( ; thread_depth > n; thread_depth--)
 		gnode = gnode->parent;
-	
+
 	iter->user_data = gnode->data;
 	return TRUE;
 }
@@ -631,7 +631,7 @@
 	/* New end of the list */
 	I7Node *last = i7_skein_get_thread_bottom(self, node);
 	GNode *gnode = old_last->gnode;
-	
+
 	/* Emit row-inserted and row-deleted signals on our tree model interface */
 	if(old_last != last) {
 		/* Find a common ancestor of the old and new list bottoms */
@@ -663,7 +663,7 @@
 			gtk_tree_path_free(path);
 		}
 	}
-	
+
 	priv->current = node;
 	g_object_notify(G_OBJECT(self), "current-node");
 	g_signal_emit_by_name(self, "needs-layout");
@@ -945,9 +945,9 @@
 {
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 	g_return_val_if_fail(file, FALSE);
-	
+
 	I7_SKEIN_USE_PRIVATE;
-	
+
 	I7Node *node = priv->root;
 	gboolean added = FALSE;
 
@@ -970,7 +970,7 @@
 				added = TRUE;
 			}
 			g_free(node_command);
-			node = newnode;	
+			node = newnode;
 		}
 		g_free(line);
 	}
@@ -1040,7 +1040,7 @@
 		}
 
 		gboolean in_current_thread = i7_skein_is_node_in_current_thread(self, node);
-		
+
 		if(i7_node_get_locked(node))
 			g_object_set(node->tree_item,
 				"stroke-color-rgba", rgba_from_gdk_color(&priv->locked),
@@ -1114,7 +1114,7 @@
 i7_skein_schedule_draw(I7Skein *self, GooCanvas *canvas)
 {
 	g_object_set_data(G_OBJECT(canvas), "waiting-for-draw", GINT_TO_POINTER(1));
-	
+
 	DrawData *draw_data = g_slice_new0(DrawData);
 	draw_data->skein = self;
 	draw_data->canvas = canvas;
@@ -1328,12 +1328,12 @@
 		i7_skein_set_played_node(self, priv->root);
 	if(i7_skein_is_node_in_current_thread(self, node))
 		i7_skein_set_current_node(self, priv->root);
-	
+
 	remove_all_from_model(self);
 	g_node_unlink(node->gnode);
 	g_node_traverse(node->gnode, G_POST_ORDER, G_TRAVERSE_ALL, -1, (GNodeTraverseFunc)remove_node_from_canvas, self);
 	reinstate_all_in_model(self);
-	
+
 	g_signal_emit_by_name(self, "needs-layout");
 	g_signal_emit_by_name(self, "modified");
 	return TRUE;
@@ -1364,7 +1364,7 @@
 	g_node_unlink(node->gnode);
 	remove_node_from_canvas(node->gnode, self);
 	reinstate_all_in_model(self);
-	
+
 	g_signal_emit_by_name(self, "needs-layout");
 	g_signal_emit_by_name(self, "modified");
 	return TRUE;
diff -ru gnome-inform7-6M62/src/story.c gnome-inform7-6M62-patched/src/story.c
--- gnome-inform7-6M62/src/story.c	2015-10-18 00:34:55.000000000 -0400
+++ gnome-inform7-6M62-patched/src/story.c	2019-01-14 18:38:29.935119815 -0500
@@ -709,7 +709,7 @@
 	gtk_text_view_set_buffer(GTK_TEXT_VIEW(panel->results_tabs[I7_RESULTS_TAB_INFORM6]), GTK_TEXT_BUFFER(priv->i6_source));
 	i7_skein_view_set_skein(I7_SKEIN_VIEW(panel->tabs[I7_PANE_SKEIN]), priv->skein);
 	gtk_tree_view_set_model(GTK_TREE_VIEW(panel->tabs[I7_PANE_TRANSCRIPT]), GTK_TREE_MODEL(priv->skein));
-	
+
 	/* Set the Results/Progress to a monospace font */
 	gtk_widget_modify_font(GTK_WIDGET(panel->results_tabs[I7_RESULTS_TAB_PROGRESS]), font);
 
@@ -824,7 +824,7 @@
 	priv->compiler_output_file = NULL;
 	priv->test_me = FALSE;
 	priv->manifest = NULL;
-	
+
 	/* Set up the Skein */
 	priv->skein = i7_skein_new();
 	g_object_set(priv->skein,
@@ -833,7 +833,7 @@
 	g_signal_connect(priv->skein, "node-activate", G_CALLBACK(on_node_activate), self);
 	g_signal_connect(priv->skein, "differs-badge-activate", G_CALLBACK(on_differs_badge_activate), self);
 	g_signal_connect(priv->skein, "modified", G_CALLBACK(on_skein_modified), self);
-	priv->skein_settings = g_settings_new("com.inform7.GUI.preferences.skein");
+	priv->skein_settings = g_settings_new("com.inform7.IDE.preferences.skein");
 	g_settings_bind(priv->skein_settings, "horizontal-spacing",
 		gtk_range_get_adjustment(GTK_RANGE(self->skein_spacing_horizontal)), "value", G_SETTINGS_BIND_DEFAULT);
 	g_settings_bind(priv->skein_settings, "vertical-spacing",
@@ -906,7 +906,7 @@
 	/* Make the Skein dialogs transient */
 	gtk_window_set_transient_for(GTK_WINDOW(self->skein_spacing_dialog), GTK_WINDOW(self));
 	gtk_window_set_transient_for(GTK_WINDOW(self->skein_trim_dialog), GTK_WINDOW(self));
-	
+
 	/* Create a callback for the delete event */
 	g_signal_connect(self, "delete-event", G_CALLBACK(on_storywindow_delete_event), NULL);
 }
