diff --git a/desktopApp/build.gradle.kts b/desktopApp/build.gradle.kts
index d684fafae..dea96d20c 100644
--- a/desktopApp/build.gradle.kts
+++ b/desktopApp/build.gradle.kts
@@ -94,7 +94,6 @@ sourceSets.main { kotlin.srcDir(generateBuildConfig.map { buildConfigOutputDir }
 kotlin {
     jvmToolchain {
         languageVersion.set(JavaLanguageVersion.of(21))
-        vendor.set(JvmVendorSpec.JETBRAINS)
     }
     compilerOptions {
         jvmTarget.set(JvmTarget.JVM_21)
