--- hugin-10/platforms/linux/appdata/hugin.appdata.xml	2014-03-19 14:33:21.000000000 -0400
+++ hugin-10/platforms/linux/appdata/hugin.appdata.xml-new	2020-06-13 23:36:59.516505541 -0400
@@ -23,12 +23,16 @@
   </ul>
  </description>
  <screenshots>
-  <screenshot type="default" width="800" height="600">http://hugin.sourceforge.net/screenshots/appdata/hugin-1.png</screenshot>
-  <screenshot width="800" height="600">http://hugin.sourceforge.net/screenshots/appdata/hugin-2.png</screenshot>
-  <screenshot width="800" height="600">http://hugin.sourceforge.net/screenshots/appdata/hugin-3.png</screenshot>
-  <screenshot width="800" height="600">http://hugin.sourceforge.net/screenshots/appdata/hugin-4.png</screenshot>
-  <screenshot width="800" height="600">http://hugin.sourceforge.net/screenshots/appdata/hugin-5.png</screenshot>
+  <screenshot type="default">http://hugin.sourceforge.net/screenshots/appdata/hugin-1.png</screenshot>
+  <screenshot>http://hugin.sourceforge.net/screenshots/appdata/hugin-2.png</screenshot>
+  <screenshot>http://hugin.sourceforge.net/screenshots/appdata/hugin-3.png</screenshot>
+  <screenshot>http://hugin.sourceforge.net/screenshots/appdata/hugin-4.png</screenshot>
+  <screenshot>http://hugin.sourceforge.net/screenshots/appdata/hugin-5.png</screenshot>
  </screenshots>
  <url type="homepage">http://hugin.sourceforge.net/</url>
  <updatecontact>tduell_at_iinet.net.au</updatecontact>
+ <content_rating type="oars-1.1"/>
+ <releases>
+   <release date="2019-12-01" version="2019.2.0" />
+ </releases>
 </application>
