Install icons, destop file, and AppStream data

diff --git a/Makefile.in b/Makefile.in
index 7dfbee1..5dd21e5 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -241,6 +241,12 @@ install-plugins:
 
 .PHONY: install-data
 install-data:
+	$(INSTALL_DATA) -D icons/ultrastardx-icon_512.png $(datarootdir)/icons/hicolor/512x512/apps/ultrastardx.png
+	$(INSTALL_DATA) -D icons/ultrastardx-icon_256.png $(datarootdir)/icons/hicolor/256x256/apps/ultrastardx.png
+	$(INSTALL_DATA) -D icons/ultrastardx-icon_32.png $(datarootdir)/icons/hicolor/32x32/apps/ultrastardx.png
+	$(INSTALL_DATA) -D icons/ultrastardx-icon.svg $(datarootdir)/icons/hicolor/scalable/apps/ultrastardx.svg
+	$(INSTALL_DATA) -D dists/ultrastardx.desktop $(datarootdir)/applications/ultrastardx.desktop
+	$(INSTALL_DATA) -D dists/ultrastardx.appdata.xml $(datarootdir)/metainfo/eu.usdx.UltraStarDeluxe.metainfo.xml
 	$(MAKE) RECURSIVE_SRC_DIR="$(USDX_GAME_DIR)/avatars" \
             RECURSIVE_DST_DIR="$(DESTDIR)$(INSTALL_DATADIR)/avatars" \
             install-data-recursive
