diff --git a/extensions/git/src/ipc/ipcServer.ts b/extensions/git/src/ipc/ipcServer.ts
index 5cea9faf98..0e4271162b 100644
--- a/extensions/git/src/ipc/ipcServer.ts
+++ b/extensions/git/src/ipc/ipcServer.ts
@@ -16,8 +16,14 @@ function getIPCHandlePath(id: string): string {
 		return `\\\\.\\pipe\\vscode-git-${id}-sock`;
 	}
 
-	if (process.env['XDG_RUNTIME_DIR']) {
-		return path.join(process.env['XDG_RUNTIME_DIR'] as string, `vscode-git-${id}.sock`);
+	const xdgRuntimeDir = process.env['XDG_RUNTIME_DIR'];
+	if (xdgRuntimeDir) {
+		const flatpakID = process.env['FLATPAK_ID'];
+		if (flatpakID) {
+			return path.join(xdgRuntimeDir, 'app', flatpakID, `vscode-git-${id}.sock`);
+		} else {
+			return path.join(xdgRuntimeDir, `vscode-git-${id}.sock`);
+		}
 	}
 
 	return path.join(os.tmpdir(), `vscode-git-${id}.sock`);
diff --git a/src/vs/platform/environment/node/environmentService.ts b/src/vs/platform/environment/node/environmentService.ts
index a67032aae3..d527616e5d 100644
--- a/src/vs/platform/environment/node/environmentService.ts
+++ b/src/vs/platform/environment/node/environmentService.ts
@@ -284,7 +284,13 @@ function getNixIPCHandle(userDataPath: string, type: string): string {
 	let result: string;
 	if (xdgRuntimeDir && !vscodePortable) {
 		const scope = crypto.createHash('md5').update(userDataPath).digest('hex').substr(0, 8);
-		result = path.join(xdgRuntimeDir, `vscode-${scope}-${product.version}-${type}.sock`);
+		const socketName = `vscode-${scope}-${product.version}-${type}.sock`
+		const flatpakID = process.env['FLATPAK_ID'];
+		if (flatpakID) {
+			result = path.join(xdgRuntimeDir, 'app', flatpakID, socketName);
+		} else {
+			result = path.join(xdgRuntimeDir, socketName);
+		}
 	} else {
 		result = path.join(userDataPath, `${product.version}-${type}.sock`);
 	}
