diff --git a/CMakeLists.txt b/CMakeLists.txt
index 635bcf6..9996510 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ FetchContent_MakeAvailable(juce)
 # If you are building a VST2 or AAX plugin, CMake needs to be told where to find these SDKs.
 # This setup should be done before calling `juce_add_plugin`.
 
-juce_set_vst2_sdk_path("${PROJECT_SOURCE_DIR}/vstsdk2.4")
+# juce_set_vst2_sdk_path("${PROJECT_SOURCE_DIR}/vstsdk2.4")
 # juce_set_aax_sdk_path(...)
 
 # `juce_add_plugin` adds a static library target with the name passed as the first argument.
@@ -57,7 +57,7 @@ juce_set_vst2_sdk_path("${PROJECT_SOURCE_DIR}/vstsdk2.4")
 # Check the readme at `docs/CMake API.md` in the JUCE repo for the full list.
 
 juce_add_plugin("${PROJECT_NAME}"
-	FORMATS AU VST VST3 LV2                 # The formats to build. Valid formats: Standalone Unity VST3 AU AUv3 AAX VST LV2.
+	FORMATS AU VST3 LV2                 # The formats to build. Valid formats: Standalone Unity VST3 AU AUv3 AAX VST LV2.
                                                 # AU and AUv3 plugins will only be enabled when building on macOS.
     PRODUCT_NAME "JC303"                        # The name of the final executable, which can differ from the target name.
     PLUGIN_NAME "JC303"                         # Name of the plugin that will be displayed in the DAW. Can differ from PRODUCT_NAME.
