diff --git a/modules/JUCE/modules/juce_gui_basics/windows/juce_ComponentPeer.h b/modules/JUCE/modules/juce_gui_basics/windows/juce_ComponentPeer.h
index 06c0a72..119f146 100644
--- a/modules/JUCE/modules/juce_gui_basics/windows/juce_ComponentPeer.h
+++ b/modules/JUCE/modules/juce_gui_basics/windows/juce_ComponentPeer.h
@@ -23,6 +23,8 @@
   ==============================================================================
 */
 
+#include <utility>
+
 namespace juce
 {
 
