diff --git a/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp b/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
index 59298c4cb..1a7ba04fb 100644
--- a/ThirdParty/JUCE/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
+++ b/ThirdParty/JUCE/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
@@ -3751,7 +3751,7 @@ FileSearchPath VST3PluginFormat::getDefaultLocationsToSearch()
    #elif JUCE_MAC
     return FileSearchPath ("/Library/Audio/Plug-Ins/VST3;~/Library/Audio/Plug-Ins/VST3");
    #else
-    return FileSearchPath ("/usr/lib/vst3/;/usr/local/lib/vst3/;~/.vst3/");
+    return FileSearchPath ("/app/extensions/Plugins/vst3/;~/.vst3/");
    #endif
 }
 
