diff --git a/JUCE/modules/juce_gui_basics/misc/juce_DropShadower.cpp b/JUCE/modules/juce_gui_basics/misc/juce_DropShadower.cpp
index cbd05cc..bbac1c5 100644
--- a/ThirdParty/JUCE/modules/juce_gui_basics/misc/juce_DropShadower.cpp
+++ b/ThirdParty/JUCE/modules/juce_gui_basics/misc/juce_DropShadower.cpp
@@ -23,6 +23,8 @@
   ==============================================================================
 */
 
+#include <utility>
+
 namespace juce
 {
 
