diff --git a/ThirdParty/JUCE/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp b/ThirdParty/JUCE/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
index 43ec447..8d3f75a 100644
--- a/ThirdParty/JUCE/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
+++ b/ThirdParty/JUCE/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
@@ -3944,7 +3944,7 @@ FileSearchPath VST3PluginFormat::getDefaultLocationsToSearch()
    #elif JUCE_MAC
     return FileSearchPath ("~/Library/Audio/Plug-Ins/VST3;/Library/Audio/Plug-Ins/VST3");
    #else
-    return FileSearchPath ("~/.vst3/;/usr/lib/vst3/;/usr/local/lib/vst3/");
+    return FileSearchPath ("~/.vst3/;/app/extensions/Plugins/vst3");
    #endif
 }
 
