diff --git a/sources/launchers/polychromatic.desktop b/sources/launchers/polychromatic.desktop
index cca893d..7cd1592 100644
--- a/sources/launchers/polychromatic.desktop
+++ b/sources/launchers/polychromatic.desktop
@@ -17,7 +17,7 @@
 Encoding=UTF-8
 Terminal=false
 Categories=Settings;HardwareSettings;
-Actions=devices;effects;tray-applet;
+Actions=devices;effects;
 StartupWMClass=polychromatic-controller
 #presets;triggers;
 
@@ -54,12 +54,3 @@
 #Name[nl]=Triggers bewerken
 #Icon=/usr/share/polychromatic/img/general/triggers.svg
 #Exec=polychromatic-controller --open triggers
-
-[Desktop Action tray-applet]
-Name=Tray Applet
-Name[de]=Tray-Anwendung
-Name[fr]=Applet de la zone de notification
-Name[nl]=Systeemvakpictogram
-Name[zh_CN]=托盘程序
-Icon=/usr/share/polychromatic/img/general/tray-applet.svg
-Exec=polychromatic-tray-applet
