
diff -uNrw -X .diff_ignore --no-dereference xtrkcad-fork-xtrkcad/app/lib/CMakeLists.txt new-fork-xtrkcad/app/lib/CMakeLists.txt
--- xtrkcad-fork-xtrkcad/app/lib/CMakeLists.txt	2025-12-20 09:20:00.607855181 -0500
+++ new-fork-xtrkcad/app/lib/CMakeLists.txt	2025-12-22 05:39:49.213081667 -0500
@@ -67,14 +67,25 @@
         )
 	install(FILES
 		xtrkcad${XTRKCAD_BETA}.desktop
-		DESTINATION "${XTRKCAD_SHARE_INSTALL_DIR}/applications"
+		DESTINATION "share/applications"
+		RENAME io.sourceforge.xtrkcad_fork.xtrkcad.desktop
 		)
 	install(FILES
 		xtrkcad.xml
-		DESTINATION "${XTRKCAD_SHARE_INSTALL_DIR}/applications"
+		DESTINATION "share/applications"
 		)
 	install(FILES
 		xtrkcad.png
 		DESTINATION "${XTRKCAD_SHARE_INSTALL_DIR}/pixmaps"
 		)
+	install(FILES
+		xtrkcad.metainfo.xml
+		DESTINATION "share/metainfo"
+		RENAME io.sourceforge.xtrkcad_fork.xtrkcad.metainfo.xml
+		)
+	install(FILES
+		xtrkcad_logo_128x128.svg
+		DESTINATION "share/icons/hicolor/scalable/apps"
+		RENAME io.sourceforge.xtrkcad_fork.xtrkcad.svg
+		)
 endif()
