diff -ur libaria2-1.33.1/src/a2io.h libaria2-1.33.1.patched/src/a2io.h
--- libaria2-1.33.1/src/a2io.h	2017-11-08 13:39:06.000000000 +0100
+++ libaria2-1.33.1.patched/src/a2io.h	2018-03-27 17:02:45.651414060 +0200
@@ -147,7 +147,7 @@
 // # define a2fseek(fp, offset, origin): No fseek64 and not used in aria2
 #define a2fstat(fd, buf) fstat64(fd, buf)
 // # define a2ftell(fd): No ftell64 and not used in aria2
-#define a2_struct_stat struct stat
+#define a2_struct_stat struct stat64
 #define a2stat(path, buf) stat64(path, buf)
 #define a2mkdir(path, openMode) mkdir(path, openMode)
 #define a2utimbuf utimbuf
diff -ur libaria2-1.33.1/src/crypto_endian.h libaria2-1.33.1.patched/src/crypto_endian.h
--- libaria2-1.33.1/src/crypto_endian.h	2017-11-08 13:39:06.000000000 +0100
+++ libaria2-1.33.1.patched/src/crypto_endian.h	2018-03-27 16:56:55.264371501 +0200
@@ -33,9 +33,13 @@
 #define BIG_ENDIAN 4321
 #define BYTE_ORDER LITTLE_ENDIAN
 #else // !  defined(_WIN32) || defined(__INTEL_COMPILER) || defined (_MSC_VER)
+#if defined(__ANDROID__)
+#include <endian.h>
+#else // defined(__ANDROID__)
 #ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
 #endif // HAVE_SYS_PARAM_H
+#endif // defined(__ANDROID__)
 #endif // !  defined(_WIN32) || defined(__INTEL_COMPILER) || defined (_MSC_VER)
 
 #if !defined(LITTLE_ENDIAN) || !defined(BIG_ENDIAN) || !defined(BYTE_ORDER) || \
