--- /configure.ac	2012-07-11 10:28:51.115114265 -0700
+++ /configure.ac	2016-04-09 17:00:02.114046934 -0700
@@ -48,29 +48,15 @@
 DBUSMENUGTK_REQUIRED_VERSION=0.5.90
 DBUS_GLIB_REQUIRED_VERSION=0.82
 
-AC_ARG_WITH([gtk],
-  [AS_HELP_STRING([--with-gtk],
-    [Which version of gtk to use @<:@default=2@:>@])],
-  [],
-  [with_gtk=2])
-AS_IF([test "x$with_gtk" = x3],
-        [PKG_CHECK_MODULES(LIBRARY,  gtk+-3.0 >= $GTK3_REQUIRED_VERSION
-                                     glib-2.0 >= $GLIB_REQUIRED_VERSION
-                                     gio-2.0 >= $GIO_REQUIRED_VERSION
-                                     indicator3-0.4 >= $INDICATOR_REQUIRED_VERSION
-                                     dbusmenu-gtk3-0.4 >= $DBUSMENUGTK_REQUIRED_VERSION)
-         AC_DEFINE(HAVE_GTK3, 1, [whether gtk3 is available])
-        ],
-      [test "x$with_gtk" = x2],
-        [PKG_CHECK_MODULES(LIBRARY,  gtk+-2.0 >= $GTK_REQUIRED_VERSION
-                                     glib-2.0 >= $GLIB_REQUIRED_VERSION
-                                     gio-2.0 >= $GIO_REQUIRED_VERSION
-                                     indicator-0.4 >= $INDICATOR_REQUIRED_VERSION
-                                     dbusmenu-gtk-0.4 >= $DBUSMENUGTK_REQUIRED_VERSION)
-        ],
-      [AC_MSG_FAILURE([Value for --with-gtk was neither 2 nor 3])]
-)
-AM_CONDITIONAL(USE_GTK3, [test "x$with_gtk" = x3])
+PKG_CHECK_MODULES(
+        LIBRARY, gtk+-3.0 >= $GTK3_REQUIRED_VERSION
+                            glib-2.0 >= $GLIB_REQUIRED_VERSION
+                            gio-2.0 >= $GIO_REQUIRED_VERSION
+                            indicator3-0.4 >= $INDICATOR_REQUIRED_VERSION
+                            dbusmenu-gtk3-0.4 >= $DBUSMENUGTK_REQUIRED_VERSION)
+
+AC_DEFINE(HAVE_GTK3, 1, [whether gtk3 is available])
+AM_CONDITIONAL(USE_GTK3, true)
 
 PKG_CHECK_MODULES(TESTDEPS, dbus-glib-1)
 
@@ -121,7 +107,7 @@
       RUNTIME=mono
    fi
 
-   AC_PATH_PROG(CSC, gmcs, no)
+   AC_PATH_PROG(CSC, mcs, no)
    LIB_PREFIX=.so
    LIB_SUFFIX=
 fi
@@ -142,16 +128,16 @@
 AC_SUBST(LIB_SUFFIX)
 AC_SUBST(GENERATED_SOURCES)
 
-GLIB_SHARP_REQ_VERSION=2.12.1
-GTK_SHARP_REQ_VERSION=2.12
-PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0 >= $GTK_SHARP_REQ_VERSION, has_gtksharp=true, has_gtksharp=false)
+GLIB_SHARP_REQ_VERSION=2.99
+GTK_SHARP_REQ_VERSION=2.99
+PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-3.0 >= $GTK_SHARP_REQ_VERSION, has_gtksharp=true, has_gtksharp=false)
 
-PKG_CHECK_MODULES(GAPI, gapi-2.0 >= $GLIB_SHARP_REQ_VERSION, has_gapi=true, has_gapi=false)
-AC_PATH_PROG(GAPI_PARSER, gapi2-parser, no)
+PKG_CHECK_MODULES(GAPI, gapi-3.0 >= $GLIB_SHARP_REQ_VERSION, has_gapi=true, has_gapi=false)
+AC_PATH_PROG(GAPI_PARSER, gapi3-parser, no)
 AC_SUBST(GAPI_PARSER)
-AC_PATH_PROG(GAPI_CODEGEN, gapi2-codegen, no)
+AC_PATH_PROG(GAPI_CODEGEN, gapi3-codegen, no)
 AC_SUBST(GAPI_CODEGEN)
-AC_PATH_PROG(GAPI_FIXUP, gapi2-fixup, no)
+AC_PATH_PROG(GAPI_FIXUP, gapi3-fixup, no)
 AC_SUBST(GAPI_FIXUP)
 
 # Check for nunit
@@ -185,7 +171,6 @@
 # Python
 ###########################
 
-PYGTK_REQUIRED=2.14.0
 PYGOBJECT_REQUIRED=0.22
 
 AM_PATH_PYTHON(2.3.5)
@@ -193,20 +178,19 @@
 
 PKG_CHECK_MODULES(APPINDICATOR_PYTHON,
                   [
-					pygtk-2.0 >= $PYGTK_REQUIRED
-					gtk+-2.0 >= $GTK_REQUIRED_VERSION
-					pygobject-2.0 >= $PYGOBJECT_REQUIRED
+                                        gtk+-3.0 >= $GTK_REQUIRED_VERSION
+                                        pygobject-3.0 >= $PYGOBJECT_REQUIRED
                   ])
 
-AC_MSG_CHECKING(for pygtk defs)
-PYGTK_DEFSDIR=`$PKG_CONFIG --variable=defsdir pygtk-2.0`
-AC_SUBST(PYGTK_DEFSDIR)
-AC_MSG_RESULT($PYGTK_DEFSDIR)
-
-AC_MSG_CHECKING(for pygtk codegen)
-PYGTK_CODEGEN="$PYTHON `$PKG_CONFIG --variable=codegendir pygtk-2.0`/codegen.py"
-AC_SUBST(PYGTK_CODEGEN)
-AC_MSG_RESULT($PYGTK_CODEGEN)
+# AC_MSG_CHECKING(for pygtk defs)
+# PYGTK_DEFSDIR=`$PKG_CONFIG --variable=defsdir pygtk-3.0`
+# AC_SUBST(PYGTK_DEFSDIR)
+# AC_MSG_RESULT($PYGTK_DEFSDIR)
+
+# AC_MSG_CHECKING(for pygtk codegen)
+# PYGTK_CODEGEN="$PYTHON `$PKG_CONFIG --variable=codegendir pygtk-3.0`/codegen.py"
+# AC_SUBST(PYGTK_CODEGEN)
+# AC_MSG_RESULT($PYGTK_CODEGEN)
 
 #########################
 # Check if build tests

