From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Thu, 14 Mar 2019 14:00:00 +0200
Subject: [PATCH 1/1] Fix build errors + add install targets


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1111111..2222222 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,3 +66,12 @@ endif()
 add_subdirectory (libde265)
 add_subdirectory (dec265)
 add_subdirectory (enc265)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix "\${prefix}/bin")
+set(libdir "\${prefix}/lib")
+set(includedir "\${prefix}/include")
+set(VERSION ${PACKAGE_VERSION})
+
+configure_file(libde265.pc.in libde265.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libde265.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
diff --git a/dec265/CMakeLists.txt b/dec265/CMakeLists.txt
index 1111111..2222222 100644
--- a/dec265/CMakeLists.txt
+++ b/dec265/CMakeLists.txt
@@ -18,7 +18,7 @@ endif()
 
 install (TARGETS dec265 DESTINATION ${CMAKE_INSTALL_BINDIR})
 
-if(NOT MSVC)
+if(NOT WIN32)
   # hdrcopy uses internal APIs that are not available when compiled for Windows
   add_executable (hdrcopy hdrcopy.cc)
 
diff --git a/libde265/CMakeLists.txt b/libde265/CMakeLists.txt
index 1111111..2222222 100644
--- a/libde265/CMakeLists.txt
+++ b/libde265/CMakeLists.txt
@@ -108,6 +108,10 @@ if(NOT DISABLE_SSE)
 endif()
 
 add_library(${PROJECT_NAME} ${libde265_sources} ${ENCODER_OBJECTS} ${X86_OBJECTS})
+set_target_properties(${PROJECT_NAME} PROPERTIES
+    PREFIX ""
+    IMPORT_PREFIX ""
+)
 target_link_libraries(${PROJECT_NAME} PRIVATE Threads::Threads)
 
 write_basic_package_version_file(${PROJECT_NAME}ConfigVersion.cmake COMPATIBILITY ExactVersion)
 