diff --git a/clang-tools-extra/clang-include-fixer/plugin/CMakeLists.txt b/clang-tools-extra/clang-include-fixer/plugin/CMakeLists.txt
index 6d0328ed831..d531e44743a 100644
--- a/clang-tools-extra/clang-include-fixer/plugin/CMakeLists.txt
+++ b/clang-tools-extra/clang-include-fixer/plugin/CMakeLists.txt
@@ -2,15 +2,18 @@ add_clang_library(clangIncludeFixerPlugin
   IncludeFixerPlugin.cpp
 
   LINK_LIBS
+  clangIncludeFixer
+  ${LLVM_PTHREAD_LIB}
+
+  DEPENDS
+  omp_gen
+  )
+clang_target_link_libraries(clangIncludeFixerPlugin
+  PRIVATE
   clangAST
   clangBasic
   clangFrontend
-  clangIncludeFixer
   clangParse
   clangSema
   clangTooling
-  ${LLVM_PTHREAD_LIB}
-
-  DEPENDS
-  omp_gen
   )
diff --git a/clang-tools-extra/clangd/refactor/tweaks/CMakeLists.txt b/clang-tools-extra/clangd/refactor/tweaks/CMakeLists.txt
index 8f708cacfdf..b08e9cee954 100644
--- a/clang-tools-extra/clangd/refactor/tweaks/CMakeLists.txt
+++ b/clang-tools-extra/clangd/refactor/tweaks/CMakeLists.txt
@@ -27,10 +27,13 @@ add_clang_library(clangDaemonTweaks OBJECT
   SwapIfBranches.cpp
 
   LINK_LIBS
-  clangAST
-  clangBasic
   clangDaemon
   clangdSupport
+  )
+
+clang_target_link_libraries(clangDaemonTweaks INTERFACE
+  clangAST
+  clangBasic
   clangFormat
   clangLex
   clangToolingCore
