diff --git a/scripts/packaging/linux_distro.py b/scripts/packaging/linux_distro.py
index 7ecb7e1..456aa36 100644
--- a/scripts/packaging/linux_distro.py
+++ b/scripts/packaging/linux_distro.py
@@ -199,18 +199,6 @@ class LinuxWebotsPackage(WebotsPackage):
               .format(self.distribution_path, self.application_name_lowercase_and_dashes, self.package_version))
 
         # add specific libraries needed for tarball package (Ubuntu 24.04)
-        usr_lib_x68_64 = self.USR_LIB_X68_64
-        usr_lib_x68_64.append('libjpeg.so.8')
-        if distro.version() == '22.04':
-            usr_lib_x68_64 += self.USR_LIB_X68_64_22_04
-            usr_lib_x68_64.append('libraw.so.20')
-        if distro.version() == '24.04':
-            usr_lib_x68_64 += self.USR_LIB_X68_64_24_04
-            usr_lib_x68_64.append('libraw.so.23')
-        system_lib_path = os.path.join('/usr', 'lib', 'x86_64-linux-gnu')
-        package_webots_lib = os.path.join(self.package_webots_path, 'lib', 'webots')
-        for lib in usr_lib_x68_64:
-            shutil.copy(os.path.join(system_lib_path, lib), package_webots_lib)
 
         os.chdir(self.package_webots_path)
         os.chdir('..')
