diff --git a/lightcrafts/jnisrc/lensfun/LC_lensfun.cpp b/lightcrafts/jnisrc/lensfun/LC_lensfun.cpp
index 34d12df..ff62311 100644
--- a/lightcrafts/jnisrc/lensfun/LC_lensfun.cpp
+++ b/lightcrafts/jnisrc/lensfun/LC_lensfun.cpp
@@ -4,6 +4,7 @@
 
 #include <cmath>
 #include <cstring>
+#include <cstdint>
 #include <vector>
 #include <omp.h>
 #include <jni.h>
diff --git a/linux/products/lightzone b/linux/products/lightzone
index a11e2ea..3e06672 100644
--- a/linux/products/lightzone
+++ b/linux/products/lightzone
@@ -9,7 +9,11 @@ java -version
 arch=`getconf LONG_BIT`
 PLATFORM=`uname`
 if [ "${PLATFORM}" = "Linux" ]; then
-  usrdir=/usr
+  if [ -f /.flatpak-info ]; then
+    usrdir=/app
+  else
+    usrdir=/usr
+  fi
 elif [ "${PLATFORM}" = "SunOS" ]; then
   usrdir=/usr
 elif [ "${PLATFORM}" = "FreeBSD" ]; then
