diff --git a/src/meson.build b/src/meson.build
index 205f13c..86db83d 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -144,6 +144,7 @@ vklayer_mesa_overlay = shared_library(
     dep_vulkan],
   include_directories : [inc_common],
   link_args : link_args,
+  install_rpath: join_paths(get_option('prefix'), libdir_mangohud),
   install_dir : libdir_mangohud,
   install : true
 )
@@ -166,6 +167,7 @@ mangohud_dlsym = shared_library(
   dependencies : [dep_dl],
   include_directories : [inc_common],
   link_args : link_args,
+  install_rpath: join_paths(get_option('prefix'), libdir_mangohud),
   install_dir : libdir_mangohud,
   install : true
 )
