diff --git a/src/3rd_party/faiss/Index.h b/src/3rd_party/faiss/Index.h
index deaabcaa..61827ea8 100644
--- a/src/3rd_party/faiss/Index.h
+++ b/src/3rd_party/faiss/Index.h
@@ -15,6 +15,7 @@
 #include <typeinfo>
 #include <string>
 #include <sstream>
+#include <cstdint>
 
 #define FAISS_VERSION_MAJOR 1
 #define FAISS_VERSION_MINOR 6
diff --git a/src/common/signal_handling.h b/src/common/signal_handling.h
index 25111b2c..9985e643 100644
--- a/src/common/signal_handling.h
+++ b/src/common/signal_handling.h
@@ -1,6 +1,7 @@
 #pragma once
 #include <csignal>
 #include <string>
+#include <cstdint>
 
 // SIGNAL HANDLING
 
