diff --git a/MONSTR.jucer b/MONSTR.jucer
index 5e5ddeb..09d073d 100644
--- a/MONSTR.jucer
+++ b/MONSTR.jucer
@@ -19,24 +19,24 @@
             file="Source/Graphics/MONSTR Background.png"/>
     </GROUP>
     <GROUP id="{54DDA0CD-EF70-78A2-BADE-B0BCA860E60F}" name="DSPFilters">
-      <FILE id="NEx5Yr" name="Bessel.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Bessel.cpp"/>
-      <FILE id="ZAhsNd" name="Biquad.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Biquad.cpp"/>
-      <FILE id="DiKHjv" name="Butterworth.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Butterworth.cpp"/>
-      <FILE id="y1OMz4" name="Cascade.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Cascade.cpp"/>
-      <FILE id="WSU68D" name="ChebyshevI.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/ChebyshevI.cpp"/>
-      <FILE id="QDCfEZ" name="ChebyshevII.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/ChebyshevII.cpp"/>
-      <FILE id="Yy0xPk" name="Custom.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Custom.cpp"/>
-      <FILE id="TrlGPp" name="Design.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Design.cpp"/>
+      <FILE id="NEx5Yr" name="Bessel.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Bessel.cpp"/>
+      <FILE id="ZAhsNd" name="Biquad.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Biquad.cpp"/>
+      <FILE id="DiKHjv" name="Butterworth.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Butterworth.cpp"/>
+      <FILE id="y1OMz4" name="Cascade.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Cascade.cpp"/>
+      <FILE id="WSU68D" name="ChebyshevI.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/ChebyshevI.cpp"/>
+      <FILE id="QDCfEZ" name="ChebyshevII.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/ChebyshevII.cpp"/>
+      <FILE id="Yy0xPk" name="Custom.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Custom.cpp"/>
+      <FILE id="TrlGPp" name="Design.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Design.cpp"/>
       <FILE id="k2unxy" name="Documentation.cpp" compile="1" resource="0"
-            file="../WECore/DSPFilters/shared/DSPFilters/source/Documentation.cpp"/>
-      <FILE id="foM98L" name="Elliptic.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Elliptic.cpp"/>
-      <FILE id="kCX8Hy" name="Filter.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Filter.cpp"/>
-      <FILE id="jFDZyh" name="Legendre.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Legendre.cpp"/>
-      <FILE id="OkUeWe" name="Param.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/Param.cpp"/>
-      <FILE id="NTQmNl" name="PoleFilter.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/PoleFilter.cpp"/>
-      <FILE id="cms4R0" name="RBJ.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/RBJ.cpp"/>
-      <FILE id="QItWYH" name="RootFinder.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/RootFinder.cpp"/>
-      <FILE id="NAjeKN" name="State.cpp" compile="1" resource="0" file="../WECore/DSPFilters/shared/DSPFilters/source/State.cpp"/>
+            file="WE-Core/DSPFilters/shared/DSPFilters/source/Documentation.cpp"/>
+      <FILE id="foM98L" name="Elliptic.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Elliptic.cpp"/>
+      <FILE id="kCX8Hy" name="Filter.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Filter.cpp"/>
+      <FILE id="jFDZyh" name="Legendre.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Legendre.cpp"/>
+      <FILE id="OkUeWe" name="Param.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/Param.cpp"/>
+      <FILE id="NTQmNl" name="PoleFilter.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/PoleFilter.cpp"/>
+      <FILE id="cms4R0" name="RBJ.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/RBJ.cpp"/>
+      <FILE id="QItWYH" name="RootFinder.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/RootFinder.cpp"/>
+      <FILE id="NAjeKN" name="State.cpp" compile="1" resource="0" file="WE-Core/DSPFilters/shared/DSPFilters/source/State.cpp"/>
     </GROUP>
     <GROUP id="{893CC731-9921-30A1-C72A-2337643309C3}" name="Source">
       <GROUP id="{AB2D53B4-98E6-6EC7-2DE3-F1C924AB3ADD}" name="GUI">
@@ -98,29 +98,29 @@
                 vst3Folder="~/SDKs/VST3 SDK">
       <CONFIGURATIONS>
         <CONFIGURATION name="Debug" libraryPath="/usr/X11R6/lib/" isDebug="0" optimisation="3"
-                       targetName="MONSTR" headerPath="~/WEA/WECore/WECore&#10;~/WEA/WECore/DSPFilters/shared/DSPFilters/include"
+                       targetName="MONSTR" headerPath="../../WE-Core/WECore&#10;../../WE-Core/DSPFilters/shared/DSPFilters/include"
                        linkTimeOptimisation="0"/>
         <CONFIGURATION name="Release" libraryPath="/usr/X11R6/lib/" isDebug="0" optimisation="3"
                        targetName="MONSTR"/>
       </CONFIGURATIONS>
       <MODULEPATHS>
-        <MODULEPATH id="juce_video" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_opengl" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_gui_extra" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_gui_basics" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_graphics" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_events" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_data_structures" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_cryptography" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_core" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_audio_processors" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_audio_plugin_client" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_audio_formats" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_audio_devices" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_audio_basics" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_audio_utils" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_box2d" path="../JUCE/modules"/>
-        <MODULEPATH id="juce_osc" path="../JUCE/modules"/>
+        <MODULEPATH id="juce_video" path="JUCE/modules"/>
+        <MODULEPATH id="juce_opengl" path="JUCE/modules"/>
+        <MODULEPATH id="juce_gui_extra" path="JUCE/modules"/>
+        <MODULEPATH id="juce_gui_basics" path="JUCE/modules"/>
+        <MODULEPATH id="juce_graphics" path="JUCE/modules"/>
+        <MODULEPATH id="juce_events" path="JUCE/modules"/>
+        <MODULEPATH id="juce_data_structures" path="JUCE/modules"/>
+        <MODULEPATH id="juce_cryptography" path="JUCE/modules"/>
+        <MODULEPATH id="juce_core" path="JUCE/modules"/>
+        <MODULEPATH id="juce_audio_processors" path="JUCE/modules"/>
+        <MODULEPATH id="juce_audio_plugin_client" path="JUCE/modules"/>
+        <MODULEPATH id="juce_audio_formats" path="JUCE/modules"/>
+        <MODULEPATH id="juce_audio_devices" path="JUCE/modules"/>
+        <MODULEPATH id="juce_audio_basics" path="JUCE/modules"/>
+        <MODULEPATH id="juce_audio_utils" path="JUCE/modules"/>
+        <MODULEPATH id="juce_box2d" path="JUCE/modules"/>
+        <MODULEPATH id="juce_osc" path="JUCE/modules"/>
       </MODULEPATHS>
     </LINUX_MAKE>
     <VS2017 targetFolder="Builds/VisualStudio2017" extraCompilerFlags="/std:c++17">
@@ -169,7 +169,7 @@
     <MODULE id="juce_osc" showAllCode="1" useLocalCopy="0"/>
     <MODULES id="juce_video" showAllCode="1" useLocalCopy="0"/>
   </MODULES>
-  <JUCEOPTIONS JUCE_QUICKTIME="disabled"/>
+  <JUCEOPTIONS JUCE_QUICKTIME="disabled" JUCE_WEB_BROWSER="disabled"/>
   <LIVE_SETTINGS>
     <OSX/>
   </LIVE_SETTINGS>
