commit c1de677483f7222160ac3eb0fbe73ab4606b04a3
Author: Hubert Figuière <hub@figuiere.net>
Date:   Sat Jan 19 09:18:48 2019 -0500

    Fix build failure

diff --git a/muse3/muse/lv2Support/sord-0.14.0/src/sord.c b/muse3/muse/lv2Support/sord-0.14.0/src/sord.c
index 8728a224..f8c394dd 100644
--- a/muse3/muse/lv2Support/sord-0.14.0/src/sord.c
+++ b/muse3/muse/lv2Support/sord-0.14.0/src/sord.c
@@ -1060,7 +1060,7 @@ sord_new_literal_counted(SordWorld*     world,
 	key.meta.lit.datatype = sord_node_copy(datatype);
 	memset(key.meta.lit.lang, 0, sizeof(key.meta.lit.lang));
 	if (lang) {
-		strncpy(key.meta.lit.lang, lang, sizeof(key.meta.lit.lang));
+		strncpy(key.meta.lit.lang, lang, sizeof(key.meta.lit.lang) - 1);
 	}
 
 	return sord_insert_node(world, &key, true);
