diff -u mustache/mustache.hpp mustache.patched/mustache.hpp
--- mustache/mustache.hpp	2018-07-23 05:13:12.000000000 +0200
+++ mustache.patched/mustache.hpp	2019-01-07 18:08:24.601917507 +0100
@@ -423,6 +423,7 @@
 template <typename string_type>
 class basic_context {
 public:
+    virtual ~basic_context() = default;
     virtual void push(const basic_data<string_type>* data) = 0;
     virtual void pop() = 0;
 
