--- a/Engine/AppManager.cpp.orig	2018-05-12 13:49:44.000000000 +0400
+++ b/Engine/AppManager.cpp	2018-05-12 14:59:43.887702689 +0400
@@ -473,7 +473,7 @@
     // set fontconfig path on all platforms
     if ( qgetenv("FONTCONFIG_PATH").isNull() ) {
         // set FONTCONFIG_PATH to Natron/Resources/etc/fonts (required by plugins using fontconfig)
-        QString path = QCoreApplication::applicationDirPath() + QString::fromUtf8("/../Resources/etc/fonts");
+        QString path = QCoreApplication::applicationDirPath() + QString::fromUtf8("/../etc/fonts");
         QFileInfo fileInfo(path);
         if ( !fileInfo.exists() ) {
             std::cerr <<  "Fontconfig configuration file " << path.toStdString() << " does not exist, not setting FONTCONFIG_PATH "<< std::endl;
@@ -3091,7 +3091,7 @@
     QString pluginPath = binPath + QString::fromUtf8("\\..\\Plugins");
 #else
 #  if defined(__NATRON_LINUX__)
-    static std::string pythonHome = binPath.toStdString() + "/.."; // must use static storage
+    static std::string pythonHome = binPath.toStdString() + "/lib/python" NATRON_PY_VERSION_STRING; // must use static storage
 #  elif defined(__NATRON_OSX__)
     static std::string pythonHome = binPath.toStdString() + "/../Frameworks/Python.framework/Versions/" NATRON_PY_VERSION_STRING; // must use static storage
 #  else

--- a/Engine/Settings.cpp.orig	2018-05-12 13:49:47.000000000 +0400
+++ b/Engine/Settings.cpp	2018-05-12 18:10:44.596550312 +0400
@@ -93,9 +93,9 @@
 
 #ifdef __NATRON_LINUX__
     QStringList ret;
-    ret.push_back( QString::fromUtf8("/usr/share/OpenColorIO-Configs") );
+    //ret.push_back( QString::fromUtf8("/usr/share/OpenColorIO-Configs") );
     ret.push_back( QString( binaryPath + QString::fromUtf8("../share/OpenColorIO-Configs") ) );
-    ret.push_back( QString( binaryPath + QString::fromUtf8("../Resources/OpenColorIO-Configs") ) );
+    //ret.push_back( QString( binaryPath + QString::fromUtf8("../Resources/OpenColorIO-Configs") ) );
 
     return ret;
 #elif defined(__NATRON_WIN32__)
