diff --git a/receivers/electron/forge.config.js b/receivers/electron/forge.config.js
index 8fbbdf9..cb33c83 100644
--- a/receivers/electron/forge.config.js
+++ b/receivers/electron/forge.config.js
@@ -22,6 +22,12 @@ module.exports = {
       appleApiKey: process.env.FCAST_APPLE_API_KEY,
       appleApiKeyId: process.env.FCAST_APPLE_API_KEY_ID,
       appleApiIssuer: process.env.FCAST_APPLE_API_ISSUER
+    },
+    download: {
+      checksums: {
+        "electron-v35.2.0-linux-x64.zip": "c3bc0fefe238e092ab2a26276d6c4bd5d70b85d08d8bb298c5f7499a4456a119",
+        "electron-v35.2.0-linux-arm64.zip": "c3b5224bbaac7b69890dbf6f69d78ad6dc2622231fa7554e5cf32333623a6d74"
+      }
     }
   },
   rebuildConfig: {},
