diff --git a/fluid/wscript b/fluid/wscript
index ab6f898..6ca58c6 100644
--- a/fluid/wscript
+++ b/fluid/wscript
@@ -32,10 +32,12 @@ def build(bld):
 ''',
                  target='ntk-fluid',
                  includes = [ '.', '../' ],
-                 use = [ 'ntk_images_shared', 'ntk_shared' ],
-		 # FIXME: why is this now neccessary with new waf version?
-                 uselib = [
- 'LIBJPEG', 'LIBPNG', 'LIBZ', 'DL', 'M', 'PTHREAD' ,
-                    'X11', 'FONTCONFIG', 'XFT', 'CAIRO', 'DL', 'M', 'PTHREAD' ], 
+                 cflags = [ "-fPIC" ],
+                 cxxflags = [ "-fPIC" ],
+                 use = [
+                     'LIBJPEG', 'LIBPNG', 'LIBZ', 'DL', 'M', 'PTHREAD',
+                     'X11', 'FONTCONFIG', 'XFT', 'CAIRO', 'DL', 'M', 'PTHREAD',
+                     'ntk_images_static', 'ntk_static',
+                 ],
                  install_path = '${BINDIR}' )
     
diff --git a/ntk.pc.in b/ntk.pc.in
index c273965..95ffc14 100644
--- a/ntk.pc.in
+++ b/ntk.pc.in
@@ -7,7 +7,6 @@ Name: ntk
 Description: Non ToolKit
 Version: @VERSION@
 
-Requires: cairo >= 1.9.0
-Requires.private: x11 xft
+Requires: cairo >= 1.9.0 x11 xft
 Libs: -L${libdir} -lntk
 Cflags: -I${includedir}/ntk @CFLAGS@
diff --git a/wscript b/wscript
index 2b10dd8..c5534e1 100644
--- a/wscript
+++ b/wscript
@@ -42,10 +42,10 @@ def makelib(bld,*k,**kw):
     kw['install_path'] = '${LIBDIR}'
     kw['features' ] = 'c cxx cxxstlib'
     kw['name'] = kw['target'] + '_static'
-    bld.shlib(*k,**kw)
-    kw['features' ] = 'c cxx cxxshlib'
-    kw['name'] = kw['target'] + '_shared'
     bld.stlib(*k,**kw)
+#    kw['features' ] = 'c cxx cxxshlib'
+#    kw['name'] = kw['target'] + '_shared'
+#    bld.shlib(*k,**kw)
 
 # from autowaf
 def run_ldconfig(ctx):
@@ -78,9 +78,9 @@ def configure(conf):
     conf.load('gnu_dirs')
     # conf.load('ntk_fluid')
     conf.line_just = 52
-    conf.env.append_value('CFLAGS', ['-Wall'])
+    conf.env.append_value('CFLAGS', ['-Wall', '-fPIC'])
 #    conf.env.append_value('CXXFLAGS',['-Wall','-fno-exceptions', '-fno-rtti'])
-    conf.env.append_value('CXXFLAGS', ['-Wall'])
+    conf.env.append_value('CXXFLAGS', ['-Wall', '-fPIC'])
 
     conf.check_cfg(package='x11', uselib_store='X11', args="--cflags --libs",
                    mandatory=True)
@@ -506,7 +506,10 @@ src/Fl_Gl_Window.cxx
 	source = 'src/ntk-chtheme.cxx',
 	target = 'ntk-chtheme',
         includes = [ '.' ], 
-        use = ['ntk_images_shared', 'ntk_shared'],
+        use = [
+            'X11', 'FONTCONFIG', 'XFT', 'CAIRO', 'DL',
+            'ntk_images_static', 'ntk_static'
+        ],
 	install_path = "${BINDIR}" )
 
     # bld( features = 'subst',
