diff --git a/src/obs-glcapture b/src/obs-glcapture
index 2efcf51..7432be0 100755
--- a/src/obs-glcapture
+++ b/src/obs-glcapture
@@ -1,6 +1,7 @@
 #!/bin/sh
 
 OBS_GLCAPTURE_LIB="libobs_glcapture.so"
+LIBDIR="/usr/lib/extensions/vulkan/OBSVkCapture/\$LIB"
 
 if [ "$#" -eq 0 ]; then
     programname=`basename "$0"`
@@ -10,5 +11,11 @@ if [ "$#" -eq 0 ]; then
     exit 1
 fi
 
+if [ -z "${LD_LIBRARY_PATH}" ]; then
+	LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${LIBDIR}"
+else
+	LD_LIBRARY_PATH="${LIBDIR}"
+fi
+
 LD_PRELOAD="${LD_PRELOAD}:${OBS_GLCAPTURE_LIB}"
-exec env LD_PRELOAD="${LD_PRELOAD}" "$@"
+exec env LD_PRELOAD="${LD_PRELOAD}" LD_LIBRARY_PATH="${LD_LIBRARY_PATH}" "$@"
