diff --git a/src/obs-glcapture b/src/obs-glcapture
index 2efcf51..aecf1ef 100755
--- a/src/obs-glcapture
+++ b/src/obs-glcapture
@@ -1,6 +1,9 @@
 #!/bin/sh
 
 OBS_GLCAPTURE_LIB="libobs_glcapture.so"
+LIBDIR_BASE="/usr/lib/extensions/vulkan/OBSVkCapture/lib"
+LIBDIR_32="${LIBDIR_BASE}/i386-linux-gnu/"
+LIBDIR_64="${LIBDIR_BASE}/x86_64-linux-gnu/"
 
 if [ "$#" -eq 0 ]; then
     programname=`basename "$0"`
@@ -11,4 +14,5 @@ if [ "$#" -eq 0 ]; then
 fi
 
 LD_PRELOAD="${LD_PRELOAD}:${OBS_GLCAPTURE_LIB}"
-exec env LD_PRELOAD="${LD_PRELOAD}" "$@"
+LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${LIBDIR_32}:${LIBDIR_64}"
+exec env LD_PRELOAD="${LD_PRELOAD}" LD_LIBRARY_PATH="${LD_LIBRARY_PATH}" "$@"
