diff --git a/Source/gui/PatchBrowser.cpp b/Source/gui/PatchBrowser.cpp
index 3ab8d00..32d43bc 100644
--- a/Source/gui/PatchBrowser.cpp
+++ b/Source/gui/PatchBrowser.cpp
@@ -43,6 +43,11 @@ PatchBrowser::PatchBrowser(OdinAudioProcessor &p_processor, AudioProcessorValueT
 	m_category_selector.setCopyTargetName("Soundbank");
 	m_patch_selector.setCopyTargetName("Category");
 
+	File dir_to_create(DEFAULT_SOUNDBANK_LOCATION_STRING);
+	if (!dir_to_create.exists()) {
+		dir_to_create.createDirectory();
+	}
+
 	m_soundbank_selector.setDirectory(DEFAULT_SOUNDBANK_LOCATION_STRING);
 	
 	m_category_selector.setDirectoryFactoryPresetCategory();
@@ -1200,4 +1205,4 @@ void PatchBrowser::setFirstSoundbankActive() {
 	m_category_selector.setDirectoryFactoryPresetCategory();
 	m_category_selector.highlightFirstEntry();
 	m_patch_selector.setDirectoryFactoryPresetPreset("Arps & Sequences");
-}
\ No newline at end of file
+}
