--- of_v0.12.0_linux64gcc6_release/libs/openFrameworks/app/ofAppGLFWWindow.cpp	2023-12-11 01:12:45.637425848 -0300
+++ ofAppGLFWWindow.cpp	2023-12-11 01:13:20.449799973 -0300
@@ -1632,7 +1632,7 @@
 //--------------------------------------------
 void ofAppGLFWWindow::exit_cb(GLFWwindow* windowP_){
 	ofAppGLFWWindow * instance = setCurrent(windowP_);
-	instance->events().notifyExit();
+if(instance->events().notifyExit()) glfwSetWindowShouldClose(windowP_, false);
 }
 
 //------------------------------------------------------------
