diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2cbbe51..99a8453 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
 include(glib-resource-compiler)
 find_package(PkgConfig REQUIRED)
 # Additional search path since gtkmm-4.0 will usually be installed in /usr
-set(ENV{PKG_CONFIG_PATH} "/usr/lib/pkgconfig")
+set(ENV{PKG_CONFIG_PATH} "/app/lib/pkgconfig")
 pkg_check_modules(GTKMM gtkmm-4.0>=4.4.0 REQUIRED IMPORTED_TARGET)
 
 # Lib adwaita
@@ -69,22 +69,26 @@ pkg_check_modules(GTKMM gtkmm-4.0>=4.4.0 REQUIRED IMPORTED_TARGET)
 # include_directories(${ADWAITA_INCLUDE_DIRS})
 
 # Disable linting for fetch content projects
-clear_variable(DESTINATION CMAKE_CXX_CLANG_TIDY BACKUP CMAKE_CXX_CLANG_TIDY_BKP)
+# clear_variable(DESTINATION CMAKE_CXX_CLANG_TIDY BACKUP CMAKE_CXX_CLANG_TIDY_BKP)
 
-FetchContent_Declare(cpr GIT_REPOSITORY https://github.com/libcpr/cpr.git
-                         GIT_TAG 6ea2dec23c3df14ac3b27b7d2d6bbff0cb7ba1b0) # The commit hash for 1.8.1. Replace with the latest from: https://github.com/libcpr/cpr/releases
-FetchContent_MakeAvailable(cpr)
+# FetchContent_Declare(cpr GIT_REPOSITORY https://github.com/libcpr/cpr.git
+#                          GIT_TAG 6ea2dec23c3df14ac3b27b7d2d6bbff0cb7ba1b0) # The commit hash for 1.8.1. Replace with the latest from: https://github.com/libcpr/cpr/releases
+# FetchContent_MakeAvailable(cpr)
 
-FetchContent_Declare(json GIT_REPOSITORY https://github.com/nlohmann/json.git
-                          GIT_TAG v3.10.5)
-FetchContent_MakeAvailable(json)
+# FetchContent_Declare(json GIT_REPOSITORY https://github.com/nlohmann/json.git
+#                           GIT_TAG v3.10.5)
+# FetchContent_MakeAvailable(json)
 
-set(SPDLOG_INSTALL ON)
-FetchContent_Declare(spdlog GIT_REPOSITORY https://github.com/gabime/spdlog.git
-                          GIT_TAG v1.9.2)
-FetchContent_MakeAvailable(spdlog)
+# set(SPDLOG_INSTALL ON)
+# FetchContent_Declare(spdlog GIT_REPOSITORY https://github.com/gabime/spdlog.git
+#                           GIT_TAG v1.9.2)
+# FetchContent_MakeAvailable(spdlog)
 
-restore_variable(DESTINATION CMAKE_CXX_CLANG_TIDY BACKUP CMAKE_CXX_CLANG_TIDY_BKP)
+# restore_variable(DESTINATION CMAKE_CXX_CLANG_TIDY BACKUP CMAKE_CXX_CLANG_TIDY_BKP)
+
+find_package(spdlog CONFIG REQUIRED)
+find_package(nlohmann_json CONFIG REQUIRED)
+find_package(cpr CONFIG REQUIRED)
 
 include_directories(${CMAKE_SOURCE_DIR}/src)
 add_subdirectory(src)
