From 8bb601f6d1f95822ec35ba61b3a321aae181dbb7 Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Sun, 1 Mar 2026 09:54:43 +0100
Subject: [PATCH] Fix build with GCC 15+

---
 lib/ome/files/CoreMetadata.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/ome/files/CoreMetadata.h b/lib/ome/files/CoreMetadata.h
index d155372..3d6a00f 100644
--- a/lib/ome/files/CoreMetadata.h
+++ b/lib/ome/files/CoreMetadata.h
@@ -44,6 +44,7 @@
 #include <numeric>
 #include <string>
 #include <vector>
+#include <memory>
 
 #include <ome/files/MetadataMap.h>
 #include <ome/files/Modulo.h>
-- 
GitLab

