--- a/CImg/Makefile	2018-04-30 02:19:49.129905724 +0400
+++ b/CImg/Makefile	2018-04-30 13:46:04.153643550 +0400
@@ -179,12 +179,12 @@
 CIMGVERSION=88fab6de7bfc141a1f577e3cf1b17b9fb1e4f438
 
 CImg.h: Inpaint/inpaint.h
-	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/CImg.h
+#	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/CImg.h
 #	patch -p0 -d. < CImg-2.0.0-anisotropic.patch
 #	patch -p0 -d. < CImg-1.7.1-omp.patch
 
 Inpaint/inpaint.h:
-	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/plugins/inpaint.h
+#	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/plugins/inpaint.h
 	patch -p0 -d. < Inpaint/inpaint.h.patch
 
 nlmeans.h:
