--- a/openfx/HostSupport/src/ofxhPluginCache.cpp
+++ b/openfx/HostSupport/src/ofxhPluginCache.cpp
@@ -314,7 +314,7 @@ PluginCache::PluginCache()
   _pluginPath.push_back("C:\\Program Files\\Common Files\\OFX\\Plugins");
 #endif
 #if defined(__linux__) || defined(__FreeBSD__)
-  _pluginPath.push_back("/usr/OFX/Plugins");
+  _pluginPath.push_back("/Plugins");
 #endif
 #if defined(__APPLE__)
   _pluginPath.push_back("/Library/OFX/Plugins");
--- a/openfx/Support/Plugins/Makefile.master
+++ b/openfx/Support/Plugins/Makefile.master
@@ -55,7 +55,7 @@ ifeq ($(OS),$(filter $(OS),MINGW64_NT-6.
   PLUGINPATH=C:\\Program Files\\Common Files\\OFX\\Plugins
 endif
 ifeq ($(OS),Linux)
-  PLUGINPATH=/usr/OFX/Plugins
+  PLUGINPATH=/Plugins
 endif
 ifeq ($(OS),FreeBSD)
   PLUGINPATH=/usr/OFX/Plugins
