--- a/DebugProxy/DebugProxy.cpp
+++ b/DebugProxy/DebugProxy.cpp
@@ -74,7 +74,7 @@ OFXS_NAMESPACE_ANONYMOUS_ENTER
 #define BINARY_PATH "C:\\Program Files\\Common Files\\OFX\\Plugins.disabled\\Sapphire.ofx.bundle\\Contents\\Win64\\Sapphire.ofx"
 #endif
 #if defined(__linux__)
-#define OFX_PATH "/usr/OFX/Plugins/"
+#define OFX_PATH "/Plugins/OFX/Natron/"
 #define BINARY_PATH "/usr/OFX/Plugins.disabled/Sapphire.ofx.bundle/Contents/Linux-x86-64/Sapphire.ofx"
 #endif
 #if defined(__FreeBSD__)
--- a/openfx/HostSupport/src/ofxhPluginCache.cpp
+++ b/openfx/HostSupport/src/ofxhPluginCache.cpp
@@ -288,7 +288,7 @@ PluginCache::PluginCache() : _hostSpec(0
   _pluginPath.push_back("C:\\Program Files\\Common Files\\OFX\\Plugins");
 #endif
 #if defined(__linux__) || defined(__FreeBSD__)
-  _pluginPath.push_back("/usr/OFX/Plugins");
+  _pluginPath.push_back("/Plugins/OFX/Natron");
 #endif
 #if defined(__APPLE__)
   _pluginPath.push_back("/Library/OFX/Plugins");
--- a/openfx/Support/Plugins/Makefile.master
+++ b/openfx/Support/Plugins/Makefile.master
@@ -55,7 +55,7 @@ ifeq ($(OS),$(filter $(OS),MINGW64_NT-6.
   PLUGINPATH=C:\\Program Files\\Common Files\\OFX\\Plugins
 endif
 ifeq ($(OS),Linux)
-  PLUGINPATH=/usr/OFX/Plugins
+  PLUGINPATH=/Plugins/OFX/Natron
 endif
 ifeq ($(OS),FreeBSD)
   PLUGINPATH=/usr/OFX/Plugins
