diff --git a/toonz/sources/toonzlib/preferences.cpp b/toonz/sources/toonzlib/preferences.cpp
index 288a26c..ddc39c2 100644
--- a/toonz/sources/toonzlib/preferences.cpp
+++ b/toonz/sources/toonzlib/preferences.cpp
@@ -497,7 +497,7 @@ void Preferences::definePreferenceItems() {
   setCallBack(autosavePeriod, &Preferences::setAutosavePeriod);
 
   // Import / Export
-  define(ffmpegPath, "ffmpegPath", QMetaType::QString, "");
+  define(ffmpegPath, "ffmpegPath", QMetaType::QString, "/app/bin");
   define(ffmpegTimeout, "ffmpegTimeout", QMetaType::Int, 600, 1,
          std::numeric_limits<int>::max());
   define(fastRenderPath, "fastRenderPath", QMetaType::QString, "desktop");
diff --git a/toonz/sources/toonzlib/thirdparty.cpp b/toonz/sources/toonzlib/thirdparty.cpp
index e5b5cae..12d66ab 100644
--- a/toonz/sources/toonzlib/thirdparty.cpp
+++ b/toonz/sources/toonzlib/thirdparty.cpp
@@ -91,6 +91,7 @@ QString autodetectFFmpeg() {
   if (findFFmpeg("./FFmpeg/bin")) return "./FFmpeg/bin";
 
 #ifndef _WIN32
+  if (findFFmpeg("/app/bin")) return "/app/bin";
   if (findFFmpeg("/usr/local/bin")) return "/usr/local/bin";
   if (findFFmpeg("/usr/bin")) return "/usr/bin";
   if (findFFmpeg("/bin")) return "/bin";
