--- a/Bundle/Makefile.orig	2018-05-06 14:06:27.000000000 +0400
+++ b/Bundle/Makefile	2018-05-06 14:45:04.725326942 +0400
@@ -168,11 +168,11 @@
 # c7bb19ad54a53f0d9917da37de7288b1aadf6d0f is Oct 23, 2017
 PNGVERSION = c7bb19ad54a53f0d9917da37de7288b1aadf6d0f
 
-lodepng.cpp:
-	curl -o $@ https://raw.githubusercontent.com/lvandeve/lodepng/$(PNGVERSION)/lodepng.cpp
+#lodepng.cpp:
+#	curl -o $@ https://raw.githubusercontent.com/lvandeve/lodepng/$(PNGVERSION)/lodepng.cpp
 
-lodepng.h:
-	curl -o $@ https://raw.githubusercontent.com/lvandeve/lodepng/$(PNGVERSION)/lodepng.h
+#lodepng.h:
+#	curl -o $@ https://raw.githubusercontent.com/lvandeve/lodepng/$(PNGVERSION)/lodepng.h
 
 $(OBJECTPATH)/lodepng.o: lodepng.cpp lodepng.h
 $(OBJECTPATH)/ReadKrita.o: ReadKrita.cpp lodepng.h
