diff --git a/data/org.gnome.baobab.appdata.xml.in b/data/org.gnome.baobab.appdata.xml.in
index 2167e6d..a8b3a93 100644
--- a/data/org.gnome.baobab.appdata.xml.in
+++ b/data/org.gnome.baobab.appdata.xml.in
@@ -34,4 +34,14 @@
   <developer_name>The GNOME Project</developer_name>
   <compulsory_for_desktop>GNOME</compulsory_for_desktop>
   <translation type="gettext">baobab</translation>
+
+  <releases>
+    <release version="3.34.0" date="2019-09-09">
+      <description>
+        <p>Updated translations (da, en_GB, eu, fa, hr, mjw, sk, zh_TW)</p>
+      </description>
+    </release>
+  </releases>
+
+  <content_rating type="oars-1.1" />
 </component>
