--- a/data/org.gnome.baobab.appdata.xml.in
+++ b/data/org.gnome.baobab.appdata.xml.in
@@ -18,5 +18,5 @@
  <screenshots>
    <screenshot height="618" width="932" type="default">
-      <image>https://git.gnome.org/browse/baobab/plain/data/baobab.png</image>
+      <image>https://git.gnome.org/browse/baobab/plain/data/default.png</image>
    </screenshot>
  </screenshots>
@@ -34,4 +34,10 @@
   <developer_name>The GNOME Project</developer_name>
   <compulsory_for_desktop>GNOME</compulsory_for_desktop>
   <translation type="gettext">baobab</translation>
+
+  <releases>
+    <release version="3.34.1" date="2019-06-28" />
+  </releases>
+
+  <content_rating type="oars-1.1" />
 </component>
