*** png/MD5	2013-12-04 08:25:05.000000000 +1100
***************
*** 7,13 ****
  7381224c65138a2acdf3a8346f8275c4 *inst/img/Rlogo.png
  0ee7cd3abffb5dd15a5785a694246a8a *man/readPNG.Rd
  653b98a2f5c34e372796df456f220933 *man/writePNG.Rd
! 7fc91ecfbf95133433e23f2e50b4a66d *src/Makevars
  6c1ccc946d45a3351a32f63d8498f712 *src/Makevars.win
  99e46bd4b410b68b8fb9c1f3d66c859a *src/read.c
  b03bb7476f2872fef94827c1af6c8c13 *src/write.c
--- 7,13 ----
  7381224c65138a2acdf3a8346f8275c4 *inst/img/Rlogo.png
  0ee7cd3abffb5dd15a5785a694246a8a *man/readPNG.Rd
  653b98a2f5c34e372796df456f220933 *man/writePNG.Rd
! 1535cd98b8189376907191152502e257 *src/Makevars
  6c1ccc946d45a3351a32f63d8498f712 *src/Makevars.win
  99e46bd4b410b68b8fb9c1f3d66c859a *src/read.c
  b03bb7476f2872fef94827c1af6c8c13 *src/write.c
*** png/src/Makevars	2013-12-04 07:09:13.000000000 +1100
***************
*** 1,2 ****
! PKG_LIBS=$(PNG_LIBS) `libpng-config --static --ldflags`
! PKG_CFLAGS=$(PNG_CFLAGS) `libpng-config --cflags`
--- 1,2 ----
! PKG_LIBS=$(PNG_LIBS)     `pkg-config --static --libs libpng`
! PKG_CFLAGS=$(PNG_CFLAGS) `pkg-config --cflags        libpng`
