diff --git a/qpdfview.pri b/qpdfview.pri
index 776b6e4..a399652 100644
--- a/qpdfview.pri
+++ b/qpdfview.pri
@@ -1,12 +1,12 @@
 isEmpty(APPLICATION_VERSION):APPLICATION_VERSION = 0.5.0
 
-isEmpty(TARGET_INSTALL_PATH):TARGET_INSTALL_PATH = /usr/bin
-isEmpty(PLUGIN_INSTALL_PATH):PLUGIN_INSTALL_PATH = /usr/lib/qpdfview
-isEmpty(DATA_INSTALL_PATH):DATA_INSTALL_PATH = /usr/share/qpdfview
-isEmpty(MANUAL_INSTALL_PATH):MANUAL_INSTALL_PATH = /usr/share/man/man1
-isEmpty(ICON_INSTALL_PATH):ICON_INSTALL_PATH = /usr/share/icons/hicolor/scalable/apps
-isEmpty(LAUNCHER_INSTALL_PATH):LAUNCHER_INSTALL_PATH = /usr/share/applications
-isEmpty(APPDATA_INSTALL_PATH):APPDATA_INSTALL_PATH = /usr/share/metainfo
+isEmpty(TARGET_INSTALL_PATH):TARGET_INSTALL_PATH = /app/bin
+isEmpty(PLUGIN_INSTALL_PATH):PLUGIN_INSTALL_PATH = /app/lib/qpdfview
+isEmpty(DATA_INSTALL_PATH):DATA_INSTALL_PATH = /app/share/qpdfview
+isEmpty(MANUAL_INSTALL_PATH):MANUAL_INSTALL_PATH = /app/share/man/man1
+isEmpty(ICON_INSTALL_PATH):ICON_INSTALL_PATH = /app/share/icons/hicolor/scalable/apps
+isEmpty(LAUNCHER_INSTALL_PATH):LAUNCHER_INSTALL_PATH = /app/share/applications
+isEmpty(APPDATA_INSTALL_PATH):APPDATA_INSTALL_PATH = /app/share/metainfo
 isEmpty(APP_DIR_DATA_PATH):APP_DIR_DATA_PATH = data
 
 win32:include(qpdfview_win32.pri)
