diff --git a/CMakeLists.txt b/CMakeLists.txt
index 421de73..5bb455a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,7 @@ juce_add_plugin(Prince
     PLUGIN_CODE Tsm3 
     FORMATS ${JUCE_FORMATS}
     ProductName "Prince"
-    LV2_URI https://github.com/GuitarML/Prince
+    LV2URI https://github.com/GuitarML/Prince
     ICON_BIG resources/logo.png
     MICROPHONE_PERMISSION_ENABLED TRUE
 )
