--- a/main/core/sandbox/pyodide/Makefile
+++ b/main/core/sandbox/pyodide/Makefile
@@ -11,7 +11,7 @@
 	echo "setup  # get pyodide node package, and python packages"
 
 fetch_packages:
-	node ./preparePackages.js https://s3.amazonaws.com/grist-pynbox/pyodide/packages/v$(GRIST_PYODIDE_VERSION)/ _build/packages/
+	node ./preparePackages.js /run/build/grist/pyodide-packages _build/packages/
 
 build_packages:
 	./build_packages.sh
