--- qscintilla/src/qscintilla.pro-orig	2022-07-23 19:45:02.205793490 -0400
+++ qscintilla/src/qscintilla.pro	2022-07-23 19:47:07.705049272 -0400
@@ -70,22 +70,22 @@
 # Scintilla namespace rather than pollute the global namespace.
 #DEFINES += SCI_NAMESPACE
 
-target.path = $$[QT_INSTALL_LIBS]
+target.path = /app/lib
 INSTALLS += target
 
-header.path = $$[QT_INSTALL_HEADERS]
+header.path = /app/include
 header.files = Qsci
 INSTALLS += header
 
-trans.path = $$[QT_INSTALL_TRANSLATIONS]
+trans.path = /app/share/qt5/translations
 trans.files = qscintilla_*.qm
 INSTALLS += trans
 
-qsci.path = $$[QT_INSTALL_DATA]
+qsci.path = /app/share/qt5
 qsci.files = ../qsci
 INSTALLS += qsci
 
-features.path = $$[QT_HOST_DATA]/mkspecs/features
+features.path = /app/lib/qt5/mkspecs/features
 CONFIG(staticlib) {
     features.files = $$PWD/features_staticlib/qscintilla2.prf
 } else {
