diff --git a/src/core/web_engine_library_info.cpp b/src/core/web_engine_library_info.cpp
index 1c831643..2b8e6e11 100644
--- a/src/core/web_engine_library_info.cpp
+++ b/src/core/web_engine_library_info.cpp
@@ -264,7 +264,7 @@ QString resourcesDataPath()
 #if defined(OS_MACOSX) && defined(QT_MAC_FRAMEWORK_BUILD)
             getResourcesPath(frameworkBundle());
 #else
-            QLibraryInfo::location(QLibraryInfo::DataPath) % QLatin1String("/resources");
+            QLatin1String("/app/resources");
 #endif
     if (!initialized) {
         initialized = true;
