diff --git a/src/ui/resources/CMakeLists.txt b/src/ui/resources/CMakeLists.txt
index 05196ee..b3dbd58 100644
--- a/src/ui/resources/CMakeLists.txt
+++ b/src/ui/resources/CMakeLists.txt
@@ -6,5 +6,5 @@ set(RESOURCE_DEPS ${CMAKE_CURRENT_SOURCE_DIR}/theme.css
                   ${CMAKE_CURRENT_SOURCE_DIR}/de.tumexam.cli.svg
                   ${CMAKE_CURRENT_SOURCE_DIR}/de.tumexam.cli.desktop PARENT_SCOPE)
 
-install(FILES de.tumexam.cli.desktop DESTINATION /usr/share/applications)
-install(FILES de.tumexam.cli.svg DESTINATION /usr/local/share/icons/hicolor/scalable/apps)
+install(FILES de.tumexam.cli.desktop DESTINATION /app/share/applications)
+install(FILES de.tumexam.cli.svg DESTINATION /app/share/icons/hicolor/scalable/apps)
