diff -ruN rhvoice-org/cmake/VersionFromGit.cmake rhvoice-patched/cmake/VersionFromGit.cmake
--- rhvoice-org/cmake/VersionFromGit.cmake	2023-05-20 17:16:22.518560362 +0200
+++ rhvoice-patched/cmake/VersionFromGit.cmake	2023-05-19 20:15:21.026264385 +0200
@@ -12,46 +12,7 @@
 		set(VERSION_EXPORT "${${variablesPrefix}_VERSION_EXPORT}")
 		message(STATUS "Using cached versions: ${VERSION_FROM_GIT}, ${VERSION_TIME}, ${VERSION_EXPORT}")
 	else()
-		if(GIT_FOUND)
-			execute_process(
-				COMMAND ${GIT_EXECUTABLE} describe --dirty --long --tags
-				WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-				OUTPUT_VARIABLE VERSION_FROM_GIT
-				RESULT_VARIABLE GIT_RESULT
-				ERROR_QUIET
-				OUTPUT_STRIP_TRAILING_WHITESPACE
-			)
-			if(NOT GIT_RESULT EQUAL 0)
-				message(WARNING "git returned error ${VERSION_FROM_GIT}")
-				set(VERSION_FROM_GIT "unknown")
-			endif()
-
-			execute_process(
-				COMMAND ${GIT_EXECUTABLE} log -1 --pretty=format:%ct
-				WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-				OUTPUT_VARIABLE VERSION_TIME
-				RESULT_VARIABLE GIT_RESULT
-				ERROR_QUIET
-				OUTPUT_STRIP_TRAILING_WHITESPACE
-			)
-			if(NOT GIT_RESULT EQUAL 0)
-				set(VERSION_TIME "0000000000" PARENT_SCOPE)
-			endif()
-
-			execute_process(
-				COMMAND ${GIT_EXECUTABLE} log -1 --pretty=format:%D
-				WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-				OUTPUT_VARIABLE VERSION_EXPORT
-				RESULT_VARIABLE GIT_RESULT
-				ERROR_QUIET
-				OUTPUT_STRIP_TRAILING_WHITESPACE
-			)
-			if(NOT GIT_RESULT EQUAL 0)
-				set(VERSION_EXPORT "HEAD -> master")
-			endif()
-		else(GIT_FOUND)
-			set(VERSION_FROM_GIT ${defaultVersion} PARENT_SCOPE)
-		endif(GIT_FOUND)
+		set(VERSION_FROM_GIT ${defaultVersion})
 		set(${variablesPrefix}_VERSION_FROM_GIT "${VERSION_FROM_GIT}" CACHE STRING "Version from git")
 		set(${variablesPrefix}_VERSION_TIME "${VERSION_TIME}" CACHE STRING "Commit time from git")
 		set(${variablesPrefix}_VERSION_EXPORT "${VERSION_EXPORT}" CACHE STRING "Commit time from git")
Binary files rhvoice-org/.git/index and rhvoice-patched/.git/index differ
diff -ruN rhvoice-org/src/hts_engine/HTS_engine.c rhvoice-patched/src/hts_engine/HTS_engine.c
--- rhvoice-org/src/hts_engine/HTS_engine.c	2023-05-20 17:16:22.608559590 +0200
+++ rhvoice-patched/src/hts_engine/HTS_engine.c	2023-05-20 13:28:38.314853379 +0200
@@ -74,6 +74,7 @@
 
 HTS_ENGINE_C_START;
 
+#include <locale.h>
 #include <stdlib.h>             /* for atof() */
 #include <string.h>             /* for strcpy() */
 #include <math.h>               /* for pow() */
@@ -81,6 +82,13 @@
 /* hts_engine libraries */
 #include "HTS_hidden.h"
 
+static double locale_independent_atof(const char *str) {
+    const char *old_locale = setlocale(LC_NUMERIC, "C");
+    double value = atof(str);
+    setlocale(LC_NUMERIC, old_locale);
+    return value;
+}
+
 /* HTS_Engine_initialize: initialize engine */
 void HTS_Engine_initialize(HTS_Engine * engine)
 {
@@ -166,7 +174,7 @@
       engine->condition.use_log_gain = atoi(&find[strlen("LN_GAIN=")]) == 1 ? TRUE : FALSE;
    find = strstr(option, "ALPHA=");
    if (find != NULL)
-      engine->condition.alpha = atof(&find[strlen("ALPHA=")]);
+      engine->condition.alpha = locale_independent_atof(&find[strlen("ALPHA=")]);
 
    /* interpolation weights */
    engine->condition.duration_iw = (double *) HTS_calloc(num_voices, sizeof(double));
diff -ruN rhvoice-org/src/lib/CMakeLists.txt rhvoice-patched/src/lib/CMakeLists.txt
--- rhvoice-org/src/lib/CMakeLists.txt	2023-05-20 17:16:22.621892810 +0200
+++ rhvoice-patched/src/lib/CMakeLists.txt	2023-05-21 17:30:03.809199639 +0200
@@ -20,9 +20,13 @@
 target_include_directories(RHVoice PRIVATE "${INCLUDE_DIR}" "${HTS_LABELS_KIT_INCLUDES}")
 target_link_libraries("RHVoice" "RHVoice_core")
 
+file(GLOB RHVoice_INCLUDE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/../include/*.h")
+set_target_properties("RHVoice" PROPERTIES PUBLIC_HEADER "${RHVoice_INCLUDE_FILES}")
+
 install(TARGETS "RHVoice"
 	LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 	COMPONENT "lib"
+	PUBLIC_HEADER DESTINATION include
 )
 
 set("CPACK_DEBIAN_LIB_PACKAGE_NAME" "librhvoice" PARENT_SCOPE)
