
diff -uNrw -X .diff_ignore --no-dereference xtrkcad-fork-xtrkcad/distribution/flatpak/run-xtrkcad.sh new-fork-xtrkcad/distribution/flatpak/run-xtrkcad.sh
--- xtrkcad-fork-xtrkcad/distribution/flatpak/run-xtrkcad.sh	1969-12-31 19:00:00.000000000 -0500
+++ new-fork-xtrkcad/distribution/flatpak/run-xtrkcad.sh	2025-12-05 09:08:50.113933429 -0500
@@ -0,0 +1,3 @@
+#!/usr/bin/env sh
+source $(dirname $0)/common_source.sh
+exec /app/bin/xtrkcad "$@"
