diff --git a/saga-gis/src/tools/io/io_gdal/MLB_Interface.cpp b/saga-gis/src/tools/io/io_gdal/MLB_Interface.cpp
index 9464d086e..270ac6ce2 100644
--- a/saga-gis/src/tools/io/io_gdal/MLB_Interface.cpp
+++ b/saga-gis/src/tools/io/io_gdal/MLB_Interface.cpp
@@ -139,7 +139,7 @@ CSG_Tool *		Create_Tool(int i)
 	case  9:	return( new CGDAL_Import_WMS );
 	case 11:	return( new CGDAL_Import_ASTER );
 
-	#ifdef USE_GDAL_V2
+	#ifdef GDAL_V2_1_OR_NEWER
 	case 12:	return( new CGDAL_BuildVRT );
 	#endif
 	case 13:	return( new CGDAL_Import_VRT );
