--- a/CMakeLists.txt	2024-09-24 02:25:16.101994478 +0300
+++ b/CMakeLists.txt	2024-09-24 02:50:02.126905110 +0300
@@ -105,7 +105,7 @@
     find_package(PkgConfig REQUIRED)
     pkg_check_modules(ARGON2 libargon2 REQUIRED)
     pkg_check_modules(SQLITE sqlite3 REQUIRED)
-    pkg_check_modules(CRYPTOPP libcryptopp REQUIRED)
+    pkg_check_modules(CRYPTOPP cryptopp REQUIRED)
 
     target_include_directories(
         securefs-static SYSTEM
