--- a/CMakeLists.txt.orig	2016-06-28 00:30:39.000000000 +0400
+++ b/CMakeLists.txt	2018-05-02 19:03:55.973596599 +0400
@@ -46,7 +46,11 @@
   ADD_DEFINITIONS (-Wall  -Wextra)
   ADD_DEFINITIONS (-pthread)
 
-  SET( CMAKE_CXX_FLAGS "-fPIC -msse4.1")
+  SET( CMAKE_CXX_FLAGS -fPIC )
+  IF(USE_SSE41)
+    SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.1" )
+    ENDIF()
+
 ENDIF(WIN32)
 
 ## Choose build options
