--- Mnemosyne-2.6.1/setup.py	2017-11-29 09:16:37.000000000 +0100
+++ setup_updated.py	2018-05-14 13:25:10.951156066 +0200
@@ -86,7 +87,7 @@
 if sys.platform != "win32":
     for mo in [x for x in glob.glob(os.path.join("mo", "*"))
                if os.path.isdir(x)]:
-        data_files.append((os.path.join(sys.exec_prefix, "share", "locale",
+        data_files.append((os.path.join("share", "locale",
             os.path.split(mo)[1], "LC_MESSAGES"),
             [os.path.join(mo, "LC_MESSAGES", "mnemosyne.mo")]))
 
