From 66cf66d63dbe88b8af702d9fde0a543bada788e1 Mon Sep 17 00:00:00 2001
From: Didier Raboud <odyx@debian.org>
Date: Sat, 9 Jul 2016 16:24:33 +0200
Subject: Cast ifstream to bool explicitely, to work with GCC-6

Bug-Debian: #811669
---
 tests/libsample/simplefile.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/libsample/simplefile.cpp b/tests/libsample/simplefile.cpp
index deac166..20369f2 100644
--- a/tests/libsample/simplefile.cpp
+++ b/tests/libsample/simplefile.cpp
@@ -90,13 +90,13 @@ bool
 SimpleFile::exists() const
 {
     std::ifstream ifile(p->m_filename);
-    return ifile;
+    return static_cast<bool>(ifile);
 }
 
 bool
 SimpleFile::exists(const char* filename)
 {
     std::ifstream ifile(filename);
-    return ifile;
+    return static_cast<bool>(ifile);
 }
 
