--- a/pdf_viewer_build_config.pro	2022-02-18 07:47:57.000000000 -0400
+++ b/pdf_viewer_build_config.pro	2022-03-07 17:50:49.489579924 -0400
@@ -5,9 +5,9 @@
               zlib
           
 
-QT += core sql opengl gui widgets quickwidgets 3dcore 3danimation 3dextras 3dinput 3dlogic 3drender openglextensions
+QT += core sql opengl gui widgets openglextensions network
 CONFIG += c++17
-DEFINES += QT_3DCORE_LIB QT_3DANIMATION_LIB QT_3DEXTRAS_LIB QT_3DINPUT_LIB QT_3DLOGIC_LIB QT_3DRENDER_LIB QT_OPENGL_LIB QT_OPENGLEXTENSIONS_LIB QT_QUICKWIDGETS_LIB QT_SQL_LIB QT_WIDGETS_LIB
+DEFINES += QT_OPENGL_LIB QT_OPENGLEXTENSIONS_LIB QT_SQL_LIB QT_WIDGETS_LIB QT_NETWORK_LIB
 
 
 CONFIG(non_portable){
@@ -72,7 +72,7 @@
     QMAKE_CXXFLAGS += -std=c++17
     LIBS += -ldl -Lmupdf/build/release -lmupdf -lmupdf-third -lmupdf-threads -lharfbuzz -lz
     isEmpty(PREFIX){
-        PREFIX = /usr
+        PREFIX = /app
     }
     target.path = $$PREFIX/bin
     shortcutfiles.files = resources/sioyek.desktop
--- a/pdf_viewer/ui.h	2022-02-18 07:47:57.000000000 -0400
+++ b/pdf_viewer/ui.h	2022-03-07 17:55:32.000002361 -0400
@@ -14,7 +14,7 @@
 #include <qopenglfunctions.h>
 #include <qopengl.h>
 #include <qwindow.h>
-#include <qkeyevent.h>
+#include <QKeyEvent>
 #include <qlineedit.h>
 #include <qtreeview.h>
 #include <qsortfilterproxymodel.h>
--- a/pdf_viewer/input.cpp	2022-02-18 07:47:57.000000000 -0400
+++ b/pdf_viewer/input.cpp	2022-03-07 18:07:06.847726694 -0400
@@ -4,7 +4,7 @@
 #include <fstream>
 #include <sstream>
 
-#include <qkeyevent.h>
+#include <QKeyEvent>
 //#include <SDL.h>
 #include "input.h"
 
--- a/pdf_viewer/main.cpp	2022-02-18 07:47:57.000000000 -0400
+++ b/pdf_viewer/main.cpp	2022-03-07 17:57:24.106376344 -0400
@@ -29,7 +29,7 @@
 #include <qopenglfunctions.h>
 #include <qopengl.h>
 #include <qwindow.h>
-#include <qkeyevent.h>
+#include <QKeyEvent>
 #include <qlineedit.h>
 #include <qtreeview.h>
 #include <qsortfilterproxymodel.h>
--- a/pdf_viewer/main_widget.cpp	2022-02-18 07:47:57.000000000 -0400
+++ b/pdf_viewer/main_widget.cpp	2022-03-07 18:09:25.807652975 -0400
@@ -14,7 +14,7 @@
 #include <qboxlayout.h>
 #include <qdatetime.h>
 #include <qdesktopwidget.h>
-#include <qkeyevent.h>
+#include <QKeyEvent>
 #include <qlabel.h>
 #include <qlineedit.h>
 #include <qlistview.h>
--- a/pdf_viewer/pdf_view_opengl_widget.h	2022-02-18 07:47:57.000000000 -0400
+++ b/pdf_viewer/pdf_view_opengl_widget.h	2022-03-07 17:59:16.241230870 -0400
@@ -19,7 +19,7 @@
 #include <qopenglfunctions.h>
 #include <qopengl.h>
 #include <qwindow.h>
-#include <qkeyevent.h>
+#include <QKeyEvent>
 #include <qlineedit.h>
 #include <qtreeview.h>
 #include <qsortfilterproxymodel.h>
