From e5c133c958c1a2d811b0fee85b36652b6e8a4866 Mon Sep 17 00:00:00 2001
From: Wilbert Berendsen <wbsoft@xs4all.nl>
Date: Wed, 18 Sep 2019 21:43:11 +0200
Subject: [PATCH 1/2] fix #1180; remove unneeded sip.setapi calls

---
 frescobaldi_app/main.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/frescobaldi_app/main.py b/frescobaldi_app/main.py
index f2e657fb..0507d45e 100644
--- a/frescobaldi_app/main.py
+++ b/frescobaldi_app/main.py
@@ -22,10 +22,6 @@ Entry point of Frescobaldi.
 """
 
 
-import sip
-sip.setapi("QString", 2)
-sip.setapi("QVariant", 2)
-
 import os
 import re
 import sys
-- 
2.21.0


From 92abb8dc7d0522166fdd90fe9921194678e229b1 Mon Sep 17 00:00:00 2001
From: Wilbert Berendsen <wbsoft@xs4all.nl>
Date: Wed, 18 Sep 2019 21:48:42 +0200
Subject: [PATCH 2/2] allow sip being in PyQt5

---
 frescobaldi_app/debuginfo.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/frescobaldi_app/debuginfo.py b/frescobaldi_app/debuginfo.py
index 25a269c2..6a9ba30e 100644
--- a/frescobaldi_app/debuginfo.py
+++ b/frescobaldi_app/debuginfo.py
@@ -48,7 +48,10 @@ def app_version():
 
 @_catch_unknown
 def sip_version():
-    import sip
+    try:
+        import sip
+    except ImportError:
+        import PyQt5.sip as sip
     return sip.SIP_VERSION_STR
 
 @_catch_unknown
-- 
2.21.0

