diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -81,7 +81,7 @@ DMG:=$(DISTDIR)/dmg/libdmg-hfsplus/build/dmg/dmg
 NSIS:=makensis
 LAUNCH4J:=$(DISTDIR)/launch4j/launch4j/launch4j
 
-SKIPS:=
+SKIPS:= -Dasciidoctor.skip=true -Dspotbugs.skip=true -Dlicense.skipDownloadLicenses -Dclirr.skip=true -Dmaven.javadoc.skip=true -Dpmd.skip=true -Dmaven.test.skip=true
 
 # -Dasciidoctor.skip=true
 # -Dspotbugs.skip=true
@@ -95,7 +95,7 @@ jar: SKIPS:=-Dasciidoctor.skip=true -Dspotbugs.skip=true -Dlicense.skipDownloadL
 jar: $(LIBDIR)/Vengine.jar
 
 compile:
-	$(MVN) compile
+	$(MVN) compile $(SKIPS)
 
 version-set:
 	$(MVN) versions:set -DnewVersion=$(MAVEN_VERSION) -DgenerateBackupPoms=false
