--- a/skytemple-rust-1.3.7/setup.py
+++ b/skytemple-rust-1.3.7/setup.py
@@ -13,7 +13,7 @@
 setup(
     name="skytemple-rust",
     version=__version__,
-    rust_extensions=[RustExtension(f"skytemple_rust", binding=Binding.PyO3)],
+    rust_extensions=[RustExtension(f"skytemple_rust", binding=Binding.PyO3, cargo_manifest_args=["--offline"])],
     packages=["skytemple_rust"],
     package_data={"skytemple_rust": ["py.typed", "*.pyi"]},
     description='Binary Rust extension for skytemple-files',
