From db50fad0979b9ff61eba0baa4cbfc0c3eb51b3c9 Mon Sep 17 00:00:00 2001
From: Keith Bloemer <32459398+GuitarML@users.noreply.github.com>
Date: Wed, 31 Aug 2022 10:02:52 -0500
Subject: [PATCH] Update CMakeLists.txt

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f8e8f0e..2a6a941 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(CMAKE_CXX_STANDARD 17)
 add_subdirectory(modules)
 include_directories(modules)
 
-juce_set_aax_sdk_path(C:/SDKs/AAX_SDK/)
+#juce_set_aax_sdk_path(C:/SDKs/AAX_SDK/)
 
 set(JUCE_FORMATS AU VST3)
 
