diff --git a/CMakeLists.txt b/CMakeLists.txt
index f8e8f0e..3d01d26 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,10 +24,10 @@ endif()
 juce_add_plugin(SmartAmp
     COMPANY_NAME GuitarML
     PLUGIN_MANUFACTURER_CODE GtML
-    PLUGIN_CODE Npi3 
+    PLUGIN_CODE Npi3
     FORMATS ${JUCE_FORMATS}
     ProductName "SmartAmp"
-    LV2_URI https://github.com/GuitarML/SmartAmp
+    LV2URI https://github.com/GuitarML/SmartAmp
     ICON_BIG resources/logo.png
     MICROPHONE_PERMISSION_ENABLED TRUE
 )
