From: Stefano Rivera <stefanor@debian.org>
Date: Sun, 24 Jan 2021 19:07:24 -0700
Subject: Honor build flags passed as command line arguments

Forwarded: not-needed
---
 GNUmakefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index 98eb1fa..d5e29cd 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -170,8 +170,8 @@ C_OTHER_OBJECTS = $(C_OTHER_SOURCES:.c=.o)
 JAVA_CLASSES = $(JAVA_SOURCES:.java=.class)
 JAVA_RUNTIME_CLASSES=$(JAVARUNTIME_SOURCES:.java=.class)
 
-CFLAGS=-O2 -W -Wall -Wmissing-prototypes -Wmissing-declarations
-CPPFLAGS=
+override CFLAGS += -O2 -W -Wall -Wmissing-prototypes -Wmissing-declarations
+CPPFLAGS?=
 
 INCLUDES=-Iinclude
 
