diff --git a/gui/sorcer_ui.cxx b/gui/sorcer_ui.cxx
index 60edbe4..3bd25a3 100644
--- a/gui/sorcer_ui.cxx
+++ b/gui/sorcer_ui.cxx
@@ -44,7 +44,7 @@ typedef struct {
   LV2UI_Controller controller;
 } SorcerGUI;
 
-static LV2UI_Handle instantiate(const struct _LV2UI_Descriptor * descriptor,
+static LV2UI_Handle instantiate(const struct LV2UI_Descriptor * descriptor,
                 const char * plugin_uri,
                 const char * bundle_path,
                 LV2UI_Write_Function write_function,
