diff -up Squeak-4.10.2.2614-src-no-mp3/unix/CMakeLists.txt.orig Squeak-4.10.2.2614-src-no-mp3/unix/CMakeLists.txt
--- Squeak-4.10.2.2614-src-no-mp3/unix/CMakeLists.txt.orig	2012-09-17 03:03:14.000000000 +0200
+++ Squeak-4.10.2.2614-src-no-mp3/unix/CMakeLists.txt	2012-11-22 18:21:16.413017772 +0100
@@ -135,9 +139,9 @@ SET (ioUtcWithOffset sqUnixUtcWithOffset
 CONFIG_DEFINE (ioUtcWithOffset)
 
 SET (CMAKE_SKIP_BUILD_RPATH TRUE)
-SET (CMAKE_BUILD_WITH_INSTALL_RPATH TRUE) 
+SET (CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
 SET (CMAKE_INSTALL_RPATH "")
-SET (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+SET (CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
 
 INCLUDE (${unix}/vm/config.cmake)
 
