diff --git a/SRC/cpanel_bmod.c b/SRC/cpanel_bmod.c
index fdb9696..7755411 100644
--- a/SRC/cpanel_bmod.c
+++ b/SRC/cpanel_bmod.c
@@ -43,7 +43,6 @@ at the top-level directory.
  */
 void clsolve(int, int, complex *, complex *);
 void cmatvec(int, int, int, complex *, complex *, complex *);
-extern void ccheck_tempv();
 
 /*! \brief
  *
diff --git a/SRC/creadtriple.c b/SRC/creadtriple.c
index 97cf70a..e8c6f22 100644
--- a/SRC/creadtriple.c
+++ b/SRC/creadtriple.c
@@ -134,7 +134,7 @@ creadtriple(int *m, int *n, int *nonz,
 
 void creadrhs(int m, complex *b)
 {
-    FILE *fp, *fopen();
+    FILE *fp;
     int i;
     /*int j;*/
 
diff --git a/SRC/dpanel_bmod.c b/SRC/dpanel_bmod.c
index 2d7c016..2b50913 100644
--- a/SRC/dpanel_bmod.c
+++ b/SRC/dpanel_bmod.c
@@ -43,7 +43,6 @@ at the top-level directory.
  */
 void dlsolve(int, int, double *, double *);
 void dmatvec(int, int, int, double *, double *, double *);
-extern void dcheck_tempv();
 
 /*! \brief
  *
diff --git a/SRC/dreadtriple.c b/SRC/dreadtriple.c
index 3e2451f..549755e 100644
--- a/SRC/dreadtriple.c
+++ b/SRC/dreadtriple.c
@@ -134,7 +134,7 @@ dreadtriple(int *m, int *n, int *nonz,
 
 void dreadrhs(int m, double *b)
 {
-    FILE *fp, *fopen();
+    FILE *fp;
     int i;
     /*int j;*/
 
diff --git a/SRC/spanel_bmod.c b/SRC/spanel_bmod.c
index b1e7f91..3fb7218 100644
--- a/SRC/spanel_bmod.c
+++ b/SRC/spanel_bmod.c
@@ -43,7 +43,6 @@ at the top-level directory.
  */
 void slsolve(int, int, float *, float *);
 void smatvec(int, int, int, float *, float *, float *);
-extern void scheck_tempv();
 
 /*! \brief
  *
diff --git a/SRC/sreadtriple.c b/SRC/sreadtriple.c
index a3d9233..0a21ec1 100644
--- a/SRC/sreadtriple.c
+++ b/SRC/sreadtriple.c
@@ -134,7 +134,7 @@ sreadtriple(int *m, int *n, int *nonz,
 
 void sreadrhs(int m, float *b)
 {
-    FILE *fp, *fopen();
+    FILE *fp;
     int i;
     /*int j;*/
 
diff --git a/SRC/zpanel_bmod.c b/SRC/zpanel_bmod.c
index 9e21cab..396a1e1 100644
--- a/SRC/zpanel_bmod.c
+++ b/SRC/zpanel_bmod.c
@@ -43,7 +43,6 @@ at the top-level directory.
  */
 void zlsolve(int, int, doublecomplex *, doublecomplex *);
 void zmatvec(int, int, int, doublecomplex *, doublecomplex *, doublecomplex *);
-extern void zcheck_tempv();
 
 /*! \brief
  *
diff --git a/SRC/zreadtriple.c b/SRC/zreadtriple.c
index fc834ec..599d0b9 100644
--- a/SRC/zreadtriple.c
+++ b/SRC/zreadtriple.c
@@ -134,7 +134,7 @@ zreadtriple(int *m, int *n, int *nonz,
 
 void zreadrhs(int m, doublecomplex *b)
 {
-    FILE *fp, *fopen();
+    FILE *fp;
     int i;
     /*int j;*/
 
