From ffdd8433509c92fc634bb1eeeb058b0f15363966 Mon Sep 17 00:00:00 2001
From: manongjohn <19245851+manongjohn@users.noreply.github.com>
Date: Sun, 24 Dec 2023 10:34:10 -0500
Subject: [PATCH 1/2] Add Flatpak bin directory for ffmpeg & rhubarb detection

---
 toonz/sources/toonzlib/thirdparty.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/toonz/sources/toonzlib/thirdparty.cpp b/toonz/sources/toonzlib/thirdparty.cpp
index 36691317d..30a3f2ef6 100644
--- a/toonz/sources/toonzlib/thirdparty.cpp
+++ b/toonz/sources/toonzlib/thirdparty.cpp
@@ -110,6 +110,7 @@ QString autodetectFFmpeg() {
 #endif
 
 #ifndef _WIN32
+  folderList.append("/app/bin");
   folderList.append("/usr/local/bin");
   folderList.append("/usr/bin");
   folderList.append("/bin");
@@ -282,6 +283,7 @@ QString autodetectRhubarb() {
 #endif
 
 #ifndef _WIN32
+  folderList.append("/app/bin");
   folderList.append("/usr/local/bin");
   folderList.append("/usr/bin");
   folderList.append("/bin");

From 18a0f0b7fa4a76de24d9e903fb842ea380852395 Mon Sep 17 00:00:00 2001
From: manongjohn <19245851+manongjohn@users.noreply.github.com>
Date: Tue, 26 Dec 2023 22:40:03 -0500
Subject: [PATCH 2/2] Fix autodetecting and setting ffmpeg/rhubarb path

---
 toonz/sources/toonzlib/thirdparty.cpp | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/toonz/sources/toonzlib/thirdparty.cpp b/toonz/sources/toonzlib/thirdparty.cpp
index 30a3f2ef6..fa84c7501 100644
--- a/toonz/sources/toonzlib/thirdparty.cpp
+++ b/toonz/sources/toonzlib/thirdparty.cpp
@@ -75,10 +75,10 @@ bool findFFmpeg(QString dir) {
 
 bool checkFFmpeg() {
   // Path in preferences
-  bool found = findFFmpeg(Preferences::instance()->getFfmpegPath());
-  if (found) return true;
+  QString path = Preferences::instance()->getFfmpegPath();
+  if (!path.isEmpty() && findFFmpeg(path)) return true;
 
-  QString path = autodetectFFmpeg();
+  path = autodetectFFmpeg();
   if (path.isEmpty()) return false;
 
   setFFmpegDir(path);
@@ -90,7 +90,7 @@ bool checkFFmpeg() {
 
 QString autodetectFFmpeg() {
   QString dir = Preferences::instance()->getFfmpegPath();
-  if (findFFmpeg(dir)) return dir;
+  if (!dir.isEmpty() && findFFmpeg(dir)) return dir;
 
   // Let's try and autodetect the exe included with release
   QStringList folderList;
@@ -249,10 +249,10 @@ bool findRhubarb(QString dir) {
 
 bool checkRhubarb() {
   // Path in preferences
-  bool found = findRhubarb(Preferences::instance()->getRhubarbPath());
-  if (found) return true;
+  QString path = Preferences::instance()->getRhubarbPath();
+  if (!path.isEmpty() && findRhubarb(path)) return true;
 
-  QString path = autodetectRhubarb();
+  path = autodetectRhubarb();
   if (path.isEmpty()) return false;
 
   setRhubarbDir(path);
@@ -263,7 +263,7 @@ bool checkRhubarb() {
 
 QString autodetectRhubarb() {
   QString dir = Preferences::instance()->getRhubarbPath();
-  if (findRhubarb(dir)) return dir;
+  if (!dir.isEmpty() && findRhubarb(dir)) return dir;
 
   // Let's try and autodetect the exe included with release
   QStringList folderList;