diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49ac81b..a1ee31f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(CMAKE_CXX_STANDARD 17)
 add_subdirectory(modules)
 include_directories(modules)
 
-juce_set_aax_sdk_path(C:/SDKs/AAX_SDK/)
+#juce_set_aax_sdk_path(C:/SDKs/AAX_SDK/)
 
 set(JUCE_FORMATS AU VST3)
 
@@ -29,7 +29,7 @@ juce_add_plugin(TS-M1N3
     PLUGIN_CODE Tsm3 
     FORMATS ${JUCE_FORMATS}
     ProductName "TS-M1N3"
-    LV2_URI https://github.com/GuitarML/TS-M1N3
+    LV2URI https://github.com/GuitarML/TS-M1N3
     ICON_BIG resources/logo.png
     MICROPHONE_PERMISSION_ENABLED TRUE
 )
