From 8e21f259ac373f17e84b3a29970020afb263ca96 Mon Sep 17 00:00:00 2001
From: PolyCat <contact.polycat@proton.me>
Date: Fri, 17 Apr 2026 13:19:47 +0300
Subject: [PATCH] version-number-injector

---
 chirp/__init__.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/chirp/__init__.py b/chirp/__init__.py
index 3f25801d..5765736e 100644
--- a/chirp/__init__.py
+++ b/chirp/__init__.py
@@ -13,4 +13,6 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-CHIRP_VERSION = "py3dev"
+from os import environ
+
+CHIRP_VERSION = environ.get("VERSION_NUMBER", "py3dev")
-- 
2.53.0

