diff --git a/GUI.fl b/GUI.fl
index 9dfa388..febb297 100644
--- a/GUI.fl
+++ b/GUI.fl
@@ -45,8 +45,8 @@ decl {\#include "version.h"} {public
 Function {hex4n(char c)} {return_type int
 } {
   code {if (c>96) c-=32;
-if ((c>='0')||(c<='9')) return (c-'0');
-if ((c>='A')||(c<='F')) return (c-'A')+10;
+if ((c>='0')&&(c<='9')) return (c-'0');
+if ((c>='A')&&(c<='F')) return (c-'A')+10;
 
 return 0;} {}
 } 
diff --git a/compile_linux_fftw.sh b/compile_linux_fftw.sh
index 13adc03..fee7b4e 100755
--- a/compile_linux_fftw.sh
+++ b/compile_linux_fftw.sh
@@ -6,7 +6,7 @@ fluid -c GUI.fl
 fluid -c FreeEditUI.fl
 
 g++ -ggdb GUI.cxx FreeEditUI.cxx *.cpp Input/*.cpp Output/*.cpp `fltk-config --cflags` \
- `fltk-config --ldflags`  -laudiofile -lfftw3f -lvorbisenc -lvorbisfile -lportaudio -lpthread -lmad -lmxml -o $outfile
+ `fltk-config --ldflags`  -laudiofile -lfftw3f -lz -logg -lvorbis -lvorbisenc -lvorbisfile -lportaudio -lpthread -lmad -lmxml -o $outfile
 
 rm -f GUI.h GUI.cxx FreeEditUI.h FreeEditUI.cxx
 
diff --git a/compile_linux_fftw_jack.sh b/compile_linux_fftw_jack.sh
index 1a03497..b9e6a33 100755
--- a/compile_linux_fftw_jack.sh
+++ b/compile_linux_fftw_jack.sh
@@ -7,7 +7,7 @@ fluid -c FreeEditUI.fl
 
 g++ -ggdb GUI.cxx FreeEditUI.cxx *.cpp Input/*.cpp Output/*.cpp `fltk-config --cflags` \
  `fltk-config --ldflags` \
- -laudiofile -lfftw3f -lvorbisenc -lvorbisfile -lportaudio -lpthread -lmad -lmxml \
+ -laudiofile -lfftw3f -lz -logg -lvorbis  -lvorbisenc -lvorbisfile -lportaudio -lpthread -lmad -lmxml \
  `pkg-config --cflags --libs jack samplerate` \
  -DHAVE_JACK -DENABLE_RESAMPLING \
  -o $outfile
diff --git a/compile_linux_kissfft.sh b/compile_linux_kissfft.sh
index 16dc0b2..37118c3 100755
--- a/compile_linux_kissfft.sh
+++ b/compile_linux_kissfft.sh
@@ -6,6 +6,6 @@ fluid -c FreeEditUI.fl
 
 
 g++ -ggdb -DKISSFFT -I./contrib GUI.cxx FreeEditUI.cxx *.cpp Input/*.cpp Output/*.cpp contrib/*.c `fltk-config --cflags` \
- `fltk-config --ldflags`  -laudiofile  -lvorbisenc -lvorbisfile -lportaudio -lpthread -lmad -lmxml -o $outfile
+ `fltk-config --ldflags` -lz -logg -lvorbis   -laudiofile  -lvorbisenc -lvorbisfile -lportaudio -lpthread -lmad -lmxml -o $outfile
 
 rm -f GUI.h GUI.cxx FreeEditUI.h FreeEditUI.cxx
