diff --git a/cmake/modules/SMTG_PlatformToolset.cmake b/cmake/modules/SMTG_PlatformToolset.cmake
index 9f0bb96..a788176 100644
--- a/cmake/modules/SMTG_PlatformToolset.cmake
+++ b/cmake/modules/SMTG_PlatformToolset.cmake
@@ -58,7 +58,7 @@ macro(smtg_setup_platform_toolset)
                     set(CMAKE_ANDROID_STL_TYPE c++_static)
                     link_libraries(dl)
                  else()
-                    link_libraries(stdc++fs pthread dl)
+                    link_libraries(pthread dl)
                 endif(ANDROID)
             endif(SMTG_MAC)
         endif(XCODE)
