--- vst3sdk/vstgui4/vstgui/lib/vstguifwd.h-orig	2022-04-10 18:42:46.993868834 -0400
+++ vst3sdk/vstgui4/vstgui/lib/vstguifwd.h	2022-04-10 18:43:04.854030034 -0400
@@ -5,6 +5,7 @@
 #pragma once
 
 #include "vstguibase.h"
+#include <limits>
 #include <functional>
 #include <memory>
 
--- vst3sdk/public.sdk/source/vst/moduleinfo/moduleinfoparser.cpp-orig	2023-02-08 23:57:21.862745883 -0500
+++ vst3sdk/public.sdk/source/vst/moduleinfo/moduleinfoparser.cpp	2023-02-08 23:57:44.943101631 -0500
@@ -39,6 +39,7 @@
 #include "jsoncxx.h"
 #include "pluginterfaces/base/ipluginbase.h"
 #include <stdexcept>
+#include <limits>
 
 //------------------------------------------------------------------------
 namespace Steinberg::ModuleInfoLib {
