--- webkitgtk/Source/WebCore/platform/graphics/SegmentedFontData.cpp-old	2021-03-25 17:15:13.708931765 -0400
+++ webkitgtk/Source/WebCore/platform/graphics/SegmentedFontData.cpp	2021-03-25 17:15:33.537092264 -0400
@@ -61,7 +61,7 @@
 {
     UChar32 c;
     for (int i = 0; i < length; ) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!containsCharacter(c))
             return false;
     }
--- webkitgtk/Source/WebCore/dom/Document.cpp-old	2021-03-25 17:51:55.769204459 -0400
+++ webkitgtk/Source/WebCore/dom/Document.cpp	2021-03-25 17:51:22.560931015 -0400
@@ -3912,12 +3912,12 @@
     unsigned i = 0;
 
     UChar32 c;
-    U16_NEXT(characters, i, length, c)
+    U16_NEXT(characters, i, length, c);
     if (!isValidNameStart(c))
         return false;
 
     while (i < length) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!isValidNamePart(c))
             return false;
     }
@@ -3980,7 +3980,7 @@
     const UChar* s = qualifiedName.deprecatedCharacters();
     for (unsigned i = 0; i < length;) {
         UChar32 c;
-        U16_NEXT(s, i, length, c)
+        U16_NEXT(s, i, length, c);
         if (c == ':') {
             if (sawColon) {
                 ec = NAMESPACE_ERR;
