diff --git a/Source/WebCore/css/CSSGrammar.y.in b/Source/WebCore/css/CSSGrammar.y.in
index 0aad4566359..a3033a778ee 100644
--- a/Source/WebCore/css/CSSGrammar.y.in
+++ b/Source/WebCore/css/CSSGrammar.y.in
@@ -21,7 +21,8 @@
  *
  */
 
-%pure_parser
+%define api.pure
+%define api.header.include {"CSSGrammar.h"}
 
 %parse-param { CSSParser* parser }
 %lex-param { CSSParser* parser }
diff --git a/Source/WebCore/xml/XPathGrammar.y b/Source/WebCore/xml/XPathGrammar.y
index 356e11889ed..f098b4dac32 100644
--- a/Source/WebCore/xml/XPathGrammar.y
+++ b/Source/WebCore/xml/XPathGrammar.y
@@ -47,7 +47,8 @@
 
 %}
 
-%pure_parser
+%define api.pure
+%define api.header.include {"XPathGrammar.h"}
 %lex-param { parser }
 %parse-param { Parser& parser }

diff --git a/Source/WebCore/platform/text/icu/UTextProvider.h b/Source/WebCore/platform/text/icu/UTextProvider.h
--- a/Source/WebCore/platform/text/icu/UTextProvider.h	2021-12-20 13:32:23.148669678 -0500
+++ b/Source/WebCore/platform/text/icu/UTextProvider.h	2021-12-20 13:32:59.745965735 -0500
@@ -28,6 +28,9 @@
 
 #include <unicode/utext.h>
 #include <wtf/unicode/Unicode.h>
+#include <unicode/umachine.h>
+#define TRUE 1
+#define FALSE 0
 
 namespace WebCore {
 
diff --git a/Source/WebCore/platform/text/TetCodecICU.cpp b/Source/WebCore/platform/text/TextCodecICU.cpp
--- a/Source/WebCore/platform/text/TextCodecICU.cpp	2021-12-20 14:22:54.095676115 -0500
+++ b/Source/WebCore/platform/text/TextCodecICU.cpp	2021-12-20 14:23:08.653794005 -0500
@@ -308,7 +308,7 @@
     m_converterICU = ucnv_open(m_canonicalConverterName, &err);
     ASSERT(U_SUCCESS(err));
     if (m_converterICU)
-        ucnv_setFallback(m_converterICU, TRUE);
+        ucnv_setFallback(m_converterICU, true);
 }
 
 int TextCodecICU::decodeToBuffer(UChar* target, UChar* targetLimit, const char*& source, const char* sourceLimit, int32_t* offsets, bool flush, UErrorCode& err)
