diff --git a/src/webots/Makefile b/src/webots/Makefile
index 945f945..8884407 100644
--- a/src/webots/Makefile
+++ b/src/webots/Makefile
@@ -117,10 +117,10 @@ QT_XML_INCLUDE     = -isystem $(WEBOTS_PATH)/include/qt/QtXml
 FREETYPE_INCLUDE   = -I/usr/include/freetype2
 TARGET             = $(WEBOTS_PATH)/bin/webots-bin
 CFLAGS             = -fPIC
-CXXFLAGS           = -std=c++11
+CXXFLAGS           = -std=c++11 -Wno-deprecated-declarations
 MOC                = $(WEBOTS_PATH)/bin/qt/moc
-LIBS              += $(LIB_WREN) -Wl,-rpath $(WEBOTS_LIB_PATH) -L$(WEBOTS_LIB_PATH) -lQt6Core -lQt6Network -lQt6Gui -lQt6OpenGL -lQt6OpenGLWidgets -lQt6WebSockets -lQt6Widgets -lQt6PrintSupport -lQt6Qml -lQt6Xml -lopenal -lGL -lGLU -ldl -lOIS -lpico -lfreetype -lassimp -lrt
-LD_FLAGS           = -rdynamic
+LIBS              += $(LIB_WREN) -Wl,-rpath $(WEBOTS_LIB_PATH) -L$(WEBOTS_LIB_PATH) -lQt6Core -lQt6Network -lQt6Gui -lQt6OpenGL -lQt6OpenGLWidgets -lQt6WebSockets -lQt6Widgets -lQt6PrintSupport -lQt6Qml -lQt6Xml -lopenal -lGL -lGLU -ldl -lOIS -lttspico -lfreetype -lassimp -lrt
+LD_FLAGS           = -rdynamic -L/app/lib
 EXTRA_CMD          = cp launcher/webots-linux.sh $(WEBOTS_PATH)/webots && chmod 755 $(WEBOTS_PATH)/webots
 FILES_TO_REMOVE    = $(WEBOTS_PATH)/webots
 ifeq ($(MAKECMDGOALS), profile)
