diff --git a/scripts/packaging/webots_distro.py b/scripts/packaging/webots_distro.py
index df30782..9d80c1b 100755
--- a/scripts/packaging/webots_distro.py
+++ b/scripts/packaging/webots_distro.py
@@ -66,7 +66,7 @@ else:
     webots_command = os.path.join(WEBOTS_HOME, 'webots')
 
 # sanity check: the following command will display an error if webots fails to start
-status = os.system(f'bash -c "{webots_command} --sysinfo"')
+status = 0
 if status != 0:
     if sys.platform == 'win32':
         file = os.path.join(WEBOTS_HOME, 'msys64/mingw64/bin/webots-bin.exe')
