diff -ruN a/python/lib/lng.py b/python/lib/lng.py
--- a/python/lib/lng.py	2020-05-17 16:05:33.000000000 +0200
+++ b/python/lib/lng.py	2023-07-19 15:56:50.021695465 +0200
@@ -13,7 +13,7 @@
 class iLang(object):
     def __init__(self):
         if(os.environ["DEBIAN_PACKAGE"] == "TRUE"):
-            languages = os.listdir('/usr/share/locale')
+            languages = os.listdir('/app/share/locale')
         else:
             languages = os.listdir(Variables.playonlinux_env+'/lang/locale')
 
@@ -28,7 +28,7 @@
             langid = wx.LANGUAGE_DEFAULT
 
         if(os.environ["DEBIAN_PACKAGE"] == "TRUE"):
-            localedir = "/usr/share/locale"
+            localedir = "/app/share/locale"
         else:
             localedir = os.path.join(Variables.playonlinux_env, "lang/locale")
 
