--- x42-plugins/convoLV2/Makefile	2020-03-17 08:44:16.000000000 -0400
+++ x42-plugins/convoLV2/Makefile-new	2020-07-02 22:24:13.387006295 -0400
@@ -76,7 +76,7 @@
 endif
 
 ifeq ($(LIBZITACONVOLVER),)
-  ifeq ($(shell test -f /usr/include/zita-convolver.h -o -f /usr/local/include/zita-convolver.h || echo no ), no)
+  ifeq ($(shell test -f $(PREFIX)/include/zita-convolver.h -o -f /usr/local/include/zita-convolver.h || echo no ), no)
     $(error "libzita-convolver3 or 4, is required")
   endif
   LOADLIBES += -lzita-convolver
