diff -ur xapian-core-1.4.7/backends/glass/glass_dbcheck.h xapian-core-1.4.7.patched/backends/glass/glass_dbcheck.h
--- xapian-core-1.4.7/backends/glass/glass_dbcheck.h	2017-10-16 04:32:24.000000000 +0200
+++ xapian-core-1.4.7.patched/backends/glass/glass_dbcheck.h	2018-06-13 14:55:58.574898188 +0200
@@ -24,7 +24,7 @@
 
 #include "xapian/types.h"
 
-#include <cstring> // For size_t.
+#include <sys/types.h> // For size_t and off_t.
 #include <iosfwd>
 #include <string>
 #include <vector>
