diff -ruN a/playonlinux b/playonlinux
--- a/playonlinux	2020-05-17 16:05:33.000000000 +0200
+++ b/playonlinux	2023-07-04 20:11:00.935584956 +0200
@@ -108,16 +108,16 @@
 	else
 		# SetDebugState(game, False)
 		# export WINEDEBUG= found and is not WINEDEBUG="-all"?
-		if grep '^export WINEDEBUG=' "$HOME/.PlayOnLinux/shortcuts/$NAME"|grep -qv 'WINEDEBUG="-all"'; then
-			cp "$HOME/.PlayOnLinux/shortcuts/$NAME" "$HOME/.PlayOnLinux/tmp/debug_$NAME" && \
-			sed -e 's/^export WINEDEBUG="\(.*\)"/export WINEDEBUG="-all"/' "$HOME/.PlayOnLinux/tmp/debug_$NAME" > "$HOME/.PlayOnLinux/shortcuts/$NAME"
+		if grep '^export WINEDEBUG=' "$XDG_DATA_HOME/PlayOnLinux/shortcuts/$NAME"|grep -qv 'WINEDEBUG="-all"'; then
+			cp "$XDG_DATA_HOME/PlayOnLinux/shortcuts/$NAME" "$XDG_DATA_HOME/PlayOnLinux/tmp/debug_$NAME" && \
+			sed -e 's/^export WINEDEBUG="\(.*\)"/export WINEDEBUG="-all"/' "$XDG_DATA_HOME/PlayOnLinux/tmp/debug_$NAME" > "$XDG_DATA_HOME/PlayOnLinux/shortcuts/$NAME"
 		fi
 
-		if [ -e "$HOME/.PlayOnLinux/configurations/pre_shortcut/$NAME" ]; then
-			source "$HOME/.PlayOnLinux/configurations/pre_shortcut/$NAME"
+		if [ -e "$XDG_DATA_HOME/PlayOnLinux/configurations/pre_shortcut/$NAME" ]; then
+			source "$XDG_DATA_HOME/PlayOnLinux/configurations/pre_shortcut/$NAME"
 		fi
 
-		exec ./playonlinux-bash "$HOME/.PlayOnLinux/shortcuts/$NAME" "$@"
+		exec ./playonlinux-bash "$XDG_DATA_HOME/PlayOnLinux/shortcuts/$NAME" "$@"
 	fi
 else
 	cd "$POLDIR/python"
diff -ruN a/python/lib/Variables.py b/python/lib/Variables.py
--- a/python/lib/Variables.py	2020-05-17 16:05:33.000000000 +0200
+++ b/python/lib/Variables.py	2023-07-04 20:12:31.231915596 +0200
@@ -43,7 +43,7 @@
 
 # Variables PlayOnLinux
 if os.environ["POL_OS"] == "Linux":
-    os.environ["POL_USER_ROOT"] = os.environ["HOME"]+"/.PlayOnLinux/"
+    os.environ["POL_USER_ROOT"] = os.environ["XDG_DATA_HOME"]+"/PlayOnLinux/"
     os.environ["APPLICATION_TITLE"] = "PlayOnLinux"
     os.environ["POL_DNS"] = "playonlinux.com"
     os.environ["WINE_SITE"] = "https://phoenicis.playonlinux.com/index.php/wine?os=linux"
