--- xjadeo/src/xjadeo/configfile.c	2015-11-10 15:54:37.000000000 -0500
+++ xjadeo/src/xjadeo/configfile.c-new	2020-12-28 16:07:43.152482321 -0500
@@ -90,8 +90,8 @@
 extern char   OSD_text[128];
 extern int    OSD_fx, OSD_tx, OSD_sx, OSD_fy, OSD_sy, OSD_ty;
 
-int start_ontop;
-int start_fullscreen;
+extern int start_ontop;
+extern int start_fullscreen;
 
 extern char	*current_file;
 #ifdef JACK_SESSION
