
diff -uNrw -X .diff_ignore --no-dereference xtrkcad-fork-xtrkcad/app/lib/xtrkcad.desktop new-fork-xtrkcad/app/lib/xtrkcad.desktop
--- xtrkcad-fork-xtrkcad/app/lib/xtrkcad.desktop	2025-11-21 20:39:19.129449339 -0500
+++ new-fork-xtrkcad/app/lib/xtrkcad.desktop	2025-12-05 06:12:07.796856977 -0500
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Name=XTrackCAD
 Comment=Design model railroad layouts
-Exec=xtrkcad
-Icon=xtrkcad
+Exec=run-xtrkcad.sh
+Icon=io.sourceforge.xtrkcad_fork.xtrkcad
 Terminal=false
 Type=Application
 Categories=Graphics
-
