--- a/configure.ac
+++ b/configure.ac
@@ -169,7 +169,7 @@
   [AC_CHECK_HEADERS([pthread.h], [],
      [AC_MSG_FAILURE([test for pthread support failed!
 configure --disable-pthread to skip threaded support.])])
-   AC_CHECK_LIB([pthread], [pthread_create], [],
+   AC_CHECK_LIB([c], [pthread_create], [],
      [AC_MSG_FAILURE([unable to link against -lpthread, although you
 appear to have pthread.h? set LDFLAGS and/or LIBS to help the linker,
 or configure --disable-pthread to skip threaded support.])])
