--- zita-resampler-1.6.2/source/Makefile	2018-08-24 14:41:47.000000000 -0400
+++ zita-resampler-1.6.2/source/Makefile-new	2020-05-03 22:12:43.207618109 -0400
@@ -38,6 +38,7 @@
 LDLIBS +=
 
 
+ZITA-RESAMPLER_A = libzita-resampler.a
 ZITA-RESAMPLER_SO = libzita-resampler.so
 ZITA-RESAMPLER_MAJ = $(ZITA-RESAMPLER_SO).$(MAJVERS)
 ZITA-RESAMPLER_MIN = $(ZITA-RESAMPLER_MAJ).$(MINVERS)
@@ -47,19 +48,19 @@
 	zita-resampler/vresampler.h zita-resampler/cresampler.h
 
 
-$(ZITA-RESAMPLER_MIN): $(ZITA-RESAMPLER_O)
-	$(CXX) -shared $(LDFLAGS) -Wl,-soname,$(ZITA-RESAMPLER_MAJ) -o $(ZITA-RESAMPLER_MIN) $(ZITA-RESAMPLER_O) $(ZITA-RESAMPLER_DEP)
+$(ZITA-RESAMPLER_A): $(ZITA-RESAMPLER_O)
+	$(AR) rcs $(ZITA-RESAMPLER_A) $(ZITA-RESAMPLER_O)
 
 $(ZITA-RESAMPLER_O):	$(ZITA-RESAMPLER_H)
 
 
-install:	$(ZITA-RESAMPLER_MIN)
+install:	$(ZITA-RESAMPLER_A)
 	install -d $(DESTDIR)$(INCDIR)/zita-resampler
 	install -d $(DESTDIR)$(LIBDIR)
 	install -m 644 $(ZITA-RESAMPLER_H)   $(DESTDIR)$(INCDIR)/zita-resampler
-	install -m 755 $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)
-	ldconfig
-	ln -sf $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-RESAMPLER_SO)
+	install -m 755 $(ZITA-RESAMPLER_A) $(DESTDIR)$(LIBDIR)
+	#ldconfig
+	#ln -sf $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-RESAMPLER_SO)
 
 uninstall:
 	/bin/rm -rf $(DESTDIR)$(INCDIR)/zita-resampler
