--- a/./src/mobile-config-autoconfig.js
+++ b/./src/mobile-config-autoconfig.js
@@ -214,6 +214,8 @@ function css_file_merge(name, file) {
                   createInstance(Components.interfaces.nsIFileOutputStream);
     ostream.init(file, 0x02 | 0x08 | 0x20, 0644, 0);

+    write_line(ostream, '@import url("file:///app/lib/firefox-gnome-theme/' + name + '.css");');
+
     var fragments = css_file_get_fragments(name);
     for (var i in fragments) {
         var line;

