diff --git i/include/sol/compatibility/lua_version.hpp w/include/sol/compatibility/lua_version.hpp
index b0bf1e45..9b39811a 100644
--- i/include/sol/compatibility/lua_version.hpp
+++ w/include/sol/compatibility/lua_version.hpp
@@ -192,7 +192,7 @@
 #else
 	// Lua 5.2 only (deprecated in 5.3 (503)) (Can be turned on with Compat flags)
 	// Lua 5.2, or other versions of Lua with the compat flag, or Lua that is not 5.2 with the specific define (5.4.1 either removed it entirely or broke it)
-	#if (SOL_LUA_VERSION_I_ == 502) || (defined(LUA_COMPAT_BITLIB) && (LUA_COMPAT_BITLIB != 0)) || (SOL_LUA_VERSION_I_ < 504 && (defined(LUA_COMPAT_5_2) && (LUA_COMPAT_5_2 != 0)))
+	#if (SOL_LUA_VERSION_I_ == 502) || (defined(LUA_COMPAT_BITLIB)) || (SOL_LUA_VERSION_I_ < 504 && (defined(LUA_COMPAT_5_2)))
 		#define SOL_LUA_BIT32_LIB_I_ SOL_ON
 	#else
 		#define SOL_LUA_BIT32_LIB_I_ SOL_DEFAULT_OFF
