From cdb3ae92eba9ee22ed5b264784e2c1bf72541666 Mon Sep 17 00:00:00 2001
From: Dylan McCall <dylan@endlessos.org>
Date: Thu, 2 Dec 2021 15:31:30 -0800
Subject: [PATCH] Allow superuser to be null in device provision API

---
 kolibri/core/device/serializers.py | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/kolibri/core/device/serializers.py b/kolibri/core/device/serializers.py
index 9103e49f93..c84de06979 100644
--- a/kolibri/core/device/serializers.py
+++ b/kolibri/core/device/serializers.py
@@ -40,8 +40,8 @@ class DeviceSerializerMixin(object):
 class DeviceProvisionSerializer(DeviceSerializerMixin, serializers.Serializer):
     facility = FacilitySerializer()
     preset = serializers.ChoiceField(choices=choices)
-    superuser = NoFacilityFacilityUserSerializer()
-    language_id = serializers.CharField(max_length=15)
+    superuser = NoFacilityFacilityUserSerializer(allow_null=True)
+    language_id = serializers.CharField(max_length=15, allow_null=True)
     device_name = serializers.CharField(max_length=50, allow_null=True)
     settings = serializers.JSONField()
     allow_guest_access = serializers.BooleanField(allow_null=True)
@@ -78,12 +78,16 @@ class DeviceProvisionSerializer(DeviceSerializerMixin, serializers.Serializer):
             facility.dataset.save()

             # Create superuser
-            superuser = FacilityUser.objects.create_superuser(
-                validated_data["superuser"]["username"],
-                validated_data["superuser"]["password"],
-                facility=facility,
-                full_name=validated_data["superuser"].get("full_name"),
-            )
+            superuser_data = validated_data.pop("superuser")
+            if superuser_data:
+                superuser = FacilityUser.objects.create_superuser(
+                    superuser_data["username"],
+                    superuser_data["password"],
+                    facility=facility,
+                    full_name=superuser_data.get("full_name"),
+                )
+            else:
+                superuser = None

             # Create device settings
             language_id = validated_data.pop("language_id")
--
2.33.1
