diff --git a/Main/Scripts/SaveAndLoad.gd b/Main/Scripts/SaveAndLoad.gd
index d92990a..11eec31 100644
--- a/Main/Scripts/SaveAndLoad.gd
+++ b/Main/Scripts/SaveAndLoad.gd
@@ -8,7 +8,7 @@ const YIELD_EVERY : int = 25
 var import_trimmed : bool = false
 var import_resized : bool = false
 var import_percent : float = 50.0
-@onready var dire = Settings.path_helper(OS.get_executable_path().get_base_dir(), "/ExportedAssets")
+@onready var dire = Settings.path_helper("user://", "/ExportedAssets")
 
 func save_file(path : String):
 	save_model(path)
diff --git a/Scripts/AutoLoads/Settings.gd b/Scripts/AutoLoads/Settings.gd
index 2170d54..0be5380 100644
--- a/Scripts/AutoLoads/Settings.gd
+++ b/Scripts/AutoLoads/Settings.gd
@@ -29,7 +29,7 @@ const SAVED_LAYOUT_PATH := "user://layout.tres"
 	properties = 0,
 	layers = -100,
 	file_manager = 0,
-	lipsync_file_path = path_helper(OS.get_executable_path().get_base_dir(), "/DefaultTraining.tres"),
+	lipsync_file_path = path_helper("user://", "/DefaultTraining.tres"),
 	microphone = null,
 	enable_trimmer = false,
 	save_raw_sprite = true, #save the original sprite even when trimmed
@@ -59,9 +59,9 @@ var websocket_api = ""
 
 
 func _enter_tree() -> void:
-	save_location = path_helper(OS.get_executable_path().get_base_dir(), "/Preferences.pRDat")
-	autosave_location = path_helper(OS.get_executable_path().get_base_dir(), "/autosaves")
-	websocket_api = path_helper(OS.get_executable_path().get_base_dir(), "/WebsocketDocumentation.txt")
+	save_location = path_helper("user://", "/Preferences.pRDat")
+	autosave_location = path_helper("user://", "/autosaves")
+	websocket_api = path_helper("user://", "/WebsocketDocumentation.txt")
 
 func _notification(what: int) -> void:
 	if what == NOTIFICATION_WM_CLOSE_REQUEST:
