From 6ce3b5a643dba02d2ebe45f4e2ef8ec9fc4b864d Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 20 Oct 2018 19:44:39 +0200
Subject: [PATCH] StProcess - Look for data in /app rather than /usr

---
 StShared/StProcess.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/StShared/StProcess.cpp b/StShared/StProcess.cpp
index 12f255eb..c82d26e9 100644
--- a/StShared/StProcess.cpp
+++ b/StShared/StProcess.cpp
@@ -35,7 +35,7 @@ namespace {
     static const StString STCORE_NAME = StString("StCore")    + ST_DLIB_SUFFIX;
 #else
     static const StString STCORE_NAME = StString("libStCore") + ST_DLIB_SUFFIX;
-    static const StString ST_DEFAULT_PATH = "/usr/share/sView/";
+    static const StString ST_DEFAULT_PATH = "/app/share/sView/";
 #endif
 
     inline bool isValidStSharePath(const StString& thePath) {
-- 
2.17.2

