diff --git a/src/MainWindow.cpp b/src/MainWindow.cpp
index 752b363..2c40763 100644
--- a/src/MainWindow.cpp
+++ b/src/MainWindow.cpp
@@ -72,6 +71,7 @@ MainWindow::MainWindow(SharedObjects *sharedObjects)
       m_geoDataModel(sharedObjects->geoDataModel())
 {
     setWindowTitle(i18n("KGeoTag"));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("org.kde.kgeotag")));
 
     connect(m_elevationEngine, &ElevationEngine::elevationProcessed,
             this, &MainWindow::elevationProcessed);
diff --git a/src/main.cpp b/src/main.cpp
index 245012f..832b046 100644
--- a/src/main.cpp
+++ b/src/main.cpp
-- 
2.35.1

