diff --git a/bsnes/target-bsnes/settings/settings.cpp b/bsnes/target-bsnes/settings/settings.cpp
index 0c17d723..a7916997 100644
--- a/bsnes/target-bsnes/settings/settings.cpp
+++ b/bsnes/target-bsnes/settings/settings.cpp
@@ -35,9 +35,9 @@ auto Settings::save() -> void {
 auto Settings::process(bool load) -> void {
   if(load) {
     //initialize non-static default settings
-    video.driver = ruby::Video::safestDriver();
-    audio.driver = ruby::Audio::safestDriver();
-    input.driver = ruby::Input::safestDriver();
+    video.driver = ruby::Video::optimalDriver();
+    audio.driver = ruby::Audio::optimalDriver();
+    input.driver = ruby::Input::optimalDriver();
   }
 
   #define bind(type, path, name) \
