diff --git a/CustomMesh/CustomMesh.gdextension b/CustomMesh/CustomMesh.gdextension
index 056d832..38198c0 100644
--- a/CustomMesh/CustomMesh.gdextension
+++ b/CustomMesh/CustomMesh.gdextension
@@ -10,6 +10,9 @@ windows.x86_64.single.release = "res://CustomMesh/windows/CustomMesh.windows.tem
 linux.x86_64.single.debug = "res://CustomMesh/linux/CustomMesh.linux.template_debug.x86_64.so"
 linux.x86_64.single.release = "res://CustomMesh/linux/CustomMesh.linux.template_release.x86_64.so"
 
+linux.arm64.single.debug = "res://CustomMesh/linux/CustomMesh.linux.template_debug.arm64.so"
+linux.arm64.single.release = "res://CustomMesh/linux/CustomMesh.linux.template_release.arm64.so"
+
 [dependencies]
 windows = {
     "res://CustomMesh/windows/libgcc_s_seh-1.dll" : "",
diff --git a/GI/GlobalInputs.gdextension b/GI/GlobalInputs.gdextension
index b26d1f8..edd01ea 100644
--- a/GI/GlobalInputs.gdextension
+++ b/GI/GlobalInputs.gdextension
@@ -7,8 +7,10 @@ compatibility_minimum = "4.2"
 windows.x86_64.single.debug = "res://GI/windows/GlobalInput.windows.template_debug.x86_64.dll"
 windows.x86_64.single.release = "res://GI/windows/GlobalInput.windows.template_release.x86_64.dll"
 
-linux.x86_64.single.debug = "res://GI/linux/GlobalInput.linux.template_debug.x86_64.so"
-linux.x86_64.single.release = "res://GI/linux/GlobalInput.linux.template_release.x86_64.so"
+linux.x86_64.single.debug = "res://GI/linux/libGlobalInput.linux.template_debug.x86_64.so"
+linux.x86_64.single.release = "res://GI/linux/libGlobalInput.linux.template_release.x86_64.so"
+linux.arm64.single.debug = "res://GI/linux/libGlobalInput.linux.template_debug.arm64.so"
+linux.arm64.single.release = "res://GI/linux/libGlobalInput.linux.template_release.arm64.so"
 
 [dependencies]
 windows = {
@@ -18,6 +20,6 @@ windows = {
     "res://GI/windows/libwinpthread-1.dll" : "",
 }
 linux = {
-    "res://GI/linux/libuiohook.so" : "",
+    "res://GI/linux/libuiohook.so.1" : "",
 }
 
