From 031d464284d3244bf993038d51a868cea5535869 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 18 Nov 2019 12:43:53 +0100
Subject: [PATCH] build: Add flag to disable update checker

make DISABLE_UPDATER=1
---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 53712792..c655ec76 100644
--- a/Makefile
+++ b/Makefile
@@ -158,6 +158,10 @@ else ifneq (,$(findstring mingw,$(CCMACHINE)))
 TARGET_ARCH2 = x86
 endif
 
+ifneq ($(DISABLE_UPDATER),)
+EXTRA_CXXFLAGS += -DST_NO_UPDATES_CHECK
+endif
+
 # to activate debug build
 #EXTRA_CXXFLAGS = -DST_DEBUG_LOG_TO_FILE=\"/sdcard/Android/data/com.sview/files/sview.log\" -DST_DEBUG
 #EXTRA_CXXFLAGS += -DST_DEBUG_GL
-- 
2.23.0

