From d0c153b4d3cccc87535042015a8e4e7f630c5ba6 Mon Sep 17 00:00:00 2001
From: Robin Lee <cheeselee@fedoraproject>
Date: Wed, 24 Jan 2018 18:15:58 +0800
Subject: [PATCH] Force gtk2

---
 configure.ac | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index b8e3338e..9fad1071 100644
--- a/configure.ac
+++ b/configure.ac
@@ -72,16 +72,9 @@ if test "$enable_deprecations" = "yes"; then
 	AC_SUBST(DISABLE_DEPRECATED_CFLAGS)
 fi
 
-PKG_CHECK_EXISTS([gtk+-3.0 >= 2.90],
-		 [
-		 	GTK_REQUIRED=gtk+-3.0
-		 	GTK_REQUIRED_VERSION=2.90.0
-			GTK_TYPELIB=Gtk-3.0
-		 ],[
-		 	GTK_REQUIRED=gtk+-2.0
-		 	GTK_REQUIRED_VERSION=2.16.0
-			GTK_TYPELIB=Gtk-2.0
-		 ])
+GTK_REQUIRED=gtk+-2.0
+GTK_REQUIRED_VERSION=2.16.0
+GTK_TYPELIB=Gtk-2.0
 
 AC_SUBST(GTK_TYPELIB)
 PKG_CHECK_MODULES(DEP, [
-- 
2.14.3

