From 9b0e3fe3562d91601cb67c21675b019e24bf794e Mon Sep 17 00:00:00 2001
From: Antoine Fontaine <antoine.fontaine@epfl.ch>
Date: Sun, 6 Jun 2021 21:16:03 +0200
Subject: [PATCH] Suppress a warning

---
 src/engine/framework/System.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/engine/framework/System.cpp b/src/engine/framework/System.cpp
index b0cda3cb..cc7233cd 100644
--- a/src/engine/framework/System.cpp
+++ b/src/engine/framework/System.cpp
@@ -440,7 +440,7 @@ static void ParseCmdline(int argc, char** argv, cmdlineArgs_t& cmdlineArgs)
 		// Any arguments after the URI are discarded.
 		if (Str::IsIEqual("-connect", argv[i])) {
 			if (argc < i + 2) {
-				Log::Warn("Missing command line parameter for -connect");
+				Log::Notice("Ignoring -connect parameter as no server was provided");
 				break;
 			}
 			cmdlineArgs.uriText = argv[i + 1];
-- 
2.31.1

