From 16acc446b573370761e49c46110ce6978f5fd2a6 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Sat, 7 Sep 2019 11:41:56 -0500
Subject: [PATCH 1/4] Use /usr/bin/env in the clamtk hashbang

---
 clamtk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clamtk b/clamtk
index 448ad99..cfd4a7a 100755
--- a/clamtk
+++ b/clamtk
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 #
 # ClamTk, copyright (C) 2004-2019 Dave M
 #
-- 
2.21.0

