From 815cbb49e940456649947e4b38e273bf179352c1 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 10 Oct 2024 02:22:00 +0200
Subject: [PATCH] Use better theme

---
 NiimPrintX/ui/main.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/NiimPrintX/ui/main.py b/NiimPrintX/ui/main.py
index 390ff8dbbd8a..774ea8f87792 100644
--- a/NiimPrintX/ui/main.py
+++ b/NiimPrintX/ui/main.py
@@ -64,12 +64,13 @@ class LabelPrinterApp(tk.Tk):
         threading.Thread(target=self.start_asyncio_loop, daemon=True).start()
 
         self.app_config = AppConfig()
+        style = ttk.Style(self)
         if self.app_config.os_system == "Darwin":
-            style = ttk.Style(self)
             style.theme_use('aqua')
         elif self.app_config.os_system == "Windows":
-            style = ttk.Style(self)
             style.theme_use('xpnative')
+        else:
+            style.theme_use('radiance')
 
         self.create_widgets()
         self.create_menu()
-- 
2.46.2

