diff --git a/data/appdata/org.gnome.Firmware.metainfo.xml.in b/data/appdata/org.gnome.Firmware.metainfo.xml.in
index def382d..2f91b8e 100644
--- a/data/appdata/org.gnome.Firmware.metainfo.xml.in
+++ b/data/appdata/org.gnome.Firmware.metainfo.xml.in
@@ -11,10 +11,19 @@
   <description>
     <!-- TRANSLATORS: AppData description marketing paragraph -->
     <p>
-      This application can update, reinstall and downgrade firmware on devices
-      supported by fwupd.
+      Update, reinstall and downgrade firmware on devices supported by fwupd.
     </p>
   </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://gitlab.gnome.org/hughsie/gnome-firmware-updater/raw/master/data/appdata/ss-devices.png</image>
+      <caption>Devices panel</caption>
+    </screenshot>
+    <screenshot>
+      <image>https://gitlab.gnome.org/hughsie/gnome-firmware-updater/raw/master/data/appdata/ss-releases.png</image>
+      <caption>Releases panel</caption>
+    </screenshot>
+  </screenshots>
   <project_group>GNOME</project_group>
   <update_contact>richard_at_hughsie.com</update_contact>
   <translation type="gettext">gnome-firmware</translation>
@@ -33,4 +42,11 @@
       </description>
     </release>
   </releases>
+  <url type="homepage">https://gitlab.gnome.org/hughsie/gnome-firmware-updater</url>
+  <url type="bugtracker">https://gitlab.gnome.org/hughsie/gnome-firmware-updater/issues</url>
+  <url type="donation">http://www.gnome.org/friends/</url>
+  <update_contact>richard_at_hughsie.com</update_contact>
+  <content_rating type="oars-1.1">
+    <content_attribute id="social-info">moderate</content_attribute>
+  </content_rating>
 </component>
