From 1c9c337e3bbff5ea2c91925c547ced9e92afdb21 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Fri, 26 Apr 2024 07:15:03 +0300
Subject: [PATCH 1/2] Mark flathub as the distributor

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 configure  | 2 +-
 fc_version | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index eb6a2b25c5..1a3ad6a8da 100755
--- a/configure
+++ b/configure
@@ -5686,7 +5686,7 @@ DEFAULT_FOLLOW_TAG=stable
 #     so would break network capability of supposedly "compatible" releases.
 #
 NETWORK_CAPSTRING="+Freeciv-3.1-network city-original rsdesc32 obsinv ids32"
-FREECIV_DISTRIBUTOR=""
+FREECIV_DISTRIBUTOR="flathub"
 
 if test "x$FREECIV_LABEL_FORCE" != "x" ; then
   VERSION_LABEL=$(echo $FREECIV_LABEL_FORCE | sed "s/<base>/$VERSION_LABEL/g")
diff --git a/fc_version b/fc_version
index df857c28f0..8806fe52df 100755
--- a/fc_version
+++ b/fc_version
@@ -73,7 +73,7 @@ DEFAULT_FOLLOW_TAG=stable
 #     so would break network capability of supposedly "compatible" releases.
 #
 NETWORK_CAPSTRING="+Freeciv-3.1-network city-original rsdesc32 obsinv ids32"
-FREECIV_DISTRIBUTOR=""
+FREECIV_DISTRIBUTOR="flathub"
 
 if test "x$FREECIV_LABEL_FORCE" != "x" ; then
   VERSION_LABEL=$(echo $FREECIV_LABEL_FORCE | sed "s/<base>/$VERSION_LABEL/g")
-- 
2.43.0

