From 7dfebf373c14d8006d54b8803ce0f0faef7df959 Mon Sep 17 00:00:00 2001
From: "Kowalski Dragon (kowalski7cc)" <kowalski7cc@users.noreply.github.com>
Date: Thu, 29 Sep 2022 14:13:06 +0200
Subject: [PATCH] Change error message for flatpak

Signed-off-by: Kowalski Dragon (kowalski7cc) <kowalski7cc@users.noreply.github.com>
---
 application/imports/Misc/ErrorStrings.qml | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/application/imports/Misc/ErrorStrings.qml b/application/imports/Misc/ErrorStrings.qml
index f74f2a3..ef89b10 100644
--- a/application/imports/Misc/ErrorStrings.qml
+++ b/application/imports/Misc/ErrorStrings.qml
@@ -29,10 +29,9 @@ QtObject {
 <p>=========== HOW TO FIX ============</p>
 <p>1. Ensure that Flipper is not connected via Bluetooth or Terminal session.</p>
 <p>2. Grant user permissions to access Serial devices.</p>
-<p>Run \"qFlipper-x86_64-%1.AppImage rules install\" to do so automatically.</p>
 <p>3. Reconnect your Flipper.</p>
 <p>-----------------------------------</p>
-<center><a href='https://docs.flipperzero.one/'>READ MORE</a></center>".arg(App.version)
+<center><a href='https://github.com/flathub/one.flipperzero.qFlipper/wiki/'>READ MORE</a></center>".arg(App.version)
 
     readonly property string errorRecovery:
 "<p>Cannot connect to Flipper in Update & Recovery mode. Device not found.</p>
-- 
2.30.2

