From d73455025d71226472e7eb880da36f6fb85c5df0 Mon Sep 17 00:00:00 2001
From: Johann ELSASS <circular@fastmail.com>
Date: Thu, 6 Oct 2022 21:44:47 +0200
Subject: [PATCH] avoid crash on Qt5

---
 lazpaint/lazpaintinstance.pas | 1 +
 lazpaint/lazpaintmainform.pas | 1 +
 2 files changed, 2 insertions(+)

diff --git a/lazpaint/lazpaintinstance.pas b/lazpaint/lazpaintinstance.pas
index 5da9ac6..b5f624b 100644
--- a/lazpaint/lazpaintinstance.pas
+++ b/lazpaint/lazpaintinstance.pas
@@ -374,6 +374,7 @@ procedure TLazPaintInstance.ReportActionProgress(AProgressPercent: integer);
 var
   delay: Integer;
 begin
+  {$IFDEF LCLqt5}exit;{$ENDIF}
   if AProgressPercent < 100 then delay := 10000 else delay := 1000;
   if Assigned(FMain) then FMain.UpdatingPopup:= true;
   try
diff --git a/lazpaint/lazpaintmainform.pas b/lazpaint/lazpaintmainform.pas
index 0fe875c..708cb24 100644
--- a/lazpaint/lazpaintmainform.pas
+++ b/lazpaint/lazpaintmainform.pas
@@ -2695,6 +2695,7 @@ end;
 
 procedure TFMain.TimerUpdateTimer(Sender: TObject);
 begin
+  if FLazPaintInstance = nil then exit;
   TimerUpdate.Enabled := false;
   if ToolManager.ToolSleeping and not spacePressed and
      ([ssLeft,ssRight,ssMiddle] * FLayout.MouseButtonState = []) then
-- 
2.43.0

