From 986d577381604450b98447298a1e25143e47e65c Mon Sep 17 00:00:00 2001
From: "Luna D. Dragon" <advaith.madhukar@gmail.com>
Date: Sat, 8 Jan 2022 11:06:12 +0530
Subject: [PATCH] set auto update and auto download to false if in flatpak

---
 main.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/main.js b/main.js
index 1b7f481..5392133 100644
--- a/main.js
+++ b/main.js
@@ -111,6 +111,11 @@ console.log(enterpriseConfig);
 autoUpdater.logger = log;
 autoUpdater.logger.transports.file.level = "info";
 
+if (isFlatpak) {
+	autoUpdater.autoInstallOnAppQuit = false;
+	autoUpdater.autoDownload = false;
+}
+
 switch(enterpriseConfig.autoUpdatePolicy) {
 	case "disabled":
 	case "manual":
-- 
2.34.1

