diff --git a/build.gradle b/build.gradle
index ebb8502c..7b7949ab 100644
--- a/build.gradle
+++ b/build.gradle
@@ -6,6 +6,7 @@ plugins {
 	id 'org.openjfx.javafxplugin' version '0.1.0'
 	id 'com.gradleup.shadow' version '9.0.0-beta11'
 	id 'com.github.ben-manes.versions' version '0.52.0'
+	id "io.github.jwharm.flatpak-gradle-generator" version "1.5.0"
 }
 
 group 'net.querz'
@@ -34,6 +35,7 @@ repositories {
 	maven {
 		url 'https://plugins.gradle.org/m2/'
 	}
+	maven { url "./offline-repository" }
 }
 
 dependencies {
@@ -98,6 +100,11 @@ tasks.register('generator', JavaExec) {
 	classpath = sourceSets.main.runtimeClasspath
 }
 
+tasks.flatpakGradleGenerator {
+  outputFile = file("flatpak-sources.json")
+  downloadDirectory = "./offline-repository"
+}
+
 jar {
 	archiveFileName = "${project.name}-${project.version}-min.jar"
 	manifest.attributes (
diff --git a/settings.gradle b/settings.gradle
index 6f42e8a9..4436e33b 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1 +1,14 @@
+pluginManagement {
+  repositories {
+  	mavenCentral()
+  	maven {
+  		url 'https://jitpack.io/'
+  	}
+  	maven {
+  		url 'https://plugins.gradle.org/m2/'
+  	}
+  	maven { url "./offline-repository" }
+  }
+}
+
 rootProject.name = 'mcaselector'
\ No newline at end of file
-- 
2.51.0

