--- a/program/CMakeLists.txt
+++ b/program/CMakeLists.txt
@@ -16,8 +16,11 @@ qt5_add_resources(okteta_SRCS
 ecm_add_app_icon(okteta_SRCS
   ICONS
     ${CMAKE_CURRENT_SOURCE_DIR}/16-apps-okteta.png
+    ${CMAKE_CURRENT_SOURCE_DIR}/24-apps-okteta.png
     ${CMAKE_CURRENT_SOURCE_DIR}/32-apps-okteta.png
     ${CMAKE_CURRENT_SOURCE_DIR}/48-apps-okteta.png
+    ${CMAKE_CURRENT_SOURCE_DIR}/64-apps-okteta.png
+    ${CMAKE_CURRENT_SOURCE_DIR}/128-apps-okteta.png
 )
 
 add_executable( okteta  ${okteta_SRCS} )
@@ -43,6 +46,12 @@ install( PROGRAMS org.kde.okteta.desktop  DESTINATION ${KDE_INSTALL_APPDIR} )
 install( FILES org.kde.okteta.appdata.xml DESTINATION ${KDE_INSTALL_METAINFODIR} )
 
 ecm_install_icons(
-  ICONS 16-apps-okteta.png  32-apps-okteta.png  48-apps-okteta.png
+  ICONS
+    16-apps-okteta.png
+    24-apps-okteta.png
+    32-apps-okteta.png
+    48-apps-okteta.png
+    64-apps-okteta.png
+    128-apps-okteta.png
   DESTINATION ${KDE_INSTALL_ICONDIR}
 )
