diff --git a/build.gradle b/build.gradle
index 949e05584..4ee5723ba 100644
--- a/build.gradle
+++ b/build.gradle
@@ -63,8 +63,7 @@ run {
 
 repositories {
     maven {
-        name = 'JBoss repository' // required to obtain non-free JAI
-        url = 'https://repository.jboss.org/nexus/content/repositories/thirdparty-releases'
+        url 'dependencies'
     }
     maven { // required for JAI Core 1.1.3
         name = 'SpringSource Enterprise Bundle Repository - External Bundle Releases'
diff --git a/schemas/build.gradle b/schemas/build.gradle
index c20dd44e3..256fd12a4 100644
--- a/schemas/build.gradle
+++ b/schemas/build.gradle
@@ -54,11 +54,10 @@ apply plugin: 'java'
 def String docFilesPattern = "src=\"doc-files/([^\"]*)\""
 
 repositories { 
-    mavenCentral() 
     maven {
-        name = 'mvnrepository' 
-        url = 'https://mvnrepository.com/artifact/net.sf.saxon/Saxon-HE'
+        url '../dependencies'
     }
+    mavenCentral() 
 }
 
 dependencies { 
