From 6e4a5b981801945456e5314ab4c615cc14993c6f Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 20 Oct 2018 19:26:39 +0200
Subject: [PATCH] build: Install in /app

---
 Makefile | 42 +++++++++++++++++++++---------------------
 1 file changed, 21 insertions(+), 21 deletions(-)

diff --git a/Makefile b/Makefile
index 79e48985..8d9969e2 100644
--- a/Makefile
+++ b/Makefile
@@ -224,27 +224,27 @@ outputs_all: $(aStOutAnaglyph) $(aStOutDual) $(aStOutInterlace) $(aStOutPageFlip
 endif
 
 install:
-	mkdir -p $(DESTDIR)/usr/bin
-	mkdir -p $(DESTDIR)/usr/$(USR_LIB)/sView
-	mkdir -p $(DESTDIR)/usr/$(USR_LIB)/firefox/plugins
-	mkdir -p $(DESTDIR)/usr/$(USR_LIB)/mozilla/plugins
-	mkdir -p $(DESTDIR)/usr/share
-	mkdir -p $(DESTDIR)/usr/share/sView/info
-	mkdir -p $(DESTDIR)/usr/share/sView/lang
-	mkdir -p $(DESTDIR)/usr/share/sView/shaders
-	mkdir -p $(DESTDIR)/usr/share/sView/textures
-	mkdir -p $(DESTDIR)/usr/share/sView/web
-	cp -f -r share/*                     $(DESTDIR)/usr/share/
-	cp -f -r $(BUILD_ROOT)/lang/*        $(DESTDIR)/usr/share/sView/lang/
-	cp -f -r $(BUILD_ROOT)/shaders/*     $(DESTDIR)/usr/share/sView/shaders/
-	cp -f -r $(BUILD_ROOT)/textures/*    $(DESTDIR)/usr/share/sView/textures/
-	cp -f -r $(BUILD_ROOT)/web/*         $(DESTDIR)/usr/share/sView/web/
-	cp -f    license-gpl-3.0.txt         $(DESTDIR)/usr/share/sView/info/license.txt
-	cp -f -r $(BUILD_ROOT)/*.$(LIBSUFFIX) $(DESTDIR)/usr/$(USR_LIB)/
-	cp -f    $(BUILD_ROOT)/sView         $(DESTDIR)/usr/$(USR_LIB)/sView/sView
-	ln --force --symbolic ../$(USR_LIB)/sView/sView       $(DESTDIR)/usr/bin/sView
-	ln --force --symbolic ../../share/sView/demo/demo.jps $(DESTDIR)/usr/$(USR_LIB)/sView/demo.jps
-	rm -f    $(DESTDIR)/usr/$(USR_LIB)/sView/*.a
+	mkdir -p $(DESTDIR)/app/bin
+	mkdir -p $(DESTDIR)/app/$(USR_LIB)/sView
+	mkdir -p $(DESTDIR)/app/$(USR_LIB)/firefox/plugins
+	mkdir -p $(DESTDIR)/app/$(USR_LIB)/mozilla/plugins
+	mkdir -p $(DESTDIR)/app/share
+	mkdir -p $(DESTDIR)/app/share/sView/info
+	mkdir -p $(DESTDIR)/app/share/sView/lang
+	mkdir -p $(DESTDIR)/app/share/sView/shaders
+	mkdir -p $(DESTDIR)/app/share/sView/textures
+	mkdir -p $(DESTDIR)/app/share/sView/web
+	cp -f -r share/*                     $(DESTDIR)/app/share/
+	cp -f -r $(BUILD_ROOT)/lang/*        $(DESTDIR)/app/share/sView/lang/
+	cp -f -r $(BUILD_ROOT)/shaders/*     $(DESTDIR)/app/share/sView/shaders/
+	cp -f -r $(BUILD_ROOT)/textures/*    $(DESTDIR)/app/share/sView/textures/
+	cp -f -r $(BUILD_ROOT)/web/*         $(DESTDIR)/app/share/sView/web/
+	cp -f    license-gpl-3.0.txt         $(DESTDIR)/app/share/sView/info/license.txt
+	cp -f -r $(BUILD_ROOT)/*.$(LIBSUFFIX) $(DESTDIR)/app/$(USR_LIB)/
+	cp -f    $(BUILD_ROOT)/sView         $(DESTDIR)/app/$(USR_LIB)/sView/sView
+	ln --force --symbolic ../$(USR_LIB)/sView/sView       $(DESTDIR)/app/bin/sView
+	ln --force --symbolic ../../share/sView/demo/demo.jps $(DESTDIR)/app/$(USR_LIB)/sView/demo.jps
+	rm -f    $(DESTDIR)/app/$(USR_LIB)/sView/*.a
 
 install_android:
 	mkdir -p $(aDestAndroid)/assets/info
-- 
2.17.2

