From b6d11363efac9ddeb0e48dd7baa7621e9c74ca08 Mon Sep 17 00:00:00 2001
From: Matthieu Gallien <matthieu_gallien@yahoo.fr>
Date: Wed, 18 Apr 2018 10:05:19 +0200
Subject: [PATCH] hardcode icon theme name to breeze to ensure consistent look

---
 src/main.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/main.cpp b/src/main.cpp
index 4600f6d..1f27e0f 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -126,6 +126,8 @@ int main(int argc, char *argv[])
 
     KLocalizedString::setApplicationDomain("elisa");
 
+    QIcon::setThemeName(QStringLiteral("breeze"));
+
 #if defined UPNPQT_FOUND && UPNPQT_FOUND
     qmlRegisterType<UpnpSsdpEngine>("org.kde.elisa", 1, 0, "UpnpSsdpEngine");
     qmlRegisterType<UpnpDiscoverAllMusic>("org.kde.elisa", 1, 0, "UpnpDiscoverAllMusic");
-- 
2.17.0

