From a06a3041eec370bf2382330161d2bbe414295ff1 Mon Sep 17 00:00:00 2001
From: Christoph Settgast <csett86@web.de>
Date: Mon, 7 Nov 2022 18:05:11 +0100
Subject: [PATCH] flatpak

---
 src/libs/vmisc/vcommonsettings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libs/vmisc/vcommonsettings.cpp b/src/libs/vmisc/vcommonsettings.cpp
index fa7f34b4b..1a6051ed6 100644
--- a/src/libs/vmisc/vcommonsettings.cpp
+++ b/src/libs/vmisc/vcommonsettings.cpp
@@ -225,7 +225,7 @@ QStringList ClearFormats(const QStringList &predefinedFormats, QStringList forma
 static const QString commonIniFilename = QStringLiteral("common");
 
 #if !defined(Q_OS_WIN)
-const QString VCommonSettings::unixStandardSharePath = QStringLiteral("/usr/share/seamly2d");
+const QString VCommonSettings::unixStandardSharePath = QStringLiteral("/app/share/seamly2d");
 #endif
 
 namespace
-- 
2.34.1

