From d06bd3d712c7e1400a64f246f7fe38033e78c035 Mon Sep 17 00:00:00 2001
From: Dennis M. Pöpperl <free-software@dm-poepperl.de>
Date: Thu, 6 Jun 2019 12:44:03 +0200
Subject: [PATCH] Remove manpage support

---
 misc/Makefile | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/misc/Makefile b/misc/Makefile
index 526f7ad..0216e07 100644
--- a/misc/Makefile
+++ b/misc/Makefile
@@ -12,15 +12,10 @@ endif
 
 .PHONY: install
 
-doc/qutebrowser.1.html:
-	a2x -f manpage doc/qutebrowser.1.asciidoc
-
-install: doc/qutebrowser.1.html
+install:
 	$(PYTHON) setup.py install --prefix="$(PREFIX)" --optimize=1 $(SETUPTOOLSOPTS)
 	install -Dm644 misc/org.qutebrowser.qutebrowser.appdata.xml \
 		"$(DESTDIR)$(DATADIR)/metainfo/org.qutebrowser.qutebrowser.appdata.xml"
-	install -Dm644 doc/qutebrowser.1 \
-		"$(DESTDIR)$(MANDIR)/man1/qutebrowser.1"
 	install -Dm644 misc/qutebrowser.desktop \
 		"$(DESTDIR)$(DATADIR)/applications/qutebrowser.desktop"
 	$(foreach i,$(ICONSIZES),install -Dm644 "icons/qutebrowser-$(i)x$(i).png" \
--
libgit2 0.27.8

