diff --git a/src/main.cpp b/src/main.cpp
index e326a44..5d23485 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -12,6 +12,7 @@
 #include <QQuickStyle>
 #include <QUrl>
 #include <QtQml>
+#include <QIcon>
 
 #ifdef Q_OS_ANDROID
 #include <QGuiApplication>
@@ -48,6 +49,7 @@ Q_DECL_EXPORT int main(int argc, char *argv[])
     if(qEnvironmentVariableIsEmpty("QT_QUICK_CONTROLS_STYLE")) {
         QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"));
     }
+    QIcon::setThemeName(QStringLiteral("breeze"));
 #endif
     QQmlApplicationEngine engine;
 
