From 1479b07c1a16797241473171a72f5f2f8f94162e Mon Sep 17 00:00:00 2001
From: Bastien Nocera <bnocera@redhat.com>
Date: Thu, 12 Jun 2025 10:34:19 +0200
Subject: [PATCH] Use forked imara-diff

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 7aca77417a7a..12de55dcfff1 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,7 +33,7 @@ format-bytes = "0.3.0"
 gtk = { version = "0.9.1", package = "gtk4", features = ["v4_12"] }
 http-body-util = "0.1.3"
 hyper = { version = "1.6.0", features = ["client", "http1"] }
-imara-diff = { git = "https://github.com/pascalkuthe/imara-diff", rev = "c0b655a86b5b" }
+imara-diff = { git = "https://github.com/pascalkuthe/imara-diff", rev = "7e6c3bd80322" }
 json = "0.12.4"
 regex = "1.10.2"
 tempfile = "3.8.1"
-- 
2.49.0

