From 8ea87194fb50ff052a75212f9d2a2ae217edad95 Mon Sep 17 00:00:00 2001
From: Mathias <mathias@localhost.localdomain>
Date: Sun, 2 May 2021 12:06:15 +0200
Subject: [PATCH] fixed DCRComplexSolve compile error

---
 fem/src/modules/DCRComplexSolve.F90 | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/fem/src/modules/DCRComplexSolve.F90 b/fem/src/modules/DCRComplexSolve.F90
index cfa71dee..b6bb59bc 100644
--- a/fem/src/modules/DCRComplexSolve.F90
+++ b/fem/src/modules/DCRComplexSolve.F90
@@ -426,7 +426,7 @@ CONTAINS
    SUBROUTINE InputTensor( Tensor, IsScalar, Name, Material, n, NodeIndexes )
 !------------------------------------------------------------------------------
       REAL(KIND=dp) :: Tensor(:,:,:)
-      INTEGER :: n, NodeIndexes(:)
+      INTEGER :: n, i, j, NodeIndexes(:)
       LOGICAL :: IsScalar
       CHARACTER(LEN=*) :: Name
       TYPE(ValueList_t), POINTER :: Material
@@ -485,7 +485,7 @@ CONTAINS
    SUBROUTINE InputVector( Tensor, IsScalar, Name, Material, n, NodeIndexes )
 !------------------------------------------------------------------------------
       REAL(KIND=dp) :: Tensor(:,:)
-      INTEGER :: n, NodeIndexes(:)
+      INTEGER :: n, i, NodeIndexes(:)
       LOGICAL :: IsScalar
       CHARACTER(LEN=*) :: Name
       TYPE(ValueList_t), POINTER :: Material
@@ -543,7 +543,7 @@ CONTAINS
     REAL(KIND=dp) :: DiffCoef(3,3), Velo(3)
     COMPLEX(KIND=dp) :: LSTIFF(2*n,2*n), LFORCE(2*n), A
     LOGICAL :: Stat
-    INTEGER :: i,p,q,t,dim, NBasis, CoordSys
+    INTEGER :: i,j,p,q,t,dim, NBasis, CoordSys
     TYPE(GaussIntegrationPoints_t) :: IntegStuff
 
     REAL(KIND=dp) :: X,Y,Z,Metric(3,3),SqrtMetric,Symb(3,3,3),dSymb(3,3,3,3)
@@ -678,7 +678,7 @@ CONTAINS
          C1(3), C0
     COMPLEX(KIND=dp) :: LSTIFF(n,n), LFORCE(n), A
     LOGICAL :: Stat
-    INTEGER :: i,p,q,t,dim,CoordSys
+    INTEGER :: i,j,p,q,t,dim,CoordSys
     TYPE(GaussIntegrationPoints_t) :: IntegStuff
 !------------------------------------------------------------------------------
     dim = CoordinateSystemDimension()
@@ -1374,7 +1374,7 @@ contains
    SUBROUTINE InputTensor( Tensor, IsScalar, Name, Material, n, NodeIndexes )
 !------------------------------------------------------------------------------
       REAL(KIND=dp) :: Tensor(:,:,:)
-      INTEGER :: n, NodeIndexes(:)
+      INTEGER :: n, i, NodeIndexes(:)
       LOGICAL :: IsScalar
       CHARACTER(LEN=*) :: Name
       TYPE(ValueList_t), POINTER :: Material
@@ -1426,7 +1426,7 @@ contains
    SUBROUTINE InputVector( Tensor, IsScalar, Name, Material, n, NodeIndexes )
 !------------------------------------------------------------------------------
       REAL(KIND=dp) :: Tensor(:,:)
-      INTEGER :: n, NodeIndexes(:)
+      INTEGER :: n, i, NodeIndexes(:)
       LOGICAL :: IsScalar
       CHARACTER(LEN=*) :: Name
       TYPE(ValueList_t), POINTER :: Material
-- 
2.31.1

