diff --git a/GI/GlobalInputs.gdextension b/GI/GlobalInputs.gdextension
index b26d1f8..bb5f902 100644
--- a/GI/GlobalInputs.gdextension
+++ b/GI/GlobalInputs.gdextension
@@ -7,8 +7,8 @@ compatibility_minimum = "4.2"
 windows.x86_64.single.debug = "res://GI/windows/GlobalInput.windows.template_debug.x86_64.dll"
 windows.x86_64.single.release = "res://GI/windows/GlobalInput.windows.template_release.x86_64.dll"
 
-linux.x86_64.single.debug = "res://GI/linux/GlobalInput.linux.template_debug.x86_64.so"
-linux.x86_64.single.release = "res://GI/linux/GlobalInput.linux.template_release.x86_64.so"
+linux.x86_64.single.debug = "res://GI/linux/libGlobalInput.linux.template_debug.x86_64.so"
+linux.x86_64.single.release = "res://GI/linux/libGlobalInput.linux.template_release.x86_64.so"
 
 [dependencies]
 windows = {
@@ -18,6 +18,6 @@ windows = {
     "res://GI/windows/libwinpthread-1.dll" : "",
 }
 linux = {
-    "res://GI/linux/libuiohook.so" : "",
+    "res://GI/linux/libuiohook.so.1" : "",
 }
 
