From f9645c905f21134ce2e883d0e36bf059ef6aa10b Mon Sep 17 00:00:00 2001
From: Sebastian Jastrzebski <sebby2k@gmail.com>
Date: Sun, 4 Mar 2018 10:54:33 -0500
Subject: [PATCH] Fix arm64 build

---
 src/include/sysdeps.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/include/sysdeps.h b/src/include/sysdeps.h
index 607ff320..fd9b618f 100644
--- a/src/include/sysdeps.h
+++ b/src/include/sysdeps.h
@@ -47,6 +47,8 @@ using namespace std;
 #define CPU_i386 1
 #elif defined(__arm__) || defined(_M_ARM)
 #define CPU_arm 1
+#elif defined(__aarch64__) || defined(_M_AARCH64)
+#define CPU_aarch64 1
 #elif defined(__powerpc__) || defined(__ppc__) || defined(_M_PPC)
 #define CPU_powerpc 1
 #else
-- 
2.14.3

