From 9c50e3ec73783b6e78db7263a0e0aa712d95435e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=86=B7=E6=99=B6=E5=B7=9D?= <j.c.leng@foxmail.com>
Date: Tue, 5 Sep 2023 15:20:09 +0800
Subject: [PATCH] install
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: jcleng <j.c.leng@foxmail.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5477141..01261a3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,5 +9,5 @@ add_executable(oneko
         oneko.c
         oneko.h
         patchlevel.h)
-
+install(TARGETS oneko DESTINATION bin)
 target_link_libraries(oneko X11 m Xext)
--
2.36.1.windows.1
