From 1507f172d2301709a66f8c189802d863c2c97ef6 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 7 Dec 2018 19:15:29 +0100
Subject: [PATCH] Use XDG_CONFIG_HOME as the root of the config path

---
 src/common/config.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/config.cpp b/src/common/config.cpp
index 4a1acb5..38b4b30 100644
--- a/src/common/config.cpp
+++ b/src/common/config.cpp
@@ -781,7 +781,7 @@ bool parse_ini_file()
 
 	if(!file.is_open())
 	{
-		std::string unix_str = getenv("HOME");
+		std::string unix_str = getenv("XDG_CONFIG_HOME");
 		std::string last_chr = "";
 
 		last_chr = unix_str[unix_str.length() - 1];
-- 
2.19.2

