From 30456eddb351e6b9977b73144329df881143ecfb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb@kaidan.im>
Date: Wed, 3 Jul 2024 16:55:07 +0200
Subject: [PATCH 1/2] mobile: Clip DocumentView

---
 mobile/components/DocumentView.qml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mobile/components/DocumentView.qml b/mobile/components/DocumentView.qml
index 3d5ada172..4ebd7de21 100644
--- a/mobile/components/DocumentView.qml
+++ b/mobile/components/DocumentView.qml
@@ -20,6 +20,8 @@ QQC2.ScrollView {
     property DocumentItem document
     property PageItem page: mouseArea.currPageDelegate.pageItem
     signal clicked
+
+    clip: true
     
     //NOTE: on some themes it tries to set the flickable to interactive
     //but we need it always non interactive as we need to manage
-- 
2.43.0

