From e4e97f4036620052d1c6f74f6955ad199531a16c Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 10 Oct 2024 10:43:14 +0200
Subject: [PATCH] Use better theme

---
 NiimPrintX/ui/main.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/NiimPrintX/ui/main.py b/NiimPrintX/ui/main.py
index 390ff8dbbd8a..d9fdf9560a08 100644
--- a/NiimPrintX/ui/main.py
+++ b/NiimPrintX/ui/main.py
@@ -1,6 +1,7 @@
 import os
 import tkinter as tk
 from tkinter import ttk
+import ttkthemes
 from tkinter import messagebox
 
 from .AppConfig import AppConfig
@@ -70,6 +71,9 @@ class LabelPrinterApp(tk.Tk):
         elif self.app_config.os_system == "Windows":
             style = ttk.Style(self)
             style.theme_use('xpnative')
+        else:
+            style = ttkthemes.ThemedStyle()
+            style.theme_use('yaru')
 
         self.create_widgets()
         self.create_menu()
@@ -131,4 +135,4 @@ if __name__ == "__main__":
         app.load_resources()
         app.mainloop()
     except Exception as e:
-        raise e
\ No newline at end of file
+        raise e
-- 
2.46.2

