From 11b9c647dd96edaf4a3240a3683493fb37a0e0e0 Mon Sep 17 00:00:00 2001
From: Johann ELSASS <circular@operamail.com>
Date: Thu, 28 Dec 2023 16:40:48 +0100
Subject: [PATCH] #515 runtime fix for Qt

---
 lazpaint/lazpaintmainform.lfm | 1 +
 lazpaint/lazpaintmainform.pas | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/lazpaint/lazpaintmainform.lfm b/lazpaint/lazpaintmainform.lfm
index a2dffa1..2765760 100644
--- a/lazpaint/lazpaintmainform.lfm
+++ b/lazpaint/lazpaintmainform.lfm
@@ -5769,6 +5769,7 @@ object FMain: TFMain
     Top = 514
   end
   object TimerUpdate: TTimer
+    Enabled = False
     Interval = 50
     OnTimer = TimerUpdateTimer
     Left = 653
diff --git a/lazpaint/lazpaintmainform.pas b/lazpaint/lazpaintmainform.pas
index 708cb24..551c7d0 100644
--- a/lazpaint/lazpaintmainform.pas
+++ b/lazpaint/lazpaintmainform.pas
@@ -1251,6 +1251,7 @@ begin
   UpdateToolBar;
   FShouldArrange := true;
   QueryArrange;
+  TimerUpdate.Enabled := true;
 end;
 
 procedure TFMain.OnLatestVersionUpdate(ANewVersion: string);
@@ -2593,6 +2594,7 @@ end;
 
 procedure TFMain.FormHide(Sender: TObject);
 begin
+  TimerUpdate.Enabled := false;
   FShouldArrange := false;
   FTopMostInfo := LazPaintInstance.HideTopmost;
   LazPaintInstance.SaveMainWindowPosition;
-- 
2.43.0

