From ad1a26afbe982ed6b85dbd208bfbe3473d6fd2be Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@nowhere.net>
Date: Tue, 19 May 2020 21:52:51 +0200
Subject: [PATCH] appdata: Update screenshot.

---
 data/opencpn.appdata.xml.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/opencpn.appdata.xml.in b/data/opencpn.appdata.xml.in
index 3bda17b76..341712e08 100644
--- a/data/opencpn.appdata.xml.in
+++ b/data/opencpn.appdata.xml.in
@@ -42,7 +42,7 @@
   <screenshots>
     <screenshot type="default">
       <caption>Main window</caption>
-      <image>https://en.wikipedia.org/wiki/OpenCPN#/media/File:OCPN_5_0.jpg</image>
+      <image>http://opencpn.duckdns.org/screenshots/shot-1.png</image>
     </screenshot>
   </screenshots>
   <url type="homepage">https://opencpn.org/</url>
-- 
2.24.1

