From aadfc71200adf45fe03b01c5953c926673f50f29 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb@kaidan.im>
Date: Sat, 28 May 2022 00:30:10 +0200
Subject: [PATCH] Fix crash on exit

---
 src/asyncytmusic.cpp | 9 ++++++---
 src/asyncytmusic.h   | 2 +-
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/asyncytmusic.cpp b/src/asyncytmusic.cpp
index b7fa5cb..7f58b79 100644
--- a/src/asyncytmusic.cpp
+++ b/src/asyncytmusic.cpp
@@ -9,6 +9,7 @@
 
 #include <QThread>
 #include <QDebug>
+#include <QCoreApplication>
 
 #include <pybind11/embed.h>
 
@@ -152,17 +153,19 @@ YTMusicThread::~YTMusicThread()
 
 AsyncYTMusic *YTMusicThread::operator->()
 {
-    return &m_ytm;
+    return m_ytm;
 }
 
 AsyncYTMusic &YTMusicThread::get()
 {
-    return m_ytm;
+    return *m_ytm;
 }
 
 YTMusicThread::YTMusicThread()
+    : m_ytm(new AsyncYTMusic())
 {
+    connect(QCoreApplication::instance(), &QCoreApplication::aboutToQuit, m_ytm, &QObject::deleteLater);
     setObjectName(QStringLiteral("YTMusicAPI"));
-    m_ytm.moveToThread(this);
+    m_ytm->moveToThread(this);
     start();
 }
diff --git a/src/asyncytmusic.h b/src/asyncytmusic.h
index ed5b6ca..48ac25a 100644
--- a/src/asyncytmusic.h
+++ b/src/asyncytmusic.h
@@ -156,5 +156,5 @@ public:
 private:
     YTMusicThread();
 
-    AsyncYTMusic m_ytm;
+    AsyncYTMusic *m_ytm;
 };
-- 
2.35.1

