From 463bd74b26d715434bbcde31b613717d8320ee34 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timoth=C3=A9e=20Ravier?= <tim@siosm.fr>
Date: Mon, 5 Sep 2022 18:24:48 +0200
Subject: [PATCH 2/2] Add option to not build KCM & KAuth support

---
 CMakeLists.txt     | 6 +++++-
 src/CMakeLists.txt | 4 +++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dd2432c..031dab7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,9 +32,13 @@ ecm_setup_version(${RELEASE_SERVICE_VERSION} VARIABLE_PREFIX KWALLETMANAGER
                   VERSION_HEADER "${CMAKE_CURRENT_BINARY_DIR}/src/kwalletmanager_version.h"
 )
 
+option(ENABLE_KAUTH "Enable KAuth support (Disabled for Flatpak builds)" ON)
+if (ENABLE_KAUTH)
+    find_package(KF5 ${KF5_MIN_VERSION} REQUIRED Auth)
+endif()
+
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED
     Archive
-    Auth
     Config
     ConfigWidgets
     CoreAddons
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bc9fa21..69ff480 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,2 +1,4 @@
-add_subdirectory(konfigurator)
+if (ENABLE_KAUTH)
+    add_subdirectory(konfigurator)
+endif()
 add_subdirectory(manager)
-- 
2.37.3

