diff --git a/src/main.cpp b/src/main.cpp
index 4c6bdf6..01236a0 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -15,6 +15,7 @@
 #include <QString>
 #include <QStringList>
 #include <QVariant>
+#include <QIcon>
 
 #ifdef Q_OS_ANDROID
 #include <QGuiApplication>
@@ -76,6 +77,8 @@ int main(int argc, char *argv[])
     engine.rootContext()->setContextObject(new KLocalizedContext(&engine));
     KLocalizedString::setApplicationDomain("kasts");
 
+    QIcon::setThemeName(QStringLiteral("breeze"));
+
     QCommandLineParser parser;
     parser.setApplicationDescription(i18n("Podcast Application"));
     QCommandLineOption addFeedOption(QStringList() << QStringLiteral("a") << QStringLiteral("add"),
