diff --git a/org.kde.kasts.appdata.xml b/org.kde.kasts.appdata.xml
index 4736b7f..c44b0bf 100644
--- a/org.kde.kasts.appdata.xml
+++ b/org.kde.kasts.appdata.xml
@@ -42,6 +42,7 @@
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-2.0-or-later</project_license>
   <url type="bugtracker">https://invent.kde.org/plasma-mobile/kasts/-/issues</url>
+  <url type="homepage">https://invent.kde.org/plasma-mobile/kasts</url>
   <developer_name>The KDE Community</developer_name>
   <developer_name xml:lang="ca">La comunitat KDE</developer_name>
   <developer_name xml:lang="ca-valencia">La comunitat KDE</developer_name>
diff --git a/src/main.cpp b/src/main.cpp
index 4c6bdf6..01236a0 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -15,6 +15,7 @@
 #include <QString>
 #include <QStringList>
 #include <QVariant>
+#include <QIcon>
 
 #ifdef Q_OS_ANDROID
 #include <QGuiApplication>
@@ -76,6 +77,8 @@ int main(int argc, char *argv[])
     engine.rootContext()->setContextObject(new KLocalizedContext(&engine));
     KLocalizedString::setApplicationDomain("kasts");
 
+    QIcon::setThemeName(QStringLiteral("breeze"));
+
     QCommandLineParser parser;
     parser.setApplicationDescription(i18n("Podcast Application"));
     QCommandLineOption addFeedOption(QStringList() << QStringLiteral("a") << QStringLiteral("add"),
