From 12d650e2f0d0e067ce828b8c4be97178d993703e Mon Sep 17 00:00:00 2001
From: Ryo Nakano <ryonakaknock3@gmail.com>
Date: Sat, 27 Dec 2025 01:47:56 +0900
Subject: [PATCH] Application: Force elementary stylesheet to prevent broken UI

---
 src/Application.vala | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Application.vala b/src/Application.vala
index 40d1149..58d7b6b 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -117,6 +117,9 @@ public class Sequeler.Application : Gtk.Application {
             // Maybe due to this issue in vala: https://gitlab.gnome.org/GNOME/vala/-/issues/957
             (BindingTransformFunc) granite_color_scheme_to_gtk_dark_theme
         );
+
+        gtk_settings.set_property("gtk-theme-name", "elementary");
+        gtk_settings.set_property("gtk-icon-theme-name", "elementary");
     }
 
     protected override void activate () {
-- 
2.43.0

