From c09cdc0ef7ec85ccf8fd4c2f27ec163096115e47 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Thu, 4 Jul 2024 00:25:40 +0200
Subject: [PATCH 2/2] Mobile: Fix switch pages

---
 mobile/components/DocumentView.qml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mobile/components/DocumentView.qml b/mobile/components/DocumentView.qml
index 4ebd7de21..7ad6a4229 100644
--- a/mobile/components/DocumentView.qml
+++ b/mobile/components/DocumentView.qml
@@ -187,33 +187,39 @@ QQC2.ScrollView {
                     target: mouseArea.currPageDelegate
                     property: "pageNumber"
                     value: root.document.currentPage
+                    restoreMode: Binding.RestoreNone
                 }
                 Binding {
                     target: mouseArea.currPageDelegate
                     property: "visible"
                     value: true
+                    restoreMode: Binding.RestoreNone
                 }
 
                 Binding {
                     target: mouseArea.prevPageDelegate
                     property: "pageNumber"
                     value: root.document.currentPage - 1
+                    restoreMode: Binding.RestoreNone
                 }
                 Binding {
                     target: mouseArea.prevPageDelegate
                     property: "visible"
                     value: !mouseArea.incrementing && root.document.currentPage > 0
+                    restoreMode: Binding.RestoreNone
                 }
 
                 Binding {
                     target: mouseArea.nextPageDelegate
                     property: "pageNumber"
                     value: root.document.currentPage + 1
+                    restoreMode: Binding.RestoreNone
                 }
                 Binding {
                     target: mouseArea.nextPageDelegate
                     property: "visible"
                     value: mouseArea.incrementing && root.document.currentPage < document.pageCount-1
+                    restoreMode: Binding.RestoreNone
                 }
                 
                 SequentialAnimation {
-- 
2.43.0

