From c5508160a0052f9e0f394682358dc54160e5b208 Mon Sep 17 00:00:00 2001
From: Sefa Eyeoglu <contact@scrumplex.net>
Date: Fri, 31 Dec 2021 19:47:48 +0100
Subject: [PATCH 2/2] refactor(packaging): install to /app

---
 vibrantLinux.pro | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/vibrantLinux.pro b/vibrantLinux.pro
index 9792f94..78c0e2c 100644
--- a/vibrantLinux.pro
+++ b/vibrantLinux.pro
@@ -59,7 +59,7 @@ FORMS += \
 
 # Default rules for deployment.
 qnx: target.path = /tmp/$${TARGET}/bin
-else: unix:!android: target.path = /usr/bin
+else: unix:!android: target.path = /app/bin
 !isEmpty(target.path): INSTALLS += target
 
 
@@ -69,13 +69,13 @@ desktopconfig.output = io.github.zeemzha.vibrantLinux.desktop
 QMAKE_SUBSTITUTES += \
 	desktopconfig
 
-desktop.path = /usr/share/applications
+desktop.path = /app/share/applications
 desktop.files += $${desktopconfig.output}
 
-icons.path = /usr/share/icons/hicolor/512x512/apps
+icons.path = /app/share/icons/hicolor/512x512/apps
 icons.files += assets/io.github.zeemzha.vibrantLinux.png
 
-appdata.path = /usr/share/metainfo
+appdata.path = /app/share/metainfo
 appdata.files += assets/io.github.zeemzha.vibrantLinux.appdata.xml
 
 INSTALLS += desktop icons appdata
-- 
2.34.1

