diff --git a/misc/xca.desktop b/misc/xca.desktop
index 4d6cf69c..d423e5bb 100644
--- a/misc/xca.desktop
+++ b/misc/xca.desktop
@@ -9,7 +9,7 @@ Comment[fr]=Création et gestion de certificats conformes à la norme X.509
 Comment[hr]=Grafičko korisničko sučelje za upravljanje životnim ciklusom X.509 sukladnih potvrda
 Comment[nl]=Een grafische gebruikers interface voor behandeling en uitgifte van X.509 conforme certificaten
 Exec=xca %F
-Icon=xca
+Icon=io.github.chris2511.xca
 Terminal=false
 Categories=Application;Utility;Qt;
 MimeType=application/x-xca-database;application/x-xca-template;application/x-x509-ca-cert;application/pkcs10;application/x-pkcs7-certificates;application/x-pkcs12;
