From 2af9f5f20c471e9bb5e1232634364c0a6fad0eb8 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Sat, 7 Sep 2019 12:26:42 -0500
Subject: [PATCH 3/4] Migrate away from deprecated icon names

---
 lib/GUI.pm | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/lib/GUI.pm b/lib/GUI.pm
index a68207a..7f328f7 100644
--- a/lib/GUI.pm
+++ b/lib/GUI.pm
@@ -247,17 +247,17 @@ sub add_config_panels {
         },
         {   link        => _( 'Whitelist' ),
             description => _( 'View or update scanning whitelist' ),
-            image       => 'gtk-new',
+            image       => 'document-new',
             button      => FALSE,
         },
         {   link        => _( 'Network' ),
             description => _( 'Edit proxy settings' ),
-            image       => 'gtk-network',
+            image       => 'network-workgroup',
             button      => FALSE,
         },
         {   link        => _( 'Scheduler' ),
             description => _( 'Schedule a scan or signature update' ),
-            image       => 'gtk-properties',
+            image       => 'document-properties',
             button      => FALSE,
         },
     );
@@ -319,7 +319,7 @@ sub add_update_panels {
     my @data = (
         {   link        => _( 'Update' ),
             description => _( 'Update antivirus signatures' ),
-            image       => 'gtk-goto-bottom',
+            image       => 'go-bottom',
             button      => FALSE,
         },
         {   link        => _( 'Update Assistant' ),
@@ -385,12 +385,12 @@ sub add_history_panels {
     my @data = (
         {   link        => _( 'History' ),
             description => _( 'View previous scans' ),
-            image       => 'gtk-edit',
+            image       => 'document-open-recent',
             button      => FALSE,
         },
         {   link        => _( 'Quarantine' ),
             description => _( 'Manage quarantined files' ),
-            image       => 'gtk-refresh',
+            image       => 'view-refresh',
             button      => FALSE,
         },
     );
@@ -451,12 +451,12 @@ sub add_analysis_panels {
     my @scan_data = (
         {   link        => _( 'Scan a file' ),
             description => _( 'Scan a file' ),
-            image       => 'gtk-file',
+            image       => 'text-x-preview',
             button      => FALSE,
         },
         {   link        => _( 'Scan a directory' ),
             description => _( 'Scan a directory' ),
-            image       => 'gtk-directory',
+            image       => 'folder',
             button      => FALSE,
         },
     );
@@ -476,7 +476,7 @@ sub add_analysis_panels {
     my @data = (
         {   link        => _( 'Analysis' ),
             description => _( "View a file's reputation" ),
-            image       => 'gtk-find',
+            image       => 'edit-find',
             button      => FALSE,
         },
     );
-- 
2.21.0

