diff --git a/src/SConstruct b/src/SConstruct
index 8592f42..0d7dbac 100644
--- a/src/SConstruct
+++ b/src/SConstruct
@@ -7,7 +7,7 @@ build_type = ARGUMENTS.get("build", "debug")
 
 godot_cpp_path = "../godot-cpp"
 libuiohook_path = "../libuiohook"
-libuiohook_path_linux = "../libuiohookLinux"
+libuiohook_path_linux = "../libuiohook/dist"
 libuiohook_path_macos = "../libuiohookMac"
 
 sources = [
@@ -114,7 +114,7 @@ elif platform.startswith("linux"):
 
     env.Append(LIBPATH=[
         f"{fixpath(godot_cpp_path)}/bin",
-        f"{fixpath(libuiohook_path_linux)}/lib",
+        f"{fixpath(libuiohook_path_linux)}/lib64",
         "/usr/lib/x86_64-linux-gnu"
     ])
 
@@ -132,8 +132,8 @@ elif platform.startswith("linux"):
 
     env.Append(CXXFLAGS=["-std=c++17", "-O2", "-fPIC"])
 
-    libuiohook_src = os.path.join(fixpath(libuiohook_path_linux), "lib", "libuiohook.so")
-    libuiohook_dst = os.path.join(output_dir, "libuiohook.so")
+    libuiohook_src = os.path.join(fixpath(libuiohook_path_linux), "lib64", "libuiohook.so.1.2.0")
+    libuiohook_dst = os.path.join(output_dir, "libuiohook.so.1")
     env.Command(target=libuiohook_dst, source=libuiohook_src, action=Copy("$TARGET", "$SOURCE"))
 
 elif platform.startswith("mac"):
