diff --git a/data/org.gnome.Ptyxis.metainfo.xml.in.in b/data/org.gnome.Ptyxis.metainfo.xml.in.in
index 59d8b72..8ebe0ac 100644
--- a/data/org.gnome.Ptyxis.metainfo.xml.in.in
+++ b/data/org.gnome.Ptyxis.metainfo.xml.in.in
@@ -27,12 +27,8 @@
       <li>Terminal inspector to help when writing terminal-based applications</li>
     </ul>
   </description>
-  <project_group>GNOME</project_group>
   <!-- developer_name tag deprecated with Appstream 1.0 -->
   <developer_name translatable="no">Christian Hergert</developer_name>
-  <developer id="org.gnome">
-    <name translatable="no">Christian Hergert</name>
-  </developer>
   <url type="homepage">https://gitlab.gnome.org/chergert/ptyxis</url>
   <url type="bugtracker">https://gitlab.gnome.org/chergert/ptyxis/issues</url>
   <url type="translate">https://gitlab.gnome.org/chergert/ptyxis/-/tree/main/po</url>
