--- a/SamTFE/serioussam.desktop	2024-01-25 13:57:23.483929853 +0300
+++ a/SamTFE/serioussam.desktop	2024-01-25 14:00:54.558511402 +0300
@@ -5,5 +5,6 @@
 Comment=Linux port of Serious Sam Classic The First Encounter
 Exec=serioussam
 TryExec=serioussam
-Icon=serioussam
+Icon=io.itch.tx00100xt.SeriousSamClassic
 Categories=Application;Game;
+Keywords=serious;sam;first;person;shooter;singleplayer;multiplayer
--- a/SamTSE/serioussamse.desktop	2024-01-25 13:57:23.483929853 +0300
+++ a/SamTSE/serioussamse.desktop	2024-01-25 14:00:54.558511402 +0300
@@ -5,5 +5,6 @@
 Comment=Linux port of Serious Sam Classic The Second Encounter
 Exec=serioussamse
 TryExec=serioussamse
-Icon=serioussam
+Icon=io.itch.tx00100xt.SeriousSamClassic
 Categories=Application;Game;
+Keywords=serious;sam;first;person;shooter;singleplayer;multiplayer;
