diff --git a/CustomMeshDeform/CustomMesh.cpp b/CustomMeshDeform/CustomMesh.cpp
index 9c11d80..e22f797 100644
--- a/CustomMeshDeform/CustomMesh.cpp
+++ b/CustomMeshDeform/CustomMesh.cpp
@@ -354,7 +354,7 @@ bool CustomMesh::remove_nearest_internal_point(Vector2 p, double max_dist) {
     return true;
 }
 
-void CustomMesh::deformations_3x3(double u, double v, bool redraw = false) { compute_interpolated_vertices(u,v,redraw); }
+void CustomMesh::deformations_3x3(double u, double v) { compute_interpolated_vertices(u,v); }
 
 void CustomMesh::sync_deformation_arrays() {
     std::lock_guard<std::recursive_mutex> lock(deform_mutex);
@@ -526,7 +526,7 @@ for (auto &layer_ref : deform_layers) {
 
 double CustomMesh::move_toward_double(double current, double target, double max_delta) { return move_toward_fast(current,target,max_delta); }
 
-void CustomMesh::compute_interpolated_vertices(double u_in, double v_in, bool redraw) {
+void CustomMesh::compute_interpolated_vertices(double u_in, double v_in) {
     std::lock_guard<std::recursive_mutex> lock(deform_mutex);
 
     int n = original_vertices.size();
@@ -542,7 +542,7 @@ void CustomMesh::compute_interpolated_vertices(double u_in, double v_in, bool re
     last_u = u;
     last_v = v;
 
-    if (redraw){
+    if (true){
 
         queue_redraw();
     }
diff --git a/CustomMeshDeform/CustomMesh.h b/CustomMeshDeform/CustomMesh.h
index 1eac7ba..a7c60a5 100644
--- a/CustomMeshDeform/CustomMesh.h
+++ b/CustomMeshDeform/CustomMesh.h
@@ -252,9 +252,9 @@ public:
     void remove_deform_layer(int index);
     int get_layer_count() const;
 
-    void compute_interpolated_vertices(double u_in, double v_in, bool redraw);
+    void compute_interpolated_vertices(double u_in, double v_in);
 
-    void deformations_3x3(double u, double v, bool redraw);
+    void deformations_3x3(double u, double v);
     void sync_deformation_arrays();
     void toggle_mesh_view();
 
