From 3285b62e89560bfa31e86686047c52c06910d9f9 Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Sat, 23 May 2026 14:46:47 +0200
Subject: [PATCH 5/5] Use the svg icon

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 mobile-datovka.pro | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mobile-datovka.pro b/mobile-datovka.pro
index 0673ae17..9bb41dd8 100644
--- a/mobile-datovka.pro
+++ b/mobile-datovka.pro
@@ -565,29 +565,29 @@ linux {
     target.path = "$${BINDIR}"
 
     desktop.path = "$${DATADIR}/applications"
     desktop.files += "linux/cz.nic.datovka.mobile.desktop"
 
     metainfo.path = "$${DATADIR}/metainfo"
     metainfo.files += "linux/cz.nic.datovka.mobile.metainfo.xml"
 
-    icon64.path = "$${DATADIR}/icons/hicolor/64x64/apps"
-    icon64.files += "res/datovka.png"
+    icon.path = "$${DATADIR}/icons/hicolor/scalable/apps"
+    icon.files += "res/cz.nic.datovka.mobile.svg"
 
     localisation.path = "$${LOCALE_INST_DIR}"
     localisation.files += \
 		res/locale/datovka_cs.qm \
 		res/locale/datovka_en.qm \
 		res/locale/datovka_uk.qm
 
     INSTALLS += target \
 		desktop \
 		metainfo \
-		icon64
+		icon
 }
 
 # iOS target.
 ios {
 
     exists(ios/codeversion.pri) {
         include(ios/codeversion.pri)
     }
-- 
2.53.0

