From 45bb2351c1df61651e4219321593ceaf64400672 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb.prv@gmx.de>
Date: Tue, 11 Dec 2018 18:45:20 +0100
Subject: [PATCH 7/8] Use QStandardPaths to locate resources

---
 src/frameworks/UBPlatformUtils_linux.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/frameworks/UBPlatformUtils_linux.cpp b/src/frameworks/UBPlatformUtils_linux.cpp
index 5bab1445..ebad94b1 100644
--- a/src/frameworks/UBPlatformUtils_linux.cpp
+++ b/src/frameworks/UBPlatformUtils_linux.cpp
@@ -32,6 +32,7 @@
 #include <QtGui>
 #include <QApplication>
 #include <QStandardPaths>
+#include <QDir>
 
 #include <unistd.h>
 #include <X11/keysym.h>
@@ -46,7 +47,10 @@ void UBPlatformUtils::init()
 
 QString UBPlatformUtils::applicationResourcesDirectory()
 {
-    return QString("/usr/share/openboard/");
+    QString appDataPath = QStandardPaths::locate(QStandardPaths::GenericDataLocation,
+                                                 QApplication::applicationName() + "/library/applications/",
+                                                 QStandardPaths::LocateDirectory);
+    return appDataPath + "../../";
 }
 
 void UBPlatformUtils::hideFile(const QString &filePath)
-- 
2.19.2

