--- a/src-tauri/src/geoip.rs
+++ b/src-tauri/src/geoip.rs
@@ -34,10 +34,9 @@ pub async fn lookup(app: &AppHandle, ips: Vec<IpAddr>) -> Vec<LookupResult> {
     let reader_handle: State<MmdbReaderHandle> = app.state();
     let mut reader = reader_handle.0.lock().await;
     if reader.is_none() {
-        let dbip_path = app
-            .path_resolver()
-            .resolve_resource("dbip.mmdb")
-            .expect("failed to resolve resource");
+        let dbip_path =
+            <std::path::PathBuf as std::str::FromStr>::from_str("/app/lib/trgui-ng/dbip.mmdb")
+                .unwrap();

         if dbip_path.is_file() {
             match maxminddb::Reader::open_readfile(&dbip_path) {
