--- a/webpack.common.js
+++ b/webpack.common.js
@@ -29,7 +29,7 @@ export const __dirname = url.fileURLToPath(new URL(".", import.meta.url));
 const tsConfig = JSON.parse(await readFile("./tsconfig.json"));
 const tauriConf = JSON.parse(await readFile("./src-tauri/tauri.conf.json"));

-const gitVersion = execaSync("git", ["describe", "--tags", "--dirty", "--always"]);
+const gitVersion = execaSync("git", ["describe", "--tags", "--always"]);
 const versionInfo = `{
     "gitVersion": "${gitVersion.stdout}",
     "backendVersion": "${tauriConf.package.version}",
