diff -ur mupdf-1.23.3-source.orig/Makefile mupdf-1.23.3-source/Makefile
--- mupdf-1.23.3-source.orig/Makefile	2023-09-05 13:51:19.000000000 +0200
+++ mupdf-1.23.3-source/Makefile	2023-09-08 09:28:01.165486441 +0200
@@ -3,7 +3,7 @@
 -include user.make
 
 ifndef build
-  build := release
+  build := debug
 endif
 
 default: all
@@ -290,17 +290,19 @@
 $(THREAD_LIB) : $(THREAD_OBJ)
 $(PKCS7_LIB) : $(PKCS7_OBJ)
 else
-MUPDF_LIB = $(OUT)/libmupdf.a
+MUPDF_LIB = libmupdf.so
 LIBS_TO_INSTALL_IN_LIB = $(MUPDF_LIB) $(THIRD_LIB)
-THIRD_LIB = $(OUT)/libmupdf-third.a
+THIRD_LIB = 
+MUPDF_STATIC = $(OUT)/libmupdf.a
 ifneq ($(USE_SYSTEM_GLUT),yes)
 THIRD_GLUT_LIB = $(OUT)/libmupdf-glut.a
 endif
 THREAD_LIB = $(OUT)/libmupdf-threads.a
 PKCS7_LIB = $(OUT)/libmupdf-pkcs7.a
 
-$(MUPDF_LIB) : $(MUPDF_OBJ)
-$(THIRD_LIB) : $(THIRD_OBJ)
+$(MUPDF_LIB) : $(MUPDF_OBJ) $(THIRD_OBJ)
+	$(QUIET_LINK) $(CC) $(LDFLAGS) --shared -Wl,-soname -Wl,$(MUPDF_LIB) -o $@ $^ $(THIRD_LIBS) $(LIBS)
+$(MUPDF_STATIC): $(MUPDF_OBJ) $(THIRD_OBJ)
 $(THIRD_GLUT_LIB) : $(THIRD_GLUT_OBJ)
 $(THREAD_LIB) : $(THREAD_OBJ)
 $(PKCS7_LIB) : $(PKCS7_OBJ)
diff -ur mupdf-1.23.3-source.orig/Makethird mupdf-1.23.3-source/Makethird
--- mupdf-1.23.3-source.orig/Makethird	2023-09-05 13:51:19.000000000 +0200
+++ mupdf-1.23.3-source/Makethird	2023-09-08 01:16:56.785811250 +0200
@@ -2,16 +2,16 @@
 
 ifeq ($(USE_SYSTEM_LIBS),yes)
   USE_SYSTEM_FREETYPE := yes
-  USE_SYSTEM_GUMBO := yes
+  USE_SYSTEM_GUMBO := no
   USE_SYSTEM_HARFBUZZ := yes
-  USE_SYSTEM_JBIG2DEC := yes
+  USE_SYSTEM_JBIG2DEC := no
   USE_SYSTEM_JPEGXR := no # not available
   USE_SYSTEM_LCMS2 := no # lcms2mt is strongly preferred
   USE_SYSTEM_LIBJPEG := yes
   USE_SYSTEM_MUJS := no # not available
-  USE_SYSTEM_OPENJPEG := yes
+  USE_SYSTEM_OPENJPEG := no
   USE_SYSTEM_ZLIB := yes
-  USE_SYSTEM_GLUT := yes
+  USE_SYSTEM_GLUT := no
   USE_SYSTEM_CURL := yes
   USE_SYSTEM_LEPTONICA := yes
   USE_SYSTEM_TESSERACT := yes
