diff --git a/src/main.cpp b/src/main.cpp
index 1633d837506ccc78f095ef9427f2e399dc2fa3d8..2c329f544cffdc6c8d3531e99b59686db2f3be65 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -59,6 +59,10 @@ int main(int argc, char *argv[])
     QQuickStyle::setStyle(QStringLiteral("Material"));
 #else
     QApplication app(argc, argv);
+    // Default to org.kde.desktop style unless the user forces another style
+    if (qEnvironmentVariableIsEmpty("QT_QUICK_CONTROLS_STYLE")) {
+        QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"));
+    }
 #endif
 
     QApplication::setOrganizationName("KDE");
