From d377de0f3aeea4c451c924c650ea97b63cd70840 Mon Sep 17 00:00:00 2001
From: Waqar Ahmed <waqar.17a@gmail.com>
Date: Tue, 14 Mar 2023 22:42:59 +0500
Subject: [PATCH] Use `test -x exec` when in flatpak

---
 src/session/Session.cpp | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/session/Session.cpp b/src/session/Session.cpp
index 0a5d813e7..7c62a4627 100644
--- a/src/session/Session.cpp
+++ b/src/session/Session.cpp
@@ -378,6 +378,22 @@ QString Session::checkProgram(const QString &program)
         return QString();
     }
 
+#ifndef Q_OS_WIN
+#if KCOREADDONS_VERSION >= QT_VERSION_CHECK(5, 97, 0)
+    if (KSandbox::isFlatpak()) {
+        QProcess proc;
+        // run "test -x exec" on the host to see if the shell is executable
+        proc.setProgram(QStringLiteral("test"));
+        proc.setArguments(QStringList{QStringLiteral("-x"), exec});
+        KSandbox::startHostProcess(proc, QProcess::ReadOnly);
+        if (proc.waitForStarted() && proc.waitForFinished(-1)) {
+            return proc.exitCode() == 0 ? exec : QString();
+        }
+        return {};
+    }
+#endif
+#endif // Q_OS_WIN
+
     QFileInfo info(exec);
     if (info.isAbsolute() && info.exists() && info.isExecutable()) {
         return exec;
-- 
GitLab

