diff --git a/d1/include/pstypes.h b/d1/include/pstypes.h
index 190b938..e6abd6b 100644
--- a/d1/include/pstypes.h
+++ b/d1/include/pstypes.h
@@ -72,7 +72,7 @@ typedef ubyte bool;
 // and since this file is included everywhere, it's here.
 #if  defined(__i386__) || defined(__ia64__) || defined(_WIN32) || \
 (defined(__alpha__) || defined(__alpha)) || \
-defined(__arm__) || defined(ARM) || \
+defined(__arm__) || defined(ARM) || defined(__aarch64__) || \
 (defined(__mips__) && defined(__MIPSEL__)) || \
 defined(__SYMBIAN32__) || \
 defined(__x86_64__) || \
