diff -r 214a94e9366c src/cpu/aarch64/vm/nativeInst_aarch64.cpp
--- openjdk/hotspot/src/cpu/aarch64/vm/nativeInst_aarch64.cpp	Mon Jul 17 12:11:32 2017 +0000
+++ openjdk/hotspot/src/cpu/aarch64/vm/nativeInst_aarch64.cpp	Mon Jul 24 16:23:14 2017 +0100
@@ -343,7 +343,7 @@
   CodeBuffer cb(code_pos, instruction_size);
   MacroAssembler a(&cb);
 
-  a.mov(rscratch1, entry);
+  a.movptr(rscratch1, (uintptr_t)entry);
   a.br(rscratch1);
 
   ICache::invalidate_range(code_pos, instruction_size);


