From b3a0f7d6f5568e8685800f6123d368691e4f227c Mon Sep 17 00:00:00 2001
From: Keeyou <keeyou-cn@outlook.com>
Date: Thu, 8 Aug 2024 12:04:36 +0800
Subject: [PATCH] flatpak: support aarch64 build

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bbf0cf48..dd4fc42d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1837,6 +1837,11 @@ if (FREEBSD AND OS_X86)
   add_compile_options(-msse2)
 endif()
 
+# make boringssl capable on aarch64
+if (LINUX AND OS_AARCH64)
+  add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-march=armv8-a+crypto>)
+endif()
+
 # *****************************************************************************************
 #           Libc++ Library
 # *****************************************************************************************
-- 
2.45.2

