diff --git a/activity/activity.info b/activity/activity.info
index dd81b6e..9e0be89 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,7 +1,10 @@
 [Activity]
 name = Abacus
 activity_version = 60.1
+release_date = 2018-04-12
 license = GPLv3+
+metadata_license = CC0-1.0
+update_contact = tch@sugarlabs.org
 bundle_id = org.sugarlabs.AbacusActivity
 exec = sugar-activity3 AbacusActivity.AbacusActivity
 icon = activity-abacus
@@ -11,4 +14,5 @@ summary = A tool for simple arithmetic calculations
 category = maths
 repository = https://github.com/sugarlabs/activity-abacus
 tags =  Maths
-screenshots = https://github.com/sugarlabs/activity-abacus/blob/master/screenshots/en/1.png https://github.com/sugarlabs/activity-abacus/blob/master/screenshots/en/2.png https://github.com/sugarlabs/activity-abacus/blob/master/screenshots/en/3.png
+description = Abacus lets the learner explore different representations of numbers using different mechanical counting systems developed by the ancient Romans and Chinese. There are several different variants available for exploration.
+screenshots = https://raw.githubusercontent.com/sugarlabs/activity-abacus/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/activity-abacus/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/activity-abacus/master/screenshots/en/3.png
