--- vara-0.1.230619/src/ui.c	2024-05-19 05:37:52.000000000 +0530
+++ /tmp/ui.c	2024-05-19 08:41:10.122656344 +0530
@@ -1620,6 +1620,7 @@
 	GtkAboutDialog *dlg = (GtkAboutDialog*) (gtk_about_dialog_new());
 	gtk_about_dialog_set_program_name(dlg, PROGRAM_NAME);
 	gtk_about_dialog_set_comments(dlg, "Just draw.");
+	gtk_about_dialog_set_logo_icon_name(dlg, "in.co.nandakumar.vara");
 	const char * _ngg_tmp_65[2];
 	_ngg_tmp_65[0] = "Nandakumar Edamana";
 	_ngg_tmp_65[1] = NULL;
