--- XaraLX-0.7r1766/Scripts/build-resources.pl.orig	2007-01-31 13:42:10.000000000 +0100
+++ XaraLX-0.7r1766/Scripts/build-resources.pl	2007-02-09 19:57:01.930281000 +0100
@@ -461,8 +461,10 @@
 print CHECK "$checksum\n";
 close CHECK;
 
-$ret = system("$zip","-9","-j","-q","$outputdir/xrc/resources.xrs","$outputdir/xrc/$xaralanguage/xrc.check","$outputdir/xrc/dialogs.xrc","$outputdir/xrc/strings.xrc","$outputdir/xrc/strings.lst",@resfiles);
-die "Could not make resources.xrs: $!" if ($ret);
+my @cmd =    ("$zip","-9","-j","-q","$outputdir/xrc/resources.xrs","$outputdir/xrc/$xaralanguage/xrc.check","$outputdir/xrc/dialogs.xrc","$outputdir/xrc/strings.xrc","$outputdir/xrc/strings.lst",@resfiles);
+
+$ret = system(@cmd);
+die "@cmd\nCould not make resources.xrs: $! $@" if ($ret);
 $ret = system("$topdir/Scripts/bin2cpp.pl -f CamResource::GetBinaryFileInfo $outputdir/xrc/resources.xrs $outputdir/resources.h");
 die "Could not make resources.cpp: $!" if ($ret);
 
--- XaraLX-0.7r1766/wxOil/stdwx.h.orig	2007-01-31 13:42:25.000000000 +0100
+++ XaraLX-0.7r1766/wxOil/stdwx.h	2007-02-09 19:32:14.966725000 +0100
@@ -99,6 +99,8 @@
 // First the main include file
 #include <wx/wx.h>
 
+#undef ATTRIBUTE_PRINTF		// wx/defs.h defines a bad one, if wxUSE_UNICODE. Get a better one from libxml/xmlversion.h
+
 // Now specialized bits we might not have got through the above
 #include <wx/bitmap.h>
 #include <wx/cmdline.h>
