diff -ruN a/avogadro/CMakeLists.txt b/avogadro/CMakeLists.txt
--- a/avogadro/CMakeLists.txt	2021-04-01 14:29:44.973903417 +0700
+++ b/avogadro/CMakeLists.txt	2021-04-01 14:30:42.634838987 +0700
@@ -91,8 +91,13 @@
 
 elseif(UNIX)
   install(FILES "icons/avogadro2.desktop" DESTINATION "${INSTALL_XDG_APPS_DIR}")
-  install(FILES "icons/avogadro.png" DESTINATION "${INSTALL_XDG_ICON_DIR}"
-    RENAME "avogadro2.png")
+  foreach (iconsize IN ITEMS 64 128 256 512)
+   install(
+     FILES       "icons/avogadro2_${iconsize}.png"
+     DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/${iconsize}/apps"
+     RENAME      avogadro2.png
+     COMPONENT   runtime)
+  endforeach ()
 elseif(WIN32)
   list(APPEND avogadro_srcs icons/avogadro.rc)
 endif()
