From d0f8fe6d45ad15dfc38a8931f88eb7888fab561a Mon Sep 17 00:00:00 2001
From: Lukas Holecek <hluk@email.cz>
Date: Sun, 25 Feb 2018 15:29:28 +0100
Subject: [PATCH] Add releases section to AppData

---
 shared/copyq.appdata.xml.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/shared/copyq.appdata.xml.in b/shared/copyq.appdata.xml.in
index 18070627..51d8377f 100644
--- a/shared/copyq.appdata.xml.in
+++ b/shared/copyq.appdata.xml.in
@@ -25,6 +25,10 @@
     <url type="donation">https://www.bountysource.com/teams/copyq</url>
     <url type="translate">https://hosted.weblate.org/engage/copyq/</url>
 
+    <releases>
+        <release version="3.2.0" date="2018-02-18" />
+    </releases>
+
     <screenshots>
         <screenshot type="default">
             <image>https://hluk.github.io/CopyQ/images/application.png</image>
