diff -ruN a/Clients/ParaView/CMakeLists.txt b/Clients/ParaView/CMakeLists.txt
--- a/Clients/ParaView/CMakeLists.txt	2021-01-24 01:01:44.000000000 +0700
+++ b/Clients/ParaView/CMakeLists.txt	2021-03-30 15:53:35.373084134 +0700
@@ -127,7 +127,7 @@
     FILES       "${CMAKE_CURRENT_BINARY_DIR}/org.paraview.ParaView.desktop"
     DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications"
     COMPONENT   runtime)
-  foreach (iconsize IN ITEMS 22x22 32x32 96x96)
+  foreach (iconsize IN ITEMS 22x22 32x32 96x96 128x128)
     install(
       FILES       "pvIcon-${iconsize}.png"
       DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/${iconsize}/apps"
