diff -ruN a/ElmerGUI/Application/twod/renderarea.cpp b/ElmerGUI/Application/twod/renderarea.cpp
--- a/ElmerGUI/Application/twod/renderarea.cpp	2020-11-10 20:52:44.000000000 +0100
+++ b/ElmerGUI/Application/twod/renderarea.cpp	2022-04-05 13:39:36.794890606 +0200
@@ -47,6 +47,7 @@
 #include <math.h>
 #include "renderarea.h"
 #include "curveeditor.h"
+#include <QPainterPath>
 
 using namespace std;
 
