diff --git a/pkg/desktop-appdata/speedcrunch.appdata.xml.in b/pkg/desktop-appdata/speedcrunch.appdata.xml.in
index 3e89f973..37bc190a 100644
--- a/pkg/desktop-appdata/speedcrunch.appdata.xml.in
+++ b/pkg/desktop-appdata/speedcrunch.appdata.xml.in
@@ -5,6 +5,7 @@
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-2.0+</project_license>
   <update_contact>speedcrunch@googlegroups.com</update_contact>
+  <content_rating type="oars-1.0" />
 
   <!-- TRANSLATORS: Name of the application, usually shouldn't be translated. -->
   <_name>SpeedCrunch</_name>
@@ -54,6 +55,13 @@
   <url type="bugtracker">https://bitbucket.org/heldercorreia/speedcrunch/issues</url>
   <url type="help">http://speedcrunch.org/introduction.html</url>
 
+  <releases>
+    <release version="0.12" date="2016-11-29"/>
+    <release version="0.11" date="2014-01-30"/>
+    <release version="0.10.1" date="2008-06-03"/>
+    <release version="0.10" date="2008-04-01"/>
+  </releases>
+
   <!-- We can't use the <translations> tag since our qm files are baked into the
        executable so we just use a static list instead. See:
        https://blogs.gnome.org/hughsie/2016/01/25/appdata-and-the-gettext-domain/#comment-5137
diff --git a/pkg/speedcrunch.appdata.xml b/pkg/speedcrunch.appdata.xml
index cc3a33c8..6d18b7ab 100644
--- a/pkg/speedcrunch.appdata.xml
+++ b/pkg/speedcrunch.appdata.xml
@@ -4,6 +4,7 @@
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-2.0+</project_license>
   <update_contact>speedcrunch@googlegroups.com</update_contact>
+  <content_rating type="oars-1.0"/>
 
   
   <name>SpeedCrunch</name>
@@ -103,6 +104,13 @@
   <url type="bugtracker">https://bitbucket.org/heldercorreia/speedcrunch/issues</url>
   <url type="help">http://speedcrunch.org/introduction.html</url>
 
+  <releases>
+    <release version="0.12" date="2016-11-29"/>
+    <release version="0.11" date="2014-01-30"/>
+    <release version="0.10.1" date="2008-06-03"/>
+    <release version="0.10" date="2008-04-01"/>
+  </releases>
+
   
   <translation/>
   <languages>
