diff -Naur Aegisub.orig/automation/Makefile Aegisub/automation/Makefile
--- Aegisub.orig/automation/Makefile	2019-09-02 10:05:32.129969717 +0200
+++ Aegisub/automation/Makefile	2019-09-02 10:50:18.689017377 +0200
@@ -12,10 +12,16 @@
 $(DESTDIR)$(P_DATA)/automation/%: $(d)%
 	$(MKDIR_INSTALL)
 
-aegisub-lua_OBJ := $(d)tests/aegisub.o $(TOP)lib/libaegisub.a $(TOP)lib/libluabins.a $(LIBS_LUA)
+aegisub-lua_OBJ := $(d)tests/aegisub.o $(TOP)lib/libaegisub.a $(TOP)lib/libluabins.a
 aegisub-lua_CPPFLAGS := $(CPPFLAGS_BOOST) $(CFLAGS_LUA) -I$(TOP)libaegisub/include -I$(TOP)src $(CXXFLAGS_WX)
 aegisub-lua_LIBS := $(LIBS_WX) $(LIBS_BOOST) $(LIBS_ICU) $(LIBS_PTHREAD)
 
+ifeq (no, $(SYSTEM_LUAJIT))
+aegisub-lua_OBJ += $(LIBS_LUA)
+else
+aegisub-lua_LIBS += $(LIBS_LUA)
+endif
+
 PROGRAM += $(d)aegisub-lua
 
 test-automation: $(PROGRAM)
diff -Naur Aegisub.orig/configure.ac Aegisub/configure.ac
--- Aegisub.orig/configure.ac	2019-09-02 10:05:32.160970021 +0200
+++ Aegisub/configure.ac	2019-09-02 10:29:33.619427536 +0200
@@ -429,13 +429,15 @@
       [AC_SUBST([LUAJIT_CFLAGS], ['-I$(TOP)vendor/luajit/include'])
        AC_SUBST([LUAJIT_LIBS], ['$(TOP)vendor/luajit/src/libluajit.a'])])
 
+AC_SUBST(with_system_luajit)
+
 # We also need a Lua binary to run part of the build system
 # Which version doesn't matter as the scripts are portable between them
-AC_CHECK_PROGS([LUA], [lua luajit lua5.3 lua-5.3 lua5.2 lua-5.2 lua5.1 lua-5.1])
+AC_PATH_PROGS([LUA], [luajit lua lua5.3 lua-5.3 lua5.2 lua-5.2 lua5.1 lua-5.1])
 
 # If the user doesn't have an installed copy of Lua, just use the one built
 # as part of building LuaJIT
-AS_IF([test -z $LUA], [LUA="$srcdir/vendor/luajit/src/host/minilua"])
+AS_IF([test -z $LUA], [LUA='$(TOP)vendor/luajit/src/host/minilua'])
 
 ######################################################
 # Debugging support
diff -Naur Aegisub.orig/header.mk Aegisub/header.mk
--- Aegisub.orig/header.mk	2019-09-02 10:05:32.172970138 +0200
+++ Aegisub/header.mk	2019-09-02 10:10:07.683712563 +0200
@@ -12,8 +12,11 @@
 	src \
 	tests \
 	tools \
-	vendor/luabins \
-	vendor/luajit
+	vendor/luabins
+
+ifeq (no, $(SYSTEM_LUAJIT))
+subdirs += vendor/luajit
+endif
 
 subdirs := $(addprefix $(TOP),$(addsuffix /Makefile,$(subdirs)))
 
diff -Naur Aegisub.orig/Makefile.inc.in Aegisub/Makefile.inc.in
--- Aegisub.orig/Makefile.inc.in	2019-09-02 10:05:32.129969717 +0200
+++ Aegisub/Makefile.inc.in	2019-09-02 10:35:49.901228138 +0200
@@ -9,6 +9,7 @@
 HAVE_OSS        = @with_oss@
 HAVE_PORTAUDIO  = @with_portaudio@
 HAVE_UCHARDET   = @with_uchardet@
+SYSTEM_LUAJIT   = @with_system_luajit@
 
 ##############
 # BUILD OUTPUT
@@ -79,7 +80,11 @@
 CFLAGS_ICU         = @ICU_I18N_CFLAGS@
 CFLAGS_LIBASS      = @LIBASS_CFLAGS@
 CFLAGS_LIBPULSE    = @LIBPULSE_CFLAGS@
+ifeq (no, $(SYSTEM_LUAJIT))
 CFLAGS_LUA         = @LUAJIT_CFLAGS@
+else
+CFLAGS_LUA         = `pkg-config --cflags luajit`
+endif
 CFLAGS_OPENAL      = @OPENAL_CFLAGS@
 CFLAGS_OSS         = @OSS_CFLAGS@
 CFLAGS_PORTAUDIO   = @PORTAUDIO_CFLAGS@
@@ -97,14 +102,22 @@
 LIBS_ICU           = @ICU_UC_LIBS@ @ICU_I18N_LIBS@
 LIBS_LIBASS        = @LIBASS_LIBS@
 LIBS_LIBPULSE      = @LIBPULSE_LIBS@
+ifeq (no, $(SYSTEM_LUAJIT))
 LIBS_LUA           = @LUAJIT_LIBS@
+else
+LIBS_LUA           = `pkg-config --libs luajit`
+endif
 LIBS_OPENAL        = @OPENAL_LIBS@
 LIBS_PORTAUDIO     = @PORTAUDIO_LIBS@
 LIBS_PTHREAD       = @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 LIBS_UCHARDET      = @UCHARDET_LIBS@
 
 ifeq (yes, $(BUILD_DARWIN))
+ifeq (no, $(SYSTEM_LUAJIT))
 LIBS_LUA           = $(TOP)vendor/luajit/src/libluajit-aegisub.so
+else
+LIBS_LUA           = `pkg-config --libs luajit`
+endif
 endif
 
 FONTCONFIG_CONF_DIR = @FONTCONFIG_CONF_DIR@
diff -Naur Aegisub.orig/src/libresrc/Makefile Aegisub/src/libresrc/Makefile
--- Aegisub.orig/src/libresrc/Makefile	2019-09-02 10:05:32.269971089 +0200
+++ Aegisub/src/libresrc/Makefile	2019-09-02 10:30:30.630003366 +0200
@@ -10,7 +10,11 @@
 
 $(resrc_OBJ): $(d)default_config.h $(d)bitmap.h $(d)default_config.cpp $(d)bitmap.cpp
 
+ifeq (no, $(SYSTEM_LUAJIT))
 RESPACK := cd $(TOP)src/libresrc; $(BIN_LUA) $(TOP)tools/respack.lua
+else
+RESPACK := cd $(TOP)src/libresrc; luajit $(TOP)tools/respack.lua
+endif
 
 $(d)bitmap.cpp: $(d)bitmap.h
 $(d)default_config.cpp: $(d)default_config.h
diff -Naur Aegisub.orig/src/Makefile Aegisub/src/Makefile
--- Aegisub.orig/src/Makefile	2019-09-02 10:56:26.446741883 +0200
+++ Aegisub/src/Makefile	2019-09-02 10:55:42.936301226 +0200
@@ -198,10 +198,17 @@
 $(d)subtitles_provider_libass.o_FLAGS   := $(CFLAGS_LIBASS) -Wno-narrowing
 $(d)text_file_reader.o_FLAGS            := -D_X86_
 $(d)video_provider_manager.o_FLAGS      := $(CFLAGS_FFMS2)
+ifeq (no, $(SYSTEM_LUAJIT))
 $(d)auto4_lua.o_FLAGS                   := $(CFLAGS_LUA)
 $(d)auto4_lua_assfile.o_FLAGS           := $(CFLAGS_LUA)
 $(d)auto4_lua_dialog.o_FLAGS            := $(CFLAGS_LUA)
 $(d)auto4_lua_progresssink.o_FLAGS      := $(CFLAGS_LUA)
+else
+$(d)auto4_lua.o_FLAGS                   := `pkg-config --cflags luajit`
+$(d)auto4_lua_assfile.o_FLAGS           := `pkg-config --cflags luajit`
+$(d)auto4_lua_dialog.o_FLAGS            := `pkg-config --cflags luajit`
+$(d)auto4_lua_progresssink.o_FLAGS      := `pkg-config --cflags luajit`
+endif
 
 $(src_OBJ): $(d)libresrc/bitmap.h $(d)libresrc/default_config.h
 
diff -Naur Aegisub.orig/tools/Makefile Aegisub/tools/Makefile
--- Aegisub.orig/tools/Makefile	2019-09-02 10:05:32.293971324 +0200
+++ Aegisub/tools/Makefile	2019-09-02 10:23:58.569045427 +0200
@@ -12,6 +12,8 @@
 
 PROGRAM += $(d)repack-thes-dict
 
+ifeq (no, $(SYSTEM_LUAJIT))
 $(TOP)tools/respack.lua: $(shell command -v "$(BIN_LUA)")
+endif
 
 include $(TOP)Makefile.target
