diff -Naur Aegisub-3.2.2.orig/header.mk Aegisub-3.2.2/header.mk
--- Aegisub-3.2.2.orig/header.mk	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/header.mk	2018-12-07 15:43:39.223405318 +0100
@@ -13,7 +13,6 @@
 	tests \
 	tools \
 	vendor/luabins \
-	vendor/luajit \
 	vendor/universalchardet
 
 subdirs := $(addprefix $(TOP),$(addsuffix /Makefile,$(subdirs)))
diff -Naur Aegisub-3.2.2.orig/Makefile.inc.in Aegisub-3.2.2/Makefile.inc.in
--- Aegisub-3.2.2.orig/Makefile.inc.in	2018-12-07 15:41:32.007726000 +0100
+++ Aegisub-3.2.2/Makefile.inc.in	2018-12-07 15:48:07.631837445 +0100
@@ -78,7 +78,7 @@
 CFLAGS_ICU         = @ICU_I18N_CFLAGS@
 CFLAGS_LIBASS      = @LIBASS_CFLAGS@
 CFLAGS_LIBPULSE    = @LIBPULSE_CFLAGS@
-CFLAGS_LUA         = -I$(TOP)vendor/luajit/include
+CFLAGS_LUA         = `pkg-config --cflags luajit`
 CFLAGS_OPENAL      = @OPENAL_CFLAGS@
 CFLAGS_OSS         = @OSS_CFLAGS@
 CFLAGS_PORTAUDIO   = @PORTAUDIO_CFLAGS@
@@ -96,13 +96,13 @@
 LIBS_ICU           = @ICU_UC_LIBS@ @ICU_I18N_LIBS@
 LIBS_LIBASS        = @LIBASS_LIBS@
 LIBS_LIBPULSE      = @LIBPULSE_LIBS@
-LIBS_LUA           = $(TOP)vendor/luajit/src/libluajit.a
+LIBS_LUA           = `pkg-config --libs luajit`
 LIBS_OPENAL        = @OPENAL_LIBS@
 LIBS_PORTAUDIO     = @PORTAUDIO_LIBS@
 LIBS_PTHREAD       = @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 
 ifeq (yes, $(BUILD_DARWIN))
-LIBS_LUA           = $(TOP)vendor/luajit/src/libluajit-aegisub.so
+LIBS_LUA           = `pkg-config --libs luajit`
 endif
 
 FONTCONFIG_CONF_DIR = @FONTCONFIG_CONF_DIR@
diff -Naur Aegisub-3.2.2.orig/src/libresrc/Makefile Aegisub-3.2.2/src/libresrc/Makefile
--- Aegisub-3.2.2.orig/src/libresrc/Makefile	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/src/libresrc/Makefile	2018-12-07 15:43:39.223405318 +0100
@@ -10,7 +10,7 @@
 
 $(resrc_OBJ): $(d)default_config.h $(d)bitmap.h $(d)default_config.cpp $(d)bitmap.cpp
 
-RESPACK := cd $(TOP)src/libresrc; $(TOP)vendor/luajit/src/host/minilua $(TOP)tools/respack.lua
+RESPACK := cd $(TOP)src/libresrc; $(TOP)tools/respack.lua
 
 $(d)bitmap.cpp: $(d)bitmap.h
 $(d)default_config.cpp: $(d)default_config.h
diff -Naur Aegisub-3.2.2.orig/src/Makefile Aegisub-3.2.2/src/Makefile
--- Aegisub-3.2.2.orig/src/Makefile	2018-12-07 15:42:28.265026000 +0100
+++ Aegisub-3.2.2/src/Makefile	2018-12-07 15:43:39.223405318 +0100
@@ -191,10 +191,10 @@
 $(d)subtitles_provider_libass.o_FLAGS   := $(CFLAGS_LIBASS)
 $(d)text_file_reader.o_FLAGS            := -D_X86_
 $(d)video_provider_manager.o_FLAGS      := $(CFLAGS_FFMS2)
-$(d)auto4_lua.o_FLAGS                   := -I$(TOP)vendor/luajit/include
-$(d)auto4_lua_assfile.o_FLAGS           := -I$(TOP)vendor/luajit/include
-$(d)auto4_lua_dialog.o_FLAGS            := -I$(TOP)vendor/luajit/include
-$(d)auto4_lua_progresssink.o_FLAGS      := -I$(TOP)vendor/luajit/include
+$(d)auto4_lua.o_FLAGS                   := `pkg-config --cflags luajit`
+$(d)auto4_lua_assfile.o_FLAGS           := `pkg-config --cflags luajit`
+$(d)auto4_lua_dialog.o_FLAGS            := `pkg-config --cflags luajit`
+$(d)auto4_lua_progresssink.o_FLAGS      := `pkg-config --cflags luajit`
 
 $(src_OBJ): $(d)libresrc/bitmap.h $(d)libresrc/default_config.h
 
diff -Naur Aegisub-3.2.2.orig/tools/Makefile Aegisub-3.2.2/tools/Makefile
--- Aegisub-3.2.2.orig/tools/Makefile	2018-12-07 15:41:32.008726000 +0100
+++ Aegisub-3.2.2/tools/Makefile	2018-12-07 15:43:39.224405324 +0100
@@ -12,6 +12,4 @@
 
 PROGRAM += $(d)repack-thes-dict
 
-$(TOP)tools/respack.lua: $(TOP)vendor/luajit/src/host/minilua
-
 include $(TOP)Makefile.target
diff -Naur Aegisub-3.2.2.orig/tools/respack.lua Aegisub-3.2.2/tools/respack.lua
--- Aegisub-3.2.2.orig/tools/respack.lua	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/tools/respack.lua	2018-12-07 15:43:39.224405324 +0100
@@ -1,4 +1,4 @@
-#!../vendor/luajit/src/host/minilua
+#!/usr/bin/env luajit
 -- Copyright (c) 2014, Thomas Goyne <plorkyeran@aegisub.org>
 --
 -- Permission to use, copy, modify, and distribute this software for any
@@ -44,7 +44,7 @@
 for line in manifest:lines() do
   if line:find('.') then
     local file = try_open(path..line, 'rb')
-    local id = line:gsub('^.*/', ''):gsub('\.[a-z]+$', '')
+    local id = line:gsub('^.*/', ''):gsub('%.[a-z]+$', '')
     out_cpp:write("const unsigned char " .. id .. "[] = {")
 
     local len = 0
