diff -Naur Aegisub-3.2.2.orig/configure Aegisub-3.2.2/configure
--- Aegisub-3.2.2.orig/configure	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/configure	2018-12-05 22:39:50.574859304 +0100
@@ -4092,7 +4092,7 @@
 if test x$build_darwin = xyes; then :
   AEGISUB_COMMAND="aegisub"
 else
-  AEGISUB_COMMAND="aegisub-3.2"
+  AEGISUB_COMMAND="aegisub"
 fi
 
 
@@ -4106,7 +4106,7 @@
 if test x$build_darwin = xyes; then :
   AEGISUB_CATALOG="aegisub"
 else
-  AEGISUB_CATALOG="aegisub-32"
+  AEGISUB_CATALOG="aegisub"
 fi
 
 
