From 7479ffbd6d9cc9bf9786d1f1d9f0e6307198681f Mon Sep 17 00:00:00 2001
From: Martin Herkt <lachs0r@srsfckn.biz>
Date: Thu, 14 Feb 2019 14:52:27 +0100
Subject: [PATCH] DataBlockCache: Fix crash in cache invalidation

Fixes #33
---
 src/block_cache.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/block_cache.h b/src/block_cache.h
index 9fcaf42e7..f2ebfa261 100644
--- a/src/block_cache.h
+++ b/src/block_cache.h
@@ -150,8 +150,8 @@ class DataBlockCache {
 		}
 
 		// Remove old entries until we're under the max size
-		for (auto it = age.rbegin(); size > max_size && it != age.rend(); )
-			KillMacroBlock(**it++);
+		for (auto it = age.rbegin(); size > max_size && it != age.rend(); it++)
+			KillMacroBlock(**it);
 	}
 
 	/// @brief Obtain a data block from the cache
