diff -Naur Aegisub-3.2.2.orig/configure.ac Aegisub-3.2.2/configure.ac
--- Aegisub-3.2.2.orig/configure.ac	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/configure.ac	2019-01-31 12:43:50.321242739 +0100
@@ -50,7 +50,7 @@
 # On OS X everything goes in an app bundle so no point in versioning the executable
 AS_IF([test x$build_darwin = xyes],
       [AEGISUB_COMMAND="aegisub"],
-      [AEGISUB_COMMAND="aegisub-3.2"])
+      [AEGISUB_COMMAND="aegisub"])
 AC_SUBST(AEGISUB_COMMAND)
 AC_DEFINE_UNQUOTED([AEGISUB_COMMAND], ["${AEGISUB_COMMAND}"], [Name of the Aegisub executable])
 
@@ -58,7 +58,7 @@
 # On OS X everything goes in an app bundle so no point in versioning the executable
 AS_IF([test x$build_darwin = xyes],
       [AEGISUB_CATALOG="aegisub"],
-      [AEGISUB_CATALOG="aegisub-32"])
+      [AEGISUB_CATALOG="aegisub"])
 AC_SUBST(AEGISUB_CATALOG)
 AC_DEFINE_UNQUOTED([AEGISUB_CATALOG], ["${AEGISUB_CATALOG}"], [Name of the Aegisub gettext catalog])
 
