From 31c8d3bff4e789c9d8a2c9f406e12511f80bd0dc Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Sat, 7 Sep 2019 19:49:51 +0200
Subject: [PATCH] Fix compilation by adding pthread dependency

---
 Makefile.inc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.inc.in b/Makefile.inc.in
index a8c0e0c48..c61afc850 100644
--- a/Makefile.inc.in
+++ b/Makefile.inc.in
@@ -29,7 +29,7 @@ CXXFLAGS            = @CXXFLAGS@
 CPPFLAGS            = @CPPFLAGS@
 CFLAGS_DEP          = -MMD -MP
 LIBS                = @LIBS@
-LDFLAGS             = @LDFLAGS@
+LDFLAGS             = @LDFLAGS@ -pthread
 INSTALL_FLAGS       = -m 644
 LIB_SHARED_LINK     = -shared -Wl,-soname -Wl,$(LIB_SHARED_FULL)
 LIB_SHARED_LINK_OSX = -dynamiclib -Wl,-undefined -Wl,dynamic_lookup -compatibility_version $(LIB_VERSION) -current_version $(LIB_VERSION) -Wl,-single_module -mmacosx-version-min=10.8 -install_name ${CURDIR}/$(LIB_SHARED_FULL)
