From dd67db47cb2203e7a14058e52549721f6ff16a49 Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@archlinux.org>
Date: Tue, 9 May 2017 04:01:47 +0200
Subject: [PATCH] Fix compilation against icu 59.1 (#46)

---
 src/utils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/utils.cpp b/src/utils.cpp
index e1cde5e83..876c4c011 100644
--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -44,6 +44,7 @@
 #include <boost/filesystem/path.hpp>
 #include <map>
 #include <unicode/locid.h>
+#include <unicode/unistr.h>
 #include <wx/clipbrd.h>
 #include <wx/filedlg.h>
 #include <wx/stdpaths.h>
