From 7479ffbd6d9cc9bf9786d1f1d9f0e6307198681f Mon Sep 17 00:00:00 2001
From: Martin Herkt <lachs0r@srsfckn.biz>
Date: Thu, 14 Feb 2019 14:52:27 +0100
Subject: [PATCH] DataBlockCache: Fix crash in cache invalidation

Fixes #33
---
 src/block_cache.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: Aegisub-524c6114a82157b143567240884de3a6d030b091/src/block_cache.h
===================================================================
--- Aegisub-524c6114a82157b143567240884de3a6d030b091.orig/src/block_cache.h	2018-07-10 00:06:51.000000000 +0200
+++ Aegisub-524c6114a82157b143567240884de3a6d030b091/src/block_cache.h	2019-03-29 13:17:48.209456221 +0200
@@ -150,8 +150,8 @@ public:
 		}
 
 		// Remove old entries until we're under the max size
-		for (auto it = age.rbegin(); size > max_size && it != age.rend(); )
-			KillMacroBlock(**it++);
+		for (auto it = age.rbegin(); size > max_size && it != age.rend(); it++)
+			KillMacroBlock(**it);
 	}
 
 	/// @brief Obtain a data block from the cache
