From b1eb9e50863781400535fa6bed3d9ec9f5297409 Mon Sep 17 00:00:00 2001
From: Ilya Tumaykin <itumaykin@gmail.com>
Date: Wed, 9 Mar 2016 21:39:26 +0300
Subject: Add missing AM_GNU_GETTEXT_VERSION macro

Close #1914.

Bug: http://devel.aegisub.org/ticket/1914
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 1649efc..c46f48a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -505,6 +505,7 @@ AC_PCH_FLAG([-fpch-preprocess])
 # Internationalisation support
 ##############################
 AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION([0.18.1])

 ################
 # Update checker
