Index: Aegisub-524c6114a82157b143567240884de3a6d030b091/Makefile.inc.in
===================================================================
--- Aegisub-524c6114a82157b143567240884de3a6d030b091.orig/Makefile.inc.in	2018-07-10 00:06:51.000000000 +0200
+++ Aegisub-524c6114a82157b143567240884de3a6d030b091/Makefile.inc.in	2019-03-29 13:17:21.692386078 +0200
@@ -29,7 +29,7 @@ CXXFLAGS            = @CXXFLAGS@
 CPPFLAGS            = @CPPFLAGS@
 CFLAGS_DEP          = -MMD -MP
 LIBS                = @LIBS@
-LDFLAGS             = @LDFLAGS@
+LDFLAGS             = @LDFLAGS@ -pthread
 INSTALL_FLAGS       = -m 644
 LIB_SHARED_LINK     = -shared -Wl,-soname -Wl,$(LIB_SHARED_FULL)
 LIB_SHARED_LINK_OSX = -dynamiclib -Wl,-undefined -Wl,dynamic_lookup -compatibility_version $(LIB_VERSION) -current_version $(LIB_VERSION) -Wl,-single_module -mmacosx-version-min=10.8 -install_name ${CURDIR}/$(LIB_SHARED_FULL)
