diff --git a/src/colour_button.cpp b/src/colour_button.cpp
index c0b1c5d58..48a87856c 100644
--- a/src/colour_button.cpp
+++ b/src/colour_button.cpp
@@ -18,7 +18,7 @@
 
 #include "dialogs.h"
 
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 
 AGI_DEFINE_EVENT(EVT_COLOR, agi::Color);
 
diff --git a/src/subtitles_provider_libass.cpp b/src/subtitles_provider_libass.cpp
index efb5cb481..fb62e3334 100644
--- a/src/subtitles_provider_libass.cpp
+++ b/src/subtitles_provider_libass.cpp
@@ -46,7 +46,7 @@
 #include <libaegisub/util.h>
 
 #include <atomic>
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 #include <memory>
 #include <mutex>
 
diff --git a/src/video_frame.cpp b/src/video_frame.cpp
index 610005879..c51f929ac 100644
--- a/src/video_frame.cpp
+++ b/src/video_frame.cpp
@@ -16,7 +16,7 @@
 
 #include "video_frame.h"
 
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 #include <wx/image.h>
 
 namespace {
diff --git a/src/video_provider_dummy.cpp b/src/video_provider_dummy.cpp
index b68b206b7..eca5ac261 100644
--- a/src/video_provider_dummy.cpp
+++ b/src/video_provider_dummy.cpp
@@ -45,7 +45,7 @@
 #include <boost/algorithm/string/predicate.hpp>
 #include <boost/filesystem/path.hpp>
 #include <libaegisub/format.h>
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 
 DummyVideoProvider::DummyVideoProvider(double fps, int frames, int width, int height, agi::Color colour, bool pattern)
 : framecount(frames)
