Subject: Fix FTBFS on some architectures
Author: Sebastian Reichel <sre@debian.org>
Last-Update: 2015-08-09

Index: aegisub/src/Makefile
===================================================================
--- aegisub.orig/src/Makefile
+++ aegisub/src/Makefile
@@ -6,7 +6,8 @@ src_CPPFLAGS := -I$(d) -I.. -I$(d)includ
 	$(CFLAGS_PTHREAD) $(CFLAGS_FFTW3) $(CFLAGS_ICU)
 src_CXXFLAGS := $(CXXFLAGS_WX)
 src_LIBS := $(LIBS_GL) $(LIBS_PTHREAD) $(LIBS_WX) $(LIBS_FREETYPE) \
-	$(LIBS_LIBASS) $(LIBS_FONTCONFIG) $(LIBS_FFTW3)  $(LIBS_BOOST) $(LIBS_ICU)
+	$(LIBS_LIBASS) $(LIBS_FONTCONFIG) $(LIBS_FFTW3)  $(LIBS_BOOST) $(LIBS_ICU) \
+        -latomic
 src_PCH := $(d)agi_pre.h
 src_INSTALLNAME := $(AEGISUB_COMMAND)
 
