diff -Naur Aegisub.orig/libaegisub/unix/path.cpp Aegisub/libaegisub/unix/path.cpp
--- Aegisub.orig/libaegisub/unix/path.cpp	2019-09-01 16:33:24.703708712 +0200
+++ Aegisub/libaegisub/unix/path.cpp	2019-09-01 16:34:28.320369936 +0200
@@ -45,7 +45,7 @@
 	SetToken("?user", home/".aegisub");
 	SetToken("?local", home/".aegisub");
 	SetToken("?data", P_DATA);
-	SetToken("?dictionary", "/usr/share/hunspell");
+	SetToken("?dictionary", "/app/share/hunspell");
 #else
 	agi::fs::path app_support = agi::util::GetApplicationSupportDirectory();
 	SetToken("?user", app_support/"Aegisub");
