diff -Naur Aegisub-3.2.2.orig/automation/Makefile Aegisub-3.2.2/automation/Makefile
--- Aegisub-3.2.2.orig/automation/Makefile	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/automation/Makefile	2018-12-06 18:55:16.112674669 +0100
@@ -14,7 +14,7 @@
 
 aegisub-lua_OBJ := $(d)tests/aegisub.o $(TOP)lib/libaegisub.a $(TOP)lib/libluabins.a $(LIBS_LUA)
 aegisub-lua_CPPFLAGS := $(CPPFLAGS_BOOST) $(CFLAGS_LUA) -I$(TOP)libaegisub/include -I$(TOP)src $(CXXFLAGS_WX)
-aegisub-lua_LIBS := $(LIBS_WX) $(LIBS_BOOST) $(LIBS_ICU)
+aegisub-lua_LIBS := $(LIBS_WX) $(LIBS_BOOST) $(LIBS_ICU) $(LIBS_PTHREAD)
 
 PROGRAM += $(d)aegisub-lua
 
diff -Naur Aegisub-3.2.2.orig/libaegisub/Makefile Aegisub-3.2.2/libaegisub/Makefile
--- Aegisub-3.2.2.orig/libaegisub/Makefile	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/libaegisub/Makefile	2018-12-06 18:55:15.193669199 +0100
@@ -41,7 +41,7 @@
 endif
 
 aegisub_PCH := $(d)lagi_pre.h
-aegisub_CPPFLAGS := -I$(d)include -I$(TOP) -fPIC $(CPPFLAGS_BOOST) $(CFLAGS_LUA)
+aegisub_CPPFLAGS := -I$(d)include -I$(TOP) -fPIC $(CPPFLAGS_BOOST) $(CFLAGS_LUA) $(CFLAGS_PTHREAD)
 
 $(d)common/charset_conv.o_FLAGS := $(CFLAGS_ICONV)
 $(d)common/parser.o_FLAGS := -ftemplate-depth=256
diff -Naur Aegisub-3.2.2.orig/Makefile.inc.in Aegisub-3.2.2/Makefile.inc.in
--- Aegisub-3.2.2.orig/Makefile.inc.in	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/Makefile.inc.in	2018-12-06 19:03:16.885536029 +0100
@@ -99,7 +99,7 @@
 LIBS_LUA           = $(TOP)vendor/luajit/src/libluajit.a
 LIBS_OPENAL        = @OPENAL_LIBS@
 LIBS_PORTAUDIO     = @PORTAUDIO_LIBS@
-LIBS_PTHREAD       = @PTHREAD_LIBS@
+LIBS_PTHREAD       = @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 
 ifeq (yes, $(BUILD_DARWIN))
 LIBS_LUA           = $(TOP)vendor/luajit/src/libluajit-aegisub.so
diff -Naur Aegisub-3.2.2.orig/tests/Makefile Aegisub-3.2.2/tests/Makefile
--- Aegisub-3.2.2.orig/tests/Makefile	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/tests/Makefile	2018-12-06 18:55:13.951661808 +0100
@@ -7,7 +7,7 @@
 run_CPPFLAGS := -I$(TOP)libaegisub/include -I$(TOP) -I$(d)support \
 	-I$(GTEST_ROOT) -I$(GTEST_ROOT)/include $(CPPFLAGS_BOOST) $(CFLAGS_LUA)
 run_CXXFLAGS := -Wno-unused-value -Wno-sign-compare
-run_LIBS := $(LIBS_BOOST) $(LIBS_ICU) -pthread
+run_LIBS := $(LIBS_BOOST) $(LIBS_ICU) $(LIBS_PTHREAD)
 run_OBJ := \
 	$(subst .cpp,.o,$(wildcard $(d)tests/*.cpp)) \
 	$(d)support/main.o \
diff -Naur Aegisub-3.2.2.orig/tools/Makefile Aegisub-3.2.2/tools/Makefile
--- Aegisub-3.2.2.orig/tools/Makefile	2014-12-08 01:07:09.000000000 +0100
+++ Aegisub-3.2.2/tools/Makefile	2018-12-06 18:55:12.815655046 +0100
@@ -7,7 +7,7 @@
 endif
 
 repack-thes-dict_OBJ  := $(d)repack-thes-dict.o $(TOP)lib/libaegisub.a
-repack-thes-dict_LIBS := $(LIBS_BOOST) $(LIBS_ICU)
+repack-thes-dict_LIBS := $(LIBS_BOOST) $(LIBS_ICU) $(LIBS_PTHREAD)
 repack-thes-dict_CPPFLAGS := -I$(TOP) -I$(TOP)libaegisub/include $(CFLAGS_ICU)
 
 PROGRAM += $(d)repack-thes-dict
