diff --git a/configuration/akregator_config_plugins.cpp b/configuration/akregator_config_plugins.cpp
index 6ed7de96..8e3d9f9b 100644
--- a/configuration/akregator_config_plugins.cpp
+++ b/configuration/akregator_config_plugins.cpp
@@ -26,11 +26,17 @@
 #include <PimCommon/ConfigurePluginsWidget>
 #include <PimCommon/ConfigurePluginsListWidget>
 #include <QHBoxLayout>
-using namespace Akregator;
 
-K_PLUGIN_FACTORY(KCMAkregatorPluginsConfigFactory, registerPlugin<KCMAkregatorPluginsConfig>();
+namespace Akregator {
+
+K_PLUGIN_FACTORY(KCMAkregatorPluginsConfigFactory, registerPlugin<Akregator::KCMAkregatorPluginsConfig>();
                  )
 
+}
+
+
+using namespace Akregator;
+
 KCMAkregatorPluginsConfig::KCMAkregatorPluginsConfig(QWidget *parent, const QVariantList &args)
     : KCModule(parent, args)
 {
