diff --git a/source/build/src/config.cpp b/source/build/src/config.cpp
index f2530438f..bf9109a71 100644
--- a/source/build/src/config.cpp
+++ b/source/build/src/config.cpp
@@ -113,7 +113,7 @@ int32_t loadsetup(const char *fn)

     editorgridextent = 131072;

-    if ((fp = buildvfs_fopen_read(fn)) == NULL) return -1;
+    if ((fp = buildvfs_fopen_read(get_writable_filename(fn).c_str())) == NULL) return -1;

     if (readconfig(fp, "usecwd", val, VL) > 0) g_useCwd = (atoi_safe(val) != 0);
     if (readconfig(fp, "forcesetup", val, VL) > 0) forcesetup = (atoi_safe(val) != 0);
@@ -324,7 +324,7 @@ void writesettings(void) // save binds and aliases to <cfgname>_m32_settings.cfg
         Bsprintf(tempbuf, "m32_settings.cfg");
     else Bsprintf(tempbuf, "%s_m32_settings.cfg", Bstrtoken(ptr, ".", &dummy, 1));

-    fp = buildvfs_fopen_write(tempbuf);
+    fp = buildvfs_fopen_write(get_writable_filename(tempbuf).c_str());

     if (fp)
     {
@@ -355,7 +355,7 @@ int32_t writesetup(const char *fn)
     buildvfs_FILE fp;
     int32_t i,j,first=1;

-    fp = buildvfs_fopen_write(fn);
+    fp = buildvfs_fopen_write(get_writable_filename(fn).c_str());
     if (!fp) return -1;

     Bfprintf(fp,
diff --git a/source/duke3d/src/astub.cpp b/source/duke3d/src/astub.cpp
index 7188d3de3..a2d616537 100644
--- a/source/duke3d/src/astub.cpp
+++ b/source/duke3d/src/astub.cpp
@@ -10136,9 +10136,11 @@ int32_t ExtInit(void)
     {
         char *dummy = NULL;
         char *ptr = Xstrdup(setupfilename), *p = Bstrtoken(ptr, ".", &dummy, 1);
+        char m32_settings_p[BMAX_PATH];
+        sprintf(m32_settings_p, "/%s_m32_settings.cfg", p);
         if (!Bstrcmp(setupfilename, SETUPFILENAME))
-            Bsprintf(tempbuf, "m32_settings.cfg");
-        else Bsprintf(tempbuf,"%s_m32_settings.cfg",p);
+            Bsprintf(tempbuf, get_writable_filename(std::string("m32_settings.cfg")).c_str());
+        else Bsprintf(tempbuf, get_writable_filename(std::string(m32_settings_p)).c_str());
         OSD_Exec(tempbuf);
         Xfree(ptr);
     }
diff --git a/source/duke3d/src/config.cpp b/source/duke3d/src/config.cpp
index c7c3774cb..84bc314f0 100644
--- a/source/duke3d/src/config.cpp
+++ b/source/duke3d/src/config.cpp
@@ -703,31 +703,33 @@ int CONFIG_ReadSetup(void)

     ud.config.setupread = 1;
     pathsearchmode = 1;
+    std::string WRITABLE_SETUPFILENAME(get_writable_filename(std::string(SETUPFILENAME)));
+    std::string WRITABLE_BACKUP_SETUPFILENAME(get_writable_filename(std::string(BACKUP_SETUPFILENAME)));

     if (ud.config.scripthandle < 0)
     {
-        if (buildvfs_exists(g_setupFileName))  // JBF 20031211
-            ud.config.scripthandle = SCRIPT_Load(g_setupFileName);
+        if (buildvfs_exists(WRITABLE_SETUPFILENAME.c_str()))  // JBF 20031211
+            ud.config.scripthandle = SCRIPT_Load(WRITABLE_SETUPFILENAME.c_str());
 #if !defined(EDUKE32_TOUCH_DEVICES) && !defined(EDUKE32_STANDALONE)
-        else if (buildvfs_exists(SETUPFILENAME))
+        else if (buildvfs_exists(WRITABLE_SETUPFILENAME.c_str()))
         {
             int const i = wm_ynbox("Import Configuration Settings",
                                    "Configuration file %s not found. "
                                    "Import configuration data from %s?",
-                                   g_setupFileName, SETUPFILENAME);
+                                   g_setupFileName, WRITABLE_SETUPFILENAME.c_str());
             if (i)
-                ud.config.scripthandle = SCRIPT_Load(SETUPFILENAME);
+                ud.config.scripthandle = SCRIPT_Load(WRITABLE_SETUPFILENAME.c_str());
         }
 #endif
 #ifdef BACKUP_SETUPFILENAME
-        else if (buildvfs_exists(BACKUP_SETUPFILENAME))
+        else if (buildvfs_exists(WRITABLE_BACKUP_SETUPFILENAME.c_str()))
         {
             int const i = wm_ynbox("Import Configuration Settings",
                                    "Configuration file %s not found. "
                                    "Import configuration data from %s?",
-                                   g_setupFileName, BACKUP_SETUPFILENAME);
+                                   g_setupFileName, WRITABLE_BACKUP_SETUPFILENAME.c_str());
             if (i)
-                ud.config.scripthandle = SCRIPT_Load(BACKUP_SETUPFILENAME);
+                ud.config.scripthandle = SCRIPT_Load(WRITABLE_BACKUP_SETUPFILENAME.c_str());
         }
 #endif
     }
@@ -848,7 +850,7 @@ void CONFIG_ReadSettings(void)

     Xfree(setupFileName);

-    OSD_Exec(tempbuf);
+    OSD_Exec(get_writable_filename(tempbuf).c_str());

     ud.config.setupread = 2;

@@ -867,7 +869,7 @@ void CONFIG_WriteSettings(void) // save binds and aliases to <cfgname>_settings.
     else
         Bsprintf(filename, "%s_settings.cfg", Bstrtoken(g_setupFileName, ".", &dummy, 1));

-    buildvfs_FILE fp = buildvfs_fopen_write(filename);
+    buildvfs_FILE fp = buildvfs_fopen_write(get_writable_filename(filename).c_str());

     if (fp)
     {
diff --git a/source/duke3d/src/gamedef.cpp b/source/duke3d/src/gamedef.cpp
index 1d310707f..d172cbcac 100644
--- a/source/duke3d/src/gamedef.cpp
+++ b/source/duke3d/src/gamedef.cpp
@@ -2486,7 +2486,7 @@ static void C_SetCfgName(const char *cfgname)
     CONFIG_WriteSetup(1);

     if (g_modDir[0] != '/')
-        Bsnprintf(g_setupFileName, sizeof(g_setupFileName), "%s/%s", g_modDir, temp);
+        Bsnprintf(g_setupFileName, sizeof(g_setupFileName), get_writable_filename(temp).c_str());
     else
         Bstrncpyz(g_setupFileName, temp, sizeof(g_setupFileName));

diff --git a/source/mact/src/scriplib.cpp b/source/mact/src/scriplib.cpp
index 3d0281447..9886ae9fe 100644
--- a/source/mact/src/scriplib.cpp
+++ b/source/mact/src/scriplib.cpp
@@ -481,7 +481,7 @@ void SCRIPT_Save(int32_t scripthandle, char const * filename)
     if (!filename) return;
     if (!SC(scripthandle)) return;

-    fp = buildvfs_fopen_write_text(filename);
+    fp = buildvfs_fopen_write_text(get_writable_filename(filename).c_str());
     if (!fp) return;

     numsect = SCRIPT_NumberSections(scripthandle);
