diff --git a/source/duke3d/src/demo.cpp b/source/duke3d/src/demo.cpp
index 3169ab92b..bd19baacd 100644
--- a/source/duke3d/src/demo.cpp
+++ b/source/duke3d/src/demo.cpp
@@ -81,6 +81,7 @@ static int32_t G_OpenDemoRead(int32_t g_whichDemo) // 0 = mine
     int32_t i;
     savehead_t saveh;

+    char demodir[6] = "demos";
     char demofn[14];
     const char *demofnptr;

@@ -94,7 +95,7 @@ static int32_t G_OpenDemoRead(int32_t g_whichDemo) // 0 = mine
         demofnptr = demofn;
     }

-    g_demo_recFilePtr = kopen4loadfrommod(demofnptr, g_loadFromGroupOnly);
+    g_demo_recFilePtr = kopen4loadfrommod(get_writable_filename_subdir(demofnptr, demodir).c_str(), g_loadFromGroupOnly);
     if (g_demo_recFilePtr == buildvfs_kfd_invalid)
         return 0;

@@ -135,6 +137,7 @@ extern int32_t krd_print(const char *filename);

 void G_OpenDemoWrite(void)
 {
+    char demodir[6] = "demos";
     char demofn[BMAX_PATH];
     int32_t i, demonum=1;

@@ -174,15 +177,14 @@ void G_OpenDemoWrite(void)
         if (demonum == MAXDEMOS)
             return;

-        if (G_ModDirSnprintf(demofn, sizeof(demofn), DEMOFN_FMT, demonum))
-        {
-            LOG_F(ERROR, "Couldn't start demo writing: INTERNAL ERROR: file name too long");
-            goto error_wopen_demo;
-        }
+        // if (G_ModDirSnprintf(demofn, sizeof(demofn), DEMOFN_FMT, demonum))
+        // {
+        //     LOG_F(ERROR, "Couldn't start demo writing: INTERNAL ERROR: file name too long");
+        //     goto error_wopen_demo;
+        // }

         demonum++;
-
-        g_demo_filePtr = buildvfs_fopen_read(demofn);
+        g_demo_filePtr = buildvfs_fopen_read(get_writable_filename_subdir(demofn, demodir).c_str());
         if (g_demo_filePtr == NULL)
             break;

@@ -190,7 +192,8 @@ void G_OpenDemoWrite(void)
     }
     while (1);

-    g_demo_filePtr = buildvfs_fopen_write(demofn);
+    create_parent_directory(get_writable_filename_subdir(demofn, demodir));
+    g_demo_filePtr = buildvfs_fopen_write(get_writable_filename_subdir(demofn, demodir).c_str());
     if (g_demo_filePtr == NULL)
         return;
