diff --git a/source/duke3d/src/gameexec.cpp b/source/duke3d/src/gameexec.cpp
index c19ce1d2b..5a52ea3ec 100644
--- a/source/duke3d/src/gameexec.cpp
+++ b/source/duke3d/src/gameexec.cpp
@@ -5928,13 +5928,13 @@ badindex:
                     int const quoteDirname = *insptr++;

                     VM_ABORT_IF(bad_quote(quoteDirname), "invalid quote %d", quoteDirname);
-                    char temp[BMAX_PATH];
+                    char temp[BMAX_PATH] = { *get_writable_filename(apStrings[quoteDirname]).c_str() };

-                    if (EDUKE32_PREDICT_FALSE(G_ModDirSnprintf(temp, sizeof(temp), "%s", apStrings[quoteDirname])))
-                    {
-                        CON_ERRPRINTF("directory name in quote %d too long\n", quoteDirname);
-                        abort_after_error();
-                    }
+                    // if (EDUKE32_PREDICT_FALSE(G_ModDirSnprintf(temp, sizeof(temp), "%s", apStrings[quoteDirname])))
+                    // {
+                    //     CON_ERRPRINTF("directory name in quote %d too long\n", quoteDirname);
+                    //     abort_after_error();
+                    // }

                     if (!buildvfs_isdir(temp))
                     {
@@ -5999,7 +5999,7 @@ badindex:

                     VM_ABORT_IF(bad_quote(quoteFilename), "invalid quote %d", quoteFilename);

-                    buildvfs_kfd kFile = kopen4loadfrommod(apStrings[quoteFilename], 0);
+                    buildvfs_kfd kFile = kopen4loadfrommod(get_writable_filename(apStrings[quoteFilename]).c_str(), 0);

                     if (kFile == buildvfs_kfd_invalid)
                         dispatch();
@@ -6068,19 +6068,20 @@ badindex:

                     VM_ABORT_IF(bad_quote(quoteFilename), "invalid quote %d", quoteFilename);

-                    char temp[BMAX_PATH];
+                    std::string temp(get_writable_filename(apStrings[quoteFilename]));

-                    if (EDUKE32_PREDICT_FALSE(G_ModDirSnprintf(temp, sizeof(temp), "%s", apStrings[quoteFilename])))
-                    {
-                        LOG_F(ERROR, "Unable to save array to %s: resulting filename is too long.", apStrings[quoteFilename]);
-                        abort_after_error();
-                    }
+                    // if (EDUKE32_PREDICT_FALSE(G_ModDirSnprintf(temp, sizeof(temp), "%s", apStrings[quoteFilename])))
+                    // {
+                    //     LOG_F(ERROR, "Unable to save array to %s: resulting filename is too long.", apStrings[quoteFilename]);
+                    //     abort_after_error();
+                    // }

-                    buildvfs_FILE const fil = buildvfs_fopen_write(temp);
+                    create_parent_directory(temp);
+                    buildvfs_FILE const fil = buildvfs_fopen_write(temp.c_str());

                     if (EDUKE32_PREDICT_FALSE(fil == NULL))
                     {
-                        LOG_F(ERROR, "Unable to save array to %s: unknown error.", temp);
+                        LOG_F(ERROR, "Unable to save array to %s: unknown error.", temp.c_str());
                         abort_after_error();
                     }
