diff --git a/source/duke3d/src/menus.cpp b/source/duke3d/src/menus.cpp
index 4a558dd2b..f1da62831 100644
--- a/source/duke3d/src/menus.cpp
+++ b/source/duke3d/src/menus.cpp
@@ -37,6 +37,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #include "music.h"
 #include "sbar.h"
 #include "joystick.h"
+#include "vfs.h"

 #ifndef __ANDROID__
 droidinput_t droidinput;
@@ -1658,11 +1659,11 @@ static MenuTextForm_t M_CHEAT_SKILL = { NULL, "Enter Skill #:", 2, 0 };
 #define MAKE_MENUFILESELECT(a, dir, b, c, d) { a, { &MMF_FileSelectLeft, &MMF_FileSelectRight }, { &MF_Minifont, &MF_Minifont }, dir, b, c, d, { NULL, NULL }, { 0, 0 }, { 3<<16, 3<<16 }, FNLIST_INITIALIZER, 0 }

 static char lastuserdir[BMAX_PATH];
-static MenuFileSelect_t M_USERMAP = MAKE_MENUFILESELECT( "Select A User Map", "/usermaps/", "*.map", boardfilename, lastuserdir);
+static MenuFileSelect_t M_USERMAP = MAKE_MENUFILESELECT( "Select A User Map", redirect_to_user_data("usermaps/").c_str(), "*.map", boardfilename, lastuserdir);

 #ifndef EDUKE32_RETAIL_MENU
 static char lastsfdir[BMAX_PATH];
-static MenuFileSelect_t M_SOUND_SF2 = MAKE_MENUFILESELECT( "Select Sound Bank", "/", "*.sf2", sf2bankfile, lastsfdir);
+static MenuFileSelect_t M_SOUND_SF2 = MAKE_MENUFILESELECT( "Select Sound Bank", redirect_to_user_data("/").c_str(), "*.sf2", sf2bankfile, lastsfdir);
 #endif

 // MUST be in ascending order of MenuID enum values due to binary search
