diff --git a/source/build/src/screenshot.cpp b/source/build/src/screenshot.cpp
index 9003e94db..d3adf693c 100644
--- a/source/build/src/screenshot.cpp
+++ b/source/build/src/screenshot.cpp
@@ -97,8 +97,8 @@ static char* getScreenshotPath(const char* filename)

 int videoCaptureScreen(const char *filename, char inverseit)
 {
-    char* fn = getScreenshotPath(filename);
-    buildvfs_FILE fp = capturecounter.opennextfile_withext(fn, "png");
+    char* fn = const_cast<char*>(get_numbered_writable_filename(get_writable_filename_subdir(filename, "screenshots"), "png").c_str());
+    buildvfs_FILE fp = buildvfs_fopen_write(fn);

     if (fp == nullptr)
     {
@@ -177,9 +177,9 @@ int videoCaptureScreenTGA(const char *filename, char inverseit)
     int32_t i;
     char head[18] = { 0,1,1,0,0,0,1,24,0,0,0,0,0/*wlo*/,0/*whi*/,0/*hlo*/,0/*hhi*/,8,0 };
     //char palette[4*256];
-    char* fn = getScreenshotPath(filename);
+    char* fn = const_cast<char*>(get_numbered_writable_filename(get_writable_filename_subdir(filename, "screenshots"), "tga").c_str());

-    buildvfs_FILE fil = capturecounter.opennextfile_withext(fn, "tga");
+    buildvfs_FILE fil = buildvfs_fopen_write(fn);
     if (fil == nullptr)
     {
         Xfree(fn);
