diff --git a/source/duke3d/src/grpscan.cpp b/source/duke3d/src/grpscan.cpp
index 8b5133efe..3258c30ce 100644
--- a/source/duke3d/src/grpscan.cpp
+++ b/source/duke3d/src/grpscan.cpp
@@ -265,8 +265,7 @@ static void FreeGameList(void)
     }
 }

-
-#define GRPCACHEFILE "grpfiles.cache"
+#define GRPCACHEFILE get_writable_filename("grpfiles.cache").c_str()
 static struct grpcache
 {
     struct grpcache *next;
