diff --git a/source/build/src/build.cpp b/source/build/src/build.cpp
index 253bcac3c..a6e873c32 100644
--- a/source/build/src/build.cpp
+++ b/source/build/src/build.cpp
@@ -625,10 +625,8 @@ void editorMaybeLockMouse(int lock)

 int app_main(int argc, char const* const* argv)
 {
-    Bstrcpy(tempbuf, AppTechnicalName);
-    Bstrcat(tempbuf, ".log");
-
-    engineSetLogFile(tempbuf);
+    std::string logfile = get_writable_filename("amcsquad.log");
+    engineSetLogFile(logfile.c_str());

 #ifdef STARTUP_SETUP_WINDOW
     char cmdsetup = 0;
diff --git a/source/duke3d/src/game.cpp b/source/duke3d/src/game.cpp
index 29760de9c..5941bb76d 100644
--- a/source/duke3d/src/game.cpp
+++ b/source/duke3d/src/game.cpp
@@ -6768,7 +6768,8 @@ static const char* dukeVerbosityCallback(loguru::Verbosity verbosity)

 int app_main(int argc, char const* const* argv)
 {
-    engineSetLogFile(APPBASENAME ".log", LOG_GAME_MAX);
+    std::string logfile_name(get_writable_filename("amcsquad.log"));
+    engineSetLogFile(logfile_name.c_str(), LOG_GAME_MAX);
     engineSetLogVerbosityCallback(dukeVerbosityCallback);

 #ifndef NETCODE_DISABLE
