diff --git a/source/build/src/build.cpp b/source/build/src/build.cpp
index 75690e91e..6813790d6 100644
--- a/source/build/src/build.cpp
+++ b/source/build/src/build.cpp
@@ -634,10 +634,8 @@ void editorMaybeLockMouse(int lock)

 int app_main(int argc, char const* const* argv)
 {
-    Bstrcpy(tempbuf, AppTechnicalName);
-    Bstrcat(tempbuf, ".log");
-
-    engineSetLogFile(tempbuf);
+    std::string logfile = get_writable_filename("amcsquad.log");
+    engineSetLogFile(logfile.c_str());

 #ifdef STARTUP_SETUP_WINDOW
     char cmdsetup = 0;
diff --git a/source/duke3d/src/game.cpp b/source/duke3d/src/game.cpp
index f3221a23a..33a413dc8 100644
--- a/source/duke3d/src/game.cpp
+++ b/source/duke3d/src/game.cpp
@@ -6808,7 +6808,8 @@ int app_main(int argc, char const* const* argv)

     G_ExtPreInit(argc, argv);

-    engineSetLogFile(APPBASENAME ".log", LOG_GAME_MAX);
+    std::string logfile_name(get_writable_filename("amcsquad.log"));
+    engineSetLogFile(logfile_name.c_str(), LOG_GAME_MAX);
     engineSetLogVerbosityCallback(dukeVerbosityCallback);

 #ifdef __APPLE__
