diff --git a/source/build/include/smmalloc.h b/source/build/include/smmalloc.h
index 4fa90d268..802967edf 100644
--- a/source/build/include/smmalloc.h
+++ b/source/build/include/smmalloc.h
@@ -685,7 +685,9 @@ struct TlsPoolBucket
     }
 };

-static_assert(std::is_pod<TlsPoolBucket>::value == true, "TlsPoolBucket must be POD type, stored in TLS");
+// Remove excessive warning spam on builds so actual errors don't get buried
+static_assert(std::is_standard_layout<TlsPoolBucket>::value == true, "TlsPoolBucket must be POD type, stored in TLS");
+static_assert(std::is_trivial<TlsPoolBucket>::value == true, "TlsPoolBucket must be POD type, stored in TLS");
 static_assert(sizeof(TlsPoolBucket) <= 64, "TlsPoolBucket sizeof must be less than CPU cache line");
 } // namespace internal

diff --git a/Common.mak b/Common.mak
index 4212d2aad..cb554f9ac 100644
--- a/Common.mak
+++ b/Common.mak
@@ -536,7 +536,7 @@ else ifeq ($(SUBPLATFORM),LINUX)
 endif
 ASFLAGS += -f $(ASFORMAT)

-COMPILERFLAGS += -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
+COMPILERFLAGS += -D_FORTIFY_SOURCE=2


 ##### Optimizations
