diff --git a/platform/android/export/export.cpp b/platform/android/export/export.cpp
index e489bce3f..ed01ff928 100644
--- a/platform/android/export/export.cpp
+++ b/platform/android/export/export.cpp
@@ -1970,9 +1970,9 @@ void register_android_exporter() {
 		exe_ext = "*.exe";
 	}
 
-	EDITOR_DEF("export/android/adb", "");
+	EDITOR_DEF("export/android/adb", "/app/bin/adb");
 	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::STRING, "export/android/adb", PROPERTY_HINT_GLOBAL_FILE, exe_ext));
-	EDITOR_DEF("export/android/jarsigner", "");
+	EDITOR_DEF("export/android/jarsigner", "/app/bin/jarsigner");
 	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::STRING, "export/android/jarsigner", PROPERTY_HINT_GLOBAL_FILE, exe_ext));
 	EDITOR_DEF("export/android/debug_keystore", "");
 	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::STRING, "export/android/debug_keystore", PROPERTY_HINT_GLOBAL_FILE, "*.keystore"));
