diff --git a/cmd/geteduroam-gui/main.go b/cmd/geteduroam-gui/main.go
index 9e926ab..febc88a 100644
--- a/cmd/geteduroam-gui/main.go
+++ b/cmd/geteduroam-gui/main.go
@@ -449,7 +449,7 @@ func (ui *ui) activate() {
 }
 
 func (ui *ui) Run(args []string) int {
-	id := fmt.Sprintf("app.%s.Linux", variant.DisplayName)
+	id := fmt.Sprintf("app.%s.%s", variant.DisplayName, variant.DisplayName)
 	ui.app = adw.NewApplication(id, gio.GApplicationFlagsNoneValue)
 	defer ui.app.Unref()
 	actcb := func(_ gio.Application) {
diff --git a/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop b/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop
index 53c8768..ba3e41c 100644
--- a/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop
+++ b/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop
@@ -4,5 +4,5 @@ Encoding=UTF-8
 Name=geteduroam
 Comment=A geteduroam client for Linux. Geteduroam simplifies the process of connecting to eduroam.
 Exec=geteduroam-gui
-Icon=app.geteduroam.Linux
+Icon=app.geteduroam.geteduroam
 Terminal=false
\ No newline at end of file
