--- a/data/org.gnome.Sudoku.appdata.xml.in	2019-06-04 14:48:52.126188662 +0200
+++ b/data/org.gnome.Sudoku.appdata.xml.in	2019-06-04 14:47:34.211892440 +0200
@@ -46,6 +46,9 @@
     <kudo>ModernToolkit</kudo>
     <kudo>UserDocs</kudo>
   </kudos>
+  <releases>
+    <release version="3.32.0" date="2019-03-11" />
+  </releases>
   <translation type="gettext">gnome-sudoku</translation>
   <developer_name>The GNOME Project</developer_name>
   <content_rating type="oars-1.0">
