--- easytag-2.4.3/data/easytag.appdata.xml.in	2016-08-14 22:44:13.000000000 +1000
+++ easytag.appdata.xml.in	2020-11-16 20:16:56.623586046 +1100
@@ -28,6 +28,12 @@
         <kudo>AppMenu</kudo>
         <kudo>ModernToolkit</kudo>
     </kudos>
+
+    <content_rating type="oars-1.1"/>
+    <releases>
+        <release version="2.4.3" date="2016-12-05"/>
+    </releases>
+
     <update_contact>amigadave@amigadave.com</update_contact>
     <translation type="gettext">easytag</translation>
 </component>
