diff -u /var/home/deathwish/src/flatpak/org.gnome.FileRoller/a/org.gnome.FileRoller.appdata.xml.in /var/home/deathwish/src/flatpak/org.gnome.FileRoller/b/org.gnome.FileRoller.appdata.xml.in
--- /var/home/deathwish/src/flatpak/org.gnome.FileRoller/a/org.gnome.FileRoller.appdata.xml.in	2020-09-09 19:59:56.019173986 +0200
+++ /var/home/deathwish/src/flatpak/org.gnome.FileRoller/b/org.gnome.FileRoller.appdata.xml.in	2020-09-09 20:00:44.623674619 +0200
@@ -46,4 +46,10 @@
   <provides>
     <id>org.gnome.FileRoller.desktop</id>
   </provides>
+
+  <releases>
+    <release version="3.36.3" date="2020-08-08"/>
+  </releases>
+
+  <content_rating type="oars-1.1" />
 </component>
