--- a/data/org.gnome.Mahjongg.appdata.xml.in	2019-06-10 16:01:57.476292999 +0200
+++ b/data/org.gnome.Mahjongg.appdata.xml.in	2019-06-10 16:03:47.839849171 +0200
@@ -21,6 +21,9 @@
    time penalty.
   </p>
   </description>
+  <releases>
+    <release version="3.32.0" date="2019-03-11" />
+  </releases>
   <screenshots>
     <screenshot height="524" width="972" type="default">
       <image>https://people.gnome.org/~mcatanzaro/gnome-mahjongg.png</image>
@@ -40,4 +43,5 @@
     <kudo>UserDocs</kudo>
   </kudos>
   <translation type="gettext">gnome-mahjongg</translation>
+  <content_rating type="oars-1.1" />
 </component>
