--- a/data/org.gnome.Robots.appdata.xml.in	2019-09-27 13:26:38.182282937 +0200
+++ b/data/org.gnome.Robots.appdata.xml.in	2019-09-27 13:26:31.785413606 +0200
@@ -32,6 +32,9 @@
   <provides>
     <id>gnome-robots.desktop</id>
   </provides>
+  <releases>
+    <release version="3.34.0" date="2019-09-08" />
+  </releases>
   <launchable type="desktop-id">org.gnome.Robots.desktop</launchable>
   <url type="homepage">https://wiki.gnome.org/Apps/Robots</url>
   <url type="bugtracker">https://gitlab.gnome.org/GNOME/gnome-robots/issues</url>
