--- a/data/org.gnome.Epiphany.appdata.xml.in.in	2020-03-27 19:08:36.106873000 +0100
+++ b/data/org.gnome.Epiphany.appdata.xml.in.in	2020-05-08 22:15:41.215513649 +0200
@@ -43,6 +43,7 @@
   <provides>
     <id>epiphany.desktop</id>
   </provides>
+  <content_rating type="oars-1.1" />
   <releases>
     <release date="2020-03-27" version="3.36.1"/>
     <release date="2020-03-07" version="3.36.0"/>
