From 9d3df1e1ad7364cd0592fa0f1da8fe06d0677bd2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Sat, 14 May 2022 14:50:10 +0000
Subject: [PATCH] Add release history

---
 data/atomix.appdata.xml.in | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/data/atomix.appdata.xml.in b/data/atomix.appdata.xml.in
index 6303010..abafb36 100644
--- a/data/atomix.appdata.xml.in
+++ b/data/atomix.appdata.xml.in
@@ -55,6 +55,25 @@
     <content_attribute id="money-purchasing">none</content_attribute>
     <content_attribute id="money-gambling">none</content_attribute>
   </content_rating>
-
+  <releases>
+    <release version="3.34" date="2019-09-09" />
+    <release version="3.33" date="2019-09-02" />
+    <release version="3.32" date="2019-04-08" />
+    <release version="3.31" date="2019-03-05" />
+    <release version="3.30" date="2019-05-16" />
+    <release version="3.29" date="2018-06-18" />
+    <release version="3.22" date="2016-09-19" />
+    <release version="3.21" date="2016-09-12" />
+    <release version="3.20" date="2019-05-16" />
+    <release version="3.19" date="2019-05-16" />
+    <release version="3.18" date="2015-09-21" />
+    <release version="3.17" date="2015-09-14" />
+    <release version="3.16" date="2019-05-16" />
+    <release version="3.15" date="2015-03-16" />
+    <release version="2.14" date="2019-05-16" />
+    <release version="2.13" date="2009-07-14" />
+    <release version="1.2" date="2009-07-14" />
+    <release version="0.4" date="2019-05-16" />
+  </releases>
 </component>
 
-- 
GitLab

