diff --git a/data/reco.appdata.xml.in b/data/reco.appdata.xml.in
index 1fa1acf..28e228b 100644
--- a/data/reco.appdata.xml.in
+++ b/data/reco.appdata.xml.in
@@ -23,15 +23,15 @@
   <screenshots>
     <screenshot type="default">
       <caption>Welcome view; perform recording configuration</caption>
-      <image>https://raw.githubusercontent.com/ryonakano/reco/main/data/screenshots/pantheon/welcome-init.png</image>
+      <image>https://raw.githubusercontent.com/ryonakano/reco/main/data/screenshots/gnome/welcome-init.png</image>
     </screenshot>
     <screenshot>
       <caption>Recording view</caption>
-      <image>https://raw.githubusercontent.com/ryonakano/reco/main/data/screenshots/pantheon/recording.png</image>
+      <image>https://raw.githubusercontent.com/ryonakano/reco/main/data/screenshots/gnome/recording.png</image>
     </screenshot>
     <screenshot>
       <caption>An indication that recording is successfully finished</caption>
-      <image>https://raw.githubusercontent.com/ryonakano/reco/main/data/screenshots/pantheon/welcome-success.png</image>
+      <image>https://raw.githubusercontent.com/ryonakano/reco/main/data/screenshots/gnome/welcome-success.png</image>
     </screenshot>
   </screenshots>
 
