--- a/data/io.github.TransmissionRemoteGtk.appdata.xml.in	2021-09-29 15:00:22.991356086 -0400
+++ b/data/io.github.TransmissionRemoteGtk.appdata.xml.in	2021-09-29 15:38:01.713347394 -0400
@@ -38,7 +38,7 @@
   </description>
   <screenshots>
     <screenshot height="576" width="1024" type="default">
-      <image>http://eth0.org.uk/~alan/transmission-remote-gtk-1.1.1.png</image>
+      <image>https://github.com/hfiguiere/flathub/raw/transmission-remote/screenshots/transmission141.png</image>
     </screenshot>
   </screenshots>
   <releases>
@@ -65,6 +65,7 @@
       </description>
     </release>
   </releases>
+  <content_rating type="oars-1.1" />
   <launchable type="desktop-id">io.github.TransmissionRemoteGtk.desktop</launchable>
   <provides>
     <id>uk.org.eth0.transmission-remote-gtk.desktop</id>
