--- a/data/com.github.raibtoffoletto.litteris.appdata.xml.in	2020-03-22 10:03:06.000000000 +0000
+++ b/data/com.github.raibtoffoletto.litteris.appdata.xml.in	2020-07-22 14:53:26.480265055 +0100
@@ -27,16 +27,4 @@
   <url type="homepage">https://toffoletto.me</url>
   <url type="bugtracker">https://github.com/raibtoffoletto/litteris/issues</url>
   <url type="help">https://github.com/raibtoffoletto/litteris/issues</url>
-
-  <releases>
-    <release version="0.1.5" date="2020-03-22">
-​      <description>
-        <ul>
-          <li>Minor Bug Fixes</li>
-          <li>Escape Characters in penpal details</li>
-        </ul>
-​      </description>
-​    </release>
-​  </releases>
-
 </component>
