--- a/data/com.github.gijsgoudzwaard.image-optimizer.appdata.xml.in	2018-08-03 20:14:24.809950114 +0200
+++ b/data/com.github.gijsgoudzwaard.image-optimizer.appdata.xml.in	2018-08-03 20:34:31.081982293 +0200
@@ -1,72 +1,75 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2017 Gijs Goudzwaard <gijs.goudzwaard@gmail.com> -->
 <component type="desktop">
-  <id>com.github.gijsgoudzwaard.image-optimizer.desktop</id>
-  <metadata_license>CC0</metadata_license>
-  <name>Image Optimizer</name>
-​  <developer_name>Gijs Goudzwaard</developer_name>
-​  <project_license>MIT</project_license>
-  <summary>Simple lossless image compression</summary>
-  <description>
-    <p>Compress your images with ease using JpegOptim and OptiPng</p>
-    <p>Features:</p>
-    <ul>
-      <li>Easily quit the application by pressing Control + q</li>
-      <li>Compress <b>jpg / jpeg, png, bmp, pnm and tiff</b> with lossless compression</li>
-    </ul>
-  </description>
-  <url type="homepage">https://github.com/gijsgoudzwaard/image-optimizer</url>
-  <url type="bugtracker">https://github.com/gijsgoudzwaard/image-optimizer/issues</url>
-  <url type="help">https://github.com/gijsgoudzwaard/image-optimizer/issues</url>
-  <update_contact>gijs.goudzwaard@gmail.com</update_contact>
-  <screenshots>
-    <screenshot type="default">
-      <image>https://raw.githubusercontent.com/GijsGoudzwaard/Image-Optimizer/master/data/screenshots/welcome-screen.png</image>
-    </screenshot>
-    <screenshot>
-      <image>https://raw.githubusercontent.com/GijsGoudzwaard/Image-Optimizer/master/data/screenshots/treeview.png</image>
-    </screenshot>
-  </screenshots>
-  <custom>
-    <value key="x-appcenter-color-primary">#687ddb</value>
-    <value key="x-appcenter-color-primary-text">#ffffff</value>
-    <value key="x-appcenter-suggested-price">2</value>
-  </custom>
-  <releases>
+   <id>com.github.gijsgoudzwaard.image-optimizer.desktop</id>
+   <metadata_license>CC0</metadata_license>
+   <name>Image Optimizer</name>
+   <developer_name>Gijs Goudzwaard</developer_name>
+   <project_license>MIT</project_license>
+   <summary>Simple lossless image compression</summary>
+   <description>
+      <p>Compress your images with ease using JpegOptim and OptiPng</p>
+      <p>Features:</p>
+      <ul>
+         <li>Easily quit the application by pressing Control + q</li>
+         <li>
+            Compress jpg / jpeg, png, bmp, pnm and tiff with lossless compression
+         </li>
+      </ul>
+   </description>
+   <url type="homepage">https://github.com/gijsgoudzwaard/image-optimizer</url>
+   <url type="bugtracker">https://github.com/gijsgoudzwaard/image-optimizer/issues</url>
+   <url type="help">https://github.com/gijsgoudzwaard/image-optimizer/issues</url>
+   <update_contact>gijs.goudzwaard@gmail.com</update_contact>
+   <screenshots>
+      <screenshot type="default">
+         <image>https://raw.githubusercontent.com/GijsGoudzwaard/Image-Optimizer/master/data/screenshots/welcome-screen.png</image>
+      </screenshot>
+      <screenshot>
+         <image>https://raw.githubusercontent.com/GijsGoudzwaard/Image-Optimizer/master/data/screenshots/treeview.png</image>
+      </screenshot>
+   </screenshots>
+   <custom>
+      <value key="x-appcenter-color-primary">#687ddb</value>
+      <value key="x-appcenter-color-primary-text">#ffffff</value>
+      <value key="x-appcenter-suggested-price">2</value>
+   </custom>
+   <releases>
       <release version="0.1.9" date="2018-08-03">
-        <description>
-          <p>Getting things ready for Juno.</p>
-        </description>
+         <description>
+            <p>Getting things ready for Juno.</p>
+         </description>
       </release>
       <release version="0.1.8" date="2018-07-03">
-        <description>
-          <p>Getting things ready for Juno.</p>
-        </description>
+         <description>
+            <p>Getting things ready for Juno.</p>
+         </description>
       </release>
       <release version="0.1.7" date="2018-01-20">
-        <description>
-          <p>This release fixes the context menu in Files.</p>
-        </description>
+         <description>
+            <p>This release fixes the context menu in Files.</p>
+         </description>
       </release>
       <release version="0.1.6" date="2018-01-18">
-        <description>
-          <p>This release adds a context menu in Files for images that will allow easier image optimizations.</p>
-        </description>
+         <description>
+            <p>This release adds a context menu in Files for images that will allow easier image optimizations.</p>
+         </description>
       </release>
       <release version="0.1.4" date="2017-10-16">
-        <description>
-          <p>This release changes the upload button to fit the Elementary OS HIG better.</p>
-        </description>
+         <description>
+            <p>This release changes the upload button to fit the Elementary OS HIG better.</p>
+         </description>
       </release>
       <release version="0.1.3" date="2017-10-09">
-        <description>
-          <p>This release changes the upload icon to fit the Elementary OS HIG better.</p>
-        </description>
+         <description>
+            <p>This release changes the upload icon to fit the Elementary OS HIG better.</p>
+         </description>
       </release>
       <release version="0.1.2" date="2017-10-04">
-        <description>
-          <p>This release changes the icon to fit the Elementary OS HIG better.</p>
-        </description>
+         <description>
+            <p>This release changes the icon to fit the Elementary OS HIG better.</p>
+         </description>
       </release>
-    </releases>
+   </releases>
 </component>
+
