diff --git a/data/com.github.raibtoffoletto.litteris.appdata.xml.in b/data/com.github.raibtoffoletto.litteris.appdata.xml.in
index 15dcb67..5376f65 100644
--- a/data/com.github.raibtoffoletto.litteris.appdata.xml.in
+++ b/data/com.github.raibtoffoletto.litteris.appdata.xml.in
@@ -28,15 +28,4 @@
   <url type="bugtracker">https://github.com/raibtoffoletto/litteris/issues</url>
   <url type="help">https://raibtoffoletto.github.io/litteris</url>
 
-  <releases>
-    <release version="0.1.6" date="2020-07-22">
-      <description>
-        <ul>
-          <li>Version number fix</li>
-          <li>Add support for flatpak</li>
-        </ul>
-      </description>
-    </release>
-  </releases>
-
 </component>
