diff --git a/data/com.github.raibtoffoletto.litteris.appdata.xml.in b/data/com.github.raibtoffoletto.litteris.appdata.xml.in
index 15dcb67..bc4969a 100644
--- a/data/com.github.raibtoffoletto.litteris.appdata.xml.in
+++ b/data/com.github.raibtoffoletto.litteris.appdata.xml.in
@@ -28,6 +28,8 @@
   <url type="bugtracker">https://github.com/raibtoffoletto/litteris/issues</url>
   <url type="help">https://raibtoffoletto.github.io/litteris</url>
 
+  <content_rating type="oars-1.1" />
+
   <releases>
     <release version="0.1.6" date="2020-07-22">
       <description>
