--- a/data/org.gnome.Nibbles.appdata.xml.in	2019-09-27 10:19:13.711024888 +0200
+++ b/data/org.gnome.Nibbles.appdata.xml.in	2019-09-27 10:19:49.891296247 +0200
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2013 Michael Catanzaro -->
-<component type="desktop">
+<component type="desktop-application">
   <id>org.gnome.Nibbles.desktop</id>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-3.0+</project_license>
   <name>GNOME Nibbles</name>
-  <_summary>Guide a worm around a maze</_summary>
+  <summary>Guide a worm around a maze</summary>
   <description>
     <p>
       Control a worm in its quest to eat bonuses and become longer. Outmaneuver
@@ -29,19 +29,23 @@
       <image>https://people.gnome.org/~mcatanzaro/gnome-nibbles.png</image>
     </screenshot>
   </screenshots>
+  <releases>
+    <release version="3.34.0" date="2019-09-10" />
+  </releases>
+  <launchable type="desktop-id">org.gnome.Nibbles.desktop</launchable>
   <url type="homepage">https://wiki.gnome.org/Apps/Nibbles</url>
-  <url type="bugtracker">https://bugzilla.gnome.org/enter_bug.cgi?product=gnome-nibbles</url>
+  <url type="bugtracker">https://gitlab.gnome.org/GNOME/gnome-nibbles/issues</url>
   <url type="donation">http://www.gnome.org/friends/</url>
   <url type="help">https://help.gnome.org/users/gnome-nibbles/stable/</url>
   <url type="translate">https://wiki.gnome.org/TranslationProject</url>
   <update_contact>mcatanzaro_at_gnome.org</update_contact>
   <project_group>GNOME</project_group>
   <kudos>
-    <kudo>AppMenu</kudo>
     <kudo>HiDpiIcon</kudo>
     <kudo>ModernToolkit</kudo>
     <kudo>UserDocs</kudo>
   </kudos>
   <translation type="gettext">gnome-nibbles</translation>
   <developer_name>The GNOME Project</developer_name>
+  <content_rating type="oars-1.0" />
 </component>
