diff --git a/data/pinit.appdata.xml.in b/data/pinit.appdata.xml.in
index 5caa3cb..8aae511 100644
--- a/data/pinit.appdata.xml.in
+++ b/data/pinit.appdata.xml.in
@@ -18,15 +18,15 @@
   <screenshots>
     <screenshot type="default">
       <caption>Welcome View</caption>
-      <image>https://raw.githubusercontent.com/ryonakano/pinit/main/data/screenshots/pantheon/screenshot-welcome-view.png</image>
+      <image>https://raw.githubusercontent.com/ryonakano/pinit/main/data/screenshots/gnome/screenshot-welcome-view.png</image>
     </screenshot>
     <screenshot>
       <caption>Files View</caption>
-      <image>https://raw.githubusercontent.com/ryonakano/pinit/main/data/screenshots/pantheon/screenshot-files-view.png</image>
+      <image>https://raw.githubusercontent.com/ryonakano/pinit/main/data/screenshots/gnome/screenshot-files-view.png</image>
     </screenshot>
     <screenshot>
       <caption>Edit View</caption>
-      <image>https://raw.githubusercontent.com/ryonakano/pinit/main/data/screenshots/pantheon/screenshot-edit-view.png</image>
+      <image>https://raw.githubusercontent.com/ryonakano/pinit/main/data/screenshots/gnome/screenshot-edit-view.png</image>
     </screenshot>
   </screenshots>
 
