From 5e95f1a16824b5bb18af1028cebc3f55162aa1b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Thu, 8 Sep 2022 09:58:45 +0000
Subject: [PATCH] Add releases

---
 org.xfce.orage.appdata.xml.in | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/org.xfce.orage.appdata.xml.in b/org.xfce.orage.appdata.xml.in
index 279f5680..8e8d88ee 100644
--- a/org.xfce.orage.appdata.xml.in
+++ b/org.xfce.orage.appdata.xml.in
@@ -40,4 +40,30 @@
   <provides>
     <binary>orage</binary>
   </provides>
+
+  <releases>
+    <release version="4.16.0" date="2022-03-17" type="stable"/>
+    <release version="4.12.1" date="2015-04-10" type="stable"/>
+    <release version="4.12.0" date="2015-04-09" type="stable"/>
+    <release version="4.10.0" date="2013-12-09" type="stable"/>
+    <release version="4.8.4" date="2013-02-01" type="stable"/>
+    <release version="4.8.3" date="2012-01-05" type="stable"/>
+    <release version="4.8.2" date="2011-09-13" type="stable"/>
+    <release version="4.8.1" date="2011-02-26" type="stable"/>
+    <release version="4.8.0" date="2011-01-13" type="stable"/>
+    <release version="4.7.5" date="2010-03-04" type="stable"/>
+    <release version="4.6.1" date="2009-04-03" type="stable"/>
+    <release version="4.6.0" date="2009-02-25" type="stable"/>
+    <release version="4.4.3" date="2008-10-26" type="stable"/>
+    <release version="4.4.2" date="2007-11-17" type="stable"/>
+    <release version="4.4.1" date="2007-04-09" type="stable"/>
+    <release version="4.4.0" date="2007-01-20" type="stable"/>
+    <release version="4.2.2" date="2005-05-14" type="stable"/>
+    <release version="4.2.1" date="2005-03-14" type="stable"/>
+    <release version="4.0.6" date="2004-07-09" type="stable"/>
+    <release version="4.0.5" date="2004-03-24" type="stable"/>
+    <release version="4.0.3" date="2004-01-07" type="stable"/>
+    <release version="4.0.0" date="2003-09-06" type="stable"/>
+  </releases>
+
 </component>
-- 
GitLab

