--- a/data/com.github.phase1geo.minder.appdata.xml.in	2020-05-16 21:20:21.000000000 +0300
+++ b/data/com.github.phase1geo.minder.appdata.xml.in	2020-05-27 22:57:04.976203000 +0300
@@ -129,14 +129,14 @@
         <p>Changes</p>
         <ul>
           <li>Updated README.md to indicate which version of valac should be used if building from source.</li>
-          <li>Changed <em>unnamed</em> default filename to use translated value.</li>
+          <li>Changed unnamed default filename to use translated value.</li>
           <li>Updated French translations (thanks to Nathan Bonnemains).</li>
           <li>Changed tab colors to match current document background.</li>
           <li>Adjusted dark theme to help with current tab contrast.</li>
         </ul>
         <p>Bug Fixes</p>
         <ul>
-          <li>Fixed issue with <em>New Document</em> plank command to load application data before opening new tab.</li>
+          <li>Fixed issue with New Document plank command to load application data before opening new tab.</li>
           <li>Fixed build issues with compiling libarchive code using newer versions of libarchive and valac.</li>
           <li>Fixed issues with translating shortcut cheetsheat strings.</li>
           <li>Fixed issue with exporting images when manual layout mode is selected.</li>
@@ -154,7 +154,7 @@
           <li>Added ability to allow descendant nodes to have different link colors than ancestor nodes.</li>
           <li>Added ability to cause a descendant node with a different color than its parent to re-use the parent color.</li>
           <li>Added ability to select children nodes using Control-click.</li>
-          <li>Added ability to export a mind-map as a new <em>Portable Minder</em> filetype which will allow Minder files to be copied between different users/filesystems.</li>
+          <li>Added ability to export a mind-map as a new Portable Minder filetype which will allow Minder files to be copied between different users/filesystems.</li>
           <li>Added ability to export to Org-Mode format.</li>
         </ul>
         <p>Changes</p>
@@ -170,7 +170,7 @@
           <li>Fixed issue with adding a new child node that doesn't cause other descendant nodes to unfold.</li>
           <li>Fixed issue with showing/hiding sidebar.</li>
           <li>Fixed segmentation faults when loading a theme.</li>
-          <li>Fixed issue with the style inspector <em>Change Affect</em> value when selection changes (thanks to Viliam K.).</li>
+          <li>Fixed issue with the style inspector Change Affect value when selection changes (thanks to Viliam K.).</li>
           <li>Fixed issue with node markup within the node text (thanks to Viliam K).</li>
           <li>Fixed issue with tab state not being properly saved when a new tab is added.</li>
           <li>Fixed bug with updating tab tooltip and saving tab state when filename is changed.</li>
