--- a/data/org.gnome.Four-in-a-row.appdata.xml.in	2019-06-10 15:47:32.111325791 +0200
+++ b/data/org.gnome.Four-in-a-row.appdata.xml.in	2019-06-10 15:48:45.402814902 +0200
@@ -40,6 +40,9 @@
     <kudo>ModernToolkit</kudo>
     <kudo>UserDocs</kudo>
   </kudos>
+  <releases>
+    <release version="3.32.0" date="2019-03-11" />
+  </releases>
   <translation type="gettext">four-in-a-row</translation>
   <content_rating type="oars-1.0">
     <content_attribute id="violence-cartoon">none</content_attribute>
