--- a/gnumeric.appdata.xml.in-orig	2021-10-01 13:39:16.109593714 -0400
+++ b/gnumeric.appdata.xml.in	2020-01-31 11:21:39.000000000 -0500
@@ -28,4 +28,8 @@
   <!-- These guys need your donations more than we do: -->
   <url type="donation">http://www.amnestyusa.org/</url>
   <url type="help">https://help.gnome.org/users/gnumeric/stable/gnumeric.html</url>
+  <releases>
+    <release version="@VERSION@" date="2021-10-01" />
+  </releases>
+  <content_rating type="oars-1.1" />
 </application>
