diff --git a/internal/icons/pack-icons/main.go b/internal/icons/pack-icons/main.go
index 955a15b..7f195f0 100644
--- a/internal/icons/pack-icons/main.go
+++ b/internal/icons/pack-icons/main.go
@@ -19,7 +19,7 @@ import (
 )
 
 const rev = "3912bae"
-const prefix = "/xyz/diamondb/gtkcord4" // xyz.diamondb.gtkcord4
+const prefix = "/so/libdb/gtkcord4" // so.libdb.gtkcord4
 
 var (
 	output = "."
diff --git a/main.go b/main.go
index 231267d..1882920 100644
--- a/main.go
+++ b/main.go
@@ -43,7 +43,7 @@ var _ = cssutil.WriteCSS(`
 
 func main() {
 	m := manager{}
-	m.app = app.New(context.Background(), "xyz.diamondb.gtkcord4", "gtkcord4")
+	m.app = app.New(context.Background(), "so.libdb.gtkcord4", "gtkcord4")
 	m.app.AddJSONActions(map[string]interface{}{
 		"app.open-channel": m.openChannel,
 		"app.preferences":  func() { prefui.ShowDialog(m.win.Context()) },
diff --git a/nix/xyz.diamondb.gtkcord4.desktop b/nix/xyz.diamondb.gtkcord4.desktop
deleted file mode 100644
index e530138..0000000
--- a/nix/xyz.diamondb.gtkcord4.desktop
+++ /dev/null
@@ -1,13 +0,0 @@
-[Desktop Entry]
-Name=gtkcord4
-GenericName=Discord Chat
-Comment=A Discord client in Go and GTK4
-Exec=gtkcord4
-Icon=gtkcord4
-Terminal=false
-Type=Application
-Categories=GNOME;GTK;Network;Chat;
-StartupNotify=true
-DBusActivatable=true
-X-GNOME-UsesNotifications=true
-X-Purism-FormFactor=Workstation;Mobile;
