diff --git a/src/Application.vala b/src/Application.vala
index 18966c8..de45788 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -34,7 +34,7 @@ public class Spice.Application : Granite.Application {
     construct {
         flags |= ApplicationFlags.HANDLES_OPEN;
 
-        application_id = "com.github.philip-scott.spice-up";
+        application_id = "com.github.philip_scott.spice-up";
         program_name = PROGRAM_NAME;
         app_years = "2016-2017";
         exec_name = "spice-up";
