diff --git a/main.go b/main.go
index 231267d..1882920 100644
--- a/main.go
+++ b/main.go
@@ -43,7 +43,7 @@ var _ = cssutil.WriteCSS(`
 
 func main() {
 	m := manager{}
-	m.app = app.New(context.Background(), "xyz.diamondb.gtkcord4", "gtkcord4")
+	m.app = app.New(context.Background(), "so.libdb.gtkcord4", "gtkcord4")
 	m.app.AddJSONActions(map[string]interface{}{
 		"app.open-channel": m.openChannel,
 		"app.preferences":  func() { prefui.ShowDialog(m.win.Context()) },
diff --git a/nix/xyz.diamondb.gtkcord4.desktop b/nix/so.libdb.gtkcord4.desktop
similarity index 100%
rename from nix/xyz.diamondb.gtkcord4.desktop
rename to nix/so.libdb.gtkcord4.desktop
