diff -ruN para/.git/config para-open/.git/config
--- para/.git/config	2021-08-21 20:38:22.064046307 +0200
+++ para-open/.git/config	2021-08-21 20:37:11.467794045 +0200
@@ -4,5 +4,5 @@
 	bare = false
 	logallrefupdates = true
 [remote "origin"]
-	url = https://github.com/Kitware/ParaView
+	url = https://github.com/OpenChemistry/ParaView
 	fetch = +refs/tags/v5.9.0:refs/tags/v5.9.0
Binary files para/.git/index and para-open/.git/index differ
diff -ruN para/.git/logs/HEAD para-open/.git/logs/HEAD
--- para/.git/logs/HEAD	2021-08-21 20:38:28.376067337 +0200
+++ para-open/.git/logs/HEAD	2021-08-21 20:37:17.783817951 +0200
@@ -1 +1 @@
-0000000000000000000000000000000000000000 0aaf903ba3e8d012a608a466f798a041b4aebe17 Andy Great <andythe_great@pm.me> 1629571108 +0200	clone: from https://github.com/Kitware/ParaView
+0000000000000000000000000000000000000000 0aaf903ba3e8d012a608a466f798a041b4aebe17 Andy Great <andythe_great@pm.me> 1629571037 +0200	clone: from https://github.com/OpenChemistry/ParaView
