commit 955217261abc3e018cce25da569f79c375c90e32
Author: Camilo Higuita <chiguitar@unal.edu.co>
Date:   Fri May 14 12:13:37 2021 -0500

    install appstream metadata

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0f3011f..83eeab9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,6 +84,7 @@ add_subdirectory(src)
 
 if(UNIX AND NOT APPLE AND NOT ANDROID)
     install(FILES org.kde.nota.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
+    install(FILES org.kde.nota.metainfo.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
 endif()
 
 feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
