diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 33a8688..03ba6e9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -5,6 +5,8 @@ if(CEMU_CXX_FLAGS)
 	add_compile_options(${CEMU_CXX_FLAGS})
 endif()
 
+check_cxx_compiler_flag(-maes CXX_SUPPORTS_MAES)
+
 if(NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
 	message( FATAL_ERROR "Pointers are not 64bit" )
 endif()
@@ -32,7 +34,9 @@ elseif(UNIX)
 			add_compile_definitions(VK_USE_PLATFORM_WAYLAND_KHR)
 		endif()
 	endif()
-	add_compile_options(-maes)
+	if(CXX_SUPPORTS_MAES)
+		add_compile_options(-maes)
+	endif()
 	# warnings
 	if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
 		add_compile_options(-Wno-ambiguous-reversed-operator)
