diff --git a/configure.ac b/configure.ac
index ea89763..d87e9f8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -546,27 +546,9 @@ dnl *************************************************
 
 if test x$armamain = xtrue; then
 
-test -z "$XML2_CONFIG" && XML2_CONFIG=xml2-config
-
-if ! $XML2_CONFIG --libs > /dev/null; then
-  AC_MSG_ERROR([Fatal error: the libxml2 configuration script $XML2_CONFIG does not work/exist.])
-fi
-
-# if test x$armamain = xtrue; then
-AC_CHECK_LIB(xml2,xmlParseMemory,
-CPPFLAGS="$CPPFLAGS `$XML2_CONFIG --cflags`"
-LIBS="`$XML2_CONFIG --libs` $LIBS"
-AC_DEFINE(HAVE_LIBXML2, 1, [Define if you have the xml2 library (-lxml2).]),
-AC_MSG_ERROR([You need the library xml2 to compile Armagetron. Read the file doc/install_linux.html.])
-,`$XML2_CONFIG --libs`)
-
-AC_CHECK_FUNC(xmlParserInputBufferCreateFilenameDefault,
-,
-AC_MSG_ERROR([
-	Your xml2 library is obsolete!
-	Please upgrade to 2.6.11 or later
-])
-)
+PKG_CHECK_MODULES(LIBXML, [libxml-2.0 >= 2.6.11])
+CXXFLAGS="${CXXFLAGS} ${LIBXML_CFLAGS}"
+LIBS="${LIBS} ${LIBXML_LIBS}"
 # fi
 
 dnl *************************************************
