diff --git a/configure.ac b/configure.ac
index d87e9f8..34276b4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -705,17 +705,9 @@ dnl AC_MSG_WARN([C++ standard library not found. You can ignore this warning on
 AC_CHECK_LIB(z, main,,
 AC_MSG_ERROR([You need libz to compile Armagetron.]))
 
-AC_MSG_CHECKING(whether libpng is installed)
-if PNGLIBS=`libpng-config --libs`; then
-    AC_MSG_RESULT(yes)
-fi
-
-CPPFLAGS="$CPPFLAGS `libpng-config --cflags`"
-LIBS="$LIBS $PNGLIBS"
-
-AC_CHECK_FUNC(png_sig_cmp,
-,
-AC_MSG_ERROR([You need libpng to compile Armagetron.]))
+PKG_CHECK_MODULES(LIBPNG, [libpng])
+CXXFLAGS="${CXXFLAGS} ${LIBPNG_CFLAGS}"
+LIBS="${LIBS} ${LIBPNG_LIBS}"
 
 AC_CHECK_LIB(jpeg, main,,
 	AC_MSG_WARN([
