diff --git a/art4apps/__init__.py b/art4apps/__init__.py
index d71dcba..4923117 100644
--- a/art4apps/__init__.py
+++ b/art4apps/__init__.py
@@ -1,13 +1,13 @@
-#!/usr/local/bin/python
+  #!/usr/local/bin/python
 # coding: utf-8
 # module to manage art4apps resources
 
 import json
 import os
 
-DATA_PATH = '/usr/share/art4apps/data/'
-IMAGES_PATH = '/usr/share/art4apps/images/'
-AUDIO_PATH = '/usr/share/art4apps/audio/'
+DATA_PATH = '/app/share/art4apps/data/'
+IMAGES_PATH = '/app/share/art4apps/images/'
+AUDIO_PATH = '/app/share/art4apps/audio/'
 
 
 class Art4Apps:
