diff --git a/src/artisanlib/main.py b/src/artisanlib/main.py
index 1f5a0523..bbfb92e0 100644
--- a/src/artisanlib/main.py
+++ b/src/artisanlib/main.py
@@ -25941,8 +25941,8 @@ def initialize_locale(my_app:Artisan) -> str:
             trans_paths.append(QApplication.applicationDirPath() + '/../translations')
         else:
             trans_paths.append(QApplication.applicationDirPath() + '/translations')
-        # add the translations path for source installations
-        trans_paths.append('translations')
+        # add the translations path for Flatpak installation
+        trans_paths.append('/app/translations')
 
         #load Qt translations
         for qt_trans_module in qt_translation_modules:
