diff --git a/Makefile.am b/Makefile.am
index ea51bb2..69c9f33 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1 +1 @@
-SUBDIRS = cursor icons gtk gtk3
+SUBDIRS = cursor icons gtk3
diff --git a/configure.ac b/configure.ac
index 82c7654..4c6edd2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -26,18 +26,11 @@ if test -z "$EMPY"; then
     AC_MSG_ERROR([empy is required])
 fi

-PKG_CHECK_MODULES(GTK2, gtk+-2.0 >= 2.16.0,,
-	          AC_MSG_ERROR([GTK+-2.0 is required to compile sugar-artwork]))
-
 PKG_CHECK_MODULES(GTK3, gtk+-3.0 >= 3.0.0,,
 	          AC_MSG_ERROR([GTK+-3.0 is required to compile sugar-artwork]))

-PKG_CHECK_MODULES(ENGINE, gtk+-2.0 >= 2.16 gobject-2.0 >= 2.0 cairo >= 0.1.1)
 PKG_CHECK_MODULES(ENGINE3, gtk+-3.0 >= 3.0 gobject-2.0 >= 2.0 cairo >= 0.1.1)

-GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
-AC_SUBST(GTK_VERSION)
-
 GTK3_VERSION=`$PKG_CONFIG --modversion gtk+-3.0`
 AC_SUBST(GTK3_VERSION)

@@ -72,9 +65,6 @@ icons/scalable/device/Makefile
 icons/scalable/emblems/Makefile
 icons/scalable/mimetypes/Makefile
 icons/scalable/status/Makefile
-gtk/Makefile
-gtk/engine/Makefile
-gtk/theme/Makefile
 gtk3/Makefile
 gtk3/theme/Makefile
 gtk3/theme/3.20/Makefile
