--- a/asarUtil.js
+++ b/asarUtil.js
@@ -271,11 +271,6 @@
         if (await this.checkAgainstRoots(file, await ALLOWLIST)) {
             return;
         }
-        // Check denylist
-        if (await this.checkAgainstRoots(file, await DENYLIST)) {
-            builder_util_1.log.error(logFields, `denied access to system or unsafe path`);
-            throw new Error(`Cannot copy file [${file}] symlinked to file [${resolved}] outside the package to a system or unsafe path`);
-        }
         // Default: outside explicit paths but not explicitly denied
         builder_util_1.log.debug(logFields, `path is outside of explicit safe paths, defaulting to safe`);
     }
