diff -up ./resources/packaging/linux/com.governikus.ausweisapp2.metainfo.xml.in.no-mailto ./resources/packaging/linux/com.governikus.ausweisapp2.metainfo.xml.in
--- ./resources/packaging/linux/com.governikus.ausweisapp2.metainfo.xml.in.no-mailto	2020-01-15 18:05:33.758355549 +0100
+++ ./resources/packaging/linux/com.governikus.ausweisapp2.metainfo.xml.in	2020-01-15 18:05:41.611456198 +0100
@@ -17,7 +17,6 @@
     <category>Utility</category>
   </categories>
   <url type="homepage">https://www.ausweisapp.bund.de/</url>
-  <url type="contact">mailto:support@ausweisapp.de</url>
   <launchable type="desktop-id">com.governikus.ausweisapp2.desktop</launchable>
   <provides>
     <binary>AusweisApp2</binary>
