diff -ru avidemux_2.7.0.orig/avidemux/qt4/ADM_jobs/src/ADM_jobControl.cpp avidemux_2.7.0/avidemux/qt4/ADM_jobs/src/ADM_jobControl.cpp
--- avidemux_2.7.0.orig/avidemux/qt4/ADM_jobs/src/ADM_jobControl.cpp	2018-05-25 22:43:37.453349626 +0200
+++ avidemux_2.7.0/avidemux/qt4/ADM_jobs/src/ADM_jobControl.cpp	2018-05-25 22:45:04.903345751 +0200
@@ -17,6 +17,7 @@
 #include "ADM_coreJobs.h"
 #include "DIA_coreToolkit.h"
 #include <QtCore/QDir>
+#include <QAction>
 
 
 static QTableWidgetItem *fromText(const string &t,int id)
diff -ru avidemux_2.7.0.orig/avidemux_plugins/ADM_videoFilters6/eq2/qt4/Q_eq2.cpp avidemux_2.7.0/avidemux_plugins/ADM_videoFilters6/eq2/qt4/Q_eq2.cpp
--- avidemux_2.7.0.orig/avidemux_plugins/ADM_videoFilters6/eq2/qt4/Q_eq2.cpp    2018-05-25 22:43:37.413349627 +0200
+++ avidemux_2.7.0/avidemux_plugins/ADM_videoFilters6/eq2/qt4/Q_eq2.cpp 2018-05-25 23:01:45.793301398 +0200
@@ -17,6 +17,7 @@
 #include "Q_eq2.h"
 #include "ADM_toolkitQt.h"
 #include <QSignalMapper>
+#include <QAction>
 
 //
 //     Video is in YV12 Colorspace

