diff --git i/Ryujinx.Common/Logging/Targets/FileLogTarget.cs w/Ryujinx.Common/Logging/Targets/FileLogTarget.cs
index 46fffd3a..f119166b 100644
--- i/Ryujinx.Common/Logging/Targets/FileLogTarget.cs
+++ w/Ryujinx.Common/Logging/Targets/FileLogTarget.cs
@@ -22,7 +22,7 @@ namespace Ryujinx.Common.Logging
         public FileLogTarget(string path, string name, FileShare fileShare, FileMode fileMode)
         {
             // Ensure directory is present
-            DirectoryInfo logDir = new DirectoryInfo(Path.Combine(path, "Logs"));
+            DirectoryInfo logDir = new DirectoryInfo(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Logs"));
             logDir.Create();
 
             // Clean up old logs, should only keep 3
diff --git i/Ryujinx/Ui/SettingsWindow.cs w/Ryujinx/Ui/SettingsWindow.cs
index 68634b30..bae66bdd 100644
--- i/Ryujinx/Ui/SettingsWindow.cs
+++ w/Ryujinx/Ui/SettingsWindow.cs
@@ -480,7 +480,7 @@ namespace Ryujinx.Ui
 
         private void OpenLogsFolder_Pressed(object sender, EventArgs args)
         {
-            string logPath = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Logs");
+            string logPath = System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Logs");
             
             DirectoryInfo directory = new DirectoryInfo(logPath);
             directory.Create();
