From b069d193b3e552b0dc27077c54cd2170e7c7d9ba Mon Sep 17 00:00:00 2001
From: danielknobe <54905085+danielknobe@users.noreply.github.com>
Date: Sat, 19 Apr 2014 16:12:58 +0000
Subject: [PATCH] - added Icon.bmp to Linux install script

---
 data/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index d9029b36..d59bdf6d 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -24,6 +24,7 @@ set(install_files
 	${CMAKE_CURRENT_BINARY_DIR}/scripts.zip
 	${CMAKE_CURRENT_BINARY_DIR}/backgrounds.zip
 	${CMAKE_CURRENT_BINARY_DIR}/rules.zip
+	Icon.bmp
 	config.xml
 	inputconfig.xml
 	server/server.xml
