diff --git a/profex/projectWidget/bgmnbackendconfig.cpp b/profex/projectWidget/bgmnbackendconfig.cpp
index 0b1e1ee..6477e95 100644
--- a/profex/projectWidget/bgmnbackendconfig.cpp
+++ b/profex/projectWidget/bgmnbackendconfig.cpp
@@ -229,7 +229,7 @@ int BgmnBackendConfig::guessExecutablesUnix()
             << QDir(QString("%1/BGMN").arg(QDir::homePath()))
             << QDir(QString("%1/BGMNwin").arg(QDir::homePath()))
             << QDir("/opt/bgmnwin")
-            << QDir("/opt/bgmn")
+            << QDir("/app/bgmn")
             << QDir("/opt/bgmn-4.2.23")
             << QDir("/opt/bgmn-4.2.22")
             << QDir("/opt/BGMNwin")
@@ -398,7 +398,7 @@ int BgmnBackendConfig::guessRepositoriesUnix()
 
     QVector<QDir> defaultRepoLocations;
     defaultRepoLocations << QDir("/opt/Profex-BGMN/BGMN-Templates")
-                         << QDir("/opt/BGMN-Templates")
+                         << QDir("/app/BGMN-Templates")
                          << QDir(QString("%1/Documents/BGMN-Templates").arg(QDir::homePath()))
                          << createUserRepos();
 
