From 9e08543926e68ac776f1aad461954936afb37ed0 Mon Sep 17 00:00:00 2001
From: Stan Janssen <stan@finetuned.nl>
Date: Wed, 15 Dec 2021 11:14:46 +0100
Subject: [PATCH] Add StartupWMClass to .desktop

---
 blender.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/blender/blender.desktop b/blender/blender.desktop
index 792f98b..b9d74d9 100644
--- a/blender/blender.desktop
+++ b/blender/blender.desktop
@@ -87,3 +87,4 @@ PrefersNonDefaultGPU=true
 X-KDE-RunOnDiscreteGpu=true
 Categories=Graphics;3DGraphics;
 MimeType=application/x-blender;
+StartupWMClass=Blender
-- 
2.25.1

