From 5d9a1da6931b57247c4c03346fd1c7ab08fab00c Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Mon, 20 Feb 2017 10:26:45 -0800
Subject: [PATCH] Fix link/include paths

---
 BlockOut/Makefile | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/BlockOut/Makefile b/BlockOut/Makefile
index 0cdfa00..711803a 100644
--- a/BlockOut/Makefile
+++ b/BlockOut/Makefile
@@ -41,18 +41,14 @@ OBJ=    BlockOut.o         PageChooseSetup.o       PageStartGame.o \
 
 ADD_LIBS = ../ImageLib/src/libimagelib.a
 
-ifdef _linux64
-SDL_LIBS = /usr/lib/x86_64-linux-gnu
-else
-SDL_LIBS = /usr/lib/i386-linux-gnu
-endif
+SDL_LIBS = /app/lib
 
 CXX		= g++
 
 ifdef _release
 
 CXXFLAGS =     -O2 -Dlinux -c \
-               -I/usr/include/SDL -I../ImageLib/src -I.
+               -I/app/include/SDL -I../ImageLib/src -I.
 
 LFLAGS   =      -Wl,--strip-all \
 		$(ADD_LIBS) \
@@ -62,7 +58,7 @@ LFLAGS   =      -Wl,--strip-all \
 else
 
 CXXFLAGS =     -g -D_DEBUG -Dlinux -c \
-               -I/usr/include/SDL -I../ImageLib/src -I.
+               -I/app/include/SDL -I../ImageLib/src -I.
 
 LFLAGS   =      -L $(SDL_LIBS) \
 		$(ADD_LIBS) \
-- 
2.9.3

