From 896e3dc83c4cc268041f90a471b368fcad5fdeb6 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Mon, 20 Feb 2017 11:41:36 -0800
Subject: [PATCH] Don't forget linker path

---
 BlockOut/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/BlockOut/Makefile b/BlockOut/Makefile
index 711803a..7cd1a3b 100644
--- a/BlockOut/Makefile
+++ b/BlockOut/Makefile
@@ -51,6 +51,7 @@ CXXFLAGS =     -O2 -Dlinux -c \
                -I/app/include/SDL -I../ImageLib/src -I.
 
 LFLAGS   =      -Wl,--strip-all \
+		-L $(SDL_LIBS) \
 		$(ADD_LIBS) \
 		-lSDL_mixer -lSDL -lasound \
 		-lGL -lGLU -lpthread -lm -ldl
-- 
2.9.3

