From f8ea15cd2eea3c232b2ea67f3343e4e455536177 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Mon, 20 Feb 2017 11:00:29 -0800
Subject: [PATCH] Fix data path

---
 BlockOut/Utils.cpp | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/BlockOut/Utils.cpp b/BlockOut/Utils.cpp
index 4ccd8df..ea8747b 100644
--- a/BlockOut/Utils.cpp
+++ b/BlockOut/Utils.cpp
@@ -207,12 +207,7 @@ BOOL CheckEnv() {
     return FALSE;
   }
     
-  char *blockoutHome = getenv("BL2_HOME");
-  if( blockoutHome==NULL ) {
-    printf("BL2_HOME environement variable if not defined !\n");
-    printf("Please set the BL2_HOME to the BlockOut II installation directory (ex: BL2_HOME=/usr/local/bl2).\n");
-    return FALSE;
-  }
+  char *blockoutHome = "/app/share/BlockOut";
   strcpy( bl2Home , blockoutHome );
     
   char bl2Dir[512];
-- 
2.9.3

