From 9a49d7bacde302b21889fbea84f7bb2caa4e893f Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jozef.mlich@greycortex.com>
Date: Sat, 20 Apr 2024 11:28:57 +0200
Subject: [PATCH] /run not into read-only fs

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3153bf5..ca3c8b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,7 +108,7 @@ install(FILES
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     set(INSTALL_UDEV_RULE ON CACHE BOOL "Whether to install udev rule")
-    set(UDEV_RULES_INSTALL_DIR /lib/udev/rules.d CACHE PATH "The location of udev rules")
+    set(UDEV_RULES_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/lib/udev/rules.d CACHE PATH "The location of udev rules")
 
     if(${INSTALL_UDEV_RULE})
         install(FILES 61-kde-bluetooth-rfkill.rules DESTINATION ${UDEV_RULES_INSTALL_DIR})
-- 
2.44.0

