diff -Naur a/boost_1_68_0/boost/spirit/home/karma/numeric/detail/numeric_utils.hpp b/boost_1_68_0/boost/spirit/home/karma/numeric/detail/numeric_utils.hpp
--- a/boost_1_68_0/boost/spirit/home/karma/numeric/detail/numeric_utils.hpp	2018-10-31 09:14:52.429280271 +0100
+++ b/boost_1_68_0/boost/spirit/home/karma/numeric/detail/numeric_utils.hpp	2018-10-31 09:28:44.874929790 +0100
@@ -71,8 +71,13 @@
                 /* implementation is well-defined for one's complement, */    \
                 /* two's complement, and signed magnitude architectures */    \
                 /* by the C++ Standard. [conv.integral] [expr.unary.op] */    \
-                return (n >= 0) ?  static_cast<type>(n)                       \
-                                : -static_cast<type>(n);                      \
+                /* return (n >= 0) ?  static_cast<type>(n)                 */ \
+                /*                 : -static_cast<type>(n);                */ \
+                if (n >= 0)                                                   \
+                    return static_cast<type>(n);                              \
+                if (n == std::numeric_limits<signedtype>::min())              \
+                    return (unsignedtype)n;                                   \
+                return (unsignedtype)(-n);                                    \
             }                                                                 \
         }                                                                     \
     /**/
