--- boost_1_66_0/tools/build/src/tools/gcc.jam~ 2018-01-19 13:09:56.041685502 +0000
+++ boost_1_66_0/tools/build/src/tools/gcc.jam  2018-01-19 13:09:56.042685500 +0000
@@ -421,7 +421,9 @@
 
 rule set-address-model-options ( targets * : sources * : properties * )
 {
-    local model = [ feature.get-values address-model : $(properties) ] ;
+    # For RPM builds the address model flag is passed in %{optflags}.
+    # local model = [ feature.get-values address-model : $(properties) ] ;
+    local model ;
     if $(model)
     {
         local option ;
