diff -Naur boost_1_67_0/libs/units/example/autoprefixes.cpp boost_1_67_0_patch4/libs/units/example/autoprefixes.cpp
--- boost_1_67_0/libs/units/example/autoprefixes.cpp	2018-06-30 03:15:27.680736712 +0200
+++ boost_1_67_0_patch4/libs/units/example/autoprefixes.cpp	2018-06-30 03:40:08.348541062 +0200
@@ -67,7 +67,7 @@
 struct euro_base_unit : boost::units::base_unit<euro_base_unit, boost::units::dimensionless_type, 5>
 {
   static constexpr const char* name() { return("EUR"); }
-  static constexpr const char* symbol() { return("€"); }
+  static constexpr const char* symbol() { return("â‚¬"); }
 };
 
 int main()
@@ -140,7 +140,7 @@
 
   quantity<euro_base_unit::unit_type> ce = 2048. * euro_base_unit::unit_type();
   cout << name_format << engineering_prefix << ce << endl;  // 2.048 kiloEUR
-  cout << symbol_format << engineering_prefix << ce << endl;  // 2.048 k€
+  cout << symbol_format << engineering_prefix << ce << endl;  // 2.048 kâ‚¬
 
 
     return 0;
