diff -up btanks-0.9.8083/engine/luaxx/state.cpp.lua-52 btanks-0.9.8083/engine/luaxx/state.cpp
--- btanks-0.9.8083/engine/luaxx/state.cpp.lua-52	2008-06-08 05:02:39.000000000 -0400
+++ btanks-0.9.8083/engine/luaxx/state.cpp	2013-05-13 10:43:52.493334782 -0400
@@ -5,6 +5,10 @@
 #include <assert.h>
 #include "mrt/chunk.h"
 
+#ifndef lua_open
+#define lua_open()     luaL_newstate()
+#endif
+
 using namespace luaxx;
 /*
 static void *l_alloc (void *ud, void *ptr, size_t osize, size_t nsize) {
@@ -59,7 +63,7 @@ static const char * chunk_reader(lua_Sta
 void State::load(const std::string &fname, const mrt::Chunk &data) {
 	//throw_ex(("implement me[%s]", fname.c_str()));
 	reader_state x(data);
-	int err = lua_load(state, chunk_reader, &x, fname.c_str());
+	int err = lua_load(state, chunk_reader, &x, fname.c_str(), NULL);
 	check_error(state, err);
 }
 
