--- engine/sl08/sl08.py.gcc	2008-08-13 04:10:45.000000000 -0500
+++ engine/sl08/sl08.py	2011-06-26 11:36:40.000000000 -0500
@@ -152,7 +152,7 @@
 			typedef base_signalXXX %s signal_type; 
 			typedef return_type (object_type::*func_t) %s; 
 
-			inline slotXXX () : object(NULL), func(NULL) {}
+			inline slotXXX () : object(((object_type*)NULL)), func(0) {}
 			inline slotXXX(object_type *object, func_t func, signal_type * signal = NULL) : object(object), func(func) {}
 
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
@@ -187,7 +187,7 @@
 			typedef base_signalXXX %s signal_type; 
 			typedef void (object_type::*func_t) %s ;
 
-			inline slotXXX () : object(NULL), func(NULL) {}
+			inline slotXXX () : object(((object_type*)NULL), func(0) {}
 			inline slotXXX (object_type *object, func_t func) : object(object), func(func) {}
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
 			inline void assign(object_type *o, func_t f, signal_type &signal_ref) { object = o; func = f; connect(signal_ref); }
--- engine/sl08/sl08.h.gcc	2008-08-13 04:10:45.000000000 -0500
+++ engine/sl08/sl08.h	2011-06-26 11:35:21.000000000 -0500
@@ -92,7 +92,7 @@
 			typedef base_signal0 <return_type> signal_type; 
 			typedef return_type (object_type::*func_t) (); 
 
-			inline slot0 () : object(NULL), func(NULL) {}
+			inline slot0 () : object(((object_type*)NULL)), func(0) {}
 			inline slot0(object_type *object, func_t func, signal_type * signal = NULL) : object(object), func(func) {}
 
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
@@ -114,7 +114,7 @@
 			typedef base_signal0 <void> signal_type; 
 			typedef void (object_type::*func_t) () ;
 
-			inline slot0 () : object(NULL), func(NULL) {}
+			inline slot0 () : object(((object_type*)NULL)), func(0) {}
 			inline slot0 (object_type *object, func_t func) : object(object), func(func) {}
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
 			inline void assign(object_type *o, func_t f, signal_type &signal_ref) { object = o; func = f; connect(signal_ref); }
@@ -267,7 +267,7 @@
 			typedef base_signal1 <return_type, arg1_type> signal_type; 
 			typedef return_type (object_type::*func_t) (arg1_type a1); 
 
-			inline slot1 () : object(NULL), func(NULL) {}
+			inline slot1 () : object(((object_type*)NULL)), func(0) {}
 			inline slot1(object_type *object, func_t func, signal_type * signal = NULL) : object(object), func(func) {}
 
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
@@ -289,7 +289,7 @@
 			typedef base_signal1 <void, arg1_type> signal_type; 
 			typedef void (object_type::*func_t) (arg1_type a1) ;
 
-			inline slot1 () : object(NULL), func(NULL) {}
+			inline slot1 () : object(((object_type*)NULL)), func(0) {}
 			inline slot1 (object_type *object, func_t func) : object(object), func(func) {}
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
 			inline void assign(object_type *o, func_t f, signal_type &signal_ref) { object = o; func = f; connect(signal_ref); }
@@ -442,7 +442,7 @@
 			typedef base_signal2 <return_type, arg1_type, arg2_type> signal_type; 
 			typedef return_type (object_type::*func_t) (arg1_type a1, arg2_type a2); 
 
-			inline slot2 () : object(NULL), func(NULL) {}
+			inline slot2 () : object(((object_type*)NULL)), func(0) {}
 			inline slot2(object_type *object, func_t func, signal_type * signal = NULL) : object(object), func(func) {}
 
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
@@ -464,7 +464,7 @@
 			typedef base_signal2 <void, arg1_type, arg2_type> signal_type; 
 			typedef void (object_type::*func_t) (arg1_type a1, arg2_type a2) ;
 
-			inline slot2 () : object(NULL), func(NULL) {}
+			inline slot2 () : object(((object_type*)NULL)), func(0) {}
 			inline slot2 (object_type *object, func_t func) : object(object), func(func) {}
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
 			inline void assign(object_type *o, func_t f, signal_type &signal_ref) { object = o; func = f; connect(signal_ref); }
@@ -617,7 +617,7 @@
 			typedef base_signal3 <return_type, arg1_type, arg2_type, arg3_type> signal_type; 
 			typedef return_type (object_type::*func_t) (arg1_type a1, arg2_type a2, arg3_type a3); 
 
-			inline slot3 () : object(NULL), func(NULL) {}
+			inline slot3 () : object(((object_type*)NULL)), func(0) {}
 			inline slot3(object_type *object, func_t func, signal_type * signal = NULL) : object(object), func(func) {}
 
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
@@ -639,7 +639,7 @@
 			typedef base_signal3 <void, arg1_type, arg2_type, arg3_type> signal_type; 
 			typedef void (object_type::*func_t) (arg1_type a1, arg2_type a2, arg3_type a3) ;
 
-			inline slot3 () : object(NULL), func(NULL) {}
+			inline slot3 () : object(((object_type*)NULL)), func(0) {}
 			inline slot3 (object_type *object, func_t func) : object(object), func(func) {}
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
 			inline void assign(object_type *o, func_t f, signal_type &signal_ref) { object = o; func = f; connect(signal_ref); }
@@ -792,7 +792,7 @@
 			typedef base_signal4 <return_type, arg1_type, arg2_type, arg3_type, arg4_type> signal_type; 
 			typedef return_type (object_type::*func_t) (arg1_type a1, arg2_type a2, arg3_type a3, arg4_type a4); 
 
-			inline slot4 () : object(NULL), func(NULL) {}
+			inline slot4 () : object(((object_type*)NULL)), func(0) {}
 			inline slot4(object_type *object, func_t func, signal_type * signal = NULL) : object(object), func(func) {}
 
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
@@ -814,7 +814,7 @@
 			typedef base_signal4 <void, arg1_type, arg2_type, arg3_type, arg4_type> signal_type; 
 			typedef void (object_type::*func_t) (arg1_type a1, arg2_type a2, arg3_type a3, arg4_type a4) ;
 
-			inline slot4 () : object(NULL), func(NULL) {}
+			inline slot4 () : object(((object_type*)NULL)), func(0) {}
 			inline slot4 (object_type *object, func_t func) : object(object), func(func) {}
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
 			inline void assign(object_type *o, func_t f, signal_type &signal_ref) { object = o; func = f; connect(signal_ref); }
@@ -967,7 +967,7 @@
 			typedef base_signal5 <return_type, arg1_type, arg2_type, arg3_type, arg4_type, arg5_type> signal_type; 
 			typedef return_type (object_type::*func_t) (arg1_type a1, arg2_type a2, arg3_type a3, arg4_type a4, arg5_type a5); 
 
-			inline slot5 () : object(NULL), func(NULL) {}
+			inline slot5 () : object(((object_type*)NULL)), func(0) {}
 			inline slot5(object_type *object, func_t func, signal_type * signal = NULL) : object(object), func(func) {}
 
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
@@ -989,7 +989,7 @@
 			typedef base_signal5 <void, arg1_type, arg2_type, arg3_type, arg4_type, arg5_type> signal_type; 
 			typedef void (object_type::*func_t) (arg1_type a1, arg2_type a2, arg3_type a3, arg4_type a4, arg5_type a5) ;
 
-			inline slot5 () : object(NULL), func(NULL) {}
+			inline slot5 () : object(((object_type*)NULL)), func(0) {}
 			inline slot5 (object_type *object, func_t func) : object(object), func(func) {}
 			inline void assign(object_type *o, func_t f) { object = o; func = f; }
 			inline void assign(object_type *o, func_t f, signal_type &signal_ref) { object = o; func = f; connect(signal_ref); }
