--- tap/Makefile	2017-11-16 09:49:42.000000000 -0500
+++ tap/Makefile-new	2020-04-21 21:57:46.261111758 -0400
@@ -23,8 +23,8 @@
 # Change this if you want to install somewhere else. In particular
 # you may wish to remove the middle "local/" part of the path.
 
-INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
-INSTALL_LRDF_DIR	=	/usr/local/share/ladspa/rdf/
+INSTALL_PLUGINS_DIR	=	/app/extensions/LadspaPlugins/TAP/ladspa/
+INSTALL_LRDF_DIR	=	/app/extensions/LadspaPlugins/TAP/ladspa/rdf/
 
 # NO EDITING below this line is required
 # if all you want to do is install and use the plugins.
