--- ./src/BuildUnix.cmake	2020-10-07 16:25:47.975549677 +0200
+++ ./src/BuildUnix-nostatic.cmake	2020-10-07 16:25:59.594653271 +0200
@@ -8,8 +8,5 @@
 
 message (STATUS "Building Linux version")
 
-# Static where possible (i.e. only not on the APPLE)
-set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -static")
-
 set (scythername "scyther-linux")
 add_executable (${scythername} ${Scyther_sources})
\ Pas de fin de ligne à la fin du fichier
