diff -u -r -u FreeFileSync_9.5_Source.orig/FreeFileSync/Source/Makefile FreeFileSync_9.5_Source/FreeFileSync/Source/Makefile
--- FreeFileSync_9.5_Source.orig/FreeFileSync/Source/Makefile	2017-11-06 17:24:14.000000000 +0100
+++ FreeFileSync_9.5_Source/FreeFileSync/Source/Makefile	2017-11-29 15:58:14.272865268 +0100
@@ -1,5 +1,5 @@
 APPNAME     = FreeFileSync
-prefix      = /usr
+prefix      = /app
 BINDIR      = $(DESTDIR)$(prefix)/bin
 SHAREDIR    = $(DESTDIR)$(prefix)/share
 APPSHAREDIR = $(SHAREDIR)/$(APPNAME)
@@ -122,5 +122,5 @@
 	$(APPSHAREDIR)
 
 	mkdir -p $(DOCSHAREDIR)
-	cp ../Build/Changelog.txt $(DOCSHAREDIR)/changelog
+	cp ../../Changelog.txt $(DOCSHAREDIR)/changelog
 	gzip $(DOCSHAREDIR)/changelog
diff -u -r -u FreeFileSync_9.5_Source.orig/FreeFileSync/Source/RealTimeSync/Makefile FreeFileSync_9.5_Source/FreeFileSync/Source/RealTimeSync/Makefile
--- FreeFileSync_9.5_Source.orig/FreeFileSync/Source/RealTimeSync/Makefile	2017-11-05 19:53:03.000000000 +0100
+++ FreeFileSync_9.5_Source/FreeFileSync/Source/RealTimeSync/Makefile	2017-11-29 15:58:22.883863201 +0100
@@ -1,5 +1,5 @@
 APPNAME = RealTimeSync
-prefix = /usr
+prefix = /app
 BINDIR = $(DESTDIR)$(prefix)/bin
 
 CXXFLAGS  = -std=c++14 -pipe -DWXINTL_NO_GETTEXT_MACRO -I../../.. -I../../../zenXml -include "zen/i18n.h" -Wall -O3 -DNDEBUG `wx-config --cxxflags --debug=no` -pthread
diff -u -r -u FreeFileSync_9.5_Source.orig/FreeFileSync/Source/ui/main_dlg.cpp FreeFileSync_9.5_Source/FreeFileSync/Source/ui/main_dlg.cpp
--- FreeFileSync_9.5_Source.orig/FreeFileSync/Source/ui/main_dlg.cpp	2017-11-05 19:53:04.000000000 +0100
+++ FreeFileSync_9.5_Source/FreeFileSync/Source/ui/main_dlg.cpp	2017-11-29 15:58:01.844868251 +0100
@@ -1025,7 +1025,7 @@
         history.resize(globalSettings.gui.cfgFileHistMax);
 
     globalSettings.gui.cfgFileHistory = history;
-    globalSettings.gui.cfgFileHistFirstItemPos = m_listBoxHistory->GetTopItem();
+    // globalSettings.gui.cfgFileHistFirstItemPos = m_listBoxHistory->GetTopItem();
     //--------------------------------------------------------------------------------
     globalSettings.gui.lastUsedConfigFiles.clear();
     for (const Zstring& cfgFilePath : activeConfigFiles_)
@@ -4871,13 +4871,13 @@
 
     m_menuItemCheckVersionAuto->Check(updateCheckActive(globalCfg_.gui.lastUpdateCheck));
 
-    if (shouldRunPeriodicUpdateCheck(globalCfg_.gui.lastUpdateCheck))
+    /* if (shouldRunPeriodicUpdateCheck(globalCfg_.gui.lastUpdateCheck))
     {
         flashStatusInformation(_("Searching for program updates..."));
         //synchronous update check is sufficient here:
         periodicUpdateCheckEval(this, globalCfg_.gui.lastUpdateCheck, globalCfg_.gui.lastOnlineVersion,
                                 periodicUpdateCheckRunAsync(periodicUpdateCheckPrepare().get()).get());
-    }
+    } */
 }
 
 
@@ -4886,7 +4886,7 @@
     //execute just once per startup!
     Disconnect(wxEVT_IDLE, wxIdleEventHandler(MainDialog::OnRegularUpdateCheck), nullptr, this);
 
-    if (shouldRunPeriodicUpdateCheck(globalCfg_.gui.lastUpdateCheck))
+    /* if (shouldRunPeriodicUpdateCheck(globalCfg_.gui.lastUpdateCheck))
     {
         flashStatusInformation(_("Searching for program updates..."));
 
@@ -4898,7 +4898,7 @@
             periodicUpdateCheckEval(this, globalCfg_.gui.lastUpdateCheck, globalCfg_.gui.lastOnlineVersion,
                                     resultAsync.get()); //run on main thread:
         });
-    }
+    } */
 }
 
 
diff -u -r -u FreeFileSync_9.5_Source.orig/zen/shell_execute.h FreeFileSync_9.5_Source/zen/shell_execute.h
--- FreeFileSync_9.5_Source.orig/zen/shell_execute.h	2017-11-05 19:53:04.000000000 +0100
+++ FreeFileSync_9.5_Source/zen/shell_execute.h	2017-11-29 15:58:01.845868251 +0100
@@ -7,6 +7,7 @@
 #ifndef SHELL_EXECUTE_H_23482134578134134
 #define SHELL_EXECUTE_H_23482134578134134
 
+#include "warn_static.h"
 #include "file_error.h"
 
     #include "thread.h"
