diff --git a/packaging/flatpak/flatpak-buildgen.sh b/packaging/flatpak/flatpak-buildgen.sh
index 6d2e15df9..2f5781f32 100644
--- a/packaging/flatpak/flatpak-buildgen.sh
+++ b/packaging/flatpak/flatpak-buildgen.sh
@@ -9,10 +9,10 @@ set -e
 set -x
 
 # Untar deutex to wad directory
-tar --zstd -xvf deutex-5.2.2.tar.zst -C wad/
+tar --zstd -xvf deutex-5.2.2.tar.zst -C wad/ --no-same-owner
 
 # Untar wxWidgets to libraries directory
-tar -xvf wxWidgets-3.0.5.tar.bz2 -C libraries/
+tar -xvf wxWidgets-3.0.5.tar.bz2 -C libraries/ --no-same-owner
 
 mkdir -p build && cd build
 # Generate build
