--- CMT/src/Makefile	2019-01-03 10:08:22.000000000 -0500
+++ CMT/src/Makefile-new	2020-04-21 21:45:27.861132115 -0400
@@ -4,7 +4,7 @@
 #
 # Change this if you want to install somewhere else.
 
-INSTALL_PLUGINS_DIR	=	/usr/lib/ladspa/
+INSTALL_PLUGINS_DIR	=	/app/extensions/LadspaPlugins/CMT/ladspa/
 
 ###############################################################################
 #
@@ -76,7 +76,7 @@
 		$(PLUGIN_OBJECTS)					
 
 install:	$(PLUGIN_LIB)
-	cp $(PLUGIN_LIB) $(INSTALL_PLUGINS_DIR)
+	mkdir -p $(INSTALL_PLUGINS_DIR) && cp $(PLUGIN_LIB) $(INSTALL_PLUGINS_DIR)
 
 test:	/tmp/test.wav ../../ladspa_sdk/snd/noise.wav always
 	@echo ---------------------------------------------
