From c5329cbc5014771958c19f777042ca8c62b6a0be Mon Sep 17 00:00:00 2001
From: ngc92 <7938269+ngc92@users.noreply.github.com>
Date: Wed, 25 Feb 2015 12:00:33 +0000
Subject: [PATCH] added inlude

---
 src/NetworkMessage.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/NetworkMessage.cpp b/src/NetworkMessage.cpp
index 9178e745..112d282f 100644
--- a/src/NetworkMessage.cpp
+++ b/src/NetworkMessage.cpp
@@ -23,6 +23,7 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 /* includes */
 #include <cstring>
+#include <ostream>
 
 #include "UserConfig.h"
 #include "SpeedController.h"
