diff --git a/layout.py b/layout.py
index 426e1de..e0a5c98 100644
--- a/layout.py
+++ b/layout.py
@@ -35,6 +35,8 @@ from toolbars import MiscToolbar
 
 from numerals import local as _n
 
+from sugarapp.widgets import ExtendedActivityToolbarButton
+
 try:
     from sugar3.graphics.toolbarbox import ToolbarButton, ToolbarBox
     from sugar3.activity.widgets import ActivityToolbarButton
@@ -138,7 +140,7 @@ class CalcLayout:
 
 # Toolbar
         self._toolbar_box = ToolbarBox()
-        activity_button = ActivityToolbarButton(self._parent)
+        activity_button = ExtendedActivityToolbarButton(self._parent)
         self._toolbar_box.toolbar.insert(activity_button, 0)
 
         def append(icon_name, label, page, position):
diff --git a/shareable_activity.py b/shareable_activity.py
index 46d02b2..1e8de97 100644
--- a/shareable_activity.py
+++ b/shareable_activity.py
@@ -28,8 +28,9 @@ class ShareableObject(dbus.service.Object):
     def SendMessageTo(self, busname, msg, kwargs):
         pass
 
+from sugarapp.widgets import SugarCompatibleActivity
 
-class ShareableActivity(activity.Activity):
+class ShareableActivity(SugarCompatibleActivity):
 
     '''
     A shareable activity.
@@ -47,7 +48,7 @@ class ShareableActivity(activity.Activity):
             service_path
         '''
 
-        activity.Activity.__init__(self, handle, *args, **kwargs)
+        SugarCompatibleActivity.__init__(self, handle, *args, **kwargs)
 
         self._sync_hid = None
         self._message_cbs = {}
