diff --git a/toolbars.py b/toolbars.py
index ca24c2c..9924a00 100644
--- a/toolbars.py
+++ b/toolbars.py
@@ -13,6 +13,8 @@ from sugar3.graphics.toolbutton import ToolButton
 from sugar3.graphics.toggletoolbutton import ToggleToolButton
 from sugar3.graphics.style import GRID_CELL_SIZE
 
+from sugarapp.helpers import PrimaryMonitor
+
 import logging
 _logger = logging.getLogger('calc-activity')
 
@@ -383,7 +385,7 @@ class MiscToolbar(Gtk.Toolbar):
         self.show_all()
 
     def update_layout(self):
-        if Gdk.Screen.width() < 14 * GRID_CELL_SIZE or \
+        if PrimaryMonitor.width() < 14 * GRID_CELL_SIZE or \
                 self._target_toolbar is None:
             target_toolbar = self
             if self._target_toolbar is not None:
