diff --git a/activity/activity.info b/activity/activity.info
index cc53226..9e25207 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,12 +1,19 @@
 [Activity]
 name = Calculate
-bundle_id = org.laptop.Calculate
+bundle_id = org.sugarlabs.Calculate
 icon = calculate
 exec = sugar-activity3 calculate.Calculate -s
 activity_version = 47
 license = GPLv2+
-max_participants = 10
-summary = This is the place to get the answer to a quick problem, but that is not the limit! You can also explore Algebra, Trigonometry, Boolean and more!
+metadata_license = CC0-1.0
+max_participants = 1
+summary = An infix-notation graphing calculator
+description = This is the place to get the answer to a quick problem, but that is not the limit! You can also explore Algebra, Trigonometry, Boolean and more!
 repository = https://github.com/sugarlabs/sugarlabs-calculate
 url = https://help.sugarlabs.org/calculate.html
-tags = Maths
+tags = Education
+release_date = 2024-06-15
+developer_name = Sugar Labs Community
+developer_id = org.sugarlabs
+update_contact = tch@sugarlabs.org
+screenshots = https://i.imgur.com/qC6lQrp.png https://i.imgur.com/R84JBRc.png https://i.imgur.com/XY9hFTQ.png
\ No newline at end of file
