diff --git a/Cargo.toml b/Cargo.toml
index 53076de..00b1ef3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,8 +16,8 @@ serde_json = "1.0.64"
 itertools = "0.10.0"
 
 # relm: take in style_class after 0.21
-relm = { git = "https://github.com/antoyo/relm", rev = "1e4dee58f2536857420756b533f0673e60beffa7" }
-relm-derive = { git = "https://github.com/antoyo/relm", rev = "1e4dee58f2536857420756b533f0673e60beffa7" }
+relm = "0.21.0"
+relm-derive = "0.21.0"
 
 glib = "0.10.3"
 gio = "0.9.1"
