diff --git a/activity.py b/activity.py
index f84da99..b40c4f2 100644
--- a/activity.py
+++ b/activity.py
@@ -22,16 +22,18 @@ from gi.repository import Gtk
 import pygame
 import sugargame
 import sugargame.canvas
-from sugar3.activity import activity
 from sugar3.graphics.toolbarbox import ToolbarBox
-from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.activity.widgets import StopButton
 import cells
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
+from sugarapp.helpers import PrimaryMonitor
+from sugar3.graphics.style import GRID_CELL_SIZE
 
-class Activity(activity.Activity):
+class Activity(SugarCompatibleActivity):
     def __init__(self, handle):
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self.max_participants = 1
         self.game = cells.Game()
@@ -43,13 +45,18 @@ class Activity(activity.Activity):
         self.game.canvas.grab_focus()
         self.build_toolbar()
 
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height()
+        # 6 added in order to remove space b/w top panel and pygame canvas
+        self.game.canvas.set_size_request(w, h - GRID_CELL_SIZE + 6)
+
     def build_toolbar(self):
 
         toolbar_box = ToolbarBox()
         self.set_toolbar_box(toolbar_box)
         toolbar_box.show()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
 
@@ -65,7 +72,6 @@ class Activity(activity.Activity):
         stop_button.show()
         stop_button.connect('clicked', self._stop_cb)
 
-        self.show_all()
 
     def _stop_cb(self, button):
         self.game.running = False
