diff --git a/activity/activity.info b/activity/activity.info
index b47e41a..4bc88f0 100755
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,9 +1,19 @@
 [Activity]
-name = Cells
-bundle_id = edu.centenary.CellManagement
+name = CellGame
+bundle_id = org.sugarlabs.CellGame
 icon = cellgame.sugar
 activity_version = 5
 show_launcher = yes
 exec = sugar-activity3 activity.Activity
-summary = This game is based on the mechanisms present in gene regulatory networks
+summary = Based on gene regulatory networks
 repository = https://github.com/sugarlabs/cellgame.git
+url = https://github.com/sugarlabs/cellgame
+license = GPLv3+
+release_date = 2024-06-22
+description = Aliens have abducted two each of six species from Earth. All are currently held captive on a spaceship returning to the alien home-planet, with each species in a different cell. The player's goal is to have one of each type of species escape to the center of the ship. A group of unique individuals, one for each species, will give the player the strength they need to overpower the alien guards and release everyone.
+tags = Education
+metadata_license = CC0-1.0
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+developer_id = org.sugarlabs
+screenshots = https://i.imgur.com/VdHweQE.png https://i.imgur.com/HyPb02d.png https://i.imgur.com/2pAJMrx.png
