diff --git a/torbrowser_launcher/common.py b/torbrowser_launcher/common.py
index 25bb984..ffa0cd7 100644
--- a/torbrowser_launcher/common.py
+++ b/torbrowser_launcher/common.py
@@ -193,7 +193,7 @@ class Common(object):
         else:
             print('Refreshing local keyring...')
 
-        p = subprocess.Popen(['/usr/bin/gpg2', '--status-fd', '2',
+        p = subprocess.Popen(['/usr/bin/gpg', '--status-fd', '2',
                               '--homedir', self.paths['gnupg_homedir'],
                               '--keyserver', 'hkps://keys.openpgp.org',
                               '--refresh-keys'], stderr=subprocess.PIPE)
