From c8694bf435ba28b704ee310c9623c8f260cb8177 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Thu, 12 Dec 2019 13:44:14 -0300
Subject: [PATCH 3/3] monitors

---
 activity.py   | 5 +++--
 helpbutton.py | 6 ++++--
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/activity.py b/activity.py
index 1b6baee..d80436a 100644
--- a/activity.py
+++ b/activity.py
@@ -62,6 +62,7 @@ from readers import ClipboardReader
 import charthelp
 import chart as charts
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import DesktopOpenChooser
 from sugarapp.widgets import DesktopSaveChooser
 from sugarapp.widgets import SugarCompatibleActivity
@@ -426,7 +427,7 @@ class ChartActivity(SugarCompatibleActivity):
                                            size_allocate_cb)
 
         scroll = Gtk.ScrolledWindow()
-        scroll.set_min_content_width(min(170, Gdk.Screen.width() / 6))
+        scroll.set_min_content_width(min(170, PrimaryMonitor.width() / 6))
         scroll.set_policy(Gtk.PolicyType.AUTOMATIC, Gtk.PolicyType.AUTOMATIC)
         self.labels_and_values = ChartData(self)
         scroll.add(self.labels_and_values)
@@ -704,7 +705,7 @@ class ChartActivity(SugarCompatibleActivity):
 
     def _configure_cb(self, event=None):
         # If we have room, put buttons on the main toolbar
-        if Gdk.Screen.width() / 14 > style.GRID_CELL_SIZE:
+        if PrimaryMonitor.width() / 14 > style.GRID_CELL_SIZE:
             self._chart_button.set_expanded(False)
             self._chart_button.hide()
             for i in range(4):
diff --git a/helpbutton.py b/helpbutton.py
index d27e8c1..6d8c8ce 100644
--- a/helpbutton.py
+++ b/helpbutton.py
@@ -28,6 +28,8 @@ from sugar3.graphics.toolbutton import ToolButton
 from sugar3.graphics.icon import Icon
 from sugar3.graphics import style
 
+from sugarapp.helpers import PrimaryMonitor
+
 
 class HelpButton(Gtk.ToolItem):
 
@@ -41,8 +43,8 @@ class HelpButton(Gtk.ToolItem):
         self._palette = help_button.get_palette()
 
         sw = Gtk.ScrolledWindow()
-        sw.set_size_request(int(Gdk.Screen.width() / 2.8),
-                            Gdk.Screen.height() - style.GRID_CELL_SIZE * 3)
+        sw.set_size_request(int(PrimaryMonitor.width() / 2.8),
+                            PrimaryMonitor.height() - style.GRID_CELL_SIZE * 3)
         sw.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
 
         self._vbox = Gtk.Box()
-- 
2.19.1

