From 4d82e95ecc513c95b9ddd800b7ea6c0748c4d2ee Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Thu, 12 Dec 2019 12:10:42 -0300
Subject: [PATCH 1/3] info

---
 activity/activity.info | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index dc0f769..7ef5ed6 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,12 +1,18 @@
 [Activity]
 name = Chart
 activity_version = 13
+release_date = 2019-10-24
 license = GPLv3+;GPLv2+
-bundle_id = org.sugarlabs.SimpleGraph
+metadata_license = CC0-1.0
+bundle_id = org.sugarlabs.Chart
 exec = sugar-activity3 activity.ChartActivity -s
 icon = chart-activity
 mime_types = application/x-chart-activity;
-summary = Create your charts and organize your information! Try the best chart for your case.
-tags = Maths;Science
+summary = Create your charts
+description = Chart lets you do graphical representation of simple tabular data. It can draw Horizontal/Vertical bar charts, line charts and pie charts.
+tags = Education
 url = https://help.sugarlabs.org/en/chart.html
 repository = https://github.com/sugarlabs/chart
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/chart/master/screenshots/1.png https://raw.githubusercontent.com/sugarlabs/chart/master/screenshots/2.png
-- 
2.19.1

