diff --git a/activity/activity.info b/activity/activity.info
index 96e3c08..16ccc24 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,11 +1,19 @@
 [Activity]
-name = SimpleGNUChess
+name = Sugar Chess
 activity_version = 18
-license = GPLv3
-bundle_id = org.sugarlabs.GNUChessActivity
+release_date = 2020-05-23
+license = GPLv3+
+metadata_license = CC0-1.0
+bundle_id = org.sugarlabs.Chess
 exec = sugar-activity3 GNUChessActivity.GNUChessActivity
 icon = activity-gnuchess
 show_launcher = yes
 mime_types = application/x-chess-pgn;text/plain
-summary = a Sugar interface to the gnuchess program
-repository = https://github.com/walterbender/sugarchess
+tags = Game
+summary = A sweet Chess game
+description = Sugar Chess is a front end to the gnuchess program. It features varying levels of difficulty, undo, PGN output, customization of graphics, et al.
+repository = https://github.com/sugarlabs/sugarchess
+url = https://github.com/sugarlabs/sugarchess
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/sugarchess/master/screenshots/en/1.png
