diff --git a/chess.py b/chess.py
index 065fe17..fd63b51 100644
--- a/chess.py
+++ b/chess.py
@@ -121,21 +121,14 @@ class Gnuchess():
         # Generate the sprites we'll need...
         self._sprites = Sprites(self._canvas)
         self._generate_sprites(colors)
-
-        p = subprocess.Popen(['uname', '-m'], # in my virtual machine, -p returns unknown
-                             stdin=subprocess.PIPE,
-                             stdout=subprocess.PIPE,
-                             stderr=subprocess.STDOUT)
-        tmp_p = p.communicate()
-        self._bin_path = 'bin/%s' % (BIN[tmp_p[0].decode().replace('\n', '')])
+        self._bin_path = '/app/bin'
         self._all_clear()
         os.system('chmod -R 755 bin -f')
 
     def move(self, my_move):
         ''' Send a command to gnuchess. '''
         # Permission to play
-        p = subprocess.Popen(['{}/{}/gnuchess'.format(self._bundle_path,
-                                                      self._bin_path)],
+        p = subprocess.Popen(['{}/gnuchess'.format(self._bin_path)],
                              stdin=subprocess.PIPE,
                              stdout=subprocess.PIPE,
                              stderr=subprocess.STDOUT)
