diff -p -up circuslinux-1.0.3/Makefile.am.fix64bits circuslinux-1.0.3/Makefile.am
--- circuslinux-1.0.3/Makefile.am.fix64bits	2001-04-29 08:53:26.000000000 +0200
+++ circuslinux-1.0.3/Makefile.am	2009-06-12 04:43:00.000000000 +0200
@@ -18,7 +18,7 @@ VERSION = @VERSION@
 CC=@CC@
 DATA_PREFIX=$(pkgdatadir)/data/
 JOY=@JOY@
-TARGET_DEF=@TARGET_DEF@
+TARGET_DEF=LINUX
 
 
 CFLAGS  = @CFLAGS@ -Wall -O2 -DDATA_PREFIX=\"$(DATA_PREFIX)\" \
diff -p -up circuslinux-1.0.3/Makefile.in.fix64bits circuslinux-1.0.3/Makefile.in
--- circuslinux-1.0.3/Makefile.in.fix64bits	2001-04-29 08:53:31.000000000 +0200
+++ circuslinux-1.0.3/Makefile.in	2009-06-12 04:43:16.000000000 +0200
@@ -83,7 +83,7 @@ VERSION = @VERSION@
 CC = @CC@
 DATA_PREFIX = $(pkgdatadir)/data/
 JOY = @JOY@
-TARGET_DEF = @TARGET_DEF@
+TARGET_DEF = LINUX
 
 CFLAGS = @CFLAGS@ -Wall -O2 -DDATA_PREFIX=\"$(DATA_PREFIX)\" 	  -D$(TARGET_DEF) -DJOY_$(JOY)
 
