diff -ur circuslinux-1.0.3/circuslinux.c circuslinux-1.0.3_novos/circuslinux.c
--- circuslinux-1.0.3/circuslinux.c	2001-04-29 09:38:01.000000000 +0300
+++ circuslinux-1.0.3_novos/circuslinux.c	2019-12-31 10:50:09.783266954 +0200
@@ -3439,8 +3439,8 @@
   /* Get home directory (from $HOME variable)... if we can't determine it,
      use the current directory ("."): */
 
-  if (getenv("HOME") != NULL)
-    home = getenv("HOME");
+  if (getenv("XDG_CONFIG_HOME") != NULL)
+    home = getenv("XDG_CONFIG_HOME");
   else
     home = ".";
   
