From 1dfc304f233128b939c0244a5da7517bc809941a Mon Sep 17 00:00:00 2001
From: "Andre A. Gomes" <andremegafone@gmail.com>
Date: Tue, 29 Aug 2023 17:43:55 +0300
Subject: [PATCH] Fix order of shared libraries.

---
 src/reload.lisp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/reload.lisp b/src/reload.lisp
index 8ef9190..12fd576 100644
--- a/_build/cl-plus-ssl/src/reload.lisp
+++ b/_build/cl-plus-ssl/src/reload.lisp
@@ -127,7 +127,8 @@ defun detect-custom-openssl-installations-if-macos
                   ;; See https://github.com/cl-plus-ssl/cl-plus-ssl/pull/115
                   "libcrypto.dylib"
                   "/usr/lib/libcrypto.dylib"))
-    ((and :unix (not :cygwin)) (:or "libcrypto.so.1.1"
+    ((and :unix (not :cygwin)) (:or "libcrypto.so.3"
+                                    "libcrypto.so.1.1"
                                     "libcrypto.so.1.0.0"
                                     "libcrypto.so"))
     (:cygwin (:or "cygcrypto-1.1.dll" "cygcrypto-1.0.0.dll"))))
@@ -163,7 +164,8 @@ defun detect-custom-openssl-installations-if-macos
     ;; so we can just use just "libssl.so".
     ;; More info at https://github.com/cl-plus-ssl/cl-plus-ssl/pull/2.
     (:openbsd "libssl.so")
-    ((and :unix (not :cygwin)) (:or "libssl.so.1.1"
+    ((and :unix (not :cygwin)) (:or "libssl.so.3"
+                                    "libssl.so.1.1"
                                     "libssl.so.1.0.2m"
                                     "libssl.so.1.0.2k"
                                     "libssl.so.1.0.2"
--
2.41.0
