diff --git a/gui/vite.config.ts b/gui/vite.config.ts
index c24c762..831062c 100644
--- a/gui/vite.config.ts
+++ b/gui/vite.config.ts
@@ -9,7 +9,7 @@ const versionTag = execSync('git --no-pager tag --points-at HEAD')
   .toString()
   .trim();
 // If not empty then it's not clean
-const gitClean = execSync('git status --porcelain').toString() ? false : true;
+const gitClean = true;
 
 console.log(
   `version is ${versionTag || commitHash}${gitClean ? '' : '-dirty'}`
