--- a/frontend_qt/mainwindow.cpp	2021-04-07 04:08:10.142848556 +0200
+++ b/frontend_qt/mainwindow.cpp	2021-04-07 04:10:04.877707078 +0200
@@ -28,6 +28,7 @@
 #include <QClipboard>
 #include <QMimeData>
 #include <QColor>
+#include <QtGlobal>
 
 #include "mainwindow.h"
 #include "datawindow.h"
@@ -371,11 +372,10 @@
 {
     QClipboard *clipboard = QGuiApplication::clipboard();
     QMimeData *data = new QMimeData;
-    QString filename = ".zint.svg";
+    QString filename = QString::fromLocal8Bit(qgetenv("XDG_RUNTIME_DIR")).append("/app/uk.org.zint.zint-qt/.zint.svg");
     double scale = spnScale->value();
 
     spnScale->setValue(5);
-
     if (!m_bc.bc.save_to_file(filename)) {
         return;
     }
@@ -383,7 +383,7 @@
     data->setImageData(QImage(filename));
     clipboard->setMimeData(data, QClipboard::Clipboard);
 
-    QFile::remove(filename);
+    //QFile::remove(filename);
 
     spnScale->setValue(scale);
 }
@@ -392,7 +392,7 @@
 {
     QClipboard *clipboard = QGuiApplication::clipboard();
     QMimeData *data = new QMimeData;
-    QString filename = ".zint.bmp";
+    QString filename = QString::fromLocal8Bit(qgetenv("XDG_RUNTIME_DIR")).append("/app/uk.org.zint.zint-qt/.zint.bmp");
 
     if (!m_bc.bc.save_to_file(filename)) {
         return;
@@ -401,7 +401,7 @@
     data->setImageData(QImage(filename));
     clipboard->setMimeData(data, QClipboard::Clipboard);
 
-    QFile::remove(filename);
+    //QFile::remove(filename);
 }
 
 void MainWindow::change_options()
