--- a/cplusplus/singletons/pqc_scripts.cpp
+++ b/cplusplus/singletons/pqc_scripts.cpp
@@ -34,6 +34,8 @@
 #include <QMimeDatabase>
 #include <QImageReader>
 #include <QKeyEvent>
+#include <QApplication>
+#include <QClipboard>
 #include <thread>
 #include <chrono>

@@ -1285,7 +1287,7 @@ void PQCScripts::copyTextToClipboard(QString txt) {

     qDebug() << "args: txt.length =" << txt.length();

-    qApp->clipboard()->setText(txt, QClipboard::Clipboard);
+    QApplication::clipboard()->setText(txt, QClipboard::Clipboard);

 }

