--- CMakeLists.txt	2025-06-01 16:11:11.789836098 -0700
+++ CMakeLists.txt	2025-06-02 15:11:06.770048743 -0700
@@ -70,11 +70,6 @@
     target_precompile_headers(abaddon PRIVATE <gtkmm.h> src/abaddon.hpp src/util.hpp)
 endif ()
 
-if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR
-(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
-((CMAKE_SYSTEM_NAME STREQUAL "Linux") OR (CMAKE_CXX_COMPILER_VERSION LESS 9))))
-    target_link_libraries(abaddon stdc++fs)
-endif ()
 
 if (NOT WIN32)
     target_sources(abaddon PRIVATE src/notifications/notifier_gio.cpp)
