diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0b051270..7c58d7a3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,6 +128,9 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
                 OUTPUT_VARIABLE LINUX_DISTRO_REL
                 OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_QUIET)
             set(LINUX_DISTRO "Arch")
+        elseif(EXISTS /etc/os-release)
+            cmake_host_system_information(RESULT LINUX_DISTRO QUERY DISTRIB_NAME)
+            cmake_host_system_information(RESULT LINUX_DISTRO_REL QUERY DISTRIB_VERSION)
         else()
             message(SEND_ERROR "lsb_release tool not found.")
         endif()
@@ -626,14 +629,14 @@ if(LINUX)
     # .desktop File
     install(FILES
         resources/mmex.desktop
-        DESTINATION share/applications)
+        DESTINATION share/applications RENAME org.moneymanagerex.MMEX.desktop)
 endif()
 
 # Icons
 if(LINUX)
     install(FILES
         resources/mmex.svg
-        DESTINATION share/icons/hicolor/scalable/apps)
+        DESTINATION share/icons/hicolor/scalable/apps RENAME org.moneymanagerex.MMEX.svg)
 elseif(APPLE)
     install(FILES
         "${MACOSX_APP_ICON_FILE}"
