diff -Naur openclonk-release-8.0-orig/cmake/FindAudio.cmake openclonk-release-8.0-src/cmake/FindAudio.cmake
--- openclonk-release-8.0-orig/cmake/FindAudio.cmake	2018-02-05 00:39:08.000000000 +0900
+++ openclonk-release-8.0-src/cmake/FindAudio.cmake	2018-03-07 09:54:09.328625319 +0900
@@ -145,3 +145,5 @@
 		message(STATUS "Not enabling audio output.")
 	endif()
 endif()
+
+set(Audio_FIND_REQUIRED ${Audio_FIND_REQUIRED}) # Shut up the warning about this variable being unset.
diff -Naur openclonk-release-8.0-orig/CMakeLists.txt openclonk-release-8.0-src/CMakeLists.txt
--- openclonk-release-8.0-orig/CMakeLists.txt	2018-02-05 00:39:08.000000000 +0900
+++ openclonk-release-8.0-src/CMakeLists.txt	2018-03-07 09:54:09.328625319 +0900
@@ -1194,7 +1194,7 @@
 	endif()
 endif()
 
-add_executable(openclonk-server
+add_executable(openclonk-server EXCLUDE_FROM_ALL
 	${OC_SYSTEM_SOURCES}
 	${OC_CLONK_SOURCES}
 	src/platform/C4AppT.cpp
@@ -1429,27 +1429,27 @@
 if (NOT APPLE)
 	install(
 		FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/res/oc32.png
-		DESTINATION share/icons/hicolor/32x32/apps
+		DESTINATION app/share/icons/hicolor/32x32/apps
 		RENAME openclonk.png
 	)
 	install(
 		FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/res/oc64.png
-		DESTINATION share/icons/hicolor/64x64/apps
+		DESTINATION app/share/icons/hicolor/64x64/apps
 		RENAME openclonk.png
 	)
 	install(
 		FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/res/oc128.png
-		DESTINATION share/icons/hicolor/128x128/apps
+		DESTINATION app/share/icons/hicolor/128x128/apps
 		RENAME openclonk.png
 	)
 	install(
 		FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/res/oc256.png
-		DESTINATION share/icons/hicolor/256x256/apps
+		DESTINATION app/share/icons/hicolor/256x256/apps
 		RENAME openclonk.png
 	)
 	install(
 		FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/res/oc512.png
-		DESTINATION share/icons/hicolor/512x512/apps
+		DESTINATION app/share/icons/hicolor/512x512/apps
 		RENAME openclonk.png
 	)
 endif()
@@ -1535,7 +1535,7 @@
 if(NOT HEADLESS_ONLY)
 	if(NOT APPLE)
 		# Install new files
-		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.desktop DESTINATION share/applications)
+		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.desktop DESTINATION app/share/applications)
 		# Update the MIME cache -- this makes the clonk:// protocol handler actually work
 		install(CODE "message(\"Note: Run update-desktop-database to install clonk:// protocol handler\")")
 
@@ -1543,7 +1543,7 @@
 		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.appdata.xml DESTINATION share/appdata)
 
 		# Install binaries
-		install(TARGETS openclonk DESTINATION games)
+		install(TARGETS openclonk DESTINATION bin)
 	else()
 		install(TARGETS openclonk
 			BUNDLE DESTINATION .
