diff -Naur openclonk-9/cmake/FindAudio.cmake openclonk-11/cmake/FindAudio.cmake
--- openclonk-9/cmake/FindAudio.cmake	2018-03-03 04:39:09.776666655 +0000
+++ openclonk-11/cmake/FindAudio.cmake	2018-03-03 04:34:51.800000001 +0000
@@ -145,3 +145,5 @@
 		message(STATUS "Not enabling audio output.")
 	endif()
 endif()
+
+set(Audio_FIND_REQUIRED ${Audio_FIND_REQUIRED}) # Shut up the warning about this variable being unset.
diff -Naur openclonk-9/CMakeLists.txt openclonk-11/CMakeLists.txt
--- openclonk-9/CMakeLists.txt	2018-02-04 15:39:08.000000000 +0000
+++ openclonk-11/CMakeLists.txt	2018-03-03 04:38:50.166666658 +0000
@@ -1194,7 +1194,7 @@
 	endif()
 endif()
 
-add_executable(openclonk-server
+add_executable(openclonk-server EXCLUDE_FROM_ALL
 	${OC_SYSTEM_SOURCES}
 	${OC_CLONK_SOURCES}
 	src/platform/C4AppT.cpp
@@ -1543,7 +1543,7 @@
 		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.appdata.xml DESTINATION share/appdata)
 
 		# Install binaries
-		install(TARGETS openclonk DESTINATION games)
+		install(TARGETS openclonk DESTINATION bin)
 	else()
 		install(TARGETS openclonk
 			BUNDLE DESTINATION .
