diff --git a/CMakeLists.txt b/CMakeLists.txt
index 10d667a..4ead59a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,12 +62,14 @@ if (NOT (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
     target_precompile_headers(abaddon PRIVATE <gtkmm.h> src/abaddon.hpp src/util.hpp)
 endif ()
 
-if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR
-(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
-((CMAKE_SYSTEM_NAME STREQUAL "Linux") OR (CMAKE_CXX_COMPILER_VERSION LESS 9))))
+if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "9.1"))
     target_link_libraries(abaddon stdc++fs)
 endif ()
 
+if ((CMAKE_CXX_COMPILER_ID STREQUAL "Clang") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "9.0"))
+    target_link_libraries(abaddon c++fs)
+endif ()
+
 if (NOT WIN32)
     target_sources(abaddon PRIVATE src/notifications/notifier_gio.cpp)
 else ()
