diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0b051270..fb42d1f40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,6 +128,9 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
                 OUTPUT_VARIABLE LINUX_DISTRO_REL
                 OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_QUIET)
             set(LINUX_DISTRO "Arch")
+        elseif(EXISTS /etc/os-release)
+            cmake_host_system_information(RESULT LINUX_DISTRO QUERY DISTRIB_NAME)
+            cmake_host_system_information(RESULT LINUX_DISTRO_REL QUERY DISTRIB_VERSION)
         else()
             message(SEND_ERROR "lsb_release tool not found.")
         endif()
