diff --git a/CMakeLists.txt b/CMakeLists.txt
index 38ff055..b1b8867 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,10 @@ else ()
 	set(PROJECT_VERSION_PATCH "0")
 endif ()
 
+set(PROJECT_VERSION_MAJOR "0")
+set(PROJECT_VERSION_MINOR "6")
+set(PROJECT_VERSION_PATCH "0")
+
 set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
 message(STATUS "Building slowmoVideo ${PROJECT_VERSION}")
 configure_file(src/version.h.in version.h)
