diff -Naur openclonk-release-8.0-orig/cmake/FindAudio.cmake openclonk-release-8.0-src/cmake/FindAudio.cmake
--- openclonk-release-8.0-orig/cmake/FindAudio.cmake	2018-02-05 00:39:08.000000000 +0900
+++ openclonk-release-8.0-src/cmake/FindAudio.cmake	2018-03-07 09:54:09.328625319 +0900
@@ -145,3 +145,5 @@
 		message(STATUS "Not enabling audio output.")
 	endif()
 endif()
+
+set(Audio_FIND_REQUIRED ${Audio_FIND_REQUIRED}) # Shut up the warning about this variable being unset.
diff -Naur openclonk-release-8.0-orig/CMakeLists.txt openclonk-release-8.0-src/CMakeLists.txt
--- openclonk-release-8.0-orig/CMakeLists.txt	2018-02-05 00:39:08.000000000 +0900
+++ openclonk-release-8.0-src/CMakeLists.txt	2018-03-07 09:54:09.328625319 +0900
@@ -1194,7 +1194,7 @@
 	endif()
 endif()
 
-add_executable(openclonk-server
+add_executable(openclonk-server EXCLUDE_FROM_ALL
 	${OC_SYSTEM_SOURCES}
 	${OC_CLONK_SOURCES}
 	src/platform/C4AppT.cpp
@@ -1543,7 +1543,7 @@
 		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.appdata.xml DESTINATION share/appdata)
 
 		# Install binaries
-		install(TARGETS openclonk DESTINATION games)
+		install(TARGETS openclonk DESTINATION bin)
 	else()
 		install(TARGETS openclonk
 			BUNDLE DESTINATION .
