--- a/CMakeLists.txt	2018-01-15 21:42:19.209794986 +0100
+++ b/CMakeLists.txt	2018-01-15 21:42:41.723128030 +0100
@@ -3,6 +3,7 @@
 
 project (com.github.artemanufrij.playmyvideos)
 include (GNUInstallDirs)
+list (APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR})
 set (DATADIR "${CMAKE_INSTALL_FULL_DATAROOTDIR}")
 set (PKGDATADIR "${DATADIR}/${CMAKE_PROJECT_NAME}")
 set (GETTEXT_PACKAGE ${CMAKE_PROJECT_NAME})
