--- a/trunk/src/plugins/openfileslist/openfileslistplugin.h
+++ b/trunk/src/plugins/openfileslist/openfileslistplugin.h
@@ -10,6 +10,7 @@
 #include <projectfile.h>
 
 #include <wx/dynarray.h>
+#include <functional>
 
 class wxTreeCtrl;
 class wxTreeEvent;
@@ -20,12 +21,8 @@
 struct TargetFilesData
 {
     TargetFilesData() : activeFile(nullptr) {} // ctor
-    // Functor for the std::set predicate to sort the opened editor files according to their tab order
-    struct compareLess
-    {
-        bool operator()(const ProjectFile* lhs, const ProjectFile* rhs) { return lhs->editorTabPos < rhs->editorTabPos; }
-    };
-    typedef std::set<ProjectFile*, compareLess> OpenFilesSet;
+
+    typedef std::set<ProjectFile*, std::less<ProjectFile*>> OpenFilesSet;
     ProjectFile* activeFile;
     OpenFilesSet openFiles;
 };
