diff --git a/activity/activity.info b/activity/activity.info
index 9379315..c4a901b 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,10 +1,19 @@
 [Activity]
 name = ColorDeducto
 exec = sugar-activity3 ColorDeductoActivity.ColorDeductoActivity
-bundle_id = in.seeta.ColorDeducto
+bundle_id = org.sugarlabs.ColorDeducto
 icon = activity-color-deducto
-activity_version = 7
+activity_version = 7.1
 show_launcher = yes
-license = GPLv3
-summary = a game based on logical induction
-category = game
+release_date = 2019-12-10
+metadata_license = CC0-1.0
+license = GPLv3+
+summary = A game based on logical induction
+description = Color Deducto is a learning activity to help children learn the art of inductive logic through pattern recognition.
+repository = https://github.com/sugarlabs/colordeducto
+url = https://github.com/sugarlabs/colordeducto
+category = Game
+tags = Game
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://i.imgur.com/jUFQPTx.png
