diff --git a/ColorDeductoActivity.py b/ColorDeductoActivity.py
index 3a24aa2..dc0c7e5 100644
--- a/ColorDeductoActivity.py
+++ b/ColorDeductoActivity.py
@@ -20,7 +20,6 @@ from sugar3 import profile
 from sugar3.graphics.toolbarbox import ToolbarBox
 from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.activity.widgets import StopButton
-from sugar3.graphics.objectchooser import ObjectChooser
 
 from toolbar_utils import button_factory, label_factory, separator_factory
 from utils import json_load, json_dump
@@ -32,6 +31,9 @@ from dbus.gi_service import ExportedGObject
 from sugar3.presence import presenceservice
 from sugar3.presence.tubeconn import TubeConnection
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import DesktopOpenChooser
+
 from gettext import gettext as _
 
 from game import Game, LEVELS_TRUE, LEVELS_FALSE
@@ -45,7 +47,7 @@ IFACE = SERVICE
 PATH = '/in/seeta/ColorDeductoActivity'
 
 
-class ColorDeductoActivity(activity.Activity):
+class ColorDeductoActivity(SugarCompatibleActivity):
     """ Logic puzzle game """
 
 
@@ -280,32 +282,21 @@ class ColorDeductoActivity(activity.Activity):
         ''' Read the Python code from the Journal object '''
         self._python_code = None
         try:
-            _logger.debug("opening %s " % dsobject.file_path)
-            file_handle = open(dsobject.file_path, "r")
+            _logger.debug("opening %s " % dsobject)
+            file_handle = open(dsobject, "r")
             self._python_code = file_handle.read()
             file_handle.close()
         except IOError:
-            _logger.debug("couldn't open %s" % dsobject.file_path)
+            _logger.debug("couldn't open %s" % dsobject)
 
     def _chooser(self, filter, action):
         ''' Choose an object from the datastore and take some action '''
-        chooser = None
-        try:
-            chooser = ObjectChooser(parent=self, what_filter=filter)
-        except TypeError:
-            chooser = ObjectChooser(
-                None, self,
-                Gtk.DialogFlags.MODAL | Gtk.DialogFlags.DESTROY_WITH_PARENT)
-        if chooser is not None:
-            try:
-                result = chooser.run()
-                if result == Gtk.ResponseType.ACCEPT:
-                    dsobject = chooser.get_selected_object()
-                    action(dsobject)
-                    dsobject.destroy()
-            finally:
-                chooser.destroy()
-                del chooser
+        chooser = DesktopOpenChooser(self)
+        chooser.add_filter('.py', 'Python Script')
+        chooser.add_filter('.py3', 'Python3 Script')
+        filename = chooser.get_filename()
+        if filename:
+            action(filename)
 
     # Collaboration-related methods
 
diff --git a/game.py b/game.py
index 9684c6c..37576a4 100644
--- a/game.py
+++ b/game.py
@@ -244,7 +244,7 @@ class Game():
         self._canvas.connect("draw", self.__draw_cb)
 
         self._width = Gdk.Screen.width()
-        self._height = Gdk.Screen.height() - (GRID_CELL_SIZE * 1.5)
+        self._height = Gdk.Screen.height() - (GRID_CELL_SIZE)
         self._scale = self._width / (10 * DOT_SIZE * 1.2)
         self._dot_size = int(DOT_SIZE * self._scale)
         self._space = int(self._dot_size / 5.)
