diff --git a/game.py b/game.py
index 37576a4..e7dd8dd 100644
--- a/game.py
+++ b/game.py
@@ -198,6 +198,7 @@ LEVELS_FALSE = ['def generate_pattern(self):\n\
                 0, 10, 5, 5, 3, 3))\n\
     return dot_list\n']
 
+from sugarapp.helpers import PrimaryMonitor
 
 from gi.repository import Gdk, GdkPixbuf, GObject, Gtk
 import cairo
@@ -243,9 +244,9 @@ class Game():
 
         self._canvas.connect("draw", self.__draw_cb)
 
-        self._width = Gdk.Screen.width()
-        self._height = Gdk.Screen.height() - (GRID_CELL_SIZE)
-        self._scale = self._width / (10 * DOT_SIZE * 1.2)
+        self._width = PrimaryMonitor.width()
+        self._height = PrimaryMonitor.height() - (GRID_CELL_SIZE * 1.5)
+        self._scale = self._height / (DOT_SIZE * 7)
         self._dot_size = int(DOT_SIZE * self._scale)
         self._space = int(self._dot_size / 5.)
         self.max_levels = len(LEVELS_TRUE)
