--- a/lib/configure.py	2023-01-01 18:36:03.000000000 -0500
+++ b/lib/configure.py	2023-11-29 15:18:50.451112183 -0500
@@ -477,7 +477,7 @@
     ''' The same as checkProgAlternatives, but for viewers '''
     alt_rc_entry = listAlternatives(progs, 'viewer', rc_entry)
     return checkProgAlternatives(description, progs, rc_entry,
-                                 alt_rc_entry, path, not_found = 'auto')
+                                 alt_rc_entry, path, not_found = 'xdg-open')
 
 
 def checkEditor(description, progs, rc_entry = [], path = []):
@@ -1149,10 +1149,10 @@
 \converter svgz       pdf6       "%%"    ""'''],
             path = ['', inkscape_path])
         #
-        checkProg('a SVG -> EPS converter', ['rsvg-convert -f ps -o $$o $$i', inkscape_cl + ' $$i --export-area-drawing --export-filename=$$o'],
+        checkProg('a SVG -> EPS converter', [inkscape_cl + ' $$i --export-area-drawing --export-filename=$$o'],
             rc_entry = [ r'''\converter svg        eps        "%%"    ""
 \converter svgz       eps        "%%"    ""'''],
-            path = ['', inkscape_path])
+            path = [inkscape_path])
         #
         checkProg('a SVG -> PNG converter', ['rsvg-convert -f png -o $$o $$i', inkscape_cl + ' $$i --export-filename=$$o'],
             rc_entry = [ r'''\converter svg        png        "%%"    "",
@@ -1888,7 +1888,77 @@
         removeFiles(['chkconfig.vars', 'chklatex.ltx', 'chklatex.log',
             'chklayouts.tex', 'chkmodules.tex', 'chkciteengines.tex',
             'missfont.log', 'wrap_chkconfig.ltx', 'wrap_chkconfig.log'])
+            
+            
+def additional_settings():
+	
+	__Aname="/.tmplyx"
+	
+	__Aoutput = subprocess.run(['xdg-user-dir', 'DOCUMENTS'], capture_output=True, text=True)
+	
+	if __Aoutput.returncode == 0:
+		__Apath = __Aoutput.stdout.strip()
+	else:
+		print(f"Error executing xdg-user-dir: {__Aoutput.stderr}")
+
+	try:
+		os.mkdir(f"{__Apath}{__Aname}")
+	except FileExistsError:
+		print("The directory already exists")
+    
+	__Aconfig=f'''# LyX 2.3.7 generated this file. If you want to make your own
+# modifications you should do them from inside LyX and save.
+\nFormat 24\n\n
+#
+# MISC SECTION ######################################
+#\n\n
+#
+# SCREEN & FONTS SECTION ############################
+#\n
+\\screen_font_sizes 5 7 8 9 10 12 14.4 17.26 20.74 24.88
+\\single_instance false\n
+#
+# COLOR SECTION ###################################
+#\n\n
+#
+# PRINTER SECTION ###################################
+#\n\n
+#
+# TEX SECTION #######################################
+#\n\n
+#
+# FILE SECTION ######################################
+#\n
+\\tempdir_path "{__Apath}{__Aname}"\n
+#
+# PLAIN TEXT EXPORT SECTION ##############################
+#\n\n
+#
+# SPELLCHECKER SECTION ##############################
+#\n
+\\spellchecker hunspell\n
+#
+# LANGUAGE SUPPORT SECTION ##########################
+#\n\n
+#
+# 2nd MISC SUPPORT SECTION ##########################
+#\n\n
+#
+# FORMATS SECTION ##########################
+#\n
+\\default_otf_view_format dvi3\n
+#
+# CONVERTERS SECTION ##########################
+#\n\n
+#
+# COPIERS SECTION ##########################
+#
+	'''
+	__Aruta_split=__Apath.split("/")
+	__Aruta="/"+__Aruta_split[1]+"/"+__Aruta_split[2]+"/.lyx/preferences"
 
+	with open(__Aruta, 'w') as arch:
+		arch.write(__Aconfig)
 
 if __name__ == '__main__':
     lyx_check_config = True
@@ -1984,4 +2054,5 @@
     # The return error code can be 256. Because most systems expect an error code
     # in the range 0-127, 256 can be interpretted as 'success'. Because we expect
     # a None for success, 'ret is not None' is used to exit.
+    additional_settings()
     sys.exit(ret is not None)
