--- a/lib/scripts/convertDefault.py	2023-01-01 18:36:04.000000000 -0500
+++ b/lib/scripts/convertDefault.py	2023-12-02 09:57:37.436354356 -0500
@@ -61,11 +61,11 @@
     command = 'lyxconvert'
 
 # If supported, add the -define option for pdf source formats
-if sys.argv[1] == 'pdf' and (version >= 0x060206 or gm):
+if sys.argv[1] == 'pdf':
     sopts = '-define pdf:use-cropbox=true ' + sopts
 
 # If supported, add the -flatten option for ppm target formats (see bug 4749)
-if sys.argv[3] == 'ppm' and (im and version >= 0x060305 or gm):
+if sys.argv[3] == 'ppm':
     topts = '-flatten'
 
 # print (command, sys.argv[2], sys.argv[4], file= sys.stdout)
