commit 8962dfc7880d08ee77c2f81dacdb30c9114f2204
Author: Toby Lane <mrtobylane@gmail.com>
Date:   Tue Sep 1 18:54:58 2020 +0100

    Update appdata.xml to metainfo.xml

diff --git a/CMakeLists.txt b/CorsixTH/CMakeLists.txt
index 649eabc0..a522a3c2 100644
--- a/CorsixTH/CMakeLists.txt
+++ b/CorsixTH/CMakeLists.txt
@@ -267,8 +267,8 @@ if(NOT USE_SOURCE_DATADIRS)

   if(UNIX AND NOT APPLE)
     install(FILES corsix-th.6 DESTINATION ${CMAKE_INSTALL_MANDIR}/man6)
-    install(FILES com.corsixth.CorsixTH.appdata.xml DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/metainfo)
-    install(FILES com.corsixth.CorsixTH.desktop DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/applications)
+    install(FILES com.corsixth.corsixth.metainfo.xml DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/metainfo)
+    install(FILES com.corsixth.corsixth.desktop DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/applications)
     install(FILES Original_Logo.svg DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps RENAME corsix-th.svg)
   endif()

diff --git a/CorsixTH/com.corsixth.CorsixTH.desktop b/CorsixTH/com.corsixth.corsixth.desktop
similarity index 100%
rename from CorsixTH/com.corsixth.CorsixTH.desktop
rename to CorsixTH/com.corsixth.corsixth.desktop

diff --git a/CorsixTH/com.corsixth.corsixth.appdata.xml b/CorsixTH/com.corsixth.corsixth.metainfo.xml
similarity index 97%
rename from CorsixTH/com.corsixth.corsixth.appdata.xml
rename to CorsixTH/com.corsixth.corsixth.metainfo.xml
index b252d5bc..c423c25d 100644
--- a/CorsixTH/com.corsixth.corsixth.appdata.xml
+++ b/CorsixTH/com.corsixth.corsixth.metainfo.xml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<component type="desktop">
+<component type="desktop-application">
   <id>com.corsixth.corsixth</id>
   <!-- Copyright 2019 Artem Polishchuk <ego.cordatus@gmail.com> -->
   <metadata_license>CC0</metadata_license>
@@ -23,6 +23,7 @@
   <provides>
     <binary>corsix-th</binary>
   </provides>
+  <launchable type="desktop-id">com.corsixth.corsixth.desktop</launchable>
   <releases>
     <release version="0.64" date="2020-06-16"></release>
   </releases>
