diff --git a/meson.build b/meson.build
index 9b423f3..50e5247 100755
--- a/meson.build
+++ b/meson.build
@@ -68,7 +68,7 @@ exe = executable('aerosol',
 		'Aerosol/ext/Photon/ruby/ext/rb_photon_chipmunk.c',
 	],
 	include_directories: [
-		include_directories(ruby_dir),
+		include_directories('/app/lib/ruby/1.8/x86_64-linux'),
 		include_directories('Aerosol/ext/Chipmunk/include/chipmunk'),
 		include_directories('Aerosol/ext/Chipmunk/ruby/ext'),
 		include_directories('Aerosol/ext/Photon/src'),
@@ -83,7 +83,7 @@ exe = executable('aerosol',
 		cc.find_library('OpenGL'),
 		cc.find_library('GLU'),
 		cc.find_library('openal'),
-		cc.find_library('ruby-static', dirs: join_paths(meson.current_source_dir(), ruby_dir)),
+		cc.find_library('ruby-static'),
 	],
 	c_args: [
 		'-Wno-unused-parameter',
