diff --git a/SlugChess/SlugChess.csproj b/SlugChess/SlugChess.csproj
index 64e1cbc..76660a4 100644
--- a/SlugChess/SlugChess.csproj
+++ b/SlugChess/SlugChess.csproj
@@ -14,12 +14,7 @@
     <Version>1.0.1</Version>
     <Authors>spaceslug</Authors>
     <Company>spaceslug.no</Company>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <PlatformTarget>x64</PlatformTarget>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
-    <PlatformTarget>x64</PlatformTarget>
+    <PlatformTarget>anycpu</PlatformTarget>
   </PropertyGroup>
   <ItemGroup>
      <RuntimeHostConfigurationOption Include="System.Globalization.Invariant" Value="true" />
