--- a/src/server/servermain.cpp	2014-03-22 14:31:22.000000000 +0100
+++ b/src/server/servermain.cpp	2020-01-13 20:53:23.473706499 +0100
@@ -60,7 +60,7 @@
 static bool g_run_in_foreground = false;
 static bool g_print_syslog_to_stderr = false;
 static bool g_workaround_memleaks = false;
-static std::string g_config_file = "server/server.xml";
+static std::string g_config_file = "server.xml";
 static std::string g_rules_file = "";
 
 // ...
@@ -85,7 +85,7 @@
 	process_arguments(argc, argv);
 
 	FileSystem fileSys(argv[0]);
-	fileSys.addToSearchPath("data" + fileSys.getDirSeparator() + "rules.zip");
+	fileSys.addToSearchPath("/app/share/blobby" + fileSys.getDirSeparator() + "rules.zip");
 
 	if (!g_run_in_foreground)
 	{
@@ -234,7 +234,7 @@
 					printHelp();
 					exit(1);
 				}
-				g_config_file = std::string("server/") + argv[i];
+				g_config_file = argv[i];
 				continue;
 			}
 			if (strcmp(argv[i], "--rules-file") == 0 || strcmp(argv[i], "-r") == 0)
@@ -312,7 +312,7 @@
 void setup_physfs(char* argv0)
 {
 	FileSystem& fs = FileSystem::getSingleton();
-	fs.addToSearchPath("data");
+	fs.addToSearchPath("/app/share/blobby");
 }
 
 
