diff --git i/src/main/java/mediathek/tool/GuiFunktionenProgramme.java w/src/main/java/mediathek/tool/GuiFunktionenProgramme.java
index 5f25443..6a1d737 100644
--- i/src/main/java/mediathek/tool/GuiFunktionenProgramme.java
+++ w/src/main/java/mediathek/tool/GuiFunktionenProgramme.java
@@ -77,7 +77,7 @@ public static String getPathToApplicationJar() {
         return s;
     }
 
-    private static final String PFAD_LINUX_VLC = "/usr/bin/vlc";
+    private static final String PFAD_LINUX_VLC = "/app/bin/vlc";
     private static final String PFAD_MAC_VLC = "/Applications/VLC.app/Contents/MacOS/VLC";
     private static final String PFAD_WIN = "\\VideoLAN\\VLC\\vlc.exe";
 
