From 059c1afbf390c1dde1b94a54b081e5a00e05697c Mon Sep 17 00:00:00 2001
From: JakobDev <jakobdev@gmx.de>
Date: Fri, 25 Apr 2025 21:42:14 +0200
Subject: [PATCH] Fix broken default settings

---
 gui/src/core/Settings.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gui/src/core/Settings.cpp b/gui/src/core/Settings.cpp
index 443b7de..bdf1d15 100644
--- a/gui/src/core/Settings.cpp
+++ b/gui/src/core/Settings.cpp
@@ -32,8 +32,8 @@ void Settings::initDefaultValues() {
         processTabColumnList.append(ProcessProperty::toString(property));
     }
 
-    this->set(SettingsKey::DefaultTab, processTabColumnList);
-    this->set(SettingsKey::ProcessTabColumns, QStringLiteral("Systeminfo"));
+    this->set(SettingsKey::DefaultTab, QStringLiteral("Systeminfo"));
+    this->set(SettingsKey::ProcessTabColumns, processTabColumnList);
     this->set(SettingsKey::ProcessTabUpdateInterval, 3);
     this->set(SettingsKey::ProcessActions, QVariantList());
     this->set(SettingsKey::PerformanceTabUpdateInterval, 2);
