From b8de622113e856f0cc4ae5fa1cbad279a322fd99 Mon Sep 17 00:00:00 2001
From: Akseli Lahtinen <akselmo@akselmo.dev>
Date: Tue, 5 Aug 2025 20:49:54 +0300
Subject: [PATCH] Add txt as default suffix

---
 src/ui/TodoPage.qml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ui/TodoPage.qml b/src/ui/TodoPage.qml
index 09e6be1..4217bb4 100644
--- a/src/ui/TodoPage.qml
+++ b/src/ui/TodoPage.qml
@@ -53,6 +53,7 @@ Kirigami.ScrollablePage {
         }
         fileMode: Qt.SaveFile
         modality: Qt.ApplicationModal
+        defaultSuffix: "txt"
         nameFilters: ["Text files (*.txt)"]
     }
 
-- 
GitLab

