diff --git a/src/d_iwad.cpp b/src/d_iwad.cpp
index b421b1b35..9f0b16948 100644
--- a/src/d_iwad.cpp
+++ b/src/d_iwad.cpp
@@ -683,9 +683,10 @@ int FIWadManager::IdentifyVersion (TArray<FString> &wadfiles, const char *iwad,
 					  "2. Edit your ~/Library/Preferences/" GAMENAMELOWERCASE ".ini and add the directories\n"
 					  "of your iwads to the list beneath [IWADSearch.Directories]");
 #else
-					  "1. Place one or more of these wads in ~/.config/" GAMENAMELOWERCASE "/.\n"
-					  "2. Edit your ~/.config/" GAMENAMELOWERCASE "/" GAMENAMELOWERCASE ".ini and add the directories of your\n"
-					  "iwads to the list beneath [IWADSearch.Directories]");
+					  "1. Place one or more of these wads in ~/.var/app/org.zdoom.GZDoom/.config/" GAMENAMELOWERCASE "/\n"
+					  "2. Edit your ~/.var/app/org.zdoom.GZDoom/.config/" GAMENAMELOWERCASE "/" GAMENAMELOWERCASE ".ini and\n"
+					  "     add the directories of your iwads to the list beneath [IWADSearch.Directories]\n"
+					  "3. Validate your Flatpak permissions, so that Flatpak has access to your directories with wads");
 #endif
 	}
 	int pick = 0;
