diff -ur ../puzzles-20230313.adf2a09/cmake/platforms/unix.cmake ./cmake/platforms/unix.cmake
--- ../puzzles-20230313.adf2a09/cmake/platforms/unix.cmake	2023-03-13 03:30:02.000000000 +0200
+++ ./cmake/platforms/unix.cmake	2023-03-19 23:36:58.932379628 +0200
@@ -109,10 +109,12 @@
       # CMAKE_INSTALL_BINDIR.
       install(TARGETS ${TARGET})
     endif()
-    configure_file(${CMAKE_SOURCE_DIR}/puzzle.desktop.in ${binary_name}.desktop)
+    configure_file(${CMAKE_SOURCE_DIR}/puzzle.desktop.in uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}.desktop)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/icons/${NAME}-48d24.png
-      DESTINATION share/pixmaps OPTIONAL RENAME ${binary_name}-48d24.png)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${binary_name}.desktop
+      DESTINATION share/pixmaps OPTIONAL RENAME uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}-48d24.png)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/icons/${NAME}-96d24.png
+      DESTINATION share/icons/hicolor/96x96/apps/ OPTIONAL RENAME uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}.png)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}.desktop
       DESTINATION share/applications)
   endif()
 endfunction()
diff -ur ../puzzles-20230313.adf2a09/puzzle.desktop.in ./puzzle.desktop.in
--- ../puzzles-20230313.adf2a09/puzzle.desktop.in	2023-03-13 03:30:02.000000000 +0200
+++ ./puzzle.desktop.in	2023-03-19 23:25:30.809299623 +0200
@@ -4,7 +4,7 @@
 Name=${displayname}
 Comment=${description}
 Exec=${binary_name}
-Icon=${binary_name}-48d24
+Icon=uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}
 StartupNotify=false
 Categories=Game;LogicGame;
 Terminal=false
