diff --git a/usr/share/applications/butt.desktop b/usr/share/applications/de.danielnoethen.butt.desktop
similarity index 91%
rename from usr/share/applications/butt.desktop
rename to usr/share/applications/de.danielnoethen.butt.desktop
index ea9b861..d373bf0 100644
--- a/butt/usr/share/applications/butt.desktop
+++ b/butt/usr/share/applications/de.danielnoethen.butt.desktop
@@ -8,8 +8,7 @@ Exec=butt %u
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Icon=butt
+Icon=de.danielnoethen.butt
 StartupWMClass=butt_FLTK
 Categories=AudioVideo;Audio;AudioVideoEditing;
 StartupNotify=true
-
