--- a/CMake/ComposeDesktopFile.cmake	2024-03-13 12:39:52.677976682 -0500
+++ b/CMake/ComposeDesktopFile.cmake	2024-03-13 12:47:03.292404818 -0500
@@ -7,7 +7,7 @@
     file(APPEND "${fname}" "GenericName=Previewer\n")
     file(APPEND "${fname}" "Comment=Preview all kinds of files\n")
     file(APPEND "${fname}" "Exec=previewqt %f\n")
-    file(APPEND "${fname}" "Icon=org.photoqt.PreviewQt\n")
+    file(APPEND "${fname}" "Icon=org.previewqt.PreviewQt\n")
     file(APPEND "${fname}" "Type=Application\n")
     file(APPEND "${fname}" "Terminal=false\n")
     file(APPEND "${fname}" "Categories=Graphics;Viewer;\n")
