diff --git a/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop b/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop
index ba3e41c..faa306f 100644
--- a/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop
+++ b/cmd/geteduroam-gui/resources/share/applications/app.geteduroam.Linux.desktop
@@ -5,4 +5,5 @@ Name=geteduroam
 Comment=A geteduroam client for Linux. Geteduroam simplifies the process of connecting to eduroam.
 Exec=geteduroam-gui
 Icon=app.geteduroam.geteduroam
-Terminal=false
\ No newline at end of file
+Terminal=false
+Categories=Network;Utility;
