--- a/src/DialogOptions.ui	2024-03-23 08:55:37.485097843 -0500
+++ b/src/DialogOptions.ui	2024-03-23 08:56:14.683099168 -0500
@@ -572,7 +572,7 @@
           <item row="1" column="1">
            <widget class="QLineEdit" name="txtTTY">
             <property name="text">
-             <string>/usr/bin/xterm</string>
+             <string>xterm</string>
             </property>
            </widget>
           </item>
