From 0d9f5a8b0a558bd12ffc7b235596d943db2e9142 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Wed, 1 Jan 2020 18:26:58 -0300
Subject: [PATCH 4/4] info

---
 activity/activity.info | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index 5350fdf..7a6c34b 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,12 +1,18 @@
 [Activity]
 name = Dimensions
-activity_version = 59
+activity_version = 59.1
+release_date = 2019-08-16
 license = GPLv3+;MIT;LGPLv2+
-bundle_id = org.sugarlabs.VisualMatchActivity
+metadata_license = CC0-1.0
+bundle_id = org.sugarlabs.Dimensions
 exec = sugar-activity Dimensions.Dimensions
 icon = activity-dimensions
 show_launcher = yes
-summary = A pattern-matching game.
-tags = Maths;Game
+summary = A pattern-matching game
+description = Dimensions is a pattern-matching game. The object of the game is to find sets of three cards where each attribute—color, shape, texture, and number of elements—are either the same or different.
+tags = Game
 url = https://help.sugarlabs.org/en/dimensions.html
 repository = https://github.com/sugarlabs/dimensions
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/dimensions/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/dimensions/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/dimensions/master/screenshots/en/3.png
-- 
2.23.0

