From d7bc13836cbe405f64ed7f8d2ab1accca4a67bed Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Wed, 1 Jan 2020 18:16:37 -0300
Subject: [PATCH 1/4] port

---
 Dimensions.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Dimensions.py b/Dimensions.py
index 85298cd..69e1f06 100644
--- a/Dimensions.py
+++ b/Dimensions.py
@@ -25,7 +25,6 @@ from gi.repository import Gdk
 from gi.repository import GdkPixbuf
 from gi.repository import GObject
 
-from sugar3.activity import activity
 from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.activity.widgets import StopButton
 from sugar3.graphics.toolbarbox import ToolbarBox
@@ -61,6 +60,8 @@ from constants import (DECKSIZE, PRODUCT, HASH, ROMAN, WORD, CHINESE, MAYAN,
 
 from game import Game
 
+from sugarapp.widgets import SugarCompatibleActivity
+
 MODE = 'pattern'
 
 help_palettes = {}
@@ -91,7 +92,7 @@ ROBOT_TIMER_LABELS = {15: _('15 seconds'), 30: _('30 seconds'),
 ROBOT_TIMER_DEFAULT = 15
 
 
-class Dimensions(activity.Activity):
+class Dimensions(SugarCompatibleActivity):
 
     ''' Dimension matching game '''
 
@@ -935,7 +936,7 @@ class ChatTube(ExportedGObject):
 
 
 def image_from_svg_file(filename):
-    path = os.path.join(activity.get_bundle_path(), 'icons', filename)
+    path = os.path.join(os.environ['SUGAR_BUNDLE_PATH'], 'icons', filename)
     fd = open(path)
     svg = fd.read()
     fd.close()
-- 
2.23.0

