From d63b988ffe0a2ceb5ba2aa20c8ff07e5a6f9cbc9 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Wed, 1 Jan 2020 18:16:53 -0300
Subject: [PATCH 2/4] disable-broken-modes

---
 constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/constants.py b/constants.py
index 6f33ec3..4a54906 100644
--- a/constants.py
+++ b/constants.py
@@ -32,7 +32,7 @@ COL = 3
 KEYMAP = ['1', '2', '3', 'q', 'w', 'e', 'a', 's', 'd', 'z', 'x', 'c', '8',
           '9', '0']
 
-CARD_STYLES = ['pattern', 'number', 'word', 'custom']
+CARD_STYLES = ['pattern', 'number']
 NUMBER_STYLES_O = ['roman', 'product', 'chinese', 'word', 'mayan', 'incan']
 NUMBER_STYLES_C = ['lines', 'dots', 'star', 'hash', 'dice']
 
-- 
2.23.0

