diff --git a/config_manager.c b/config_manager.c
index 128af09b..0ff5eadd 100644
--- a/config_manager.c
+++ b/config_manager.c
@@ -777,10 +777,7 @@ void DumpConfig(char *name)
 	FILE	*f;
 	char	*outfile, *newlines = "\n";
 
-	if (cfg_use_home.integer) // homedir
-		outfile = va("%s/%s/%s", com_homedir, (strcmp(com_gamedirfile, "qw") == 0 || !cfg_use_gamedir.integer) ? "" : com_gamedirfile, name);
-	else // basedir
-		outfile = va("%s/%s/configs/%s", com_basedir, (strcmp(com_gamedirfile, "qw") == 0 || !cfg_use_gamedir.integer) ? "ezquake" : com_gamedirfile, name);
+	outfile = va("%s/%s/%s", com_homedir, (strcmp(com_gamedirfile, "qw") == 0 || !cfg_use_gamedir.integer) ? "" : com_gamedirfile, name);
 
 	if (!(f	= fopen	(outfile, "w"))) {
 		FS_CreatePath(outfile);
diff --git a/fs.c b/fs.c
index 6a6cb59b..5f74fdd8 100644
--- a/fs.c
+++ b/fs.c
@@ -767,11 +767,12 @@ void FS_InitFilesystemEx( qbool guess_cwd ) {
 	}
 
 	// start up with id1 by default
-	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_basedir, "id1");
+	printf("DEBUG path: %s %s \n", com_homedir, "/id1");
+	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_homedir, "/id1");
 	FS_AddGameDirectory(tmp_path, FS_LOAD_FILE_ALL);
-	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_basedir, "ezquake");
+	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_homedir, "/ezquake");
 	FS_AddGameDirectory(tmp_path, FS_LOAD_FILE_ALL);
-	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_basedir, "qw");
+	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_homedir, "/qw");
 	FS_AddGameDirectory(tmp_path, FS_LOAD_FILE_ALL);
 	if (*com_homedir) {
 	        FS_AddHomeDirectory(com_homedir, FS_LOAD_FILE_ALL);
