diff --git a/fs.c b/fs.c
index 6a6cb59b..581f2199 100644
--- a/fs.c
+++ b/fs.c
@@ -766,6 +766,10 @@ void FS_InitFilesystemEx( qbool guess_cwd ) {
 		Com_Printf("Using home directory \"%s\"\n", com_homedir);
 	}
 
+	if (*com_homedir) {
+	        FS_AddHomeDirectory(com_homedir, FS_LOAD_FILE_ALL);
+	}
+
 	// start up with id1 by default
 	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_basedir, "id1");
 	FS_AddGameDirectory(tmp_path, FS_LOAD_FILE_ALL);
@@ -773,9 +777,7 @@ void FS_InitFilesystemEx( qbool guess_cwd ) {
 	FS_AddGameDirectory(tmp_path, FS_LOAD_FILE_ALL);
 	snprintf(&tmp_path[0], sizeof(tmp_path), "%s/%s", com_basedir, "qw");
 	FS_AddGameDirectory(tmp_path, FS_LOAD_FILE_ALL);
-	if (*com_homedir) {
-	        FS_AddHomeDirectory(com_homedir, FS_LOAD_FILE_ALL);
-	}
+
 
 	// -data <datadir>
 	// Adds datadirs similar to "-game"
