diff --git a/externals/Makefile b/externals/Makefile
index 96c184243..8f6489589 100644
--- a/externals/Makefile
+++ b/externals/Makefile
@@ -125,7 +125,7 @@ CXXFLAGS = $(CFLAGS)
 #
 # WARNING!  this MUST be all on one line because the automatic package
 # building scripts rely on it being that way.
-LIB_TARGETS = adaptive arraysize bassemu boids bsaylor chaos comport creb cxc cyclone earplug ekext ext13 extra flatgui freeverb ggee hcs iem_ambi iem_bin_ambi iemlib iemgui iemguts iem_adaptfilt iemmatrix iemnet iemxmlrpc iem_delay iem_roomsim iem_spec2 iem_tab jasch_lib loaders-hexloader loaders-libdir loaders-pdlua loaders-tclpd log mapping markex maxlib mediasettings mjlib moocow moonlib motex mrpeach net osc oscx pan pdcontainer pddp pdogg pduino plugin pmpd sigpack smlib testtools tof unauthorized vanilla vbap windowing zexy
+LIB_TARGETS = adaptive arraysize bassemu boids bsaylor chaos comport creb cxc cyclone earplug ekext ext13 extra flatgui freeverb ggee hcs iem_ambi iem_bin_ambi iemlib iemgui iemguts iem_adaptfilt iemmatrix iemnet iemxmlrpc iem_delay iem_roomsim iem_spec2 iem_tab jasch_lib loaders-hexloader loaders-libdir loaders-pdlua loaders-tclpd log mapping markex maxlib mediasettings mjlib moocow moonlib motex mrpeach net osc oscx pan pdcontainer pddp pdogg pduino plugin pmpd sigpack smlib testtools tof unauthorized vanilla vbap windowing
 
 # this is for libraries that don't compile (yet) on all platforms
 ifeq ($(OS_NAME),windows)
diff --git a/externals/Makefile b/externals/Makefile
index 96c184243..97b4394a2 100644
--- a/externals/Makefile
+++ b/externals/Makefile
@@ -139,7 +139,7 @@ else
     endif
   else
     # GNU/Linux, BSD, IRIX, etc.
-    LIB_TARGETS += gem hid pdp gem2pdp iem16 syslog
+    LIB_TARGETS += hid syslog
   endif
 endif
 
