diff --git a/src/HexEditorFrame.cpp b/src/HexEditorFrame.cpp
index 6370b76..b741f1d 100644
--- a/src/HexEditorFrame.cpp
+++ b/src/HexEditorFrame.cpp
@@ -877,7 +877,7 @@ void HexEditorFrame::OnUpdateUI(wxUpdateUIEvent& event){
 #endif
 		this->Disconnect( wxID_ANY, wxEVT_COMMAND_MENU_SELECTED, wxCommandEventHandler( HexEditorFrame::OnDevicesMenu ) );
 		#ifdef __WXGTK__
-		mbar->Enable(idProcessRAM, true);
+		//mbar->Enable(idProcessRAM, true); //Disable for Flatpak - not applicable
 		#endif
 		wxMenuItemList devMen = menuDeviceDisk->GetMenuItems();
 		for( wxMenuItemList::iterator it = devMen.begin(); it != devMen.end() ; it++ )
diff --git a/src/HexEditorGui.cpp b/src/HexEditorGui.cpp
index 112c4e0..dc66ac1 100644
--- a/src/HexEditorGui.cpp
+++ b/src/HexEditorGui.cpp
@@ -231,7 +231,7 @@ HexEditorGui::HexEditorGui( wxWindow* parent, wxWindowID id, const wxString& tit
 	devicesMenu->Append( menuDeviceDiskItem );
 	
 	wxMenuItem* menuDevicesProcessRAM;
-	menuDevicesProcessRAM = new wxMenuItem( devicesMenu, idProcessRAM, wxString( _("Open Process RAM") ) , wxEmptyString, wxITEM_NORMAL );
+	menuDevicesProcessRAM = new wxMenuItem( devicesMenu, idProcessRAM, wxString( _("Open Process RAM") ) , wxString("Not available with Flatpak"), wxITEM_NORMAL );
 	devicesMenu->Append( menuDevicesProcessRAM );
 	menuDevicesProcessRAM->Enable( false );
 	
