diff --git a/transit/world_feed/CMakeLists.txt b/transit/world_feed/CMakeLists.txt
index 83cd080230..3b87431398 100644
--- a/transit/world_feed/CMakeLists.txt
+++ b/transit/world_feed/CMakeLists.txt
@@ -25,4 +25,4 @@ target_link_libraries(${PROJECT_NAME}
 omim_add_tool_subdirectory(gtfs_converter)
 
 omim_add_test_subdirectory(world_feed_tests)
-omim_add_test_subdirectory(world_feed_integration_tests)
+#omim_add_test_subdirectory(world_feed_integration_tests)
