diff -u ./Project/QtCreator/old/QCTools.pro ./Project/QtCreator/QCTools.pro
--- ./Project/QtCreator/old/QCTools.pro	2017-08-15 13:05:18.000000000 +0200
+++ ./Project/QtCreator/QCTools.pro	2018-01-21 02:41:35.617715264 +0100
@@ -2,7 +2,7 @@
 
+CONFIG += ordered
 SUBDIRS = \
         qctools-lib/qctools-lib.pro \
-        qctools-cli/qctools-cli.pro \
         qctools-gui/qctools-gui.pro
 
 FORMS += \
