--- a/buildconfig/config_unix.py	2020-11-23 19:13:01.379167734 +0100
+++ b/buildconfig/config_unix.py	2020-11-23 19:13:07.622524262 +0100
@@ -213,7 +213,7 @@
     ])
     is_freebsd = 'FreeBSD' in platform.system()
     is_hurd = platform.system() == 'GNU'
-    if not is_freebsd and not is_hurd:
+    if False:
         porttime_dep = get_porttime_dep()
         DEPS.append(
             Dependency('PORTMIDI', 'portmidi.h', 'libportmidi.so', ['portmidi'])
