diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-01-07 12:52:40.527360183 +0100
+++ b/CMakeLists.txt	2022-01-07 12:59:29.143358661 +0100
@@ -503,7 +503,7 @@
 set_property(TARGET install_test_dll PROPERTY FOLDER "CMake")
 
 # java targets
-find_program(MVN_EXECUTABLE mvn)
+#find_program(MVN_EXECUTABLE mvn)
 find_package(Java COMPONENTS Development)
 if (MVN_EXECUTABLE AND Java_FOUND)
     if (NOT DEFINED ENV{JAVA_HOME})
Binary files a/.git/index and b/.git/index differ
