--- a/distribute.py
+++ b/distribute.py
@@ -275,7 +275,7 @@
     if os.path.exists(STEAMWORKS_SDK_PATH):
         print("Existing SDK found. Removing, and re-downloading fresh copy.")
         shutil.rmtree(STEAMWORKS_SDK_PATH)
-    with ZipFile(BytesIO(handle_request(STEAMWORKS_SDK_URL).content)) as zipobj:
+    with ZipFile('steamworks_sdk_155.zip') as zipobj:
         zipobj.extractall(STEAMWORKS_PY_PATH)

     print("Getting Steam headers...")
