diff --git a/CMakeLists.txt b/CMakeLists.txt
index 080fc5fea..928dd3265 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,9 +133,9 @@ install(FILES
     COMPONENT Devel
 )
 
-configure_file(org.kde.dolphin.FileManager1.service.in
-               ${CMAKE_CURRENT_BINARY_DIR}/org.kde.dolphin.FileManager1.service)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/org.kde.dolphin.FileManager1.service
+configure_file(org.freedesktop.FileManager1.service.in
+               ${CMAKE_CURRENT_BINARY_DIR}/org.freedesktop.FileManager1.service)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/org.freedesktop.FileManager1.service
         DESTINATION ${DBUS_SERVICES_INSTALL_DIR})
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git a/org.kde.dolphin.FileManager1.service.in b/org.freedesktop.FileManager1.service.in
similarity index 100%
rename from org.kde.dolphin.FileManager1.service.in
rename to org.freedesktop.FileManager1.service.in
diff --git a/src/main.cpp b/src/main.cpp
index 789a52996..18c6c2c78 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -104,7 +104,7 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
 
     KAboutData::setApplicationData(aboutData);
 
-    KDBusService dolphinDBusService;
+    KDBusService dolphinDBusService(KDBusService::NoExitOnFailure);
     DBusInterface interface;
 
     QCommandLineParser parser;
