From e6d923235d6f22e8270df41e168227647ddfdce8 Mon Sep 17 00:00:00 2001
From: BEN ENGLISCH <EMAILBEN145@gmail.com>
Date: Mon, 29 Apr 2019 22:27:51 -0500
Subject: [PATCH] Fix scrollkeeper check

---
 doc/user/Makefile.am | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/doc/user/Makefile.am b/doc/user/Makefile.am
index 729b295..3477fa0 100644
--- a/doc/user/Makefile.am
+++ b/doc/user/Makefile.am
@@ -9,11 +9,11 @@ noinst_SCRIPTS=$(OMF)
 
 
 $(OMF): $(OMF).in
-	if [ -x `which scrollkeeper-preinstall` ]; then \
-	    scrollkeeper-preinstall file:$(datadir)/doc/$(PACKAGE_NAME)/user/bless-manual.xml $(OMF).in $(OMF); \
-	else \
-	    cp $(OMF).in $(OMF); \
-	fi
+if WITH_SCROLLKEEPER
+	scrollkeeper-preinstall file:$(datadir)/doc/$(PACKAGE_NAME)/user/bless-manual.xml $(OMF).in $(OMF)
+else
+	cp $(OMF).in $(OMF)
+endif
 
 doc:
 	xsltproc /usr/share/xml/docbook/stylesheet/nwalsh/html/chunk.xsl bless-manual.xml
