--- src/Makefile.in.orig	2016-12-19 03:25:00.000000000 -0700
+++ src/Makefile.in	2017-02-23 10:37:33.138140753 -0700
@@ -183,6 +183,30 @@ c/%.o: $(srcdir)/c/%.c
 # 	if test -f CROSS-DPP ; then ./CROSS-DPP $< $@.c ; else ./$(DPP) $< $@.c ; fi
 # 	$(TRUE_CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -S -o $@ $@.c
 
+# GCC doesn't understand #pragma STDC FENV_ACCESS ON, so we do this instead:
+c/numbers/atan.o: c/numbers/atan.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/cos.o: c/numbers/cos.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/cosh.o: c/numbers/cosh.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/exp.o: c/numbers/exp.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/expt.o: c/numbers/expt.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/log.o: c/numbers/log.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/sin.o: c/numbers/sin.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/sinh.o: c/numbers/sinh.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/sqrt.o: c/numbers/sqrt.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/tan.o: c/numbers/tan.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+c/numbers/tanh.o: c/numbers/tanh.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -fsignaling-nans -o $@ $<
+
 # prerequisite libraries
 libeclgc.a: libeclatomic.a
 	test -d ecl/gc || mkdir ecl/gc
