--- a/src/res/translations/edb_zh_CN.ts	2024-03-22 13:13:48.000000000 -0500
+++ b/src/res/translations/edb_zh_CN.ts	2024-03-23 09:09:40.240127856 -0500
@@ -1584,7 +1584,7 @@
     </message>
     <message>
         <location filename="../../DialogOptions.ui" line="575"/>
-        <source>/usr/bin/xterm</source>
+        <source>xterm</source>
         <translation></translation>
     </message>
     <message>
