--- efax-gtk-3.2.15/efax-gtkrc.orig	2021-06-20 21:02:12.657519275 -0600
+++ efax-gtk-3.2.15/efax-gtkrc	2021-06-20 21:18:12.011887139 -0600
@@ -176,13 +176,13 @@
 # fax lists, and also to view faxes to be sent identified via the
 # "Files to fax" file selector dialog.  Specify here the command to be
 # used to view postscript files.  If none is specified, the program
-# will default to 'gv'.  Other possibilities you may want to specify
+# will default to 'xdg-open'.  Other possibilities you may want to specify
 # are 'evince' (the Gnome image viewer) or 'kghostview' (the KDE
 # postscript viewer). If using gv, you may need to use the '-media A4'
 # or '-media Letter' option to select correct paper size.  evince and
 # kghostview do this automatically
 
-PS_VIEWER: gv
+PS_VIEWER: xdg-open
 
 
 # This specifies whether a socket server is to be run to receive faxes
