--- a/setup.py	2019-07-11 08:51:54.000000000 -0400
+++ b/setup.py	2019-09-10 02:12:25.236638896 -0300
@@ -36,11 +36,10 @@
     parser = argparse.ArgumentParser()
     parser.add_argument('--root=', dest='root_path', metavar='dir', default='/')
     opts, _ = parser.parse_known_args(sys.argv[1:])
-    usr_share = os.path.join(sys.prefix, "share")
-    icons_dirname = 'pixmaps'
-    if not os.access(opts.root_path + usr_share, os.W_OK) and \
+    usr_share = "/app/share"
+    icons_dirname = 'icons/hicolor/128x128/apps'
+    if not os.access(opts.root_path + usr_share, os.W_OK) or \
        not os.access(opts.root_path, os.W_OK):
-        icons_dirname = 'icons'
         if 'XDG_DATA_HOME' in os.environ.keys():
             usr_share = os.environ['XDG_DATA_HOME']
         else:
