diff --git a/download-chromedriver.js b/download-chromedriver.js
index 4614a66..0ed8eae 100644
--- a/download-chromedriver.js
+++ b/download-chromedriver.js
@@ -6,6 +6,7 @@ var extractZip = require('extract-zip')
 var versionSegments = require('./package').version.split('.')

 electronDownload({
+  cache: '/run/build/sideka-app/chromedriver-cache',
   version: versionSegments[0] + '.' + versionSegments[1] + '.0',
   chromedriver: true,
   platform: process.env.npm_config_platform,
