From bf10efdb5dc8b218d032b0a5f6505b2241b19deb Mon Sep 17 00:00:00 2001
From: alex285 <alexis.diavatis@hotmail.com>
Date: Fri, 14 Dec 2018 18:58:14 +0200
Subject: [PATCH 1/1] force elementary theme

---
 src/MainWindow.vala | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/MainWindow.vala b/src/MainWindow.vala
index e623394..21c1d34 100644
--- a/src/MainWindow.vala
+++ b/src/MainWindow.vala
@@ -69,6 +69,8 @@ namespace Timetable {
             insert_action_group ("win", actions);
 
             var settings = AppSettings.get_default ();
+            Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+            Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
             int x = settings.window_x;
             int y = settings.window_y;
             int h = settings.window_height;
-- 
2.20.0