--- a/src/Application.vala	2019-06-10 15:43:20.733715669 +0000
+++ b/src/Application.vala	2019-06-10 15:43:21.418703307 +0000
@@ -44,6 +44,9 @@
     /* Add the application-specific icons */
     weak IconTheme default_theme = IconTheme.get_default();
     default_theme.add_resource_path( "/com/github/phase1geo/minder" );
+    
+    Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+    Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
 
     /* Create the main window */
     var appwin = new MainWindow( this, settings );