--- a/src/Application.vala	2018-08-27 11:47:19.028592715 +0200
+++ b/src/Application.vala	2018-08-27 11:48:05.652714587 +0200
@@ -154,6 +154,7 @@
         protected override void activate () {
             if (mainwindow == null) {
                 mainwindow = new MainWindow ();
+                Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
                 mainwindow.application = this;
                 Interfaces.MediaKeyListener.listen ();
                 Interfaces.SoundIndicator.listen ();
