--- a/src/widgets/Application.vala	2019-03-05 13:23:59.000000000 +0200
+++ b/src/widgets/Application.vala	2023-09-11 15:52:39.439283335 +0300
@@ -40,6 +40,9 @@
             ErrorToast.show ();
             ErrorToast.send_notification ();
 
+            Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+            Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
+
             var window = new Gtk.ApplicationWindow (this);
 
             var headerbar = new Wineglass.Headerbar ();
