--- a/src/Application.vala	2018-08-27 11:47:19.028592715 +0200
+++ b/src/Application.vala	2018-08-27 11:48:05.652714587 +0200
@@ -154,6 +154,8 @@
         protected override void activate () {
             if (main_window == null) {
                 main_window = new MainWindow (); 
+                Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+                Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
                 main_window.application = this;
                 Services.MediaKey.listen ();
                 indicator.initialize ();