From bd3005a9933294d730c5c1a6a09f14914cf12d7e Mon Sep 17 00:00:00 2001
From: alex285 <alexis.diavatis@hotmail.com>
Date: Sun, 23 Sep 2018 13:03:50 +0300
Subject: [PATCH] explicit ask for elementary Gtk and Icons

---
 src/Application.vala | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Application.vala b/src/Application.vala
index d1550a4..c76cc35 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -38,6 +38,9 @@ public class Harvey : Gtk.Application {

         var app_window = new MainWindow (this);

+        Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+        Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
+
         var window_x = settings.get_int ("window-x");
         var window_y = settings.get_int ("window-y");

--
2.19.0.rc0
