From 24e454699075538840d0b0ad51e9167bb1aa0598 Mon Sep 17 00:00:00 2001
From: Matt Harris <matth281@outlook.com>
Date: Thu, 31 May 2018 09:59:31 +0100
Subject: [PATCH] use elementary theme

---
 src/Application.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Application.vala b/src/Application.vala
index 3be4ba2..b108408 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -27,6 +27,7 @@ public class Docs : Gtk.Application {
 
     public override void activate () {
         var window = new MainWindow (this);
+        Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
 
         var quit_action = new SimpleAction ("quit", null);
         add_action (quit_action);
-- 
2.17.0

