--- a/src/Application.vala	2020-03-22 10:03:06.000000000 +0000
+++ b/src/Application.vala	2020-07-22 14:59:51.786810766 +0100
@@ -41,6 +41,8 @@
     protected override void activate () {
         database.sync_databases ();
         database.reload_app.connect (reload_application);
+        Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+        Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
 
         if (get_windows ().length () > 0) {
             main_window.present ();
