--- a/src/Application.vala	2018-10-31 20:55:17.000000000 +0100
+++ b/src/Application.vala	2018-11-01 14:40:39.569330799 +0100
@@ -107,6 +107,8 @@
                 mainwindow.present ();
                 return;
             }
+            Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+            Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");

             mainwindow = new MainWindow ();
             mainwindow.set_application (this);
