diff --git a/src/Application.vala b/src/Application.vala
index dc1f1a7..a8a5fbf 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -41,6 +41,9 @@ namespace App {
                 flags: ApplicationFlags.FLAGS_NONE
             );
 
+            Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
+            Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+
             var quit_action = new SimpleAction ("quit", null);
             add_action (quit_action);
             set_accels_for_action ("app.quit", {"<Control>q"});
