--- a/src/Application.vala	2018-10-25 22:38:30.000000000 +0200
+++ b/src/Application.vala	2018-10-31 13:59:00.027116997 +0100
@@ -49,7 +49,8 @@
                 mainwindow.present ();
                 return;
             }
-
+            Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+            Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
             mainwindow = new MainWindow ();
             mainwindow.set_application(this);
         }
