From 8fa5d0161b795795645f181f49c9e445cca918c4 Mon Sep 17 00:00:00 2001
From: Marvin Ahlgrimm <marv.ahlgrimm@gmail.com>
Date: Wed, 11 Nov 2020 16:17:54 +0100
Subject: [PATCH] Add Elementary OS style

---
 src/Application.vala | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Application.vala b/src/Application.vala
index fb0f78b..5799f22 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -112,6 +112,8 @@ namespace Spectator {
                 this.load_legacy (cs, rs);
 
                 var window = new Spectator.Window(this, rs, cs, os);
+                Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+                Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
                 this.add_window (window);
 
                 window.show_content ();
-- 
2.25.1

