From bd3742f7345746a6e32a58f35956d36e41d9c09f Mon Sep 17 00:00:00 2001
From: Matheus Fantinel <fantinel@pop-os.localdomain>
Date: Sat, 11 Jan 2020 17:11:04 -0300
Subject: [PATCH] Patch elementary theme

---
 src/Application.vala | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Application.vala b/src/Application.vala
index 71b5693..8666d70 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -29,6 +29,9 @@ namespace Moneta {
         }
 
         protected override void activate() {
+            Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
+            Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
+            
             if(get_windows().length() > 0) {
                 app_window.present();
                 return;
-- 
2.20.1

