diff --git a/src/Application.vala b/src/Application.vala
index 16ae0f6..c0e3284 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -51,6 +51,8 @@ namespace Reminduck {
         }
 
         protected override void activate() {
+            Gtk.Settings.get_default().set_property("gtk-icon-theme-name", "elementary");
+            Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
             stdout.printf("\n✔️ Activated");
             database.verify_database();
 