diff -Naur entangle-2.0/src/entangle.appdata.xml.in entangle-2.0/src/entangle.appdata.xml.in
index bbb8351..8695f82 100644
--- entangle-2.0/src/entangle.appdata.xml.in
+++ entangle-2.0/src/entangle.appdata.xml.in
@@ -38,5 +38,36 @@
   <url type="homepage">http://entangle-photo.org/</url>
   <developer_name>The Entangle Photo project</developer_name>
   <updatecontact>https://groups.google.com/forum/#!forum/entangle-devel</updatecontact>
+  <content_rating type="oars-1.1" />
+  <releases>
+    <release version="2.0" date="2018-01-13">
+      <description>
+        <ul>
+          <li>Require gobject introspection >= 1.54</li>
+          <li>Require GTK3 >= 3.22</li>
+          <li>Fix dependency on libraw</li>
+          <li>Fix variable name in photobox plugin</li>
+          <li>Document some missing keyboard shortcuts</li>
+          <li>Fix upper bound in histogram to display clipped pixel</li>
+          <li>Refresh translations</li>
+          <li>Option to highlight over exposed pixels in red</li>
+          <li>Disable noisy compiler warning</li>
+          <li>Remove use of deprecated application menu concept</li>
+          <li>Fix image redraw when changing some settings</li>
+          <li>Update mailing list address in appdaat</li>
+          <li>Add more fields to appdata content</li>
+          <li>Fix reference counting during window close</li>
+          <li>Use correct API for destroying top level windows</li>
+          <li>Fix unmounting of cameras with newer gvfs URI naming scheme</li>
+          <li>Avoid out of bounds read of property values</li>
+          <li>Fix many memory leaks</li>
+          <li>Workaround for combo boxes not displaying on Wayland</li>
+          <li>Fix race condition in building enums</li>
+          <li>Fix setting of gschema directory during startup</li>
+          <li>Set env to ensure plugins can find introspection typelib</li>
+        </ul>
+      </description>
+    </release>
+  </releases>
   <translation type="gettext">entangle</translation>
 </component>
