commit 04bb229a5ed8d225b2bdf952dbb0ef3932d581bb
Author: Hubert Figuière <hub@figuiere.net>
Date:   Wed Oct 19 14:39:03 2022 -0400

    Use xdg-open to open directiories

diff --git a/src/Standalone/savedestfolder.cpp b/src/Standalone/savedestfolder.cpp
index 4ab51a4..be11c4b 100644
--- a/src/Standalone/savedestfolder.cpp
+++ b/src/Standalone/savedestfolder.cpp
@@ -37,7 +37,14 @@ bool SaveDestFolder::open_folder(Supervisor* sv)
     if (result) {
         //同じディレクトリを保存後に何度も開いてしまうためコマンドで開く仕様に
         //上に記載されているほど使用優先度が高い
-        if(system("which nautilus > /dev/null") == 0){
+        if(system("which xdg-open > /dev/null") == 0){
+            cmd_size = strlen("xdg-open ");
+            failer = (char *)malloc(cmd_size+1);
+            if(failer){
+               memset(failer, 0, cmd_size+1);
+               memcpy(failer, "xdg-open", cmd_size);
+            }
+        }else if(system("which nautilus > /dev/null") == 0){
             cmd_size = strlen("nautilus ");
             failer = (char *)malloc(cmd_size+1);
             if(failer){
