diff --git a/Eqonomize.pro b/Eqonomize.pro
index e5b3276..de28715 100644
--- a/Eqonomize.pro
+++ b/Eqonomize.pro
@@ -48,6 +48,9 @@ unix:!equals(COMPILE_RESOURCES,"yes"):!android:!macx {
 isEmpty(MAN_DIR) {
 	MAN_DIR = $$PREFIX/share/man
 }
+isEmpty(APPDATA_DIR) {
+	APPDATA_DIR = $$PREFIX/share/metainfo
+}
 TEMPLATE = app
 TARGET = eqonomize
 INCLUDEPATH += src
@@ -171,6 +174,8 @@ unix:!equals(COMPILE_RESOURCES,"yes"):!android:!macx {
 	doc.path = $$DOCUMENTATION_DIR
 	desktop.files = data/eqonomize.desktop
 	desktop.path = $$DESKTOP_DIR
+	appdata.files = data/io.github.eqonomize.eqonomize.metainfo.xml
+	appdata.path = $$APPDATA_DIR
 	mime.files = data/eqonomize.xml
 	mime.path = $$MIME_DIR
 
@@ -377,7 +382,7 @@ unix:!equals(COMPILE_RESOURCES,"yes"):!android:!macx {
 				data/64/eqz-overtime-report.png
 	actionicons64.path = $$ICON_DIR/hicolor/64x64/actions
 
-	INSTALLS += 	target qm data doc desktop mime \ 
+	INSTALLS += 	target qm data doc desktop appdata mime \
 			appicon16 appicon22 appicon32 appicon64 appicon128 appiconsvg \
 			mimeicon16 mimeicon22 mimeicon32 mimeicon48 mimeicon64 mimeicon128 mimeiconsvg \
 			actionicons16 actionicons22 actionicons32 actionicons48 actionicons64 actioniconssvg
