diff --git a/cmake/deps.cmake b/cmake/deps.cmake
index 3e105c7..fa979f2 100644
--- a/cmake/deps.cmake
+++ b/cmake/deps.cmake
@@ -15,12 +15,7 @@ if (SONIC_LIB AND SONIC_INC)
   set(HAVE_LIBSONIC ON)
 else()
-  FetchContent_Declare(sonic-git
-    GIT_REPOSITORY https://github.com/waywardgeek/sonic.git
-    GIT_TAG fbf75c3d6d846bad3bb3d456cbc5d07d9fd8c104
-  )
-  FetchContent_MakeAvailable(sonic-git)
-  FetchContent_GetProperties(sonic-git)
+  set(sonic-git_SOURCE_DIR "${CMAKE_SOURCE_DIR}/sonic")
   add_library(sonic OBJECT ${sonic-git_SOURCE_DIR}/sonic.c)
   target_include_directories(sonic PUBLIC ${sonic-git_SOURCE_DIR})
   set(HAVE_LIBSONIC ON)
