--- a/cmake/ETLBuildPack.cmake	2019-01-05 14:54:35.000000000 -0500
+++ b/cmake/ETLBuildPack.cmake	2019-03-21 22:24:17.055858577 -0400
@@ -13,6 +13,7 @@
 
 add_custom_target(
 	pak3_pk3 ALL
+	COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/legacy/"
 	COMMAND ${CMAKE_COMMAND} -E tar c "${CMAKE_CURRENT_BINARY_DIR}/legacy/pak3_${ETL_CMAKE_VERSION_SHORT}.pk3" --format=zip ${etmain_files_pack}
 	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/etmain/
 	VERBATIM
