From ba7a8cfe30342592e277ad41837f0eb5c6e88648 Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Wed, 20 Jan 2016 17:00:19 +0000
Subject: [PATCH] Disable builddir for exiv2

It's not supported.
---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index 4cd76eb..7848532 100755
--- a/configure
+++ b/configure
@@ -19914,6 +19914,8 @@ fi
 # Configuration summary.
 # ---------------------------------------------------------------------------
 
+echo \#buildapi-variable-no-builddir >/dev/null
+
 echo ""
 echo "------------------------------------------------------------------"
 echo "-- Exiv2 $VERSION feature configuration summary"
-- 
2.5.0
