diff --git a/src/main.cpp b/src/main.cpp
index fc6caff..303c33d 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <string.h>
 #include "stub.h"
+#include <filesystem>

 const char *g_caption = "Fade2Black/OpenGL";

@@ -162,7 +163,8 @@ static void setupKeyMap() {
 	gGamepadMap[SDL_CONTROLLER_BUTTON_DPAD_RIGHT] = kKeyCodeRight;

 	// read controls mapping from file
-	FILE *fp = fopen(kControlsCfg, "rb");
+	FILE *fp = fopen((std::filesystem::path(std::getenv("XDG_DATA_HOME"))/"user"/kControlsCfg).c_str(), "rb");
+
 	if (fp) {
 		const int mappingCount = readKeyMap(fp);
 		fclose(fp);
