--- mosh-1.3.2/scripts/mosh.pl~	2019-06-19 15:55:11.717241450 +0100
+++ mosh-1.3.2/scripts/mosh.pl	2019-06-19 16:40:34.134466931 +0100
@@ -394,7 +394,14 @@
     # Non-standard shells and broken shrc files cause the ssh
     # proxy to break mysteriously.
     $ENV{ 'SHELL' } = '/bin/sh';
-    my $quoted_proxy_command = shell_quote( $0, "--family=$family" );
+    my @proxy_command;
+    if ( -f '/.flatpak-info' ) {
+      push @proxy_command, 'flatpak', 'run', 'org.mosh.mosh';
+    } else {
+      push @proxy_command, $0;
+    }
+    push @proxy_command, "--family=$family";
+    my $quoted_proxy_command = shell_quote( @proxy_command );
     push @sshopts, ( '-S', 'none', '-o', "ProxyCommand=$quoted_proxy_command --fake-proxy -- %h %p" );
   }
   my @exec_argv = ( @ssh, @sshopts, $userhost, '--', $ssh_connection . "$server " . shell_quote( @server ) );
