diff --git a/activity/activity.info b/activity/activity.info
index 57a2158..45d2252 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,10 +1,20 @@
 [Activity]
-name = falabracman
+name = Falabracman
 activity_version = 2
 host_version = 1
-bundle_id = org.laptop.community.falabracman
+release_date = 2024-06-02
+bundle_id = org.sugarlabs.Falabracman
 icon = activity
 license = GPLv3+
 exec = sugar-activity3 activity.Falabracman
+metadata_license = CC0-1.0
 show_launcher = yes
 repository = https://github.com/sugarlabs/falabracman-activity.git
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+developer_id = org.sugarlabs
+tags = Education
+url = https://github.com/sugarlabs/falabracman-activity
+summary = Sugar activity to learn words
+description =  It helps in learning some words, by collecting all the letters in the right order and by avoiding the lake otherwise you will fall into it.
+screenshots = https://i.imgur.com/z6XfKCv.png https://i.imgur.com/lmaDLVr.png https://i.imgur.com/KJod9ZO.png
