diff --git a/activity.py b/activity.py
index 693ceba..01e7b39 100644
--- a/activity.py
+++ b/activity.py
@@ -17,13 +17,17 @@ import sugargame.canvas
 
 import falabracman
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
+from sugarapp.helpers import PrimaryMonitor
+
 sys.path.append('..')  # Import sugargame package from top directory.
 
 
-class Falabracman(Activity):
+class Falabracman(SugarCompatibleActivity):
 
     def __init__(self, handle):
-        Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self.paused = False
 
@@ -41,8 +45,8 @@ class Falabracman(Activity):
 
         # Note that set_canvas implicitly calls read_file when
         # resuming from the Journal.
-        w = Gdk.Screen.width()
-        h = Gdk.Screen.height() - GRID_CELL_SIZE
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height() - GRID_CELL_SIZE
 
         self._pygamecanvas.set_size_request(w, h)
         self.set_canvas(self._pygamecanvas)
@@ -53,7 +57,7 @@ class Falabracman(Activity):
         self.set_toolbar_box(toolbar_box)
         toolbar_box.show()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
