diff --git a/fdroidserver/common.py b/fdroidserver/common.py
index 9800955f..b8c69477 100644
--- a/fdroidserver/common.py
+++ b/fdroidserver/common.py
@@ -321,6 +321,7 @@ def find_sdk_tools_cmd(cmd):
         if os.path.exists(sdk_platform_tools):
             tooldirs.append(sdk_platform_tools)
     tooldirs.append('/usr/bin')
+    tooldirs.append('/app/bin')
     for d in tooldirs:
         path = os.path.join(d, cmd)
         if os.path.isfile(path):
