diff -up feh-1.10.1/src/feh.h.orig feh-1.10.1/src/feh.h
--- feh-1.10.1/src/feh.h.orig	2010-12-29 19:42:58.249577910 -0800
+++ feh-1.10.1/src/feh.h	2010-12-29 19:44:18.840802975 -0800
@@ -80,10 +80,10 @@ CONNECTION WITH THE SOFTWARE OR THE USE 
 #define XY_IN_RECT(x, y, rx, ry, rw, rh) \
 (((x) >= (rx)) && ((y) >= (ry)) && ((x) < ((rx) + (rw))) && ((y) < ((ry) + (rh))))
 
-#define DEFAULT_FONT "yudit/11"
-#define DEFAULT_MENU_FONT "yudit/10"
-#define DEFAULT_FONT_BIG "yudit/12"
-#define DEFAULT_FONT_TITLE "yudit/14"
+#define DEFAULT_FONT "DejaVuSans/11"
+#define DEFAULT_MENU_FONT "DejaVuSans/10"
+#define DEFAULT_FONT_BIG "DejaVuSans/12"
+#define DEFAULT_FONT_TITLE "DejaVuSans/14"
 
 enum mode_type { MODE_NORMAL = 0, MODE_PAN, MODE_ZOOM, MODE_ROTATE, MODE_BLUR, MODE_NEXT
 };
diff -up feh-1.10.1/src/imlib.c.orig feh-1.10.1/src/imlib.c
--- feh-1.10.1/src/imlib.c.orig	2010-12-29 19:45:00.074336772 -0800
+++ feh-1.10.1/src/imlib.c	2010-12-29 19:45:06.425802860 -0800
@@ -69,6 +69,8 @@ void init_imlib_fonts(void)
 {
 	/* Set up the font stuff */
 	imlib_add_path_to_font_path(".");
+	imlib_add_path_to_font_path("/usr/share/fonts/dejavu");
+	imlib_add_path_to_font_path("/usr/share/fonts/dejavu-sans-fonts");
 	imlib_add_path_to_font_path(PREFIX "/share/feh/fonts");
 
 	return;
