Index: micropolis-activity-0.0.20071228/Micropolis
===================================================================
--- micropolis-activity-0.0.20071228.orig/Micropolis	2008-01-14 17:05:57.000000000 +0000
+++ micropolis-activity-0.0.20071228/Micropolis	2008-01-14 17:05:57.000000000 +0000
@@ -1,7 +1,7 @@
 #!/bin/bash
-export SIMHOME=`pwd`
+export SIMHOME=/app/share/micropolis
 echo "Starting Micropolis in $SIMHOME ... "
-res/sim $*
+cd $SIMHOME && /app/lib/micropolis/sim $*
 
 #
 # Micropolis, Unix Version.  This game was released for the Unix platform
