From d39328709056a89160373f33139862c823dcf953 Mon Sep 17 00:00:00 2001
From: dginovker <danielginovker@gmail.com>
Date: Mon, 11 Jul 2022 13:16:32 +0900
Subject: [PATCH] Patch .desktop file for use in Flatpak

---
 beeref.desktop | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/beeref.desktop b/beeref.desktop
index 74d9362..cec63d9 100644
--- a/beeref.desktop
+++ b/beeref.desktop
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Name=BeeRef
 Comment=BeeRef 0.2.0
 Terminal=false
-Exec=/home/yourname/path/to/BeeRef-0.2.0-linux
+Exec=beeref
 Type=Application
-Icon=/home/yourname/path/to/logo.png
+Icon=org.beeref.BeeRef
 Name[en_US]=BeeRef
 
 MimeType=application/x-beeref;
-- 
2.32.0

