diff --git a/filtertoolitem.py b/filtertoolitem.py
index 2b042db..1da1b06 100644
--- a/filtertoolitem.py
+++ b/filtertoolitem.py
@@ -2,6 +2,7 @@ from gi.repository import GObject
 from gi.repository import Gtk
 from gi.repository import Gdk
 from gi.repository import Pango
+from sugarapp.helpers import PrimaryMonitor
 
 from sugar3.graphics import style
 from sugar3.graphics.palette import Palette
@@ -124,8 +125,8 @@ class FilterToolItem(Gtk.ToolButton):
         menuitem_width = req2.width
         menuitem_height = req2.height
 
-        palette_width = Gdk.Screen.width() - style.GRID_CELL_SIZE
-        palette_height = Gdk.Screen.height() - style.GRID_CELL_SIZE * 3
+        palette_width = PrimaryMonitor.width() - style.GRID_CELL_SIZE
+        palette_height = PrimaryMonitor.height() - style.GRID_CELL_SIZE * 3
 
         nx = min(self._MAXIMUM_PALETTE_COLUMNS,
                  int(palette_width / menuitem_width))
diff --git a/helpbutton.py b/helpbutton.py
index 97a5342..5533580 100644
--- a/helpbutton.py
+++ b/helpbutton.py
@@ -23,6 +23,7 @@ from gettext import gettext as _
 from gi.repository import Gtk
 from gi.repository import Gdk
 from gi.repository import GObject
+from sugarapp.helpers import PrimaryMonitor
 
 from sugar3.graphics.toolbutton import ToolButton
 from sugar3.graphics.icon import Icon
@@ -42,11 +43,11 @@ class HelpButton(Gtk.ToolItem):
 
         sw = Gtk.ScrolledWindow()
         sw.set_size_request(
-            int(Gdk.Screen.width() / 2.8),
-            Gdk.Screen.height() - style.GRID_CELL_SIZE * 3)
+            int(PrimaryMonitor.width() / 2.8),
+            PrimaryMonitor.height() - style.GRID_CELL_SIZE * 3)
         sw.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
 
-        self._max_text_width = int(Gdk.Screen.width() / 3) - 600
+        self._max_text_width = int(PrimaryMonitor.width() / 3) - 600
         self._vbox = Gtk.Box()
         self._vbox.set_orientation(Gtk.Orientation.VERTICAL)
         self._vbox.set_homogeneous(False)
