diff --git a/finance.py b/finance.py
index 24022e0..bcd43b0 100644
--- a/finance.py
+++ b/finance.py
@@ -40,14 +40,15 @@ from sugar3.graphics.radiotoolbutton import RadioToolButton
 from sugar3.graphics import style
 
 from sugar3.activity.widgets import StopButton
-from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.activity import activity
-from sugar3.datastore import datastore
 from sugar3.graphics.alert import Alert
 from sugar3.graphics.icon import Icon
 from sugar3.graphics.palettemenu import PaletteMenuItem
 from sugar3.graphics.palettemenu import PaletteMenuBox
 from sugar3 import profile
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import DesktopSaveChooser, DesktopOpenChooser
+from sugarapp.widgets import ExtendedActivityToolbarButton
 
 # Import screen classes.
 import registerscreen
@@ -82,9 +83,9 @@ FOREVER = 4
 # and options. Screens are stored in a stack, with the currently
 # active screen on top.
 
-class Finance(activity.Activity):
+class Finance(SugarCompatibleActivity):
     def __init__(self, handle):
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
         self.set_title(_("Finance"))
         self.max_participants = 1
 
@@ -214,7 +215,7 @@ class Finance(activity.Activity):
 
         self.toolbar_box = ToolbarBox()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         self.toolbar_box.toolbar.insert(activity_button, 0)
         activity_button.show()
 
@@ -248,7 +249,7 @@ class Finance(activity.Activity):
         export_data.props.hide_tooltip_on_click = False
         export_data.palette_invoker.props.toggle_palette = True
         export_data.show()
-
+        self.export_data = export_data
         menu_box = PaletteMenuBox()
         export_data.props.palette.set_content(menu_box)
 
@@ -904,29 +905,11 @@ class Finance(activity.Activity):
             fd.close()
 
     def __save_image_cb(self, widget):
-        image_file = tempfile.NamedTemporaryFile(mode='w+b', suffix='.png')
-        journal_entry = datastore.create()
-        journal_entry.metadata['title'] = self.chart.title
-        journal_entry.metadata['keep'] = '0'
-        journal_entry.metadata['mime_type'] = 'image/png'
-
-        # generate the image
-        self.chart.generate_image(image_file.file, 800, 600)
-        image_file.file.close()
-        journal_entry.file_path = image_file.name
-
-        # generate the preview
-        preview_str = io.StringIO()
-        self.chart.generate_image(preview_str, activity.PREVIEW_SIZE[0],
-                                  activity.PREVIEW_SIZE[1])
-        journal_entry.metadata['preview'] = dbus.ByteArray(
-            preview_str.getvalue())
-
-        logging.error('Create %s image file', image_file.name)
-        datastore.write(journal_entry)
-        self._show_journal_alert(
-            _('Chart created'), _('Open in the Journal'),
-            journal_entry.object_id)
+        chooser = DesktopSaveChooser(self, filename=_("Untitled") + '.png')
+        filename = chooser.get_filename()
+        if filename:
+            with open(filename, 'wb') as image_file:            
+                self.chart.generate_image(image_file, 800, 600)
 
     def _show_journal_alert(self, title, msg, object_id):
         open_alert = Alert()
@@ -1027,21 +1010,12 @@ class Finance(activity.Activity):
         chart_params['chart_data'] = data
 
         logging.debug('chart_data %s', chart_params)
+        self.export_data.props.palette.popdown(immediate=True)
 
+        chooser = DesktopSaveChooser(self, filename=_("Untitled")+'.json')
+        filename = chooser.get_filename()
         # save to the journal
-        data_file = tempfile.NamedTemporaryFile(mode='w+b', suffix='.json')
-        journal_entry = datastore.create()
-        journal_entry.metadata['title'] = title
-        journal_entry.metadata['keep'] = '0'
-        journal_entry.metadata['mime_type'] = 'application/x-chart-activity'
-        journal_entry.metadata['activity'] = 'org.sugarlabs.SimpleGraph'
-
-        json.dump(chart_params, data_file.file)
-        data_file.file.close()
-        journal_entry.file_path = data_file.name
-
-        logging.debug('Create %s data file', data_file.name)
-        datastore.write(journal_entry)
-        self._show_journal_alert(
-            _('Exported data'), _('Open in the Journal'),
-            journal_entry.object_id)
+        if filename:
+            with open(filename, 'w') as w:
+                json.dump(chart_params, w)
+            logging.debug('Create %s data file'.format(filename))
