diff --git a/activity/activity.info b/activity/activity.info
index d51267a..071351b 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,21 +1,20 @@
 [Activity]
 name = Finance
-activity_version = 14
+activity_version = 14.1
 license = GPLv3+
 host_version = 1
-bundle_id = org.laptop.community.Finance
+release_date = 2018-04-12
+bundle_id = org.sugarlabs.Finance
 icon = Finance-activity
 exec = sugar-activity3 finance.Finance
-summary = Want to wisely manage money? Learn financial planning basics and track your family's finance, make your own budget and learn how to make every penny count!
+summary = Learn financial planning basics
 repository = https://github.com/sugarlabs/finance-activity
 max_participants = 1
+mime_types = application/json
 metadata_license = CC0-1.0
-description:
-    <p>Here is what you can use Finance activity for:</p>
-	<ul>
-		<li>Learn financial planning basics</li>
-		<li>Track your group's finance</li>
-		<li>Check every income and outcome</li>
-	</ul>
-    
-screenshots = https://github.com/sugarlabs/finance-activity/tree/master/screenshots/en/1.png https://github.com/sugarlabs/finance-activity/tree/master/screenshots/en/2.png https://github.com/sugarlabs/finance-activity/tree/master/screenshots/en/3.png
+tags = Education
+url = https://help.sugarlabs.org/en/finance.html
+description = Want to wisely manage money? Learn financial planning basics and track your family's finance, make your own budget and learn how to make every penny count!
+screenshots = https://raw.githubusercontent.com/sugarlabs/finance-activity/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/finance-activity/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/finance-activity/master/screenshots/en/3.png
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
