From 0a6605024de6b4aa01f4dcf21c723d7f5f034f4c Mon Sep 17 00:00:00 2001
From: Michal Rostecki <michal@kinvolk.io>
Date: Fri, 23 Dec 2016 12:18:23 +0100
Subject: [PATCH] Add install target

---
 GNUmakefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/GNUmakefile b/GNUmakefile
index 38656c8..7af0bf4 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -11,4 +11,9 @@ build:
 clean:
 	./mach clobber
 
-.PHONY: all build clean
+install:
+	$(MAKE) -f client.mk install $@
+	desktop-file-install --dir $(prefix)/share/applications org.mozilla.Firefox.desktop
+	install -D browser/branding/official/mozicon128.png $(prefix)/share/icons/hicolor/128x128/apps/org.mozilla.Firefox.png
+
+.PHONY: all build clean install
-- 
2.9.3

