From d13d06185de781cf4acff432f518ca3154096a4d Mon Sep 17 00:00:00 2001
From: Michal Rostecki <michal@kinvolk.io>
Date: Wed, 28 Dec 2016 17:13:05 +0100
Subject: [PATCH] Set /app/extra/lib/mozilla/plugins as an extension dir

---
 xpcom/io/nsAppFileLocationProvider.cpp | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/xpcom/io/nsAppFileLocationProvider.cpp b/xpcom/io/nsAppFileLocationProvider.cpp
index 27ccd56..86d712d 100644
--- a/xpcom/io/nsAppFileLocationProvider.cpp
+++ b/xpcom/io/nsAppFileLocationProvider.cpp
@@ -187,13 +187,7 @@ nsAppFileLocationProvider::GetFile(const char* aProp, bool* aPersistent,
   else if (nsCRT::strcmp(aProp, NS_SYSTEM_PLUGINS_DIR) == 0) {
 #ifdef ENABLE_SYSTEM_EXTENSION_DIRS
     static const char* const sysLPlgDir =
-#if defined(HAVE_USR_LIB64_DIR) && defined(__LP64__)
-      "/usr/lib64/mozilla/plugins";
-#elif defined(__OpenBSD__) || defined (__FreeBSD__)
-      "/usr/local/lib/mozilla/plugins";
-#else
-      "/usr/lib/mozilla/plugins";
-#endif
+      "/app/extra/lib/mozilla/plugins";
     rv = NS_NewNativeLocalFile(nsDependentCString(sysLPlgDir),
                                false, getter_AddRefs(localFile));
 #else
-- 
2.9.3

