--- a/OpenRGB.pro	2021-12-12 17:18:24.457156692 +0300
+++ b/OpenRGB.pro	2021-12-12 17:29:38.748106647 +0300
@@ -874,7 +874,7 @@
     desktop.files+=qt/OpenRGB.desktop
     pixmap.path=$$PREFIX/share/pixmaps/
     pixmap.files+=qt/OpenRGB.png
-    rules.path=/lib/udev/rules.d/
+    rules.path=$$PREFIX/lib/udev/rules.d/
     rules.files+=60-openrgb.rules
     INSTALLS += target desktop pixmap rules
 }
