diff --git a/CMakeLists.txt b/CMakeLists.txt
index 30d769c..3163241 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ configure_file (${CMAKE_SOURCE_DIR}/src/config.vala.cmake ${CMAKE_SOURCE_DIR}/sr
 
 #GSettings
 include(GSettings)
-add_schema("com.github.parnold-x.nasc.gschema.xml")
+add_schema("com.github.parnoldx.nasc.gschema.xml")
 
 set(NEEDED_PKG
     glib-2.0
@@ -110,16 +110,16 @@ target_link_libraries(nasc
 
 install (TARGETS nasc RUNTIME DESTINATION bin)
 
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/com.github.parnold-x.nasc.desktop DESTINATION share/applications)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/com.github.parnold-x.nasc.appdata.xml DESTINATION share/metainfo)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/com.github.parnoldx.nasc.desktop DESTINATION share/applications)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/com.github.parnoldx.nasc.appdata.xml DESTINATION share/metainfo)
 # install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/datasets/country.xml DESTINATION share/qalculate)
 # install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/datasets/datasetsNasc.xml DESTINATION share/qalculate)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/16/com.github.parnold-x.nasc.svg DESTINATION share/icons/hicolor/16x16/apps)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/24/com.github.parnold-x.nasc.svg DESTINATION share/icons/hicolor/24x24/apps)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/32/com.github.parnold-x.nasc.svg DESTINATION share/icons/hicolor/32x32/apps)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/48/com.github.parnold-x.nasc.svg DESTINATION share/icons/hicolor/48x48/apps)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/64/com.github.parnold-x.nasc.svg DESTINATION share/icons/hicolor/64x64/apps)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/128/com.github.parnold-x.nasc.svg DESTINATION share/icons/hicolor/128x128/apps)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/16/com.github.parnoldx.nasc.svg DESTINATION share/icons/hicolor/16x16/apps)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/24/com.github.parnoldx.nasc.svg DESTINATION share/icons/hicolor/24x24/apps)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/32/com.github.parnoldx.nasc.svg DESTINATION share/icons/hicolor/32x32/apps)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/48/com.github.parnoldx.nasc.svg DESTINATION share/icons/hicolor/48x48/apps)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/64/com.github.parnoldx.nasc.svg DESTINATION share/icons/hicolor/64x64/apps)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/icons/128/com.github.parnoldx.nasc.svg DESTINATION share/icons/hicolor/128x128/apps)
 install (FILES ${QALCULATE_LIB_PATH} DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install (DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/data/styles DESTINATION share/qalculate)
 #overwrite the standard libqalculate datasets.xml with a extended one
diff --git a/README.md b/README.md
index e9ab024..8155b86 100644
--- a/README.md
+++ b/README.md
@@ -10,7 +10,7 @@ NaSC is an app where you do maths like a normal person. It lets you type whateve
 
 
 ## Installation
-[![Get it on AppCenter](https://appcenter.elementary.io/badge.svg)](https://appcenter.elementary.io/com.github.parnold-x.nasc)﻿
+[![Get it on AppCenter](https://appcenter.elementary.io/badge.svg)](https://appcenter.elementary.io/com.github.parnoldx.nasc)﻿
 
 PPA: nasc-team/daily
 
@@ -18,7 +18,7 @@ PPA: nasc-team/daily
 ```
 sudo apt-add-repository ppa:nasc-team/daily
 sudo apt-get update
-sudo apt-get install com.github.parnold-x.nasc
+sudo apt-get install com.github.parnoldx.nasc
 ```
 
 ## Building
diff --git a/com.github.parnold-x.nasc.gschema.xml b/com.github.parnold-x.nasc.gschema.xml
index 3563982..9c13b15 100755
--- a/com.github.parnold-x.nasc.gschema.xml
+++ b/com.github.parnold-x.nasc.gschema.xml
@@ -1,5 +1,5 @@
 <schemalist>
-	<schema id="com.github.parnold-x.nasc" path="/com/github/parnold-x/nasc/" gettext-domain="nasc">
+	<schema id="com.github.parnoldx.nasc" path="/com/github/parnold-x/nasc/" gettext-domain="nasc">
 		
 		<key name="show-tutorial" type="b">
 			<default>true</default>
diff --git a/data/com.github.parnold-x.nasc.appdata.xml b/data/com.github.parnold-x.nasc.appdata.xml
index 4c0ea9d..9166ac1 100644
--- a/data/com.github.parnold-x.nasc.appdata.xml
+++ b/data/com.github.parnold-x.nasc.appdata.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2017 Peter Arnold <parnold1@gmail.com> -->
 <component type="desktop">
-  <id>com.github.parnold-x.nasc.desktop</id>
+  <id>com.github.parnoldx.nasc.desktop</id>
   <metadata_license>CC0</metadata_license>
   <project_license>GPL-3.0+</project_license>
   <name>NaSC</name>
diff --git a/data/com.github.parnold-x.nasc.desktop b/data/com.github.parnold-x.nasc.desktop
index fb91aa9..478cdb6 100755
--- a/data/com.github.parnold-x.nasc.desktop
+++ b/data/com.github.parnold-x.nasc.desktop
@@ -5,7 +5,7 @@ Comment=Do maths like a normal person
 GenericName=NaSC
 TryExec=nasc
 Exec=nasc
-Icon=com.github.parnold-x.nasc
+Icon=com.github.parnoldx.nasc
 Terminal=false
 Categories=Utility;Calculator;Education;Science;
 Actions=AboutDialog;
diff --git a/debian/changelog b/debian/changelog
index 2f62c82..fc379c3 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,4 +1,4 @@
-com.github.parnold-x.nasc (0.4) UNSTABLE; urgency=low
+com.github.parnoldx.nasc (0.4) UNSTABLE; urgency=low
 
   * update appdata.xml
   * Move to github
diff --git a/debian/control b/debian/control
index 4d44bb5..827d3ca 100644
--- a/debian/control
+++ b/debian/control
@@ -1,4 +1,4 @@
-Source: com.github.parnold-x.nasc
+Source: com.github.parnoldx.nasc
 Section: misc
 Priority: optional
 Maintainer: Peter Arnold <parnold1@gmail.com>
@@ -15,7 +15,7 @@ Build-Depends: cmake (>= 2.8),
                valac
 Standards-Version: 3.9.3
 
-Package: com.github.parnold-x.nasc
+Package: com.github.parnoldx.nasc
 Architecture: any
 Depends: ${misc:Depends}, ${shlibs:Depends}
 Pre-Depends: dpkg (>= 1.15.6)
diff --git a/src/Nasc.vala b/src/Nasc.vala
index 3190b27..b0648a5 100755
--- a/src/Nasc.vala
+++ b/src/Nasc.vala
@@ -33,9 +33,9 @@ namespace Nasc {
             build_version_info = Constants.VERSION_INFO;
 
             app_years = "2017";
-            app_icon = "com.github.parnold-x.nasc";
-            app_launcher = "com.github.parnold-x.nasc.desktop";
-            application_id = "com.github.parnold-x.nasc";
+            app_icon = "com.github.parnoldx.nasc";
+            app_launcher = "com.github.parnoldx.nasc.desktop";
+            application_id = "com.github.parnoldx.nasc";
 
             main_url = "https://github.com/parnold-x/nasc/tree/master";
             bug_url = "https://github.com/parnold-x/nasc/issues";
diff --git a/src/NascSettings.vala b/src/NascSettings.vala
index 41ee88d..2b712d9 100755
--- a/src/NascSettings.vala
+++ b/src/NascSettings.vala
@@ -37,7 +37,7 @@ public class NascSettings : Granite.Services.Settings {
     public int open_sheet { get; set; }
 
     private NascSettings () {
-        base ("com.github.parnold-x.nasc");
+        base ("com.github.parnoldx.nasc");
     }
 
     public static NascSettings get_instance () {
