diff --git a/meson.build b/meson.build
index 8ea9792..aa47829 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('imcompressor',  
-          version: '0.1.0',
+          version: '0.2.1',
     meson_version: '>= 0.50.0',
   default_options: [ 'warning_level=2',
                    ],
@@ -12,4 +12,4 @@ subdir('data')
 subdir('src')
 subdir('po')
 
-meson.add_install_script('build-aux/meson/postinstall.py')
\ No newline at end of file
+meson.add_install_script('build-aux/meson/postinstall.py')
