diff --git a/data/icons/Makefile.am b/data/icons/Makefile.am
index 14fc348ae..37eb702fd 100644
--- a/data/icons/Makefile.am
+++ b/data/icons/Makefile.am
@@ -9,11 +9,11 @@ icons_DATA = manaplus.png \
 	     manaplus.xpm
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(datarootdir)/pixmaps
-	$(INSTALL) $(srcdir)/manaplus.png $(DESTDIR)$(datarootdir)/pixmaps/manaplus.png
+	$(mkinstalldirs) $(DESTDIR)$(datarootdir)/icons/hicolor/scalable/apps
+	$(INSTALL) $(srcdir)/manaplus.svg $(DESTDIR)$(datarootdir)/icons/hicolor/scalable/apps/manaplus.svg
 
 uninstall-local:
-	rm -f $(DESTDIR)$(datarootdir)/pixmaps/manaplus.png
+	rm -f $(DESTDIR)$(datarootdir)/icons/hicolor/scalable/apps/manaplus.svg
 
 EXTRA_DIST =		\
 	$(icons_DATA) \
