diff -r -U4 colorful-2.0/Makefile.in colorful-2.0--patched/Makefile.in
--- colorful-2.0/Makefile.in	2024-05-22 18:14:54.465229707 +0200
+++ colorful-2.0--patched/Makefile.in	2024-05-22 18:14:28.722011075 +0200
@@ -37,34 +37,34 @@
 	rm -rf build/
 
 install: all
 	# Executable
-	install -m 755 -p -D "$(EXECUTABLE)" "$(DESTDIR)/usr/bin/colorful"
+	install -m 755 -p -D "$(EXECUTABLE)" "$(DESTDIR)/app/bin/colorful"
 	# Data files
-	install -m 755 -d "$(DESTDIR)/usr/share/suve/colorful"
-	cp -t "$(DESTDIR)/usr/share/suve/colorful/" -a build/gfx build/map build/sfx build/slides
+	install -m 755 -d "$(DESTDIR)/app/share/suve/colorful"
+	cp -t "$(DESTDIR)/app/share/suve/colorful/" -a build/gfx build/map build/sfx build/slides
 	# Icons
-	install -m 644 -p -D "pkg/colorful.svg"         "$(DESTDIR)/usr/share/icons/hicolor/scalable/apps/colorful.svg"
-	install -m 644 -p -D "pkg/colorful-32x32.png"   "$(DESTDIR)/usr/share/icons/hicolor/32x32/apps/colorful.png"
-	install -m 644 -p -D "pkg/colorful-64x64.png"   "$(DESTDIR)/usr/share/icons/hicolor/64x64/apps/colorful.png"
-	install -m 644 -p -D "pkg/colorful-128x128.png" "$(DESTDIR)/usr/share/icons/hicolor/128x128/apps/colorful.png"
+	install -m 644 -p -D "pkg/colorful.svg"         "$(DESTDIR)/app/share/icons/hicolor/scalable/apps/pl.suve.colorful.svg"
+	install -m 644 -p -D "pkg/colorful-32x32.png"   "$(DESTDIR)/app/share/icons/hicolor/32x32/apps/pl.suve.colorful.png"
+	install -m 644 -p -D "pkg/colorful-64x64.png"   "$(DESTDIR)/app/share/icons/hicolor/64x64/apps/pl.suve.colorful.png"
+	install -m 644 -p -D "pkg/colorful-128x128.png" "$(DESTDIR)/app/share/icons/hicolor/128x128/apps/pl.suve.colorful.png"
 	# Desktop file and AppStream metadata
-	install -m 644 -p -D "pkg/pl.suve.colorful.desktop"      "$(DESTDIR)/usr/share/applications/pl.suve.colorful.desktop"
-	install -m 644 -p -D "pkg/pl.suve.colorful.metainfo.xml" "$(DESTDIR)/usr/share/metainfo/pl.suve.colorful.metainfo.xml"
+	install -m 644 -p -D "pkg/pl.suve.colorful.desktop"      "$(DESTDIR)/app/share/applications/pl.suve.colorful.desktop"
+	install -m 644 -p -D "pkg/pl.suve.colorful.metainfo.xml" "$(DESTDIR)/app/share/metainfo/pl.suve.colorful.metainfo.xml"
 	# Man pages
-	install -m 644 -p -D "pkg/man/en.man"    "$(DESTDIR)/usr/share/man/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/de.man"    "$(DESTDIR)/usr/share/man/de/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/es.man"    "$(DESTDIR)/usr/share/man/es/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/fr.man"    "$(DESTDIR)/usr/share/man/fr/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/he.man"    "$(DESTDIR)/usr/share/man/he/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/id.man"    "$(DESTDIR)/usr/share/man/id/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/nl.man"    "$(DESTDIR)/usr/share/man/nl/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/pl.man"    "$(DESTDIR)/usr/share/man/pl/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/pt.man"    "$(DESTDIR)/usr/share/man/pt/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/pt_BR.man" "$(DESTDIR)/usr/share/man/pt_BR/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/ro.man"    "$(DESTDIR)/usr/share/man/ro/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/ru.man"    "$(DESTDIR)/usr/share/man/ru/man6/colorful.6"
-	install -m 644 -p -D "pkg/man/uk.man"    "$(DESTDIR)/usr/share/man/uk/man6/colorful.6"
+	install -m 644 -p -D "pkg/man/en.man"    "$(DESTDIR)/app/share/man/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/de.man"    "$(DESTDIR)/app/share/man/de/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/es.man"    "$(DESTDIR)/app/share/man/es/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/fr.man"    "$(DESTDIR)/app/share/man/fr/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/he.man"    "$(DESTDIR)/app/share/man/he/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/id.man"    "$(DESTDIR)/app/share/man/id/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/nl.man"    "$(DESTDIR)/app/share/man/nl/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/pl.man"    "$(DESTDIR)/app/share/man/pl/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/pt.man"    "$(DESTDIR)/app/share/man/pt/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/pt_BR.man" "$(DESTDIR)/app/share/man/pt_BR/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/ro.man"    "$(DESTDIR)/app/share/man/ro/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/ru.man"    "$(DESTDIR)/app/share/man/ru/man6/pl.suve.colorful.6"
+	install -m 644 -p -D "pkg/man/uk.man"    "$(DESTDIR)/app/share/man/uk/man6/pl.suve.colorful.6"
 
 # -- Real targets
 
 build/gfx/%.png: gfx/%.png
Only in colorful-2.0--patched: Makefile.in.orig
diff -r -U4 colorful-2.0/pkg/pl.suve.colorful.desktop colorful-2.0--patched/pkg/pl.suve.colorful.desktop
--- colorful-2.0/pkg/pl.suve.colorful.desktop	2024-05-22 18:14:54.465229707 +0200
+++ colorful-2.0--patched/pkg/pl.suve.colorful.desktop	2024-05-22 18:13:37.175573306 +0200
@@ -2,8 +2,8 @@
 Type=Application
 Name=Colorful
 Comment=Simple 2D sideview shooter
 Exec=colorful
-Icon=colorful
+Icon=pl.suve.colorful
 Terminal=false
 Categories=Game;
 StartupNotify=false
diff -r -U4 colorful-2.0/src/configfiles.pas colorful-2.0--patched/src/configfiles.pas
--- colorful-2.0/src/configfiles.pas	2022-12-16 00:56:34.000000000 +0100
+++ colorful-2.0--patched/src/configfiles.pas	2024-05-22 18:13:37.175573306 +0200
@@ -332,9 +332,9 @@
 			(*
 			 * If we are building in "systemwide" mode, data files should be found
 			 * in this pre-determined location.
 			 *)
-			DataPath:='/usr/share/suve/colorful/';
+			DataPath:='/app/share/suve/colorful/';
 		{$ELSE}
 			(*
 			 * If we're not building in "systemwide" mode, grab the path to the executable.
 			 * For "bundle" mode, go two directories up. Otherwise, do nothing.
