diff -ruN a/cmake/internal/SearchForStuff.cmake b/cmake/internal/SearchForStuff.cmake
--- a/cmake/internal/SearchForStuff.cmake	2021-08-06 00:15:49.144169413 +0700
+++ b/cmake/internal/SearchForStuff.cmake	2021-08-06 00:18:25.104200656 +0700
@@ -171,7 +171,7 @@
         LIST_TO_STRING (GTK_CFLAGS "${GTK_PKG_CFLAGS_OTHER}")
     ENDIF (GTK_PKG_FOUND)
 
-    pkg_check_modules (GDKPIXBUF_PKG gdk-pixbuf-2.0)
+    pkg_check_modules (GDKPIXBUF_PKG gdk-pixbuf)
     IF (GDKPIXBUF_PKG_FOUND)
         SET (WITH_GDKPIXBUF TRUE)
         LIST_TO_STRING (GDKPIXBUF_LINKFLAGS "${GDKPIXBUF_PKG_LDFLAGS_OTHER}")
