diff --git a/com.github.ImCompressor.json b/com.github.ImCompressor.json
index c456977..68b3557 100644
--- a/com.github.ImCompressor.json
+++ b/com.github.ImCompressor.json
@@ -52,7 +52,7 @@
             "sources": [
                 {
                     "type": "git",
-                    "url": "file:///home/hugoposnic/Documents/Dev/ImCompressor"
+                    "url": "https://github.com/Huluti/ImCompressor"
                 }
             ]
         }
diff --git a/po/LINGUAS b/po/LINGUAS
index e69de29..527e861 100644
--- a/po/LINGUAS
+++ b/po/LINGUAS
@@ -0,0 +1 @@
+fr
diff --git a/src/imcompressor.in b/src/imcompressor.in
index c969984..625d453 100755
--- a/src/imcompressor.in
+++ b/src/imcompressor.in
@@ -21,10 +21,12 @@ import os
 import sys
 import signal
 import gettext
+import locale
 
 VERSION = '@VERSION@'
 pkgdatadir = '@pkgdatadir@'
 localedir = '@localedir@'
+locale.bindtextdomain('imcompressor', localedir)
 
 sys.path.insert(1, pkgdatadir)
 signal.signal(signal.SIGINT, signal.SIG_DFL)
diff --git a/src/ui/menu.ui b/src/ui/menu.ui
index b79a259..e54520d 100644
--- a/src/ui/menu.ui
+++ b/src/ui/menu.ui
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<interface domain="imeditor">
+<interface domain="imcompressor">
   <menu id="window-menu">
     <section>
       <item>
