diff --git a/data/io.github.ImEditor.appdata.xml.in b/data/io.github.ImEditor.appdata.xml.in
index 2c8f1b8..57ef07c 100644
--- a/data/io.github.ImEditor.appdata.xml.in
+++ b/data/io.github.ImEditor.appdata.xml.in
@@ -5,7 +5,7 @@
 	<project_license>GPL-3.0-or-later</project_license>
 	<content_rating type="oars-1.1" />
 	<name>ImEditor</name>
-	<summary>Simple &amp; versatile image editor.</summary>
+	<summary>Simple &amp; versatile image editor</summary>
 	<description>
 		<p>ImEditor is simple image editor, supporting PNG, JPEG, WEBP, BMP, ICO file types.</p>
 		<p>It offers several tools to modify an image.</p>
@@ -24,4 +24,18 @@
 
 	<translation type="gettext">imeditor</translation>
 	<launchable type="desktop-id">io.github.ImEditor.desktop</launchable>
+
+	<releases>
+		<release version="0.9" date="2019-08-01">
+			<description>
+				<ul>
+					<li>Adopt meson and flatpak over snap</li>
+					<li>Better GNOME integration</li>
+					<li>Translations (add French as first additional language)</li>
+					<li>New icon</li>
+					<li>Revert continuous line (too bad)</li>
+				</ul>
+			</description>
+	​    </release>
+	</releases>
 </component>
diff --git a/data/screenshots/screen1.png b/data/screenshots/screen1.png
index 38ab9ff..461e278 100644
Binary files a/data/screenshots/screen1.png and b/data/screenshots/screen1.png differ
diff --git a/io.github.ImEditor.json b/io.github.ImEditor.json
index 11f12e2..4b0bd30 100644
--- a/io.github.ImEditor.json
+++ b/io.github.ImEditor.json
@@ -14,13 +14,6 @@
         "--talk-name=ca.desrt.dconf",
         "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
     ],
-    "build-options": {
-        "cflags": "-O2 -g",
-        "cxxflags": "-O2 -g",
-        "env": {
-            "V": "1"
-        }
-    },
     "cleanup": [
         "/include",
         "/lib/pkgconfig",
@@ -49,7 +42,6 @@
         },
         {
             "name": "imeditor",
-            "builddir": true,
             "buildsystem": "meson",
             "sources": [
                 {
