--- a/minigalaxy/paths.py	2020-03-10 23:13:14.584954584 +0200
+++ b/minigalaxy/paths.py	2020-03-11 01:06:01.208302886 +0200
@@ -9,12 +9,12 @@
 
 UI_DIR = "data/ui"
 if not os.path.exists(UI_DIR):
-    UI_DIR = "/usr/share/minigalaxy/ui"
+    UI_DIR = "/app/share/minigalaxy/ui"
 
 LOGO_IMAGE_PATH = "data/minigalaxy.png"
 if not os.path.exists(LOGO_IMAGE_PATH):
-    LOGO_IMAGE_PATH = "/usr/share/icons/hicolor/192x192/apps/minigalaxy.png"
+    LOGO_IMAGE_PATH = "/app/share/icons/hicolor/192x192/apps/com.github.sharkwouter.minigalaxy.png"
 
 LOCALE_DIR = "data/mo"
 if not os.path.exists(LOCALE_DIR):
-    LOCALE_DIR = "/usr/share/minigalaxy/translations"
+    LOCALE_DIR = "/app/minigalaxy/translations"
