diff -ruN a/configure b/configure
--- a/configure	2021-07-15 18:54:56.000000000 +0700
+++ b/configure	2021-07-22 03:28:08.289535604 +0700
@@ -14559,7 +14559,7 @@
 $as_echo_n "checking jgraphx... " >&6; }
     PACKAGE_JAR_FILE=
 
-    DEFAULT_JAR_DIR="$(pwd)/thirdparty/ $(pwd)/jar/ /usr/local/java/ /usr/local/java/jar /usr/local/share/java/ /usr/local/share/java/jar/ /usr/local/lib/java/ $(find /usr/share/java/ -maxdepth 1 -type d 2>/dev/null | sort) $(find /usr/lib64/ -maxdepth 1 -type d 2>/dev/null) $(find  /usr/lib/ -maxdepth 1 -type d 2>/dev/null) $(find /usr/share/*/lib -maxdepth 1 -type d 2>/dev/null) /opt/java/lib/"
+    DEFAULT_JAR_DIR="$(pwd)/thirdparty/ $(pwd)/jar/ /usr/local/java/ /usr/local/java/jar /app/share/java/ /app/share/java/jgraphx/ /usr/local/share/java/jar/ /usr/local/lib/java/ $(find /app/share/java/ -maxdepth 1 -type d 2>/dev/null | sort) $(find /app/share/java/jgraphx/ -maxdepth 1 -type d 2>/dev/null | sort) $(find /app/lib64/ -maxdepth 1 -type d 2>/dev/null) $(find  /app/lib/ -maxdepth 1 -type d 2>/dev/null) $(find /app/share/*/lib -maxdepth 1 -type d 2>/dev/null) /opt/java/lib/"
 
     jar_resolved="$(find $DEFAULT_JAR_DIR -maxdepth 1 \( -type f -name 'jgraphx.jar' -or -name 'libjgraphx.jar' -or -name 'libjgraphx-java.jar' -or -name 'jgraphx*.jar' \) 2>/dev/null |tr '\n' ':')."
 
