diff --git a/fluxer_desktop/electron-builder.config.cjs b/fluxer_desktop/electron-builder.config.cjs
index f8f8328c..aa5175cb 100644
--- a/fluxer_desktop/electron-builder.config.cjs
+++ b/fluxer_desktop/electron-builder.config.cjs
@@ -145,12 +145,12 @@ module.exports = {
 				arch: ['x64', 'arm64'],
 			},
 		],
-		desktop: {
+		desktop:{entry:{
 			Name: productName,
 			Comment: 'Instant messaging and VoIP application',
 			Categories: 'Network;InstantMessaging;',
 			StartupWMClass: isCanary ? 'fluxer-canary' : 'fluxer',
-		},
+		}},
 	},
 
 	deb: {
diff --git a/fluxer_desktop/scripts/build.mjs b/fluxer_desktop/scripts/build.mjs
index 4ee9ac8a..e688604a 100644
--- a/fluxer_desktop/scripts/build.mjs
+++ b/fluxer_desktop/scripts/build.mjs
@@ -83,7 +83,7 @@ async function buildMain() {
 			'process.env.NODE_ENV': JSON.stringify(isProduction ? 'production' : 'development'),
 		},
 		banner: {
-			js: `import { createRequire } from 'module'; const require = createRequire(import.meta.url);`,
+			js: ``,
 		},
 	});
 
