diff -ruN a/avogadro/core/avospglib.cpp b/avogadro/core/avospglib.cpp
--- a/avogadro/core/avospglib.cpp	2021-02-18 23:15:04.000000000 +0700
+++ b/avogadro/core/avospglib.cpp	2021-04-10 01:27:15.689680870 +0700
@@ -25,7 +25,7 @@
 #include <iostream>
 
 extern "C" {
-#include "spglib/spglib.h"
+#include "spglib.h"
 }
 
 namespace Avogadro {
diff -ruN a/cmake/FindSpglib.cmake b/cmake/FindSpglib.cmake
--- a/cmake/FindSpglib.cmake	2021-02-18 23:15:04.000000000 +0700
+++ b/cmake/FindSpglib.cmake	2021-04-10 01:27:56.490132974 +0700
@@ -6,7 +6,7 @@
 #  SPGLIB_INCLUDE_DIRS - the Spglib include directories
 #  SPGLIB_LIBRARY      - The Spglib library
 #
-find_path(SPGLIB_INCLUDE_DIR spglib/spglib.h)
+find_path(SPGLIB_INCLUDE_DIR spglib.h)
 find_library(SPGLIB_LIBRARY NAMES spglib symspg)
 
 set(SPGLIB_INCLUDE_DIRS "${SPGLIB_INCLUDE_DIR}")
