diff -ruN a/applications/utilities/mesh/generation/foamyMesh/conformalVoronoiMesh/Make/options b/applications/utilities/mesh/generation/foamyMesh/conformalVoronoiMesh/Make/options
--- a/applications/utilities/mesh/generation/foamyMesh/conformalVoronoiMesh/Make/options	2025-06-27 10:47:59.000000000 +0200
+++ b/applications/utilities/mesh/generation/foamyMesh/conformalVoronoiMesh/Make/options	2025-11-06 00:37:35.590545262 +0100
@@ -30,4 +30,5 @@
     -lmeshTools \
     -ldecompose \
     -ldynamicMesh \
-    -lsnappyHexMesh
+    -lsnappyHexMesh \
+    -lmpfr -lgmp
diff -ruN a/applications/utilities/mesh/generation/foamyMesh/foamyHexMesh/Make/options b/applications/utilities/mesh/generation/foamyMesh/foamyHexMesh/Make/options
--- a/applications/utilities/mesh/generation/foamyMesh/foamyHexMesh/Make/options	2025-06-27 10:47:59.000000000 +0200
+++ b/applications/utilities/mesh/generation/foamyMesh/foamyHexMesh/Make/options	2025-11-06 00:32:26.001589648 +0100
@@ -33,4 +33,5 @@
     -ldecompose \
     -L$(FOAM_LIBBIN)/dummy \
     -lkahipDecomp -lmetisDecomp -lptscotchDecomp -lscotchDecomp \
-    -ldynamicMesh
+    -ldynamicMesh \
+    -lmpfr -lgmp
diff -ruN a/applications/utilities/mesh/generation/foamyMesh/foamyQuadMesh/Make/options b/applications/utilities/mesh/generation/foamyMesh/foamyQuadMesh/Make/options
--- a/applications/utilities/mesh/generation/foamyMesh/foamyQuadMesh/Make/options	2025-06-27 10:47:59.000000000 +0200
+++ b/applications/utilities/mesh/generation/foamyMesh/foamyQuadMesh/Make/options	2025-11-06 00:32:00.289975775 +0100
@@ -32,4 +32,5 @@
     -ldynamicMesh \
     -ldecompositionMethods \
     -L$(FOAM_LIBBIN)/dummy \
-    -lkahipDecomp -lmetisDecomp -lptscotchDecomp -lscotchDecomp
+    -lkahipDecomp -lmetisDecomp -lptscotchDecomp -lscotchDecomp \
+    -lmpfr -lgmp
