Index: cdrtools-3.02/libhfs_iso/internal.h
===================================================================
--- cdrtools-3.02.orig/libhfs_iso/internal.h
+++ cdrtools-3.02/libhfs_iso/internal.h
@@ -26,7 +26,7 @@
 # define ERROR(code, str)	(hfs_error = (str), errno = (code))
 
 # define SIZE(type, n)		((size_t) (sizeof(type) * (n)))
-# define ALLOC(type, n)		((type *) malloc(SIZE(type, n)))
+# define ALLOC(type, n)		((type *) calloc(sizeof(type), n))
 # define ALLOCX(type, n)	((n) ? ALLOC(type, n) : (type *) 0)
 # define FREE(ptr)		{if (ptr) free((void *) ptr) ;}
 
