diff -r 64c520d897aa src/CMakeLists.txt
--- a/src/CMakeLists.txt	Wed Feb 21 11:23:35 2018 +0100
+++ b/src/CMakeLists.txt	Mon Mar 05 21:47:25 2018 +0100
@@ -15,10 +15,10 @@
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 
 if(UNIX AND NOT APPLE)
-  if (PACKAGE_ARCHITECTURE MATCHES "arm")
+  if (PACKAGE_ARCHITECTURE MATCHES "arm" OR PACKAGE_ARCHITECTURE MATCHES "aarch64")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsigned-char")
-    message("Set additional flag -fsigned-char for ARM arch")
-  endif(PACKAGE_ARCHITECTURE MATCHES "arm")
+    message("Set additional flag -fsigned-char for ${PACKAGE_ARCHITECTURE} arch")
+  endif(PACKAGE_ARCHITECTURE MATCHES "arm" OR PACKAGE_ARCHITECTURE MATCHES "aarch64")
 endif(UNIX AND NOT APPLE)
 
 if(UNIX AND NOT APPLE)
