Index: cdrtools-3.02/mkisofs/mkisofs.c
===================================================================
--- cdrtools-3.02.orig/mkisofs/mkisofs.c
+++ cdrtools-3.02/mkisofs/mkisofs.c
@@ -379,7 +379,7 @@ LOCAL	int	get_boot_table	__PR((char *opt
 #ifdef	APPLE_HYB
 #ifdef PREP_BOOT
 LOCAL	int	get_prep_boot	__PR((char *opt_arg));
-LOCAL	int	get_chrp_boot	__PR((char *opt_arg));
+LOCAL	int	get_chrp_boot	__PR((void));
 #endif
 LOCAL	int	get_bsize	__PR((char *opt_arg));
 
@@ -562,8 +562,7 @@ get_prep_boot(opt_arg)
 }
 
 LOCAL int
-get_chrp_boot(opt_arg)
-	char	*opt_arg;
+get_chrp_boot()
 {
 	if (use_chrp_boot)
 		return (1);		/* silently allow duplicates */
@@ -1426,7 +1425,7 @@ LOCAL const struct mki_option mki_option
 #ifdef PREP_BOOT
 	{{"prep-boot&", NULL, (getpargfun)get_prep_boot },
 	__("\1FILE\1PReP boot image file -- up to 4 are allowed")},
-	{{"chrp-boot&", NULL, (getpargfun)get_chrp_boot },
+	{{"chrp-boot~", NULL, (getpargfun)get_chrp_boot },
 	__("Add CHRP boot header")},
 #endif	/* PREP_BOOT */
 	{{"cap~", NULL, (getpargfun)hfs_cap },
