diff --git a/dist/linux/rigelengine.desktop b/dist/linux/rigelengine.desktop
index c17bb019..d2c20a2b 100644
--- a/dist/linux/rigelengine.desktop
+++ b/dist/linux/rigelengine.desktop
@@ -3,6 +3,6 @@ Type=Application
 Name=RigelEngine
 Comment=A modern re-implementation of the classic DOS game Duke Nukem II
 Icon=rigelengine
-Exec=RigelEngine
+Exec=run.sh
 Terminal=false
 Categories=Game
diff --git a/dist/linux/rigelengine.metainfo.xml b/dist/linux/rigelengine.metainfo.xml
index c31edf49..97058a1b 100644
--- a/dist/linux/rigelengine.metainfo.xml
+++ b/dist/linux/rigelengine.metainfo.xml
@@ -19,6 +19,10 @@
             In order to run RigelEngine, the game data from the original game is required.
             Both the shareware version and the registered version work.
         </p>
+        <p>
+            Place the game data files into a directory called 'rigelengine-game-data' inside
+            your home directory.
+        </p>
         <p>
             You can grab the shareware version from archive.org:
             https://archive.org/download/msdos_DUKE2_shareware/DUKE2.zip
diff --git a/src/game_main.cpp b/src/game_main.cpp
index f5548f41..3d7dad2e 100644
--- a/src/game_main.cpp
+++ b/src/game_main.cpp
@@ -115,7 +115,7 @@ for more info.)");
   if (!isValidGamePath(gamePath))
   {
     throw std::runtime_error(
-      "No game data (NUKEM2.CMP file) found in game path");
+      "No game data (NUKEM2.CMP file) found. Place your game data inside '~/rigelengine-game-data'.");
   }
 
   // Import original game's profile data, if our profile is still 'empty'
