diff --git a/src/misc.py b/src/misc.py
index 937d078..b789edc 100644
--- a/src/misc.py
+++ b/src/misc.py
@@ -53,7 +53,7 @@ def init(doWX = True):
             progPath = str(
                 os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
-            confPath = str(os.environ["HOME"]) + "/.trelby"
+            confPath = str(os.environ["XDG_CONFIG_HOME"]) + "/trelby"
         else:
             progPath = getPathFromRegistry()
 
diff --git a/src/util.py b/src/util.py
index 20126ac..981fece 100644
--- a/src/util.py
+++ b/src/util.py
@@ -1055,13 +1055,7 @@ def getPDFViewer():
         ]
     elif misc.isUnix:
         progs = [
-            ("/usr/local/Adobe/Acrobat7.0/bin/acroread", "-tempFile"),
-            ("acroread", "-tempFile"),
-            ("xpdf", ""),
-            ("evince", ""),
-            ("gpdf", ""),
-            ("kpdf", ""),
-            ("okular", ""),
+            ("/usr/bin/xdg-open", "")
             ]
     elif misc.isWindows:
         # get value via registry if possible, or fallback to old method.
@@ -1097,9 +1091,7 @@ def getPDFViewer():
             if fileExists(name):
                 return (name, args)
         else:
-            name = findFileInPath(name)
-            if name:
-                return (name, args)
+            return (name, args)
     return (None, None)
 
 # get a windows environment variable in its native unicode format, or None
