diff --git a/Hearts.pro b/Hearts.pro
index 16e5f3c..19120c0 100644
--- a/Hearts.pro
+++ b/Hearts.pro
@@ -25,9 +25,8 @@
 # ------------------------------------------------------------------------------------------------
 # NOTE: require a rebuild all when modifing the defines options. (QTCREATOR)
 #       or on the shell: make clean; qmake Hearts.pro; make
-DEFINES += ONLINE_PLAY USE_LIBALLEGRO5 FULL_SCREEN DEBUG
+DEFINES += ONLINE_PLAY USE_LIBALLEGRO5 FULL_SCREEN
 
-equals(QMAKE_CXX, g++) { QMAKE_LFLAGS += -no-pie }
 
 QT      += core gui svg
 
@@ -106,3 +105,6 @@ FORMS    += mainwindow.ui \
 
 RESOURCES += \
     resources.qrc
+
+target.path = /app/bin
+INSTALLS += target
