diff --git a/d1/main/CMakeLists.txt b/d1/main/CMakeLists.txt
index 09c2bed..7a19f95 100644
--- a/d1/main/CMakeLists.txt
+++ b/d1/main/CMakeLists.txt
@@ -176,4 +176,12 @@ target_include_directories(d1x-redux PUBLIC ${SDL_INCLUDE_DIR} ${PHYSFS_INCLUDE_
 target_link_libraries(d1x-redux PRIVATE 2d 3d arch_sdl iff maths mem misc texmap)
 target_link_libraries(d1x-redux PUBLIC ${OPENGL_LIBRARY} ${SDL_LIBRARY} ${SDL_MIXER_LIBRARY} ${PHYSFS_LIBRARY} ${PNG_LIBRARY} GLEW::GLEW ${EXTRA_LIBRARIES})
 target_sources(d1x-redux PRIVATE vers_id.c)
+
+if(WIN32)
 install(TARGETS d1x-redux RUNTIME DESTINATION .)
+else()
+include(GNUInstallDirs)
+install(TARGETS d1x-redux RUNTIME DESTINATION ./${CMAKE_INSTALL_BINDIR})
+install(FILES ../d1x-redux.png DESTINATION ./${CMAKE_INSTALL_DATADIR}/icons/hicolor/128x128/apps)
+install(FILES ../d1x-redux.desktop DESTINATION ./${CMAKE_INSTALL_DATADIR}/applications)
+endif()
