diff --git a/packages/Makefile b/packages/Makefile
index e9964496c..b877e20e0 100644
--- a/packages/Makefile
+++ b/packages/Makefile
@@ -10,7 +10,7 @@
 CWD := $(shell pwd)
 
 cvs_root_dir := $(shell cd $(CWD)/.. && pwd)
-DESTDIR = $(CWD)/build
+DESTDIR = /
 BUILDLAYOUT_DIR = $(CWD)
 
 
diff --git a/packages/linux_make/Makefile b/packages/linux_make/Makefile
index 91fea1a84..04ca0109b 100755
--- a/packages/linux_make/Makefile
+++ b/packages/linux_make/Makefile
@@ -5,28 +5,14 @@
 CWD := $(shell pwd)
 
 # DESTDIR needs the trailing slash
-DESTDIR = $(CWD)/build/
+DESTDIR = /
 cvs_root_dir := $(shell cd $(CWD)/../.. && pwd)
 BUILDLAYOUT_DIR = $(cvs_root_dir)/packages
 
 include $(BUILDLAYOUT_DIR)/Makefile.buildlayout
 
-# if machine has dpkg-deb, build a Debian package
-ifeq ($(shell test -x /usr/bin/dpkg-shlibdeps && echo YES),YES)
-  PACKAGE_TYPE = deb
-else
-  PACKAGE_TYPE = tarbz2
-endif
-
-# for command-line UNIX versions, you need to install Pd into the same
-# directory as $(prefix) otherwise Pd won't be able to find extra, doc, etc.
-ifeq ($(PACKAGE_TYPE),deb)
-  prefix = /usr
-  # generates the dependencies for all externals and pd itself
-  PACKAGE_DEPENDS = $(shell find $(DESTDIR) \( -name "*.pd_linux" -or -name pd-gui -or -wholename \*bin/pd \) -print0 | xargs -0 dpkg-shlibdeps -O | sed 's|shlibs:Depends=|, |' )
-else
-  prefix = /usr/local
-endif
+PACKAGE_TYPE = flatpak
+prefix = /app
 
 # to control whether the debug symbols are kept
 #STRIP = strip --strip-unneeded -R .note -R .comment
@@ -112,59 +98,44 @@ tarbz2: installer_makefile installer_readme
 	mv $(DESTDIR)$(PACKAGE_NAME) $(DESTDIR)$(prefix)
 
 
-deb: DEB_BUILD_ARCH := $(shell dpkg-architecture -qDEB_BUILD_ARCH)
-deb: DEB_PD_VERSION := $(shell echo $(PD_VERSION) | sed 's|\(.*\)-extended-\(.*\)|\1~cvs\2-1|')
-deb: $(bindir)
+flatpak: $(bindir)
 # free desktop menu, mime, and icon support
-	install -d $(DESTDIR)/usr/share/icons/hicolor/128x128/apps
+	install -d $(DESTDIR)/app/share/icons/hicolor/128x128/apps
 	install -p -m0644 $(packages_src)/linux_make/pd-extended.png \
-		$(DESTDIR)/usr/share/icons/hicolor/128x128/apps/pd-extended.png
-	install -d $(DESTDIR)/usr/share/icons/hicolor/48x48/apps/
+		$(DESTDIR)/app/share/icons/hicolor/128x128/apps/pd-extended.png
+	install -d $(DESTDIR)/app/share/icons/hicolor/48x48/apps/
 	install -p -m0644 $(packages_src)/linux_make/48x48/pd-extended.png \
-		$(DESTDIR)/usr/share/icons/hicolor/48x48/apps/
-	install -d $(DESTDIR)/usr/share/pixmaps
+		$(DESTDIR)/app/share/icons/hicolor/48x48/apps/
+	install -d $(DESTDIR)/app/share/pixmaps
 	install -p -m0644 $(packages_src)/linux_make/pd-extended.xpm \
-		$(DESTDIR)/usr/share/pixmaps/
-	install -d $(DESTDIR)/usr/share/icons/hicolor/128x128/mimetypes
+		$(DESTDIR)/app/share/pixmaps/
+	install -d $(DESTDIR)/app/share/icons/hicolor/128x128/mimetypes
 	install -p -m0644 $(packages_src)/linux_make/text-x-puredata.png \
-		$(DESTDIR)/usr/share/icons/hicolor/128x128/mimetypes
-	install -d $(DESTDIR)/usr/share/icons/hicolor/48x48/mimetypes
+		$(DESTDIR)/app/share/icons/hicolor/128x128/mimetypes
+	install -d $(DESTDIR)/app/share/icons/hicolor/48x48/mimetypes
 	install -p -m0644 $(packages_src)/linux_make/48x48/text-x-puredata.png \
-		$(DESTDIR)/usr/share/icons/hicolor/48x48/mimetypes/
-	install -d $(DESTDIR)/usr/share/applications/ 
+		$(DESTDIR)/app/share/icons/hicolor/48x48/mimetypes/
+	install -d $(DESTDIR)/app/share/applications/ 
 	install -p -m0644 $(packages_src)/linux_make/pd-extended.desktop \
-		$(DESTDIR)/usr/share/applications/
-	install -d $(DESTDIR)/usr/share/menu/
+		$(DESTDIR)/app/share/applications/
+	install -d $(DESTDIR)/app/share/menu/
 	install -p -m0644 $(packages_src)/linux_make/pd-extended.menu \
-		$(DESTDIR)/usr/share/menu/pd-extended
+		$(DESTDIR)/app/share/menu/pd-extended
 # add mime type to .pd files
-	install -d $(DESTDIR)/usr/share/mime/packages/
+	install -d $(DESTDIR)/app/share/mime/packages/
 	install -p -m0644 $(packages_src)/linux_make/pd-extended.xml \
-		$(DESTDIR)/usr/share/mime/packages/
-# files for /etc
-	install -d $(DESTDIR)/etc/bash_completion.d/
-	install -p -m0644 $(scripts_src)/bash_completion/pdextended \
-		$(DESTDIR)/etc/bash_completion.d/pd-extended
-	install -d $(DESTDIR)/etc/default
-	install -p -m0644 $(scripts_src)/debian/default/pd-extended \
-		$(DESTDIR)/etc/default
-	install -d $(DESTDIR)/etc/init.d
-	install -p -m0755 $(scripts_src)/debian/init.d/pd-extended \
-		$(DESTDIR)/etc/init.d/
+		$(DESTDIR)/app/share/mime/packages/
 # default prefs file
-	install -d $(DESTDIR)/etc/pd-extended
-	ln -s -f $(libpddir)/default.pdextended \
-		$(DESTDIR)/etc/pd-extended/default.pdextended
 	install -p $(packages_src)/linux_make/default.pdextended $(DESTDIR)$(libpddir)
 # emacs mode for .pd files
-	install -d $(DESTDIR)/usr/share/emacs/site-lisp/
+	install -d $(DESTDIR)/app/share/emacs/site-lisp/
 	install -p -m0644 $(scripts_src)/pd-mode.el \
-		$(DESTDIR)/usr/share/emacs/site-lisp/
+		$(DESTDIR)/app/share/emacs/site-lisp/
 # the 'pd' command should be handled by update-alternatives
 	mv $(DESTDIR)$(bindir)/pd $(DESTDIR)$(bindir)/pd-extended
-	mv $(DESTDIR)/usr/share/man/man1/pd.1 \
-		$(DESTDIR)/usr/share/man/man1/pd-extended.1
-	gzip $(DESTDIR)/usr/share/man/man1/pd-extended.1
+	mv $(DESTDIR)/app/share/man/man1/pd.1 \
+		$(DESTDIR)/app/share/man/man1/pd-extended.1
+	gzip $(DESTDIR)/app/share/man/man1/pd-extended.1
 # Pd-extended uses the one in /usr/lib/pd-extended
 	rm -- $(DESTDIR)$(bindir)/pd-gui.tcl
 # Pd-related scripts
@@ -173,50 +144,36 @@ deb: $(bindir)
 # delete these since they are provided by the 'puredata' package suite
 	rm -- $(DESTDIR)$(bindir)/pdsend
 	rm -- $(DESTDIR)$(bindir)/pdreceive
-	rm -- $(DESTDIR)/usr/share/man/man1/pdsend.1
-	rm -- $(DESTDIR)/usr/share/man/man1/pdreceive.1
-	rm -- $(DESTDIR)/usr/include/m_pd.h
+	rm -- $(DESTDIR)/app/share/man/man1/pdsend.1
+	rm -- $(DESTDIR)/app/share/man/man1/pdreceive.1
+	rm -- $(DESTDIR)/app/include/m_pd.h
 # Debian docs dir
-	install -d $(DESTDIR)/usr/share/doc/pd-extended/
+	install -d $(DESTDIR)/app/share/doc/pd-extended/
 	ln -s /usr/lib/pd-extended/doc/1.manual \
-		$(DESTDIR)/usr/share/doc/pd-extended/html
+		$(DESTDIR)/app/share/doc/pd-extended/html
 	ln -s /usr/lib/pd-extended/doc/manuals/Pd/License.html \
-		$(DESTDIR)/usr/share/doc/pd-extended/License.html
+		$(DESTDIR)/app/share/doc/pd-extended/License.html
 	ln -s /usr/lib/pd-extended/doc/manuals/Pd/notes.txt \
-		$(DESTDIR)/usr/share/doc/pd-extended/notes.txt
+		$(DESTDIR)/app/share/doc/pd-extended/notes.txt
 	ln -s /usr/lib/pd-extended/doc/manuals/Pd/Pd_File_Format.html \
-		$(DESTDIR)/usr/share/doc/pd-extended/Pd_File_Format.html
+		$(DESTDIR)/app/share/doc/pd-extended/Pd_File_Format.html
 	ln -s /usr/lib/pd-extended/doc/manuals/Pd/Pd-LICENSE.txt \
-		$(DESTDIR)/usr/share/doc/pd-extended/Pd-LICENSE.txt
+		$(DESTDIR)/app/share/doc/pd-extended/Pd-LICENSE.txt
 	ln -s /usr/lib/pd-extended/doc/manuals/Pd/ReadMe.html \
-		$(DESTDIR)/usr/share/doc/pd-extended/ReadMe.html
+		$(DESTDIR)/app/share/doc/pd-extended/ReadMe.html
 	ln -s /usr/lib/pd-extended/doc/manuals/Pd/Welcome.html \
-		$(DESTDIR)/usr/share/doc/pd-extended/Welcome.html
+		$(DESTDIR)/app/share/doc/pd-extended/Welcome.html
 	ln -s /usr/share/common-licenses/GPL-3 \
-		$(DESTDIR)/usr/share/doc/pd-extended/copyright
-# debian packaging
-	test -d $(DESTDIR)/DEBIAN || mkdir -p $(DESTDIR)/DEBIAN
-	cp $(packages_src)/linux_make/debian/control $(DESTDIR)/DEBIAN
-	echo -n "Installed-Size: " >> $(DESTDIR)/DEBIAN/control
-	du -sk $(DESTDIR) | sed 's|\(^[0-9][0-9]*\).*|\1|' >>  $(DESTDIR)/DEBIAN/control
-	install -p -m0755  $(packages_src)/linux_make/debian/pd-extended.prerm \
-		$(packages_src)/linux_make/debian/pd-extended.postrm \
-	 	$(packages_src)/linux_make/debian/pd-extended.postinst \
-		$(DESTDIR)/DEBIAN
-# set build architecture and version for the package
-	sed -i 's|^Version:.*|Version: $(DEB_PD_VERSION)|' \
-		$(DESTDIR)/DEBIAN/control
-	sed -i 's|^Architecture:.*|Architecture: $(DEB_BUILD_ARCH)|' \
-		$(DESTDIR)/DEBIAN/control
-# add auto-generated dependencies
-	sed -i 's!^\(Depends:.*\)!\1 $(PACKAGE_DEPENDS)!' $(DESTDIR)/DEBIAN/control
+		$(DESTDIR)/app/share/doc/pd-extended/copyright
 # sanitize permissions
-	chmod -R go-w $(DESTDIR)
-	chmod -R a+r $(DESTDIR)
-	find $(DESTDIR) -name \*.pd_linux -print0 | xargs -0 chmod a-x
-	find $(DESTDIR) -name \*.pd -print0 | xargs -0 chmod a-x
-	find $(DESTDIR) -type d -print0 | xargs -0 chmod a+x
-	fakeroot dpkg-deb --build $(DESTDIR) $(packages_src)/linux_make/$(PACKAGE_NAME).deb
+	chmod -R go-w $(DESTDIR)/app/share/doc/pd-extended $(DESTDIR)$(libpddir)
+	chmod -R a+r $(DESTDIR)/app/share/doc/pd-extended $(DESTDIR)$(libpddir)
+	find $(DESTDIR)/app/share/doc/pd-extended $(DESTDIR)$(libpddir) \
+		-name \*.pd_linux -print0 | xargs -0 chmod a-x
+	find $(DESTDIR)/app/share/doc/pd-extended $(DESTDIR)$(libpddir) \
+		-name \*.pd -print0 | xargs -0 chmod a-x
+	find $(DESTDIR)/app/share/doc/pd-extended $(DESTDIR)$(libpddir) \
+		-type d -print0 | xargs -0 chmod a+x
 
 
 INSTALLER_README_FILE = $(DESTDIR)$(prefix)/README.txt
