diff -Naur dist/linux/applications/jclicauthor.desktop dist-flatpak/linux/applications/jclicauthor.desktop
--- dist/linux/applications/jclicauthor.desktop	2020-04-15 12:48:32.737513585 +0200
+++ dist-flatpak/linux/applications/jclicauthor.desktop	2020-04-15 13:00:51.167190011 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 Type=Application
 Name=JClic Author
 Name[es]=JClic Author
@@ -30,7 +29,7 @@
 Keywords=JClic;game;education;school;multimedia;
 Exec=jclicauthor %U
 TryExec=jclicauthor
-Icon=jclicauthor
+Icon=edu.xtec.JClic.jclicauthor
 Terminal=false
 Categories=Education;
 StartupNotify=true
diff -Naur dist/linux/applications/jclic.desktop dist-flatpak/linux/applications/jclic.desktop
--- dist/linux/applications/jclic.desktop	2020-04-15 12:48:32.737513585 +0200
+++ dist-flatpak/linux/applications/jclic.desktop	2020-04-15 13:20:45.275543295 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 Type=Application
 Name=JClic Player
 Name[es]=JClic Player
@@ -30,7 +29,7 @@
 Keywords=JClic;game;education;school;
 Exec=jclic %U
 TryExec=jclic
-Icon=jclic
+Icon=edu.xtec.JClic.jclic
 Terminal=false
 Categories=Education;Game;KidsGame;
 StartupNotify=false
diff -Naur dist/linux/bin/jclic dist-flatpak/linux/bin/jclic
--- dist/linux/bin/jclic	2020-04-15 12:48:32.737513585 +0200
+++ dist-flatpak/linux/bin/jclic	2020-04-15 13:22:28.638096413 +0200
@@ -7,5 +7,5 @@
 export LANGUAGE=""
 
 # Launch JClic Player
-java -jar /usr/share/java/jclic.jar "$@"
+java -jar /app/JClic/jclic.jar "$@"
 
diff -Naur dist/linux/bin/jclicauthor dist-flatpak/linux/bin/jclicauthor
--- dist/linux/bin/jclicauthor	2020-04-15 12:48:32.737513585 +0200
+++ dist-flatpak/linux/bin/jclicauthor	2020-04-15 13:22:26.096082809 +0200
@@ -7,5 +7,5 @@
 export LANGUAGE=""
 
 # Launch JClic Author
-java -jar /usr/share/java/jclicauthor.jar "$@"
+java -jar /app/JClic/jclicauthor.jar "$@"
 
diff -Naur dist/linux/edu.xtec.JClic.metainfo.xml dist-flatpak/linux/edu.xtec.JClic.metainfo.xml
--- dist/linux/edu.xtec.JClic.metainfo.xml	2020-04-15 12:48:32.738513590 +0200
+++ dist-flatpak/linux/edu.xtec.JClic.metainfo.xml	2020-04-15 13:00:12.838997838 +0200
@@ -56,8 +56,8 @@
    <update_contact>fbusquets@xtec.cat</update_contact>
    <project_group>Projectes TAC, Catalonia Department of Education - ICT in Education Unit</project_group>
 
-   <launchable type="desktop-id">jclic.desktop</launchable>
-   <launchable type="desktop-id">jclicauthor.desktop</launchable>
+   <launchable type="desktop-id">edu.xtec.JClic.jclic.desktop</launchable>
+   <launchable type="desktop-id">edu.xtec.JClic.jclicauthor.desktop</launchable>
 
    <screenshots>
       <screenshot type="default">
