diff --git a/d1/arch/sdl/digi_mixer.c b/d1/arch/sdl/digi_mixer.c
index 9e589bf..2c7bb94 100644
--- a/d1/arch/sdl/digi_mixer.c
+++ b/d1/arch/sdl/digi_mixer.c
@@ -77,6 +77,18 @@ int digi_mixer_init()
 			"%s/usr/share/sounds/sf3/default-GM.sf3", getenv("APPDIR"));
 		Mix_SetSoundFonts(soundfonts);
 	}
+	if (!Mix_EachSoundFont(digi_mixer_check_soundfont, NULL) && getenv("XDG_DATA_DIRS"))
+	{
+		char soundfonts[PATH_MAX];
+		char *data_dir = getenv("XDG_DATA_DIRS");
+		char *data_dir_end;
+		if (!(data_dir_end = strchr(data_dir, ':')))
+			data_dir_end = data_dir + strlen(data_dir);
+		snprintf(soundfonts, sizeof(soundfonts),
+			"%.*s/sounds/sf3/default-GM.sf3",
+			(int)(data_dir_end - data_dir), data_dir);
+		Mix_SetSoundFonts(soundfonts);
+	}
 	#endif
 
 	if (Mix_OpenAudio(digi_sample_rate, MIX_OUTPUT_FORMAT, MIX_OUTPUT_CHANNELS, SOUND_BUFFER_SIZE))
