diff --git a/cmake/FilesystemResolver.cmake b/cmake/FilesystemResolver.cmake
index acef652..1bd3a15 100644
--- a/cmake/FilesystemResolver.cmake
+++ b/cmake/FilesystemResolver.cmake
@@ -12,7 +12,7 @@ else()
     set(USING_BOOST_FILESYSTEM FALSE PARENT_SCOPE)
     set(USING_BOOST_FILESYSTEM FALSE)
     message(STATUS "[FilesystemResolver]: Using standard C++ filesystem API")
-    if(APPLE OR "${CMAKE_SYSTEM}" MATCHES "Linux")
+    if(APPLE)
         set(FILESYSTEM_LIBS "stdc++fs")
     else()
         set(FILESYSTEM_LIBS)
