diff --git a/CHANGELOG.md b/CHANGELOG.md
index 56117af..1eab6a7 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,11 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## [0.2.1] - DEV
+### Changed
+- Stick back and forward buttons.
+- APP id.
+
 ## [0.2] - 2019-10-08
 ### Changed
 - Various optimizations.
diff --git a/com.github.ImCompressor.json b/com.github.huluti.ImCompressor.json
similarity index 96%
rename from com.github.ImCompressor.json
rename to com.github.huluti.ImCompressor.json
index 68b3557..36ba07a 100644
--- a/com.github.ImCompressor.json
+++ b/com.github.huluti.ImCompressor.json
@@ -1,12 +1,11 @@
 {
-    "app-id": "com.github.ImCompressor",
+    "app-id": "com.github.huluti.ImCompressor",
     "runtime": "org.gnome.Platform",
     "runtime-version": "3.34",
     "sdk": "org.gnome.Sdk",
     "command": "imcompressor",
     "finish-args": [
         "--share=ipc",
-        "--share=network",
         "--socket=fallback-x11",
         "--socket=wayland",
         "--filesystem=home"
diff --git a/data/com.github.ImCompressor.appdata.xml.in b/data/com.github.huluti.ImCompressor.appdata.xml.in
similarity index 92%
rename from data/com.github.ImCompressor.appdata.xml.in
rename to data/com.github.huluti.ImCompressor.appdata.xml.in
index bfea905..51a93d8 100644
--- a/data/com.github.ImCompressor.appdata.xml.in
+++ b/data/com.github.huluti.ImCompressor.appdata.xml.in
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
-  <id>com.github.ImCompressor.desktop</id>
+  <id>com.github.huluti.ImCompressor.desktop</id>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-3.0-or-later</project_license>
   <content_rating type="oars-1.1" />
@@ -30,7 +30,7 @@
   </screenshots>
 
   <translation type="gettext">imcompressor</translation>
-  <launchable type="desktop-id">com.github.ImCompressor.desktop</launchable>
+  <launchable type="desktop-id">com.github.huluti.ImCompressor.desktop</launchable>
 
   <releases>
     <release version="0.2" date="2019-10-08">
diff --git a/data/com.github.ImCompressor.desktop.in b/data/com.github.huluti.ImCompressor.desktop.in
similarity index 88%
rename from data/com.github.ImCompressor.desktop.in
rename to data/com.github.huluti.ImCompressor.desktop.in
index 03992d6..c4de427 100644
--- a/data/com.github.ImCompressor.desktop.in
+++ b/data/com.github.huluti.ImCompressor.desktop.in
@@ -6,7 +6,7 @@ Terminal=false
 Type=Application
 Categories=GNOME;GTK;Graphics;
 # Icon name, do not translate
-Icon=com.github.ImCompressor
+Icon=com.github.huluti.ImCompressor
 StartupWMClass=imcompressor
 MimeType=image/jpeg;image/png;
 StartupNotify=true
diff --git a/data/com.github.ImCompressor.gschema.xml b/data/com.github.huluti.ImCompressor.gschema.xml
similarity index 55%
rename from data/com.github.ImCompressor.gschema.xml
rename to data/com.github.huluti.ImCompressor.gschema.xml
index 731a5aa..1565d10 100644
--- a/data/com.github.ImCompressor.gschema.xml
+++ b/data/com.github.huluti.ImCompressor.gschema.xml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <schemalist gettext-domain="imcompressor">
-	<schema id="com.github.ImCompressor" path="/com/github/ImCompressor/">
+	<schema id="com.github.huluti.ImCompressor" path="/com/github/huluti/ImCompressor/">
 	</schema>
 </schemalist>
diff --git a/data/icons/hicolor/scalable/apps/com.github.ImCompressor.svg b/data/icons/hicolor/scalable/apps/com.github.huluti.ImCompressor.svg
similarity index 99%
rename from data/icons/hicolor/scalable/apps/com.github.ImCompressor.svg
rename to data/icons/hicolor/scalable/apps/com.github.huluti.ImCompressor.svg
index b047358..536ee11 100644
--- a/data/icons/hicolor/scalable/apps/com.github.ImCompressor.svg
+++ b/data/icons/hicolor/scalable/apps/com.github.huluti.ImCompressor.svg
@@ -15,7 +15,7 @@
    viewBox="0 0 33.866666 33.866668"
    version="1.1"
    id="svg8"
-   sodipodi:docname="com.github.ImCompressor.svg"
+   sodipodi:docname="com.github.huluti.ImCompressor.svg"
    inkscape:version="0.92.4 5da689c313, 2019-01-14">
   <defs
      id="defs2">
diff --git a/data/icons/hicolor/symbolic/apps/com.github.ImCompressor-symbolic.svg b/data/icons/hicolor/symbolic/apps/com.github.huluti.ImCompressor-symbolic.svg
similarity index 98%
rename from data/icons/hicolor/symbolic/apps/com.github.ImCompressor-symbolic.svg
rename to data/icons/hicolor/symbolic/apps/com.github.huluti.ImCompressor-symbolic.svg
index 6113596..e0f0dca 100644
--- a/data/icons/hicolor/symbolic/apps/com.github.ImCompressor-symbolic.svg
+++ b/data/icons/hicolor/symbolic/apps/com.github.huluti.ImCompressor-symbolic.svg
@@ -15,7 +15,7 @@
    version="1.1"
    id="svg930"
    inkscape:version="0.92.4 5da689c313, 2019-01-14"
-   sodipodi:docname="com.github.ImCompressor-symbolic.svg">
+   sodipodi:docname="com.github.huluti.ImCompressor-symbolic.svg">
   <defs
      id="defs924" />
   <sodipodi:namedview
diff --git a/data/meson.build b/data/meson.build
index 7634eda..160e6b6 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,6 +1,6 @@
 desktop_file = i18n.merge_file(
-  input: 'com.github.ImCompressor.desktop.in',
-  output: 'com.github.ImCompressor.desktop',
+  input: 'com.github.huluti.ImCompressor.desktop.in',
+  output: 'com.github.huluti.ImCompressor.desktop',
   type: 'desktop',
   po_dir: '../po',
   install: true,
@@ -15,8 +15,8 @@ if desktop_utils.found()
 endif
 
 appstream_file = i18n.merge_file(
-  input: 'com.github.ImCompressor.appdata.xml.in',
-  output: 'com.github.ImCompressor.appdata.xml',
+  input: 'com.github.huluti.ImCompressor.appdata.xml.in',
+  output: 'com.github.huluti.ImCompressor.appdata.xml',
   po_dir: '../po',
   install: true,
   install_dir: join_paths(get_option('datadir'), 'appdata')
@@ -29,7 +29,7 @@ if appstream_util.found()
   )
 endif
 
-install_data('com.github.ImCompressor.gschema.xml',
+install_data('com.github.huluti.ImCompressor.gschema.xml',
   install_dir: join_paths(get_option('datadir'), 'glib-2.0/schemas')
 )
 
@@ -45,15 +45,15 @@ icon_sizes = ['scalable']
 
 foreach i : icon_sizes
   install_data(
-    join_paths('icons/hicolor', i , 'apps', 'com.github.ImCompressor.svg'),
+    join_paths('icons/hicolor', i , 'apps', 'com.github.huluti.ImCompressor.svg'),
     install_dir: join_paths(get_option('prefix'), get_option('datadir'), 'icons', 'hicolor', i , 'apps'),
-    rename: '@0@.svg'.format('com.github.ImCompressor')
+    rename: '@0@.svg'.format('com.github.huluti.ImCompressor')
   )
 endforeach
 
 # Installing the symbolic icon
 install_data(
-    join_paths('icons/hicolor/symbolic/apps', 'com.github.ImCompressor-symbolic.svg'),
+    join_paths('icons/hicolor/symbolic/apps', 'com.github.huluti.ImCompressor-symbolic.svg'),
     install_dir: join_paths(get_option('prefix'), get_option('datadir'), 'icons/hicolor/symbolic/apps'),
-    rename: '@0@-symbolic.svg'.format('com.github.ImCompressor')
+    rename: '@0@-symbolic.svg'.format('com.github.huluti.ImCompressor')
 )
diff --git a/po/POTFILES b/po/POTFILES
index c42d77d..79d8c36 100644
--- a/po/POTFILES
+++ b/po/POTFILES
@@ -1,6 +1,6 @@
-data/com.github.ImCompressor.desktop.in
-data/com.github.ImCompressor.appdata.xml.in
-data/com.github.ImCompressor.gschema.xml
+data/com.github.huluti.ImCompressor.desktop.in
+data/com.github.huluti.ImCompressor.appdata.xml.in
+data/com.github.huluti.ImCompressor.gschema.xml
 src/ui/window.ui
 src/ui/menu.ui
 src/main.py
diff --git a/po/fr.po b/po/fr.po
index 4222848..2619620 100644
--- a/po/fr.po
+++ b/po/fr.po
@@ -18,25 +18,25 @@ msgstr ""
 "X-Generator: Poedit 2.2.3\n"
 "Plural-Forms: nplurals=2; plural=(n > 1);\n"
 
-#: data/com.github.ImCompressor.desktop.in:3
-#: data/com.github.ImCompressor.appdata.xml.in:7
+#: data/com.github.huluti.ImCompressor.desktop.in:3
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:7
 msgid "ImCompressor"
 msgstr ""
 
-#: data/com.github.ImCompressor.desktop.in:5
+#: data/com.github.huluti.ImCompressor.desktop.in:5
 msgid "Simple & lossless image compressor."
 msgstr "Compresseur d'image simple et sans-perte."
 
 #. Icon name, do not translate
-#: data/com.github.ImCompressor.desktop.in:10
-msgid "com.github.ImCompressor"
+#: data/com.github.huluti.ImCompressor.desktop.in:10
+msgid "com.github.huluti.ImCompressor"
 msgstr ""
 
-#: data/com.github.ImCompressor.appdata.xml.in:8
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:8
 msgid "Simple &amp; lossless image compressor"
 msgstr "Compresseur d'image simple &amp; sans-perte"
 
-#: data/com.github.ImCompressor.appdata.xml.in:10
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:10
 msgid ""
 "ImCompressor is a lossless image compressor, supporting PNG and JPEG file "
 "types."
@@ -44,7 +44,7 @@ msgstr ""
 "ImCompressor est un compresseur d'image sans-perte qui supporte les "
 "formats PNG et JPG."
 
-#: data/com.github.ImCompressor.appdata.xml.in:14
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:14
 msgid "Hugo Posnic"
 msgstr ""
 
diff --git a/po/imcompressor.pot b/po/imcompressor.pot
index 62a5222..a1f3348 100644
--- a/po/imcompressor.pot
+++ b/po/imcompressor.pot
@@ -17,31 +17,31 @@ msgstr ""
 "Content-Type: text/plain; charset=CHARSET\n"
 "Content-Transfer-Encoding: 8bit\n"
 
-#: data/com.github.ImCompressor.desktop.in:3
-#: data/com.github.ImCompressor.appdata.xml.in:7
+#: data/com.github.huluti.ImCompressor.desktop.in:3
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:7
 msgid "ImCompressor"
 msgstr ""
 
-#: data/com.github.ImCompressor.desktop.in:5
+#: data/com.github.huluti.ImCompressor.desktop.in:5
 msgid "Simple & lossless image compressor."
 msgstr ""
 
 #. Icon name, do not translate
-#: data/com.github.ImCompressor.desktop.in:10
-msgid "com.github.ImCompressor"
+#: data/com.github.huluti.ImCompressor.desktop.in:10
+msgid "com.github.huluti.ImCompressor"
 msgstr ""
 
-#: data/com.github.ImCompressor.appdata.xml.in:8
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:8
 msgid "Simple &amp; lossless image compressor"
 msgstr ""
 
-#: data/com.github.ImCompressor.appdata.xml.in:10
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:10
 msgid ""
 "ImCompressor is a lossless image compressor, supporting PNG and JPEG file "
 "types."
 msgstr ""
 
-#: data/com.github.ImCompressor.appdata.xml.in:14
+#: data/com.github.huluti.ImCompressor.appdata.xml.in:14
 msgid "Hugo Posnic"
 msgstr ""
 
diff --git a/src/imcompressor.gresource.xml b/src/imcompressor.gresource.xml
index dfdbd69..e3ad265 100644
--- a/src/imcompressor.gresource.xml
+++ b/src/imcompressor.gresource.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <gresources>
-  <gresource prefix="/com/github/ImCompressor">
+  <gresource prefix="/com/github/huluti/ImCompressor">
     <file>ui/window.ui</file>
     <file>ui/menu.ui</file>
   </gresource>
diff --git a/src/main.py b/src/main.py
index 5969763..e767bb8 100644
--- a/src/main.py
+++ b/src/main.py
@@ -25,7 +25,7 @@ from gi.repository import Gtk, Gio
 from .window import ImCompressorWindow
 
 
-APP_ID = 'com.github.ImCompressor'
+APP_ID = 'com.github.huluti.ImCompressor'
 
 
 class Application(Gtk.Application):
diff --git a/src/ui/window.ui b/src/ui/window.ui
index 4e26f74..72a7c98 100644
--- a/src/ui/window.ui
+++ b/src/ui/window.ui
@@ -12,11 +12,6 @@
     <property name="can_focus">False</property>
     <property name="stock">gtk-open</property>
   </object>
-  <object class="GtkImage" id="forward_image">
-    <property name="visible">True</property>
-    <property name="can_focus">False</property>
-    <property name="icon_name">go-next-symbolic</property>
-  </object>
   <template class="ImCompressorWindow" parent="GtkApplicationWindow">
     <property name="can_focus">False</property>
     <property name="default_width">600</property>
@@ -29,31 +24,46 @@
         <property name="subtitle" translatable="yes">Simple &amp; lossless image compressor</property>
         <property name="show_close_button">True</property>
         <child>
-          <object class="GtkButton" id="back_button">
+          <object class="GtkButtonBox">
             <property name="visible">True</property>
-            <property name="sensitive">False</property>
-            <property name="can_focus">True</property>
-            <property name="receives_default">True</property>
-            <property name="action_name">win.back</property>
-            <property name="image">back_image</property>
-            <property name="always_show_image">True</property>
-          </object>
-          <packing>
-            <property name="position">1</property>
-          </packing>
-        </child>
-        <child>
-          <object class="GtkButton" id="forward_button">
-            <property name="visible">True</property>
-            <property name="sensitive">False</property>
-            <property name="can_focus">True</property>
-            <property name="receives_default">True</property>
-            <property name="action_name">win.forward</property>
-            <property name="image">forward_image</property>
-            <property name="always_show_image">True</property>
+            <property name="can_focus">False</property>
+            <property name="homogeneous">True</property>
+            <property name="layout_style">expand</property>
+            <child>
+              <object class="GtkButton" id="back_button">
+                <property name="visible">True</property>
+                <property name="sensitive">False</property>
+                <property name="can_focus">True</property>
+                <property name="receives_default">True</property>
+                <property name="action_name">win.back</property>
+                <property name="image">back_image</property>
+                <property name="always_show_image">True</property>
+              </object>
+              <packing>
+                <property name="expand">True</property>
+                <property name="fill">True</property>
+                <property name="position">0</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkButton" id="forward_button">
+                <property name="visible">True</property>
+                <property name="sensitive">False</property>
+                <property name="can_focus">True</property>
+                <property name="receives_default">True</property>
+                <property name="action_name">win.forward</property>
+                <property name="image">forward_image</property>
+                <property name="always_show_image">True</property>
+              </object>
+              <packing>
+                <property name="expand">True</property>
+                <property name="fill">True</property>
+                <property name="position">1</property>
+              </packing>
+            </child>
           </object>
           <packing>
-            <property name="position">2</property>
+            <property name="position">3</property>
           </packing>
         </child>
         <child>
@@ -173,4 +183,9 @@
       </object>
     </child>
   </template>
+  <object class="GtkImage" id="forward_image">
+    <property name="visible">True</property>
+    <property name="can_focus">False</property>
+    <property name="icon_name">go-next-symbolic</property>
+  </object>
 </interface>
diff --git a/src/window.py b/src/window.py
index 280795c..1ecb670 100644
--- a/src/window.py
+++ b/src/window.py
@@ -24,7 +24,7 @@ from .compressor import Compressor
 from .tools import message_dialog
 
 
-UI_PATH = '/com/github/ImCompressor/ui/'
+UI_PATH = '/com/github/huluti/ImCompressor/ui/'
 
 
 @Gtk.Template(resource_path=UI_PATH + 'window.ui')
@@ -207,7 +207,7 @@ class ImCompressorWindow(Gtk.ApplicationWindow):
 
     def about_window(self, *args):
         dialog = Gtk.AboutDialog(transient_for=self)
-        dialog.set_logo_icon_name('com.github.ImCompressor')
+        dialog.set_logo_icon_name('com.github.huluti.ImCompressor')
         dialog.set_program_name('ImCompressor')
         dialog.set_version('0.2')
         dialog.set_website('https://github.com/Huluti/ImCompressor')
