diff --git a/activity/activity.info b/activity/activity.info
index 9ceb1c3..65d147a 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,7 +1,7 @@
 [Activity]
 implement = flipsticks
 name      = Flip Sticks
-summary   = Using keyframes, program a stick figure to twist and dance
+summary   = Animate stick figures with keyframes
 description = Flipsticks is a keyframe animation activity that lets you pose
             and program a stick figure to walk, run, rotate, twist, tumble and
             dance. You can save your animations to the journal and will soon be
@@ -13,11 +13,19 @@ license   = GPLv2+
 
 version   = 14
 stability = stable
-
+metadata_license = CC0-1.0
 icon      = activity-flipsticks
 exec      = sugar-activity3 activity.flipsticksActivity
 requires  = sdk/sugar-toolkit
 
 # support original activity.info fields
 activity_version = 14
-bundle_id = org.worldwideworkshop.olpc.FlipSticks
+release_date = 2024-09-13
+tags = Game
+bundle_id = org.sugarlabs.FlipSticks
+url = https://github.com/sugarlabs/flipsticks
+repository = https://github.com/sugarlabs/flipsticks.git
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+developer_id = org.sugarlabs
+screenshots = https://i.imgur.com/49VPVly.png https://i.imgur.com/ILrONHd.png https://i.imgur.com/6VDUGTn.png https://i.imgur.com/94k7X2i.png
\ No newline at end of file
