diff --git a/activity.py b/activity.py
index 40a91c3..2d31088 100644
--- a/activity.py
+++ b/activity.py
@@ -19,7 +19,7 @@ from gettext import gettext as _
 from sugar3.graphics.toggletoolbutton import ToggleToolButton
 from sugar3.graphics.toolbutton import ToolButton
 from sugar3.graphics.toolbarbox import ToolbarBox
-from sugar3.activity.widgets import ActivityToolbarButton
+from sugarapp.widgets import ExtendedActivityToolbarButton
 from sugar3.activity.widgets import StopButton
 
 from toolkit.activity import SharedActivity
@@ -48,7 +48,7 @@ class flipsticksActivity(SharedActivity):
 
         toolbar_box = ToolbarBox()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, 0)
         activity_button.show()
 
diff --git a/model.py b/model.py
index 478a224..a937798 100644
--- a/model.py
+++ b/model.py
@@ -14,7 +14,7 @@
 
 import os
 import math
-from sugar3.datastore import datastore
+from sugarapp.widgets import DesktopSaveChooser
 
 try:
     import json
@@ -208,20 +208,15 @@ def getparentjoint(jname, joints, middle):
 
 
 def screen_shot(pixbuf):
-    tmpdir = '/tmp'
-
-    filename = 'fp%03d.png' % i
-    filepath = os.path.join(tmpdir, filename)
-    pixbuf.savev(filepath, 'png', (), ())
-
-    mediaObject = datastore.create()
-    mediaObject.metadata['title'] = 'FlipSticks PNG'
-    thumbData = _get_base64_pixbuf_data(pixbuf)
-    mediaObject.metadata['preview'] = thumbData
-    # medaiObject.metadata['icon-color'] = ''
-    mediaObject.metadata['mime_type'] = 'image/png'
-    mediaObject.file_path = filepath
-    datastore.write(mediaObject)
+    # Open a file chooser dialog for the user to select the filename
+    chooser = DesktopSaveChooser(None, filename='screenshot.png')
+    filename = chooser.get_filename()
+    
+    # If the user cancels the dialog, exit the function
+    if not filename:
+        return
+
+    pixbuf.savev(filename, 'png', (), ())
 
 
 def _get_base64_pixbuf_data(pixbuf):
diff --git a/toolkit/activity.py b/toolkit/activity.py
index 20cadfe..6f95957 100644
--- a/toolkit/activity.py
+++ b/toolkit/activity.py
@@ -24,7 +24,7 @@ from gi.repository import Gdk
 from gi.repository import TelepathyGLib
 import logging
 
-from sugar3.activity import activity
+from sugarapp.widgets import SugarCompatibleActivity
 from sugar3.presence.sugartubeconn import SugarTubeConnection
 from sugar3.graphics.alert import ConfirmationAlert, NotifyAlert
 
@@ -49,7 +49,7 @@ class CursorFactory:
         return self.cursors[cur_type]
 
 
-class Activity(activity.Activity):
+class Activity(SugarCompatibleActivity):
 
     """Basic activity class"""
 
@@ -102,22 +102,22 @@ class Activity(activity.Activity):
         pass
 
     def set_toolbar_box(self, toolbox):
-        if hasattr(activity.Activity, 'set_toolbar_box'):
-            activity.Activity.set_toolbar_box(self, toolbox)
+        if hasattr(SugarCompatibleActivity, 'set_toolbar_box'):
+            SugarCompatibleActivity.set_toolbar_box(self, toolbox)
         else:
             self.set_toolbox(toolbox)
 
     def get_toolbar_box(self):
-        if hasattr(activity.Activity, 'get_toolbar_box'):
-            return activity.Activity.get_toolbar_box(self)
+        if hasattr(SugarCompatibleActivity, 'get_toolbar_box'):
+            return SugarCompatibleActivity.get_toolbar_box(self)
         else:
             return self.get_toolbox()
 
     toolbar_box = property(get_toolbar_box, set_toolbar_box)
 
     def get_shared_activity(self):
-        if hasattr(activity.Activity, 'get_shared_activity'):
-            return activity.Activity.get_shared_activity(self)
+        if hasattr(SugarCompatibleActivity, 'get_shared_activity'):
+            return SugarCompatibleActivity.get_shared_activity(self)
         else:
             return self.shared_activity
 
@@ -154,8 +154,9 @@ class Activity(activity.Activity):
 
         if self._cursor != cursor:
             self._cursor = cursor
-            self.get_window().set_cursor(self._cursor)
-            Gdk.flush()
+            if self.get_window():
+                self.get_window().set_cursor(self._cursor)
+                Gdk.flush()
 
     def __init__(self, canvas, handle):
         """
@@ -167,7 +168,7 @@ class Activity(activity.Activity):
         handle -- sugar.activity.activityhandle.ActivityHandle
             instance providing the activity id and access to the
         """
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         if handle.object_id:
             self.__state = _NEW_INSTANCE
