diff --git a/theme.py b/theme.py
index 9568535..794a84f 100644
--- a/theme.py
+++ b/theme.py
@@ -18,6 +18,8 @@ from gettext import gettext as _
 
 from sugar3.graphics import style
 
+from sugarapp.helpers import PrimaryMonitor
+
 GRAY = "#B7B7B7"
 PINK = "#FF0099"
 YELLOW = "#FFFF00"
@@ -79,11 +81,11 @@ PAD = 10
 LOGO_WIDTH = 276
 TOLLBAR_HEIGHT = style.LARGE_ICON_SIZE
 
-KEYFRAMEWIDTH = Gdk.Screen.width() - PAD * 3
+KEYFRAMEWIDTH = PrimaryMonitor.width() - PAD * 3
 KEYFRAMEHEIGHT = 80
 
-DRAWWIDTH = Gdk.Screen.width() - LOGO_WIDTH - PAD * 4
-DRAWHEIGHT = Gdk.Screen.height() - KEYFRAMEHEIGHT - PAD * 6 - TOLLBAR_HEIGHT
+DRAWWIDTH = PrimaryMonitor.width() - LOGO_WIDTH - PAD * 4
+DRAWHEIGHT = PrimaryMonitor.height() - KEYFRAMEHEIGHT - PAD * 6 - TOLLBAR_HEIGHT
 
 KEYFRAMES = []
 KEYFRAMES_NUMBER = 5
