--- foma-0.9.18/foma.h	2015-06-12 22:38:57.000000000 +0300
+++ b/foma.h	2021-06-21 12:12:55.739363135 +0300
@@ -24,8 +24,8 @@
 #define PROMPT_MAIN 0 /* Regular prompt */
 #define PROMPT_A 1    /* Apply prompt   */
 
-struct defined_networks   *g_defines;
-struct defined_functions  *g_defines_f;
+extern struct defined_networks   *g_defines;
+extern struct defined_functions  *g_defines_f;
 
 /** User stack */
 struct stack_entry {
--- foma-0.9.18/foma.c	2015-06-13 06:24:13.000000000 +0300
+++ b/foma.c	2021-06-21 12:14:59.764143662 +0300
@@ -24,6 +24,10 @@
 #include <readline/readline.h>
 #include "foma.h"
 
+struct defined_networks   *g_defines;
+struct defined_functions  *g_defines_f;
+
+
 /* Front-end behavior variables */
 int pipe_mode = 0;
 int quiet_mode = 0;
--- foma-0.9.18/determinize.c	2015-06-13 06:24:13.000000000 +0300
+++ b/determinize.c	2021-06-21 12:13:56.387744770 +0300
@@ -59,12 +59,12 @@
     unsigned int set_offset;
 };
 
-struct trans_list {
+static struct trans_list {
     int inout;
     int target;
 } *trans_list;
 
-struct trans_array {
+static struct trans_array {
     struct trans_list *transitions;
     unsigned int size;
     unsigned int tail;
--- foma-0.9.18/minimize.c	2015-06-13 06:24:13.000000000 +0300
+++ b/minimize.c	2021-06-21 12:13:35.157611165 +0300
@@ -66,12 +66,12 @@
   _Bool index;
 };
 
-struct trans_list {
+static struct trans_list {
     int inout;
     int source;
 } *trans_list;
 
-struct trans_array {
+static struct trans_array {
     struct trans_list *transitions;
     unsigned int size;
     unsigned int tail;
