--- a/Makefile	2021-06-21 11:41:45.764583598 +0300
+++ foma-0.9.17/Makefile	2021-06-21 11:57:35.945567018 +0300
@@ -1,4 +1,4 @@
-prefix = /usr/local
+prefix = /app/libreoffice/share/extensions/Voikko
 exec_prefix = $(prefix)
 libdir = $(exec_prefix)/lib
 bindir = $(exec_prefix)/bin
@@ -13,7 +13,7 @@
 LEXIFACE = flex -8 --prefix=interface
 LEXCMATRIX = flex -8 --prefix=cmatrix
 RM = /bin/rm -f
-LDFLAGS = -lreadline -lz -ltermcap
+LDFLAGS = -lreadline -lz
 FLOOKUPLDFLAGS = libfoma.a -lz
 CFLAGS = -O3 -Wall -D_GNU_SOURCE -std=c99 -fvisibility=hidden -fPIC
 FOMAOBJS = foma.o stack.o iface.o lex.interface.o
