diff --git a/forge.config.ts b/forge.config.ts
index 9a6a54db..4d1c6ff5 100644
--- a/forge.config.ts
+++ b/forge.config.ts
@@ -10,6 +10,7 @@ import rendererConfig from "./webpack.renderer.config";
 import mainConfig from "./webpack.main.config";
 
 const packagerConfig: ForgePackagerOptions = {
+  download: { unsafelyDisableChecksums: true },
   appBundleId: "com.dygmalab.bazecor",
   darwinDarkModeSupport: true,
   asar: false,
@@ -86,6 +87,16 @@ const config: ForgeConfig = {
   ],
   hooks: {
     packageAfterPrune: async (_forgeConfig, buildPath, _electronVersion, platform, _arch) => {
+      const rootYarnRc = path.resolve(__dirname, ".yarnrc");
+      if (fs.existsSync(rootYarnRc)) {
+        fs.copyFileSync(rootYarnRc, path.resolve(buildPath, ".yarnrc"));
+      }
+
+      const rootYarnLock = path.resolve(__dirname, "yarn.lock");
+      if (fs.existsSync(rootYarnLock)) {
+        fs.copyFileSync(rootYarnLock, path.resolve(buildPath, "yarn.lock"));
+      }
+
       /**
        * Serialport, usb and uiohook-napi are problematic libraries to run in Electron.
        * When Electron app is been built, these libraries are not included properly in the final executable.
@@ -101,7 +112,7 @@ const config: ForgeConfig = {
       };
 
       fs.writeFileSync(path.resolve(buildPath, "package.json"), JSON.stringify(packageJson));
-      spawnSync("npm", ["install", "--omit=dev"], {
+      spawnSync("yarn", ["install", "--production", "--offline"], {
         cwd: buildPath,
         stdio: "inherit",
         shell: true,
