From 24a3dd440d0065ffaf9d6733cd7e000930498e22 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Fri, 10 Jan 2020 09:24:05 -0300
Subject: [PATCH 2/4] fix-reload

---
 historietaactivity.py | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/historietaactivity.py b/historietaactivity.py
index cc189fc..003f1cd 100644
--- a/historietaactivity.py
+++ b/historietaactivity.py
@@ -307,10 +307,39 @@ class HistorietaActivity(SugarCompatibleActivity):
         if active_globe is not None:
             active_globe.set_selected(False)
 
+    def _cleanup_page(self):
+        # clean all selections
+        for box in self.page.boxs:
+            for globe in box.globos:
+                globe.set_selected(False)
+            box.set_globo_activo(None)
+
+        # start from the last box
+        self.page.set_active_box(self.page.boxs[-1])
+
+        # clean current page
+        for _ in range(len(self.page.table)):
+            self.globes_manager.borrar(None)
+            self.page.set_active_box(self.page.boxs[-1])
+
+        # clean current title globe
+        box = self.page.get_active_box()
+        if box.title_globe in box.globos:
+            box.globos.remove(box.title_globe)
+        if box.title_globe is not None:
+            box.title_globe.set_selected(False)
+            box.title_globe = None
+
     def read_file(self, file_path):
+        self._cleanup_page()
+
         persistence = persistencia.Persistence()
         persistence.read(file_path, self.page)
 
+        if self.page.empty_page:
+            self.set_canvas(self._notebook)
+            self.page.empty_page = False
+
     def _get_image_size(self):
         image_height, image_width = 0, 0
         posi = 0
-- 
2.23.0

