From 0c15a76f6be8b11dd3f9b12b10c044ba12155eea Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Fri, 10 Jan 2020 09:29:05 -0300
Subject: [PATCH 3/4] monitors

---
 historietaactivity.py | 5 +++--
 reorderwindow.py      | 6 ++++--
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/historietaactivity.py b/historietaactivity.py
index 003f1cd..9c27a95 100644
--- a/historietaactivity.py
+++ b/historietaactivity.py
@@ -40,6 +40,7 @@ from slideview import SlideView
 from reorderwindow import ReorderView
 from reorderwindow import ImageEditorView
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import ExtendedActivityToolbarButton
 from sugarapp.widgets import DesktopSaveChooser
 from sugarapp.widgets import SugarCompatibleActivity
@@ -580,7 +581,7 @@ class HistorietaActivity(SugarCompatibleActivity):
 DEF_SPACING = 6
 DEF_WIDTH = 4
 
-SCREEN_HEIGHT = Gdk.Screen.height()
+SCREEN_HEIGHT = PrimaryMonitor.height()
 # HACK: This is to calculate the scrollbar width
 # defined in sugar-artwork gtk-widgets.css.em
 if style.zoom(1):
@@ -588,7 +589,7 @@ if style.zoom(1):
 else:
     scrollbar_width = 11
 
-SCREEN_WIDTH = Gdk.Screen.width() - scrollbar_width - 5
+SCREEN_WIDTH = PrimaryMonitor.width() - scrollbar_width - 5
 BOX_HEIGHT = 450
 
 THUMB_SIZE = activity.PREVIEW_SIZE
diff --git a/reorderwindow.py b/reorderwindow.py
index 8c95740..5581f0c 100644
--- a/reorderwindow.py
+++ b/reorderwindow.py
@@ -9,6 +9,8 @@ from gi.repository import GdkPixbuf
 from sugar3.graphics import style
 from sugar3.graphics.toolbutton import ToolButton
 
+from sugarapp.helpers import PrimaryMonitor
+
 HANDLE_SIZE = 18
 MIN_IMAGE_SIZE = 50
 
@@ -22,9 +24,9 @@ class BaseWindow(Gtk.Window):
         self.set_decorated(False)
         self.set_resizable(False)
         if width == -1:
-            width = Gdk.Screen.width() - style.GRID_CELL_SIZE * 2
+            width = PrimaryMonitor.width() - style.GRID_CELL_SIZE * 2
         if height == -1:
-            height = Gdk.Screen.height() - style.GRID_CELL_SIZE * 2
+            height = PrimaryMonitor.height() - style.GRID_CELL_SIZE * 2
         self.set_size_request(width, height)
 
 
-- 
2.23.0

