From 29d4521419b2177f30d33fe595697e8ae82d0dba Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Fri, 10 Jan 2020 09:34:54 -0300
Subject: [PATCH 4/4] info

---
 activity/activity.info | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index 35c7553..22466d7 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,14 +1,22 @@
 [Activity]
 name = FotoToon
-bundle_id = org.eq.FotoToon
+bundle_id = org.sugarlabs.FotoToon
 icon = historieta
 exec = sugar-activity3 historietaactivity.HistorietaActivity
 show_launcher = yes
 activity_version = 24
+release_date = 2019-12-03
 host_version = 2
 mime_types = application/x-fototoon-activity;
-summary = Interested in telling a story? Easy! Use your pictures and your imagination to create your own comics!
 categories = media language art
 repository = https://github.com/sugarlabs/fototoon-activity.git
+url = https://help.sugarlabs.org/en/fototoon.html
 max_participants = 1
 license = GPLv3+
+metadata_license = CC0-1.0
+summary = Create your own comic
+description = FotoToon allows you to create everything from a simple “comic” strip to a long elaborate illustrated document. You will be able to import many kinds of images into the panes of your project and add text bubbles to help tell the story.
+tags = Graphics
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/fototoon-activity/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/fototoon-activity/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/fototoon-activity/master/screenshots/en/3.png
-- 
2.23.0

