diff --git a/activity/activity.info b/activity/activity.info
index 2b1e77a..320eede 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,11 +1,19 @@
 [Activity]
 name = FractionBounce
-activity_version = 27
-license = GPLv3
-bundle_id = org.sugarlabs.FractionBounceActivity
+activity_version = 27.1
+release_date = 2019-10-22
+metadata_license = CC0-1.0
+license = GPLv3+
+bundle_id = org.sugarlabs.FractionBounce
 exec = sugar-activity3 FractionBounceActivity.FractionBounceActivity
 icon = activity-fractionbounce
 show_launcher = yes
-summary = interactive fraction game
+tags = Education
+summary = A game about fractions
 category = maths game
+url = https://github.com/sugarlabs/fractionbounce
 repository = https://github.com/sugarlabs/fractionbounce
+description = FractionBounce is a game to nudge a bouncing ball to land at an estimate of a given fraction. For example, if 2/3 is displayed on the ball, then the ball must land 2/3 the distance along the bottom.
+developer_name = Sugar Labs Community
+update_contact = tch@sugarlabs.org
+screenshots = https://raw.githubusercontent.com/sugarlabs/fractionbounce/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/fractionbounce/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/fractionbounce/master/screenshots/en/3.png
