--- freeplane-1.8.1/freeplane.sh	2020-03-28 13:58:58.000000000 +0100
+++ freeplane2.sh	2020-04-07 19:04:03.267133991 +0200
@@ -180,6 +180,7 @@
 
 _debug "Calling: "\
 "${JAVACMD}" -Xmx512m\
+ "-DWORKINGDIRECTORY=$XDG_DATA_HOME" \
  "-Dorg.freeplane.userfpdir=$userfpdir"\
  "-Dorg.freeplane.old_userfpdir=$old_userfpdir"\
  "-Dorg.freeplane.globalresourcedir=${freedir}/resources"\
@@ -192,6 +193,7 @@
 
 # now actually launch Freeplane
 "${JAVACMD}" -Xmx512m\
+ "-DWORKINGDIRECTORY=$XDG_DATA_HOME" \
  "-Dorg.freeplane.userfpdir=$userfpdir"\
  "-Dorg.freeplane.old_userfpdir=$old_userfpdir"\
  "-Dorg.freeplane.globalresourcedir=${freedir}/resources"\
