From abf3daa97f96fb1df8f68eec9730dd37e9ea504a Mon Sep 17 00:00:00 2001
From: freanux <pw@earthwave.ch>
Date: Thu, 12 Oct 2017 10:22:19 +0200
Subject: [PATCH] append standard freetype2 include path

---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index bd9420a..87ed876 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,6 +14,9 @@ AC_CONFIG_MACRO_DIR([m4])
 AC_HEADER_STDC
 AM_SILENT_RULES([yes])
 
+# --- append freetype2 standard include path ---
+CPPFLAGS="$CPPFLAGS -I/usr/include/freetype2"
+
 # --- check headers ---
 AC_CHECK_HEADERS([ft2build.h], [], [missing_headers="$missing_headers ft2build.h"])
 
