Index: ftgl-2.1.3~rc5/Makefile.am
===================================================================
--- ftgl-2.1.3~rc5.orig/Makefile.am	2008-05-12 11:04:08.000000000 -0700
+++ ftgl-2.1.3~rc5/Makefile.am	2013-02-26 08:38:09.000000000 -0800
@@ -26,13 +26,13 @@
 
 # Print out an informative summary.
 all-local:
-	@$(ECHO) "Done."
-	@$(ECHO)
+	@echo "Done."
+	@echo
 	@if test "x$(MAKECMDGOALS)" = "xall-am" -o "x$(.TARGETS)" = "xall-am" -o "x$(MAKECMDGOALS)" = "x" -o "x$(.TARGETS)" = "x" ; then \
-	  $(ECHO) "---" ;\
-	  $(ECHO) "Run 'make install' to begin installation into $(prefix)" ;\
+	  echo "---" ;\
+	  echo "Run 'make install' to begin installation into $(prefix)" ;\
 	fi
-	@$(ECHO)
+	echo
 
 # Upload documentation
 DOC = docs/html docs/latex/ftgl.pdf
