--- fuse-2.9.2/util/ulockmgr_server.c.closefromfix	2019-01-04 05:33:33.000000000 -0800
+++ fuse-2.9.2/util/ulockmgr_server.c	2022-07-12 12:29:56.445402244 -0700
@@ -124,7 +124,7 @@
 	return res;
 }
 
-static int closefrom(int minfd)
+static int _closefrom(int minfd)
 {
 	DIR *dir = opendir("/proc/self/fd");
 	if (dir) {
@@ -384,7 +384,7 @@
 		dup2(nullfd, 1);
 	}
 	close(3);
-	closefrom(5);
+	_closefrom(5);
 	while (1) {
 		char c;
 		int sock;
