Author: Alberto Garcia <berto@igalia.com>
Subject: Store configuration in $XDG_CONFIG_HOME/fuserc, not in $HOME/.fuserc
diff --git a/compat/linux/paths.c b/compat/linux/paths.c
index 3bc76b0b..e6f05a65 100644
--- a/compat/linux/paths.c
+++ b/compat/linux/paths.c
@@ -49,7 +49,7 @@ const char*
 compat_get_config_path( void )
 {
   const char *dir;
-  dir = getenv( "HOME" ); if( dir ) return dir;
+  dir = getenv( "XDG_CONFIG_HOME" ); if( dir ) return dir;
   return ".";
 }
 
diff --git a/settings.pl b/settings.pl
index 8017b6b7..e4536244 100644
--- a/settings.pl
+++ b/settings.pl
@@ -99,7 +99,7 @@ print hashline( __LINE__ ), << 'CODE';
 #ifdef WIN32
 #define CONFIG_FILE_NAME "fuse.cfg"
 #else				/* #ifdef WIN32 */
-#define CONFIG_FILE_NAME ".fuserc"
+#define CONFIG_FILE_NAME "fuserc"
 #endif				/* #ifdef WIN32 */
 
 /* The current settings of options, etc */
