diff --git a/TheForceEngine/TFE_Settings/settings.cpp b/TheForceEngine/TFE_Settings/settings.cpp
index 84ae4a99..9f4e8c26 100644
--- a/TheForceEngine/TFE_Settings/settings.cpp
+++ b/TheForceEngine/TFE_Settings/settings.cpp
@@ -200,12 +200,19 @@ namespace TFE_Settings
 			// If the registry approach fails, just try looking in the various hardcoded paths.
 			if (!pathValid)
 			{
-				// Try various possible locations.
-				const char** locations = c_gameLocations[gameId];
-				for (u32 i = 0; i < c_hardcodedPathCount; i++)
+				const char* locations[] =
 				{
-					if (FileUtil::directoryExits(locations[i]))
+					strcat(std::getenv("XDG_DATA_HOME"), "/"),
+					strcat(std::getenv("HOME"), "/.steam/steam/steamapps/common/Dark Forces/Game/"),
+					strcat(std::getenv("HOME"), "/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Dark Forces/Game/"),
+				};
+				for (u32 i = 0; i < TFE_ARRAYSIZE(locations); i++)
+				{
+					char testFile[TFE_MAX_PATH];
+					sprintf(testFile, "%s/%s", locations[i], "DARK.GOB");
+					if (FileUtil::exists(testFile))
 					{
+						TFE_System::logWrite(LOG_MSG, "Main", locations[i]);
 						strcpy(s_gameSettings.header[gameId].sourcePath, locations[i]);
 						pathValid = true;
 						break;
