diff -ruN a/gazebo/msgs/CMakeLists.txt b/gazebo/msgs/CMakeLists.txt
--- a/gazebo/msgs/CMakeLists.txt	2021-04-20 23:56:51.000000000 +0700
+++ b/gazebo/msgs/CMakeLists.txt	2021-04-27 05:34:12.968614243 +0700
@@ -143,7 +143,7 @@
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 add_executable(gazebomsgs_out generator/GazeboGenerator.cc generator/gazebo_generator.cc)
-target_link_libraries(gazebomsgs_out ${GZ_PROTOBUF_LIBRARY} ${GZ_PROTOBUF_PROTOC_LIBRARY})
+target_link_libraries(gazebomsgs_out ${GZ_PROTOBUF_LIBRARY} ${GZ_PROTOBUF_PROTOC_LIBRARY} protobuf::libprotoc protobuf::libprotobuf)
 if (UNIX)
   target_link_libraries(gazebomsgs_out pthread)
 endif()
