diff -up gc-7.6.4/pthread_start.c.dont_disable_exceptions gc-7.6.4/pthread_start.c
--- gc-7.6.4/pthread_start.c.dont_disable_exceptions	2018-01-26 07:16:17.000000000 -0600
+++ gc-7.6.4/pthread_start.c	2018-03-14 09:21:34.263477197 -0500
@@ -22,7 +22,7 @@
 /* case at the top of the file.  FIXME: it's still unclear whether this */
 /* will actually cause the exit handler to be invoked last when         */
 /* thread_exit is called (and if -fexceptions is used).                 */
-#if defined(__GNUC__) && defined(__linux__)
+#if 0 && defined(__GNUC__) && defined(__linux__)
   /* We undefine __EXCEPTIONS to avoid using GCC __cleanup__ attribute. */
   /* The current NPTL implementation of pthread_cleanup_push uses       */
   /* __cleanup__ attribute when __EXCEPTIONS is defined (-fexceptions). */
