--- gcc-3.3.6/gcc/f/com.h       2003-03-22 16:24:51.000000000 +0100
+++ gcc-3.3.6-fix/gcc/f/com.h   2006-03-15 18:23:29.000000000 +0100
@@ -233,7 +233,7 @@
 void ffecom_finish_progunit (void);
 tree ffecom_get_invented_identifier (const char *pattern, ...)
   ATTRIBUTE_PRINTF_1;
-ffeinfoKindtype ffecom_gfrt_basictype (ffecomGfrt ix);
+ffeinfoBasictype ffecom_gfrt_basictype (ffecomGfrt ix);
 ffeinfoKindtype ffecom_gfrt_kindtype (ffecomGfrt ix);
 void ffecom_init_0 (void);
 void ffecom_init_2 (void);

--- gcc-3.3.6/gcc/collect2.c	2003-12-08 20:02:39.000000000 +0100
+++ gcc-3.3.6-fix/gcc/collect2.c	2023-09-22 18:04:31.158974582 +0200
@@ -1572,7 +1572,7 @@
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, 0666);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
