diff -up ./src/graphics/model_obj.cpp.gcc6 ./src/graphics/model_obj.cpp
--- ./src/graphics/model_obj.cpp.gcc6	2016-02-09 19:02:50.282554264 -0500
+++ ./src/graphics/model_obj.cpp	2016-02-09 19:03:10.566308088 -0500
@@ -221,7 +221,7 @@ bool ModelObj::Save(const std::string &
 	std::ofstream f(strFileName.c_str());
 	if (!f)
 	{
-		error_output << "Error opening file for writing: " << error_output << endl;
+		error_output << "Error opening file for writing: " << strFileName << endl;
 		return false;
 	}

