diff --git a/buildSrc/DesktopBuilder.js b/buildSrc/DesktopBuilder.js
index 56e84c93..9a4a8a6e 100644
--- a/buildSrc/DesktopBuilder.js
+++ b/buildSrc/DesktopBuilder.js
@@ -108,7 +108,6 @@ function build(dirname, version, targets, updateUrl, nameSuffix) {
 				fs.removeAsync(path.join(distDir, '/node_modules/')),
 				fs.removeAsync(path.join(distDir, '/cache.json')),
 				fs.removeAsync(path.join(distDir, '/package.json')),
-				fs.removeAsync(path.join(distDir, '/package-lock.json')),
 				fs.removeAsync(path.join(distDir, '/src/')),
 			]))
 		})
