diff -Naur gImageReader-3.3.0.orig/gtk/src/TessdataManager.cc gImageReader-3.3.0/gtk/src/TessdataManager.cc
--- gImageReader-3.3.0.orig/gtk/src/TessdataManager.cc	2018-09-26 23:06:06.000000000 +0200
+++ gImageReader-3.3.0/gtk/src/TessdataManager.cc	2019-02-02 12:03:53.466818444 +0100
@@ -159,8 +159,28 @@
 		for(GList* l = elementArray; l; l = l->next) {
 			JsonNode* value = static_cast<JsonNode*>(l->data);
 			JsonObject* treeObj = json_node_get_object(value);
-			Glib::ustring fileName = json_object_get_string_member(treeObj, "name");
-			Glib::ustring url = json_object_get_string_member(treeObj, "download_url");
+			Glib::ustring fileName;
+			try
+			{
+				fileName = json_object_get_string_member(treeObj, "name");
+			}
+			catch (const std::exception&)
+			{
+				continue;
+			}
+			if (fileName == "script")
+			{
+				continue;
+			}
+			Glib::ustring url;
+			try
+			{
+				url = json_object_get_string_member(treeObj, "download_url");
+			}
+			catch (const std::exception&)
+			{
+				continue;
+			}
 			Glib::ustring subdir;
 			// If filename starts with upper case letter, it is a script
 			if(fileName.substr(0, 1).uppercase() == fileName.substr(0, 1)) {
