diff -Naur gImageReader-3.3.0.orig/qt/src/MainWindow.cc gImageReader-3.3.0/qt/src/MainWindow.cc
--- gImageReader-3.3.0.orig/qt/src/MainWindow.cc	2018-09-26 23:06:06.000000000 +0200
+++ gImageReader-3.3.0/qt/src/MainWindow.cc	2019-01-02 16:16:51.821467726 +0100
@@ -198,6 +198,8 @@
 	}
 #endif
 
+	ui.actionManageLanguages->setVisible(!getConfig()->useSystemDataLocations());
+
 	m_sourceManager->addSources(files);
 }
 
@@ -313,6 +315,7 @@
 
 void MainWindow::showConfig() {
 	m_config->showDialog();
+	ui.actionManageLanguages->setVisible(!getConfig()->useSystemDataLocations());
 	m_recognizer->updateLanguagesMenu();
 }
 
