--- gimp-lqr-plugin-0.7.2/configure.ac	2013-08-05 09:05:28.000000000 -0400
+++ gimp-lqr-plugin-0.7.2/configure.ac.new	2020-06-13 10:42:58.321279223 -0400
@@ -58,10 +58,10 @@
 AC_SUBST(GIMP_CFLAGS)
 AC_SUBST(GIMP_LIBS)
 
-GIMP_LIBDIR=`$PKG_CONFIG --variable=gimplibdir gimp-2.0`
+GIMP_LIBDIR?=`$PKG_CONFIG --variable=gimplibdir gimp-2.0`
 AC_SUBST(GIMP_LIBDIR)
 
-GIMP_DATADIR=`$PKG_CONFIG --variable=gimpdatadir gimp-2.0`
+GIMP_DATADIR?=`$PKG_CONFIG --variable=gimpdatadir gimp-2.0`
 AC_SUBST(GIMP_DATADIR)
 
 
--- gimp-lqr-plugin-0.7.2/configure	2013-08-05 09:07:12.000000000 -0400
+++ gimp-lqr-plugin-0.7.2/configure.new	2020-06-13 10:51:43.672081887 -0400
@@ -4929,10 +4929,10 @@
 
 
 
-GIMP_LIBDIR=`$PKG_CONFIG --variable=gimplibdir gimp-2.0`
+GIMP_LIBDIR?=`$PKG_CONFIG --variable=gimplibdir gimp-2.0`
 
 
-GIMP_DATADIR=`$PKG_CONFIG --variable=gimpdatadir gimp-2.0`
+GIMP_DATADIR?=`$PKG_CONFIG --variable=gimpdatadir gimp-2.0`
 
 
 
