commit ca4511c1a4dd8edabe86e4a943861fda07b7e86c
Author: Neil Mayhew <neil@neil.mayhew.name>
Date:   Fri Jul 5 13:37:39 2019 -0600

    Add missing header file

diff --git a/src/gimplensfun.cpp b/src/gimplensfun.cpp
index da2a94c..8e9e12e 100644
--- a/src/gimplensfun.cpp
+++ b/src/gimplensfun.cpp
@@ -31,6 +31,7 @@
 #include <libgimp/gimp.h>
 #include <libgimp/gimpui.h>
 
+#include <exiv2/error.hpp>
 #include <exiv2/image.hpp>
 #include <exiv2/exif.hpp>
 
