diff --git a/Makefile b/Makefile
index e6f08a7..7917554 100644
--- a/Makefile
+++ b/Makefile
@@ -93,7 +93,7 @@ LIB.OBJS.MX        := $(LIB.OBJS.MX:.c=.o)
 LIB.SOBJS.MX       := $(addprefix tmp/$(SYSTEM)/mx/shared/,$(LIB.SRCS.NAMES))
 LIB.SOBJS.MX       := $(LIB.SOBJS.MX:.c=.o)

-glew.lib: lib lib/$(LIB.SHARED) lib/$(LIB.STATIC) glew.pc
+glew.lib: lib lib/$(LIB.SHARED) glew.pc

 lib:
 	mkdir lib
@@ -250,7 +250,6 @@ endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.DEVLNK)
 endif
-	$(INSTALL) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/

 install.lib.mx: glew.lib.mx
 	$(INSTALL) -d -m 0755 $(LIBDIR)
diff --git a/config/Makefile.linux b/config/Makefile.linux
index 234adc6..6958782 100644
--- a/config/Makefile.linux
+++ b/config/Makefile.linux
@@ -10,10 +10,10 @@ ifeq (ppc64,${M_ARCH})
   ARCH64 = true
 endif
 ifeq (${ARCH64},true)
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
+  LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64 -L/app/lib
   LIBDIR = $(GLEW_DEST)/lib64
 else
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
+  LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib -L/app/lib
   LIBDIR = $(GLEW_DEST)/lib
 endif
 LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
