--- a/src/AppInfoViewSaver.vala	2018-02-18 13:19:28.436621354 +0100
+++ b/src/AppInfoViewSaver.vala	2018-02-18 13:19:24.104751382 +0100
@@ -93,7 +93,7 @@
 
         key.set_string (KeyFileDesktop.GROUP, KeyFileDesktop.KEY_TYPE, KeyFileDesktop.TYPE_APPLICATION);
         key.set_locale_string (KeyFileDesktop.GROUP, KeyFileDesktop.KEY_NAME, lang, name);
-        key.set_locale_string (KeyFileDesktop.GROUP, KeyFileDesktop.KEY_FULLNAME, lang, name);
+        //key.set_locale_string (KeyFileDesktop.GROUP, KeyFileDesktop.KEY_FULLNAME, lang, name);
         key.set_locale_string (KeyFileDesktop.GROUP, KeyFileDesktop.KEY_COMMENT, lang, format_desktop_entry_string (target.save_description));
         key.set_string (KeyFileDesktop.GROUP, KeyFileDesktop.KEY_ICON, format_desktop_entry_string (target.save_icon));
         key.set_boolean (KeyFileDesktop.GROUP, KeyFileDesktop.KEY_NO_DISPLAY, !target.save_display);
