--- gnuclad-0.2.4/src/gnuclad.cpp	2011-01-07 13:28:37.000000000 -0500
+++ gnuclad-0.2.4/src/gnuclad-fixed.cpp	2023-02-24 12:05:06.706818738 -0500
@@ -142,7 +142,7 @@
     cout << "\nError: " << err;
   } catch(string err) {
     cout << "\nError: " << err;
-  } catch(exception e) {
+  } catch(exception &e) {
     cout << "\nError: " << e.what();
   } catch(...) {
     cout << "\nError: unknown reason";
