diff -up godot-3.0.6-stable/platform/x11/detect.py.O2 godot-3.0.6-stable/platform/x11/detect.py
--- godot-3.0.6-stable/platform/x11/detect.py.O2	2019-02-06 16:05:50.878344886 +0100
+++ godot-3.0.6-stable/platform/x11/detect.py	2019-02-06 16:05:57.425278478 +0100
@@ -77,7 +77,7 @@ def configure(env):
     if (env["target"] == "release"):
         # -O3 -ffast-math is identical to -Ofast. We need to split it out so we can selectively disable
         # -ffast-math in code for which it generates wrong results.
-        env.Prepend(CCFLAGS=['-O3', '-ffast-math'])
+        env.Prepend(CCFLAGS=['-O2', '-ffast-math'])
         if (env["debug_symbols"] == "yes"):
             env.Prepend(CCFLAGS=['-g1'])
         if (env["debug_symbols"] == "full"):
