diff -up godot-release-template-1/platform/x11/detect.py.O2 godot-release-template-1/platform/x11/detect.py
--- godot-release-template-1/platform/x11/detect.py.O2	2019-03-11 09:58:09.934768314 +0100
+++ godot-release-template-1/platform/x11/detect.py	2019-03-11 09:58:17.139663676 +0100
@@ -86,7 +86,7 @@ def configure(env):
 
     if (env["target"] == "release"):
         if (env["optimize"] == "speed"): #optimize for speed (default)
-            env.Prepend(CCFLAGS=['-O3'])
+            env.Prepend(CCFLAGS=['-O2'])
         else: #optimize for size
             env.Prepend(CCFLAGS=['-Os'])
 
