diff --git a/configure.in b/configure.in
index fd40d53b..32ceee6d 100644
--- a/configure.in
+++ b/configure.in
@@ -512,12 +512,30 @@ if test $gtk_disable_deprecated = yes; then
 fi
 
 if test "x$glib_has_G_REGEX_ERROR_STRAY_BACKSLASH" = "xno"; then
-  dnl Check for working pcre
-  AC_CHECK_HEADER(pcre.h,[], AC_MSG_ERROR([Header files for PCRE were not found]))
-  AC_CHECK_LIB(pcre,pcre_compile2,[], AC_MSG_ERROR([The PCRE library is missing or too old]))
-  AC_MSG_CHECKING([for UTF-8 support in pcre])
-  if (pcretest -C 2>&1) | grep -i "^ *UTF-8 support" >/dev/null; then
-      AC_MSG_RESULT(yes)
+  PKG_CHECK_MODULES(PCRE, libpcre)
+  AC_CACHE_CHECK([for Unicode support in PCRE],goffice_cv_pcre_has_unicode,[
+		  goffice_save_CFLAGS="$CFLAGS"
+		  goffice_save_LIBS="$LIBS"
+		  CFLAGS="$CFLAGS $PCRE_CFLAGS" LIBS="$PCRE_LIBS"
+		  AC_TRY_RUN([#include <pcre.h>
+			      int main () {
+			      int support;
+			      pcre_config (PCRE_CONFIG_UTF8, &support);
+			      if (!support)
+			      return 1;
+			      pcre_config (PCRE_CONFIG_UNICODE_PROPERTIES, &support);
+			      if (!support)
+			      return 1;
+			      return 0;
+			    }],
+			    goffice_cv_pcre_has_unicode=yes,
+			    goffice_cv_pcre_has_unicode=no,
+			    goffice_cv_pcre_has_unicode=yes)
+		  CFLAGS="$goffice_save_CFLAGS"
+		  LIBS="$goffice_save_LIBS"
+		  ])
+
+  if test "$goffice_cv_pcre_has_unicode" = "yes"; then
       LIBS="$LIBS -lpcre"
       EXTRA_LIBS="$EXTRA_LIBS -lpcre"
   else
@@ -525,7 +543,6 @@ if test "x$glib_has_G_REGEX_ERROR_STRAY_BACKSLASH" = "xno"; then
   fi
 fi
 
-
 AC_DEFUN([GOFFICE_CHECK_FUNC],
 [AC_CHECK_FUNC([$1],
 	[],
