diff -Naru -x .git gperftools/include/gperftools/heap-checker.h gperftools-2.7-enve-mod/include/gperftools/heap-checker.h
--- gperftools/include/gperftools/heap-checker.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/heap-checker.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/heap-checker.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/heap-profiler.h gperftools-2.7-enve-mod/include/gperftools/heap-profiler.h
--- gperftools/include/gperftools/heap-profiler.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/heap-profiler.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/heap-profiler.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/malloc_extension.h gperftools-2.7-enve-mod/include/gperftools/malloc_extension.h
--- gperftools/include/gperftools/malloc_extension.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/malloc_extension.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/malloc_extension.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/malloc_extension_c.h gperftools-2.7-enve-mod/include/gperftools/malloc_extension_c.h
--- gperftools/include/gperftools/malloc_extension_c.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/malloc_extension_c.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/malloc_extension_c.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/malloc_hook.h gperftools-2.7-enve-mod/include/gperftools/malloc_hook.h
--- gperftools/include/gperftools/malloc_hook.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/malloc_hook.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/malloc_hook.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/malloc_hook_c.h gperftools-2.7-enve-mod/include/gperftools/malloc_hook_c.h
--- gperftools/include/gperftools/malloc_hook_c.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/malloc_hook_c.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/malloc_hook_c.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/nallocx.h gperftools-2.7-enve-mod/include/gperftools/nallocx.h
--- gperftools/include/gperftools/nallocx.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/nallocx.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/nallocx.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/profiler.h gperftools-2.7-enve-mod/include/gperftools/profiler.h
--- gperftools/include/gperftools/profiler.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/profiler.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/profiler.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/stacktrace.h gperftools-2.7-enve-mod/include/gperftools/stacktrace.h
--- gperftools/include/gperftools/stacktrace.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/stacktrace.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/stacktrace.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/tcmalloc.h gperftools-2.7-enve-mod/include/gperftools/tcmalloc.h
--- gperftools/include/gperftools/tcmalloc.h	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/tcmalloc.h	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/tcmalloc.h"
\ No newline at end of file
diff -Naru -x .git gperftools/include/gperftools/tcmalloc.h.in gperftools-2.7-enve-mod/include/gperftools/tcmalloc.h.in
--- gperftools/include/gperftools/tcmalloc.h.in	1969-12-31 16:00:00.000000000 -0800
+++ gperftools-2.7-enve-mod/include/gperftools/tcmalloc.h.in	2020-04-05 22:40:36.000000000 -0700
@@ -0,0 +1 @@
+#include "../../src/gperftools/tcmalloc.h.in"
\ No newline at end of file
diff -Naru -x .git gperftools/src/gperftools/malloc_extension.h gperftools-2.7-enve-mod/src/gperftools/malloc_extension.h
--- gperftools/src/gperftools/malloc_extension.h	2020-04-11 14:00:46.000000000 -0700
+++ gperftools-2.7-enve-mod/src/gperftools/malloc_extension.h	2020-04-05 22:40:36.000000000 -0700
@@ -218,6 +218,10 @@
   // REQUIRES: property != NULL; value != NULL
   virtual bool GetNumericProperty(const char* property, size_t* value);
 
+  virtual void eMemoryStats(size_t* virtual_memory_used,
+                            size_t* physical_memory_used,
+                            size_t* bytes_in_use_by_app);
+
   // Set the named "property"'s value.  Returns true if the property
   // is known and writable.  Returns false if the property is not a
   // valid property name for the current malloc implementation, or
diff -Naru -x .git gperftools/src/gperftools/malloc_extension_c.h gperftools-2.7-enve-mod/src/gperftools/malloc_extension_c.h
--- gperftools/src/gperftools/malloc_extension_c.h	2020-04-11 14:00:46.000000000 -0700
+++ gperftools-2.7-enve-mod/src/gperftools/malloc_extension_c.h	2020-04-05 22:40:36.000000000 -0700
@@ -72,6 +72,9 @@
 /* void MallocExtension_GetHeapGrowthStacks(string* result); */
 
 PERFTOOLS_DLL_DECL int MallocExtension_GetNumericProperty(const char* property, size_t* value);
+PERFTOOLS_DLL_DECL void MallocExtension_eMemoryStats(size_t* virtual_memory_used,
+                                                     size_t* physical_memory_used,
+                                                     size_t* bytes_in_use_by_app);
 PERFTOOLS_DLL_DECL int MallocExtension_SetNumericProperty(const char* property, size_t value);
 PERFTOOLS_DLL_DECL void MallocExtension_MarkThreadIdle(void);
 PERFTOOLS_DLL_DECL void MallocExtension_MarkThreadBusy(void);
diff -Naru -x .git gperftools/src/malloc_extension.cc gperftools-2.7-enve-mod/src/malloc_extension.cc
--- gperftools/src/malloc_extension.cc	2020-04-11 14:00:46.000000000 -0700
+++ gperftools-2.7-enve-mod/src/malloc_extension.cc	2020-04-05 22:40:36.000000000 -0700
@@ -118,6 +118,12 @@
   return false;
 }
 
+void MallocExtension::eMemoryStats(size_t* virtual_memory_used,
+                                   size_t* physical_memory_used,
+                                   size_t* bytes_in_use_by_app) {
+  return;
+}
+
 bool MallocExtension::SetNumericProperty(const char* property, size_t value) {
   return false;
 }
diff -Naru -x .git gperftools/src/tcmalloc.cc gperftools-2.7-enve-mod/src/tcmalloc.cc
--- gperftools/src/tcmalloc.cc	2020-04-11 14:00:46.000000000 -0700
+++ gperftools-2.7-enve-mod/src/tcmalloc.cc	2020-04-05 22:40:36.000000000 -0700
@@ -689,6 +689,23 @@
     IterateOverRanges(arg, func);
   }
 
+  virtual void eMemoryStats(size_t* virtual_memory_used,
+                            size_t* physical_memory_used,
+                            size_t* bytes_in_use_by_app) {
+    TCMallocStats stats;
+    ExtractStats(&stats, NULL, NULL, NULL);
+    *virtual_memory_used = (stats.pageheap.system_bytes
+                            + stats.metadata_bytes);
+    *physical_memory_used = (*virtual_memory_used
+                             - stats.pageheap.unmapped_bytes);
+    *bytes_in_use_by_app = (*physical_memory_used
+                            - stats.metadata_bytes
+                            - stats.pageheap.free_bytes
+                            - stats.central_bytes
+                            - stats.transfer_bytes
+                            - stats.thread_bytes);
+  }
+
   virtual bool GetNumericProperty(const char* name, size_t* value) {
     ASSERT(name != NULL);
 
