diff -ru gst-plugins-base-0.10.36/gst-libs/gst/app/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/app/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/app/Makefile.in	2012-02-21 00:33:14.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/app/Makefile.in	2016-12-06 17:50:34.225382882 +0100
@@ -973,7 +973,7 @@
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-base-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-app-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/audio/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/audio/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/audio/Makefile.in	2012-02-21 00:33:15.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/audio/Makefile.in	2016-12-06 17:50:39.471263440 +0100
@@ -1134,7 +1134,7 @@
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-base-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-interfaces-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-audio-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/interfaces/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/interfaces/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/interfaces/Makefile.in	2012-02-21 00:33:16.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/interfaces/Makefile.in	2016-12-06 17:50:53.306948423 +0100
@@ -1142,7 +1142,7 @@
 @HAVE_INTROSPECTION_TRUE@		--libtool="$(top_builddir)/libtool" \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-interfaces-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/netbuffer/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/netbuffer/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/netbuffer/Makefile.in	2012-02-21 00:33:16.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/netbuffer/Makefile.in	2016-12-06 17:50:57.476853480 +0100
@@ -863,7 +863,7 @@
 @HAVE_INTROSPECTION_TRUE@		--libtool="$(top_builddir)/libtool" \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-netbuffer-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/pbutils/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/pbutils/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/pbutils/Makefile.in	2012-02-21 00:33:16.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/pbutils/Makefile.in	2016-12-06 17:51:02.673735155 +0100
@@ -1116,7 +1116,7 @@
 @HAVE_INTROSPECTION_TRUE@		--libtool="$(top_builddir)/libtool" \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-pbutils-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/riff/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/riff/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/riff/Makefile.in	2012-02-21 00:33:16.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/riff/Makefile.in	2016-12-06 17:51:07.250630948 +0100
@@ -906,7 +906,7 @@
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-audio-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-interfaces-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-riff-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/rtp/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/rtp/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/rtp/Makefile.in	2012-02-21 00:33:17.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/rtp/Makefile.in	2016-12-06 17:51:11.867525826 +0100
@@ -922,7 +922,7 @@
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-base-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-rtp-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/rtsp/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/rtsp/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/rtsp/Makefile.in	2012-02-21 00:33:17.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/rtsp/Makefile.in	2016-12-06 17:51:15.858434959 +0100
@@ -1068,7 +1068,7 @@
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-sdp-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-rtsp-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/tag/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/tag/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/tag/Makefile.in	2012-02-21 00:33:18.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/tag/Makefile.in	2016-12-06 17:51:21.977295644 +0100
@@ -1081,7 +1081,7 @@
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-base-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-tag-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
diff -ru gst-plugins-base-0.10.36/gst-libs/gst/video/Makefile.in gst-plugins-base-0.10.36.new/gst-libs/gst/video/Makefile.in
--- gst-plugins-base-0.10.36/gst-libs/gst/video/Makefile.in	2012-02-21 00:33:18.000000000 +0100
+++ gst-plugins-base-0.10.36.new/gst-libs/gst/video/Makefile.in	2016-12-06 17:51:30.387104159 +0100
@@ -1132,7 +1132,7 @@
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg gstreamer-base-@GST_MAJORMINOR@ \
 @HAVE_INTROSPECTION_TRUE@		--pkg-export gstreamer-video-@GST_MAJORMINOR@ \
-@HAVE_INTROSPECTION_TRUE@		--add-init-section="gst_init(NULL,NULL);" \
+@HAVE_INTROSPECTION_TRUE@		--add-init-section="void gst_init(void*,void*); gst_init(NULL,NULL);" \
 @HAVE_INTROSPECTION_TRUE@		--output $@ \
 @HAVE_INTROSPECTION_TRUE@		$(gir_headers) \
 @HAVE_INTROSPECTION_TRUE@		$(gir_sources)
