--- guitarix-11/src/LV2/gx_fuzzfacefm.lv2/wscript	2020-05-27 03:00:12.000000000 -0400
+++ guitarix-11/src/LV2/gx_fuzzfacefm.lv2/wscript-new	2020-05-27 12:27:33.715394833 -0400
@@ -10,7 +10,7 @@
     bld.lv2(
         lv2_base = lv2_base,
         source   = ['gx_fuzzfacefm.cpp'],
-        includes = ['../faust','./', '../DSP', '../DSP/tube_tables'],
+        includes = ['../faust','./', '../DSP', '../DSP/tube_tables', '../../zita-resampler-1.1.0'],
         cxxflags=['-fvisibility=hidden','-Wl,-z,noexecstack','-Wl,-z,relro,-z,now','-Wl,--exclude-libs,ALL'],
         use = ['LV2CORE'],
         )
