commit 7a8529b87bb308d5624d01c615360421d0c178a8
Author: Hubert Figuière <hub@figuiere.net>
Date:   Sat Sep 5 23:25:27 2020 -0400

    LV2/gx_livelooper: remove dependency on glibmm

diff --git a/trunk/src/LV2/gx_livelooper.lv2/gx_livelooper.cpp b/trunk/src/LV2/gx_livelooper.lv2/gx_livelooper.cpp
index 2d350a92..eccf49b7 100644
--- a/trunk/src/LV2/gx_livelooper.lv2/gx_livelooper.cpp
+++ b/trunk/src/LV2/gx_livelooper.lv2/gx_livelooper.cpp
@@ -19,7 +19,6 @@
 
 
 #include <sndfile.hh>
-#include <glibmm.h>
 #include <sys/stat.h>
 #include <time.h>
 ////////////////////////////// LOCAL INCLUDES //////////////////////////
diff --git a/trunk/src/LV2/gx_livelooper.lv2/livelooper.cc b/trunk/src/LV2/gx_livelooper.lv2/livelooper.cc
index 44262405..0af7d288 100644
--- a/trunk/src/LV2/gx_livelooper.lv2/livelooper.cc
+++ b/trunk/src/LV2/gx_livelooper.lv2/livelooper.cc
@@ -178,9 +178,9 @@ private:
 	bool RP2;
 	bool RP3;
 	bool RP4;
-    Glib::ustring pfreset_name;
-    Glib::ustring cur_name;
-    Glib::ustring loop_dir;
+    std::string pfreset_name;
+    std::string cur_name;
+    std::string loop_dir;
     bool save_p;
 	bool mem_allocated;
 	volatile bool ready;
@@ -195,7 +195,7 @@ private:
     void load_array(std::string name);
     void save_to_wave(std::string fname, float *tape, float fSize);
     int load_from_wave(std::string fname, float *tape);
-    void set_p_state(Glib::ustring preset_file_name, bool save_file);
+    void set_p_state(std::string preset_file_name, bool save_file);
     
 public:
 	static void clear_state_f_static(LiveLooper*);
@@ -204,7 +204,7 @@ public:
 	static void compute_static(int count, float *input0, float *output0, LiveLooper*);
 	static void del_instance(LiveLooper *p);
 	static void connect_static(uint32_t port,void* data, LiveLooper *p);
-    static void set_preset_state(Glib::ustring preset_file_name, bool save_file, LiveLooper*);
+    static void set_preset_state(std::string preset_file_name, bool save_file, LiveLooper*);
 	LiveLooper();
 	~LiveLooper();
 };
@@ -421,7 +421,7 @@ int LiveLooper::activate_static(bool start, LiveLooper *p)
 	return (p)->activate(start);
 }
 
-void LiveLooper::set_p_state(Glib::ustring preset_file_name, bool save_file) {
+void LiveLooper::set_p_state(std::string preset_file_name, bool save_file) {
     if (!preset_file_name.empty()) {
         pfreset_name = preset_file_name;
         save_p = save_file;
@@ -441,7 +441,7 @@ void LiveLooper::set_p_state(Glib::ustring preset_file_name, bool save_file) {
     }
 }
 
-void LiveLooper::set_preset_state(Glib::ustring preset_file_name, bool save_file,LiveLooper* p) {
+void LiveLooper::set_preset_state(std::string preset_file_name, bool save_file,LiveLooper* p) {
 	(p)->set_p_state(preset_file_name, save_file);
 }
 
diff --git a/trunk/src/LV2/gx_livelooper.lv2/wscript b/trunk/src/LV2/gx_livelooper.lv2/wscript
index 2447b1c1..023beef8 100644
--- a/trunk/src/LV2/gx_livelooper.lv2/wscript
+++ b/trunk/src/LV2/gx_livelooper.lv2/wscript
@@ -11,7 +11,7 @@ def build(bld):
         lv2_base = lv2_base,
         source   = ['gx_livelooper.cpp'],
         includes = ['../faust','./', '../DSP'],
-        use = ['LV2CORE','GLIBMM', 'SNDFILE'],
+        use = ['LV2CORE', 'SNDFILE'],
         )
     bld.lv2_gui(
         lv2_base = lv2_base,
