--- a/love.desktop	2024-11-22 21:39:59.329932099 -0300
+++ b/love.desktop	2024-11-22 21:44:24.377810132 -0300
@@ -1,10 +1,7 @@
 [Desktop Entry]
 Name=Journey to the Center of Hawkthorne
 Comment=A 2D platformer based on Community's Digital Estate Planning
-MimeType=application/x-love-game;
-Exec=hawkthorne %f
+Exec=Hawkthorne.sh
 Type=Application
 Categories=Game;
-Terminal=false
-Icon=hawkthorne
-NoDisplay=true
+Icon=com.projecthawkthorne.Hawkthorne
