diff --git a/src/org.hdfgroup.hdfview/hdf/view/DefaultFileFilter.java b/src/org.hdfgroup.hdfview/hdf/view/DefaultFileFilter.java
index 2c3a3a7..5c172c4 100644
--- a/src/org.hdfgroup.hdfview/hdf/view/DefaultFileFilter.java
+++ b/src/org.hdfgroup.hdfview/hdf/view/DefaultFileFilter.java
@@ -123,7 +123,10 @@ public class DefaultFileFilter {
         String extString = "";
 
         while (extensions.hasMoreElements()) {
-            extString += "*." + extensions.nextElement() + ";";
+            extString += "*." + extensions.nextElement();
+            if (extensions.hasMoreElements()) {
+                extString += ";";
+            }
         }
 
         return extString;
