diff --git a/activity/activity.info b/activity/activity.info
index 4c1a510..2aaade6 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,7 +2,16 @@
 name = HelloWorld
 activity_version = 7
 bundle_id = org.sugarlabs.HelloWorld
+release_date = 2020-03-12
 exec = sugar-activity3 activity.HelloWorldActivity
 icon = activity-helloworld
 license = GPLv2+
+metadata_license = CC0-1.0
 repository = https://github.com/sugarlabs/hello-world.git
+summary = A demo Sugar application
+description = HelloWorld is a demo application used in tutorials for porting Sugar applications to Flatpak.
+url = https://github.com/sugarlabs/hello-world
+tags = Education
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://i.imgur.com/N1uXt6S.png

