diff --git a/activity.py b/activity.py
index f6f1634..9a9038f 100644
--- a/activity.py
+++ b/activity.py
@@ -30,13 +30,16 @@ from sugar3.activity.widgets import StopButton
 from sugar3.activity.widgets import ShareButton
 from sugar3.activity.widgets import DescriptionItem
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
 
-class HelloWorldActivity(activity.Activity):
+
+class HelloWorldActivity(SugarCompatibleActivity):
     """HelloWorldActivity class as specified in activity.info"""
 
     def __init__(self, handle):
         """Set up the HelloWorld activity."""
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         # we do not have collaboration features
         # make the share option insensitive
@@ -45,22 +48,10 @@ class HelloWorldActivity(activity.Activity):
         # toolbar with the new toolbar redesign
         toolbar_box = ToolbarBox()
 
-        activity_button = ActivityButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, 0)
         activity_button.show()
 
-        title_entry = TitleEntry(self)
-        toolbar_box.toolbar.insert(title_entry, -1)
-        title_entry.show()
-
-        description_item = DescriptionItem(self)
-        toolbar_box.toolbar.insert(description_item, -1)
-        description_item.show()
-
-        share_button = ShareButton(self)
-        toolbar_box.toolbar.insert(share_button, -1)
-        share_button.show()
-
         separator = Gtk.SeparatorToolItem()
         separator.props.draw = False
         separator.set_expand(True)

