--- a/setup.py	2020-02-07 23:32:40.000000000 +0100
+++ b/setup.py	2020-02-16 20:01:41.435479073 +0100
@@ -16,7 +16,7 @@
 if '--with-system-hidapi' in sys.argv:
     sys.argv.remove('--with-system-hidapi')
     system_hidapi = 1
-    hidapi_include = '/usr/include/hidapi'
+    hidapi_include = '/app/include/hidapi'
 
 if sys.platform.startswith('linux'):
     modules = []
@@ -33,7 +33,7 @@
         modules.append(
             Extension('hid',
                 sources = src,
-                include_dirs = [hidapi_include, '/usr/include/libusb-1.0'],
+                include_dirs = [hidapi_include, '/app/include/libusb-1.0'],
                 libraries = libs,
             )
         )
