diff --git a/io/hpmud/model.c b/io/hpmud/model.c
index 4ea8990..72c630b 100644
--- a/io/hpmud/model.c
+++ b/io/hpmud/model.c
@@ -420,7 +420,10 @@ static int ParseFile(char *datFile, char *model, char *attr, int attrSize, int *
          strncpy(section, rcbuf+1, sizeof(section)); /* found new section */
          n = strlen(section);
          section[n-2]=0; /* remove ']' and CR */
-         if (strcasecmp(model, section) == 0)
+         if (strcasecmp(model, section) == 0 ||
+            (section[0] == 'h' && section[1] == 'p' &&
+             section[2] == '_' &&
+             strcasecmp(model, section + 3) == 0))
          {
             /* Found model match. */
             *bytes_read = ResolveAttributes(fp, attr, attrSize); 
