diff -Naur hplip-3.19.6.orig/configure.in hplip-3.19.6/configure.in
--- hplip-3.19.6.orig/configure.in	2019-06-26 15:07:21.000000000 +0200
+++ hplip-3.19.6/configure.in	2019-08-12 14:37:37.281939838 +0200
@@ -27,8 +27,8 @@
 
 #AC_PREREQ(2.59)
 AC_INIT([HP Linux Imaging and Printing], [3.19.6], [3.19.6], [hplip])
-#AM_INIT_AUTOMAKE([1.9 foreign])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([1.9 foreign])
+#AM_INIT_AUTOMAKE
 AC_DISABLE_STATIC
 
 # Checks for programs.
