diff --git a/io/hpmud/musb.c b/io/hpmud/musb.c
index 755493e..6d9c7fc 100644
--- a/io/hpmud/musb.c
+++ b/io/hpmud/musb.c
@@ -1454,7 +1454,7 @@ enum HPMUD_RESULT __attribute__ ((visibility ("hidden"))) musb_raw_channel_write
                     BUG("unable to write data %s: %d second io timeout\n", msp->device[pc->dindex].uri, sec_timeout);
             }
             else
-                BUG("unable to write data (len = %d) %s: %m\n", msp->device[pc->dindex].uri, len);
+                BUG("unable to write data (len = %d) %s: %m\n", len, msp->device[pc->dindex].uri);
             goto bugout;
         }
 		if(len == 0 && size > 0)
