diff -up hplip-3.15.2/base/logger.py.silence-ioerror hplip-3.15.2/base/logger.py
--- hplip-3.15.2/base/logger.py.silence-ioerror	2015-03-16 16:48:12.750337907 +0000
+++ hplip-3.15.2/base/logger.py	2015-03-16 16:48:34.850253564 +0000
@@ -199,6 +199,8 @@ class Logger(object):
                         out.write('\n')
 
                     out.flush()
+                except IOError:
+                    pass
                 finally:
                     self._lock.release()
 
