diff -up hplip-3.16.5/scan.py.sourceoption hplip-3.16.5/scan.py
--- hplip-3.16.5/scan.py.sourceoption	2016-06-22 10:25:52.742799941 +0200
+++ hplip-3.16.5/scan.py	2016-06-22 10:34:51.945766487 +0200
@@ -658,6 +658,7 @@ try:
                 sys.exit(1)
         except:
             log.error("Failed to get the source from device.")
+            sys.exit(1)
 
         #check if device has only ADF
         if len(source_option) == 1 and 'ADF' in source_option:
