diff -up hplip-3.18.6/base/validation.py.keyserver hplip-3.18.6/base/validation.py
--- hplip-3.18.6/base/validation.py.keyserver	2018-11-13 17:35:38.085322569 +0100
+++ hplip-3.18.6/base/validation.py	2018-11-13 17:36:18.026991983 +0100
@@ -42,7 +42,7 @@ class DigiSign_Verification(object):
 
 
 class GPG_Verification(DigiSign_Verification):
-    def __init__(self, pgp_site = 'pgp.mit.edu', key = 0x4ABA2F66DBD5A95894910E0673D770CDA59047B9):
+    def __init__(self, pgp_site = 'sks-keyservers.net', key = 0x4ABA2F66DBD5A95894910E0673D770CDA59047B9):
         self.__pgp_site = pgp_site
         self.__key = key
         self.__gpg = utils.which('gpg',True)
