diff -up hplip-3.18.12/scan.py.scan-tmp hplip-3.18.12/scan.py
--- hplip-3.18.12/scan.py.scan-tmp	2019-01-08 12:48:20.194815147 +0100
+++ hplip-3.18.12/scan.py	2019-01-08 12:50:47.942620000 +0100
@@ -1887,7 +1887,7 @@ try:
         if ('editor' in dest or 'viewer' in dest or 'email' in dest or 'print' in dest) \
             and not file_saved:
 
-            output_fd, output = utils.make_temp_file(suffix='.png')
+            output_fd, output = utils.make_temp_file(dir='/var/tmp', suffix='.png')
             try:
                 '''pyPlatform = platform.python_version()
                 num = pyPlatform.split('.')
diff -up hplip-3.18.12/scan/sane.py.scan-tmp hplip-3.18.12/scan/sane.py
--- hplip-3.18.12/scan/sane.py.scan-tmp	2018-12-03 08:10:22.000000000 +0100
+++ hplip-3.18.12/scan/sane.py	2019-01-08 12:48:20.194815147 +0100
@@ -425,7 +425,8 @@ class ScanThread(threading.Thread):
         self.dev = device
         self.update_queue = update_queue
         self.event_queue = event_queue
-        self.buffer_fd, self.buffer_path = utils.make_temp_file(prefix='hpscan')
+        self.buffer_fd, self.buffer_path = utils.make_temp_file(dir='/var/tmp',
+                                                                prefix='hpscan')
         self.buffer = os.fdopen(self.buffer_fd, "w+b")
         self.format = -1
         self.format_name = ''
