diff -Naur hplip-3.19.11.orig/configure.in hplip-3.19.11/configure.in
--- hplip-3.19.11.orig/configure.in	2019-11-04 12:03:35.000000000 +0100
+++ hplip-3.19.11/configure.in	2019-11-30 17:36:17.908657332 +0100
@@ -28,7 +28,7 @@
 #AC_PREREQ(2.59)
 AC_INIT([HP Linux Imaging and Printing], [3.19.11], [3.19.11], [hplip])
 #AM_INIT_AUTOMAKE([1.9 foreign])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 AC_DISABLE_STATIC
 
 # Checks for programs.
