diff -up hplip-3.9.8/ip/xfax.c.no-asm hplip-3.9.8/ip/xfax.c
--- hplip-3.9.8/ip/xfax.c.no-asm	2009-08-04 22:37:28.000000000 +0100
+++ hplip-3.9.8/ip/xfax.c	2009-08-25 14:44:04.502202418 +0100
@@ -1680,19 +1680,6 @@ static BOOL bits_flush_to_eol (
                  & ((1u<<n_bits) - 1u);                                      \
 }
 
-#if 0
-
-#define BITS_LOAD(g, num_bits, par_result) {                                 \
-    int n_bits = (int)(num_bits);                                            \
-                                                                             \
-    par_result = g->gb_cache;                                                \
-                                                                             \
-    asm ("extract %1,%2,%0"                                                  \
-        : "=d" (par_result)                                                  \
-        : "dI" (g->gb_cache_cnt - n_bits), "dI" (n_bits), "0" (par_result)); \
-}
-
-#endif
 
 
 
diff -up hplip-3.9.8/scan/sane/common.h.no-asm hplip-3.9.8/scan/sane/common.h
--- hplip-3.9.8/scan/sane/common.h.no-asm	2009-08-25 16:35:14.577203106 +0100
+++ hplip-3.9.8/scan/sane/common.h	2009-08-25 16:35:29.948216281 +0100
@@ -53,8 +53,6 @@
 
 #define BACKEND_NAME hpaio
 
-#define BREAKPOINT __asm( "int3" )
-
 #define OK 1
 #define ERROR 0
 #define MAX_LIST_SIZE 32
diff -up hplip-3.9.8/scan/sane/hpaio.c.no-asm hplip-3.9.8/scan/sane/hpaio.c
