diff -Naur hunspell-1.7.0.orig/src/tools/hunspell.cxx hunspell-1.7.0/src/tools/hunspell.cxx
--- hunspell-1.7.0.orig/src/tools/hunspell.cxx	2018-11-12 21:38:56.000000000 +0100
+++ hunspell-1.7.0/src/tools/hunspell.cxx	2019-09-01 16:25:53.924051520 +0200
@@ -116,6 +116,9 @@
 #include "../parsers/odfparser.hxx"
 
 #define LIBDIR                \
+  "/app/share/hunspell:"      \
+  "/app/share/myspell:"       \
+  "/app/share/myspell/dicts:" \
   "/usr/share/hunspell:"      \
   "/usr/share/myspell:"       \
   "/usr/share/myspell/dicts:" \
