--- a/CMakeLists.txt	2024-03-12 13:58:23.848152416 -0500
+++ b/CMakeLists.txt	2024-03-12 15:12:53.260812789 -0500
@@ -474,10 +474,6 @@
         DESTINATION share/icons/hicolor/512x512/apps/
     )
     install(
-        FILES icons/1024x1024/org.photoqt.PreviewQt.png
-        DESTINATION share/icons/hicolor/1024x1024/apps/
-    )
-    install(
         FILES org.photoqt.PreviewQt.metainfo.xml
         DESTINATION share/metainfo/
     )
