--- ./CMakeLists.txt
+++ ./CMakeLists.txt
@@ -192,7 +192,7 @@
 install (FILES ${assets}
   DESTINATION ${DATADIR}/notes-up/)
 install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/data/com.github.philip-scott.notes-up.svg"
-  DESTINATION ${DATADIR}/pixmaps/)
+  DESTINATION share/icons/hicolor/scalable/apps)
 
 # App DATA
 #configure_file_translation (io.github.notes-up.xml.in ${CMAKE_CURRENT_BINARY_DIR}/io.github.notes-up.xml ${CMAKE_SOURCE_DIR}/po/)

