--- kdeconnect-1/indicator/deviceindicator.cpp	2018-01-29 16:57:11.112055312 -0200
+++ kdeconnect-1/indicator/deviceindicator.cpp	2018-01-29 16:56:49.012153249 -0200
@@ -66,7 +66,7 @@
 #ifdef Q_OS_WIN
     setIcon(QIcon(QStandardPaths::locate(QStandardPaths::AppDataLocation, "icons/hicolor/scalable/status/"+device->iconName()+".svg")));
 #else
-    setIcon(QIcon::fromTheme(device->iconName()));
+    setIcon(QIcon::fromTheme("org.kde.kdeconnect"+device->iconName()));
 #endif
 
     connect(device, SIGNAL(nameChanged(QString)), this, SLOT(setText(QString)));

