From cbf443482b8f4996a2f68bea6e959cbf459a6057 Mon Sep 17 00:00:00 2001
Message-Id: <cbf443482b8f4996a2f68bea6e959cbf459a6057.1671457097.git.matthias@mailaender.name>
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Mon, 19 Dec 2022 14:37:41 +0100
Subject: [PATCH 1/2] This is actually 64x64.

---
 cutepeaks.pro | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cutepeaks.pro b/cutepeaks.pro
index ac2357b..3a78334 100644
--- a/cutepeaks.pro
+++ b/cutepeaks.pro
@@ -11,10 +11,10 @@ linux-g++ {
 
     desktop.path = $$PREFIX/share/applications/
     desktop.files += CutePeaks.desktop
-    icon512.path = $$PREFIX/share/icons/hicolor/512x512/apps
-    icon512.files += cutepeaks.png
+    icon64.path = $$PREFIX/share/icons/hicolor/64x64/apps
+    icon64.files += cutepeaks.png
 
-    INSTALLS += icon512
+    INSTALLS += icon64
     INSTALLS += desktop
     INSTALLS += target
 }
-- 
2.39.0

