diff --git a/redist/org.goldendict.GoldenDict.desktop b/redist/org.goldendict.GoldenDict.desktop
index 2eb159b..ea135da 100755
--- a/redist/org.goldendict.GoldenDict.desktop
+++ b/redist/org.goldendict.GoldenDict.desktop
@@ -5,5 +5,5 @@ Categories=Office;Dictionary;Education;Qt;
 Name=GoldenDict
 GenericName=Multiformat Dictionary
 Comment=A feature-rich dictionary lookup program
-Icon=goldendict
+Icon=org.goldendict.GoldenDict
 Exec=goldendict
