--- a/CMakeLists.txt	2023-12-14 16:30:06.601471417 -0600
+++ b/CMakeLists.txt	2023-12-14 23:05:24.060241964 -0600
@@ -569,10 +569,6 @@
         DESTINATION share/icons/hicolor/512x512/apps/
     )
     install(
-        FILES icons/1024x1024/org.photoqt.PhotoQt.png
-        DESTINATION share/icons/hicolor/1024x1024/apps/
-    )
-    install(
         FILES org.photoqt.PhotoQt.metainfo.xml
         DESTINATION share/metainfo/
     )
