--- a/src/mednaffe.c	2021-02-03 21:48:35.000000000 +0100
+++ b/src/mednaffe.c	2021-05-20 02:44:43.429379933 +0200
@@ -45,7 +45,7 @@
 {
   MednaffeApp * self = (MednaffeApp*) g_object_new (mednaffe_app_get_type(),
                                                     "application-id",
-                                                    "com.github.mednaffe",
+                                                    "com.github.AmatCoder.mednaffe",
                                                     "flags",
                                                     G_APPLICATION_FLAGS_NONE,
                                                     NULL);
