diff -Nur ImageMagick-6.9.9-3.orig/wand/MagickWand-config.in ImageMagick-6.9.9-3/wand/MagickWand-config.in
--- ImageMagick-6.9.9-3.orig/wand/MagickWand-config.in	2017-07-24 06:37:55.000000000 -0700
+++ ImageMagick-6.9.9-3/wand/MagickWand-config.in	2017-07-27 15:17:55.343675457 -0700
@@ -38,19 +38,19 @@
       echo '@PACKAGE_VERSION@ Q@QUANTUM_DEPTH@ @MAGICK_HDRI@'
       ;;
     --cflags)
-      @PKG_CONFIG@ --cflags MagickWand
+      pkg-config --cflags MagickWand
       ;;
     --cxxflags)
-      @PKG_CONFIG@ --cflags MagickWand
+      pkg-config --cflags MagickWand
       ;;
     --cppflags)
-      @PKG_CONFIG@ --cflags MagickWand
+      pkg-config --cflags MagickWand
       ;;
     --ldflags)
-      @PKG_CONFIG@ --libs MagickWand
+      pkg-config --libs MagickWand
       ;;
     --libs)
-      @PKG_CONFIG@ --libs MagickWand
+      pkg-config --libs MagickWand
       ;;
     *)
       echo "${usage}" 1>&2
