diff --git a/helpwidget.py b/helpwidget.py
index 12b9a38..0149abf 100644
--- a/helpwidget.py
+++ b/helpwidget.py
@@ -34,6 +34,8 @@ import board
 from anim import Anim
 from gridwidget import BoardDrawer, RemovalDrawer, WinDrawer
 
+from sugarapp.helpers import PrimaryMonitor
+
 if 'SUGAR_BUNDLE_PATH' in os.environ:
     from sugar3.graphics import style
     _DEFAULT_SPACING = style.DEFAULT_SPACING
@@ -42,7 +44,7 @@ if 'SUGAR_BUNDLE_PATH' in os.environ:
     _TOOLBAR_COLOR = tuple(style.COLOR_TOOLBAR_GREY.get_rgba()[:3])
     _PREVIEW_TRIM_WIDTH = style.GRID_CELL_SIZE * 4
 
-    aspect_ratio = Gdk.Screen.width() * 100 // Gdk.Screen.height()
+    aspect_ratio = PrimaryMonitor.width() * 100 // PrimaryMonitor.height()
     if aspect_ratio > 159:
         _PREVIEW_TRIM_WIDTH = style.GRID_CELL_SIZE * 11
 else:
diff --git a/implodeactivity.py b/implodeactivity.py
index 84467a5..3e839e1 100644
--- a/implodeactivity.py
+++ b/implodeactivity.py
@@ -42,6 +42,7 @@ from gi.repository import Gdk
 
 from keymap import KEY_MAP
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import SugarCompatibleActivity
 from sugarapp.widgets import ExtendedActivityToolbarButton
 
@@ -227,7 +228,7 @@ class ImplodeActivity(SugarCompatibleActivity):
         self._seps[-1].show()
 
     def _configure_cb(self, event=None):
-        if Gdk.Screen.width() < Gdk.Screen.height():
+        if PrimaryMonitor.width() < PrimaryMonitor.height():
             _logger.debug('TRUE')
             hide = True
         else:
@@ -246,8 +247,8 @@ class _DialogWindow(Gtk.Window):
         super(_DialogWindow, self).__init__()
 
         self.set_border_width(style.LINE_WIDTH)
-        width = Gdk.Screen.width() // 2
-        height = Gdk.Screen.height() // 2
+        width = PrimaryMonitor.width() // 2
+        height = PrimaryMonitor.height() // 2
         self.set_size_request(width, height)
         self.set_position(Gtk.WindowPosition.CENTER_ALWAYS)
         self.set_decorated(False)
@@ -289,8 +290,8 @@ class _HelpWindow(_DialogWindow):
         super(_HelpWindow, self).__init__('toolbar-help', _("Help"))
 
         offset = style.GRID_CELL_SIZE
-        width = Gdk.Screen.width() - offset * 2
-        height = Gdk.Screen.height() - offset * 2
+        width = PrimaryMonitor.width() - offset * 2
+        height = PrimaryMonitor.height() - offset * 2
         self.set_size_request(width, height)
 
         self._help_widget = HelpWidget(self._icon_file)
