diff --git a/activity/activity.info b/activity/activity.info
index b05bdd8..3d6c3dc 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,12 +1,18 @@
 [Activity]
 name = Implode
 activity_version = 20
-bundle_id = com.jotaro.ImplodeActivity
+release_date = 2019-10-15
+bundle_id = org.sugarlabs.ImplodeActivity
 icon = activity-implode
 exec = sugar-activity3 implodeactivity.ImplodeActivity
 license = GPLv2+
+metadata_license = CC0-1.0
 show_launcher = yes
-repository = https://github.com/quozl/implode-activity.git
-summary = Implode blocks of the same colour until they are all gone.
+repository = https://github.com/sugarlabs/implode-activity.git
+summary = Implode blocks of the same colour until they are all gone
+description = Implode is a logic game based on the "falling block" model of Tetris. The game starts with a grid partially filled with blocks. The player makes a move by removing adjacent blocks of the same color in groups of three or more.
 url = https://help.sugarlabs.org/en/implode.html
 tags = Game
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://help.sugarlabs.org/en/_images/implode-image1.png

