--- indicator-kdeconnect/src/main.vala	2017-07-29 02:53:53.492812341 -0300
+++ indicator-kdeconnect/src/main.vala	2017-07-29 13:28:17.784019292 -0300
@@ -12,7 +12,7 @@
         private FirstTimeWizard ftw;
 
         public Application () {
-            Object (application_id: "com.bajoja.indicator-kdeconnect",
+            Object (application_id: "com.github.bajoja.indicator-kdeconnect",
                     flags: ApplicationFlags.FLAGS_NONE);
 
         }
          
--- indicator-kdeconnect/src/StartupManager.vala	2017-07-29 13:59:59.751564867 -0300
+++ indicator-kdeconnect/src/StartupManager.vala	2017-07-29 14:00:59.863372685 -0300
@@ -5,7 +5,7 @@
  */
 namespace KDEConnectIndicator {
     public class StartupManager {
-        private const string desktop_file = "indicator-kdeconnect.desktop";
+        private const string desktop_file = "com.github.bajoja.indicator-kdeconnect.desktop";
         private string prefix_dir;
         private string autostart_folder;
         private File startup_file;
         
--- indicator-kdeconnect/src/settings/Settings.vala	2017-07-29 02:53:53.492812341 -0300
+++ indicator-kdeconnect/src/settings/Settings.vala	2017-07-29 14:27:26.702453963 -0300
@@ -22,12 +22,12 @@
 
 
 		public SettingsDialog () {
-			Object (application_id: "com.bajoja.indicator-kdeconnect-settings",
+			Object (application_id: "com.github.bajoja.indicator-kdeconnect.settings",
 				flags: ApplicationFlags.FLAGS_NONE);
 		}
 
 		protected override void activate () {
 			this.settings = new GLib.Settings("com.bajoja.indicator-kdeconnect");
 			create_window ();
 		}
 

--- indicator-kdeconnect/src/DeviceIndicator.vala	2017-07-30 18:35:08.073869901 -0300
+++ indicator-kdeconnect/src/DeviceIndicator.vala	2017-07-30 22:45:31.176073120 -0300
@@ -28,7 +28,7 @@
             menu = new Gtk.Menu ();
 
             indicator = new AppIndicator.Indicator (path,
-                    				    device.icon,
+                    				    "com.github.bajoja.indicator-kdeconnect." + device.icon + "-symbolic",
                                                     AppIndicator.IndicatorCategory.HARDWARE);
 
             name_item = new Gtk.MenuItem ();
@@ -134,7 +134,7 @@
             	try{
 		    Process.spawn_async (null,
 		    			new string[]{
-				        "/usr/share/indicator-kdeconnect/Sms.py",
+				        "/app/share/indicator-kdeconnect/Sms.py",
 					"-d",
 					device.id},
 				        null,
@@ -216,7 +216,7 @@
         }
 
         private void update_icon_item() {
-	    indicator.set_icon_full (device.icon, "");
+	    indicator.set_icon_full ("com.github.bajoja.indicator-kdeconnect." + device.icon + "-symbolic", "");
 	}
         
         private void update_battery_item () {

--- indicator-kdeconnect/src/sms/Sms.py	2017-08-09 20:19:50.201059304 -0300
+++ indicator-kdeconnect/src/sms/Sms.py	2017-08-12 13:35:09.928443571 -0300
@@ -27,7 +27,7 @@
 
 _ = gettext.gettext
 locale.setlocale(locale.LC_ALL, '')
-gettext.bindtextdomain('indicator-kdeconnect', '/usr/share/locale')
+gettext.bindtextdomain('indicator-kdeconnect', '/app/share/locale')
 gettext.textdomain('indicator-kdeconnect')
 
 parser = argparse.ArgumentParser(

--- indicator-kdeconnect/src/contractor/Contractor.vala	2017-08-09 20:19:50.201059304 -0300
+++ indicator-kdeconnect/src/contractor/Contractor.vala	2017-08-12 12:49:50.547395390 -0300
@@ -25,7 +25,7 @@
 		private SList<Device> device_list;
 
 		public SendDialog () {
-			Object (application_id: "com.bajoja.kdeconnect-send",
+			Object (application_id: "com.github.bajoja.kdeconnect-indicator.kdeconnect-send",
 				flags: ApplicationFlags.HANDLES_OPEN);
 		}

--- indicator-kdeconnect/data/desktop/indicator-kdeconnect-settings.desktop	2017-07-29 02:07:28.573346739 -0300
+++ indicator-kdeconnect/data/desktop/indicator-kdeconnect-settings.desktop	2017-07-30 20:29:46.927678806 -0300
@@ -5,5 +5,5 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Icon=kdeconnect
+Icon=com.github.bajoja.indicator-kdeconnect
 Categories=Network;System;

