diff --git a/audio/midi_drivers/MT32EmuMidiDriver.cpp b/audio/midi_drivers/MT32EmuMidiDriver.cpp
index 9947891c1..3d042c46b 100644
--- a/audio/midi_drivers/MT32EmuMidiDriver.cpp
+++ b/audio/midi_drivers/MT32EmuMidiDriver.cpp
@@ -39,23 +39,11 @@ const MidiDriver::MidiDriverDesc MT32EmuMidiDriver::desc
 static bool openROMFile(
 		FileStream& file, const std::string& filename, bool writable) {
 	std::string basedir;
-	if (!writable) {
-		// May be in bundle.
-		if (is_system_path_defined("<BUNDLE>")) {
-			basedir = "<BUNDLE>/" + filename;
-			if (file.open(get_system_path(basedir).c_str())) {
-				return true;
-			}
-		}
-		// Now try data dir.
-		basedir = "<DATA>/" + filename;
-		if (file.open(get_system_path(basedir).c_str())) {
-			return true;
-		}
-		// We now fall back to a writable data dir, as the emulator
-		// may have written something there.
-	}
-	basedir = "<SAVEHOME>/data/" + filename;
+	basedir = std::getenv("XDG_DATA_HOME");
+ basedir.append("/mt32emu/");
+ basedir.append(filename);
+	std::cout << "Trying to load MT-32 ROM file:" << std::endl;
+	std::cout << basedir << std::endl;
 	return file.open(get_system_path(basedir).c_str());
 }
 
