diff --git a/configure.ac b/configure.ac
index b99729518..fc490f487 100644
--- a/configure.ac
+++ b/configure.ac
@@ -955,7 +955,7 @@ AC_ARG_WITH([desktopdir],
 	esac])
 AC_SUBST([DESKTOPDIR])
 
-ICONDIR="${datadir}/icons"
+ICONDIR="${datadir}/icons/hicolor/512x512/apps"
 AC_ARG_WITH([icondir],
 	AS_HELP_STRING([--with-icondir=DIR],[change icon directory]),
 	[case "${withval}" in
@@ -969,6 +969,23 @@ AC_ARG_WITH([icondir],
 	esac])
 AC_SUBST([ICONDIR])
 
+SVGDIR="${datadir}/icons/hicolor/scalable/apps"
+AC_ARG_WITH([svgdir],
+	AS_HELP_STRING([--with-svgdir=DIR],[change svg directory]),
+	[case "${withval}" in
+		yes)
+		;;
+		no)
+		;;
+		*)
+			SVGDIR="${withval}"
+		;;
+	esac])
+AC_SUBST([SVGDIR])
+
+METADIR="${datadir}/metainfo"
+AC_SUBST([METADIR])
+
 # ---------------------------------------------------------------------
 # Workaround for 'ar' warning in libtool
 # ---------------------------------------------------------------------
diff --git a/desktop/Makefile.am b/desktop/Makefile.am
index 65238af55..d604ac205 100644
--- a/desktop/Makefile.am
+++ b/desktop/Makefile.am
@@ -1,10 +1,16 @@
 icondir = $(ICONDIR)
 desktopdir = $(DESKTOPDIR)
+svgdir = $(SVGDIR)
+metadir = $(METADIR)
 
 if BUILD_STUDIO
 dist_desktop_DATA = info.exult.exult.desktop info.exult.studio.desktop
-dist_icon_DATA = info.exult.exult.png info.exult.studio.png info.exult.exult.svg info.exult.studio.svg
+dist_icon_DATA = info.exult.exult.png info.exult.studio.png
+dist_svg_DATA = info.exult.exult.svg info.exult.studio.svg
+dist_meta_DATA = info.exult.exult.metainfo.xml
 else
 dist_desktop_DATA = info.exult.exult.desktop
-dist_icon_DATA = info.exult.exult.png info.exult.exult.svg
+dist_icon_DATA = info.exult.exult.png
+dist_svg_DATA = info.exult.exult.svg
+dist_meta_DATA = info.exult.exult.metainfo.xml
 endif
