Index: library/alsa/CMakeLists.txt
===================================================================
--- library/alsa/CMakeLists.txt	(revisión: 622)
+++ library/alsa/CMakeLists.txt	(copia de trabajo)
@@ -34,6 +34,7 @@
     ../include/drumstick/playthread.h
     ../include/drumstick/sequencererror.h
     ../include/drumstick/subscription.h
+    ../include/drumstick/macros.h
 )
 
 set(drumstick-alsa_SRCS
