diff --git i/vesc_tool.pro w/vesc_tool.pro
index 0fe4872..4ac3d9a 100644
--- i/vesc_tool.pro
+++ w/vesc_tool.pro
@@ -100,7 +100,6 @@ contains(DEFINES, HAS_GAMEPAD) {
 android: QT += androidextras
 
 android: TARGET = vesc_tool
-!android: TARGET = vesc_tool_$$VT_VERSION
 
 ANDROID_VERSION = 1
 
@@ -258,3 +257,31 @@ DISTFILES += \
     android/src/com/vedder/vesc/Utils.java
 
 ANDROID_PACKAGE_SOURCE_DIR = $$PWD/android
+
+unix {
+    isEmpty(PREFIX) {
+        PREFIX = /usr/local
+    }
+    BINDIR  = $$PREFIX/bin
+    DATADIR = $$PREFIX/share
+    CONFDIR = /etc
+    isEmpty(VARDIR) {
+        VARDIR  = /var/lib/$${TARGET}
+    }
+
+    INSTALLS += target config desktop icon32 vardir
+
+    target.path = $$BINDIR
+
+    config.path = $$CONFDIR
+    config.files += $${TARGET}.conf
+
+    desktop.path = $$DATADIR/applications
+    desktop.files += $${TARGET}.desktop
+
+    icon32.path = $$DATADIR/icons/hicolor/32x32/apps
+    icon32.files += ./data/32x32/$${TARGET}.png
+
+    vardir.path = $$VARDIR
+    vardir.commands = true
+}
