Binary files a/.git/index and b/.git/index differ
diff -ruN a/src/cpp/desktop/CMakeLists.txt b/src/cpp/desktop/CMakeLists.txt
--- a/src/cpp/desktop/CMakeLists.txt	2021-10-23 21:39:52.500213658 +0200
+++ b/src/cpp/desktop/CMakeLists.txt	2021-10-23 21:58:31.185247774 +0200
@@ -614,11 +614,11 @@
    configure_file (${RSTUDIO_FREEDESKTOP_SOURCE_DIR}/rstudio.desktop.in
                    ${RSTUDIO_FREEDESKTOP_BINARY_DIR}/rstudio.desktop)
    install(FILES ${RSTUDIO_FREEDESKTOP_BINARY_DIR}/rstudio.desktop
-           DESTINATION /usr/share/applications)
+           DESTINATION /app/share/applications)
 
    # mime types
    install(FILES ${RSTUDIO_FREEDESKTOP_SOURCE_DIR}/rstudio.xml
-           DESTINATION /usr/share/mime/packages)
+           DESTINATION /app/share/mime/packages)
 
    # define icon dirs
    set(RSTUDIO_ICONS_16 ${RSTUDIO_FREEDESKTOP_SOURCE_DIR}/icons/16x16)
@@ -629,40 +629,40 @@
 
    # application icon
    install(FILES ${RSTUDIO_ICONS_32}/rstudio.png
-           DESTINATION /usr/share/pixmaps)
+           DESTINATION /app/share/pixmaps)
    install(FILES ${RSTUDIO_ICONS_16}/rstudio.png
-           DESTINATION /usr/share/icons/hicolor/16x16/apps)
+           DESTINATION /app/share/icons/hicolor/16x16/apps)
    install(FILES ${RSTUDIO_ICONS_24}/rstudio.png
-           DESTINATION /usr/share/icons/hicolor/24x24/apps)
+           DESTINATION /app/share/icons/hicolor/24x24/apps)
    install(FILES ${RSTUDIO_ICONS_32}/rstudio.png
-           DESTINATION /usr/share/icons/hicolor/32x32/apps)
+           DESTINATION /app/share/icons/hicolor/32x32/apps)
    install(FILES ${RSTUDIO_ICONS_48}/rstudio.png
-           DESTINATION /usr/share/icons/hicolor/48x48/apps)
+           DESTINATION /app/share/icons/hicolor/48x48/apps)
    install(FILES ${RSTUDIO_ICONS_256}/rstudio.png
-           DESTINATION /usr/share/icons/hicolor/256x256/apps)
+           DESTINATION /app/share/icons/hicolor/256x256/apps)
 
    # .RData icon
    install(FILES ${RSTUDIO_ICONS_16}/application-x-r-data.png
-           DESTINATION /usr/share/icons/hicolor/16x16/mimetypes)
+           DESTINATION /app/share/icons/hicolor/16x16/mimetypes)
    install(FILES ${RSTUDIO_ICONS_24}/application-x-r-data.png
-           DESTINATION /usr/share/icons/hicolor/24x24/mimetypes)
+           DESTINATION /app/share/icons/hicolor/24x24/mimetypes)
    install(FILES ${RSTUDIO_ICONS_32}/application-x-r-data.png
-           DESTINATION /usr/share/icons/hicolor/32x32/mimetypes)
+           DESTINATION /app/share/icons/hicolor/32x32/mimetypes)
    install(FILES ${RSTUDIO_ICONS_48}/application-x-r-data.png
-           DESTINATION /usr/share/icons/hicolor/48x48/mimetypes)
+           DESTINATION /app/share/icons/hicolor/48x48/mimetypes)
    install(FILES ${RSTUDIO_ICONS_256}/application-x-r-data.png
-           DESTINATION /usr/share/icons/hicolor/256x256/mimetypes)
+           DESTINATION /app/share/icons/hicolor/256x256/mimetypes)
 
    # .Rproj icon
    install(FILES ${RSTUDIO_ICONS_16}/application-x-r-project.png
-           DESTINATION /usr/share/icons/hicolor/16x16/mimetypes)
+           DESTINATION /app/share/icons/hicolor/16x16/mimetypes)
    install(FILES ${RSTUDIO_ICONS_24}/application-x-r-project.png
-           DESTINATION /usr/share/icons/hicolor/24x24/mimetypes)
+           DESTINATION /app/share/icons/hicolor/24x24/mimetypes)
    install(FILES ${RSTUDIO_ICONS_32}/application-x-r-project.png
-           DESTINATION /usr/share/icons/hicolor/32x32/mimetypes)
+           DESTINATION /app/share/icons/hicolor/32x32/mimetypes)
    install(FILES ${RSTUDIO_ICONS_48}/application-x-r-project.png
-           DESTINATION /usr/share/icons/hicolor/48x48/mimetypes)
+           DESTINATION /app/share/icons/hicolor/48x48/mimetypes)
    install(FILES ${RSTUDIO_ICONS_256}/application-x-r-project.png
-           DESTINATION /usr/share/icons/hicolor/256x256/mimetypes)
+           DESTINATION /app/share/icons/hicolor/256x256/mimetypes)
 
 endif()
