From 000268c8ac575eaf38eff8278699d36804954db6 Mon Sep 17 00:00:00 2001
Message-Id: <000268c8ac575eaf38eff8278699d36804954db6.1653572468.git.matthias@mailaender.name>
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Thu, 26 May 2022 15:41:02 +0200
Subject: [PATCH] Fix mod installation.

---
 Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 7cdfcb18..882b96a4 100644
--- a/Makefile
+++ b/Makefile
@@ -163,9 +163,10 @@ install-metadata:
 	@sh -c '. ./packaging/linux/functions.sh; install_metadata $(DESTDIR)$(datadir) . ./engine hv $(VERSION) OpenHV 730762985772941312 ./packaging/linux ./packaging/artwork'
 
 install-data:
-	@sh -c '. ./engine/packaging/functions.sh; install_data ./engine $(DESTDIR)$(gamedir) hv'
+	@mkdir -p $(DESTDIR)$(gamedir)/mods/
+	@sh -c '. ./engine/packaging/functions.sh; install_data ./engine $(DESTDIR)$(gamedir)'
 	@rm -f "$(DESTDIR)$(gamedir)/global mix database.dat"
-	@cp -Lr mods/hv $(DESTDIR)$(gamedir)
+	@cp -Lr mods/hv $(DESTDIR)$(gamedir)/mods/
 	@mkdir -p $(DESTDIR)$(mandir)/man6/
 	@./utility.sh all --man-page > $(DESTDIR)$(mandir)/man6/openhv.6
 
-- 
2.35.3

