diff --git a/src/main.rs b/src/main.rs
index f30cc29..4834494 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -62,7 +62,7 @@ fn init_folders() {
 fn init_internationalization() -> Result<(), Box<dyn std::error::Error>> {
     gettextrs::TextDomain::new("de.schmidhuberj.DieBahn")
         .locale_category(gettextrs::LocaleCategory::LcAll)
-        .prepend("./po")
+        .prepend("/app/sharelocale")
         .init()?;
     Ok(())
 }
