diff --git a/dist/io.github.jardon.Outlet.desktop b/dist/io.github.jardon.Outlet.desktop
index da245af..9f4d5a8 100644
--- a/dist/io.github.jardon.Outlet.desktop
+++ b/dist/io.github.jardon.Outlet.desktop
@@ -3,7 +3,7 @@ Type=Application
 Version=1.0
 Name=Outlet
 Comment=App management client
-Exec=outlet
+Exec=/app/bin/outlet
 Icon=io.github.jardon.Outlet
 Terminal=false
-Categories=Utility;
\ No newline at end of file
+Categories=Utility;
