--- a/resources/linux/tev.metainfo.xml.in
+++ b/resources/linux/tev.metainfo.xml.in
@@ -5,12 +5,13 @@
     <summary>High dynamic range (HDR) image viewer for people who care about colors</summary>
     <metadata_license>CC0-1.0</metadata_license>
     <project_license>GPL-3.0-or-later</project_license>
-    <developer id="io.github.Tom94">
+    <developer id="io.github.tom94">
         <name translate="no">Thomas Müller</name>
     </developer>
-    <description>
     <content_rating type="oars-1.1" />
-    <p>High dynamic range (HDR) image viewer for people who care about colors.<br />It is</p>
+    <description>
+    <p>High dynamic range (HDR) image viewer for people who care about colors.</p>
+    <p>It is:</p>
     <ul>
         <li>Lightning fast: starts up instantly, loads hundreds of images in seconds.</li>
         <li>Accurate: understands HDR and color profiles (ICC, CICP, etc.). Displays HDR on all operating systems.</li>
@@ -48,4 +49,18 @@
         <color type="primary" scheme_preference="light">#e89df4</color>
         <color type="primary" scheme_preference="dark">#854091</color>
     </branding>
+    <releases>
+        <release version="2.5.1" date="2025-09-30">
+            <url type="details">https://github.com/Tom94/tev/releases/tag/v2.5.1</url>
+            <description>
+                <ul>
+                    <li>Support for CICP tags embedded in ICC color profiles
+                    (found in the wild as ImageMagick / Discord converts from HDR PNG images;
+                    see <a href="https://github.com/Tom94/tev/issues/351">#351</a>)</li>
+                    <li>Reintroduced shortcut for toggling LDR clipping mode: <code>U</code></li>
+                    <li>Fixed crash when loading WEBP images with embedded ICC color profiles</li>
+                </ul>
+            </description>
+        </release>
+    </releases>
 </component>
