diff --git a/gui/src-tauri/src/main.rs b/gui/src-tauri/src/main.rs
index 3d2a1b3..9d7e564 100644
--- a/gui/src-tauri/src/main.rs
+++ b/gui/src-tauri/src/main.rs
@@ -74,6 +74,11 @@ fn get_launch_path(cli: Cli) -> Option<PathBuf> {
 		return Some(path);
 	}
 
+	let path = PathBuf::from("/app/share/slimevr/");
+	if is_valid_path(&path) {
+		return Some(path);
+	}
+
 	let path = PathBuf::from("/usr/share/slimevr/");
 	if is_valid_path(&path) {
 		return Some(path);
