diff --git a/api/logic/java/JavaUtils.cpp b/api/logic/java/JavaUtils.cpp
index 1f9719a8..6d8fc793 100644
--- a/api/logic/java/JavaUtils.cpp
+++ b/api/logic/java/JavaUtils.cpp
@@ -324,10 +324,11 @@ QList<QString> JavaUtils::FindJavaPaths()
         }
     };
     // oracle RPMs
-    scanJavaDir("/usr/java");
+    scanJavaDir("/app/jre");
+    //scanJavaDir("/usr/java");
     // general locations used by distro packaging
-    scanJavaDir("/usr/lib/jvm");
-    scanJavaDir("/usr/lib32/jvm");
+    //scanJavaDir("/usr/lib/jvm");
+    //scanJavaDir("/usr/lib32/jvm");
     // javas stored in MultiMC's folder
     scanJavaDir("java");
     return javas;
