diff --git a/JUCE/modules/juce_core/native/juce_linux_Files.cpp b/JUCE/modules/juce_core/native/juce_linux_Files.cpp
index c51a97e6a..86107dbc8 100644
--- a/JUCE/modules/juce_core/native/juce_linux_Files.cpp
+++ b/JUCE/modules/juce_core/native/juce_linux_Files.cpp
@@ -122,7 +122,7 @@ File File::getSpecialLocation (const SpecialLocationType type)
         case userDesktopDirectory:            return resolveXDGFolder ("XDG_DESKTOP_DIR",   "~/Desktop");
         case userApplicationDataDirectory:    return resolveXDGFolder ("XDG_CONFIG_HOME",   "~/.config");
         case commonDocumentsDirectory:
-        case commonApplicationDataDirectory:  return File ("/opt");
+        case commonApplicationDataDirectory:  return File ("/app/extensions/Plugins/Odin2");
         case globalApplicationsDirectory:     return File ("/usr");
 
         case tempDirectory:
