diff -Naur kchmviewer-7.7.orig/src/config.cpp kchmviewer-7.7/src/config.cpp
--- kchmviewer-7.7.orig/src/config.cpp	2015-11-15 05:20:36.739480000 +0100
+++ kchmviewer-7.7/src/config.cpp	2018-12-26 05:06:45.568575843 +0100
@@ -50,7 +50,7 @@
 	m_advUseInternalEditor = settings.value( "advanced/internaleditor", true ).toBool();
 	m_advLayoutDirectionRL = settings.value( "advanced/layoutltr", false ).toBool();
 	m_advAutodetectEncoding = settings.value( "advanced/autodetectenc", false ).toBool();
-	m_advExternalEditorPath = settings.value( "advanced/editorpath", "/usr/bin/kate" ).toString();
+	m_advExternalEditorPath = settings.value( "advanced/editorpath", "kate" ).toString();
 	m_advCheckNewVersion = settings.value( "advanced/checknewver", true ).toBool();
 	m_toolbarMode = (Config::ToolbarMode) settings.value( "advanced/toolbarmode", TOOLBAR_LARGEICONSTEXT ).toInt();
 	m_lastOpenedDir = settings.value( "advanced/lastopendir", "." ).toString();
