diff --git a/ffmpeg/libavdevice/v4l.c b/ffmpeg/libavdevice/v4l.c
index 5d57a2b..619eb26 100644
--- a/ffmpeg/libavdevice/v4l.c
+++ b/ffmpeg/libavdevice/v4l.c
@@ -26,7 +26,7 @@
 #include <sys/mman.h>
 #include <sys/time.h>
 #define _LINUX_TIME_H 1
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <time.h>
 
 typedef struct {
diff --git a/src/v4l.h b/src/v4l.h
index 50bbd98..0c05199 100644
--- a/src/v4l.h
+++ b/src/v4l.h
@@ -40,7 +40,7 @@ using std::deque;
 
 #define _DEVICE_H_
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 #include "displayer.h"
 
