--- xbmc-18.0b3-Leia/tools/Linux/kodi.sh.in-	2018-10-10 21:06:03.867255720 -0700
+++ xbmc-18.0b3-Leia/tools/Linux/kodi.sh.in	2018-10-10 21:07:21.651249748 -0700
@@ -27,7 +27,8 @@
 LIBDIR="@libdir@"
 APP_BINARY=$LIBDIR/${bin_name}/@APP_BINARY@
 CRASHLOG_DIR=${CRASHLOG_DIR:-$HOME}
-KODI_DATA=${KODI_DATA:-"${HOME}/.${bin_name}"} # mapped to special://home/
+export KODI_DATA=${XDG_DATA_HOME}
+export KODI_HOME=/app/share/kodi
 WINDOWING=${WINDOWING:-"auto"}
 
 # Workaround for high CPU load with nvidia GFX
