diff --git a/src/utilssystem_unix.cpp b/src/utilssystem_unix.cpp
index 59f0a2f..99185b6 100644
--- a/src/utilssystem_unix.cpp
+++ b/src/utilssystem_unix.cpp
@@ -10,11 +10,6 @@ QString getTerminalCommand()
 #else // Linux
 	// Linux does not have a uniform way to determine the default terminal application
 	// gnome
-	QString command = execCommand("gsettings get org.gnome.desktop.default-applications.terminal exec");
-	command = command.replace('\'', "");
-	if (!command.isEmpty()) {
-		return command;
-	}
 	// fallback
 	QStringList fallbacks = QStringList() << "konsole" << "xterm";
 	foreach (const QString &fallback, fallbacks) {
