From 7f837a786b48497680f9c7462284366d310e52b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 4 Jun 2020 13:42:30 +0200
Subject: [PATCH] Fix build with Qt 5.15+

---
 src/modules/iograph/libkviiograph.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/modules/iograph/libkviiograph.cpp b/src/modules/iograph/libkviiograph.cpp
index 8e08bb427..265376231 100644
--- a/src/modules/iograph/libkviiograph.cpp
+++ b/src/modules/iograph/libkviiograph.cpp
@@ -35,6 +35,7 @@
 
 #include <QPainter>
 #include <QPaintEvent>
+#include <QPainterPath>
 
 #ifdef COMPILE_PSEUDO_TRANSPARENCY
 extern KVIRC_API QPixmap * g_pShadedChildGlobalDesktopBackground;
