diff --git a/org.kde.labplot2.appdata.xml b/org.kde.labplot2.appdata.xml
index 9a5f6a29b..a069fbc1b 100644
--- a/org.kde.labplot2.appdata.xml
+++ b/org.kde.labplot2.appdata.xml
@@ -201,4 +201,9 @@
   <provides>
     <binary>labplot2</binary>
   </provides>
+  <content_rating type="oars-1.1" />
+  <releases>
+    <release version="2.8.1" date="2020-11-13"/>
+    <release version="2.8.0" date="2020-09-16"/>
+  </releases>
 </component>
diff --git a/src/backend/cantorWorksheet/VariableParser.cpp b/src/backend/cantorWorksheet/VariableParser.cpp
index a3c9a2db9..1f240ce62 100644
--- a/src/backend/cantorWorksheet/VariableParser.cpp
+++ b/src/backend/cantorWorksheet/VariableParser.cpp
@@ -38,9 +38,9 @@ VariableParser::VariableParser(QString name, QString value)
 	PERFTRACE("parsing variable");
 	if (m_backendName.compare(QStringLiteral("Maxima"), Qt::CaseInsensitive) == 0)
 		parseMaximaValues();
-	else if (m_backendName.compare(QStringLiteral("Python 3"), Qt::CaseInsensitive) == 0)
-		parsePythonValues();
-	else if (m_backendName.compare(QStringLiteral("Python 2"), Qt::CaseInsensitive) == 0)
+	else if ( (m_backendName.compare(QStringLiteral("Python"), Qt::CaseInsensitive) == 0)
+			|| (m_backendName.compare(QStringLiteral("Python 3"), Qt::CaseInsensitive) == 0)
+			|| (m_backendName.compare(QStringLiteral("Python 2"), Qt::CaseInsensitive) == 0) )
 		parsePythonValues();
 	else if (m_backendName.compare(QStringLiteral("Sage"), Qt::CaseInsensitive) == 0)
 		parsePythonValues();
diff --git a/src/backend/gsl/parser.y b/src/backend/gsl/parser.y
index 033d814a3..0fec18082 100644
--- a/src/backend/gsl/parser.y
+++ b/src/backend/gsl/parser.y
@@ -298,6 +298,7 @@ double parse(const char* string, const char* locale) {
 
 	/* parameter for yylex */
 	res = NAN;	/* default value */
+	yynerrs = 0;	/* reset error count */
 	yyparse(&p);
 
 	pdebug("PARSER: parse() DONE (result = %g, errors = %d)\n*******************************\n", res, parse_errors());
