diff --git a/labyrinth_lib/Browser.py b/labyrinth_lib/Browser.py
index 5b5468e..3fb54fc 100644
--- a/labyrinth_lib/Browser.py
+++ b/labyrinth_lib/Browser.py
@@ -28,7 +28,7 @@ _ = gettext.gettext
 
 # Gtk stuff
 import gtk
-if os.name != 'nt':
+if os.name != 'nt' and False:
     import gconf
 import gtk.glade
 import pango
@@ -89,7 +89,7 @@ class Browser (gtk.Window):
         self.main_window = self.glade.get_widget ('MapBrowser')
 
         # set remembered size
-        if os.name != 'nt':
+        if os.name != 'nt' and False:
             self.config_client = gconf.client_get_default()
             self.config_client.add_dir ("/apps/labyrinth", gconf.CLIENT_PRELOAD_NONE)
 
@@ -105,7 +105,7 @@ class Browser (gtk.Window):
 
         view_sortable = self.view.get_model ()
         view_sortable.connect ('sort-column-changed', self.sort_column_changed_cb)
-        if os.name != 'nt':
+        if os.name != 'nt' and False:
             sort_order = self.config_client.get_int('/apps/labyrinth/map_sort_order')
             column_id = self.config_client.get_int('/apps/labyrinth/map_sort_order_column')
             view_sortable.set_sort_column_id (column_id, sort_order)
@@ -307,7 +307,7 @@ class Browser (gtk.Window):
 
         width, height = self.main_window.get_size()
 
-        if os.name != 'nt':
+        if os.name != 'nt' and False:
             self.config_client.set_int('/apps/labyrinth/width', width)
             self.config_client.set_int('/apps/labyrinth/height', height)
 
@@ -335,6 +335,6 @@ class Browser (gtk.Window):
 
     def sort_column_changed_cb (self, data):
         column_id, sort_order = data.get_sort_column_id ()
-        if os.name != 'nt':
+        if os.name != 'nt' and False:
             self.config_client.set_int('/apps/labyrinth/map_sort_order', sort_order)
             self.config_client.set_int('/apps/labyrinth/map_sort_order_column', column_id)
diff --git a/labyrinth_lib/MainWindow.py b/labyrinth_lib/MainWindow.py
index b73fb56..76be263 100644
--- a/labyrinth_lib/MainWindow.py
+++ b/labyrinth_lib/MainWindow.py
@@ -31,7 +31,7 @@ import xml.dom.minidom as dom
 import gtk
 import cairo, pangocairo
 import gobject
-if os.name != 'nt':
+if os.name != 'nt' and False:
     import gconf
 
 # Local imports
@@ -79,7 +79,7 @@ class LabyrinthWindow (gobject.GObject):
         self.MainArea.connect ("thought_selection_changed", self.thought_selected_cb)
         self.MainArea.connect ("set_focus", self.main_area_focus_cb)
         self.MainArea.connect ("set_attrs", self.attrs_cb)
-        if os.name != 'nt':
+        if os.name != 'nt' and False:
             self.MainArea.connect ("text_selection_changed", self.selection_changed_cb)
             self.config_client = gconf.client_get_default()
 
@@ -321,7 +321,7 @@ class LabyrinthWindow (gobject.GObject):
 
     def view_change_cb(self, base, activated):
         utils.use_bezier_curves = activated.get_current_value() == MMapArea.VIEW_BEZIER
-        if os.name != 'nt':
+        if os.name != 'nt' and False:
             self.config_client.set_bool('/apps/labyrinth/curves', utils.use_bezier_curves)
         self.MainArea.update_all_links()
         self.MainArea.invalidate()
