diff --git a/labyrinth b/labyrinth
index 118884f..07dcd44 100755
--- a/labyrinth
+++ b/labyrinth
@@ -38,7 +38,7 @@ try:
     from labyrinth_lib import defs
     localedir = osp.abspath(osp.join(defs.DATA_DIR, "locale"))
 except:
-    localedir = "/usr/share/locale"
+    localedir = "/app/share/locale"
 
 gettext.bindtextdomain('labyrinth', localedir)
 if hasattr(gettext, 'bind_textdomain_codeset'):
diff --git a/labyrinth_lib/utils.py b/labyrinth_lib/utils.py
index a535e55..90f406f 100644
--- a/labyrinth_lib/utils.py
+++ b/labyrinth_lib/utils.py
@@ -111,7 +111,7 @@ def get_data_dir():
             if exists(path) and isdir(path) and isfile(path+"/AUTHORS"):
                 __data_dir = os.sep.join([dirname(__file__), '..' , 'data'])
             else:
-                __data_dir = "/usr/share/labyrinth"
+                __data_dir = "/app/share/labyrinth"
     else:
         if __data_dir is None:
             __data_dir = join (".","data")
