diff -Naur ladspa_sdk.orig/src/makefile ladspa_sdk/src/makefile
--- ladspa_sdk.orig/src/makefile	2007-11-06 11:42:45.000000000 +0100
+++ ladspa_sdk/src/makefile	2018-12-05 21:59:25.278935059 +0100
@@ -59,9 +59,9 @@
 	@echo Test complete.
 
 install:	targets
-	-mkdirhier $(INSTALL_PLUGINS_DIR)
-	-mkdirhier $(INSTALL_INCLUDE_DIR)
-	-mkdirhier $(INSTALL_BINARY_DIR)
+	-mkdir -p $(INSTALL_PLUGINS_DIR)
+	-mkdir -p $(INSTALL_INCLUDE_DIR)
+	-mkdir -p $(INSTALL_BINARY_DIR)
 	cp ../plugins/* $(INSTALL_PLUGINS_DIR)
 	cp ladspa.h $(INSTALL_INCLUDE_DIR)
 	cp ../bin/* $(INSTALL_BINARY_DIR)
