diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1edfef2..c090f1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,6 +132,10 @@ elseif(UNIX)
 endif()
 
 target_link_libraries(BasemarkGPULauncher sfml-network sfml-graphics sfml-window sfml-system json11 ${PLATFORM_LIBRARY} ${OPENGL_gl_LIBRARY} ${HARDWARE_LIBRARIES}) 
+install(TARGETS BasemarkGPULauncher RUNTIME DESTINATION "bin")
+
+#Install assets
+install(DIRECTORY assets DESTINATION "bin")
 
 if(MSVC)
 	set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT BasemarkGPULauncher)
