diff --git a/Makefiles/Makefile.UNIX b/Makefiles/Makefile.UNIX
index 1e48361..1e7577d 100644
--- a/Makefiles/Makefile.UNIX
+++ b/Makefiles/Makefile.UNIX
@@ -27,13 +27,13 @@
 
 
 ###############################################################################
-# Build Options
-###############################################################################
-
-DELETE = rm -d -r -f
-
-
-###############################################################################
+# Build Options
+###############################################################################
+
+DELETE = rm -d -r -f
+
+
+###############################################################################
 # Platform Configuration
 #
 # Edit library and include paths as needed.
@@ -53,29 +53,29 @@ LD = gcc
 
 #
 # SDL
-#
-
-SDL2_CFLAGS = `sdl2-config --cflags`
-SDL2_LIBS = `sdl2-config --libs`
-
-#
-#	UNIX-specific
-#
-
-PLATFORM_CXXFLAGS = $(SDL2_CFLAGS) -O3
-PLATFORM_LDFLAGS = $(SDL2_LIBS) -lGL -lGLU -lz -lm -lstdc++ -lpthread -lSDL2_net
-
-
-###############################################################################
-# Core Makefile
-###############################################################################
+#
+
+SDL2_CFLAGS = `sdl2-config --cflags`
+SDL2_LIBS = `sdl2-config --libs`
+
+#
+#	UNIX-specific
+#
+
+PLATFORM_CXXFLAGS = $(SDL2_CFLAGS) -O3
+PLATFORM_LDFLAGS = $(SDL2_LIBS) -lGL -lGLU -lz -lm -lstdc++ -lpthread -lSDL2_net -L/app/lib
+
+
+###############################################################################
+# Core Makefile
+###############################################################################
 
 PLATFORM_SRC_FILES = \
 	Src/OSD/Unix/FileSystemPath.cpp
-
-include Makefiles/Rules.inc
-
-clean:
-	$(SILENT)echo Cleaning up \"$(BIN_DIR)\" and \"$(OBJ_DIR)\"...
-	$(SILENT)$(DELETE) $(BIN_DIR)
-	$(SILENT)$(DELETE) $(OBJ_DIR)
+
+include Makefiles/Rules.inc
+
+clean:
+	$(SILENT)echo Cleaning up \"$(BIN_DIR)\" and \"$(OBJ_DIR)\"...
+	$(SILENT)$(DELETE) $(BIN_DIR)
+	$(SILENT)$(DELETE) $(OBJ_DIR)
