diff --git a/Makefile.inc b/Makefile.inc
index f0f2398..8cdedf8 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -3,10 +3,10 @@ ifeq ($(shell uname), FreeBSD)
 CXX = c++
 else
 CC = gcc
 CXX = g++
 ifeq ($(shell if [ `uname -m` = x86_64 -o `uname -m` = i386 -o `uname -m` = i486 -o `uname -m` = i586 -o `uname -m` = i686 ]; then echo OK; fi),OK)
-EXPORT_3DS=-DEXPORT_3DS
+#EXPORT_3DS=-DEXPORT_3DS
 endif
 endif
 RM = rm -f
 RMDIR = rm -rf
diff --git a/QT/LDView.pro b/QT/LDView.pro
index 94900d8..ecf042d 100644
--- a/QT/LDView.pro
+++ b/QT/LDView.pro
@@ -55,7 +55,9 @@ DEFINES		+= QT_THREAD_SUPPORT _QT
 #      - Background is transparent for ModelViewerWidget
 #DEFINES	+= QOPENGLWIDGET
 
-isEmpty(PREFIX):PREFIX			= /usr
+LIBS            += -L/app/lib
+
+isEmpty(PREFIX):PREFIX			= /app
 isEmpty(BINDIR):BINDIR			= $$PREFIX/bin
 isEmpty(DATADIR):DATADIR		= $$PREFIX/share
 isEmpty(DOCDIR):DOCDIR			= $$DATADIR/doc
@@ -58,11 +60,11 @@ OBJECTS_DIR = .obj$$POSTFIX
 MAKEOPT += \"TESTING=-I$$[QT_INSTALL_HEADERS] $$QMAKE_CXXFLAGS_STATIC_LIB $(TESTING)\"
 
 linux-g++-64{
-DEFINES 	+= EXPORT_3DS
+#DEFINES 	+= EXPORT_3DS
 }
 linux-g++-32{
 DEFINES 	+= EXPORT_3DS
 }
 
 #DEFINES 	+= _NO_BOOST
 
diff --git a/QT/desktop/ldraw-thumbnailer b/QT/desktop/ldraw-thumbnailer
index b2f6574..d552bbe 100644
--- a/QT/desktop/ldraw-thumbnailer
+++ b/QT/desktop/ldraw-thumbnailer
@@ -22,8 +22,8 @@ size=$3
 if [ -f /tmp/ldraw-thumbnailer.log ] ; then
 	echo $input $output $size $dir>>/tmp/ldraw-thumbnailer.log
 fi
-/usr/bin/$LDVIEW "$uri" -SaveSnapshot=$output -CheckPartTracker=0 \
+/app/bin/$LDVIEW "$uri" -SaveSnapshot=$output -CheckPartTracker=0 \
 -SaveWidth=$size -SaveHeight=$size -ShowErrors=0 -SaveActualSize=0 $INI \
 -SnapshotTNMetas=1
 if [ -f $output.png ] ; then
 	mv -f $output.png $output
diff --git a/QT/desktop/ldview.desktop b/QT/desktop/ldview.desktop
index 57f22ac..8b5f1cc 100644
--- a/QT/desktop/ldview.desktop
+++ b/QT/desktop/ldview.desktop
@@ -7,5 +7,5 @@ Exec=LDView %f
 Icon=gnome-ldraw
 Terminal=false
 Type=Application
-MimeType=application/x-ldraw;application/x-multipart-ldraw;
+MimeType=application/x-ldraw;application/x-multipart-ldraw;application/x-multi-part-ldraw;
 Categories=Graphics;3DGraphics;Viewer;
diff --git a/QT/kde/ldviewthumbnailcreator.desktop b/QT/kde/ldviewthumbnailcreator.desktop
index eb69911..dc6e8de 100644
--- a/QT/kde/ldviewthumbnailcreator.desktop
+++ b/QT/kde/ldviewthumbnailcreator.desktop
@@ -3,7 +3,7 @@ Encoding=UTF-8
 Type=Service
 Name=LDView
 X-KDE-ServiceTypes=ThumbCreator
-MimeType=application/x-ldraw;application/x-multipart-ldraw
+MimeType=application/x-ldraw;application/x-multipart-ldraw;application/x-multi-part-ldraw
 CacheThumbnail=true
 X-KDE-Library=ldviewthumbnail
 
