--- src/dialog.c.orig	2018-05-28 08:17:58.249171564 -0400
+++ src/dialog.c	2018-05-28 08:19:37.719410214 -0400
@@ -36,7 +36,7 @@
 		GTK_DIALOG_DESTROY_WITH_PARENT,
 		type,
 		GTK_BUTTONS_NONE,
-		str);
+		(const gchar*) str, "");
 	gtk_window_set_resizable(GTK_WINDOW(dialog), FALSE);
 	gtk_dialog_add_buttons(GTK_DIALOG(dialog),
 		GTK_STOCK_OK, GTK_RESPONSE_CANCEL, NULL);
@@ -61,7 +61,7 @@
 		GTK_DIALOG_DESTROY_WITH_PARENT,
 		GTK_MESSAGE_QUESTION,
 		GTK_BUTTONS_NONE,
-		str);
+		(const gchar*) str, "");
 	gtk_window_set_resizable(GTK_WINDOW(dialog), FALSE);
 	gtk_dialog_add_buttons(GTK_DIALOG(dialog),
 		GTK_STOCK_NO, GTK_RESPONSE_NO,
--- src/gtkprint.c.orig	2018-05-28 08:18:21.958230271 -0400
+++ src/gtkprint.c	2018-05-28 08:18:56.251313162 -0400
@@ -165,7 +165,7 @@
 		GTK_DIALOG_DESTROY_WITH_PARENT,
 		GTK_MESSAGE_ERROR,
 		GTK_BUTTONS_NONE,
-		message);
+		(const gchar*) message, "");
 	gtk_window_set_resizable(GTK_WINDOW(dialog), FALSE);
 	gtk_dialog_add_buttons(GTK_DIALOG(dialog),
 		GTK_STOCK_OK, GTK_RESPONSE_CANCEL, NULL);
