diff -Naur leafpad-0.8.18.1.orig/src/dnd.c leafpad-0.8.18.1/src/dnd.c
--- leafpad-0.8.18.1.orig/src/dnd.c	2007-12-15 07:17:46.000000000 +0100
+++ leafpad-0.8.18.1/src/dnd.c	2018-12-03 00:38:10.552273691 +0100
@@ -100,7 +100,7 @@
 DV(g_print("DND start!\n"));
 	
 #if GTK_CHECK_VERSION(2, 10, 0)
-	if (g_strcasecmp(gdk_atom_name(context->targets->data),
+	if (g_ascii_strcasecmp(gdk_atom_name(context->targets->data),
 	    "GTK_TEXT_BUFFER_CONTENTS") != 0) {
 #else
 	if (info != TARGET_SELF) {
diff -Naur leafpad-0.8.18.1.orig/src/main.c leafpad-0.8.18.1/src/main.c
--- leafpad-0.8.18.1.orig/src/main.c	2009-10-24 04:38:34.000000000 +0200
+++ leafpad-0.8.18.1/src/main.c	2018-12-03 00:38:10.551273686 +0100
@@ -255,12 +255,12 @@
 #endif
 	
 	if (fi->charset 
-		&& (g_strcasecmp(fi->charset, get_default_charset()) != 0)
-		&& (g_strcasecmp(fi->charset, "UTF-8") != 0)) {
+		&& (g_ascii_strcasecmp(fi->charset, get_default_charset()) != 0)
+		&& (g_ascii_strcasecmp(fi->charset, "UTF-8") != 0)) {
 		encarray = get_encoding_items(get_encoding_code());
 		for (i = 0; i < ENCODING_MAX_ITEM_NUM; i++)
 			if (encarray->item[i])
-				if (g_strcasecmp(fi->charset, encarray->item[i]) == 0)
+				if (g_ascii_strcasecmp(fi->charset, encarray->item[i]) == 0)
 					break;
 		if (i == ENCODING_MAX_ITEM_NUM)
 			fi->charset_flag = TRUE;
diff -Naur leafpad-0.8.18.1.orig/src/selector.c leafpad-0.8.18.1/src/selector.c
--- leafpad-0.8.18.1.orig/src/selector.c	2007-07-22 03:14:50.000000000 +0200
+++ leafpad-0.8.18.1/src/selector.c	2018-12-03 00:38:10.553273697 +0100
@@ -273,7 +273,7 @@
 	i = 0;
 	if (selected_fi->charset) {
 		do {
-			if (g_strcasecmp(selected_fi->charset, ctable->charset[i]) == 0)
+			if (g_ascii_strcasecmp(selected_fi->charset, ctable->charset[i]) == 0)
 				break;
 			i++;
 		} while (i < ctable->num);
