--- libcoverart/CMakeLists.txt~bak	2012-10-09 23:51:32.000000000 +1100
+++ libcoverart/CMakeLists.txt	2026-05-22 23:19:05.205396120 +1000
@@ -19,8 +19,11 @@
 SET(coverart_SOVERSION ${coverart_SOVERSION_MAJOR})
 
 SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
+
 FIND_PACKAGE(Neon REQUIRED)
 FIND_PACKAGE(Jansson REQUIRED)
+FIND_PACKAGE(GSSAPI REQUIRED)
+FIND_PACKAGE(LibProxy REQUIRED)
 
 SET(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)")
 SET(EXEC_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX} CACHE PATH "Installation prefix for executables and object code libraries" FORCE)
--- libcoverart/cmake/modules/FindGSSAPI.cmake~bak	1970-01-01 10:00:00.000000000 +1000
+++ libcoverart/cmake/modules/FindGSSAPI.cmake	1970-01-01 10:00:00.000000000 +1000
@@ -0,0 +1,19 @@
+INCLUDE(FindPackageHandleStandardArgs)
+FIND_PACKAGE(PkgConfig)
+PKG_CHECK_MODULES(PKG_GSSAPI krb5-gssapi)
+
+FIND_PATH(GSSAPI_INCLUDE_DIR gssapi/gssapi.h
+    PATHS
+    ${PKG_GSSAPI_INCLUDE_DIRS}
+    /usr/include
+    /usr/local/include
+    PATH_SUFFIXES GSSAPI
+)
+
+FIND_LIBRARY(GSSAPI_LIBRARIES gssapi_krb5
+    ${PKG_GSSAPI_LIBRARY_DIRS}
+    /usr/lib
+    /usr/local/lib
+)
+
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(krb5-gssapi DEFAULT_MSG GSSAPI_LIBRARIES GSSAPI_INCLUDE_DIR)
--- libcoverart/cmake/modules/FindLibProxy.cmake~bak	1970-01-01 10:00:00.000000000 +1000
+++ libcoverart/cmake/modules/FindLibProxy.cmake	1970-01-01 10:00:00.000000000 +1000
@@ -0,0 +1,19 @@
+INCLUDE(FindPackageHandleStandardArgs)
+FIND_PACKAGE(PkgConfig)
+PKG_CHECK_MODULES(PKG_GSSAPI libproxy-1.0)
+
+FIND_PATH(LIBPROXY_INCLUDE_DIR libproxy/proxy.h
+    PATHS
+    ${PKG_LIBPROXY_INCLUDE_DIRS}
+    /usr/include
+    /usr/local/include
+    PATH_SUFFIXES LIBPROXY
+)
+
+FIND_LIBRARY(LIBPROXY_LIBRARIES proxy
+    ${PKG_LIBPROXY_LIBRARY_DIRS}
+    /usr/lib
+    /usr/local/lib
+)
+
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(proxy DEFAULT_MSG LIBPROXY_LIBRARIES LIBPROXY_INCLUDE_DIR)
--- libcoverart/src/CMakeLists.txt~bak	2012-10-09 23:51:32.000000000 +1100
+++ libcoverart/src/CMakeLists.txt	2026-05-22 23:24:04.268720206 +1000
@@ -5,6 +5,8 @@
 	${CMAKE_CURRENT_BINARY_DIR}/../include
 	${NEON_INCLUDE_DIR}
 	${JANSSON_INCLUDE_DIR}
+	${GSSAPI_INCLUDE_DIR}
+	${LIBPROXY_INCLUDE_DIR}
 )
 
 SET(_sources HTTPFetch.cc CoverArt.cc ReleaseInfo.cc ImageList.cc Image.cc
@@ -31,7 +33,7 @@
 	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/caa_c.cc ${CMAKE_CURRENT_BINARY_DIR}/caa_c.h ${CMAKE_CURRENT_BINARY_DIR}/../include/coverart/caa_c.h
 	COMMAND make-c-interface ${CMAKE_CURRENT_SOURCE_DIR} cinterface.xml ${CMAKE_CURRENT_BINARY_DIR} caa_c.cc caa_c.h
 	COMMAND ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/caa_c.h ${CMAKE_CURRENT_BINARY_DIR}/../include/coverart/caa_c.h
-	DEPENDS make-c-interface cinterface.xml *.inc
+	DEPENDS make-c-interface cinterface.xml
 )
 
 ADD_CUSTOM_TARGET(src_gen DEPENDS caa_c.h)
@@ -53,14 +55,14 @@
 	ENDIF(CMAKE_COMPILER_IS_GNUCXX)
 endif(CMAKE_BUILD_TYPE STREQUAL Debug)
 
-TARGET_LINK_LIBRARIES(coverart ${NEON_LIBRARIES} ${JANSSON_LIBRARIES})
+TARGET_LINK_LIBRARIES(coverart ${NEON_LIBRARIES} ${JANSSON_LIBRARIES} ${GSSAPI_LIBRARIES} ${LIBPROXY_LIBRARIES})
 
 IF(WIN32)
     TARGET_LINK_LIBRARIES(coverart wsock32 winmm ws2_32)
 ENDIF(WIN32)
 
 IF(CMAKE_COMPILER_IS_GNUCXX)
-		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wextra -pedantic-errors")
+		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
 		set_source_files_properties(mb4_c.cc PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
 		set_source_files_properties(ReleaseInfo.cc PROPERTIES COMPILE_FLAGS "-Wno-long-long")
 		set_source_files_properties(ImageList.cc PROPERTIES COMPILE_FLAGS "-Wno-long-long")
--- libcoverart/tests/CMakeLists.txt~bak	2012-10-09 23:51:32.000000000 +1100
+++ libcoverart/tests/CMakeLists.txt	2026-05-22 23:06:10.998753395 +1000
@@ -7,11 +7,11 @@
 ADD_EXECUTABLE(ctest ctest.c)
 
 IF(CMAKE_COMPILER_IS_GNUCXX)
-		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wextra -pedantic-errors")
+		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
 ENDIF(CMAKE_COMPILER_IS_GNUCXX)
 
 IF(CMAKE_COMPILER_IS_GNUCC)
-		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wextra -pedantic-errors")
+		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
 #		set_source_files_properties(ctest.c PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
 		set_source_files_properties(test.cc PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
 ENDIF(CMAKE_COMPILER_IS_GNUCC)
--- libcoverart/examples/CMakeLists.txt~bak	2012-10-09 23:51:32.000000000 +1100
+++ libcoverart/examples/CMakeLists.txt	2026-05-22 23:07:14.908751418 +1000
@@ -10,11 +10,11 @@
 ADD_EXECUTABLE(example_c example_c.c)
 
 IF(CMAKE_COMPILER_IS_GNUCXX)
-		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wextra -pedantic-errors")
+		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
 ENDIF(CMAKE_COMPILER_IS_GNUCXX)
 
 IF(CMAKE_COMPILER_IS_GNUCC)
-		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wextra -pedantic-errors")
+		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
 		set_source_files_properties(example.cc PROPERTIES COMPILE_FLAGS "-Wno-long-long")
 ENDIF(CMAKE_COMPILER_IS_GNUCC)
 
