--- a/configure.ac.orig	2014-04-16 09:22:14.000000000 +0200
+++ b/configure.ac	2017-08-16 14:47:31.362460853 +0200
@@ -22,7 +22,7 @@ dnl store CFLAGS from user
 CFLAGS_save="${CFLAGS}"
 
 dnl add features to CFLAGS
-CFLAGS_dist="-Wall -Werror --std=gnu99 -D_GNU_SOURCE"
+CFLAGS_dist="-Wall --std=gnu99 -D_GNU_SOURCE"
 CFLAGS_dist="${CFLAGS_dist} -Wpointer-arith -Wcast-align -Wcast-qual -Wstrict-prototypes -Wshadow -Waggregate-return -Wmissing-prototypes -Wnested-externs -Wsign-compare"
 
 dnl check the operating system
