diff --git a/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp b/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp
index 527708d..f5db56a 100644
--- a/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp
+++ b/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp
@@ -381,9 +382,6 @@
               'defines': [
                 'WEBRTC_POSIX',
               ],
-              'cflags_cc': [
-                '-msse2',
-              ],
               'direct_dependent_settings': {
                 'libraries': [

@@ -391,6 +389,13 @@
               },
             },
           ],
+          [
+            '"<!(uname -p)" == "i686"', {
+              'cflags_cc': [
+                '-msse2',
+              ],
+            }
+          ],
         ],
       },
     ],

