diff -pur libgnome-keyring-3.12.0.orig/configure.ac libgnome-keyring-3.12.0/configure.ac
--- libgnome-keyring-3.12.0.orig/configure.ac	2014-03-23 00:48:24.000000000 -0700
+++ libgnome-keyring-3.12.0/configure.ac	2016-11-07 16:58:37.063601464 -0800
@@ -127,24 +127,13 @@ LIBRARY_LIBS="$LIBRARY_LIBS $DBUS_LIBS"
 #
 
 GCRYPT_VERSION=1.2.2
-GCRYPT_LIBVER=1
 
-AM_PATH_LIBGCRYPT($GCRYPT_LIBVER:$GCRYPT_VERSION,,
-	AC_MSG_ERROR([[
-***
-*** libgcrypt was not found. You may want to get it from
-*** ftp://ftp.gnupg.org/gcrypt/libgcrypt/
-***
-]]))
-libgcrypt=yes
+PKG_CHECK_MODULES(LIBGCRYPT, [libgcrypt >= $GCRYPT_VERSION])
 AC_DEFINE_UNQUOTED(LIBGCRYPT_VERSION, "$GCRYPT_VERSION", [Version of GCRYPT we expect])
 
 LIBRARY_CFLAGS="$LIBRARY_CFLAGS $LIBGCRYPT_CFLAGS"
 LIBRARY_LIBS="$LIBRARY_LIBS $LIBGCRYPT_LIBS"
 
-AC_SUBST([LIBGCRYPT_CFLAGS])
-AC_SUBST([LIBGCRYPT_LIBS])
-
 # --------------------------------------------------------------------
 # Debug mode
 #
