diff -urN libid3tag-0.15.1b/configure.ac libid3tag-0.15.1b.new/configure.ac
--- libid3tag-0.15.1b/configure.ac	2004-01-23 23:22:46.000000000 +0000
+++ libid3tag-0.15.1b.new/configure.ac	2016-05-12 13:59:10.457870513 +0100
@@ -26,7 +26,7 @@
 
 AC_CONFIG_SRCDIR([id3tag.h])
 
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 
 AM_CONFIG_HEADER([config.h])
 
@@ -201,5 +201,5 @@
 dnl AC_SUBST(LTLIBOBJS)
 
 AC_CONFIG_FILES([Makefile msvc++/Makefile  \
-	libid3tag.list])
+	libid3tag.list id3tag.pc])
 AC_OUTPUT
diff -urN libid3tag-0.15.1b/id3tag.pc.in libid3tag-0.15.1b.new/id3tag.pc.in
--- libid3tag-0.15.1b/id3tag.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ libid3tag-0.15.1b.new/id3tag.pc.in	2016-05-12 13:57:40.249785719 +0100
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: id3tag
+Description: ID3 tag manipulation library
+Requires:
+Version: @PACKAGE_VERSION@
+Libs: -lid3tag
+Cflags:
diff -urN libid3tag-0.15.1b/Makefile.am libid3tag-0.15.1b.new/Makefile.am
--- libid3tag-0.15.1b/Makefile.am	2004-02-17 02:11:28.000000000 +0000
+++ libid3tag-0.15.1b.new/Makefile.am	2016-05-12 13:56:42.898460276 +0100
@@ -120,4 +120,7 @@
 	$(MAKE) clean
 	$(MAKE)
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = id3tag.pc
+
 .PHONY: again
