commit 425d7ed
Author: Timothée Ravier <tim@siosm.fr>
Date:   Wed Jan 13 19:32:22 2021 +0100

    Add option to not build/install KAuth helper

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b167ad..55db5ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,6 +91,8 @@ configure_file(config-ksysguard.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-ksysg
 option(BUILD_DESIGNERPLUGIN "Build plugin for Qt Designer" ON)
 add_feature_info(DESIGNERPLUGIN ${BUILD_DESIGNERPLUGIN} "Build plugin for Qt Designer")
 
+option(ENABLE_KAUTH_HELPER "Build and install ksysguardprocesslist_helper KAuth helper" ON)
+add_feature_info(DESIGNERPLUGIN ${ENABLE_KAUTH_HELPER} "Build and install ksysguardprocesslist_helper KAuth helper")
 
 add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 add_definitions(-DQT_USE_QSTRINGBUILDER)
diff --git a/processcore/CMakeLists.txt b/processcore/CMakeLists.txt
index 4f7b097..23962a9 100644
--- a/processcore/CMakeLists.txt
+++ b/processcore/CMakeLists.txt
@@ -74,21 +74,22 @@ install( FILES
 
 #------ KAuth stuff
 
-# Auth example helper
-set(ksysguardprocesslist_helper_srcs
-    helper.cpp
-    process.cpp
-    processes_local_p.cpp
-    processes_base_p.cpp
-    read_procsmaps_runnable.cpp
-)
-
-add_executable(ksysguardprocesslist_helper ${ksysguardprocesslist_helper_srcs})
-target_link_libraries(ksysguardprocesslist_helper Qt5::Core KF5::AuthCore KF5::I18n)
-install(TARGETS ksysguardprocesslist_helper DESTINATION ${KAUTH_HELPER_INSTALL_DIR})
-
-kauth_install_helper_files(ksysguardprocesslist_helper org.kde.ksysguard.processlisthelper root)
-kauth_install_actions(org.kde.ksysguard.processlisthelper actions.actions)
-
-set_target_properties(ksysguardprocesslist_helper PROPERTIES COMPILE_FLAGS "-Wall -ggdb")
-
+if(ENABLE_KAUTH_HELPER)
+    # Auth example helper
+    set(ksysguardprocesslist_helper_srcs
+        helper.cpp
+        process.cpp
+        processes_local_p.cpp
+        processes_base_p.cpp
+        read_procsmaps_runnable.cpp
+    )
+
+    add_executable(ksysguardprocesslist_helper ${ksysguardprocesslist_helper_srcs})
+    target_link_libraries(ksysguardprocesslist_helper Qt5::Core KF5::AuthCore KF5::I18n)
+    install(TARGETS ksysguardprocesslist_helper DESTINATION ${KAUTH_HELPER_INSTALL_DIR})
+
+    kauth_install_helper_files(ksysguardprocesslist_helper org.kde.ksysguard.processlisthelper root)
+    kauth_install_actions(org.kde.ksysguard.processlisthelper actions.actions)
+
+    set_target_properties(ksysguardprocesslist_helper PROPERTIES COMPILE_FLAGS "-Wall -ggdb")
+endif()
