diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c9cc0dd..30d3af6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -281,7 +281,7 @@ ecm_generate_pri_file(
 )
 
 install(FILES ${PRI_FILENAME}
-        DESTINATION ${ECM_MKSPECS_INSTALL_DIR})
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/mkspecs/modules)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/kdb_version.h"
         DESTINATION "${KDB_INCLUDE_INSTALL_DIR}"
