--- libmusicbrainz/src/HTTPFetch.cc~bak	2026-05-22 17:49:26.682934050 +1000
+++ libmusicbrainz/src/HTTPFetch.cc	2026-05-22 17:50:23.862672650 +1000
@@ -182,8 +182,6 @@
 		}
 
 		ne_request *req = ne_request_create(sess, Request.c_str(), URL.c_str());
-		if (Request=="PUT")
-			ne_set_request_body_buffer(req,0,0);
 
 		if (Request!="GET")
 			ne_set_request_flag(req, NE_REQFLAG_IDEMPOTENT, 0);
--- libmusicbrainz/CMakeLists.txt~bak	2026-05-22 17:49:26.680563388 +1000
+++ libmusicbrainz/CMakeLists.txt	2026-05-22 21:50:22.502227391 +1000
@@ -30,8 +30,11 @@
 SET(musicbrainz5c_SOVERSION ${musicbrainz5c_SOVERSION_MAJOR})
 
 SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
+
 FIND_PACKAGE(Neon REQUIRED)
 FIND_PACKAGE(LibXml2 REQUIRED)
+FIND_PACKAGE(GSSAPI REQUIRED)
+FIND_PACKAGE(LibProxy REQUIRED)
 
 SET(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)")
 SET(EXEC_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX} CACHE PATH "Installation prefix for executables and object code libraries" FORCE)
--- libmusicbrainz/cmake/modules/FindGSSAPI.cmake~bak	1970-01-01 10:00:00.000000000 +1000
+++ libmusicbrainz/cmake/modules/FindGSSAPI.cmake	1970-01-01 10:00:00.000000000 +1000
@@ -0,0 +1,19 @@
+INCLUDE(FindPackageHandleStandardArgs)
+FIND_PACKAGE(PkgConfig)
+PKG_CHECK_MODULES(PKG_GSSAPI krb5-gssapi)
+
+FIND_PATH(GSSAPI_INCLUDE_DIR gssapi/gssapi.h
+    PATHS
+    ${PKG_GSSAPI_INCLUDE_DIRS}
+    /usr/include
+    /usr/local/include
+    PATH_SUFFIXES GSSAPI
+)
+
+FIND_LIBRARY(GSSAPI_LIBRARIES gssapi_krb5
+    ${PKG_GSSAPI_LIBRARY_DIRS}
+    /usr/lib
+    /usr/local/lib
+)
+
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(krb5-gssapi DEFAULT_MSG GSSAPI_LIBRARIES GSSAPI_INCLUDE_DIR)
--- libmusicbrainz/cmake/modules/FindLibProxy.cmake~bak	1970-01-01 10:00:00.000000000 +1000
+++ libmusicbrainz/cmake/modules/FindLibProxy.cmake	1970-01-01 10:00:00.000000000 +1000
@@ -0,0 +1,19 @@
+INCLUDE(FindPackageHandleStandardArgs)
+FIND_PACKAGE(PkgConfig)
+PKG_CHECK_MODULES(PKG_GSSAPI libproxy-1.0)
+
+FIND_PATH(LIBPROXY_INCLUDE_DIR libproxy/proxy.h
+    PATHS
+    ${PKG_LIBPROXY_INCLUDE_DIRS}
+    /usr/include
+    /usr/local/include
+    PATH_SUFFIXES LIBPROXY
+)
+
+FIND_LIBRARY(LIBPROXY_LIBRARIES proxy
+    ${PKG_LIBPROXY_LIBRARY_DIRS}
+    /usr/lib
+    /usr/local/lib
+)
+
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(proxy DEFAULT_MSG LIBPROXY_LIBRARIES LIBPROXY_INCLUDE_DIR)
--- libmusicbrainz/src/CMakeLists.txt~bak	2026-05-22 17:49:26.682934050 +1000
+++ libmusicbrainz/src/CMakeLists.txt	2026-05-22 22:50:23.408782700 +1000
@@ -5,6 +5,8 @@
 	${CMAKE_CURRENT_BINARY_DIR}/../include
 	${NEON_INCLUDE_DIR}
 	${LIBXML2_INCLUDE_DIR}
+	${GSSAPI_INCLUDE_DIR}
+	${LIBPROXY_INCLUDE_DIR}
 )
 
 SET(_sources_cc Alias.cc Annotation.cc Artist.cc ArtistCredit.cc Attribute.cc CDStub.cc Collection.cc
@@ -67,7 +69,7 @@
 	ENDIF(CMAKE_COMPILER_IS_GNUCXX)
 endif(CMAKE_BUILD_TYPE STREQUAL Debug)
 
-TARGET_LINK_LIBRARIES(musicbrainz5cc ${NEON_LIBRARIES} ${LIBXML2_LIBRARIES})
+TARGET_LINK_LIBRARIES(musicbrainz5cc ${NEON_LIBRARIES} ${LIBXML2_LIBRARIES} ${GSSAPI_LIBRARIES} ${LIBPROXY_LIBRARIES})
 TARGET_LINK_LIBRARIES(musicbrainz5 musicbrainz5cc)
 
 IF(WIN32)
@@ -75,10 +77,7 @@
 ENDIF(WIN32)
 
 IF(CMAKE_COMPILER_IS_GNUCXX)
-		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic-errors")
-		IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../.git)
-			SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-		ENDIF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../.git)
+		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
 		#set_source_files_properties(mb5_c.cc PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
 ENDIF(CMAKE_COMPILER_IS_GNUCXX)
 
