--- a/oxt/pythonpath/VoikkoHandlePool.py
+++ b/oxt/pythonpath/VoikkoHandlePool.py
@@ -402,7 +402,7 @@
 
 	def setInstallationPath(self, path):
 		self.__installationPath = path
-		searchPath = os.path.join(path, "voikko", platform.system() + "-" + "-".join(platform.architecture()))
+		searchPath = os.path.join(path, "voikko", platform.system())
 		logging.debug("VoikkoHandlePool.setInstallationPath: library search path is " + searchPath)
 		Voikko.setLibrarySearchPath(searchPath)
 
