--- libusb-0.1.12/usb.c.memset	2006-03-04 03:52:46.000000000 +0100
+++ libusb-0.1.12/usb.c	2007-08-01 12:32:17.000000000 +0200
@@ -232,6 +232,7 @@ int usb_get_string_simple(usb_dev_handle
   char tbuf[255];	/* Some devices choke on size > 255 */
   int ret, langid, si, di;
 
+  memset(tbuf, 0, sizeof(tbuf));
   /*
    * Asking for the zero'th index is special - it returns a string
    * descriptor that contains all the language IDs supported by the
