diff --git a/build.gradle b/build.gradle
index 4fb73624..2710a983 100644
--- a/build.gradle
+++ b/build.gradle
@@ -21,6 +21,7 @@ allprojects {
     }
 
     repositories {
+        maven { url = "../maven-local" }
         mavenCentral()
         maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
         maven { url "https://oss.sonatype.org/content/repositories/releases/" }
