commit 62daf9a09bee7552aa1013501ff3a9b912a222f9
Author: Jonathan Esk-Riddell <jr@jriddell.org>
Date:   Tue Nov 15 13:13:39 2022 +0000

    fix includes for dbus

diff --git a/isoimagewriter/CMakeLists.txt b/isoimagewriter/CMakeLists.txt
index b3a78e2..3fc64a6 100644
--- a/isoimagewriter/CMakeLists.txt
+++ b/isoimagewriter/CMakeLists.txt
@@ -56,6 +56,7 @@ target_compile_definitions(isoimagewriter PRIVATE -DPROJECT_VERSION="${PROJECT_V
 target_link_libraries(isoimagewriter
     Qt5::Widgets
     Qt5::Network
+    Qt5::DBus
     KF5::I18n
     KF5::CoreAddons
     KF5::WidgetsAddons
diff --git a/isoimagewriter/physicaldevice.cpp b/isoimagewriter/physicaldevice.cpp
index 765021d..4515f24 100644
--- a/isoimagewriter/physicaldevice.cpp
+++ b/isoimagewriter/physicaldevice.cpp
@@ -7,9 +7,10 @@
 // Implementation of PhysicalDevice
 
 #include <KLocalizedString>
-#include <QtDBus>
-#include <QDBusInterface>
-#include <QDBusUnixFileDescriptor>
+#include <QtDBus/QDBusArgument>
+#include <QtDBus/QDBusReply>
+#include <QtDBus/QDBusInterface>
+#include <QtDBus/QDBusUnixFileDescriptor>
 #include <fcntl.h>
 
 #include "physicaldevice.h"
