diff --git a/src/Makefile.am b/src/Makefile.am
index 53ef0a6..0663211 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/usr/include/SDL $(FILE_OFFSET)
+AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/app/include/SDL $(FILE_OFFSET)
 AUTOMAKE_OPTIONS = subdir-objects
 DEFAULT_INCLUDES =
 SUBDIRS =
@@ -366,4 +366,4 @@ love_SOURCES = \
 ./common/Exception.h \
 ./common/Exception.cpp \
 ./common/EnumMap.h \
-./common/Data.h
\ No newline at end of file
+./common/Data.h
diff --git a/src/Makefile.in b/src/Makefile.in
index 391bb6f..f95dd99 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -393,7 +393,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/usr/include/SDL $(FILE_OFFSET)
+AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/app/include/SDL $(FILE_OFFSET)
 AUTOMAKE_OPTIONS = subdir-objects
 DEFAULT_INCLUDES = 
 SUBDIRS = 
