diff --git a/src/Makefile.am b/src/Makefile.am
index 53ef0a6..0663211 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/usr/include/SDL $(FILE_OFFSET)
+AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/app/include/SDL $(FILE_OFFSET)
 AUTOMAKE_OPTIONS = subdir-objects
 DEFAULT_INCLUDES =
 SUBDIRS =
@@ -366,4 +366,4 @@ love_SOURCES = \
 ./common/Exception.h \
 ./common/Exception.cpp \
 ./common/EnumMap.h \
-./common/Data.h
\ No newline at end of file
+./common/Data.h
diff --git a/src/Makefile.in b/src/Makefile.in
index 391bb6f..f95dd99 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -393,7 +393,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/usr/include/SDL $(FILE_OFFSET)
+AM_CPPFLAGS = -I. -I./modules -I/usr/include/AL -I/usr/include/freetype2  $(INCLUDE_LUA) -I/app/include/SDL $(FILE_OFFSET)
 AUTOMAKE_OPTIONS = subdir-objects
 DEFAULT_INCLUDES = 
 SUBDIRS = 
diff --git a/src/modules/font/freetype/Font.h b/src/modules/font/freetype/Font.h
index a98778f..d8ad364 100644
--- a/src/modules/font/freetype/Font.h
+++ b/src/modules/font/freetype/Font.h
@@ -26,14 +26,14 @@
 
 // FreeType2
 #ifdef LOVE_MACOSX
-#include <freetype/ft2build.h>
+#include <freetype2/freetype/ft2build.h>
 #else
 #include <ft2build.h>
 #endif
-#include <freetype/freetype.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
-#include <freetype/fttrigon.h>
+#include <freetype2/freetype/freetype.h>
+#include <freetype2/freetype/ftglyph.h>
+#include <freetype2/freetype/ftoutln.h>
+#include <freetype2/freetype/fttrigon.h>
 
 namespace love
 {
diff --git a/src/modules/font/freetype/TrueTypeRasterizer.h b/src/modules/font/freetype/TrueTypeRasterizer.h
index 6fc13db..7393bc2 100644
--- a/src/modules/font/freetype/TrueTypeRasterizer.h
+++ b/src/modules/font/freetype/TrueTypeRasterizer.h
@@ -27,10 +27,10 @@
 
 // TrueType2
 #include <ft2build.h>
-#include <freetype/freetype.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
-#include <freetype/fttrigon.h>
+#include <freetype2/freetype/freetype.h>
+#include <freetype2/freetype/ftglyph.h>
+#include <freetype2/freetype/ftoutln.h>
+#include <freetype2/freetype/fttrigon.h>
 
 namespace love
 {
diff --git a/src/modules/graphics/opengl/Framebuffer.cpp b/src/modules/graphics/opengl/Framebuffer.cpp
index 93575d7..ea36807 100644
--- a/src/modules/graphics/opengl/Framebuffer.cpp
+++ b/src/modules/graphics/opengl/Framebuffer.cpp
@@ -1,3 +1,5 @@
+#define GL_GLEXT_PROTOTYPES
+
 #include "Framebuffer.h"
 #include <common/Matrix.h>
 
diff --git a/src/modules/graphics/opengl/GLee.h b/src/modules/graphics/opengl/GLee.h
index 34ca16b..dd3aa2c 100644
--- a/src/modules/graphics/opengl/GLee.h
+++ b/src/modules/graphics/opengl/GLee.h
@@ -67,6 +67,7 @@
 	#define GLX_GLXEXT_PROTOTYPES
 	#include <GL/gl.h>
 	#include <GL/glx.h>
+typedef XID GLXContextID;
 #endif
 
 #ifndef APIENTRY
diff --git a/src/modules/graphics/opengl/SpriteBatch.cpp b/src/modules/graphics/opengl/SpriteBatch.cpp
index ca5762f..7a2e57a 100644
--- a/src/modules/graphics/opengl/SpriteBatch.cpp
+++ b/src/modules/graphics/opengl/SpriteBatch.cpp
@@ -18,6 +18,8 @@
 * 3. This notice may not be removed or altered from any source distribution.
 **/
 
+#define GL_GLEXT_PROTOTYPES
+
 #include "SpriteBatch.h"
 
 // STD
