diff -rup lua-1/Makefile lua-2/Makefile
--- lua-1/Makefile      2012-02-10 09:50:23.000000000 +0000
+++ lua-2/Makefile      2018-05-17 07:53:10.385840691 +0100
@@ -5,11 +5,11 @@
 # == CHANGE THE SETTINGS BELOW TO SUIT YOUR ENVIRONMENT =======================

 # Your platform. See PLATS for possible values.
-PLAT= none
+PLAT= linux

 # Where to install. The installation starts in the src and doc directories,
 # so take care if INSTALL_TOP is not an absolute path.
-INSTALL_TOP= /usr/local
+INSTALL_TOP= /app
 INSTALL_BIN= $(INSTALL_TOP)/bin
 INSTALL_INC= $(INSTALL_TOP)/include
 INSTALL_LIB= $(INSTALL_TOP)/lib
