diff -ur lua-5.1.4/etc/lua.pc lua-5.1.4-new/etc/lua.pc
--- lua-5.1.4/etc/lua.pc	2008-08-08 14:46:11.000000000 +0200
+++ lua-5.1.4-new/etc/lua.pc	2012-02-23 18:25:34.000000000 +0100
@@ -8,6 +8,6 @@
 
 # grep '^INSTALL_.*=' ../Makefile | sed 's/INSTALL_TOP/prefix/'
-prefix= /usr/local
+prefix= /app
 INSTALL_BIN= ${prefix}/bin
 INSTALL_INC= ${prefix}/include
 INSTALL_LIB= ${prefix}/lib
diff -ur lua-5.1.4/src/luaconf.h lua-5.1.4-new/src/luaconf.h
--- lua-5.1.4/src/luaconf.h	2008-02-11 17:25:08.000000000 +0100
+++ lua-5.1.4-new/src/luaconf.h	2012-02-23 18:25:34.000000000 +0100
@@ -94,7 +94,7 @@
 	".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
 
 #else
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"/app/"
 #define LUA_LDIR	LUA_ROOT "share/lua/5.1/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/5.1/"
 #define LUA_PATH_DEFAULT  \
