From 9b41062156779160b88fe5e1eb1ece1ee1fe6a74 Mon Sep 17 00:00:00 2001
From: Mike Pall <mike>
Date: Sun, 24 Jun 2018 14:10:21 +0200
Subject: [PATCH 48/72] DynASM: Fix warning.

---
 dynasm/dasm_arm64.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dynasm/dasm_arm64.h b/dynasm/dasm_arm64.h
index 47e1e07..ff21236 100644
--- a/dynasm/dasm_arm64.h
+++ b/dynasm/dasm_arm64.h
@@ -427,6 +427,7 @@ int dasm_encode(Dst_DECL, void *buffer)
 	  break;
 	case DASM_REL_LG:
 	  CK(n >= 0, UNDEF_LG);
+	  /* fallthrough */
 	case DASM_REL_PC:
 	  CK(n >= 0, UNDEF_PC);
 	  n = *DASM_POS2PTR(D, n) - (int)((char *)cp - base) + 4;
-- 
2.20.1

