diff -Naur LuaJIT-2.1.0-beta3.orig/src/Makefile LuaJIT-2.1.0-beta3/src/Makefile
--- LuaJIT-2.1.0-beta3.orig/src/Makefile	2017-05-01 21:05:00.000000000 +0200
+++ LuaJIT-2.1.0-beta3/src/Makefile	2019-09-02 16:16:07.578460773 +0200
@@ -99,7 +99,7 @@
 # enabled by default. Some other features that *might* break some existing
 # code (e.g. __pairs or os.execute() return values) can be enabled here.
 # Note: this does not provide full compatibility with Lua 5.2 at this time.
-#XCFLAGS+= -DLUAJIT_ENABLE_LUA52COMPAT
+XCFLAGS+= -DLUAJIT_ENABLE_LUA52COMPAT
 #
 # Disable the JIT compiler, i.e. turn LuaJIT into a pure interpreter.
 #XCFLAGS+= -DLUAJIT_DISABLE_JIT
