--- a/cplusplus/main.cpp       2024-02-29 20:30:39.655866641 -0600
+++ b/cplusplus/main.cpp       2024-02-29 20:31:51.479866964 -0600
@@ -111,6 +111,13 @@

 int main(int argc, char *argv[]) {

+    char env1[] = "MAGICK_CODER_MODULE_PATH=/app/lib/ImageMagick-7.1.1/modules-Q16HDRI/coders";
+    char env2[] = "MAGICK_FILTER_MODULE_PATH=/app/lib/ImageMagick-7.1.1/modules-Q16HDRI/filters";
+    char env3[] = "MAGICK_CONFIGURE_PATH=/app/lib/ImageMagick-7.1.1/config-Q16HDRI";
+    putenv(env1);
+    putenv(env2);
+    putenv(env3);
+
 #ifdef Q_OS_WIN

 #ifdef PQMEXIV2
