diff --git a/activity/activity.info b/activity/activity.info
index 9f075b4..1008e2c 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,9 +1,19 @@
 [Activity]
 name = Make Them Fall
 activity_version = 2
-bundle_id = org.sugarlabs.makethemfall
+bundle_id = org.sugarlabs.MakeThemFall
 icon = makethem
 exec = sugar-activity3 activity.MakeThemFallActivity
 license = GPLv3+
-summary = An arcade game
+metadata_license = CC0-1.0
+summary = It is a multilevel arcade game
 repository = https://github.com/sugarlabs/make-them-fall-activity.git
+release_date = 2024-06-13
+show_launcher = yes
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+developer_id = org.sugarlabs
+tags = Game
+description = Make Them Fall, is a challenging distance game that tests your multitasking skills. Control up to 6 falling stickmen and avoid spikes and obstacles. Choose from six difficulty levels and see how long you can last. Play alone or with friends for added fun!
+url = https://github.com/sugarlabs/make-them-fall-activity
+screenshots = https://i.imgur.com/yJYB6Mw.png https://i.imgur.com/4sbwa3D.png https://i.imgur.com/p6bub23.png https://i.imgur.com/qNmko6e.png https://i.imgur.com/XA6cK7D.png
\ No newline at end of file
