--- caps/Makefile	2018-10-18 03:17:36.000000000 -0400
+++ caps/Makefile-new	2020-05-18 21:35:59.981475714 -0400
@@ -1,7 +1,7 @@
 VERSION = 0.9.26
 
-PREFIX = /usr
-DESTDIR = 
+PREFIX ?= /usr
+DESTDIR ?= 
 
 CC = g++
 
@@ -22,8 +22,8 @@
 OBJECTS	= $(SOURCES:.cc=.o) 
 HEADERS = $(wildcard *.h) $(wildcard dsp/*.h) $(wildcard util/*.h) $(wildcard dsp/tonestack/*.h)
 
-DEST = $(PREFIX)/lib/ladspa
-RDFDEST = $(PREFIX)/share/ladspa/rdf
+DEST ?= $(PREFIX)/lib/ladspa
+RDFDEST ?= $(PREFIX)/share/ladspa/rdf
 
 # targets following -------------------------------------------------------------
 
