diff --git a/activity.py b/activity.py
index a0a8dd0..fe24e0e 100644
--- a/activity.py
+++ b/activity.py
@@ -6,9 +6,7 @@ gi.require_version('Gtk', '3.0')
 from gi.repository import Gtk
 import pygame
 
-from sugar3.activity.activity import Activity
 from sugar3.graphics.toolbarbox import ToolbarBox
-from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.graphics.toolbutton import ToolButton
 from sugar3.activity.widgets import StopButton
 
@@ -17,11 +15,15 @@ import sugargame.canvas
 from gettext import gettext as _
 from main import MakeThemFallGame
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
+from sugarapp.helpers import PrimaryMonitor
+from sugar3.graphics.style import GRID_CELL_SIZE
 
-class MakeThemFallActivity(Activity):
+class MakeThemFallActivity(SugarCompatibleActivity):
 
     def __init__(self, handle):
-        Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
         self.max_participants = 1
         self.sound = True
         self.game = MakeThemFallGame()
@@ -33,12 +35,17 @@ class MakeThemFallActivity(Activity):
         self.set_canvas(self.game.canvas)
         self.game.canvas.grab_focus()
 
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height()
+        #  6 added in order to remove space b/w top panel and pygame canvas
+        self.game.canvas.set_size_request(w, h - GRID_CELL_SIZE + 6)
+
     def build_toolbar(self):
         toolbar_box = ToolbarBox()
         self.set_toolbar_box(toolbar_box)
         toolbar_box.show()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
 
