From 3f58ca8961dc616810e8b9fa7083fa92877f2208 Mon Sep 17 00:00:00 2001
Message-Id: <3f58ca8961dc616810e8b9fa7083fa92877f2208.1653572934.git.matthias@mailaender.name>
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Thu, 26 May 2022 15:48:48 +0200
Subject: [PATCH] Split man page installation

as it requires the utility
---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 882b96a4..b8689aca 100644
--- a/Makefile
+++ b/Makefile
@@ -167,6 +167,8 @@ install-data:
 	@sh -c '. ./engine/packaging/functions.sh; install_data ./engine $(DESTDIR)$(gamedir)'
 	@rm -f "$(DESTDIR)$(gamedir)/global mix database.dat"
 	@cp -Lr mods/hv $(DESTDIR)$(gamedir)/mods/
+
+install-man: engine
 	@mkdir -p $(DESTDIR)$(mandir)/man6/
 	@./utility.sh all --man-page > $(DESTDIR)$(mandir)/man6/openhv.6
 
-- 
2.35.3

