diff --git a/math_hurdlerActivity.py b/math_hurdlerActivity.py
index dcf6000..9719d6c 100644
--- a/math_hurdlerActivity.py
+++ b/math_hurdlerActivity.py
@@ -2,9 +2,7 @@ import math_hurdler
 import sugargame.canvas
 from sugar3.activity.widgets import StopButton
 from sugar3.graphics.toolbutton import ToolButton
-from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.graphics.toolbarbox import ToolbarBox
-import sugar3.activity.activity
 import pygame
 
 from gettext import gettext as _
@@ -14,10 +12,15 @@ import gi
 gi.require_version('Gtk', '3.0')
 from gi.repository import Gtk
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
 
-class MathHurdlerActivity(sugar3.activity.activity.Activity):
+from sugar3.graphics.style import GRID_CELL_SIZE
+from sugarapp.helpers import PrimaryMonitor
+
+class MathHurdlerActivity(SugarCompatibleActivity):
     def __init__(self, handle):
-        super(MathHurdlerActivity, self).__init__(handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self.paused = False
         self.muted = False
@@ -28,24 +31,29 @@ class MathHurdlerActivity(sugar3.activity.activity.Activity):
         self.build_toolbar()
 
         # Build the Pygame canvas.
-        self._pygamecanvas = sugargame.canvas.PygameCanvas(
+        self.game.canvas = sugargame.canvas.PygameCanvas(
             self, main=self.game.run)
 
         # Note that set_canvas implicitly calls read_file when
         # resuming from the Journal.
-        self.set_canvas(self._pygamecanvas)
+        self.game.canvas.grab_focus()
+        self.set_canvas(self.game.canvas)
 
         # Start the game running (self.game.run is called when the
         # activity constructor returns).
         # Deprecated:Not required in sugargame v0.12
         # self._pygamecanvas.run_pygame(self.game.run)
 
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height()
+        self.game.canvas.set_size_request(w, h-GRID_CELL_SIZE)
+
     def build_toolbar(self):
         toolbar_box = ToolbarBox()
         self.set_toolbar_box(toolbar_box)
         toolbar_box.show()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
 
