diff --git a/activity/activity.info b/activity/activity.info
index 75d10e3..308a0b2 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,7 +1,18 @@
 [Activity]
 name = Math Hurdler
-bundle_id = edu.rit.math-hurdler
+bundle_id = org.sugarlabs.MathHurdler
 exec = sugar-activity3 math_hurdlerActivity.MathHurdlerActivity
 icon = activity-math-hurdler
 activity_version = 4
-license = MIT License
+release_date = 2024-07-21
+license = GPLv3+
+metadata_license = CC0-1.0
+summary = Math game for fourth grade children
+description = A game designed to help teach children math in accordance with the Common Core standard for fourth grade students.
+tags = Game
+url = https://github.com/sugarlabs/math-hurdler
+repository = https://github.com/sugarlabs/math-hurdler.git
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+developer_id = org.sugarlabs
+screenshots = https://i.imgur.com/ZRESoT0.png https://i.imgur.com/0tWmyEQ.png https://i.imgur.com/BsUEDWL.png
\ No newline at end of file
