diff -up maxima-5.37.1/src/maxima.in.clisp-noreadline maxima-5.37.1/src/maxima.in
--- maxima-5.37.1/src/maxima.in.clisp-noreadline	2015-09-14 09:49:14.367123471 -0500
+++ maxima-5.37.1/src/maxima.in	2015-09-14 09:51:24.943335449 -0500
@@ -96,6 +96,7 @@ process_commandline_arguments() {
 	    --userdir ) : ; shift;; # noop; handled by process_userdir_argument
 	    --userdir=*) : ;; # noop; handled by process_userdir_argument
 	    -v|--verbose ) verbose=true;;
+            --disable-readline ) CLISP_OPTIONS="-disable-readline";;
 	    *) ;;
 	esac
 	shift
@@ -133,9 +134,9 @@ fi
 if [ "$MAXIMA_LISP" = "clisp" ]; then
     if [ "$layout_autotools" = "true" ]; then
       if [ -x "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/maxima" ]; then
-        exec "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/maxima" $MAXIMA_LISP_OPTIONS -q "" -- "$@"
+        exec "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/maxima" $MAXIMA_LISP_OPTIONS $CLISP_OPTIONS -q "" -- "$@"
       else
-	exec "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/@CLISP_RUNTIME@" $MAXIMA_LISP_OPTIONS -q -M "$maxima_image_base.mem" "" -- "$@"
+	exec "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/@CLISP_RUNTIME@" $MAXIMA_LISP_OPTIONS $CLISP_OPTIONS -q -M "$maxima_image_base.mem" "" -- "$@"
       fi
     else
       if [ -x "$maxima_image_base" ]; then
