--- maxima-5.42.1/src/maxima.system.orig	2018-10-25 17:40:37.141080112 -0600
+++ maxima-5.42.1/src/maxima.system	2018-10-25 18:30:23.986042121 -0600
@@ -71,20 +71,16 @@ 
 	(let ((c::*ld-format* (concatenate 'string c::*ld-format* " /LIBPATH:binary-ecl")))
 	  (c::build-fasl "binary-ecl/maxima" :lisp-files '(maxima-lib))))
       #-msvc
-      (let ((obj (mapcar #'(lambda (p)
-			     ;; Convert dir/foo.fas to dir/foo.o
-			     (make-pathname :type "o" :defaults p))
-			 files)))
-	(c::build-fasl "binary-ecl/maxima" :lisp-files obj
-			  :ld-flags
-			  (let ((x (symbol-value (find-symbol "*AUTOCONF-LD-FLAGS*"
-							      (find-package "MAXIMA")))))
-			    (if (and x (not (string= x ""))) (list x))))
+      (let* ((obj (mapcar #'(lambda (p)
+			      ;; Convert dir/foo.fas to dir/foo.o
+			      (make-pathname :type "o" :defaults p))
+			  files))
+	     (ldflags (let ((x (symbol-value (find-symbol "*AUTOCONF-LD-FLAGS*"
+							  (find-package "MAXIMA")))))
+			(if (and x (not (string= x ""))) (c::split-program-options x)))))
+	(c::build-fasl "binary-ecl/maxima" :lisp-files obj :ld-flags ldflags)
 	(c::build-program "binary-ecl/maxima" :lisp-files obj
-			  :ld-flags
-			  (let ((x (symbol-value (find-symbol "*AUTOCONF-LD-FLAGS*"
-							      (find-package "MAXIMA")))))
-			    (if (and x (not (string= x ""))) (list x)))
+			  :ld-flags ldflags
 			  :epilogue-code '(progn (require :defsystem)
 					         (cl-user::run)))))))
 
