diff --git a/activity/activity.info b/activity/activity.info
index abb1d02..96ba2cd 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -3,9 +3,16 @@ name = Maze
 exec = sugar-activity3 activity.MazeActivity
 icon = activity-maze
 activity_version = 29
+release_date = 2018-04-12
 license = GPLv3+
-bundle_id = vu.lux.olpc.Maze
-summary = Good at finding your way out of hard situations? Let's see! Try to make your way through an increasingly difficult path, or you can also play with a friend!
-max_participants = 6
+metadata_license = CC0-1.0
+update_contact = tch@sugarlabs.org
+bundle_id = org.sugarlabs.Maze
+summary = A simple maze game
+max_participants = 1
 repository = https://github.com/sugarlabs/maze-activity
 tags = Game
+url = https://github.com/sugarlabs/maze-activity
+description = Good at finding your way out of hard situations? Let's see! Try to make your way through an increasingly difficult path.
+screenshots = https://i.imgur.com/imoTbzP.png https://i.imgur.com/GO11pbh.png https://i.imgur.com/RH1koq6.png
+developer_name = Sugar Labs Community
