diff --git a/game.py b/game.py
index 7bfa351..3c8085d 100644
--- a/game.py
+++ b/game.py
@@ -43,6 +43,8 @@ from maze import Maze, Rectangle
 from player import Player
 import sensors
 
+from sugarapp.helpers import PrimaryMonitor
+
 
 class MazeGame(Gtk.DrawingArea):
     """Maze game controller.
@@ -79,8 +81,7 @@ class MazeGame(Gtk.DrawingArea):
         # keep a list of all players, local and remote,
         self.allplayers = [] + self.localplayers
 
-        screen = Gdk.Screen.get_default()
-        self.aspectRatio = float(screen.width()) / screen.height()
+        self.aspectRatio = float(PrimaryMonitor.width()) / PrimaryMonitor.height()
 
         # start with a small maze using a seed that will be different
         # each time you play
@@ -152,8 +153,8 @@ class MazeGame(Gtk.DrawingArea):
 
     def __configure_cb(self, event):
         ''' Screen size has changed '''
-        width = Gdk.Screen.get_default().width()
-        height = Gdk.Screen.get_default().height() - style.GRID_CELL_SIZE
+        width = PrimaryMonitor.width()
+        height = PrimaryMonitor.height() - style.GRID_CELL_SIZE
         self.aspectRatio = width / height
 
         self._activity.busy()
