From 945bc6f5a57fc8af8964f90036c09afe1c9d4804 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Fri, 13 Dec 2019 11:32:50 -0300
Subject: [PATCH 5/5] fix-hw-detection

---
 measure.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/measure.py b/measure.py
index 6f10606..c94a921 100644
--- a/measure.py
+++ b/measure.py
@@ -109,11 +109,6 @@ def _get_firmware_number():
 
 
 def _get_hardware_model():
-    settings = Gio.Settings('org.sugarlabs.extensions.aboutcomputer')
-    model = settings.get_string('hardware-model')
-    if model:
-        return model
-
     model = _read_device_tree('mfg-data/MN')
 
     if model is None:
-- 
2.19.1

