From d1268049ca543f88e75f320ac4c81521aef1aba7 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Fri, 13 Dec 2019 10:35:50 -0300
Subject: [PATCH 2/5] monitors

---
 measure.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/measure.py b/measure.py
index b461916..ee0ad47 100644
--- a/measure.py
+++ b/measure.py
@@ -48,6 +48,7 @@ from config import ICONS_DIR, INSTRUMENT_DICT
 
 import logging
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import ExtendedActivityToolbarButton
 from sugarapp.widgets import SugarCompatibleActivity
 
@@ -326,7 +327,7 @@ class MeasureActivity(SugarCompatibleActivity):
     def __screen_size_changed_cb(self, event):
         ''' Screen size has changed, so check to see if the toolbar
         elements still fit.'''
-        self.width = Gdk.Screen.width()
+        self.width = PrimaryMonitor.width()
         if self.width < style.GRID_CELL_SIZE * 14:
             self._extras_button.show()
             if self._extra_tools in self._extra_item:
-- 
2.19.1

