From 74955778daf1bac3a2b4e95dd10cf02d5a5c5024 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Fri, 13 Dec 2019 10:48:30 -0300
Subject: [PATCH 2/4] info

---
 activity/activity.info | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index c3ee8ab..35917d0 100755
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,16 +1,17 @@
 [Activity]
 name = Measure
-bundle_id = org.laptop.MeasureActivity
+bundle_id = org.sugarlabs.Measure
 license = GPLv2+
 exec = sugar-activity measure.MeasureActivity
 icon = activity-measure
 activity_version = 102
-summary = Oscilloscope program for visualizing microphone input.
-tags = Science
+release_date = 2018-09-18
+summary = Oscilloscope program for visualizing microphone input
+tags = Education
 repository = https://github.com/sugarlabs/Measure
-url = https://help.sugarlabs.org/measure.html
+url = https://help.sugarlabs.org/en/measure.html
 metadata_license = CC0-1.0
-description:
-    <p>Program for visualizing microphone input</p>
-	<p>Using this program you can easily visualize your voice or even a musical instrument.</p>
-screenshots = https://github.com/sugarlabs/Measure/tree/master/screenshots/en/1.png https://github.com/sugarlabs/Measure/tree/master/screenshots/en/2.png https://github.com/sugarlabs/Measure/tree/master/screenshots/en/3.png
+description = Measure is a tool that turns the computer into an oscilloscope. You can easily visualize your voice or even a musical instrument. It provides an interface for the kids to connect with the physical world and an opportunity to view and understand through a visual and statistical representations.
+screenshots = https://raw.githubusercontent.com/sugarlabs/Measure/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/Measure/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/Measure/master/screenshots/en/3.png
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
-- 
2.19.1

