From 4a02981489b783008398c1e276632597d5f5a86d Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Fri, 13 Dec 2019 10:57:44 -0300
Subject: [PATCH 4/5] fix-close

---
 measure.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/measure.py b/measure.py
index ee0ad47..6f10606 100644
--- a/measure.py
+++ b/measure.py
@@ -170,7 +170,6 @@ class MeasureActivity(SugarCompatibleActivity):
         self._dsobject = None
 
         self.connect('notify::active', self._notify_active_cb)
-        self.connect('destroy', self.on_quit)
 
         self.data_logger = DataLogger(self)
 
@@ -350,9 +349,11 @@ class MeasureActivity(SugarCompatibleActivity):
             self.sensor_toolbar.trigger_label.show()
         self._extra_tools.show()
 
-    def on_quit(self, data=None):
+    def can_close(self, data=None):
         '''Clean up, close journal on quit'''
+        self.audiograb.set_logging_params(start_stop=False)
         self.audiograb.on_activity_quit()
+        return True
 
     def _notify_active_cb(self, widget, pspec):
         ''' Callback to handle starting/pausing capture when active/idle '''
-- 
2.19.1

