From 046dad45ce098f3c710500b1eee7e9ee8081594a Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Fri, 3 Jan 2020 12:53:33 -0300
Subject: [PATCH 5/5] info

---
 activity/activity.info | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index d5ed42e..1dcba65 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,17 +1,19 @@
 [Activity]
 name = Memorize
-bundle_id = org.laptop.Memorize
+bundle_id = org.sugarlabs.Memorize
 exec = sugar-activity3 activity.MemorizeActivity
 icon = activity-memorize
-activity_version = 57
+activity_version = 57.1
+release_date = 2019-11-13
 mime_types = application/x-memorize-project;
 license = GPLv2+;GPLv3+
-summary = Here you can play games that challenge your memory! But, more importantly, you can create your own game to test the memory of others!
+summary = Find matching pairs
 max_participants = 2
 repository = https://github.com/sugarlabs/memorize-activity.git
-tags = Maths;Language;Game
+tags = Game
 url = https://help.sugarlabs.org/memorize.html
 metadata_license = CC0-1.0
-description:
-    <p>Memorize is about finding matching pairs. A pair can be images, sounds and text and this could be extended to animations or movie snippets as well. Which pairs match is up to the creator of the game. Memorize is more than a predefined game you can play, it allows you to create new games yourself as well.</p>
-screenshots = https://github.com/sugarlabs/memorize-activity/tree/master/screenshots/en/1.png https://github.com/sugarlabs/memorize-activity/tree/master/screenshots/en/2.png https://github.com/sugarlabs/memorize-activity/tree/master/screenshots/en/3.png
+description = Memorize is about finding matching pairs. A pair can be images, sounds and text and this could be extended to animations or movie snippets as well. Which pairs match is up to the creator of the game. Memorize is more than a predefined game you can play, it allows you to create new games yourself as well.
+screenshots = https://raw.githubusercontent.com/sugarlabs/memorize-activity/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/memorize-activity/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/memorize-activity/master/screenshots/en/3.png
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
-- 
2.23.0

