From f8e5d8ba5f3b481809651d54aaabddc159922000 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Fri, 3 Jan 2020 12:28:15 -0300
Subject: [PATCH 3/5] fix-editor-size

---
 cardlist.py        | 2 +-
 createcardpanel.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cardlist.py b/cardlist.py
index 08a96a2..72071fd 100644
--- a/cardlist.py
+++ b/cardlist.py
@@ -64,7 +64,7 @@ class CardList(Gtk.EventBox):
         self._scroll.get_child().modify_bg(Gtk.StateType.NORMAL,
                                            Gdk.color_parse('#d7d7d7'))
         self.add(self._scroll)
-        self.set_size_request(-1, PAIR_SIZE * 2 + style.DEFAULT_SPACING * 4)
+        self.set_size_request(-1, PAIR_SIZE * 2)
         self.show_all()
 
     def load_game(self, game):
diff --git a/createcardpanel.py b/createcardpanel.py
index 688ff14..fd03d2b 100644
--- a/createcardpanel.py
+++ b/createcardpanel.py
@@ -58,7 +58,7 @@ class CreateCardPanel(Gtk.EventBox):
         def make_label(icon_name, label):
             label_box = Gtk.VBox()
             icon = Icon(icon_name=icon_name,
-                        pixel_size=style.LARGE_ICON_SIZE)
+                        pixel_size=style.MEDIUM_ICON_SIZE)
             label_box.pack_start(icon, False, False, 0)
             label = Gtk.Label(label=label)
             label.modify_fg(Gtk.StateType.NORMAL,
-- 
2.23.0

