--- a/meson/post_install.py	2018-02-10 19:52:37.000000000 +0100
+++ b/meson/post_install.py	2018-02-11 12:27:18.346549503 +0100
@@ -1,13 +1,25 @@
 #!/usr/bin/env python3
+
 import os
 import subprocess
 
-schemadir = os.path.join('/','usr', 'share', 'glib-2.0', 'schemas')
-icondir = os.path.join('/','usr', 'share', 'icons', 'hicolor')
+prefix = os.environ.get('MESON_INSTALL_PREFIX', '/usr/local')
+datadir = os.path.join(prefix, 'share')
+
+# Packaging tools define DESTDIR and this isn't needed for them
+if 'DESTDIR' not in os.environ:
+	
 
-if not os.environ.get('DESTDIR'):
     print('Compiling gsettings schemas...')
-    subprocess.call(['glib-compile-schemas', schemadir], shell=False)
+    schema_dir = os.path.join(datadir, 'glib-2.0/schemas')
+    subprocess.call(['glib-compile-schemas', schema_dir])
+
+    print('Updating icon cache...')
+    icon_cache_dir = os.path.join(datadir, 'icons', 'hicolor')
+
+    subprocess.call(['gtk-update-icon-cache', '-qtf', icon_cache_dir])
+
+    print('Updating desktop database...')
+    desktop_database_dir = os.path.join(datadir, 'applications')
+    subprocess.call(['update-desktop-database', '-q', desktop_database_dir])
 
-    print('Rebuilding desktop icons cache...')
-    subprocess.call(['update-icon-caches', icondir], shell=False)
