diff -u /var/home/deathwish/src/flatpak/org.gnome.Screenshot/a/src/org.gnome.Screenshot.metainfo.xml.in /var/home/deathwish/src/flatpak/org.gnome.Screenshot/b/src/org.gnome.Screenshot.metainfo.xml.in
--- a/src/org.gnome.Screenshot.metainfo.xml.in	2020-03-10 09:52:30.000000000 +0100
+++ b/src/org.gnome.Screenshot.metainfo.xml.in	2020-09-09 20:18:42.965479884 +0200
@@ -27,7 +27,7 @@
   <url type="donation">http://www.gnome.org/friends/</url>
   <screenshots>
     <screenshot type="default">
-      <image>https://git.gnome.org/browse/gnome-screenshot/plain/src/gnome-screenshot.png</image>
+      <image>https://gitlab.gnome.org/GNOME/gnome-screenshot/-/raw/master/data/screenshots/gnome-screenshot-3.38-2.png</image>
     </screenshot>
   </screenshots>
   <releases>
@@ -46,4 +46,5 @@
   <project_group>GNOME</project_group>
   <compulsory_for_desktop>GNOME</compulsory_for_desktop>
   <translation type="gettext">gnome-screenshot</translation>
+  <content_rating type="oars-1.0" />
 </component>

Diff finished.  Wed Sep  9 20:26:50 2020
