diff --git a/src/celestia/CMakeLists.txt b/src/celestia/CMakeLists.txt
index c76f7213..0cd8b4f1 100644
--- a/src/celestia/CMakeLists.txt
+++ b/src/celestia/CMakeLists.txt
@@ -144,6 +144,30 @@ else()
   )
 endif()
 
+find_program(GIT_FOUND git)
+if(GIT_FOUND)
+  execute_process(
+    COMMAND git log --tags=\[0-9\]* --no-walk --format=%as/%\(describe:tags\)
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+    OUTPUT_VARIABLE GIT_RELEASE_TAGS
+  )
+  string(REPLACE "\n" ";" GIT_RELEASE_TAGS "${GIT_RELEASE_TAGS}")
+  foreach(T IN LISTS GIT_RELEASE_TAGS)
+    if (T)
+      string(REPLACE "/" ";" F "${T}")
+      list(GET F 0 R_DATE)
+      list(GET F 1 R_TAG)
+      set(R_TYPE "stable")
+      string(REPLACE "-" "~" R_TAG "${R_TAG}")
+      if (R_TAG MATCHES "~")
+        set(R_TYPE "development")
+      endif()
+      list(APPEND METAINFO_RELEASES_LIST "<release version=\"${R_TAG}\" date=\"${R_DATE}\" type=\"${R_TYPE}\"><url>https://github.com/celestiaproject/Celestia/releases/tag/${R_TAG}</url></release>")
+    endif()
+  endforeach()
+  list(JOIN METAINFO_RELEASES_LIST "\n    " METAINFO_RELEASES)
+endif()
+
 add_subdirectory(gtk)
 add_subdirectory(qt5)
 add_subdirectory(qt6)
diff --git a/src/celestia/data/space.celestiaproject.Celestia.metainfo.xml.in b/src/celestia/data/space.celestiaproject.Celestia.metainfo.xml.in
index a110c6c7..fe3d09a0 100644
--- a/src/celestia/data/space.celestiaproject.Celestia.metainfo.xml.in
+++ b/src/celestia/data/space.celestiaproject.Celestia.metainfo.xml.in
@@ -45,4 +45,8 @@
       <image>https://celestiaproject.space/images/gallery/proxima.png</image>
     </screenshot>
   </screenshots>
+
+  <releases>
+    @METAINFO_RELEASES@
+  </releases>
 </component>
