--- a/src/Doom.metainfo.xml.in
+++ b/src/Doom.metainfo.xml.in
@@ -25,6 +25,15 @@
       network stack, allowing it to function on modern LANs and the
       Internet.
     </p>
+    <p>
+      The easiest way to run @PACKAGE_SHORTNAME@ Doom is to copy a
+      supported game file (known as a "WAD") to the root of your home
+      directory and launch the application.  If you have trouble with
+      the automatic game file detection or need to manage multiple WAD
+      files, you can run the application from the command line with a
+      path to the selected game file:
+      <code>flatpak run io.github.fabiangreffrath.Doom -iwad ~/PATH/TO/GAME.WAD</code>
+    </p>
   </description>
   <screenshots>
     <screenshot type="default">
@@ -44,4 +53,14 @@
       <caption>Doom the Way id Did, on Freedoom. Level 3-2: City of Corpses</caption>
     </screenshot>
   </screenshots>
+  <content_rating type="oars-1.1">
+    <content_attribute id="violence-fantasy">intense</content_attribute>
+    <content_attribute id="violence-realistic">intense</content_attribute>
+    <content_attribute id="violence-bloodshed">intense</content_attribute>
+    <content_attribute id="violence-desecration">intense</content_attribute>
+    <content_attribute id="social-chat">intense</content_attribute>
+  </content_rating>
+  <releases>
+    <release version="5.8.0" date="2020-04-17"/>
+  </releases>
 </component>
