Index: metis-5.1.0/programs/gpmetis.c
===================================================================
--- metis-5.1.0.orig/programs/gpmetis.c
+++ metis-5.1.0/programs/gpmetis.c
@@ -154,7 +154,6 @@ void GPPrintInfo(params_t *params, graph
 
   printf("******************************************************************************\n");
   printf("%s", METISTITLE);
-  printf(" (HEAD: %s, Built on: %s, %s)\n", SVNINFO, __DATE__, __TIME__);
   printf(" size of idx_t: %zubits, real_t: %zubits, idx_t *: %zubits\n", 
       8*sizeof(idx_t), 8*sizeof(real_t), 8*sizeof(idx_t *));
   printf("\n");
Index: metis-5.1.0/programs/graphchk.c
===================================================================
--- metis-5.1.0.orig/programs/graphchk.c
+++ metis-5.1.0/programs/graphchk.c
@@ -42,7 +42,6 @@ int main(int argc, char *argv[])
 
   printf("**********************************************************************\n");
   printf("%s", METISTITLE);
-  printf(" (HEAD: %s, Built on: %s, %s)\n", SVNINFO, __DATE__, __TIME__);
   printf(" size of idx_t: %zubits, real_t: %zubits, idx_t *: %zubits\n",
       8*sizeof(idx_t), 8*sizeof(real_t), 8*sizeof(idx_t *));
   printf("\n");
Index: metis-5.1.0/programs/m2gmetis.c
===================================================================
--- metis-5.1.0.orig/programs/m2gmetis.c
+++ metis-5.1.0/programs/m2gmetis.c
@@ -100,7 +100,6 @@ void M2GPrintInfo(params_t *params, mesh
 { 
   printf("******************************************************************************\n");
   printf("%s", METISTITLE);
-  printf(" (HEAD: %s, Built on: %s, %s)\n", SVNINFO, __DATE__, __TIME__);
   printf(" size of idx_t: %zubits, real_t: %zubits, idx_t *: %zubits\n", 
       8*sizeof(idx_t), 8*sizeof(real_t), 8*sizeof(idx_t *));
   printf("\n");
Index: metis-5.1.0/programs/mpmetis.c
===================================================================
--- metis-5.1.0.orig/programs/mpmetis.c
+++ metis-5.1.0/programs/mpmetis.c
@@ -120,7 +120,6 @@ void MPPrintInfo(params_t *params, mesh_
 
   printf("******************************************************************************\n");
   printf("%s", METISTITLE);
-  printf(" (HEAD: %s, Built on: %s, %s)\n", SVNINFO, __DATE__, __TIME__);
   printf(" size of idx_t: %zubits, real_t: %zubits, idx_t *: %zubits\n", 
       8*sizeof(idx_t), 8*sizeof(real_t), 8*sizeof(idx_t *));
   printf("\n");
Index: metis-5.1.0/programs/ndmetis.c
===================================================================
--- metis-5.1.0.orig/programs/ndmetis.c
+++ metis-5.1.0/programs/ndmetis.c
@@ -113,7 +113,6 @@ void NDPrintInfo(params_t *params, graph
 { 
   printf("******************************************************************************\n");
   printf("%s", METISTITLE);
-  printf(" (HEAD: %s, Built on: %s, %s)\n", SVNINFO, __DATE__, __TIME__);
   printf(" size of idx_t: %zubits, real_t: %zubits, idx_t *: %zubits\n", 
       8*sizeof(idx_t), 8*sizeof(real_t), 8*sizeof(idx_t *));
   printf("\n");
