Index: metis-5.1.0/Makefile
===================================================================
--- metis-5.1.0.orig/Makefile
+++ metis-5.1.0/Makefile
@@ -51,6 +51,10 @@ endif
 ifneq ($(cc), not-set)
     CONFIG_FLAGS += -DCMAKE_C_COMPILER=$(cc)
 endif
+ifneq ($(cflags), not-set)
+    CONFIG_FLAGS += -DCMAKE_C_FLAGS="$(cflags)" -DCMAKE_CXX_FLAGS="$(cflags)"
+endif
+
 
 VERNUM=5.1.0
 PKGNAME=metis-$(VERNUM)
