diff --git a/3rdparty/mpvqt/CMakeLists.txt b/3rdparty/mpvqt/CMakeLists.txt
index c1d491b..8237269 100644
--- a/3rdparty/mpvqt/CMakeLists.txt
+++ b/3rdparty/mpvqt/CMakeLists.txt
@@ -28,13 +28,6 @@ if(ECM_FOUND)
 
     set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/MpvQt")
 
-    include(KDEClangFormat)
-    file(GLOB_RECURSE ALL_CLANG_FORMAT_SOURCE_FILES *.cpp *.h *.hpp *.c)
-    kde_clang_format(${ALL_CLANG_FORMAT_SOURCE_FILES})
-
-    include(KDEGitCommitHooks)
-    kde_configure_git_pre_commit_hook(CHECKS CLANG_FORMAT)
-
     include(ECMSetupVersion)
     ecm_setup_version(PROJECT
         VARIABLE_PREFIX MPVQT
diff --git a/3rdparty/mpvqt/src/CMakeLists.txt b/3rdparty/mpvqt/src/CMakeLists.txt
index 5841717..aef5215 100644
--- a/3rdparty/mpvqt/src/CMakeLists.txt
+++ b/3rdparty/mpvqt/src/CMakeLists.txt
@@ -66,7 +66,7 @@ install(FILES
 install(FILES
     "${PROJECT_BINARY_DIR}/MpvQtConfig.cmake"
     "${PROJECT_BINARY_DIR}/MpvQtConfigVersion.cmake"
-    "${CMAKE_SOURCE_DIR}/cmake/FindLibmpv.cmake"
+    "${MpvQt_SOURCE_DIR}/cmake/FindLibmpv.cmake"
     DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
     COMPONENT Devel
 )
diff --git a/3rdparty/mpvqt/src/mpvrenderer.cpp b/3rdparty/mpvqt/src/mpvrenderer.cpp
index 50f4e5f..270a8ac 100644
--- a/3rdparty/mpvqt/src/mpvrenderer.cpp
+++ b/3rdparty/mpvqt/src/mpvrenderer.cpp
@@ -16,7 +16,7 @@
 #include "mpvabstractitem_p.h"
 #include "mpvcontroller.h"
 
-Q_STATIC_LOGGING_CATEGORY(MpvQt_MpvRenderer, "MpvQt.MpvRenderer")
+Q_LOGGING_CATEGORY(MpvQt_MpvRenderer, "MpvQt.MpvRenderer")
 
 static void *get_proc_address_mpv(void *ctx, const char *name)
 {
