diff -Naur mtPaint.orig/src/spawn.c mtPaint/src/spawn.c
--- mtPaint.orig/src/spawn.c	2018-12-02 00:22:37.691302000 +0100
+++ mtPaint/src/spawn.c	2018-12-02 18:18:40.735899196 +0100
@@ -782,9 +782,9 @@
 
 #else /* Linux */
 
-#define HANDBOOK_BROWSER "firefox"
-#define HANDBOOK_LOCATION "/usr/doc/mtpaint/index.html"
-#define HANDBOOK_LOCATION2 "/usr/share/doc/mtpaint/index.html"
+#define HANDBOOK_BROWSER "xdg-open"
+#define HANDBOOK_LOCATION "/app/share/doc/mtpaint-handbook/index.html"
+#define HANDBOOK_LOCATION2 "/app/share/doc/mtpaint-handbook/index.html"
 
 #endif
 
