diff -Naur mtPaint.orig/src/spawn.c mtPaint/src/spawn.c
--- mtPaint.orig/src/spawn.c	2018-12-02 00:22:37.691302000 +0100
+++ mtPaint/src/spawn.c	2018-12-03 06:07:23.684446985 +0100
@@ -421,6 +421,7 @@
 #ifndef WIN32
 	int i, j;
 	static char *row_def[][3] = {
+		/*
 		{"View EXIF data (leafpad)", "exif %f | leafpad"},
 		{"View filesystem data (xterm)", "xterm -hold -e ls -l %f"},
 		{"Edit in Gimp", "gimp %f"},
@@ -443,6 +444,18 @@
 		{"Rename *.jpeg to *.jpg", "rename .jpeg .jpg *.jpeg"},
 		{"Remove spaces from filenames", "for file in *\" \"*; do mv \"$file\" `echo $file | sed -e 's/ /_/g'`; done"},
 		{"Remove extra .jpg. from filename", "rename .jpg. . *.jpg.jpg"},
+		*/
+		{"Open via xdg-open", "xdg-open %f"},
+        {"Print image", "yad --print --type IMAGE --print-add-preview --filename %f"},
+		{"View EXIF data (viewer)", "exif %f 2>&1 | open-viewer"},
+		{"View EXIF data (editor)", "exif %f 2>&1 | open-editor"},
+		{"View file info (viewer)", "file %f 2>&1 | open-viewer"},
+		{"View file info (editor)", "file %f 2>&1 | open-editor"},
+		{"View filesystem data (viewer)", "ls -la %f 2>&1 | open-viewer"},
+		{"View filesystem data (editor)", "ls -la %f 2>&1 | open-editor"},
+		{"View image information (viewer)", "identify -verbose %f 2>&1 | open-viewer"},
+		{"View image information (editor)", "identify -verbose %f 2>&1 | open-editor"},
+		{"Open terminal", "open-terminal %f"},
 //		{"", ""},
 		{NULL, NULL, NULL}
 		},
