diff -rupN mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf/src/png.c mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf-new/src/png.c
--- mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf/src/png.c	2017-10-20 22:54:45.000000000 +0200
+++ mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf-new/src/png.c	2018-01-17 23:39:32.645262972 +0100
@@ -774,7 +774,7 @@ static int load_png(char *file_name, ls_
 	if (settings->bpp == 3)
 	{
 		png_set_strip_16(png_ptr);
-		png_set_gray_1_2_4_to_8(png_ptr);
+		png_set_expand_gray_1_2_4_to_8(png_ptr);
 		png_set_palette_to_rgb(png_ptr);
 		png_set_gray_to_rgb(png_ptr);
 
@@ -871,7 +871,7 @@ static int load_png(char *file_name, ls_
 		png_set_strip_alpha(png_ptr);
 		png_set_packing(png_ptr);
 		if ((color_type == PNG_COLOR_TYPE_GRAY) && (bit_depth < 8))
-			png_set_gray_1_2_4_to_8(png_ptr);
+		        png_set_expand_gray_1_2_4_to_8(png_ptr);
 		for (i = 0; i < height; i++)
 		{
 			row_pointers[i] = settings->img[CHN_IMAGE] + i * width;
