diff -rupN mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf/configure mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf-new/configure
--- mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf/configure	2017-10-20 22:54:45.000000000 +0200
+++ mtPaint-3884b6d6ebc18511df7bf1485a19fe774dd61dcf-new/configure	2018-01-17 23:39:32.661262971 +0100
@@ -492,7 +492,6 @@ else # $OPTS = CFLAGS
 	MARCH=
 	# And leave CFLAGS alone
 fi
-[ "$OPTS" != DEBUG ] && LDFLAGS="-s $LDFLAGS" # Strip debug info
 # Set Windows-specific flags
 if [ "$OS" != "${OS#MinGW/}" ]
 then
