diff --git a/muse3/muse/mplugins/mitplugin.cpp b/muse3/muse/mplugins/mitplugin.cpp
index 2cc2f209..be657e73 100644
--- a/muse3/muse/mplugins/mitplugin.cpp
+++ b/muse3/muse/mplugins/mitplugin.cpp
@@ -21,6 +21,8 @@
 //
 //=========================================================
 
+#include <QAction>
+
 #include "mitplugin.h"
 #include "app.h"
 #include "event.h"
diff --git a/muse3/muse/widgets/genset.cpp b/muse3/muse/widgets/genset.cpp
index 6b2a566e..568afff5 100644
--- a/muse3/muse/widgets/genset.cpp
+++ b/muse3/muse/widgets/genset.cpp
@@ -29,6 +29,7 @@
 #include <QRect>
 #include <QShowEvent>
 #include <QString>
+#include <QButtonGroup>
 
 #include "genset.h"
 #include "app.h"
