diff --git a/activity/activity.info b/activity/activity.info
index d181a74..411481f 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -5,8 +5,15 @@ bundle_id = org.sugarlabs.MusicKeyboard
 exec = sugar-activity activity.SimplePianoActivity
 icon = activity-simplepiano
 license = GPLv2+
-summary = Play music with this piano keyboard.
+summary = Play music with this piano keyboard
 tags = Music
 repository = https://github.com/godiard/music-keyboard-activity
 mime_types = application/x-musickeyboard-activity;
 max_participants = 1
+metadata_license = CC0-1.0
+description = This activity display a piano keyboard in the screen and can be used to play music selecting sounds from different instruments. In devices with touch screen, can be used touching the screen, in devices without one, the laptop keyboard can be used.
+update_contact = tch@sugarlabs.org
+release_date = 2018-09-17
+screenshots = https://raw.githubusercontent.com/godiard/music-keyboard-activity/master/screenshots/en/1.png
+url = https://help.sugarlabs.org/en/musickeyboard.html
+developer_name = Sugar Labs Community
