diff -urNp a/snmplib/transports/snmpUDPIPv6Domain.c b/snmplib/transports/snmpUDPIPv6Domain.c
--- a/snmplib/transports/snmpUDPIPv6Domain.c	2019-01-24 09:03:05.606441678 +0100
+++ b/snmplib/transports/snmpUDPIPv6Domain.c	2019-02-07 08:59:26.434587244 +0100
@@ -464,7 +464,7 @@ netsnmp_udp6_transport(const struct sock
                                               NETSNMP_DS_LIB_CLIENT_ADDR);
         if (client_socket) {
             struct sockaddr_in6 client_addr;
-            if(!netsnmp_sockaddr_in6_2(&client_addr, client_socket, NULL)) {
+            if(netsnmp_sockaddr_in6_2(&client_addr, client_socket, NULL)) {
                 return netsnmp_udp6_transport_with_source(addr, local,
                                                           &client_addr);
             }
