diff -urNp old/snmplib/transports/snmpUDPIPv4BaseDomain.c new/snmplib/transports/snmpUDPIPv4BaseDomain.c
--- old/snmplib/transports/snmpUDPIPv4BaseDomain.c	2019-06-27 08:40:48.663969034 +0200
+++ new/snmplib/transports/snmpUDPIPv4BaseDomain.c	2019-06-27 08:42:05.293723487 +0200
@@ -317,7 +317,7 @@ netsnmp_udpipv4base_tspec_transport(nets
     if (NULL != tspec->source) {
         struct sockaddr_in src_addr, *srcp = &src_addr;
         /** get sockaddr from source */
-        if (!netsnmp_sockaddr_in2(&src_addr, tspec->source, NULL))
+        if (!netsnmp_sockaddr_in2(&src_addr, tspec->source, ":0"))
             return NULL;
         return netsnmp_udpipv4base_transport_with_source(&addr, local, srcp);
      } else {
@@ -364,7 +364,7 @@ netsnmp_udpipv4base_transport(const stru
                 strcat(client_address, ":0");
                 have_port = 1;
             }
-            rc = netsnmp_sockaddr_in2(&client_addr, client_socket, NULL);
+            rc = netsnmp_sockaddr_in2(&client_addr, client_socket, ":0");
             if (client_address != client_socket)
                 free(client_address);
             if(rc) {
