diff -Naur netpbm-10.84.05.orig/buildtools/configure.pl netpbm-10.84.05/buildtools/configure.pl
--- netpbm-10.84.05.orig/buildtools/configure.pl	2019-01-19 15:59:43.599120000 +0100
+++ netpbm-10.84.05/buildtools/configure.pl	2019-01-19 16:29:32.257768682 +0100
@@ -2075,15 +2075,21 @@
 autoFlushStdout();
 
 my $configInPathArg;
-if (@ARGV > 0) {
+while (@ARGV > 0) {
     if ($ARGV[0] =~ "^-") {
         if ($ARGV[0] eq "--help") {
             help();
             exit(0);
         } else {
-            die("Unrecognized option: $ARGV[0]");
+            print("Unrecognized option: $ARGV[0]\n");
+            shift(@ARGV);
         }
-    } 
+    }
+    else {
+        last;
+    }
+}
+if (@ARGV > 0) {
     $configInPathArg = $ARGV[0];
 }
 
