diff --git a/lazpaint/uconfig.pas b/lazpaint/uconfig.pas
index ee16357..b13b491 100644
--- a/lazpaint/uconfig.pas
+++ b/lazpaint/uconfig.pas
@@ -804,7 +804,7 @@ end;
 
 function TLazPaintConfig.DefaultUseImageBrowser: boolean;
 begin
-  result := iniOptions.ReadBool('General','UseImageBrowser',{$IFDEF DARWIN}false{$ELSE}true{$ENDIF});
+  result := iniOptions.ReadBool('General','UseImageBrowser',{$IFDEF DARWIN}false{$ELSE}false{$ENDIF});
 end;
 
 procedure TLazPaintConfig.SetDefaultUseImageBrowser(value: boolean);
