diff --git a/src/openrct2/CMakeLists.txt b/src/openrct2/CMakeLists.txt
index b0bd3e9..e937b45 100644
--- a/src/openrct2/CMakeLists.txt
+++ b/src/openrct2/CMakeLists.txt
@@ -25,7 +25,7 @@ SET_CHECK_CXX_FLAGS(${PROJECT_NAME})
 # GCC / Clang likes us to pass the -lstdc++fs flag to link C++17 filesystem implementation.
 if (NOT MINGW AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
     if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-        target_link_libraries(${PROJECT_NAME} stdc++fs)
+        #target_link_libraries(${PROJECT_NAME} stdc++fs)
     endif()
 endif()
 
