--- ImageLounge/src/DkGui/DkNoMacs.h	2018-10-22 11:25:33.000000000 +0200
+++ ImageLounge/src/DkGui/DkNoMacs_patched.h	2019-04-27 11:33:04.549811250 +0200
@@ -29,6 +29,8 @@
 
 #pragma once
 
+#include <iostream>
+
 // Qt
 #pragma warning(push, 0)	// no warnings from includes - begin
 #include <QMainWindow>
--- ImageLounge/src/DkCore/DkMetaData.h	2018-10-22 11:25:33.000000000 +0200
+++ ImageLounge/src/DkCore/DkMetaData_patched.h	2019-04-27 11:51:50.646599865 +0200
@@ -27,6 +27,8 @@
 
 #pragma once
 
+#include <iostream>
+
 #pragma warning(push, 0)	// no warnings from includes - begin
 #include <QSharedPointer>
 #include <QStringList>
