diff -up app/main/main.js.platform app/main/main.js
--- app/main/main.js.platform	2020-12-10 17:19:40.477012805 +0100
+++ app/main/main.js	2020-12-10 17:20:24.974560519 +0100
@@ -85,12 +85,10 @@ async function handleReady() {
         return;
     }
     let relativeUrl;
-    if (process.platform === "win32") {
-        const { argv } = process;
-        const url = argv.find(arg => arg.startsWith(config_1.default.protocol + ":"));
-        if (url) {
-            relativeUrl = makeRelativeUrl(url);
-        }
+    const { argv } = process;
+    const url = argv.find(arg => arg.startsWith(config_1.default.protocol + ":"));
+    if (url) {
+        relativeUrl = makeRelativeUrl(url);
     }
     startup(relativeUrl);
 }
@@ -128,11 +126,9 @@ electron_1.app.on("open-url", (event, ur
 });
 if (electron_1.app.requestSingleInstanceLock()) {
     electron_1.app.on("second-instance", (event, argv, workingDirectory) => {
-        if (process.platform === "win32") {
-            const url = argv.find(arg => arg.startsWith(config_1.default.protocol + ":"));
-            const urlPath = url && makeRelativeUrl(url);
-            handleActivate(urlPath);
-        }
+        const url = argv.find(arg => arg.startsWith(config_1.default.protocol + ":"));
+        const urlPath = url && makeRelativeUrl(url);
+        handleActivate(urlPath);
     });
 }
 else {
