From 9bb1f3d41a391d086e26d069410e84a6bda3aa1b Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Sun, 27 Jan 2019 15:44:35 +0100
Subject: [PATCH] Set automake strictness to foreign

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 3b3c41d..6f26110 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6,7 +6,7 @@ AC_INIT([ocrfeeder],
 
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE([1.11])
+AM_INIT_AUTOMAKE([1.11 foreign])
 AM_MAINTAINER_MODE([enable])
 # Support silent build rules. Disable by either passing --disable-silent-rules
 # to configure or passing V=1 to make
-- 
2.18.1

