diff --git a/src/ocrfeeder/feeder/layoutAnalysis.py b/src/ocrfeeder/feeder/layoutAnalysis.py
index a2b4edc4231116ac1d942a43093e73ba5cec9ddb..a2fc873545f026b1991faef58487024c35b02d81 100644
--- a/src/ocrfeeder/feeder/layoutAnalysis.py
+++ b/src/ocrfeeder/feeder/layoutAnalysis.py
@@ -19,7 +19,7 @@
 ###########################################################################
 
 from ocrfeeder.util.lib import debug
-from ocrfeeder.util import graphics
+from ocrfeeder.util import graphics, lib
 from ocrfeeder.util.constants import OCRFEEDER_DEBUG, DTP
 from ocrfeeder.studio.dataHolder import DataBox
 from imageManipulation import ImageProcessor
@@ -547,7 +547,7 @@ class LayoutAnalysis(object):
         text = self.ocr_engine.read()
         if self.clean_text:
             text = self.__cleanText(text)
-        return text
+        return lib.ensureUnicode(text)
 
     def __cleanText(self, text):
         clean_text = re.sub(r'(?<!-)-\n(?!\n)', r'', text)
