diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/feeder/ocrEngines.py ocrfeeder-release_0.8.1/src/ocrfeeder/feeder/ocrEngines.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/feeder/ocrEngines.py	2019-01-20 02:38:20.661497000 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/feeder/ocrEngines.py	2019-01-21 08:45:16.965574886 +0100
@@ -211,7 +211,7 @@
         if not len(self.ocr_engines):
             lib.debug("Warning: no engines found!")
         elif not favorite_engine_exists:
-            self.configuration_manager.favorite_engine = self.ocr_engines[0][0].name
+            self.configuration_manager.favorite_engine = self.ocr_engines[-1][0].name
         engines_needing_update = {'auto': [],
                                   'manual': []}
         for engine, path in self.ocr_engines:
@@ -260,7 +260,7 @@
         return engine
 
     def getXmlFilesInFolder(self, folder):
-        return [os.path.join(folder, file) for file in os.listdir(folder) if file.endswith('.xml')]
+        return [os.path.join(folder, file) for file in sorted(os.listdir(folder)) if file.endswith('.xml')]
 
     def newEngine(self, name, engine_path, arguments,
                   image_format, failure_string, languages,
diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py	2019-01-20 02:38:20.673497000 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py	2019-01-21 08:44:42.281177855 +0100
@@ -1443,7 +1443,7 @@
         try:
             index = [engine.name for engine, path in self.ocr_engines].index(self.configuration_manager.favorite_engine)
         except ValueError:
-            index = 0
+            index = 0 if len(self.ocr_engines) == 0 else len(self.ocr_engines) - 1
         self.engines_combo.set_active(index)
         engines_box = Gtk.HBox(spacing = 10)
         label = Gtk.Label(_("Favorite _engine:"))
diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/util/configuration.py ocrfeeder-release_0.8.1/src/ocrfeeder/util/configuration.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/util/configuration.py	2019-01-20 02:38:20.685497000 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/util/configuration.py	2019-01-21 07:42:00.673832882 +0100
@@ -113,7 +113,7 @@
                 UNPAPER_USE_BLACK_FILTER: True,
                 UNPAPER_NOISE_FILTER_INTENSITY: 'auto',
                 UNPAPER_GRAY_FILTER_SIZE: 'auto',
-                FAVORITE_ENGINE: 'ocrad',
+                FAVORITE_ENGINE: 'tesseract',
                 IMPROVE_COLUMN_DETECTION: True,
                 COLUMN_MIN_WIDTH: 'auto',
                 CLEAN_TEXT: True,
@@ -171,7 +171,7 @@
                             language_argument = language_argument,
                             version = version)
             existing_engines.append(engine)
-        return existing_engines
+        return sorted(existing_engines, key=lambda x: x.name, reverse=False)
 
     def setConf(self, conf_key, value):
         ConfigurationManager.conf[conf_key] = value
