diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py	2019-01-21 11:30:33.264175361 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py	2019-01-21 15:13:11.517513121 +0100
@@ -1312,7 +1312,7 @@
         self.ocr_engines = ocr_engines
         self.notebook = Gtk.Notebook()
         self.__makeGeneralPreferences(self.__makeColors())
-        self.__makeToolsPreferences(self.__makeUnpaper(),
+        self.__makeToolsPreferences(#self.__makeUnpaper(),
                                     self.__makeEngines(),
                                     self.__makePreProcessorPreferences())
         self.__makeRecognitionPreferences(self.__makeLanguagePreferences(),
@@ -1320,7 +1320,7 @@
                                       self.__makeWindowSize(),
                                       self.__makeColumnDetectionPreferences(),
                                       self.__makeBoundsAdjustmentsPreferences())
-        self.unpaper_select.connect('clicked', self.__unpaperSelectDialog)
+        #self.unpaper_select.connect('clicked', self.__unpaperSelectDialog)
         self.custom_window_size.connect('toggled', self.__toggledCustomWindowSize)
         self.vbox.add(self.notebook)
         self.set_icon_from_file(WINDOW_ICON)
@@ -1342,7 +1342,7 @@
         self.configuration_manager.setBoxesStroke(
             self.__getColor(self.boxes_stroke_color))
         self.configuration_manager.setImageFill(self.__getColor(self.image_fill_color))
-        self.configuration_manager.setUnpaper(self.unpaper_entry.get_text())
+        #self.configuration_manager.setUnpaper(self.unpaper_entry.get_text())
         self.configuration_manager.improve_column_detection = \
             self.improve_column_detection.get_active()
         self.configuration_manager.column_min_width = self.__getColumnMinWidth()
