diff --git a/src/ocrfeeder/studio/widgetModeler.py b/src/ocrfeeder/studio/widgetModeler.py
index f94184effa919a4710bb21eb0da3fd27f6de9ee1..d2b0f1bcccbcb273c2cca3493e322e3100ab1fa2 100644
--- a/src/ocrfeeder/studio/widgetModeler.py
+++ b/src/ocrfeeder/studio/widgetModeler.py
@@ -61,7 +61,6 @@ class ImageReviewer(Gtk.Paned):
         self.selectable_boxes_area.connect('deselected_box',
                                            self.deselectedBoxCb)
         self.image_pixbuf = GdkPixbuf.Pixbuf.new_from_file(self.path_to_image)
-        self.set_position(500)
         self.show()
         self.ocr_engines = ocr_engines
         self.editor = Editor(self.image_pixbuf, self.ocr_engines, self)
@@ -76,7 +75,7 @@ class ImageReviewer(Gtk.Paned):
         selectable_boxes_scrolled_window.show()
 
         self.pack1(selectable_boxes_scrolled_window, True, False)
-        self.pack2(self.editor.box_editor, True, False)
+        self.pack2(self.editor.box_editor, False, True)
         self.page = page_data
         self.updatePageData(self.page)
         selectable_boxes_scrolled_window.connect_after("size-allocate", self.zoomFitCb)
diff --git a/src/ocrfeeder/studio/widgetPresenter.py b/src/ocrfeeder/studio/widgetPresenter.py
index 240f4f0af7dde4ad82dd7465df59e79089dca3d4..b06fcbfd202e437200d6d0f3dd0c960594135ad0 100644
--- a/src/ocrfeeder/studio/widgetPresenter.py
+++ b/src/ocrfeeder/studio/widgetPresenter.py
@@ -325,6 +325,8 @@ class BoxEditor(Gtk.ScrolledWindow):
 
     def __init__(self, image_width = 0, image_height = 0, pixbuf = 0, x = 0, y = 0, width = 0, height = 0, ocr_engines_list = []):
         super(BoxEditor, self).__init__()
+
+        self.set_size_request(300, -1)
         self.get_accessible().set_name(_('Area editor'))
         self.set_policy(Gtk.PolicyType.AUTOMATIC, Gtk.PolicyType.AUTOMATIC)
         self.contents = Gtk.VBox()
