From 23aa40c3c2761a7eada829f1af05ec32a210d98d Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Sun, 27 Jan 2019 00:41:32 +0100
Subject: [PATCH] Update desktop file

---
 resources/ocrfeeder.desktop.in | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/resources/ocrfeeder.desktop.in b/resources/ocrfeeder.desktop.in
index ede5108..bf1641f 100644
--- a/resources/ocrfeeder.desktop.in
+++ b/resources/ocrfeeder.desktop.in
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Version=0.5
+Version=1.1
 Type=Application
 _Name=OCRFeeder
 _Comment=The complete OCR suite.
@@ -7,4 +7,8 @@ TryExec=ocrfeeder
 Exec=ocrfeeder %F
 Icon=ocrfeeder
 MimeType=image/bmp;image/gif;image/jpeg;image/jpg;image/pjpeg;image/png;image/tiff;
-Categories=Application;Office;
+Categories=Office;Graphics;OCR;Scanning;
+Keywords=OCR;optical character recognition;Scanner;tesseract;
+Terminal=false
+StartupNotify=true
+StartupWMClass=ocrfeeder
-- 
2.18.1

