From: Joaquim Rocha <me@joaquimrocha.com>
Subject: Fix unicode problems when generating a plain text file
Origin: https://git.gnome.org/browse/ocrfeeder/commit/?id=691f54618ed17a2553f154af07a6cfb4bf887e09
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=765847
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=811571
Index: ocrfeeder/src/ocrfeeder/feeder/documentGeneration.py
===================================================================
--- ocrfeeder.orig/src/ocrfeeder/feeder/documentGeneration.py
+++ ocrfeeder/src/ocrfeeder/feeder/documentGeneration.py
@@ -358,7 +358,7 @@ class PlaintextGenerator(DocumentGenerat
         self.text = ''
 
     def addText(self, newText):
-        self.text += newText
+        self.text += unicode(newText, 'utf-8')
 
     def addPage(self, page):
         self.addText(page.getTextFromBoxes())
