diff --git a/src/ocrfeeder/studio/studioBuilder.py b/src/ocrfeeder/studio/studioBuilder.py
index 6ab2404ffc61bd3a65e57dcfd28a8c5bea6b1b39..8e1ed01cb1442aaefb254edd298b6f0bac5de3eb 100644
--- a/src/ocrfeeder/studio/studioBuilder.py
+++ b/src/ocrfeeder/studio/studioBuilder.py
@@ -19,6 +19,10 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 ###########################################################################
 
+import gi
+gi.require_version('Gtk', '3.0')
+gi.require_version('GooCanvas', '2.0')
+gi.require_version('GtkSpell', '3.0')
 from ocrfeeder.util import lib
 from ocrfeeder.util.constants import *
 import sys
