diff --git a/src/ocrfeeder/studio/widgetPresenter.py b/src/ocrfeeder/studio/widgetPresenter.py
index 43cc281168547aaa1fb0e1533042b397c985fdc3..240f4f0af7dde4ad82dd7465df59e79089dca3d4 100644
--- a/src/ocrfeeder/studio/widgetPresenter.py
+++ b/src/ocrfeeder/studio/widgetPresenter.py
@@ -367,7 +367,8 @@ class BoxEditor(Gtk.ScrolledWindow):
         self.image_width = image_width
         self.image_height = image_height
 
-        image_frame = PlainFrame(_('Clip'))
+        image_frame = PlainExpander(_('Clip'))
+        image_frame.set_expanded(True)
         image_frame.add(self.image_window)
         self.contents.pack_start(image_frame, False, False, 0)
 
