diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetModeler.py ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetModeler.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetModeler.py	2019-01-22 07:36:19.181915476 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetModeler.py	2019-01-22 09:02:35.538044182 +0100
@@ -74,7 +74,7 @@
         selectable_boxes_scrolled_window.show()
 
         self.pack1(selectable_boxes_scrolled_window, True, False)
-        self.pack2(self.editor.box_editor, False, True)
+        self.pack2(self.editor.box_editor, False, False)
         self.page = page_data
         self.updatePageData(self.page)
         selectable_boxes_scrolled_window.connect_after("size-allocate", self.zoomFitCb)
diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py	2019-01-22 07:36:19.181915476 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py	2019-01-22 08:58:16.406019238 +0100
@@ -53,7 +53,7 @@
 
         self.main_area = Gtk.Paned.new(orientation=Gtk.Orientation.HORIZONTAL)
         self.main_area.set_wide_handle(True)
-        self.main_area.set_position(150)
+        self.main_area.set_position(200)
         self.main_area.show()
         self.main_box.pack_end(self.main_area, True, True, 0)
 
@@ -61,14 +61,15 @@
         self.main_area_left = Gtk.ScrolledWindow()
         self.main_area_left.get_accessible().set_name(_('Pages'))
         self.main_area_left.set_policy(Gtk.PolicyType.AUTOMATIC, Gtk.PolicyType.AUTOMATIC)
+        self.main_area_left.set_size_request(200, -1)
         self.main_area_left.show()
 
-        self.main_area.pack1(self.main_area_left, False, True)
+        self.main_area.pack1(self.main_area_left, False, False)
         self.notebook = Gtk.Notebook()
         self.notebook.set_show_tabs(False)
         self.notebook.set_show_border(False)
         self.notebook.show()
-        self.main_area.pack2(self.notebook, False, False)
+        self.main_area.pack2(self.notebook, True, False)
         self.action_group = None
 
         self.show()
@@ -325,7 +326,7 @@
     def __init__(self, image_width = 0, image_height = 0, pixbuf = 0, x = 0, y = 0, width = 0, height = 0, ocr_engines_list = []):
         super(BoxEditor, self).__init__()
 
-        self.set_size_request(300, -1)
+        self.set_size_request(400, -1)
         self.get_accessible().set_name(_('Area editor'))
         self.set_policy(Gtk.PolicyType.AUTOMATIC, Gtk.PolicyType.AUTOMATIC)
         self.contents = Gtk.VBox()
