diff --git a/src/ocrfeeder/studio/widgetModeler.py b/src/ocrfeeder/studio/widgetModeler.py
index 9be01095d6dff1dc41c82169d0d79c2d26f6b713..f94184effa919a4710bb21eb0da3fd27f6de9ee1 100644
--- a/src/ocrfeeder/studio/widgetModeler.py
+++ b/src/ocrfeeder/studio/widgetModeler.py
@@ -46,6 +46,7 @@ class ImageReviewer(Gtk.Paned):
 
     def __init__(self, main_window, page_data, ocr_engines):
         super(ImageReviewer, self).__init__(orientation=Gtk.Orientation.HORIZONTAL)
+        self.set_wide_handle(True)
         self.configuration_manager = ConfigurationManager()
         self.main_window = main_window
         self.path_to_image = page_data.image_path
diff --git a/src/ocrfeeder/studio/widgetPresenter.py b/src/ocrfeeder/studio/widgetPresenter.py
index 957518a46e9427002d3faf57f0a518a26fcb3738..09ad1e28f2d56b71df100fa7c55f7141d9bc0ec4 100644
--- a/src/ocrfeeder/studio/widgetPresenter.py
+++ b/src/ocrfeeder/studio/widgetPresenter.py
@@ -53,6 +53,7 @@ class MainWindow(Gtk.Window):
         self.main_box.pack_end(self.statusbar, False, False, 0)
 
         self.main_area = Gtk.Paned.new(orientation=Gtk.Orientation.HORIZONTAL)
+        self.main_area.set_wide_handle(True)
         self.main_area.set_position(150)
         self.main_area.show()
         self.main_box.pack_end(self.main_area, True, True, 0)
