From efc3260c84a983c5894345a813224f5f2371e79a Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Sun, 27 Jan 2019 16:46:23 +0100
Subject: [PATCH] Increase window size

---
 src/ocrfeeder/studio/widgetPresenter.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ocrfeeder/studio/widgetPresenter.py b/src/ocrfeeder/studio/widgetPresenter.py
index e634aec..f230cfd 100644
--- a/src/ocrfeeder/studio/widgetPresenter.py
+++ b/src/ocrfeeder/studio/widgetPresenter.py
@@ -43,7 +43,7 @@ class MainWindow(Gtk.Window):
 
     def __init__(self):
         Gtk.Window.__init__(self, Gtk.WindowType.TOPLEVEL)
-        self.set_size_request(800, 600)
+        self.set_size_request(1280, 720)
         self.set_default_icon_name(APP_ID)
         self.main_box = Gtk.VBox()
         self.main_box.show()
-- 
2.18.1

