diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetPresenter.py	2019-01-21 10:27:01.138176570 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetPresenter.py	2019-01-21 11:07:50.644484434 +0100
@@ -42,7 +42,7 @@
 
     def __init__(self):
         Gtk.Window.__init__(self, Gtk.WindowType.TOPLEVEL)
-        self.set_size_request(800, 600)
+        self.set_size_request(1280, 720)
         self.set_icon_from_file(WINDOW_ICON)
         self.main_box = Gtk.VBox()
         self.main_box.show()
