diff --git a/configure.ac b/configure.ac
index fcd65bff83014eb36d6be52bad25d37b6a5fec5a..b74ee3659bdaa7b21a9daf90518cc4d6c6ab789d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -51,7 +51,7 @@ dnl == intltool check ==
 IT_PROG_INTLTOOL([0.35.0])
 
 dnl == Documentation ==
-GNOME_DOC_INIT
+YELP_HELP_INIT
 
 dnl == generate makefiles ==
 AC_OUTPUT([
diff --git a/help/Makefile.am b/help/Makefile.am
index cf71a69c8bf86b07ade1d34608039a2b0703eb50..34c314b679aacb11bc29a671489954b186ba0fc0 100644
--- a/help/Makefile.am
+++ b/help/Makefile.am
@@ -1,12 +1,12 @@
-include $(top_srcdir)/gnome-doc-utils.make
+@YELP_HELP_RULES@
 
-DOC_ID = ocrfeeder
+HELP_ID = ocrfeeder
 
-DOC_FIGURES = \
+HELP_MEDIA = \
 	figures/areas-edition.png \
 	figures/content-areas.png
 
-DOC_PAGES = \
+HELP_FILES = \
 	addingfolder.page \
 	addingimage.page \
 	automaticrecognition.page \
@@ -22,6 +22,4 @@ DOC_PAGES = \
 	projects.page \
 	unpaper.page
 
-DOC_LINGUAS = cs de el es fr gl ru sl uk
-
-dist-hook: doc-dist-hook
+HELP_LINGUAS = cs de el es fr gl ru sl uk
diff --git a/src/ocrfeeder/studio/studioBuilder.py b/src/ocrfeeder/studio/studioBuilder.py
index 7f2dff531f55c016f4a064fb1992d10501c75344..8f0ae3db210319d582c3598a89be48a8a690c31e 100644
--- a/src/ocrfeeder/studio/studioBuilder.py
+++ b/src/ocrfeeder/studio/studioBuilder.py
@@ -439,7 +439,7 @@ class Studio:
 
     def showHelpContents(self, widget = None):
         Gtk.show_uri(self.main_window.get_screen(),
-                     'ghelp:ocrfeeder',
+                     'help:ocrfeeder',
                      Gtk.get_current_event_time())
 
     def zoomIn(self, widget = None):
