diff -Naur ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetModeler.py ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetModeler.py
--- ocrfeeder-release_0.8.1.orig/src/ocrfeeder/studio/widgetModeler.py	2014-12-22 01:09:05.000000000 +0100
+++ ocrfeeder-release_0.8.1/src/ocrfeeder/studio/widgetModeler.py	2019-01-22 19:05:10.057244691 +0100
@@ -249,6 +249,9 @@
             debug(exception.message)
             return
         self.selectable_boxes_area.setBackgroundImage(self.path_to_image)
+        self.editor.updateImage(self.image_pixbuf)
+        if self.editor.box:
+            self.editor.update(self.editor.box)
 
     def updateBoxesColors(self):
         self.editor.updateBoxColor()
@@ -1033,6 +1036,9 @@
         self.__connectDataBoxSignals()
         self.__updateBoxColor(None, self.data_box.type)
 
+    def updateImage(self, pixbuf):
+        self.pixbuf = pixbuf
+
     def __connectDataBoxSignals(self):
         handler_id = self.data_box.connect('changed_x', self.__updateEditorX)
         self.__connected_signal_handles_list.append(handler_id)
