From: Alberto Garcia <berto@igalia.com>
Subject: OCRFeeder no longer uses python-gobject
Bug-Debian: https://bugs.debian.org/890160
Index: ocrfeeder/configure.ac
===================================================================
--- ocrfeeder.orig/configure.ac
+++ ocrfeeder/configure.ac
@@ -20,7 +20,6 @@ AX_PYTHON_MODULE([enchant],[needed])
 AX_PYTHON_MODULE([sane],[needed])
 AX_PYTHON_MODULE([PIL],[needed])
 AX_PYTHON_MODULE([reportlab],[needed])
-AX_PYTHON_MODULE([gobject],[needed])
 
 dnl == Introspection Dependencies ==
 
