diff -r 75a90bfc14b1 etc/icons/org.octave.Octave.appdata.xml
--- a/etc/icons/org.octave.Octave.appdata.xml	Fri Jan 31 10:45:13 2020 -0800
+++ b/etc/icons/org.octave.Octave.appdata.xml	Thu Feb 13 13:37:58 2020 +0900
@@ -19,7 +19,7 @@
   <https://www.gnu.org/licenses/>.
 -->
 <component type="desktop-application">
-  <id>org.octave.Octave.desktop</id>
+  <id>org.octave.Octave_BLAS_64.desktop</id>
   <metadata_license>FSFAP</metadata_license>
   <project_license>GPL-3.0+</project_license>
   <name>GNU Octave</name>
@@ -35,6 +35,13 @@
   <summary xml:lang="zh">数值计算交互式编程环境</summary>
   <description>
     <p>
+      This version of GNU Octave uses 64-bit linear algebra for large data.
+      Unless your computer has more than ~32GB of memory and you need to solve
+      linear algebra problems with arrays containing more than ~2 billion
+      elements, this version will offer no advantage over the default GNU
+      Octave version.
+    </p>
+    <p>
       GNU Octave is a high-level interpreted language, primarily intended for
       numerical computations. It provides capabilities for the numerical
       solution of linear and nonlinear problems, and for performing other
@@ -44,28 +51,6 @@
       non-interactive programs. The Octave language is quite similar to
       Matlab so that most programs are easily portable.
     </p>
-    <p xml:lang="de">
-      GNU Octave ist eine höhere, interpretierte Programmiersprache, die
-      hauptsächlich für numerische Berechnungen gedacht ist. Sie bietet
-      Funktionen für die numerische Lösung linearer und nichtlinearer Probleme
-      und für die Durchführung anderer numerischer Experimente. Darüber hinaus
-      bietet es umfangreiche Grafikfunktionen zur Datenvisualisierung und
-      -manipulation. Octave wird normalerweise mittels einer interaktiven
-      Befehlszeile verwendet, kann aber auch zum Schreiben nicht interaktiver
-      Programme verwendet werden. Die Octave-Programmiersprache ist sehr
-      ähnlich zu Matlab, so dass die meisten Programme leicht übertragbar sind.
-    </p>
-    <p xml:lang="fr">
-      GNU Octave est un langage de programmation interprété, destiné
-      principalement au calcul numérique. Il fournit des fonctionnalités
-      pour la résolution de problèmes linéaires et non-linéaires, et pour toutes
-      sortes d'expériences numériques.  Il s'accompagne d'un système complet
-      de visualisation et de manipulation des données. Octave s’utilise
-      généralement via son environnement en lignes de commandes interactif,
-      mais il peut aussi être utilisé pour écrire des programmes non
-      interactifs. Le langage Octave est très similaire à Matlab de telle sorte
-      que les programmes Matlab sont très faciles à porter sous Octave.
-    </p>
     <p>
       Octave has extensive tools for solving common numerical linear algebra
       problems, finding the roots of nonlinear equations, integrating ordinary
@@ -75,25 +60,6 @@
       Octave's own language, or using dynamically loaded modules written in
       C++, C, Fortran, or other languages.
     </p>
-    <p xml:lang="de">
-      Octave verfügt über umfangreiche Werkzeuge zur Lösung gängiger
-      numerischer linearer Probleme der Algebra, zum Auffinden der Nullstellen
-      nichtlinearer Gleichungen, zur Integration gewöhnlicher Funktionen, zur
-      Manipulation von Polynomen und zur Integration gewöhnlicher
-      Differentialgleichungen und differential-algebraischer Gleichungen. Es
-      ist leicht erweiter- und anpassbar mittels benutzerdefinierter
-      Funktionen, die in Octaves eigener Programmiersprache geschrieben wurden,
-      oder über dynamisch geladene, in C++, C, Fortran oder anderen Sprachen
-      geschriebene Module.
-    </p>
-    <p xml:lang="fr">
-      Octave dispose de nombreux outils pour résoudre les problèmes classiques
-      d'algèbre linéaire, trouver les racines d'équations non-linéaires,
-      intégrer les fonctions ordinaires et les équations différentielles.
-      Il peut très facilement être étendu en définissant de nouvelles
-      fonctions écrites en langage Octave ou en utilisant des modules
-      chargés dynamiquement, écrits en C++, C, Fortran ou autres langages.
-    </p>
   </description>

   <screenshots>
diff -r 75a90bfc14b1 etc/icons/org.octave.Octave.desktop.in
--- a/etc/icons/org.octave.Octave.desktop.in	Fri Jan 31 10:45:13 2020 -0800
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,22 +0,0 @@
-[Desktop Entry]
-Name=GNU Octave
-Comment=Interactive programming environment for numerical computations
-Comment[ca]=Entorn de programació interactiva per a càlculs numèrics
-Comment[de]=Interaktive Programmierumgebung für numerische Berechnungen
-Comment[es]=Entorno de programación interactiva para cálculos numéricos
-Comment[fr]=Environnement de programmation interactif pour le calcul numérique
-Comment[it]=Ambiente di programmazione interattivo per il calcolo numerico
-Comment[ja]=数値計算のための対話的なプログラミング環境
-Comment[nl]=Interactieve programmeeromgeving voor numerieke berekeningen
-Comment[pt]=Ambiente de programação interativo para computação numérica
-Comment[zh]=数值计算交互式编程环境
-TryExec=%OCTAVE_PREFIX%/bin/octave
-Exec=%OCTAVE_PREFIX%/bin/octave --gui %f
-Icon=octave
-Terminal=false
-Type=Application
-Categories=Education;Science;Math;
-StartupNotify=false
-StartupWMClass=GNU Octave
-MimeType=text/x-octave;text/x-matlab;
-Keywords=science;math;matrix;numerical computation;plotting;
diff -r 75a90bfc14b1 etc/icons/org.octave.Octave_BLAS_64.desktop.in
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/etc/icons/org.octave.Octave_BLAS_64.desktop.in	Thu Feb 13 13:37:58 2020 +0900
@@ -0,0 +1,22 @@
+[Desktop Entry]
+Name=GNU Octave BLAS 64
+Comment=Interactive programming environment for numerical computations
+Comment[ca]=Entorn de programació interactiva per a càlculs numèrics
+Comment[de]=Interaktive Programmierumgebung für numerische Berechnungen
+Comment[es]=Entorno de programación interactiva para cálculos numéricos
+Comment[fr]=Environnement de programmation interactif pour le calcul numérique
+Comment[it]=Ambiente di programmazione interattivo per il calcolo numerico
+Comment[ja]=数値計算のための対話的なプログラミング環境
+Comment[nl]=Interactieve programmeeromgeving voor numerieke berekeningen
+Comment[pt]=Ambiente de programação interativo para computação numérica
+Comment[zh]=数值计算交互式编程环境
+TryExec=%OCTAVE_PREFIX%/bin/octave
+Exec=%OCTAVE_PREFIX%/bin/octave --gui %f
+Icon=octave
+Terminal=false
+Type=Application
+Categories=Education;Science;Math;
+StartupNotify=false
+StartupWMClass=GNU Octave
+MimeType=text/x-octave;text/x-matlab;
+Keywords=science;math;matrix;numerical computation;plotting;
diff -r 75a90bfc14b1 etc/module.mk
--- a/etc/module.mk	Fri Jan 31 10:45:13 2020 -0800
+++ b/etc/module.mk	Thu Feb 13 13:37:58 2020 +0900
@@ -46,7 +46,7 @@ appdata_DATA = $(APPDATA_XML_FILE)
 desktopdir = $(datadir)/applications

 desktop_DATA = \
-  %reldir%/icons/org.octave.Octave.desktop
+  %reldir%/icons/org.octave.Octave_BLAS_64.desktop

 icon_IMAGE_FILES = \
   %reldir%/icons/octave-logo.svg \
@@ -77,7 +77,7 @@ BUILT_ICONS = \
   $(icon_IMAGE_FILES) \
   %reldir%/icons/octave_branding_samples.svg \
   %reldir%/icons/org.octave.Octave.appdata.xml \
-  %reldir%/icons/org.octave.Octave.desktop.in
+  %reldir%/icons/org.octave.Octave_BLAS_64.desktop.in

 image_DATA += \
   $(icon_IMAGE_FILES) \
@@ -88,9 +88,9 @@ DIRSTAMP_FILES += \

 all-local: all-icons

-all-icons: %reldir%/icons/org.octave.Octave.desktop $(BUILT_ICONS)
+all-icons: %reldir%/icons/org.octave.Octave_BLAS_64.desktop $(BUILT_ICONS)

-%reldir%/icons/org.octave.Octave.desktop: %reldir%/icons/org.octave.Octave.desktop.in | %reldir%/icons/$(octave_dirstamp)
+%reldir%/icons/org.octave.Octave_BLAS_64.desktop: %reldir%/icons/org.octave.Octave_BLAS_64.desktop.in | %reldir%/icons/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(SED) < $< > $@-t \
 	  -e "s|%OCTAVE_PREFIX%|${prefix}|" && \
@@ -147,7 +147,7 @@ uninstall-icons:
 EXTRA_DIST += $(%canon_reldir%_EXTRA_DIST)

 %canon_reldir%_CLEANFILES += \
-  %reldir%/icons/org.octave.Octave.desktop
+  %reldir%/icons/org.octave.Octave_BLAS_64.desktop

 %canon_reldir%_MAINTAINERCLEANFILES += \
   $(BUILT_ICONS)
diff -r 75a90bfc14b1 libgui/src/main-window.cc
--- a/libgui/src/main-window.cc	Fri Jan 31 10:45:13 2020 -0800
+++ b/libgui/src/main-window.cc	Thu Feb 13 13:37:58 2020 +0900
@@ -226,7 +226,7 @@ namespace octave
     m_active_editor = m_editor_window;

 #if defined (HAVE_QGUIAPPLICATION_SETDESKTOPFILENAME)
-    QGuiApplication::setDesktopFileName ("org.octave.Octave.desktop");
+    QGuiApplication::setDesktopFileName ("org.octave.Octave_BLAS_64.desktop");
 #endif

     m_default_style = m_qt_app->style ()->objectName ();
