diff --git a/Source/ScopedNoDenormals.h b/Source/ScopedNoDenormals.h
index 7cf45e6d..be232d04 100644
--- a/Source/ScopedNoDenormals.h
+++ b/Source/ScopedNoDenormals.h
@@ -13,7 +13,13 @@
 ** GNU General Public License for more details.
 */
 
+#if defined(__GNUC__) && (defined(__ARM_NEON__) || defined(__aarch64__))
+/* GCC-compatible compiler, targeting ARM with NEON */
+#include <arm_neon.h>
+#else
+/* This is x86_64, so you might need to fix this */
 #include "xmmintrin.h"
+#endif
 
 namespace denormals {
 class ScopedNoDenormals {
@@ -33,4 +39,4 @@ public:
 
 	int oldMXCSR;
 };
-} // namespace denormals
\ No newline at end of file
+} // namespace denormals
