diff --git a/example/Makefile b/example/Makefile
index 88be11b..a3796df 100644
--- a/example/Makefile
+++ b/example/Makefile
@@ -16,7 +16,7 @@ CXXFLAGS      = -pipe -O2 -Wall -W $(DEFINES)
 INCPATH       = -I. -I../include -I/usr/X11R6/include
 LINK          = g++
 LFLAGS        = -Wl,-O1
-LIBS          = $(SUBLIBS)  -L/usr/X11R6/lib64 -L../lib -lopencsg -lGLEW -lGLU -lglut -lGL 
+LIBS          = $(SUBLIBS) -L/app/lib -L/usr/X11R6/lib64 -L../lib -lopencsg -lGLEW -lGLU -lglut -lGL 
 AR            = ar cqs
 RANLIB        = 
 QMAKE         = /usr/lib/x86_64-linux-gnu/qt4/bin/qmake
@@ -124,13 +124,13 @@ main.o: main.cpp displaylistPrimitive.h
 ####### Install
 
 install_target: first FORCE
-	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/usr/local/bin/ || $(MKDIR) $(INSTALL_ROOT)/usr/local/bin/ 
-	-$(INSTALL_PROGRAM) "$(QMAKE_TARGET)" "$(INSTALL_ROOT)/usr/local/bin/$(QMAKE_TARGET)"
-	-$(STRIP) "$(INSTALL_ROOT)/usr/local/bin/$(QMAKE_TARGET)"
+	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/app/bin/ || $(MKDIR) $(INSTALL_ROOT)/app/bin/ 
+	-$(INSTALL_PROGRAM) "$(QMAKE_TARGET)" "$(INSTALL_ROOT)/app/bin/$(QMAKE_TARGET)"
+	-$(STRIP) "$(INSTALL_ROOT)/app/bin/$(QMAKE_TARGET)"
 
 uninstall_target:  FORCE
-	-$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/bin/$(QMAKE_TARGET)"
-	-$(DEL_DIR) $(INSTALL_ROOT)/usr/local/bin/ 
+	-$(DEL_FILE) "$(INSTALL_ROOT)/app/bin/$(QMAKE_TARGET)"
+	-$(DEL_DIR) $(INSTALL_ROOT)/app/bin/ 
 
 
 install:  install_target  FORCE
diff --git a/src/Makefile b/src/Makefile
index b70edb4..5fb079d 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -16,7 +16,7 @@ CXXFLAGS      = -pipe -O2 -Wall -W -fPIC $(DEFINES)
 INCPATH       = -I. -I../include -I.. -I/usr/X11R6/include
 LINK          = g++
 LFLAGS        = -Wl,-O1 -shared -Wl,-soname,libopencsg.so.1
-LIBS          = $(SUBLIBS)  -L/usr/X11R6/lib64 -lGLEW -lGL 
+LIBS          = $(SUBLIBS) -L/app/lib -L/usr/X11R6/lib64 -lGLEW -lGL 
 AR            = ar cqs
 RANLIB        = 
 QMAKE         = /usr/lib/x86_64-linux-gnu/qt4/bin/qmake
@@ -285,29 +285,29 @@ RenderTexture.o: ../RenderTexture/RenderTexture.cpp ../RenderTexture/RenderTextu
 ####### Install
 
 install_target: first FORCE
-	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/usr/local/lib/ || $(MKDIR) $(INSTALL_ROOT)/usr/local/lib/ 
-	-$(INSTALL_PROGRAM) "../lib/$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET)"
-	-$(STRIP) --strip-unneeded "$(INSTALL_ROOT)/usr/local/lib/$(TARGET)"
-	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET0)"
-	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET1)"
-	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET2)"
+	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/app/lib/ || $(MKDIR) $(INSTALL_ROOT)/app/lib/ 
+	-$(INSTALL_PROGRAM) "../lib/$(TARGET)" "$(INSTALL_ROOT)/app/lib/$(TARGET)"
+	-$(STRIP) --strip-unneeded "$(INSTALL_ROOT)/app/lib/$(TARGET)"
+	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/app/lib/$(TARGET0)"
+	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/app/lib/$(TARGET1)"
+	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/app/lib/$(TARGET2)"
 
 uninstall_target:  FORCE
-	-$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET)" 
-	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET0)" 
-	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET1)" 
-	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET2)"
-	-$(DEL_DIR) $(INSTALL_ROOT)/usr/local/lib/ 
+	-$(DEL_FILE) "$(INSTALL_ROOT)/app/lib/$(TARGET)" 
+	 -$(DEL_FILE) "$(INSTALL_ROOT)/app/lib/$(TARGET0)" 
+	 -$(DEL_FILE) "$(INSTALL_ROOT)/app/lib/$(TARGET1)" 
+	 -$(DEL_FILE) "$(INSTALL_ROOT)/app/lib/$(TARGET2)"
+	-$(DEL_DIR) $(INSTALL_ROOT)/app/lib/ 
 
 
 install_headers: first FORCE
-	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/usr/local/include/ || $(MKDIR) $(INSTALL_ROOT)/usr/local/include/ 
-	-$(INSTALL_FILE) ../include/opencsg.h $(INSTALL_ROOT)/usr/local/include/
+	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/app/include/ || $(MKDIR) $(INSTALL_ROOT)/app/include/ 
+	-$(INSTALL_FILE) ../include/opencsg.h $(INSTALL_ROOT)/app/include/
 
 
 uninstall_headers:  FORCE
-	-$(DEL_FILE) -r $(INSTALL_ROOT)/usr/local/include/opencsg.h
-	-$(DEL_DIR) $(INSTALL_ROOT)/usr/local/include/ 
+	-$(DEL_FILE) -r $(INSTALL_ROOT)/app/include/opencsg.h
+	-$(DEL_DIR) $(INSTALL_ROOT)/app/include/ 
 
 
 install:  install_target install_headers  FORCE
