--- libs/openFrameworks/sound/ofOpenALSoundPlayer.bck	2020-01-01 13:36:59.501973818 +0100
+++ libs/openFrameworks/sound/ofOpenALSoundPlayer.h	2020-01-01 13:37:52.486103430 +0100
@@ -10,9 +10,9 @@
 
 typedef unsigned int ALuint;
 /** Opaque device handle */
-typedef struct ALCdevice_struct ALCdevice;
+struct ALCdevice;
 /** Opaque context handle */
-typedef struct ALCcontext_struct ALCcontext;
+struct ALCcontext;
 
 
 
