diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5956a67..3f0fda1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -528,3 +528,11 @@ set_target_properties(OCOVR PROPERTIES
 get_target_property(output_dir OCOVR LIBRARY_OUTPUT_DIRECTORY)
 add_custom_command(TARGET OCOVR
 	PRE_LINK COMMAND ${CMAKE_COMMAND} -E make_directory ${output_dir})
+
+# Installation
+include(GNUInstallDirs)
+install(TARGETS OCOVR
+	LIBRARY
+	DESTINATION "${PROJECT_NAME}/${VRCLIENT_DIR}"
+	COMPONENT Runtime
+)
-- 
2.45.2

