diff --git a/api/gui/DesktopServices.cpp b/api/gui/DesktopServices.cpp
index 5368ddc8..c5f904c7 100644
--- a/api/gui/DesktopServices.cpp
+++ b/api/gui/DesktopServices.cpp
@@ -83,11 +83,7 @@ bool openDirectory(const QString &path, bool ensureExists)
     {
         return QDesktopServices::openUrl(QUrl::fromLocalFile(dir.absolutePath()));
     };
-#if defined(Q_OS_LINUX)
-    return IndirectOpen(f);
-#else
     return f();
-#endif
 }
 
 bool openFile(const QString &path)
@@ -97,11 +93,7 @@ bool openFile(const QString &path)
     {
         return QDesktopServices::openUrl(QUrl::fromLocalFile(path));
     };
-#if defined(Q_OS_LINUX)
-    return IndirectOpen(f);
-#else
     return f();
-#endif
 }
 
 bool openFile(const QString &application, const QString &path, const QString &workingDirectory, qint64 *pid)
@@ -139,11 +131,7 @@ bool openUrl(const QUrl &url)
     {
         return QDesktopServices::openUrl(url);
     };
-#if defined(Q_OS_LINUX)
-    return IndirectOpen(f);
-#else
     return f();
-#endif
 }
 
 }
