diff --git a/kwave/org.kde.kwave.desktop.in b/kwave/org.kde.kwave.desktop.in
index fc81ecc4..de94b961 100644
--- a/kwave/org.kde.kwave.desktop.in
+++ b/kwave/org.kde.kwave.desktop.in
@@ -98,7 +98,7 @@ Comment[zh_CN]=KDE 推出的声音编辑器
 Comment[zh_TW]=來自 KDE 的音效編輯器
 TryExec=kwave
 Exec=kwave %F
-Icon=kwave
+Icon=org.kde.kwave
 MimeType=${KWAVE_DESKTOP_MIME_TYPES}
 Terminal=false
 Categories=Qt;KDE;AudioVideo;Audio;Player;Recorder;AudioVideoEditing;

