diff --git a/org.garudalinux.firedragon-catppuccin.desktop b/org.garudalinux.firedragon-catppuccin.desktop
--- a/org.garudalinux.firedragon-catppuccin.desktop
+++ b/org.garudalinux.firedragon-catppuccin.desktop
@@ -113,7 +113,7 @@
 Keywords[vi]=Internet;WWW;Browser;Web;Explorer;Trình duyệt;Trang web
 Keywords[zh_CN]=Internet;WWW;Browser;Web;Explorer;网页;浏览;上网;火狐;FireDragon;ff;互联网;网站;
 Keywords[zh_TW]=Internet;WWW;Browser;Web;Explorer;網際網路;網路;瀏覽器;上網;網頁;火狐
-Exec=/usr/lib/firedragon/firedragon %u
+Exec=/app/lib/firedragon/firedragon %u
 Icon=firedragon
 Terminal=false
 X-MultipleArgs=false
@@ -230,7 +230,7 @@
 Name[xh]=Ifestile entsha
 Name[zh_CN]=新建窗口
 Name[zh_TW]=開新視窗
-Exec=/usr/lib/firedragon/firedragon --new-window %u
+Exec=/app/lib/firedragon/firedragon --new-window %u
 
 [Desktop Action new-private-window]
 Name=New Private Window
@@ -338,4 +338,4 @@
 Name[xh]=Ifestile yangasese entsha
 Name[zh_CN]=新建隐私浏览窗口
 Name[zh_TW]=新增隱私視窗
-Exec=/usr/lib/firedragon/firedragon --private-window %u
+Exec=/app/lib/firedragon/firedragon --private-window %u
