diff --git a/externals/oscx/Makefile b/externals/oscx/Makefile
index fc30328ba..40ca62407 100644
--- a/externals/oscx/Makefile
+++ b/externals/oscx/Makefile
@@ -169,10 +169,10 @@ ifeq ($(UNAME),Linux)
   OS = linux
   PD_PATH = /usr
   OPT_CFLAGS = -O6 -funroll-loops -fomit-frame-pointer
-  ALL_CFLAGS += -fPIC
-  ALL_LDFLAGS += -rdynamic -shared -fPIC -Wl,-rpath,"\$$ORIGIN",--enable-new-dtags
-  SHARED_LDFLAGS += -Wl,-soname,$(SHARED_LIB) -shared
-  ALL_LIBS += -lc $(LIBS_linux)
+  ALL_CFLAGS += -fPIC -I/app/include/tirpc
+  ALL_LDFLAGS += -rdynamic -shared -fPIC -Wl,-rpath,"\$$ORIGIN",--enable-new-dtags -L/app/lib
+  SHARED_LDFLAGS += -Wl,-soname,$(SHARED_LIB) -shared -L/app/lib
+  ALL_LIBS += -lc $(LIBS_linux) -ltirpc
   STRIP = strip --strip-unneeded -R .note -R .comment
   DISTBINDIR=$(DISTDIR)-$(OS)-$(shell uname -m)
 endif
