diff --git a/Makefile b/Makefile
index 8e3d328..acd05b7 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 DESTDIR?=/
-datadir?=$(DESTDIR)/usr/share
+datadir?=/app/share
 INSTALL=install
 
 SOURCES=$(wildcard *.desktop.in)
@@ -25,7 +25,7 @@ pos:
 
 install: all
 	rm othman-data/quran-kareem.png || :
-	python setup.py install -O2 --root $(DESTDIR)
+	python setup.py install --prefix=/app
 	$(INSTALL) -d $(datadir)/applications/
 	$(INSTALL) -m 0644 Othman.desktop $(datadir)/applications/
 	for i in 96 72 64 48 36 32 24 22 16; do \
