diff --git a/activity/activity.info b/activity/activity.info
index 6933331..745a3ab 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,12 +1,19 @@
 [Activity]
 name = Paint
 activity_version = 70
-bundle_id = org.laptop.Oficina
+release_date = 2020-02-21
+bundle_id = org.sugarlabs.Paint
 exec = sugar-activity3 OficinaActivity.OficinaActivity
 icon = activity-paint
 mime_types = image/svg+xml; image/svg; image/svg-xml; image/vnd.adobe.svg+xml; text/xml-svg; image/png; image/jpeg; image/gif; image/x-icon; image/x-ico; application/x-navi-animation; image/x-cmu-raster; image/x-sun-raster; image/x-xpixmap; image/tiff; image/x-portable-anymap; image/x-portable-bitmap; image/x-portable-graymap; image/x-portable-pixmap; image/bmp; image/x-bmp; image/x-MS-bmp; image/vnd.wap.wbmp; image/x-xbitmap; image/x-tga; image/x-pcx; image/x-wmf
-license = GPLv2;GPLv2+
-summary = Picasso? Van Gogh? You can be any of them! Just let the artist inside you free. Transform images in your mind to beautiful paintings.
+license = GPLv2+
+metadata_license = CC0-1.0
+summary = A sweet painting tool
+description = Picasso? Van Gogh? You can be any of them! Just let the artist inside you free. Transform images in your mind to beautiful paintings.
 repository = https://github.com/sugarlabs/paint-activity
 max_participants = 1
-tags = Documents;Media;Art
+tags = Education
+url = https://help.sugarlabs.org/en/paint.html
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/paint-activity/master/screenshots/en/1.png
