diff --git a/src/grapejuice_packaging/packaging_resources/bin/grapejuice b/src/grapejuice_packaging/packaging_resources/bin/grapejuice
index bcf21c1..f6785c9 100644
--- a/src/grapejuice_packaging/packaging_resources/bin/grapejuice
+++ b/src/grapejuice_packaging/packaging_resources/bin/grapejuice
@@ -1,3 +1,4 @@
 #!/usr/bin/env bash
-
-exec /usr/bin/env python3 -m grapejuice "$@"
+mkdir -p /var/config/brinkervii/grapejuice/ 
+cp -n /app/user_settings.json /var/config/brinkervii/grapejuice/     
+exec /usr/bin/env PYTHONPATH="${PYTHONPATH}:/app/lib/python3.9/pip_deps" GRAPEJUICE_LOCALE_DIRECTORY=/app/share/locale python3 -m grapejuice "$@"
diff --git a/src/grapejuice_packaging/packaging_resources/bin/grapejuice-gui b/src/grapejuice_packaging/packaging_resources/bin/grapejuice-gui
index 9a6ad30..2e19c4e 100644
--- a/src/grapejuice_packaging/packaging_resources/bin/grapejuice-gui
+++ b/src/grapejuice_packaging/packaging_resources/bin/grapejuice-gui
@@ -1,3 +1,4 @@
 #!/usr/bin/env bash
-
-exec /usr/bin/env python3 -m grapejuice gui "$@"
+mkdir -p /var/config/brinkervii/grapejuice/ 
+cp -n /app/user_settings.json /var/config/brinkervii/grapejuice/     
+exec /usr/bin/env PYTHONPATH="${PYTHONPATH}:/app/lib/python3.9/pip_deps" GRAPEJUICE_LOCALE_DIRECTORY=/app/share/locale python3 -m grapejuice gui "$@"
diff --git a/src/grapejuice_packaging/packaging_resources/bin/grapejuiced b/src/grapejuice_packaging/packaging_resources/bin/grapejuiced
index 86e58d3..c282ec9 100644
--- a/src/grapejuice_packaging/packaging_resources/bin/grapejuiced
+++ b/src/grapejuice_packaging/packaging_resources/bin/grapejuiced
@@ -1,3 +1,4 @@
 #!/usr/bin/env bash
-
-exec /usr/bin/env python3 -m grapejuiced "$@"
+mkdir -p /var/config/brinkervii/grapejuice/ 
+cp -n /app/user_settings.json /var/config/brinkervii/grapejuice/     
+exec /usr/bin/env PYTHONPATH="${PYTHONPATH}:/app/lib/python3.9/pip_deps" GRAPEJUICE_LOCALE_DIRECTORY=/app/share/locale python3 -m grapejuiced "$@"
