diff --git a/gtk/build-intel-lib.sh b/gtk/build-intel-lib.sh
--- a/gtk/build-intel-lib.sh
+++ b/gtk/build-intel-lib.sh
@@ -44,6 +44,7 @@
 tar xvfz ../inteldecimal/IntelRDFPMathLib20U1.tar.gz
 cd IntelRDFPMathLib20U1
 patch -p0 <../intel-lib-linux.patch
+patch -p0 <../patch-aarch64-include-fenv.patch
 
 # When building for architectures other than x86 or x86_64, I remove the
 # section titled "Determine host architecture" in
