--- configure2/MakefileOLD	2020-11-07 01:06:41.721888131 -0400
+++ configure2/Makefile	2020-11-07 01:09:25.127019480 -0400
@@ -7,7 +7,7 @@
 #############################################################################
 
 ####### Compiler, tools and options
-
+QTDIR=/app
 CC       = gcc
 CXX      = g++
 LEX      = flex
@@ -91,7 +91,7 @@
 uicables: $(UICDECLS) $(UICIMPLS)
 
 $(MOC): 
-	( cd $(QTDIR)/src/moc && $(MAKE) )
+	( echo "cd $(QTDIR)/src/moc && $(MAKE)" )
 
 Makefile: configure2.pro  /app/mkspecs/default/qmake.conf /app/lib/libqt.prl
 	$(QMAKE) -o Makefile configure2.pro
