diff --git a/electrum.desktop b/electrum.desktop
index e40106716..15d25a704 100644
--- a/electrum.desktop
+++ b/electrum.desktop
@@ -1,12 +1,9 @@
-# If you want Electrum to appear in a Linux app launcher ("start menu"), install this by doing:
-# sudo desktop-file-install electrum.desktop
-
 [Desktop Entry]
 Comment=Lightweight Bitcoin Client
-Exec=sh -c "PATH=\"\\$HOME/.local/bin:\\$PATH\"; electrum %u"
+Exec=/app/bin/electrum %u
 GenericName[en_US]=Bitcoin Wallet
 GenericName=Bitcoin Wallet
-Icon=electrum
+Icon=org.electrum.electrum
 Name[en_US]=Electrum Bitcoin Wallet
 Name=Electrum Bitcoin Wallet
 Categories=Finance;Network;
@@ -18,5 +15,5 @@ MimeType=x-scheme-handler/bitcoin;
 Actions=Testnet;
 
 [Desktop Action Testnet]
-Exec=sh -c "PATH=\"\\$HOME/.local/bin:\\$PATH\"; electrum --testnet %u"
+Exec=/app/bin/electrum --testnet %u
 Name=Testnet mode
diff --git a/setup.py b/setup.py
index 409664742..835baa634 100755
--- a/setup.py
+++ b/setup.py
@@ -37,14 +37,8 @@ if platform.system() in ['Linux', 'FreeBSD', 'DragonFly']:
     parser.add_argument('--root=', dest='root_path', metavar='dir', default='/')
     opts, _ = parser.parse_known_args(sys.argv[1:])
     usr_share = os.path.join(sys.prefix, "share")
-    icons_dirname = 'pixmaps'
-    if not os.access(opts.root_path + usr_share, os.W_OK) and \
-       not os.access(opts.root_path, os.W_OK):
-        icons_dirname = 'icons'
-        if 'XDG_DATA_HOME' in os.environ.keys():
-            usr_share = os.environ['XDG_DATA_HOME']
-        else:
-            usr_share = os.path.expanduser('~/.local/share')
+    icons_dirname = 'icons/hicolor/128x128/apps'
+    usr_share = "/app/share"
     data_files += [
         (os.path.join(usr_share, 'applications/'), ['electrum.desktop']),
         (os.path.join(usr_share, icons_dirname), ['electrum/gui/icons/electrum.png']),
