diff --git a/src/Launch.lua b/src/Launch.lua
index 0673314..39800ae 100644
--- a/src/Launch.lua
+++ b/src/Launch.lua
@@ -51,7 +51,7 @@ function launch:OnInit()
 	if localManXML and not self.versionBranch and not self.versionPlatform then
 		-- Looks like a remote manifest, so we're probably running from a repository
 		-- Enable dev mode to disable updates and set user path to be the script path
-		self.devMode = true
+		self.devMode = false
 	end
 	RenderInit()
 	ConPrintf("Loading main script...")
@@ -286,6 +286,8 @@ function launch:ApplyUpdate(mode)
 end
 
 function launch:CheckForUpdate(inBackground)
+	return
+	--[[
 	if self.updateCheckRunning then
 		return
 	end
@@ -302,6 +304,7 @@ function launch:CheckForUpdate(inBackground)
 		self.updateCheckRunning = true
 	end
 	update:close()
+	--]]
 end
 
 function launch:ShowPrompt(r, g, b, str, func)
diff --git a/src/Modules/Main.lua b/src/Modules/Main.lua
index 1af88e0..683fc33 100644
--- a/src/Modules/Main.lua
+++ b/src/Modules/Main.lua
@@ -42,7 +42,7 @@ function main:Init()
 	self.modes["LIST"] = LoadModule("Modules/BuildList")
 	self.modes["BUILD"] = LoadModule("Modules/Build")
 
-	if launch.devMode or (GetScriptPath() == GetRuntimePath() and not launch.installedMode) then
+	if launch.devMode and (GetScriptPath() == GetRuntimePath() and not launch.installedMode) then
 		-- If running in dev mode or standalone mode, put user data in the script path
 		self.userPath = GetScriptPath().."/"
 	else
