diff --git a/src/io/filesystem.cpp b/src/io/filesystem.cpp
index 237abc3..bd56a1f 100644
--- a/src/io/filesystem.cpp
+++ b/src/io/filesystem.cpp
@@ -27,6 +27,7 @@
 #include <QFileInfo>
 #include <QLibraryInfo>
 #include <QRegExp>
+#include <QStandardPaths>
 #include <QTemporaryFile>
 #include <QTextStream>
 
@@ -259,15 +260,7 @@ QString confDirPath(const QString &confSubdir)
 	return dirPath;
 
 #else /* !PORTABLE_APPLICATION */
-	/* StandardLocation::writableLocation(QStandardPaths::HomeLocation) ? */
-	QDir homeDir(QDir::homePath());
-
-	if (homeDir.exists(WIN_PREFIX) && !homeDir.exists(confSubdir)) {
-		/* Set windows directory. */
-		homeDir.cd(WIN_PREFIX);
-	}
-
-	return homeDir.path() + QDir::separator() + confSubdir;
+	return QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + QDir::separator();
 #endif /* PORTABLE_APPLICATION */
 
 #undef WIN_PREFIX
