diff --git a/QtScrcpy/main.cpp b/QtScrcpy/main.cpp
index 629e08bd..91760ce4 100644
--- a/QtScrcpy/main.cpp
+++ b/QtScrcpy/main.cpp
@@ -36,10 +36,10 @@ int main(int argc, char *argv[])
 #endif
 
 #ifdef Q_OS_LINUX
-    qputenv("QTSCRCPY_ADB_PATH", "../../../QtScrcpy/QtScrcpyCore/src/third_party/adb/linux/adb");
-    qputenv("QTSCRCPY_SERVER_PATH", "../../../QtScrcpy/QtScrcpyCore/src/third_party/scrcpy-server");
-    qputenv("QTSCRCPY_KEYMAP_PATH", "../../../keymap");
-    qputenv("QTSCRCPY_CONFIG_PATH", "../../../config");
+    qputenv("QTSCRCPY_ADB_PATH", "/app/bin/adb");
+    qputenv("QTSCRCPY_SERVER_PATH", "/app/qtscrcpy/scrcpy-server");
+    qputenv("QTSCRCPY_KEYMAP_PATH", "/app/qtscrcpy/keymap");
+    qputenv("QTSCRCPY_CONFIG_PATH", "/app/etc/qtscrcpy");
 #endif
 
     g_msgType = covertLogLevel(Config::getInstance().getLogLevel());