diff --git a/cbindings/Qt6Pas.pro b/cbindings/Qt6Pas.pro
index 50b69f2..890031b 100644
--- a/cbindings/Qt6Pas.pro
+++ b/cbindings/Qt6Pas.pro
@@ -49,10 +49,10 @@ haiku:isEmpty(is64) {
 
 # Available Qt6 Pascal Binding Platforms: see http://wiki.freepascal.org/Qt6_binding#Supported_Platforms  
 
-target.path = $$[QT_INSTALL_LIBS]
+target.path = '/app/lib'
 win32 : {
   PLATFORM = MSWINDOWS
-  target.path = $$[QT_INSTALL_BINS]
+  target.path = '/app/lib'
   }
                            
 embedded:PLATFORM = QTOPIA  