From 93b9cad5a20ca41303e4923ed3c9a0c674f0a98a Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Mon, 2 Dec 2019 16:57:47 -0300
Subject: [PATCH 6/6] info

---
 activity/activity.info | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index 8faedfa..bfdc6a8 100755
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,13 +1,21 @@
 [Activity]
 name = Physics
-summary = Prove Sir Isaac Newton correct. Create real-life simulations using different shapes and see them come to life with forces.
-bundle_id = org.laptop.physics
+summary = Prove Sir Isaac Newton correct
+description = Prove Sir Isaac Newton correct. Create real-life simulations using different shapes and see them come to life with forces.
+bundle_id = org.sugarlabs.Physics
 exec = sugar-activity activity.PhysicsActivity
 icon = activity-physics
-activity_version = 34~dfsg
+activity_version = 34
+release_date = 2018-09-05
 max_participants = 4
 show_launcher = yes
 mime_types = application/x-physics-activity
 category = science game art
+url = https://help.sugarlabs.org/en/physics.html
 repository = https://github.com/sugarlabs/physics
 license = GPLv3+
+metadata_license = CC0-1.0
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://help.sugarlabs.org/en/_images/Physics-activity.png
+tags = Education
-- 
2.19.1

