From 910da065a92b5deebde9d3121b1e8fdcb3f4914b Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Mon, 2 Dec 2019 17:11:32 -0300
Subject: [PATCH] fix-cursor

---
 physics.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/physics.py b/physics.py
index fd221af..ecba481 100644
--- a/physics.py
+++ b/physics.py
@@ -95,7 +95,9 @@ class PhysicsGame:
             self.show_fake_cursor = True
             pygame.mouse.set_cursor((8, 8), (0, 0), (0, 0, 0, 0, 0, 0, 0, 0),
                                     (0, 0, 0, 0, 0, 0, 0, 0))
-            self.cursor_picture = pygame.image.load('standardcursor.png')
+            file_path = os.path.dirname(os.path.abspath(__file__))
+            cursor_path = os.path.join(file_path, 'standardcursor.png')
+            self.cursor_picture = pygame.image.load(cursor_path)
             self.cursor_picture.convert_alpha()
             self.canvas.connect('enter_notify_event',
                                 self.switch_on_fake_pygame_cursor_cb)
-- 
2.19.1

