From cf375dd8c1e2b92812b741cc265cd0d1080fef7e Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Mon, 2 Dec 2019 16:50:26 -0300
Subject: [PATCH 2/6] fix-saving-world

---
 myelements/elements.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/myelements/elements.py b/myelements/elements.py
index 19b2a3e..1fcef09 100644
--- a/myelements/elements.py
+++ b/myelements/elements.py
@@ -495,6 +495,7 @@ class Elements:
         bodylist = []
         for body in self.world.bodies:
             if body == self.world.groundBody:
+                body.userData = {"saveid": 0}
                 continue
             body.userData["saveid"] = save_id_index  # set temporary data
             save_id_index += 1
-- 
2.19.1

