From 5d4589f9772b7cce1066b77df412c8d6421d2fb9 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Mon, 2 Dec 2019 16:50:43 -0300
Subject: [PATCH 3/6] fix-loading-world

---
 physics.py | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/physics.py b/physics.py
index 546695c..fd221af 100644
--- a/physics.py
+++ b/physics.py
@@ -109,27 +109,27 @@ class PhysicsGame:
         self.world.renderer.set_surface(self.screen)
         self.world.add.ground()
 
-        if self.opening_queue:
-            path = self.opening_queue.encode('ascii', 'convert')
-            if os.path.exists(path):
-                self.world.json_load(path, serialized=True)
-                if 'full_pos_list' in self.world.additional_vars:
-                    self.full_pos_list = \
-                        self.world.additional_vars['full_pos_list']
-                if 'trackinfo' in self.world.additional_vars:
-                    self.trackinfo = self.world.additional_vars['trackinfo']
-                if 'tracked_bodies' in self.world.additional_vars:
-                    self.tracked_bodies = \
-                        self.world.additional_vars['tracked_bodies']
-
         while self.running:
-
             # Pump GTK messages.
             while Gtk.events_pending():
                 Gtk.main_iteration()
             if not self.running:
                 break
 
+            if self.opening_queue:
+                path = self.opening_queue.encode('ascii', 'convert')
+                if os.path.exists(path):
+                    self.world.json_load(path, serialized=True)
+                    if 'full_pos_list' in self.world.additional_vars:
+                        self.full_pos_list = \
+                            self.world.additional_vars['full_pos_list']
+                    if 'trackinfo' in self.world.additional_vars:
+                        self.trackinfo = self.world.additional_vars['trackinfo']
+                    if 'tracked_bodies' in self.world.additional_vars:
+                        self.tracked_bodies = \
+                            self.world.additional_vars['tracked_bodies']
+                self.opening_queue = None
+
             # Pump PyGame messages.
             for event in pygame.event.get():
                 if event.type == pygame.QUIT:
-- 
2.19.1

