From e1a8b4911145537a59bc52bd2082a3d58d7201f5 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Mon, 2 Dec 2019 16:51:31 -0300
Subject: [PATCH 5/6] monitor

---
 activity.py | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/activity.py b/activity.py
index 0a78418..52b4bd6 100644
--- a/activity.py
+++ b/activity.py
@@ -57,6 +57,7 @@ from collabwrapper import CollabWrapper
 import tools
 from physics import PhysicsGame
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import ExtendedActivityToolbarButton
 from sugarapp.widgets import SugarCompatibleActivity
 
@@ -88,8 +89,8 @@ class PhysicsActivity(SugarCompatibleActivity):
         self._notebook = Gtk.Notebook(show_tabs=False)
         self._notebook.add(self.game.canvas)
 
-        w = Gdk.Screen.width()
-        h = Gdk.Screen.height() - 2 * GRID_CELL_SIZE
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height() - 2 * GRID_CELL_SIZE
 
         self.game.canvas.set_size_request(w, h)
 
@@ -115,8 +116,8 @@ class PhysicsActivity(SugarCompatibleActivity):
         ''' Screen size has changed '''
         self.write_file(os.path.join(
                         activity.get_activity_root(), 'data', 'data'))
-        w = Gdk.Screen.width()
-        h = Gdk.Screen.height() - 2 * GRID_CELL_SIZE
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height() - 2 * GRID_CELL_SIZE
         pygame.display.set_mode((w, h),
                                 pygame.RESIZABLE)
         self.read_file(os.path.join(
@@ -644,7 +645,7 @@ class PhysicsActivity(SugarCompatibleActivity):
             self._sample_window.set_border_width(4)
 
             self._sample_window.set_size_request(
-                Gdk.Screen.width() / 2, Gdk.Screen.height() / 2)
+                PrimaryMonitor.width() / 2, PrimaryMonitor.height() / 2)
             self._sample_window.show()
 
             store = Gtk.ListStore(GdkPixbuf.Pixbuf, str)
-- 
2.19.1

