From 954b0388a1fa7a56c520c8241ec95703b21ed014 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Mon, 2 Dec 2019 17:50:50 -0300
Subject: [PATCH] fix-remove-joint

---
 tools.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools.py b/tools.py
index 8ae1567..a66ff0e 100644
--- a/tools.py
+++ b/tools.py
@@ -615,7 +615,7 @@ class GrabTool(Tool):
                     if self.pm_mode_active:
                         pos = event.pos
                         body = find_body(self.game.world, pos)
-                        if body is not None:
+                        if hasattr(self._moving_pm, 'enableMotor') and body is not None:
                             if self._moving_pm.enableMotor:
                                 self.game.world.add.motor(
                                     body, pos,
-- 
2.19.1

