From c9f74ace73a78bd9abc839eb0ae4793a9075df65 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Mon, 2 Dec 2019 16:50:57 -0300
Subject: [PATCH 4/6] fix-no-badges

---
 tools.py | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/tools.py b/tools.py
index a051eae..8ae1567 100644
--- a/tools.py
+++ b/tools.py
@@ -145,23 +145,8 @@ class Tool(object):
         pass
 
     def add_badge(self, message, icon='trophy-icon-physics', name='Physics'):
-        badge = {
-            'icon': icon,
-            'from': name,
-            'message': message
-        }
-        icon_path = os.path.join(activity.get_bundle_path(),
-                                 'icons',
-                                 (icon + '.svg'))
-        sugar_icons = os.path.join(os.path.expanduser('~'), '.icons')
-        copy(icon_path, sugar_icons)
-
-        if 'comments' in self.game.activity.metadata:
-            comments = json.loads(self.game.activity.metadata['comments'])
-            comments.append(badge)
-            self.game.activity.metadata['comments'] = json.dumps(comments)
-        else:
-            self.game.activity.metadata['comments'] = json.dumps([badge])
+        # no badge system in sugarapp
+        pass
 
 
 # The circle creation tool
-- 
2.19.1

