--- Pillow-5.0.0/setup_old.py	2018-01-01 22:04:32.000000000 +0100
+++ Pillow-5.0.0/setup.py	2018-01-18 22:32:00.102552029 +0100
@@ -368,7 +368,7 @@
 
                 for platform_ in arch_tp:
                     dirs = libdirs.get(platform_, None)
-                    if not platform_:
+                    if not dirs:
                         continue
                     for path in dirs:
                         _add_directory(library_dirs, path)
