From dfa462406547865e53a6eb790c8cc273181e35cf Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Wed, 15 Jan 2020 18:01:37 -0300
Subject: [PATCH] info

---
 activity/activity.info | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index e79f720..ee79c93 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,14 +1,20 @@
 [Activity]
 name = Pippy
-bundle_id = org.laptop.Pippy
+bundle_id = org.sugarlabs.Pippy
 exec = sugar-activity3 pippy_app.PippyActivity
 icon = activity-icon
-activity_version = 74
+activity_version = 74.1
+release_date = 2019-12-16
 max_participants = 4
 mime_types = text/x-python
 show_launcher = yes
-license = GPLv2+;GPLv3+;LGPLv2
-summary = Searching for programmers! Here you can use examples to begin learning the language your computer understands called "Python".
-tags = Programming
+license = GPLv3+
+metadata_license = CC0-1.0
+summary = Talk to your computer in Python
+description = Pippy allows the student to examine, execute, and modify simple Python programs. It's possible to write Python statements to play sounds, calculate expressions, or make simple text based interactive games. It also includes PyGame examples and a tutorial for GTK basics.
+tags = Development
 repository = https://github.com/sugarlabs/Pippy
-url = https://help.sugarlabs.org/pippy.html
+url = https://help.sugarlabs.org/en/pippy.html
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://i.imgur.com/VeZWL2q.png https://i.imgur.com/7QOhdmS.png https://i.imgur.com/fNlsMcf.png
-- 
2.23.0

