From 71cf453cfce4dc4265b06b6ebcbd1a9919b8bd2c Mon Sep 17 00:00:00 2001
Message-Id: <71cf453cfce4dc4265b06b6ebcbd1a9919b8bd2c.1671455296.git.matthias@mailaender.name>
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Mon, 19 Dec 2022 14:08:12 +0100
Subject: [PATCH] Configurable prefix

---
 src/src.pro | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/src.pro b/src/src.pro
index cd201ac..fa0e24e 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -29,12 +29,12 @@ RESOURCES += \
 
 unix {
 desktop.files += ../cutepeaks.desktop
-desktop.path = /usr/share/applications/
+desktop.path = $$PREFIX/share/applications/
 
 icons.files += ../cutepeaks.png
-icons.path = /usr/share/icons/hicolor/64x64/apps
+icons.path = $$PREFIX/share/icons/hicolor/64x64/apps
 
-target.path = /usr/bin/
+target.path = $$PREFIX/bin/
 
 INSTALLS += desktop icons target
 
-- 
2.39.0

