diff --git a/externals/bsaylor/Makefile b/externals/bsaylor/Makefile
index f6cf5c731..60b76bb7e 100644
--- a/externals/bsaylor/Makefile
+++ b/externals/bsaylor/Makefile
@@ -155,7 +155,7 @@ ifeq ($(UNAME),Linux)
   PD_PATH = /usr
   OPT_CFLAGS = -O6 -funroll-loops -fomit-frame-pointer
   ALL_CFLAGS += -fPIC
-  ALL_LDFLAGS += -rdynamic -shared -fPIC
+  ALL_LDFLAGS += -rdynamic -shared -fPIC -L/app/lib
   SHARED_LDFLAGS += -Wl,-soname,$(SHARED_LIB) -shared
   ALL_LIBS += -lc $(LIBS_linux)
   STRIP = strip --strip-unneeded -R .note -R .comment
diff --git a/externals/loaders/pdlua/src/Makefile b/externals/loaders/pdlua/src/Makefile
index ed7ae23b0..a88955bb7 100644
--- a/externals/loaders/pdlua/src/Makefile
+++ b/externals/loaders/pdlua/src/Makefile
@@ -150,9 +150,9 @@ ifeq ($(UNAME),Linux)
   PD_PATH = /usr
   OPT_CFLAGS = -O6 -funroll-loops -fomit-frame-pointer
   CFLAGS += -fPIC -threepic
-  LDFLAGS += -Wl,--export-dynamic  -shared -fPIC
+  LDFLAGS += -Wl,--export-dynamic  -shared -fPIC -L/app/lib
   LUACFLAGS += -I/usr/include/lua5.1 # lua is named differently on every platform, check this and change it to fit
-  LIBS += -llua5.1  # lua is named differently on every platform, check this and change it to fit
+  LIBS += -llua  # lua is named differently on every platform, check this and change it to fit
   LIBS += -lc
   STRIP = strip --strip-unneeded -R .note -R .comment
   DISTBINDIR=$(DISTDIR)-$(OS)-$(shell uname -m)
diff --git a/externals/loaders/tclpd/Makefile b/externals/loaders/tclpd/Makefile
index a23730dd8..54bec0bc2 100644
--- a/externals/loaders/tclpd/Makefile
+++ b/externals/loaders/tclpd/Makefile
@@ -168,7 +168,7 @@ ifeq ($(UNAME),Linux)
   PD_PATH = /usr
   OPT_CFLAGS = -O6 -funroll-loops -fomit-frame-pointer
   ALL_CFLAGS += -fPIC
-  ALL_LDFLAGS += -rdynamic -shared -fPIC -Wl,-rpath,"\$$ORIGIN",--enable-new-dtags
+  ALL_LDFLAGS += -rdynamic -shared -fPIC -Wl,-rpath,"\$$ORIGIN",--enable-new-dtags -L/app/lib
   SHARED_LDFLAGS += -Wl,-soname,$(SHARED_LIB) -shared
   ALL_LIBS += -lc $(LIBS_linux)
   STRIP = strip --strip-unneeded -R .note -R .comment
diff --git a/externals/unauthorized/Makefile b/externals/unauthorized/Makefile
index 5b2c9734d..3cba50a0a 100644
--- a/externals/unauthorized/Makefile
+++ b/externals/unauthorized/Makefile
@@ -160,7 +160,7 @@ ifeq ($(UNAME),Linux)
   PD_PATH = /usr
   OPT_CFLAGS = -O6 -funroll-loops -fomit-frame-pointer
   ALL_CFLAGS += -fPIC
-  ALL_LDFLAGS += -rdynamic -shared -fPIC -Wl,-rpath,"\$$ORIGIN",--enable-new-dtags
+  ALL_LDFLAGS += -rdynamic -shared -fPIC -Wl,-rpath,"\$$ORIGIN",--enable-new-dtags -L/app/lib
   SHARED_LDFLAGS += -Wl,-soname,$(SHARED_LIB) -shared
   ALL_LIBS += -lc $(LIBS_linux)
   STRIP = strip --strip-unneeded -R .note -R .comment
