diff --git a/CMakeLists.txt b/CMakeLists.txt
index f3de258..c5b1ea4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.12)
 
-project(obs-mute-filter VERSION 0.3.0)
+project(mute-filter VERSION 0.3.0)
 
 # Replace `Your Name Here` with the name (yours or your organization's) you want
 # to see as the author of the plugin (in the plugin's metadata itself and in the installers)
