diff --git a/PukllanapacActivity.py b/PukllanapacActivity.py
index ba9a388..b852184 100644
--- a/PukllanapacActivity.py
+++ b/PukllanapacActivity.py
@@ -40,6 +40,8 @@ from toolbar_utils import radio_factory, label_factory, separator_factory, \
     button_factory
 from window import Game
 
+from sugarapp.widgets import SugarCompatibleActivity
+
 SERVICE = 'org.sugarlabs.PukllanapacActivity'
 IFACE = SERVICE
 PATH = '/org/augarlabs/PukllanapacActivity'
@@ -47,7 +49,7 @@ LEVEL_ICONS = ['level1', 'level2', 'level3']
 GAME_ICONS = ['rectangle', 'hexagon', 'hexagon2']
 
 
-class PukllanapacActivity(activity.Activity):
+class PukllanapacActivity(SugarCompatibleActivity):
     """ Sliding puzzle game """
 
     def __init__(self, handle):
