diff --git a/activity/activity.info b/activity/activity.info
index aa43ea2..cc77365 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,11 +2,17 @@
 name = Pukllanapac
 activity_version = 14
 license = GPLv3+;MIT
-bundle_id = org.sugarlabs.PukllanapacActivity
+metadata_license = CC0-1.0
+bundle_id = org.sugarlabs.Pukllanapac
+release_date = 2020-06-21
 exec = sugar-activity3 PukllanapacActivity.PukllanapacActivity
 icon = activity-pukllanapac
 show_launcher = yes
-tags = Game; Maths
+tags = Game
 repository = https://github.com/sugarlabs/pukllanapac
-summary = A sliding-puzzle game with objective to rearrange tiles so sectors become of the same color.
+summary = A sliding-puzzle game
+description = Pukllanapac is a sliding-puzzle game. The objective is to rearrange tiles so that all of the circles (and semicircles) are composed of sectors of the same color. There are three different patterns: circles, triangles, and hexagons. Drag tiles to swap their positions. Click on tiles to rotate them.
 url = https://help.sugarlabs.org/en/pukllanapac.html
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/pukllanapac/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/pukllanapac/master/screenshots/en/2.png https://raw.githubusercontent.com/sugarlabs/pukllanapac/master/screenshots/en/3.png
