--- a/extern/pybind11/tools/FindPythonLibsNew.cmake 2021-02-11 00:14:13.000000000 +0200
+++ b/extern/pybind11/tools/FindPythonLibsNew.cmake        2021-04-30 00:37:12.571898600 +0300
@@ -237,7 +237,7 @@
 # cache entries because they are meant to specify the location of a single
 # library. We now set the variables listed by the documentation for this
 # module.
-set(PYTHON_INCLUDE_DIRS "${PYTHON_INCLUDE_DIR}")
+set(PYTHON_INCLUDE_DIRS "/usr/include/python3.8")
 set(PYTHON_LIBRARIES "${PYTHON_LIBRARY}")
 if(NOT PYTHON_DEBUG_LIBRARY)
   set(PYTHON_DEBUG_LIBRARY "") 
