diff --git a/activity/activity.info b/activity/activity.info
index eabf592..07d4039 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,8 +2,16 @@
 name = PyEyes
 activity_version = 2
 host_version = 1
-bundle_id = org.laptop.community.PyEyes
+bundle_id = org.sugarlabs.PyEyes
+release_date = 2011-04-05
 icon = activity
 exec = sugar-activity3 activity.PyEyesActivity
 license = GPLv3
+metadata_license = CC0-1.0
 repository = https://github.com/sugarlabs/pyeyes-activity.git
+developer_name = Sugar Labs Community
+screenshots = http://activities.sugarlabs.org/en-US/sugar/images/t/602/1302138013
+tags = Education
+update_contact = tch@sugarlabs.org
+summary = Eyes following the cursor
+description = PyEyes is a program where eyes follow the mouse pointer and move according to it.
\ No newline at end of file
