diff --git a/activity.py b/activity.py
index 3698232..98a3a17 100755
--- a/activity.py
+++ b/activity.py
@@ -7,6 +7,7 @@ gi.require_version('Gtk', '3.0')
 from gi.repository import Gtk
 from gi.repository import Gdk
 
+from sugarapp.helpers import PrimaryMonitor
 from sugar3.activity.activity import Activity
 from sugar3.activity.widgets import StopButton
 from sugar3.activity.widgets import ActivityToolbarButton
@@ -26,9 +27,8 @@ class PyEyesActivity(Activity):
         box = Gtk.HBox()
         self.set_canvas(box)
 
-        screen = Gdk.Screen.get_default()
-        rw = screen.get_width() / 2
-        rh = screen.get_height() / 2
+        rw = PrimaryMonitor.width() / 2
+        rh = PrimaryMonitor.height() / 2
 
         self.eye1 = eyes.Eye(Gdk.Color.parse("#FFFFFF")[1])
         self.eye1.set_size_request(rw, rh)
