diff --git a/activity.py b/activity.py
index 3698232..6c2d768 100755
--- a/activity.py
+++ b/activity.py
@@ -9,14 +9,15 @@ from gi.repository import Gdk
 
 from sugar3.activity.activity import Activity
 from sugar3.activity.widgets import StopButton
-from sugar3.activity.widgets import ActivityToolbarButton
+from sugarapp.widgets import ExtendedActivityToolbarButton
 from sugar3.graphics.toolbarbox import ToolbarBox
 
+from sugarapp.widgets import SugarCompatibleActivity
 
-class PyEyesActivity(Activity):
 
+class PyEyesActivity(SugarCompatibleActivity):
     def __init__(self, handle):
-        Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self.add_events(Gdk.EventMask.POINTER_MOTION_MASK)
         self.connect("motion-notify-event", lambda w, e: self.look_at((e.x, e.y)))
@@ -46,7 +47,7 @@ class PyEyesActivity(Activity):
         toolbarbox = ToolbarBox()
         self.set_toolbar_box(toolbarbox)
 
-        toolbarbox.toolbar.insert(ActivityToolbarButton(self), -1)
+        toolbarbox.toolbar.insert(ExtendedActivityToolbarButton(self), -1)
         toolbarbox.toolbar.insert(StopButton(self), -1)
 
         separator = Gtk.SeparatorToolItem()
