diff --git a/bootloader/wscript b/bootloader/wscript
index 46650090b04f..c19f4c2fabb0 100644
--- a/bootloader/wscript
+++ b/bootloader/wscript
@@ -503,7 +503,6 @@ def configure(ctx):
         ctx.env.append_value(
             'CFLAGS', [
                 '-Wall',
-                '-Werror',
                 '-Wno-error=unused-variable',
                 '-Wno-error=unused-function',
             ]
