diff -up pymol-open-source-2.3.0/contrib/mmtf-c/mmtf_parser.cpp.orig contrib/mmtf-c/mmtf_parser.cpp
--- pymol-open-source-2.3.0.orig/contrib/mmtf-c/mmtf_parser.cpp.orig	2018-08-06 14:17:09.154969915 +0200
+++ pymol-open-source-2.3.0/contrib/mmtf-c/mmtf_parser.cpp	2018-08-06 14:18:24.045410763 +0200
@@ -27,6 +27,10 @@
 #define WIN32_LEAN_AND_MEAN
 #define __STDC_LIMIT_MACROS
 
+#if ! defined(unix)
+#define unix 1
+#endif
+
 #include "mmtf_parser.h"
 
 //*** Standard libs
