diff -ur pyotherside-1.5.4.old/qtquicktests/qtquicktests.pro pyotherside-1.5.4/qtquicktests/qtquicktests.pro
--- pyotherside-1.5.4.old/qtquicktests/qtquicktests.pro	2019-03-24 13:42:28.406420156 +0100
+++ pyotherside-1.5.4/qtquicktests/qtquicktests.pro	2019-03-24 13:45:01.644942702 +0100
@@ -1,4 +1,4 @@
 TEMPLATE = app
 TARGET = qtquicktests
-CONFIG += warn_on qmltestcase
+CONFIG += warn_on qmltestcase no_testcase_installs
 SOURCES += qtquicktests.cpp
diff -ur pyotherside-1.5.4.old/src/src.pro pyotherside-1.5.4/src/src.pro
--- pyotherside-1.5.4.old/src/src.pro	2019-03-24 13:42:28.406420156 +0100
+++ pyotherside-1.5.4/src/src.pro	2019-03-24 13:43:01.842097488 +0100
@@ -10,7 +10,7 @@
 CONFIG += qt plugin
 QT += qml quick svg
 
-target.path = $$[QT_INSTALL_QML]/$$PLUGIN_IMPORT_PATH
+target.path = $$PREFIX/lib/qml/$$PLUGIN_IMPORT_PATH
 INSTALLS += target
 
 qmldir.files += qmldir pyotherside.qmltypes
