diff -up PyQt5_gpl-5.10.2.dev1805251538/sip/QtTest/qtestmouse.sip.qt511 PyQt5_gpl-5.10.2.dev1805251538/sip/QtTest/qtestmouse.sip
--- PyQt5_gpl-5.10.2.dev1805251538/sip/QtTest/qtestmouse.sip.qt511	2018-05-25 20:33:17.000000000 -0500
+++ PyQt5_gpl-5.10.2.dev1805251538/sip/QtTest/qtestmouse.sip	2018-05-29 13:33:57.896800513 -0500
@@ -41,7 +41,6 @@ namespace QTest
     void mousePress(QWidget *widget, Qt::MouseButton button, Qt::KeyboardModifiers modifier = 0, QPoint pos = QPoint(), int delay = -1);
     void mouseRelease(QWidget *widget, Qt::MouseButton button, Qt::KeyboardModifiers modifier = 0, QPoint pos = QPoint(), int delay = -1);
     void mouseEvent(QTest::MouseAction action, QWidget *widget, Qt::MouseButton button, Qt::KeyboardModifiers modifier, QPoint pos, int delay = -1);
-    void waitForEvents() /ReleaseGIL/;
     void mouseEvent(QTest::MouseAction action, QWindow *window, Qt::MouseButton button, Qt::KeyboardModifiers modifier, QPoint pos, int delay = -1);
     void mousePress(QWindow *window, Qt::MouseButton button, Qt::KeyboardModifiers modifier = Qt::KeyboardModifiers(), QPoint pos = QPoint(), int delay = -1);
     void mouseRelease(QWindow *window, Qt::MouseButton button, Qt::KeyboardModifiers modifier = Qt::KeyboardModifiers(), QPoint pos = QPoint(), int delay = -1);
