diff -ur pyscard-release-1.9.9.old/setup.py pyscard-release-1.9.9/setup.py
--- pyscard-release-1.9.9.old/setup.py	2019-08-20 20:34:13.468821467 +0200
+++ pyscard-release-1.9.9/setup.py	2019-08-20 20:34:27.476686855 +0200
@@ -76,7 +76,7 @@
     platform_swig_opts = ['-DPCSCLITE']
     platform_sources = []
     platform_libraries = []
-    platform_include_dirs = ['/usr/include/PCSC', '/usr/local/include/PCSC']
+    platform_include_dirs = ['/app/include/PCSC']
     platform_extra_compile_args = []    # ['-ggdb', '-O0']
     platform_extra_link_args = []   # ['-ggdb']
 
