diff --git a/paperwork_backend/__init__.py b/paperwork_backend/__init__.py
index d89b3e0..c6585a2 100644
--- a/paperwork_backend/__init__.py
+++ b/paperwork_backend/__init__.py
@@ -16,7 +16,8 @@ def init_flatpak():
     """
     tessdata_files = glob.glob("/app/share/locale/*/*.traineddata")
     if len(tessdata_files) <= 0:
-        return False
+        logger.info("Is Flatpak enabled ? %s", str(os.path.exists("/app")))
+        return os.path.exists("/app")
 
     localdir = os.path.expanduser("~/.local")
     base_data_dir = os.getenv(
