diff -up PyQt5_gpl-5.9/configure.py.sipdir PyQt5_gpl-5.9/configure.py
--- PyQt5_gpl-5.9/configure.py.sipdir	2017-07-03 12:24:48.000000000 -0500
+++ PyQt5_gpl-5.9/configure.py	2017-07-05 20:31:35.319524089 -0500
@@ -25,6 +25,7 @@ import os
 import shutil
 import stat
 import sys
+import sipconfig
 
 
 # Initialise the constants.
@@ -485,7 +486,7 @@ class TargetConfiguration:
         self.pyqt_bin_dir = py_config.bin_dir
         self.pyqt_module_dir = py_config.module_dir
         self.pyqt_stubs_dir = os.path.join(py_config.module_dir, 'PyQt5')
-        self.pyqt_sip_dir = os.path.join(py_config.data_dir, 'sip', 'PyQt5')
+        self.pyqt_sip_dir = os.path.join(sipconfig.Configuration().default_sip_dir, 'PyQt5')
         self.pyuic_interpreter = py_config.pyuic_interpreter
 
         # Remaining values.
