diff --git a/data/qalculate-gtk.appdata.xml.in b/data/qalculate-gtk.appdata.xml.in
index 4c2e95b..2085e45 100644
--- a/data/qalculate-gtk.appdata.xml.in
+++ b/data/qalculate-gtk.appdata.xml.in
@@ -43,9 +43,6 @@
   <kudo>UserDocs</kudo>
  </kudos>
  <launchable type="desktop-id">qalculate-gtk.desktop</launchable>
- <provides>
-  <binary>qalculate-gtk</binary>
- </provides>
  <translation type="gettext">qalculate-gtk</translation>
  <releases>
   <release version="3.11.0" date="2020-06-22">
diff --git a/src/interface.cc b/src/interface.cc
index 07d680a..47efce9 100644
--- a/src/interface.cc
+++ b/src/interface.cc
@@ -1144,12 +1144,10 @@ void create_main_window(void) {
 	gtk_style_context_add_provider(gtk_widget_get_style_context(GTK_WIDGET(gtk_builder_get_object(main_builder, "box_rpnl"))), GTK_STYLE_PROVIDER(box_rpnl_provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
 	gtk_style_context_add_provider_for_screen(gdk_screen_get_default(), GTK_STYLE_PROVIDER(app_provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
 #if GTK_MAJOR_VERSION > 3 || GTK_MINOR_VERSION >= 16
-#	ifdef _WIN32
 	app_provider_theme = gtk_css_provider_new();
 	gtk_style_context_add_provider_for_screen(gdk_screen_get_default(), GTK_STYLE_PROVIDER(app_provider_theme), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
 	if(use_dark_theme > 0) gtk_css_provider_load_from_resource(app_provider_theme, "/org/gtk/libgtk/theme/Adwaita/gtk-contained-dark.css");
 	else if(use_dark_theme == 0) gtk_css_provider_load_from_resource(app_provider_theme, "/org/gtk/libgtk/theme/Adwaita/gtk-contained.css");
-#	endif
 #endif
 
 	set_mode_items(printops, evalops, CALCULATOR->defaultAssumptions()->type(), CALCULATOR->defaultAssumptions()->sign(), rpn_mode, CALCULATOR->getPrecision(), CALCULATOR->usesIntervalArithmetic(), CALCULATOR->variableUnitsEnabled(), adaptive_interval_display, visible_keypad, auto_calculate, complex_angle_form, true);
@@ -2057,10 +2055,8 @@ GtkWidget* get_preferences_dialog(void) {
 			}
 		}
 #if GTK_MAJOR_VERSION > 3 || GTK_MINOR_VERSION >= 16
-#	ifdef _WIN32
 		gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(preferences_builder, "preferences_checkbutton_dark_theme")), use_dark_theme > 0);
 		gtk_widget_show(GTK_WIDGET(gtk_builder_get_object(preferences_builder, "preferences_checkbutton_dark_theme")));
-#	endif
 #endif
 		gtk_builder_connect_signals(preferences_builder, NULL);
 
