diff -Naur qalculate-gtk-2.9.0.orig/data/qalculate-gtk.appdata.xml.in qalculate-gtk-2.9.0/data/qalculate-gtk.appdata.xml.in
--- qalculate-gtk-2.9.0.orig/data/qalculate-gtk.appdata.xml.in	2019-02-03 15:07:08.000000000 +0100
+++ qalculate-gtk-2.9.0/data/qalculate-gtk.appdata.xml.in	2019-02-16 16:39:45.456475107 +0100
@@ -1,25 +1,25 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2003-2007, 2008, 2016-2019 Hanna Knutsson <hanna.knutsson@protonmail.com> -->
 <component type="desktop">
- <id>qalculate-gtk.desktop</id>
+ <id>io.github.qalculate.Qalculate</id>
  <metadata_license>CC0-1.0</metadata_license>
  <project_license>GPL-2.0+</project_license>
  <_name>Qalculate! (GTK+ UI)</_name>
  <_summary>Powerful and easy to use calculator</_summary>
- <_description>
-  <p>
+ <description>
+  <_p>
    Qalculate! is a multi-purpose cross-platform desktop calculator.
    It is simple to use but provides power and versatility normally
    reserved for complicated math packages, as well as useful tools for
    everyday needs (such as currency conversion and percent calculation).
-  </p>
-  <p>
+  </_p>
+  <_p>
    Features include a large library of customizable functions, unit
    calculations and conversion, physical constants, symbolic calculations
    (including integrals and equations), arbitrary precision, uncertainty
    propagation, interval arithmetic, plotting, and a user-friendly interface.
-  </p>
- </_description>
+  </_p>
+ </description>
  <screenshots>
   <screenshot type="default">
    <image>https://github.com/Qalculate/qalculate-gtk/raw/master/data/qalculate-gtk-appdata-1.png</image>
@@ -32,4 +32,317 @@
  </screenshots>
  <url type="homepage">http://qalculate.github.io/</url>
  <url type="bugtracker">https://github.com/Qalculate/qalculate-gtk/issues</url>
+ <developer_name>Hanna Knutsson</developer_name>
+ <content_rating type="oars-1.0" />
+ <kudos>
+  <kudo>HiDpiIcon</kudo>
+  <kudo>ModernToolkit</kudo>
+  <kudo>UserDocs</kudo>
+ </kudos>
+ <launchable type="desktop-id">io.github.qalculate.Qalculate.desktop</launchable>
+ <provides>
+  <binary>qalculate-gtk</binary>
+ </provides>
+ <releases>
+  <release version="2.9.0" date="2019-02-11">
+   <description>
+    <p>This version primarily includes a large number of stability and reliability improvements, thanks to automatic testing of hundreds of thousands of randomly generated mathematical expressions.</p>
+    <p>Changes (including changes from libqalculate):</p>
+     <ul>
+      <li>Fix convert button</li>
+      <li>Propagation of uncertainty using variance formula (if interval arithmetic is disabled)</li>
+      <li>Improved and extended physical constants, now defined in relation to each other (e.g. RK = (μ0c)/(2α))</li>
+      <li>Natural, atomic, and Planck units</li>
+      <li>Reciprocal unit conversion (e.g. 5 m/s = 0.2 s/m, 60 mpg = 0.039 L/km)</li>
+      <li>Updated elements and planets data sets</li>
+      <li>+/- syntax for non-numeric values</li>
+      <li>raid() function for RAID capacity calculation</li>
+      <li>UTC+/-hh[:mm] and CET time zone conversion (using "to")</li>
+      <li>Minor bug fixes and feature improvements</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.8.2" date="2018-12-03">
+   <description>
+    <p>This version primarily includes a large number of stability and reliability improvements, thanks to automatic testing of hundreds of thousands of randomly generated mathematical expressions.</p>
+    <p>Changes:</p>
+     <ul>
+      <li>Wrap lines and right align results in history</li>
+      <li>Fix missing prefix for gram (and are) when unit exponent not one and automatic prefixes is off</li>
+      <li>Fixes for units with non-linear relation</li>
+      <li>Minor bug fixes and feature improvements</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.8.1" date="2018-11-07">
+   <description>
+    <p>This version primarily includes a large number of stability and reliability improvements, thanks to automatic testing of hundreds of thousands of randomly generated mathematical expressions.</p>
+    <p>GUI Changes:</p>
+     <ul>
+      <li>Modified keypad layout</li>
+      <li>RPN stack buttons with icons</li>
+      <li>Use 'linked' widget style where appropriate</li>
+     </ul>
+    <p>From libqalculate:</p>
+     <ul>
+      <li>Interpret a character preceded by '' as an unknown variable (same as in quotation marks)</li>
+      <li>Do not show fractions with very large numerators or denominators, by default</li>
+      <li>Hectare and decare unit names (in addition to ha/da and hectoare/decaare)</li>
+      <li>Always enable units for expressions that ends with "to [unit expression]", "to optimal", or similar</li>
+      <li>Improve appearance of "x m/s" and similar</li>
+      <li>Fix (1x+ay)^b (was not expanded)</li>
+      <li>Fix roman() when used for reversed conversion</li>
+      <li>Fix compilation without readline</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.8.0" date="2018-10-29">
+   <description>
+    <p>This version primarily includes a large number of stability and reliability improvements, thanks to automatic testing of hundreds of thousands of randomly generated mathematical expressions.</p>
+    <p>Changes (from libqalculate):</p>
+     <ul>
+      <li>Equation solved and improved simplification for expressions with trigonometric functions</li>
+      <li>General improvements to simplification and equation solving</li>
+      <li>Improved factorization</li>
+      <li>New magnitude(), entrywise() and hadamard() functions, and new n variable (unknown integer)</li>
+      <li>Parse |x| and ||x|| as abs(x) and magnitude(x)</li>
+      <li>Use kilo as prefix for gram if automatic prefix is off</li>
+      <li>Many bug fixes and minor feature improvements</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.6.2" date="2018-08-13">
+   <description>
+    <p>Changes:</p>
+     <ul>
+      <li>Polar and exponential complex forms (changed with mode or with "to" operator/command)</li>
+      <li>Fix f(x)+a=a (e.g. x+1=1, which returned false/zero)</li>
+      <li>Fix ln(-i)</li>
+      <li>Fix automatic unit conversion in RPN mode</li>
+      <li>Allow conversion of the current result using standalone "to" expression (e.g. type "to m" to convert current result to meter) in GUI</li>
+      <li>Option to disable automatic conversion to local currency (when optimal conversion activated)</li>
+      <li>Minor bug fixes</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.6.1" date="2018-07-09">
+   <description>
+    <p>Changes:</p>
+     <ul>
+      <li>Do not always download all sources during automatic exchange rates updates (updating of common currencies will be much faster)</li>
+      <li>Configurable completion suggestions</li>
+      <li>Option to keep insert function dialog open for repeated function calculations</li>
+      <li>Minor bug fixes and feature enhancements</li>
+     </ul>
+    <p>This release does not include any changes to libqalculate or the CLI application.</p>
+   </description>
+  </release>
+  <release version="2.6.0" date="2018-06-11">
+   <description>
+    <p>GUI improvements:</p>
+     <ul>
+      <li>Calendar conversion</li>
+      <li>Currency flags in completion suggestions list and result display</li>
+      <li>Extended completion suggestions, matching titles (and countries for currencies) when at least three characters are entered</li>
+      <li>Completion suggestions for prefixes and prefixed units</li>
+      <li>Completion suggestions for "to" commands</li>
+      <li>Add duodecimal and roman numerals to number base conversion dialog</li>
+      <li>Duodecimal, roman, base #, and partial fraction "to" conversion commands</li>
+      <li>Minor bug fixes and feature enhancements</li>
+     </ul>
+    <p>Changes from libqalculate:</p>
+     <ul>
+      <li>Calendar conversion</li>
+      <li>5'8" notation for feet and inches, 5°12'30" notation for degrees, arcminutes and arcseconds</li>
+      <li>Interpret 5m 7cm as 5m+7cm, and 3h 52min 20s as 3h+52min+20s, and similar</li>
+      <li>Improved ln() simplifications</li>
+      <li>Lunar phase functions</li>
+      <li>Fix secondary exchange rates source</li>
+      <li>Fix crashes with compiler flag "-D_GLIBCXX_ASSERTIONS"</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.5.0" date="2018-05-15">
+   <description>
+    <p>GUI improvements:</p>
+     <ul>
+      <li>Currency flags</li>
+      <li>Context menu for conversion list, and middle click for adding unit to expression</li>
+      <li>Search entry for functions, units and variables</li>
+      <li>Improved convert to menu in unit manager</li>
+      <li>Workarounds for bugs in GTK+ themes</li>
+     </ul>
+    <p>Changes from libqalculate:</p>
+     <ul>
+      <li>Improved interval arithmetic with dependent values (variables and function arguments). Most differentiable expressions do now return a properly narrow interval</li>
+      <li>Experimental limit() function</li>
+      <li>Two's complement representation for binary numbers</li>
+      <li>Improved currency titles</li>
+      <li>bitcmp() function</li>
+      <li>Allow all rational numbers for second argument of root()</li>
+      <li>Fix segfault when prefix is set in some cases</li>
+      <li>Fix a^infinity when a > 1</li>
+      <li>Minor bug fixes</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.4.0" date="2018-04-11">
+   <description>
+    <p>List of changes:</p>
+     <ul>
+      <li>Simplified percent calculation (e.g. 100 + 20% = 120)</li>
+      <li>Direct date and time arithmetic (e.g. "2018-03-10" + 372 h = "2018-03-25T13:00:00"), with time zone support</li>
+      <li>Automatic conversion to local currency (in "optimal" conversion mode)</li>
+      <li>Calculate each vector element (and each extra argument) separately in many functions (e.g. sin(5, 2, 3) = [-0.95892427, 0.90929743, 0.14112001])</li>
+      <li>Unit-like conversion for physical constants (using "to")</li>
+      <li>Allow user to change decimal separator (and consquently argument separator) independantly of locale</li>
+      <li>Enhanced timastamp() and stamptodate()</li>
+      <li>Fix function completion (GUI)</li>
+      <li>Fix lambertw()/productlog()</li>
+      <li>Fix display of parsed sexagesimal numbers</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.3.0" date="2018-03-28">
+   <description>
+    <p>The main focus of version 2.3 is enhanced integration capabilities and basic solving of differential equations. Qalculate! should be much more usable for calculation of integrals now, even though it is still far from perfect.</p>
+    <p>There is also some important plotting fixes, as well as many minor bug fixes and feature enhancements.</p>
+   </description>
+  </release>
+  <release version="2.2.1" date="2018-01-17">
+   <description>
+    <p>Version 2.2.1 includes bug fixes and minor feature enhancements.</p>
+   </description>
+  </release>
+  <release version="2.2.0" date="2017-12-07">
+   <description>
+    <p>The main focus of version 2.2 is interval arithmetic, which can be used for determination of the number of significant digits in the final result or direct calculation with intervals of numbers. Note that more work is needed to avoid unnecessary interval widening for non-trivial expressions with dependent variables.</p>
+    <p>List of main changes:</p>
+     <ul>
+      <li>Interval arithmetic</li>
+      <li>Digit grouping</li>
+      <li>Enhanced display of parsed expression</li>
+      <li>Use 10^n instead of E-notation for main result display by default</li>
+      <li>Bitcoin support (frequent exchange rate updates are recommended)</li>
+      <li>Use 'X' and 'E' (or appropriate Unicode characters), instead of 'A' and 'B', in duodecimal numbers (base 12)</li>
+      <li>Option to disable units in physical constants</li>
+      <li>Fixed engineering mode</li>
+      <li>Various bug fixes and minor feature enhancements</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.1.0" date="2017-10-02">
+   <description>
+    <p>The main focus of Qalculate! version 2.1 is consistent handling of roots of negative numbers and equation solving enhancements.</p>
+    <p>List of main changes:</p>
+     <ul>
+      <li>Highlighting of matching parentheses</li>
+      <li>Proper display of roots and absolute value</li>
+      <li>More than 100 additional currencies (using mycurrency.net)</li>
+      <li>Equation solving enhancements</li>
+      <li>root() and cbrt() now always returns the real root (e.g. cbrt(-8)=-2), while fractional powers return the principal root ((-8)^(1/3)=1+1,73i)</li>
+      <li>Enhanced simplification</li>
+      <li>Minor bug fixes and feature enhancements</li>
+     </ul>
+   </description>
+  </release>
+  <release version="2.0.0" date="2017-08-28">
+   <description>
+    <p>After a month of hard work Qalculate! version 2.0 is ready with a redesigned expression entry and keypad in the GUI, and a new feature complete Windows version.</p>
+    <p>List of main changes:</p>
+     <ul>
+      <li>Redesigned keypad with a lot more functionality</li>
+      <li>Multiline expression entry with undo/redo</li>
+      <li>Alternative simplified function creation dialog</li>
+      <li>Percentage calculation tool</li>
+      <li>Binary installer for Microsoft Windows</li>
+      <li>Date functions have been rewritten and now support much larger numbers</li>
+      <li>Trigonometric functions with angle unit other than radians have been fixed</li>
+      <li>Functions with more than one argument can now also be applied to the RPN stack</li>
+      <li>New functions: airy(), arg(), besselj(), bessely(), conj(), digamma(), erf(), erfc()</li>
+      <li>Many minor enhancements and bug fixes</li>
+     </ul>
+   </description>
+  </release>
+  <release version="1.0.99" date="2017-08-09">
+   <description>
+    <p>Qalculate! has long been available for GNU/Linux and Mac OS X, but now it is finally also compatible with Microsoft Windows (although there is a fork by Cheng Sun with Windows compatibility, which has contributed code to this version). This pre-release version is likely not as stable as version 1.0.0 (although it does include some important bug fixes), but you are nevertheless invited to try it, report any issues you find and share your impressions.</p>
+   </description>
+  </release>
+  <release version="1.0.0" date="2017-08-28">
+   <description>
+    <p>Qalculate! version 1.0 includes the following changes:</p>
+     <ul>
+      <li>Increased efficiency and stability. Slow calculations can in most cases now be terminated in a controlled manner and the application should no longer potentially crash when trying to calculate extremely large numbers</li>
+      <li>Enhanced factorization</li>
+      <li>Revised temperature unit handling</li>
+      <li>Enhanced automatic unit conversion, including a new default mode which keeps original non-SI units if no better alternative (that uses fewer units and exponents) is found</li>
+      <li>A long list of bug fixes and minor feature enhancements</li>
+     </ul>
+   </description>
+  </release>
+  <release version="0.9.12" date="2017-05-29">
+   <description>
+    <p>Qalculate! version 0.9.12 includes fixes to unit parsing in adaptive parsing mode, build fixes and other minor bug fixes. The fork with support for Microsoft Windows has been merged (compilation on Windows has however not been tested).</p>
+   </description>
+  </release>
+  <release version="0.9.11" date="2017-03-06">
+   <description>
+    <p>Qalculate! version 0.9.11 includes both some important bug fixes and a number of new features. Among other things, a new enhanced history view and mixed units conversion have been added. The RPN functionality has received much needed attention and should now be more usable.</p>
+    <p>Changes:</p>
+     <ul>
+      <li>Updated history view. The calculation log is now presented in an actual list. Buttons and functions have been added for calculation with previous expression and results</li>
+      <li>Enhanced RPN stack manipulation (swap, rotation and copy) and arithmetics buttons next to stack list and combine RPN options in a single menu</li>
+      <li>Calculation of text in spinboxes for integer function arguments</li>
+      <li>Option to use multiple instances by default</li>
+      <li>Mixed units conversion (e.g. "15 in = 1 ft + 3 in" and "3,2 h = 3 h + 12 min")</li>
+      <li>Fixes to unit conversion (e.g. "1/(1 ns) to GHz")</li>
+      <li>Fix slow calculation of exponentiations with big numerators in exponents (e.g. "2^0.333333333")</li>
+      <li>Critical fixes for RPN syntax</li>
+      <li>Fix degree sign</li>
+      <li>Fix transformation of last arguments to vector, when only one last argument (fixes function("1+\x"; 1))</li>
+      <li>Various other minor enhancements and bug fixes</li>
+     </ul>
+   </description>
+  </release>
+  <release version="0.9.9" date="2016-07-26">
+   <description>
+    <p>Qalculate! version 0.9.9 is a minor release with some important bug fixes, as well as a few new features.</p>
+    <p>Changes:</p>
+     <ul>
+      <li>Only a single instance of the application will run by default (use command line switch '-n' for old behaviour)</li>
+      <li>The application will no longer cause segmentation faults when using a locale unsupported by the users system</li>
+      <li>Parsing of "today" fixed and "tomorrow" and "yesterday" added</li>
+      <li>Qalculate! now compiles, without errors and warning, using clang</li>
+     </ul>
+   </description>
+  </release>
+  <release version="0.9.8" date="2016-06-15">
+   <description>
+    <p>Qalculate! version 0.9.8 includes many bug fixes, but due to limited testing some new bugs have probably been introduced.</p>
+    <p>Changes:</p>
+     <ul>
+      <li>Lots of bug fixes and minor improvements</li>
+      <li>The GTK+ UI has been ported to GTK+ 3 and has been polished in the process</li>
+      <li>The convert dialog has been transformed into a tab page in the main window, with auto-selection of unit category and continuous conversion</li>
+      <li>Less obtrusive display of errors and warnings. They are now shown as tooltips for an icon next to the expression entry</li>
+      <li>Spinning animations show calculation/display activity and an icon in the expression entry halts the process, instead of a message dialog</li>
+      <li>A tooltip for the result display will show the result in text format less open to interpretation</li>
+      <li>Automatic on demand updates of exchanges rates</li>
+      <li>Better resizing of main windows. Tabs will be resized instead of result display and keypad view will remember size independently of the history view</li>
+      <li>'to' can now not only be used for conversion to unit expressions. "to bin", "to oct", "to hex" and "to bases" changes number base displayed, "to factors" factorizes, "to fraction" displays as fraction, and "to base" and "to optimal" auto-converts units</li>
+      <li>Three different parsing modes can be chosen from (providing mainly different handling of implicit multiplication). The default 'adaptive' mode has been enhanced</li>
+      <li>By default prefixes will not be added to for example imperial units</li>
+      <li>Changed handling of degrees Celsius and Fahrenheit. Conversion will take place already during parsing</li>
+      <li>When converting to unit without prefix, no prefix will be added to the result unless the unit expression is prepended with '?' ('0' works in the opposite direction)</li>
+      <li>Option to ignore commas in numbers (for use as thousands separator)</li>
+      <li>New function: fibonacci (Qalculate! can now tell you that the millionth fibonacci number is approximately 1,953282129 * 10^208987, although for obvious reasons the exact number cannot be displayed)</li>
+      <li>New units: Hartree, Rydberg, Dalton (same as u)</li>
+      <li>Currencies have been updated to reflect the available data from ECB</li>
+      <li>Use ~/.local and ~/.conf to store configuration and definitions</li>
+      <li>The KDE UI has been abandoned for now and is not included in this release</li>
+     </ul>
+   </description>
+  </release>
+ </releases>
 </component>
