diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 3e47c90..5ac4b3e 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -1,4 +1,8 @@

+include_directories(
+    ${ADWAITAQT_INCLUDE_DIRS}
+)
+
 set(common_SRCS
     gnomesettings.cpp
 )
diff --git a/src/decoration/CMakeLists.txt b/src/decoration/CMakeLists.txt
index e98ec5c..4959a96 100644
--- a/src/decoration/CMakeLists.txt
+++ b/src/decoration/CMakeLists.txt
@@ -1,5 +1,6 @@

 include_directories(
+    ${ADWAITAQT_INCLUDE_DIRS}
     ${CMAKE_SOURCE_DIR}/src/common
 )