From b92a4b1b3bc47c6de758d549959b4edb78870f8f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Verschelde?= <rverschelde@gmail.com>
Date: Mon, 27 Jul 2020 13:38:07 +0200
Subject: [PATCH] QMake: Fix make install for Unix

---
 laigter.pro | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/laigter.pro b/laigter.pro
index 008f45c..ed58c97 100644
--- a/laigter.pro
+++ b/laigter.pro
@@ -112,9 +112,9 @@ unix{
     shortcutfiles.files = dist/laigter.desktop
     shortcutfiles.path = $$PREFIX/share/applications/
     iconfiles.files = dist/laigter.png
-    iconfiles.path = $$PREFIX/share/icons/hicolor/256x256/
-    appdatafiles.files = dist/laigter.appdata.xlm
-    appdatafiles.path = $$PREFIX/share/metainfo
+    iconfiles.path = $$PREFIX/share/icons/hicolor/256x256/apps/
+    appdatafiles.files = dist/laigter.appdata.xml
+    appdatafiles.path = $$PREFIX/share/metainfo/
 
     INSTALLS += target
     INSTALLS += shortcutfiles
@@ -123,7 +123,7 @@ unix{
 }
 
 DISTFILES += \
-	ACKNOWLEDGEMETS \
+	ACKNOWLEDGEMENTS \
 	LICENSE
 
 TRANSLATIONS = translations/laigter_da.ts \
-- 
2.25.4

