--- pdfedit-git/src/qsa/configureOLD	2020-11-07 01:16:25.223786295 -0400
+++ pdfedit-git/src/qsa/configure	2020-11-07 13:48:55.926419801 -0400
@@ -15,7 +15,10 @@
 # invoke 2nd stage
 echo Building second stage2 configure
 cd configure2
-$QMAKE && make > /dev/null
+$QMAKE
+patch -p1 < ../../../patch.patch
+export QTDIR=/app
+make > /dev/null
 if [ $? -gt 0 -o ! -x configure2 ]; 
 then
 	echo Failed to build 
