diff -ur QScintilla_src-2.13.4.orig/src/qscintilla.pro QScintilla_src-2.13.4/src/qscintilla.pro
--- QScintilla_src-2.13.4.orig/src/qscintilla.pro	2023-01-15 19:13:30.751242600 +0100
+++ QScintilla_src-2.13.4/src/qscintilla.pro	2023-04-18 01:42:25.794709139 +0200
@@ -70,22 +70,22 @@
 # Scintilla namespace rather than pollute the global namespace.
 #DEFINES += SCI_NAMESPACE
 
-target.path = $$[QT_INSTALL_LIBS]
+target.path = /app/lib
 INSTALLS += target
 
-header.path = $$[QT_INSTALL_HEADERS]
+header.path = /app/include
 header.files = Qsci
 INSTALLS += header
 
-trans.path = $$[QT_INSTALL_TRANSLATIONS]
+trans.path = /app/share/qt5/translations
 trans.files = qscintilla_*.qm
 INSTALLS += trans
 
-qsci.path = $$[QT_INSTALL_DATA]
+qsci.path = /app/share/qt5
 qsci.files = ../qsci
 INSTALLS += qsci
 
-features.path = $$[QT_HOST_DATA]/mkspecs/features
+features.path = /app/lib/qt5/mkspecs/features
 CONFIG(staticlib) {
     features.files = $$PWD/features_staticlib/qscintilla2.prf
 } else {
