--- ./Qt4Qt5/qscintilla.pro
+++ ./Qt4Qt5/qscintilla.pro
@@ -47,25 +47,25 @@
 # Scintilla namespace rather than pollute the global namespace.
 #DEFINES += SCI_NAMESPACE
 
-target.path = $$[QT_INSTALL_LIBS]
+target.path = /app/lib
 INSTALLS += target
 
-header.path = $$[QT_INSTALL_HEADERS]
+header.path = /app/include
 header.files = Qsci
 INSTALLS += header
 
-trans.path = $$[QT_INSTALL_TRANSLATIONS]
+trans.path = /app/translations
 trans.files = qscintilla_*.qm
 INSTALLS += trans
 
-qsci.path = $$[QT_INSTALL_DATA]
+qsci.path = /app/qsci
 qsci.files = ../qsci
 INSTALLS += qsci
 
 greaterThan(QT_MAJOR_VERSION, 4) {
-    features.path = $$[QT_HOST_DATA]/mkspecs/features
+    features.path = /app/lib/mkspecs/features
 } else {
-    features.path = $$[QT_INSTALL_DATA]/mkspecs/features
+    features.path = /app/lib/mkspecs/features
 }
 CONFIG(staticlib) {
     features.files = $$PWD/features_staticlib/qscintilla2.prf
