--- qt-x11-free-3.3.8b/src/tools/qvaluelistOLD.h	2020-10-25 20:45:34.118471327 -0400
+++ qt-x11-free-3.3.8b/src/tools/qvaluelist.h	2020-10-25 20:46:30.704239458 -0400
@@ -45,6 +45,7 @@
 #include "qtl.h"
 #include "qshared.h"
 #include "qdatastream.h"
+#include <cstddef>
 #endif // QT_H
 
 #ifndef QT_NO_STL
--- qt-x11-free-3.3.8b/configureOLD	2020-11-05 00:28:51.778020402 -0400
+++ qt-x11-free-3.3.8b/configure	2020-11-05 00:30:52.497451175 -0400
@@ -1720,21 +1720,25 @@
 	fi
     fi
     # auto-detect Xcursor support
-    if [ "$CFG_XCURSOR" != "no" ]; then
-        if $x11tests/xcursor.test $XQMAKESPEC $OPT_VERBOSE $L_FLAGS $I_FLAGS; then
-	    if [ "$CFG_XCURSOR" = "yes" ] && [ "$CFG_CONFIGURE_EXIT_ON_ERROR" = "yes" ]; then
-		echo "Xcursor support cannot be enabled due to functionality tests!"
-		echo " Turn on verbose messaging (-v) to $0 to see the final report."
-		echo " If you believe this message is in error you may use the continue"
-		echo " switch (-continue) to $0 to continue."
-		exit 101
-            else
-                CFG_XCURSOR=no
-            fi
-	else
-            CFG_XCURSOR=yes
-        fi
-    fi
+    #if [ "$CFG_XCURSOR" != "no" ]; then
+    #    if $x11tests/xcursor.test $XQMAKESPEC $OPT_VERBOSE $L_FLAGS $I_FLAGS; then
+	#    if [ "$CFG_XCURSOR" = "yes" ] && [ "$CFG_CONFIGURE_EXIT_ON_ERROR" = "yes" ]; then
+	#	echo "Xcursor support cannot be enabled due to functionality tests!"
+	#	echo " Turn on verbose messaging (-v) to $0 to see the final report."
+	#	echo " If you believe this message is in error you may use the continue"
+	#	echo " switch (-continue) to $0 to continue."
+	#	exit 101
+    #        else
+    #            CFG_XCURSOR=no
+    #        fi
+	#else
+    #        CFG_XCURSOR=yes
+    #    fi
+    #fi
+
+    ### Force xcursor support ###
+    CFG_XCURSOR=yes
+    
     # auto-detect XRandR support (resize and rotate extension)
     if [ "$CFG_XRANDR" != "no" ]; then
 	if $x11tests/xrandr.test $XQMAKESPEC $OPT_VERBOSE $L_FLAGS $I_FLAGS; then
--- qt-x11-free-3.3.8b/src/tools/qcstringOLD.cpp	2020-11-01 22:55:05.236781386 -0400
+++ qt-x11-free-3.3.8b/src/tools/qcstring.cpp	2020-11-01 22:55:25.603690364 -0400
@@ -374,7 +374,7 @@
     int res;
     do {
 	bazip.resize( len + 4 );
-	res = ::compress(  (uchar*)bazip.data()+4, &len, (uchar*)data, nbytes );
+	res = compress(  (uchar*)bazip.data()+4, &len, (uchar*)data, nbytes );
 
 	switch ( res ) {
 	case Z_OK:
@@ -450,7 +450,7 @@
     int res;
     do {
         if ( baunzip.resize( len ) ) {
-	    res = ::uncompress( (uchar*)baunzip.data(), &len,
+	    res = uncompress( (uchar*)baunzip.data(), &len,
 				    (uchar*)data+4, nbytes-4 );
         } else {
             res = Z_MEM_ERROR;
--- qt-x11-free-3.3.8b/src/codecs/qt_codecsOLD.pri	2020-11-01 23:23:12.776091333 -0400
+++ qt-x11-free-3.3.8b/src/codecs/qt_codecs.pri	2020-11-01 23:23:58.728881090 -0400
@@ -3,9 +3,7 @@
 !bigcodecs:DEFINES += QT_NO_BIG_CODECS
 tools {
 	CODECS_P		= codecs
-	HEADERS += \
-#$$CODECS_H/qasmocodec.h \
-		$$CODECS_H/qbig5codec.h \
+	HEADERS += $$CODECS_H/qbig5codec.h \
 		  $$CODECS_H/qeucjpcodec.h \
 		  $$CODECS_H/qeuckrcodec.h \
 		  $$CODECS_P/qisciicodec_p.h \
@@ -21,9 +19,7 @@
 		  $$CODECS_H/qtextcodecfactory.h \
 		  $$CODECS_H/qtextcodecplugin.h
 
-	SOURCES += \
-#$$CODECS_CPP/qasmocodec.cpp \
-		$$CODECS_CPP/qbig5codec.cpp \
+	SOURCES += $$CODECS_CPP/qbig5codec.cpp \
 		  $$CODECS_CPP/qeucjpcodec.cpp \
 		  $$CODECS_CPP/qeuckrcodec.cpp \
 		  $$CODECS_CPP/qisciicodec.cpp \
--- qt-x11-free-3.3.8b/src/moc/mocOLD.pro	2020-10-29 22:33:06.020796639 -0400
+++ qt-x11-free-3.3.8b/src/moc/moc.pro	2020-11-02 00:21:36.932907785 -0400
@@ -2,14 +2,13 @@
 TARGET		= moc
 
 CONFIG 		= console release qtinc yacc lex_included yacc_no_name_mangle
-DEFINES	       += QT_MOC QT_NO_CODECS QT_LITE_UNICODE QT_NO_COMPONENT \
-		  QT_NO_STL QT_NO_COMPRESS
+DEFINES	       += QT_MOC QT_LITE_UNICODE QT_NO_COMPONENT QT_NO_CODECS QT_NO_STL
 win32:DEFINES  += QT_NODLL
 DESTDIR         = ../../bin
 
 
-INCLUDEPATH	+= $$QT_SOURCE_TREE/include ../tools .
-DEPENDPATH	+= $$QT_SOURCE_TREE/include ../tools .
+INCLUDEPATH	+= $$QT_SOURCE_TREE/include ../tools ../codecs .
+DEPENDPATH	+= $$QT_SOURCE_TREE/include ../tools ../codecs .
 LIBS		=
 OBJECTS_DIR	= .
 SOURCES		= ../tools/qbuffer.cpp	    \
@@ -49,6 +48,7 @@
 win32:SOURCES	+= ../tools/qfile_win.cpp ../tools/qdir_win.cpp ../tools/qfileinfo_win.cpp
 macx:LIBS	+= -framework Carbon
 
+unix:LIBS += -lSM -lz -ljpeg
 target.path=$$bins.path
 INSTALLS += target
 
--- qt-x11-free-3.3.8b/src/tools/qt_toolsOLD.pri	2020-11-02 00:49:07.272074529 -0400
+++ qt-x11-free-3.3.8b/src/tools/qt_tools.pri	2020-11-02 00:50:04.238800133 -0400
@@ -141,6 +141,8 @@
 		  $$TOOLS_CPP/qucom.cpp \
 		  $$TOOLS_CPP/quuid.cpp
 
+unix:LIBS += -lSM -lz -ljpeg
+
         irix-cc* {
                 CXXFLAGS_PRELINK = $$QMAKE_CXXFLAGS
                 CXXFLAGS_PRELINK -= -O2
--- qt-x11-free-3.3.8b/tools/designer/uic/mainOLD.cpp	2020-11-02 23:13:16.967184468 -0400
+++ qt-x11-free-3.3.8b/tools/designer/uic/main.cpp	2020-11-02 23:13:48.181029137 -0400
@@ -47,8 +47,8 @@
 #include <stdlib.h>
 
 // see ### in widgetdatabase.cpp
-extern bool dbnounload;
-extern QStringList *dbpaths;
+extern bool dbnounload = FALSE;
+extern QStringList *dbpaths = NULL;
 
 int main( int argc, char * argv[] )
 {
--- qt-x11-free-3.3.8b/tools/designer/shared/widgetdatabaseOLD.cpp	2020-11-02 23:33:12.979144530 -0400
+++ qt-x11-free-3.3.8b/tools/designer/shared/widgetdatabase.cpp	2020-11-02 23:33:54.141927953 -0400
@@ -932,7 +932,7 @@
 bool dbnounload = FALSE;
 QStringList *dbpaths = 0;
 #else
-extern QString *qwf_plugin_dir;
+extern QString *qwf_plugin_dir = NULL;
 #endif
 
 
--- qt-x11-free-3.3.8b/tools/designer/designer/pixmapchooserOLD.cpp	2020-11-03 22:09:12.170003080 -0400
+++ qt-x11-free-3.3.8b/tools/designer/designer/pixmapchooser.cpp	2020-11-03 22:09:24.066941806 -0400
@@ -52,9 +52,7 @@
 #include <qpushbutton.h>
 #include <qiconview.h>
 
-#if defined(DESIGNER)
 static ImageIconProvider *imageIconProvider = 0;
-#endif
 
 PixmapView::PixmapView( QWidget *parent )
     : QScrollView( parent )
--- qt-x11-free-3.3.8b/tools/designer/designerOLD.pro	2020-11-03 22:48:28.959956675 -0400
+++ qt-x11-free-3.3.8b/tools/designer/designer.pro	2020-11-03 22:49:05.518770327 -0400
@@ -4,8 +4,7 @@
 
 SUBDIRS	=  uic \
 	   uilib \
-	   designer \
-	   app
+	   designer
 
 dll:SUBDIRS *=  editor plugins
 shared:SUBDIRS *=  editor plugins
--- qt-x11-free-3.3.8b/examples/examplesOLD.pro	2020-11-05 15:42:58.971842318 -0400
+++ qt-x11-free-3.3.8b/examples/examples.pro	2020-11-05 16:45:04.374433916 -0400
@@ -43,7 +43,6 @@
 		qfd \
 		qmag \
 		qtl \
-		qwerty \
 		rangecontrols \
 		richtext \
 		rot13 \
@@ -63,61 +62,61 @@
 		wizard \
 		xform
 
-thread:SUBDIRS +=   thread/prodcons \
-		    thread/semaphores
-
-canvas:SUBDIRS +=   canvas
-canvas:table:SUBDIRS += chart
-
-opengl:SUBDIRS +=   opengl/box \
-		    opengl/gear \
-		    opengl/glpixmap \
-		    opengl/overlay \
-		    opengl/sharedbox \
-		    opengl/texture
-
-x11:nas:SUBDIRS += 	    sound
-mac|win32:SUBDIRS +=	    sound
-
-iconview:SUBDIRS += fileiconview \
-		    iconview \
-		    iconview/simple_dd
-
-
-network:SUBDIRS +=  network/archivesearch \
-		    network/clientserver/client \
-		    network/clientserver/server \
-		    network/ftpclient \
-		    network/httpd \
-		    network/infoprotocol/infoclient \
-		    network/infoprotocol/infoserver \
-		    network/infoprotocol/infourlclient \
-		    network/mail \
-		    network/networkprotocol \
-		    network/remotecontrol
-
-workspace:SUBDIRS+= mdi
-
-table:SUBDIRS +=    table/statistics \
-		    table/small-table-demo \
-		    regexptester \
-		    table/bigtable \
-			helpsystem
-
-tablet:SUBDIRS += tablet
-
-sql:SUBDIRS += sql
-
-xml:SUBDIRS +=	xml/outliner \
-		xml/tagreader \
-		xml/tagreader-with-features
-
-embedded:SUBDIRS += launcher
-
-embedded:SUBDIRS -= showimg
-
-win32:SUBDIRS += trayicon
-embedded:SUBDIRS += trayicon
-
-x11:SUBDIRS	+= biff \
-		   desktop
+#thread:SUBDIRS +=   thread/prodcons \
+#		    thread/semaphores
+#
+#canvas:SUBDIRS +=   canvas
+#canvas:table:SUBDIRS += chart
+#
+#opengl:SUBDIRS +=   opengl/box \
+#		    opengl/gear \
+#		    opengl/glpixmap \
+#		    opengl/overlay \
+#		    opengl/sharedbox \
+#		    opengl/texture
+#
+#x11:nas:SUBDIRS += 	    sound
+#mac|win32:SUBDIRS +=	    sound
+#
+#iconview:SUBDIRS += fileiconview \
+#		    iconview \
+#		    iconview/simple_dd
+#
+#
+#network:SUBDIRS +=  network/archivesearch \
+#		    network/clientserver/client \
+#		    network/clientserver/server \
+#		    network/ftpclient \
+#		    network/httpd \
+#		    network/infoprotocol/infoclient \
+#		    network/infoprotocol/infoserver \
+#		    network/infoprotocol/infourlclient \
+#		    network/mail \
+#		    network/networkprotocol \
+#		    network/remotecontrol
+#
+#workspace:SUBDIRS+= mdi
+#
+#table:SUBDIRS +=    table/statistics \
+#		    table/small-table-demo \
+#		    regexptester \
+#		    table/bigtable \
+#			helpsystem
+#
+#tablet:SUBDIRS += tablet
+#
+#sql:SUBDIRS += sql
+#
+#xml:SUBDIRS +=	xml/outliner \
+#		xml/tagreader \
+#		xml/tagreader-with-features
+#
+#embedded:SUBDIRS += launcher
+#
+#embedded:SUBDIRS -= showimg
+#
+#win32:SUBDIRS += trayicon
+#embedded:SUBDIRS += trayicon
+#
+#x11:SUBDIRS	+= biff \
+#		   desktop
