--- qt4/configure.old	2018-09-10 20:04:39.190753162 +0200
+++ qt4/configure	2018-09-10 20:04:20.906777432 +0200
@@ -7731,20 +7731,6 @@ case "$XPLATFORM" in
 	canBuildWebKit="no"
 	canBuildQtXmlPatterns="no"
 	;;
-    *-g++*)
-	# Check gcc's version
-	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    5*|4*|3.4*)
-		;;
-            3.3*)
-                canBuildWebKit="no"
-                ;;
-	    *)
-		canBuildWebKit="no"
-		canBuildQtXmlPatterns="no"
-		;;
-	esac
-	;;
     solaris-cc*)
         # Check the compiler version
         case `${QMAKE_CONF_COMPILER} -V 2>&1 | awk '{print $4}'` in
