diff -Naur qtwebkit/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp qtwebkit.new/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp
--- qtwebkit/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp	2017-10-23 17:25:58.941518553 +0200
+++ qtwebkit.new/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp	2017-10-23 17:27:44.223917497 +0200
@@ -961,6 +961,8 @@
     FrameView* view = frame->view();
     ASSERT(view);
     view->resize(size);
+    if (view->needsLayout())
+        view->layout();
     view->adjustViewSize();
 }
 
