diff --git a/declarative/declarative.pro b/declarative/declarative.pro
index 230c877..8198ea6 100644
--- a/declarative/declarative.pro
+++ b/declarative/declarative.pro
@@ -24,7 +24,7 @@ SOURCES += \
 HEADERS += \
     mprisplugin.h
 
-target.path = $$[QT_INSTALL_QML]/$$PLUGIN_IMPORT_PATH
+target.path = $${PREFIX}/lib/qml/$$PLUGIN_IMPORT_PATH
 
 qml.files = qmldir plugins.qmltypes
 qml.path = $$target.path
diff --git a/example/controller/controller.pro b/example/controller/controller.pro
index 6ab85a8..6a89cd3 100644
--- a/example/controller/controller.pro
+++ b/example/controller/controller.pro
@@ -13,6 +13,6 @@ LIBS += -L../../src -l$${MPRISQTLIB}
 SOURCES += \
     controller.cpp
 
-target.path = /usr/bin
+target.path = $${PREFIX}/bin
 
 INSTALLS += target
diff --git a/example/player/player.pro b/example/player/player.pro
index 0076b0c..b569a96 100644
--- a/example/player/player.pro
+++ b/example/player/player.pro
@@ -13,6 +13,6 @@ LIBS += -L../../src -l$${MPRISQTLIB}
 SOURCES += \
     player.cpp
 
-target.path = /usr/bin
+target.path = $${PREFIX}/bin
 
 INSTALLS += target
diff --git a/src/src.pro b/src/src.pro
index 75fafca..43a6f96 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -57,11 +57,11 @@ INSTALL_HEADERS = \
 OTHER_FILES += org.mpris.MediaPlayer2.xml \
     org.mpris.MediaPlayer2.Player.xml
 
-target.path = $$[QT_INSTALL_LIBS]
+target.path = $$PREFIX/lib
 headers.files = $$INSTALL_HEADERS
-headers.path = $$[QT_INSTALL_HEADERS]/MprisQt
+headers.path = $${PREFIX}/include/MprisQt
 prf.files = $${TARGET}.prf
-prf.path = $$[QMAKE_MKSPECS]/features
+prf.path = $$PREFIX/share/qt5/mkspecs/features/
 INSTALLS += target headers prf
 
 QMAKE_PKGCONFIG_REQUIRES = Qt5Core Qt5DBus
