diff -Naur qtspell-0.8.5.orig/src/Codetable.cpp qtspell-0.8.5/src/Codetable.cpp
--- qtspell-0.8.5.orig/src/Codetable.cpp	2018-06-15 13:20:29.000000000 +0200
+++ qtspell-0.8.5/src/Codetable.cpp	2019-01-10 22:38:02.506181015 +0100
@@ -77,7 +77,7 @@
 		QString name = xml.attributes().value("name").toString();
 		QString code = xml.attributes().value("iso_639_1_code").toString();
 		if(!name.isEmpty() && !code.isEmpty()){
-			name = QString::fromUtf8(dgettext(ISO_639_DOMAIN, name.toLatin1().data()));
+			name = QString::fromUtf8(dgettext(ISO_639_DOMAIN, name.toUtf8().constData()));
 			table.insert(code, name);
 		}
 	}
@@ -89,7 +89,7 @@
 		QString name = xml.attributes().value("name").toString();
 		QString code = xml.attributes().value("alpha_2_code").toString();
 		if(!name.isEmpty() && !code.isEmpty()){
-			name = QString::fromUtf8(dgettext(ISO_3166_DOMAIN, name.toLatin1().data()));;
+			name = QString::fromUtf8(dgettext(ISO_3166_DOMAIN, name.toUtf8().constData()));
 			table.insert(code, name);
 		}
 	}
