diff -ur quazip-0.8.1.ori/CMakeLists.txt quazip-0.8.1/CMakeLists.txt
--- quazip-0.8.1.ori/CMakeLists.txt	2019-05-27 06:27:44.000000000 +0200
+++ quazip-0.8.1/CMakeLists.txt	2020-03-09 13:49:52.958631572 +0100
@@ -60,4 +60,4 @@
 
 add_subdirectory(quazip)
 
-install(FILES FindQuaZip.cmake RENAME FindQuaZip${QUAZIP_LIB_VERSION_SUFFIX}.cmake DESTINATION share/cmake/Modules)
+#install(FILES FindQuaZip.cmake RENAME FindQuaZip${QUAZIP_LIB_VERSION_SUFFIX}.cmake DESTINATION share/cmake/Modules)
