diff -Naur quazip-0.7.6.orig/CMakeLists.txt quazip-0.7.6/CMakeLists.txt
--- quazip-0.7.6.orig/CMakeLists.txt	2018-06-13 17:51:21.000000000 +0200
+++ quazip-0.7.6/CMakeLists.txt	2019-01-01 01:02:10.244032810 +0100
@@ -60,4 +60,4 @@
 
 add_subdirectory(quazip)
 
-install(FILES FindQuaZip.cmake RENAME FindQuaZip${QUAZIP_LIB_VERSION_SUFFIX}.cmake DESTINATION ${CMAKE_ROOT}/Modules)
+install(FILES FindQuaZip.cmake RENAME FindQuaZip${QUAZIP_LIB_VERSION_SUFFIX}.cmake DESTINATION "${CMAKE_INSTALL_PREFIX}/share/cmake/Modules")
