diff --git a/Qv2ray.pro b/Qv2ray.pro
index 98ec27e..c251834 100644
--- a/Qv2ray.pro
+++ b/Qv2ray.pro
@@ -232,24 +232,24 @@ unix {
     message("Configuring for unix-like (macOS and linux) environment")
     # For gRPC and protobuf in linux and macOS
     message("  --> Linking against gRPC and protobuf library.")
-    LIBS += -L/usr/local/lib -lgrpc++ -lprotobuf -lgrpc -lgpr
+    LIBS += -L/app/lib -lgrpc++ -lprotobuf -lgrpc -lgpr
 
     # macOS homebrew include path
     message("  --> Adding local include folder to search path")
-    INCLUDEPATH += /usr/local/include/
+    INCLUDEPATH += /app/include/
 
     message("  --> Adding Plasma Toolbox CPP files.")
     SOURCES += src/ui/NetSpeedBar/QvNetSpeedBar_linux.cpp
 
     message("  --> Generating desktop dependency.")
     desktop.files += ./icons/qv2ray.desktop
-    desktop.path = /usr/share/applications/
+    desktop.path = /app/share/applications/
 
     message("  --> Generating icons dependency.")
     icon.files += ./icons/qv2ray.png
-    icon.path = /usr/share/icons/hicolor/256x256/apps/
+    icon.path = /app/share/icons/hicolor/256x256/apps/
 
-    target.path = /usr/local/bin/
+    target.path = /app/bin/
     INSTALLS += target desktop icon
 }
 
