From 003253babb00989159d858bbb9434a3fda0a7131 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Wed, 18 Dec 2019 08:57:46 -0300
Subject: [PATCH] fix-pango

---
 ReadEtextsActivity.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ReadEtextsActivity.py b/ReadEtextsActivity.py
index 0265307..afd3377 100644
--- a/ReadEtextsActivity.py
+++ b/ReadEtextsActivity.py
@@ -905,7 +905,7 @@ class ReadEtextsActivity(SugarCompatibleActivity):
         label_text = label_text + '\n\n\n'
         textbuffer = self.textview.get_buffer()
         tag = textbuffer.create_tag()
-        tag.set_property('weight', Pango.WEIGHT_BOLD)
+        tag.set_property('weight', Pango.Weight.BOLD)
         tag.set_property( 'foreground', "white")
         tag.set_property( 'background', "black")
         textbuffer.set_text(label_text)
-- 
2.23.0

