From fbc10d66f9bc13e96eef3bbd1562f1fd1e9fe446 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Wed, 18 Dec 2019 08:43:04 -0300
Subject: [PATCH 2/2] fix-print

---
 ReadEtextsActivity.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ReadEtextsActivity.py b/ReadEtextsActivity.py
index 85ede66..0265307 100644
--- a/ReadEtextsActivity.py
+++ b/ReadEtextsActivity.py
@@ -886,7 +886,6 @@ class ReadEtextsActivity(SugarCompatibleActivity):
             word_tuple = self.word_tuples[i]
             marked_up_text = marked_up_text + '<mark name="' + str(i) + '"/>' + word_tuple[2]
             i = i + 1
-        print marked_up_text
         return marked_up_text + '</speak>'
 
     def show_found_page(self, page_tuple):
-- 
2.23.0

