From 83847fbe3761dc07d2aa89b8363b3b69d930d9b5 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Tue, 17 Dec 2019 20:45:07 -0300
Subject: [PATCH 3/3] monitors

---
 readtoolbar.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/readtoolbar.py b/readtoolbar.py
index fd0165a..ad6ce2f 100644
--- a/readtoolbar.py
+++ b/readtoolbar.py
@@ -34,6 +34,8 @@ from sugar3.activity import widgets
 from sugar3.graphics.toggletoolbutton import ToggleToolButton
 from speech import SpeechManager
 
+from sugarapp.helpers import PrimaryMonitor
+
 
 class ReadToolbar(Gtk.Toolbar):
     __gtype_name__ = 'ReadToolbar'
@@ -258,7 +260,7 @@ class EditToolbar(widgets.EditToolbar):
         self.search_entry = Gtk.Entry()
         self.search_entry.connect('activate', self.search_entry_activate_cb)
 
-        width = int(Gdk.Screen.width() / 3)
+        width = int(PrimaryMonitor.width() / 3)
         self.search_entry.set_size_request(width, -1)
 
         self.search_entry.props.sensitive = False
@@ -318,7 +320,7 @@ class BooksToolbar(Gtk.Toolbar):
         self.search_entry.connect('activate', self.search_entry_activate_cb)
         self.search_entry.connect("key_press_event", self.keypress_cb)
 
-        width = int(Gdk.Screen.width() / 2)
+        width = int(PrimaryMonitor.width() / 2)
         self.search_entry.set_size_request(width, -1)
 
         book_search_item.add(self.search_entry)
-- 
2.23.0

