From 504f26b07caccec909b2bfbb2d03512d2dccbc60 Mon Sep 17 00:00:00 2001
Message-Id: <504f26b07caccec909b2bfbb2d03512d2dccbc60.1671534716.git.matthias@mailaender.name>
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Tue, 20 Dec 2022 12:11:50 +0100
Subject: [PATCH] Add releases.

---
 share/metainfo/ballz.appdata.xml.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/share/metainfo/ballz.appdata.xml.in b/share/metainfo/ballz.appdata.xml.in
index d846054..4c5d598 100644
--- a/share/metainfo/ballz.appdata.xml.in
+++ b/share/metainfo/ballz.appdata.xml.in
@@ -43,4 +43,11 @@ acronym for 'TINS is not Speedhack'.
     </screenshot>
   </screenshots>
  <translation type="gettext">ballz</translation>
+ <releases>
+    <release version="1.0.4" date="2019-02-17" type="stable"/>
+    <release version="1.0.3" date="2015-07-27" type="stable"/>
+    <release version="1.0.2" date="2011-05-28" type="stable"/>
+    <release version="1.0.1" date="2008-10-16" type="stable"/>
+    <release version="1.0" date="2007-04-05" type="stable"/>
+ </releases>
 </component>
-- 
2.39.0

