--- remid/src/CMakeLists.txt	2019-04-20 02:01:52.000000000 -0400
+++ remid/src/CMakeLists.txt-new	2020-06-27 23:26:07.337216997 -0400
@@ -88,8 +88,8 @@
 
 install(TARGETS remid.lv2
   LIBRARY
-  DESTINATION lib/lv2/remid.lv2
+  DESTINATION lv2/remid.lv2
 )
 install (FILES manifest.ttl remid.ttl
-  DESTINATION lib/lv2/remid.lv2
+  DESTINATION lv2/remid.lv2
 )
--- remid/CMakeLists.txt	2019-04-20 02:01:52.000000000 -0400
+++ remid/CMakeLists.txt-new	2020-06-27 23:34:28.339694386 -0400
@@ -4,5 +4,5 @@
 add_subdirectory(lib/) 
 add_subdirectory(src/) 
 install (DIRECTORY instruments/
-    DESTINATION lib/lv2/remid.lv2
+    DESTINATION lv2/remid.lv2
 )
