diff --git a/Makefile.am b/Makefile.am
index 603cf8e16..2a62986e7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,8 +3,7 @@ SUBDIRS = data docs po src
 
 desktopdir = $(datarootdir)/applications
 
-desktop_DATA = manaplus.desktop \
-	     manaplustest.desktop
+desktop_DATA = manaplus.desktop
 
 appdatadir = $(datarootdir)/appdata
 
@@ -19,7 +18,6 @@ EXTRA_DIST = $(desktop_DATA) \
 	     CMake/Modules/FindLibIntl.cmake \
 	     CMake/Modules/FindSDL_gfx.cmake \
 	     Android.mk \
-	     manaplus.menu \
 	     manaplus.spec \
 	     AUTHORS \
 	     ChangeLog \
