diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-04-01 02:22:44.000000000 +0700
+++ b/CMakeLists.txt	2021-04-26 04:20:02.026649622 +0700
@@ -23,11 +23,11 @@
 
 # ignition-transport currently has no options that are unique to it
 
-if (UNIX AND NOT APPLE)
-  set (EXTRA_TEST_LIB_DEPS stdc++fs)
-else()
-  set (EXTRA_TEST_LIB_DEPS)
-endif()
+#if (UNIX AND NOT APPLE)
+  #set (EXTRA_TEST_LIB_DEPS stdc++fs)
+#else()
+  #set (EXTRA_TEST_LIB_DEPS)
+#endif()
 
 #============================================================================
 # Search for project-specific dependencies
