diff --git a/config/install-app-deps.js b/config/install-app-deps.js
index 01ac0e55..0b1c8f35 100644
--- a/config/install-app-deps.js
+++ b/config/install-app-deps.js
@@ -1,7 +1,6 @@
 const fs = require('fs');
 const jetpack = require('fs-jetpack');
 const path = require('path');
-const installOrRebuild = require('electron-builder-lib/out/util/yarn').installOrRebuild;
 const printErrorAndExit = require('builder-util/out/promise').printErrorAndExit;
 
 const root = process.cwd();
@@ -25,7 +24,6 @@ const options = {
   arch
 }
 
-installOrRebuild(rootPackage.build, appDir, options, true).catch(printErrorAndExit);
 
 //jetpack.remove(path.join(appDir, 'src'));
 
