--- a/Makefile.in	2020-04-25 00:45:26.380451913 +0300
+++ b/Makefile.in	2020-04-25 01:21:56.071519046 +0300
@@ -130,17 +130,10 @@
 
 install-html:
 	$(INSTALL) -d $(DESTDIR)$(docdir)/html
-	for d in $(srcdir)/doc/html/*.html; do \
-		$(INSTALL_DATA) $$d $(DESTDIR)$(docdir)/html; \
-	done
 
 install-man:
 	$(INSTALL) -d $(DESTDIR)$(man3dir)
 	$(INSTALL) -d $(DESTDIR)$(man1dir)
-	for m in $(srcdir)/doc/man/*.3; do \
-	 $(INSTALL_DATA) $$m $(DESTDIR)$(man3dir); done
-	for m in $(srcdir)/doc/man/*.1; do \
-	 $(INSTALL_DATA) $$m $(DESTDIR)$(man1dir); done
 
 install: install-@ALLOW_INSTALL@
 
@@ -182,10 +175,6 @@
 	@echo NLS not enabled.
 
 install-nls-yes:
-	@for f in $(LINGUAS); do \
-	 $(INSTALL) -d $(DESTDIR)$(localedir)/$$f/LC_MESSAGES; \
-	 $(INSTALL_DATA) $(srcdir)/po/$$f.gmo $(DESTDIR)$(localedir)/$$f/LC_MESSAGES/neon.mo; \
-	done
 
 ChangeLog:
 	svn log > $@
