diff -ruN a/setup.py b/setup.py
--- a/setup.py	2020-01-27 00:39:13.000000000 +0700
+++ b/setup.py	2021-08-03 16:33:46.015793024 +0700
@@ -171,8 +171,7 @@
       entry_points={'console_scripts':
                     ['pythran = pythran.run:run',
                      'pythran-config = pythran.config:run']},
-      setup_requires=["pytest-runner"],
-      tests_require=['pytest', 'flake8'],
+      tests_require=['flake8'],
       extras_require={
           'doc': open('docs/requirements.txt').read().splitlines(),
       },
