diff --git a/setup.py b/setup.py
index f9739ab..a79d3b3 100644
--- a/setup.py
+++ b/setup.py
@@ -71,8 +71,8 @@ with open(os.path.join(SHARE, 'torbrowser-launcher/version')) as buf:
 
 datafiles = []
 for root, dirs, files in os.walk(SHARE):
-    datafiles.append((os.path.join(sys.prefix, root),
-                      [os.path.join(root, f) for f in files]))
+    if files:
+        datafiles.append((root, [os.path.join(root, f) for f in files]))
 
 # disable shipping apparmor profiles until they work in ubuntu (#128)
 if distro != 'Ubuntu':
@@ -88,7 +88,7 @@ if distro != 'Ubuntu':
             ('/etc/apparmor.d/tunables/', ['apparmor/tunables/torbrowser'])
         ]
 
-datafiles += [('/usr/share/locale/', create_mo_files())]
+datafiles += [(os.path.dirname(f), [f]) for f in create_mo_files()]
 
 setup(
