diff --git a/Quake/common.c b/Quake/common.c
index 70e77b39..fa582660 100644
--- a/Quake/common.c
+++ b/Quake/common.c
@@ -1436,38 +1436,7 @@ being registered.
 */
 static void COM_CheckRegistered (void)
 {
-	int		h;
-	unsigned short	check[128];
 	int		i;
-
-	COM_OpenFile("gfx/pop.lmp", &h, NULL);
-
-	if (h == -1)
-	{
-		Cvar_SetROM ("registered", "0");
-		Con_Printf ("Playing shareware version.\n");
-		if (com_modified)
-			Sys_Error ("You must have the registered version to use modified games.\n\n"
-				   "Basedir is: %s\n\n"
-				   "Check that this has an " GAMENAME " subdirectory containing pak0.pak and pak1.pak, "
-				   "or use the -basedir command-line option to specify another directory.",
-				   com_basedirs[0]);
-		return;
-	}
-
-	i = Sys_FileRead (h, check, sizeof(check));
-	COM_CloseFile (h);
-	if (i != (int) sizeof(check))
-		goto corrupt;
-
-	for (i = 0; i < 128; i++)
-	{
-		if (pop[i] != (unsigned short)BigShort (check[i]))
-		{ corrupt:
-			Sys_Error ("Corrupted data file.");
-		}
-	}
-
 	for (i = 0; com_cmdline[i]; i++)
 	{
 		if (com_cmdline[i]!= ' ')
@@ -1476,7 +1445,6 @@ static void COM_CheckRegistered (void)
 
 	Cvar_SetROM ("cmdline", &com_cmdline[i]);
 	Cvar_SetROM ("registered", "1");
-	Con_Printf ("Playing registered version.\n");
 }
 
 
