From 924897e0538465615948d0549d7796e00144b970 Mon Sep 17 00:00:00 2001
Message-Id: <924897e0538465615948d0549d7796e00144b970.1664793065.git.matthias@mailaender.name>
From: =?UTF-8?q?Matthias=20Mail=C3=A4nder?= <matthias@mailaender.name>
Date: Mon, 3 Oct 2022 12:30:59 +0200
Subject: [PATCH] Rename the icon.

---
 cmake/Platform/Linux.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Platform/Linux.cmake b/cmake/Platform/Linux.cmake
index 1ba82470f..2e41b48da 100644
--- a/cmake/Platform/Linux.cmake
+++ b/cmake/Platform/Linux.cmake
@@ -119,7 +119,7 @@ function(BF_PLATFORM_INSTALL targetName)
 	# Install desktop files
 	install(FILES ${LINUX_PKG_RESOURCE_DIR}/bitfighter.desktop DESTINATION ${CMAKE_DESKTOP_DATA_PATH}/applications/)
 	install(FILES ${LINUX_PKG_RESOURCE_DIR}/bitfighter.png DESTINATION ${CMAKE_DESKTOP_DATA_PATH}/icons/hicolor/48x48/apps/)
-	install(FILES ${LINUX_PKG_RESOURCE_DIR}/bitfighter128.png DESTINATION ${CMAKE_DESKTOP_DATA_PATH}/icons/hicolor/128x128/apps/)
+	install(FILES ${LINUX_PKG_RESOURCE_DIR}/bitfighter128.png DESTINATION ${CMAKE_DESKTOP_DATA_PATH}/icons/hicolor/128x128/apps/ RENAME bitfighter.png)
 	install(FILES ${LINUX_PKG_RESOURCE_DIR}/org.bitfighter.Bitfighter.metainfo.xml DESTINATION ${CMAKE_DESKTOP_DATA_PATH}/metainfo/)
 	
 	# Manpage
-- 
2.37.3

