Author: Unrud <unrud@outlook.com>
Description: don't rely on hostname resolution for localhost
Last-Update: 2019-05-10
--- a/src/EngineTest.py
+++ b/src/EngineTest.py
@@ -33,7 +33,7 @@
     e2 = Engine()
     
     e1.startServer()
-    e2.connect("localhost")
+    e2.connect("127.0.0.1")
 
     while not e2.isConnected():
       e1.run()
--- a/src/GameEngineTest.py
+++ b/src/GameEngineTest.py
@@ -28,8 +28,8 @@
     e1 = GameEngine()
     
     e1.startServer()
-    session1 = e1.connect("localhost")
-    session2 = e1.connect("localhost")
+    session1 = e1.connect("127.0.0.1")
+    session2 = e1.connect("127.0.0.1")
 
     while not session1.isConnected() or not session2.isConnected():
       e1.run()
@@ -44,7 +44,7 @@
     assert len(session1.world.players) == 2
     assert len(session2.world.players) == 2
     
-    session3 = e1.connect("localhost")
+    session3 = e1.connect("127.0.0.1")
     
     for i in range(10):
       e1.run()
--- a/src/Network.py
+++ b/src/Network.py
@@ -176,7 +176,7 @@
     asyncore.dispatcher.__init__(self)
     self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
     self.set_reuse_addr()
-    self.bind((localOnly and "localhost" or "", port))
+    self.bind((localOnly and "127.0.0.1" or "", port))
     self.listen(5)
     self.clients = {}
     self.__idCounter = 0
--- a/src/NetworkTest.py
+++ b/src/NetworkTest.py
@@ -36,7 +36,7 @@
   def testHandshake(self):
     s = TestServer()
     c = TestConnection()
-    c.connect("localhost")
+    c.connect("127.0.0.1")
 
     c.sendPacket("moikka")
 
