diff --git a/setup.py b/setup.py
index 51083f9..2f622ea 100755
--- a/setup.py
+++ b/setup.py
@@ -149,11 +149,11 @@ DESCRIPTION = 'Persepolis Download Manager'
 
 if os_type == 'Linux':
     DATA_FILES = [
-        ('/usr/share/man/man1/', ['man/persepolis.1.gz']),
-        ('/usr/share/applications/', ['xdg/com.github.persepolisdm.persepolis.desktop']),
-        ('/usr/share/metainfo/', ['xdg/com.github.persepolisdm.persepolis.appdata.xml']),
-        ('/usr/share/icons/hicolor/scalable/apps/', ['resources/persepolis.svg']),
-        ('/usr/share/icons/hicolor/scalable/apps/', ['resources/persepolis-tray.svg'])
+        ('/app/share/man/man1/', ['man/persepolis.1.gz']),
+        ('/app/share/applications/', ['xdg/com.github.persepolisdm.persepolis.desktop']),
+        ('/app/share/metainfo/', ['xdg/com.github.persepolisdm.persepolis.appdata.xml']),
+        ('/app/share/icons/hicolor/scalable/apps/', ['resources/persepolis.svg']),
+        ('/app/share/icons/hicolor/scalable/apps/', ['resources/persepolis-tray.svg'])
     ]
 elif os_type == 'FreeBSD' or os_type == 'OpenBSD':
     DATA_FILES = [
