diff -Naur old/ReText/tab.py new/ReText/tab.py
--- old/ReText/tab.py	2018-07-01 16:42:50.378219313 +0200
+++ new/ReText/tab.py	2018-07-01 16:44:14.036135426 +0200
@@ -120,6 +120,7 @@
 			rect = doc.documentLayout().blockBoundingRect(block)
 			return rect.top()
 
+		return ReTextPreview(self)
 		if ReTextWebKitPreview and globalSettings.useWebKit:
 			preview = ReTextWebKitPreview(self,
 			                              editorPositionToSourceLine,
diff -Naur old/ReText/window.py new/ReText/window.py
--- old/ReText/window.py	2018-07-01 16:42:50.409218912 +0200
+++ new/ReText/window.py	2018-07-01 16:44:14.028135529 +0200
@@ -301,9 +301,9 @@
 		menuFormat.addAction(self.actionItalic)
 		menuFormat.addAction(self.actionUnderline)
 		if ReTextWebKitPreview is not None or ReTextWebEnginePreview is None:
-			menuEdit.addAction(self.actionWebKit)
+			pass #menuEdit.addAction(self.actionWebKit)
 		else:
-			menuEdit.addAction(self.actionWebEngine)
+			pass #menuEdit.addAction(self.actionWebEngine)
 		menuEdit.addSeparator()
 		menuEdit.addAction(self.actionViewHtml)
 		menuEdit.addAction(self.actionPreview)
@@ -562,14 +562,14 @@
 		self.currentTab.triggerPreviewUpdate()
 
 	def enableWebKit(self, enable):
-		globalSettings.useWebKit = enable
+		globalSettings.useWebKit = False
 		globalSettings.useWebEngine = False
 		for tab in self.iterateTabs():
 			tab.rebuildPreviewBox()
 
 	def enableWebEngine(self, enable):
 		globalSettings.useWebKit = False
-		globalSettings.useWebEngine = enable
+		globalSettings.useWebEngine = False
 		for tab in self.iterateTabs():
 			tab.rebuildPreviewBox()
 
