--- libs/rewrite-edn/deps.edn	2022-01-03 11:16:05.972365288 +0900
+++ libs/rewrite-edn/deps.edn.local	2022-01-03 11:18:37.209577706 +0900
@@ -1,5 +1,4 @@
-{:deps {lread/rewrite-cljc {:git/url "https://github.com/lread/rewrite-cljc-playground"
-                            :sha "ba1e4255cd1c1fd09dbe2e90556bfa66d5459234"}}
+{:deps {lread/rewrite-cljc {:local/root "/run/build/logseq/libs/rewrite-cljc"}}
  :aliases {:test
            {:extra-deps
             {org.clojure/clojure {:mvn/version "1.9.0"}
