diff -up Ri-li-2.0.1/Makefile.am~ Ri-li-2.0.1/Makefile.am
--- Ri-li-2.0.1/Makefile.am~	2007-11-07 08:55:39.000000000 +0100
+++ Ri-li-2.0.1/Makefile.am	2007-11-07 08:55:39.000000000 +0100
@@ -1,3 +1,3 @@
-SUBDIRS=src data Sounds gentoo
+SUBDIRS=src data Sounds
 datdir = $(datadir)/@PACKAGE@
 dat_HEADERS = COPYING.Music
diff -up Ri-li-2.0.1/Makefile.in~ Ri-li-2.0.1/Makefile.in
--- Ri-li-2.0.1/Makefile.in~	2007-11-07 08:55:45.000000000 +0100
+++ Ri-li-2.0.1/Makefile.in	2007-11-07 08:55:45.000000000 +0100
@@ -156,7 +156,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = src data Sounds gentoo
+SUBDIRS = src data Sounds
 datdir = $(datadir)/@PACKAGE@
 dat_HEADERS = COPYING.Music
 all: config.h
