diff -up Ri-li-2.0.1/src/jeux.h.gcc43 Ri-li-2.0.1/src/jeux.h
diff -up Ri-li-2.0.1/src/editeur.h.gcc43 Ri-li-2.0.1/src/editeur.h
diff -up Ri-li-2.0.1/src/ecran.h.gcc43 Ri-li-2.0.1/src/ecran.h
diff -up Ri-li-2.0.1/src/preference.h.gcc43 Ri-li-2.0.1/src/preference.h
diff -up Ri-li-2.0.1/src/tableau.cc.gcc43 Ri-li-2.0.1/src/tableau.cc
--- Ri-li-2.0.1/src/tableau.cc.gcc43	2007-11-02 12:48:21.000000000 +0100
+++ Ri-li-2.0.1/src/tableau.cc	2008-01-11 21:33:43.000000000 +0100
@@ -21,7 +21,8 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include "preference.h"
diff -up Ri-li-2.0.1/src/jeux.cc.gcc43 Ri-li-2.0.1/src/jeux.cc
--- Ri-li-2.0.1/src/jeux.cc.gcc43	2007-11-02 12:48:17.000000000 +0100
+++ Ri-li-2.0.1/src/jeux.cc	2008-01-11 21:33:43.000000000 +0100
@@ -25,7 +25,8 @@
 #include <windows.h>
 #endif
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -up Ri-li-2.0.1/src/mouse.cc.gcc43 Ri-li-2.0.1/src/mouse.cc
--- Ri-li-2.0.1/src/mouse.cc.gcc43	2007-11-02 12:48:20.000000000 +0100
+++ Ri-li-2.0.1/src/mouse.cc	2008-01-11 21:33:43.000000000 +0100
@@ -21,7 +21,8 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include "mouse.h"
 #include "preference.h"
diff -up Ri-li-2.0.1/src/sprite.h.gcc43 Ri-li-2.0.1/src/sprite.h
diff -up Ri-li-2.0.1/src/audio.h.gcc43 Ri-li-2.0.1/src/audio.h
diff -up Ri-li-2.0.1/src/menu.cc.gcc43 Ri-li-2.0.1/src/menu.cc
--- Ri-li-2.0.1/src/menu.cc.gcc43	2007-11-02 12:48:19.000000000 +0100
+++ Ri-li-2.0.1/src/menu.cc	2008-01-11 21:33:43.000000000 +0100
@@ -21,7 +21,8 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -up Ri-li-2.0.1/src/editeur.cc.gcc43 Ri-li-2.0.1/src/editeur.cc
--- Ri-li-2.0.1/src/editeur.cc.gcc43	2007-11-02 12:48:17.000000000 +0100
+++ Ri-li-2.0.1/src/editeur.cc	2008-01-11 21:33:43.000000000 +0100
@@ -25,7 +25,8 @@
 #include <windows.h>
 #endif
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -up Ri-li-2.0.1/src/mouse.h.gcc43 Ri-li-2.0.1/src/mouse.h
diff -up Ri-li-2.0.1/src/main.cc.gcc43 Ri-li-2.0.1/src/main.cc
--- Ri-li-2.0.1/src/main.cc.gcc43	2007-11-02 12:48:19.000000000 +0100
+++ Ri-li-2.0.1/src/main.cc	2008-01-11 21:33:43.000000000 +0100
@@ -23,7 +23,8 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <string.h>
 #include <SDL/SDL.h>
 #include <SDL/SDL_mixer.h>
diff -up Ri-li-2.0.1/src/sprite.cc.gcc43 Ri-li-2.0.1/src/sprite.cc
--- Ri-li-2.0.1/src/sprite.cc.gcc43	2007-11-02 12:48:20.000000000 +0100
+++ Ri-li-2.0.1/src/sprite.cc	2008-01-11 21:33:43.000000000 +0100
@@ -21,7 +21,8 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -up Ri-li-2.0.1/src/loco.cc.gcc43 Ri-li-2.0.1/src/loco.cc
--- Ri-li-2.0.1/src/loco.cc.gcc43	2007-11-02 12:48:18.000000000 +0100
+++ Ri-li-2.0.1/src/loco.cc	2008-01-11 21:33:43.000000000 +0100
@@ -21,7 +21,8 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -up Ri-li-2.0.1/src/tableau.h.gcc43 Ri-li-2.0.1/src/tableau.h
diff -up Ri-li-2.0.1/src/menu.h.gcc43 Ri-li-2.0.1/src/menu.h
diff -up Ri-li-2.0.1/src/utils.h.gcc43 Ri-li-2.0.1/src/utils.h
diff -up Ri-li-2.0.1/src/ecran.cc.gcc43 Ri-li-2.0.1/src/ecran.cc
--- Ri-li-2.0.1/src/ecran.cc.gcc43	2007-11-02 12:48:16.000000000 +0100
+++ Ri-li-2.0.1/src/ecran.cc	2008-01-11 21:33:43.000000000 +0100
@@ -21,7 +21,8 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
diff -up Ri-li-2.0.1/src/loco.h.gcc43 Ri-li-2.0.1/src/loco.h
diff -up Ri-li-2.0.1/src/audio.cc.gcc43 Ri-li-2.0.1/src/audio.cc
--- Ri-li-2.0.1/src/audio.cc.gcc43	2007-11-02 12:48:15.000000000 +0100
+++ Ri-li-2.0.1/src/audio.cc	2008-01-11 21:33:43.000000000 +0100
@@ -22,7 +22,8 @@
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <string.h>
 
 #include "audio.h"
diff -up Ri-li-2.0.1/src/utils.cc.gcc43 Ri-li-2.0.1/src/utils.cc
--- Ri-li-2.0.1/src/utils.cc.gcc43	2007-11-02 12:48:22.000000000 +0100
+++ Ri-li-2.0.1/src/utils.cc	2008-01-11 21:33:43.000000000 +0100
@@ -21,7 +21,8 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
