diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-04-18 01:10:17.023165680 +0200
+++ b/CMakeLists.txt	2022-04-18 17:51:00.376094695 +0200
@@ -106,7 +106,7 @@
         set(OBABEL_IFACE_NAME "obabeliface.dll")
 endif(WIN32 OR WIN64)
 
-add_definitions(-DOBABEL_IFACE_LOCATION="\\\"${MSK_INSTALL_LIBS}/${OBABEL_IFACE_NAME}\\\"")
+add_definitions(-DOBABEL_IFACE_LOCATION="${MSK_INSTALL_LIBS}/${OBABEL_IFACE_NAME}")
 
 set(CMAKE_INSTALL_PREFIX ${MSK_PREFIX})
 
Binary files a/.git/index and b/.git/index differ
