--- rosegarden/src/gui/widgets/Panner.cpp-orig	2020-09-02 23:19:25.324148026 -0400
+++ rosegarden/src/gui/widgets/Panner.cpp	2020-09-02 23:19:43.528373822 -0400
@@ -24,6 +24,7 @@
 #include "misc/Debug.h"
 #include "base/Profiler.h"
 
+#include <QPainterPath>
 #include <QPolygon>
 #include <QMouseEvent>
 
--- rosegarden/src/gui/general/ThornStyle.cpp	2020-06-03 22:06:46.000000000 -0400
+++ rosegarden/src/gui/general/ThornStyle.cpp-new	2020-09-02 23:29:22.343561125 -0400
@@ -45,6 +45,7 @@
 #include <QSpinBox>
 #include <QScrollBar>
 #include <QAbstractScrollArea>
+#include <QPainterPath>
 
 using namespace Rosegarden;
 
