--- rust-module.cjs
+++ rust-module.cjs
@@ -166,7 +166,13 @@
   case 'linux':
     switch (arch) {
       case 'x64':
-        if (isMusl()) {
+          if (process.env.NAPI_RS_NATIVE_LIBRARY_PATH !== undefined) {
+            try {
+              nativeBinding = require(process.env.NAPI_RS_NATIVE_LIBRARY_PATH);
+            } catch (e) {
+              loadError = e;
+            }
+          } else if (isMusl()) {
           localFileExisted = existsSync(
             join(__dirname, 'index.linux-x64-musl.node')
           )
@@ -195,7 +201,13 @@
         }
         break
       case 'arm64':
-        if (isMusl()) {
+        if (process.env.NAPI_RS_NATIVE_LIBRARY_PATH !== undefined) {
+          try {
+            nativeBinding = require(process.env.NAPI_RS_NATIVE_LIBRARY_PATH);
+          } catch (e) {
+            loadError = e;
+          }
+        } else if (isMusl()) {
           localFileExisted = existsSync(
             join(__dirname, 'index.linux-arm64-musl.node')
           )
