diff --git a/CMakeLists.txt b/CMakeLists.txt
index 524e5f4..3fe907a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,8 @@ target_link_libraries(${PROJECT_NAME} ${OS_LINK_LIBRARY})
 #SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE C)
 option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
 ADD_LIBRARY (libryzenadj ${OS_SOURCE} ${COMMON_SOURCES})
+set_target_properties(libryzenadj PROPERTIES PUBLIC_HEADER lib/ryzenadj.h)
 set_target_properties(libryzenadj PROPERTIES PREFIX "")
 target_link_libraries(libryzenadj ${OS_LINK_LIBRARY})
 #SET_TARGET_PROPERTIES(libryzenadj PROPERTIES LINKER_LANGUAGE C)
-install(TARGETS ${PROJECT_NAME} DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS libryzenadj PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
diff --git a/lib/api.c b/lib/api.c
index a0757c8..af29b08 100644
--- a/lib/api.c
+++ b/lib/api.c
@@ -423,7 +423,7 @@ do {                                                 \
 do {                                                \
 	if(!ry->table_values)                           \
 		return NAN;                                 \
-	return ry->table_values[OFFSET / 4];            \
+	return ry->table_values[(OFFSET) / 4];            \
 } while (0);
 
 
diff --git a/lib/ryzenadj.h b/lib/ryzenadj.h
index c2d1dce..9fedeae 100644
--- a/lib/ryzenadj.h
+++ b/lib/ryzenadj.h
@@ -32,6 +32,8 @@ enum ryzen_family {
         FAM_END
 };
 
+#include <stdint.h>
+
 #ifdef _LIBRYZENADJ_INTERNAL
 #include  "ryzenadj_priv.h"
 
