diff -up sane-backends-1.0.28/backend/genesys_gl841.cc.genesys-regression sane-backends-1.0.28/backend/genesys_gl841.cc
--- sane-backends-1.0.28/backend/genesys_gl841.cc.genesys-regression	2019-10-14 13:11:10.772763713 +0200
+++ sane-backends-1.0.28/backend/genesys_gl841.cc	2019-10-14 13:11:57.602389188 +0200
@@ -5042,6 +5042,9 @@ gl841_init (Genesys_Device * dev)
   /* Set analog frontend */
   RIE (gl841_set_fe(dev, sensor, AFE_INIT));
 
+    // FIXME: slow_back_home modifies dev->calib_reg and requires it to be filled
+    dev->calib_reg = dev->reg;
+
   /* Move home */
   RIE (gl841_slow_back_home (dev, SANE_TRUE));
 
