diff --git a/backend/xerox_mfp.c b/backend/xerox_mfp.c
index b7fcbee..2cb73ee 100644
--- a/backend/xerox_mfp.c
+++ b/backend/xerox_mfp.c
@@ -209,6 +209,7 @@ static int isSupportedDevice(struct device __sane_unused__ *dev)
     if (dev->compressionTypes & (1 << 6)) {
 	/* blacklist malfunctioning device(s) */
 	if (!strncmp(dev->sane.model, "SCX-4500W", 9) ||
+            !strncmp(dev->sane.model, "C460", 4) ||
 	    !strncmp(dev->sane.model, "M288x", 5))
 	    return 0;
         return 1;
