diff --git a/backend/genesys.cc b/backend/genesys.cc
index 0368e21..5ec37bc 100644
--- a/backend/genesys.cc
+++ b/backend/genesys.cc
@@ -778,6 +778,12 @@ void sanei_genesys_create_default_gamma_table(Genesys_Device* dev,
             size = 4096;
         }
         max = size - 1;
+    } else if (dev->model->asic_type == GENESYS_GL846
+               || dev->model->asic_type == GENESYS_GL847
+               || dev->model->asic_type == GENESYS_GL848
+               || dev->model->asic_type == GENESYS_GL124) {
+        size = 257;
+        max = 65535;
     } else {
         size = 256;
         max = 65535;
