diff -Naur sane-backends-1.0.28.orig/configure.ac sane-backends-1.0.28/configure.ac
--- sane-backends-1.0.28.orig/configure.ac	2019-07-31 14:43:52.000000000 +0200
+++ sane-backends-1.0.28/configure.ac	2019-08-12 12:09:32.148900684 +0200
@@ -5,7 +5,7 @@
 dnl  When preparing a release, modify the numeric version components
 dnl  and remove the git suffix.
 dnl ******************************************************************
-AC_INIT([sane-backends],m4_esyscmd_s([git describe --dirty]),
+AC_INIT([sane-backends],[1.0.28],
         [sane-devel@alioth-lists.debian.net])
 AC_PREREQ([2.69])               dnl minimum autoconf version required
 AC_CONFIG_MACRO_DIR([m4])
