diff --git a/client/src/game/gamesettings.cpp b/client/src/game/gamesettings.cpp
index becf9cb..a30e585 100644
--- a/client/src/game/gamesettings.cpp
+++ b/client/src/game/gamesettings.cpp
@@ -1,8 +1,11 @@
 #include "gamesettings.h"
+#include <QStandardPaths>
 
 #include <cstdint>
 
-#define SETTINGS_FILE "data/game.ini"
+QString settingsPath = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/dynablaster/game.ini";
+
+#define SETTINGS_FILE settingsPath
 
 // defaults
 
