Subject: [PATCH] Script patch for FlatPak
---
Index: package/linux/common/jarmemu
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package/linux/common/jarmemu b/package/linux/common/jarmemu
--- a/package/linux/common/jarmemu	(revision af123155c5ea7658bfa875fc95e532b31c410857)
+++ b/package/linux/common/jarmemu	(date 1700476837251)
@@ -23,4 +23,4 @@
 #     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #

-java -Dprism.dirtyopts=false -jar /usr/share/java/JArmEmu/JArmEmu.jar "$@"
\ No newline at end of file
+java -Dprism.dirtyopts=false -jar /app/share/java/JArmEmu/JArmEmu.jar "$@"
\ No newline at end of file
