--- a/Libs/Client.Native/SDL2.cs
+++ b/Libs/Client.Native/SDL2.cs
@@ -39,7 +39,7 @@ namespace SDL2
 	{
 		#region SDL2# Variables
 
-		private const string nativeLibName = "SDL2";
+		private const string nativeLibName = "SDL2-2.0.so.0";
 
 		#endregion
 
--- a/Libs/Client.Native/SDL2_image.cs
+++ b/Libs/Client.Native/SDL2_image.cs
@@ -38,7 +38,7 @@ namespace SDL2
 		#region SDL2# Variables
 
 		/* Used by DllImport to load the native library. */
-		private const string nativeLibName = "SDL2_image";
+		private const string nativeLibName = "SDL2_image-2.0.so.0";
 
 		#endregion
 
--- a/Libs/Client.Native/SDL2_mixer.cs
+++ b/Libs/Client.Native/SDL2_mixer.cs
@@ -38,7 +38,7 @@ namespace SDL2
 		#region SDL2# Variables
 
 		/* Used by DllImport to load the native library. */
-		private const string nativeLibName = "SDL2_mixer";
+		private const string nativeLibName = "SDL2_mixer-2.0.so.0";
 
 		#endregion
 
--- a/Libs/Client.Native/SDL2_ttf.cs
+++ b/Libs/Client.Native/SDL2_ttf.cs
@@ -38,7 +38,7 @@ namespace SDL2
 		#region SDL2# Variables
 
 		/* Used by DllImport to load the native library. */
-		private const string nativeLibName = "SDL2_ttf";
+		private const string nativeLibName = "SDL2_ttf-2.0.so.0";
 
 		#endregion
 
