Author: Alberto Garcia <berto@igalia.com>
Subject: Use SDL2 for audio and joystick (SDL2 UI is unsupported)
diff --git a/configure.ac b/configure.ac
index 2716b04..26fb81a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -277,9 +277,6 @@ if test -z "$UI"; then
   fi
 fi
 
-dnl Look for SDL (default=no)
-SDL_VERSION=1.2.4
-AM_PATH_SDL($SDL_VERSION, sdl_available=yes, sdl_available=no)
 if test -z "$UI"; then
   AC_MSG_CHECKING(whether SDL UI requested)
   AC_ARG_WITH(sdl,
@@ -287,6 +284,16 @@ if test -z "$UI"; then
   if test "$withval" = no; then sdl=no; else sdl=yes; fi,
   sdl=no)
   AC_MSG_RESULT($sdl)
+fi
+
+dnl Look for SDL (default=no)
+SDL_VERSION=1.2.4
+if test "$sdl" = yes; then
+  AM_PATH_SDL($SDL_VERSION, sdl_available=yes, sdl_available=no)
+else
+  PKG_CHECK_MODULES([SDL], [sdl2], [sdl_available=yes], [sdl_available=no])
+fi
+
   if test "$sdl" = yes; then
     if test "$sdl_available" = yes; then
       AC_DEFINE([UI_SDL], 1, [Defined if the SDL UI in use])
@@ -298,7 +305,6 @@ if test -z "$UI"; then
       AC_MSG_ERROR([SDL version $SDL_VERSION not found])
     fi
   fi
-fi
 
 dnl Look for null UI (default=no)
 if test -z "$UI"; then
diff --git a/ui/sdl/sdljoystick.c b/ui/sdl/sdljoystick.c
index ed37c83..94c5680 100644
--- a/ui/sdl/sdljoystick.c
+++ b/ui/sdl/sdljoystick.c
@@ -53,12 +53,11 @@ ui_joystick_init( void )
 {
   int error, retval;
 
-#ifdef UI_SDL
+#ifndef UI_SDL
+  SDL_Init(0);
+#endif
+
   error = SDL_InitSubSystem( SDL_INIT_JOYSTICK );
-#else
-  /* Other UIs could handle joysticks by the SDL library */
-  error = SDL_Init(SDL_INIT_JOYSTICK|SDL_INIT_VIDEO);
-#endif
 
   if ( error ) {
     ui_error( UI_ERROR_ERROR, "failed to initialise joystick subsystem" );
