diff --git a/b_overdrive/Makefile b/b_overdrive/Makefile
index a4679d7..22b334f 100644
--- a/b_overdrive/Makefile
+++ b/b_overdrive/Makefile
@@ -10,6 +10,7 @@ LV2NAME=b_overdrive
 
 ifeq ($(LV2AVAIL), yes)
   targets+=lv2
+  override CFLAGS+=`pkg-config --cflags lv2`
 endif
 
 ifeq ($(shell pkg-config --exists liblo && echo yes), yes)
diff --git a/common.mak b/common.mak
index 1c12f32..db0ac70 100644
--- a/common.mak
+++ b/common.mak
@@ -20,7 +20,7 @@ LV2VERSION=$(VERSION)
 
 bindir = $(PREFIX)/bin
 sharedir = $(PREFIX)/share/setBfree
-lv2dir = $(PREFIX)/lib/lv2
+lv2dir = $(PREFIX)/lv2
 
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 ifeq ($(XWIN),)
