--- setup.py	2018-12-19 21:57:01.000000000 +0000
+++ setup_fix.py	2019-04-28 00:56:05.672188401 +0100
@@ -113,9 +113,9 @@
                 'Nagstamon.thirdparty.Xlib.xobject'],
       package_dir={'Nagstamon': 'Nagstamon'},
       package_data={'Nagstamon': ['resources/*']},
-      data_files=[('%s/share/man/man1' % sys.prefix, ['Nagstamon/resources/nagstamon.1.gz']),
-          ('%s/share/pixmaps' % sys.prefix, ['Nagstamon/resources/nagstamon.svg']),
-          ('%s/share/applications' % sys.prefix, ['Nagstamon/resources/nagstamon.desktop'])],
+      data_files=[('%s/share/man/man1' % '/app', ['Nagstamon/resources/nagstamon.1.gz']),
+          ('%s/share/pixmaps' % '/app', ['Nagstamon/resources/nagstamon.svg']),
+          ('%s/share/applications' % '/app', ['Nagstamon/resources/nagstamon.desktop'])],
       options=dict(build_exe=build_exe_options,
           bdist_mac=bdist_mac_options,
           bdist_dmg=bdist_dmg_options,
