--- sfizz-1/cmake/LV2Config.cmake	2020-04-01 20:49:02.445501300 -0400
+++ sfizz-1/cmake/LV2Config.cmake-new	2020-06-27 00:22:22.927998225 -0400
@@ -18,6 +18,6 @@
     set (LV2PLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lv2" CACHE STRING
     "Install destination for LV2 bundle [default: ${CMAKE_INSTALL_PREFIX}/lv2}]")
 else()
-    set (LV2PLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib/lv2" CACHE STRING
-    "Install destination for LV2 bundle [default: ${CMAKE_INSTALL_PREFIX}/lib/lv2}]")
+    set (LV2PLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lv2" CACHE STRING
+    "Install destination for LV2 bundle [default: ${CMAKE_INSTALL_PREFIX}/lv2}]")
 endif()
--- sfizz/cmake/VSTConfig.cmake	2020-04-02 15:03:26.885996000 -0400
+++ sfizz/cmake/VSTConfig.cmake-new	2020-07-06 09:54:36.838519334 -0400
@@ -7,8 +7,8 @@
     set (VSTPLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/vst3" CACHE STRING
     "Install destination for VST bundle [default: ${CMAKE_INSTALL_PREFIX}/vst3}]")
 else()
-    set (VSTPLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib/vst3" CACHE STRING
-    "Install destination for VST bundle [default: ${CMAKE_INSTALL_PREFIX}/lib/vst3}]")
+    set (VSTPLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/vst3" CACHE STRING
+    "Install destination for VST bundle [default: ${CMAKE_INSTALL_PREFIX}/vst3}]")
 endif()
 
 if (NOT VST3_SYSTEM_PROCESSOR)
