diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6e57c5..66089e7 100644
--- a/neo/extern/ShaderMake/CMakeLists.txt
+++ b/neo/extern/ShaderMake/CMakeLists.txt
@@ -92,7 +92,7 @@ if (MSVC)
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
     target_link_libraries (ShaderMake pthread)
 else ()
-    target_link_libraries (ShaderMake stdc++fs pthread)
+    target_link_libraries (ShaderMake pthread)
 endif ()
 
 if (SHADERMAKE_SEARCH_FOR_COMPILERS)
