diff --git a/tools/shaderCompiler/CMakeLists.txt b/tools/shaderCompiler/CMakeLists.txt
index 1de06e0..0d19e05 100644
--- a/neo/extern/nvrhi/tools/shaderCompiler/CMakeLists.txt
+++ b/neo/extern/nvrhi/tools/shaderCompiler/CMakeLists.txt
@@ -41,7 +41,7 @@ if(MSVC)
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
 	target_link_libraries(shaderCompiler pthread)
 else()
-	target_link_libraries(shaderCompiler stdc++fs pthread)
+	target_link_libraries(shaderCompiler pthread)
 endif()
 
 set_target_properties(shaderCompiler PROPERTIES OUTPUT_NAME "nvrhi-scomp")
