diff --git i/shelm w/shelm
index 147c0ea..8fa98ed 100755
--- i/shelm
+++ w/shelm
@@ -122,7 +122,7 @@ fetch() {
 
 # List dependencies in the local package cache, in the form $author/$project/$version.
 list_dependencies() {
-	cd "$pkgdir" && find . -type d -depth 3 | sed 's|^./||'
+	cd "$pkgdir" && find . -mindepth 3 -maxdepth 3 -type d | sed 's|^./||'
 }
 
 # Prune dependencies from the local package cache that don't match the required
