diff --git a/tests/libsample/simplefile.cpp b/tests/libsample/simplefile.cpp
index deac166..a99cec9 100644
--- a/tests/libsample/simplefile.cpp
+++ b/tests/libsample/simplefile.cpp
@@ -90,13 +90,13 @@ bool
 SimpleFile::exists() const
 {
     std::ifstream ifile(p->m_filename);
-    return ifile;
+    return (ifile.is_open());
 }
 
 bool
 SimpleFile::exists(const char* filename)
 {
     std::ifstream ifile(filename);
-    return ifile;
+    return (ifile.is_open());
 }
 
