diff --git a/misc/shotwell.desktop.in b/misc/shotwell.desktop.in
index aa1ede2..2800ebf 100644
--- a/misc/shotwell.desktop.in
+++ b/misc/shotwell.desktop.in
@@ -5,7 +5,7 @@ GenericName=Photo Manager
 Comment=Organize your photos
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 Keywords=album;camera;cameras;crop;edit;enhance;export;gallery;image;images;import;organize;photo;photographs;photos;picture;pictures;photography;print;publish;rotate;share;tags;video;facebook;flickr;picasa;youtube;piwigo;
-Exec=shotwell %U
+Exec=shotwell-wrapper %U
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
 Icon=shotwell
 Terminal=false
