diff -up sip-4.19.3/configure.py.sip_bin sip-4.19.3/configure.py
--- sip-4.19.3/configure.py.sip_bin	2017-07-03 12:24:27.000000000 -0500
+++ sip-4.19.3/configure.py	2017-07-05 20:21:24.376350683 -0500
@@ -281,7 +281,7 @@ def create_config(module, template, macr
         "sip_version":      sip_version,
         "sip_version_str":  sip_version_str,
         "platform":         build_platform,
-        "sip_bin":          os.path.join(sip_bin_dir, "sip"),
+        "sip_bin":          os.path.join(sip_bin_dir, "python3-sip"),
         "sip_inc_dir":      sip_inc_dir,
         "sip_mod_dir":      sip_module_dir,
         "default_bin_dir":  plat_bin_dir,
