diff -up sip-4.19.9.dev1805261119/siplib/siplib.c.in.rex sip-4.19.9.dev1805261119/siplib/siplib.c.in
--- sip-4.19.9.dev1805261119/siplib/siplib.c.in.rex	2018-05-28 20:30:19.000000000 -0500
+++ sip-4.19.9.dev1805261119/siplib/siplib.c.in	2018-05-29 11:23:00.868228044 -0500
@@ -6323,6 +6323,7 @@ static PyObject *createContainerType(sip
     if (py_type == NULL)
         goto relargs;
 
+#if PY_VERSION_HEX >= 0x03030000
     /* Fix __qualname__ if there is a scope. */
     if (scope_td != NULL)
     {
@@ -6337,6 +6338,7 @@ static PyObject *createContainerType(sip
         Py_CLEAR(ht->ht_qualname);
         ht->ht_qualname = qualname;
     }
+#endif
 
     /* Add the type to the "parent" dictionary. */
     if (PyDict_SetItem(scope_dict, name, py_type) < 0)
