diff --git a/SlideruleActivity.py b/SlideruleActivity.py
index c167d04..54b50fd 100644
--- a/SlideruleActivity.py
+++ b/SlideruleActivity.py

@@ -477,7 +476,8 @@ class SlideruleActivity(activity.Activity):
                                             page=edit_toolbar,
                                             icon_name='toolbar-edit')
         edit_toolbar_button.show()
-        toolbox.toolbar.insert(edit_toolbar_button, -1)
+        # Hide copy paste for now
+        # toolbox.toolbar.insert(edit_toolbar_button, -1)
         edit_toolbar_button.show()
 
         self.set_toolbar_box(toolbox)
@@ -590,6 +590,7 @@ class SlideruleActivity(activity.Activity):
         toolbox.toolbar.insert(stop_button, -1)
         stop_button.show()
         # workaround to #2050
-        edit_toolbar_button.set_expanded(True)
+        # Hide copy paste for now
+        # edit_toolbar_button.set_expanded(True)
         # start with project toolbar enabled
         project_toolbar_button.set_expanded(True)
