diff --git a/window.py b/window.py
index 69cace4..b009af7 100644
--- a/window.py
+++ b/window.py
@@ -33,6 +33,7 @@ graphics associated with your slide and stator.
 # import pygtk
 # pygtk.require('2.0')
 import gi
+from sugarapp.helpers import PrimaryMonitor
 
 gi.require_version('Gtk', '3.0')
 gi.require_version('PangoCairo', '1.0')
@@ -113,8 +114,8 @@ class SlideRule():
         self.canvas.connect("key-press-event", self._keypress_cb)
         self.canvas.set_can_focus(True)
         self.canvas.grab_focus()
-        self.width = Gdk.Screen.width()
-        self.height = Gdk.Screen.height() - GRID_CELL_SIZE
+        self.width = PrimaryMonitor.width()
+        self.height = PrimaryMonitor.height() - GRID_CELL_SIZE
         self.sprites = Sprites(self.canvas)
         self.slides = []
         self.stators = []
