diff --git a/toolbars.py b/toolbars.py
index 30bb4d0..e5b9def 100644
--- a/toolbars.py
+++ b/toolbars.py
@@ -43,6 +43,8 @@ from sugar3.graphics.toolbarbox import ToolbarBox as SugarToolbarBox
 from sugar3.activity.widgets import StopButton
 from sugar3.activity.widgets import ActivityToolbarButton
 
+from sugarapp.helpers import PrimaryMonitor
+
 
 class ToolbarInfo(Gtk.Toolbar):
 
@@ -261,10 +263,10 @@ class HelpButton(Gtk.ToolItem):
         self._palette = help_button.get_palette()
 
         sw = Gtk.ScrolledWindow()
-        sw.set_size_request(int(Gdk.Screen.width() / 2.8), 310)
+        sw.set_size_request(int(PrimaryMonitor.width() / 2.8), 310)
         sw.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
 
-        self._max_text_width = int(Gdk.Screen.width() / 3) - 600
+        self._max_text_width = int(PrimaryMonitor.width() / 3) - 600
         self._vbox = Gtk.Box()
         self._vbox.set_orientation(Gtk.Orientation.VERTICAL)
         self._vbox.set_homogeneous(False)
