diff --git a/activity/activity.info b/activity/activity.info
index 7668f91..241f8b3 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,8 +1,18 @@
 [Activity]
 name = Solar System
 activity_version = 1
+release_date = 2020-04-22
 bundle_id = org.sugarlabs.SolarSystem
 icon = icon
 exec = sugar-activity3 activity.SolarSystem
-license = LGPLv3+
+license = LGPL-3.0-or-later
+metadata_license = CC0-1.0
 max_participants = 1
+summary = Learn about the planets
+description = The Solar System activity is a tool to encourage children to learn more about the planets and their moons (natural satellites). It is the author's hope that this tool serves as a practical and interactve way to explore astronomy.
+url = https://github.com/sugarlabs/solar-system
+repository = https://github.com/sugarlabs/solar-system
+tags = Education
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://i.imgur.com/IoEJQ0Y.png
