diff --git a/activity.py b/activity.py
index 6460e16..7517a4e 100644
--- a/activity.py
+++ b/activity.py
@@ -31,11 +31,13 @@ from gi.repository import Gtk
 from gi.repository import Gdk
 from sugar3.activity import activity
 
+from sugarapp.widgets import SugarCompatibleActivity
 
-class SolarSystem(activity.Activity):
+
+class SolarSystem(SugarCompatibleActivity):
 
     def __init__(self, handle):
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self.screen = None
 
