diff --git a/bin/soundconverter b/bin/soundconverter
index ebc3acd..1dab318 100755
--- a/bin/soundconverter
+++ b/bin/soundconverter
@@ -44,6 +44,9 @@ elif SOURCE_PATH.startswith('/usr/local/'):
     DATA_PATH = '/usr/local/share/soundconverter'
 elif SOURCE_PATH.startswith('/usr/'):
     DATA_PATH = '/usr/share/soundconverter'
+elif SOURCE_PATH.startswith('/app/'):
+    # flatpak support
+    DATA_PATH = '/app/share/soundconverter'
 else:
     # installed with -e, running from the cloned git source
     DATA_PATH = os.path.join(SOURCE_PATH, 'data')
