From 84b6173e2f582902aca71690c88439221b18c092 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Wed, 27 Nov 2019 19:50:26 -0300
Subject: [PATCH 4/6] info

---
 activity/activity.info | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index 19f9abd..1cbeac1 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,14 +2,17 @@
 name = Speak
 summary = An animated face that speaks whatever you type
 description = Speak is a talking face.  Anything you type will be spoken aloud using a speech synthesizer.  You can adjust the accent, rate and pitch of the voice as well as the shape of the eyes and mouth.  This is a great way to experiment with a speech synthesizer, learn to type or just have fun making a funny face.
-homepage = http://wiki.sugarlabs.org/go/Activities/Speak
+url = https://help.sugarlabs.org/en/speak.html
 license = GPLv3+
+metadata_license = CC0-1.0
 categories = language
-stability = stable
 icon = activity-speak
 exec = sugar-activity activity.SpeakActivity
-requires = gst-plugins-espeak; numpy
 activity_version = 57
-bundle_id = vu.lux.olpc.Speak
-category = language
+release_date = 2018-08-20
+bundle_id = org.sugarlabs.Speak
+tags = Education
 repository = https://github.com/sugarlabs/speak
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/speak/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/speak/master/screenshots/en/2.png
-- 
2.19.1

