From 2113ca9d89e1c105fc9d7786691f5ab442bbb9a8 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Thu, 28 Nov 2019 10:47:47 -0300
Subject: [PATCH 6/6] encoding

---
 face.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/face.py b/face.py
index 8a1aaa3..f9caeee 100644
--- a/face.py
+++ b/face.py
@@ -94,8 +94,9 @@ class Status:
                   3: fft_mouth.FFTMouth}
 
         data = json.loads(buf)
-        self.voice = voice.Voice(data['voice']['language'],
-                                 data['voice']['name'])
+        self.voice = voice.Voice(
+            data['voice']['language'].encode('utf-8', 'ignore'),
+            data['voice']['name'].encode('utf-8', 'ignore'))
         self.pitch = data['pitch']
         self.rate = data['rate']
         self.eyes = [eyes[i] for i in data['eyes']]
-- 
2.19.1

