From 73961d3486d25076e24cb569b995256f076e52e5 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Wed, 27 Nov 2019 18:59:23 -0300
Subject: [PATCH 2/6] greeting

---
 activity.py | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/activity.py b/activity.py
index 87b1d6e..801eb49 100644
--- a/activity.py
+++ b/activity.py
@@ -366,14 +366,8 @@ class SpeakActivity(SugarCompatibleActivity):
             if self._tablet_mode:
                 self._entry.props.text = _('Hello %s.') \
                     % self.owner.props.nick.encode('utf-8', 'ignore')
-            self.face.say_notification(_('Hello %s. Please Type something.')
-                                       % self.owner.props.nick)
-        else:
-            if self._tablet_mode:
-                self._entry.props.text = _('Welcome back %s.') \
-                    % self.owner.props.nick.encode('utf-8', 'ignore')
-            self.face.say_notification(_('Welcome back %s.')
-                                       % self.owner.props.nick)
+        self.face.say_notification(_('Hello %s. Please Type something.')
+                                   % self.owner.props.nick)
         self._set_idle_phrase(speak=False)
         self._first_time = False
 
-- 
2.19.1

