From 874cf662cb3715507924c37f4e768b3df532aed0 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Wed, 27 Nov 2019 18:59:54 -0300
Subject: [PATCH 3/6] i18n

---
 voice.py | 32 +++++++++++++++++++++++---------
 1 file changed, 23 insertions(+), 9 deletions(-)

diff --git a/voice.py b/voice.py
index 79bb10d..6307123 100644
--- a/voice.py
+++ b/voice.py
@@ -73,6 +73,12 @@ _allVoices = {}
 _defaultVoice = None
 
 
+def _friendly(full_name):
+    parts = re.split('[ _-]', full_name)
+    short_name = _(parts[0].capitalize())
+    return ' '.join([short_name] + parts[1:])
+
+
 class Voice:
     def __init__(self, language, name):
         self.language = language
@@ -113,11 +119,15 @@ def allVoices():
         voice = Voice(language, name)
         _allVoices[voice.friendlyname] = voice
 
-    if 'English' not in _allVoices:
-        _allVoices['English'] = _allVoices['English (America)']
+    en_name = _friendly('English')
+    if en_name not in _allVoices:
+        _allVoices[en_name] = _allVoices[_friendly('English (America)')]
+        _allVoices[en_name].friendlyname = en_name
 
-    if 'Spanish' not in _allVoices:
-        _allVoices['Spanish'] = _allVoices['Spanish (Latin America)']
+    es_name = _friendly('Spanish')
+    if es_name not in _allVoices:
+        _allVoices[es_name] = _allVoices[_friendly('Spanish (Latin America)')]
+        _allVoices[es_name].friendlyname = es_name
 
     return _allVoices
 
@@ -152,9 +162,9 @@ def defaultVoice():
         lang = ""
 
     voice_names = [
-        "English (America)",  # espeak-ng 1.49.2
-        "English",  # espeak-ng 1.49.1
-        "Default",  # espeak 1.48
+        _friendly("English (America)"),  # espeak-ng 1.49.2
+        _friendly("English"),  # espeak-ng 1.49.1
+        _friendly("Default"),  # espeak 1.48
     ]
 
     best = None
@@ -163,15 +173,19 @@ def defaultVoice():
             best = voices[voice_name]
             break
 
+    es_name = _friendly('Spanish')
+    es_la_name = _friendly('Spanish (Latin America)')
+    en_au_name = _friendly('English (Received Pronunciation)')
+
     for voice in voices.values():
         voiceMetric = fit(voice.language, lang)
         bestMetric = fit(best.language, lang)
         if lang == 'en_AU.UTF-8':
-            if voice.friendlyname == 'English (Received Pronunciation)':
+            if voice.friendlyname == es_au_name:
                 best = voice
                 break
         if lang[0:2] == 'es':
-            if voice.friendlyname in ['Spanish', 'Spanish (Latin America)']:
+            if voice.friendlyname in [es_name, es_la_name]:
                 best = voice
                 break
         if voiceMetric > bestMetric:
-- 
2.19.1

