diff -Naur heldercorreia-speedcrunch-ea93b21f9498.orig/pkg/desktop-appdata/speedcrunch.appdata.xml.in heldercorreia-speedcrunch-ea93b21f9498/pkg/desktop-appdata/speedcrunch.appdata.xml.in
--- heldercorreia-speedcrunch-ea93b21f9498.orig/pkg/desktop-appdata/speedcrunch.appdata.xml.in	2016-11-29 00:34:56.000000000 +0100
+++ heldercorreia-speedcrunch-ea93b21f9498/pkg/desktop-appdata/speedcrunch.appdata.xml.in	2018-12-04 00:22:08.276984082 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2016 Felix Krull <f_krull@gmx.de> -->
 <component type="desktop">
-  <id>speedcrunch.desktop</id>
+  <id>org.speedcrunch.SpeedCrunch</id>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-2.0+</project_license>
   <update_contact>speedcrunch@googlegroups.com</update_contact>
@@ -28,6 +28,8 @@
     </_p>
   </description>
 
+  <launchable type="desktop-id">org.speedcrunch.SpeedCrunch.desktop</launchable>
+
   <screenshots>
     <screenshot type="default">
       <image>http://speedcrunch.org/_static/appdata-screenshots/0.12/advanced-features.png</image>
@@ -54,6 +56,13 @@
   <url type="bugtracker">https://bitbucket.org/heldercorreia/speedcrunch/issues</url>
   <url type="help">http://speedcrunch.org/introduction.html</url>
 
+  <provides>
+    <binary>speedcrunch</binary>
+  </provides>
+  <releases>
+    <release version="0.12" date="2016-11-28"/>
+  </releases>
+
   <!-- We can't use the <translations> tag since our qm files are baked into the
        executable so we just use a static list instead. See:
        https://blogs.gnome.org/hughsie/2016/01/25/appdata-and-the-gettext-domain/#comment-5137
diff -Naur heldercorreia-speedcrunch-ea93b21f9498.orig/pkg/speedcrunch.appdata.xml heldercorreia-speedcrunch-ea93b21f9498/pkg/speedcrunch.appdata.xml
--- heldercorreia-speedcrunch-ea93b21f9498.orig/pkg/speedcrunch.appdata.xml	2016-11-29 00:34:56.000000000 +0100
+++ heldercorreia-speedcrunch-ea93b21f9498/pkg/speedcrunch.appdata.xml	2018-12-04 00:20:51.029566242 +0100
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
-  <id>speedcrunch.desktop</id>
+  <id>org.speedcrunch.SpeedCrunch</id>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-2.0+</project_license>
   <update_contact>speedcrunch@googlegroups.com</update_contact>
@@ -57,6 +57,8 @@
     <p xml:lang="zh">SpeedCrunch 提供多种语言和平台。</p>
   </description>
 
+  <launchable type="desktop-id">org.speedcrunch.SpeedCrunch.desktop</launchable>
+
   <screenshots>
     <screenshot type="default">
       <image>http://speedcrunch.org/_static/appdata-screenshots/0.12/advanced-features.png</image>
@@ -103,6 +105,13 @@
   <url type="bugtracker">https://bitbucket.org/heldercorreia/speedcrunch/issues</url>
   <url type="help">http://speedcrunch.org/introduction.html</url>
 
+  <provides>
+    <binary>speedcrunch</binary>
+  </provides>
+  <releases>
+    <release version="0.12" date="2016-11-28"/>
+  </releases>
+
   
   <translation/>
   <languages>
@@ -142,4 +151,4 @@
     <lang>vi</lang>
     <lang>en</lang>
   </languages>
-</component>
\ No newline at end of file
+</component>
