diff --git a/src/spek-artwork.cc b/src/spek-artwork.cc
index cd7e78c..322555a 100644
--- a/src/spek-artwork.cc
+++ b/src/spek-artwork.cc
@@ -2,6 +2,7 @@
 #include <wx/iconbndl.h>
 
 #include "spek-artwork.h"
+#include <iostream>
 
 class SpekArtProvider : public wxArtProvider
 {
@@ -16,7 +17,7 @@ wxBitmap SpekArtProvider::CreateBitmap(
 	(void) id, (void) client, (void) size;
 #ifdef OS_UNIX
     if (id == ART_SPEK) {
-        return wxArtProvider::GetBitmap("spek", client, size);
+        return wxArtProvider::GetBitmap("cc.spek.Spek", client, size);
     }
     if (id == ART_HELP) {
         return wxArtProvider::GetBitmap("gtk-help", client, size);
@@ -67,7 +68,7 @@ wxIconBundle SpekArtProvider::CreateIconBundle(const wxArtID& id, const wxArtCli
 	(void) id, (void) client;
 #ifdef OS_UNIX
     if (id == ART_SPEK) {
-        return wxArtProvider::GetIconBundle("spek", client);
+        return wxArtProvider::GetIconBundle("cc.spek.Spek", client);
     }
 #endif
 #ifdef OS_WIN
