--- squoosh_png.js	2025-02-17 11:24:51.956947286 +0100
+++ src/node_modules/.deno/@jsquash+png@3.0.1/node_modules/@jsquash/png/codec/pkg/squoosh_png.js	2025-02-17 11:24:55.890005251 +0100
@@ -202,7 +202,7 @@
     const imports = __wbg_get_imports();

     if (typeof input === 'string' || (typeof Request === 'function' && input instanceof Request) || (typeof URL === 'function' && input instanceof URL)) {
-        input = fetch(input);
+        input = await Deno.readFile(input);
     }

     __wbg_init_memory(imports);
