diff --git i/src/main/java/mediathek/config/StandardLocations.kt w/src/main/java/mediathek/config/StandardLocations.kt
index cda2f5e..72beee7 100644
--- i/src/main/java/mediathek/config/StandardLocations.kt
+++ w/src/main/java/mediathek/config/StandardLocations.kt
@@ -67,10 +67,7 @@ object StandardLocations {
     @Throws(InvalidPathException::class)
     fun getStandardDownloadPath(): String {
         val userHome = SystemUtils.USER_HOME
-        val path = if (SystemUtils.IS_OS_MAC_OSX)
-            Paths.get(userHome, "Downloads")
-        else
-            Paths.get(userHome, Konstanten.VERZEICHNIS_DOWNLOADS)
+        val path = Paths.get(userHome, "Downloads")
         return path.toAbsolutePath().toString()
     }
 
@@ -123,4 +120,4 @@ object StandardLocations {
     fun getLockFilePath(): Path {
         return getSettingsDirectory().resolve(LOCKFILE_NAME)
     }
-}
\ No newline at end of file
+}
