diff --git a/pkg/desktop-appdata/speedcrunch.desktop.in b/pkg/desktop-appdata/speedcrunch.desktop.in
index 7c300674..6301a001 100644
--- a/pkg/desktop-appdata/speedcrunch.desktop.in
+++ b/pkg/desktop-appdata/speedcrunch.desktop.in
@@ -3,6 +3,7 @@ Type=Application
 _Name=SpeedCrunch
 Exec=speedcrunch
 Icon=speedcrunch
+StartupWMClass=speedcrunch
 _Comment=A high-precision scientific calculator
 Categories=Qt;Utility;Calculator;
 # TRANSLATORS: Extra keywords for finding the application in the desktop environment. The trailing ';' is important!
diff --git a/pkg/speedcrunch.desktop b/pkg/speedcrunch.desktop
index daab281b..93eebb75 100644
--- a/pkg/speedcrunch.desktop
+++ b/pkg/speedcrunch.desktop
@@ -12,6 +12,7 @@ Name[ru]=SpeedCrunch
 Name[zh]=SpeedCrunch
 Exec=speedcrunch
 Icon=speedcrunch
+StartupWMClass=speedcrunch
 Comment=A high-precision scientific calculator
 Comment[de]=Ein hochpräziser wissenschaftlicher Taschenrechner
 Comment[el]=Επιστημονική αριθμομηχανή υψηλής ακρίβειας
