--- rayfowler/Rotp.java	2022-02-04 05:41:49.212552595 +0530
+++ Aman9das/Rotp.java	2022-02-04 05:42:17.389056011 +0530
@@ -42,7 +42,7 @@
     public static String jarFileName = "Remnants.jar";
     public static String exeFileName = "Remnants.exe";
     public static boolean countWords = false;
-    private static String startupDir;
+    private static String startupDir = System.getProperty("startupdir");
     private static JFrame frame;
     public static String releaseId = "1.02a";
     public static long startMs = System.currentTimeMillis();
