diff --git a/src/sdl/sdl_display.c b/src/sdl/sdl_display.c
index 12fc8f14dabb..77e6b8e5b6c4 100644
--- a/src/sdl/sdl_display.c
+++ b/src/sdl/sdl_display.c
@@ -205,6 +205,8 @@ static ALLEGRO_DISPLAY *sdl_create_display_locked(int w, int h)
    d->extra_settings.settings[ALLEGRO_OPENGL_MAJOR_VERSION] = (v >> 24) & 0xFF;
    d->extra_settings.settings[ALLEGRO_OPENGL_MINOR_VERSION] = (v >> 16) & 0xFF;
 
+   SDL_StopTextInput();
+
    return d;
 }
 
