From 544b3454ddb6370f29107ec24be8585dfb581790 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Wed, 11 Dec 2019 01:59:48 -0300
Subject: [PATCH] fix-espeak-ng

---
 utils.py | 111 +++++++++++++++++++++++++++++++++++++++++++++++--------
 1 file changed, 95 insertions(+), 16 deletions(-)

diff --git a/utils.py b/utils.py
index d0c4c2a..44e1e3b 100644
--- a/utils.py
+++ b/utils.py
@@ -38,19 +38,97 @@ def json_dump(data):
     _io = StringIO()
     jdump(data, _io)
     return _io.getvalue()
-VOICES = {'af': 'afrikaans', 'cy': 'welsh-test', 'el': 'greek',
-          'es': 'spanish', 'hi': 'hindi-test', 'hy': 'armenian',
-          'ku': 'kurdish', 'mk': 'macedonian-test', 'pt': 'brazil',
-          'sk': 'slovak', 'sw': 'swahili', 'bs': 'bosnian',
-          'da': 'danish', 'en': 'english', 'fi': 'finnish',
-          'hr': 'croatian', 'id': 'indonesian-test', 'la': 'latin',
-          'nl': 'dutch-test', 'sq': 'albanian', 'ta': 'tamil',
-          'vi': 'vietnam-test', 'ca': 'catalan', 'de': 'german',
-          'eo': 'esperanto', 'fr': 'french', 'hu': 'hungarian',
-          'is': 'icelandic-test', 'lv': 'latvian', 'no': 'norwegian',
-          'ro': 'romanian', 'sr': 'serbian', 'zh': 'Mandarin',
-          'cs': 'czech', 'it': 'italian', 'pl': 'polish',
-          'ru': 'russian_test', 'sv': 'swedish', 'tr': 'turkish'}
+VOICES = {
+    'af': 'Afrikaans',
+    'am': 'Amharic',
+    'an': 'Aragonese',
+    'ar': 'Arabic',
+    'as': 'Assamese',
+    'az': 'Azerbaijani',
+    'bg': 'Bulgarian',
+    'bn': 'Bengali',
+    'bpy': 'Bishnupriya_Manipuri',
+    'bs': 'Bosnian',
+    'ca': 'Catalan',
+    'cmn': 'Chinese_(Mandarin)',
+    'cs': 'Czech',
+    'cy': 'Welsh',
+    'da': 'Danish',
+    'de': 'German',
+    'el': 'Greek',
+    'en': 'English_(America)',
+    'eo': 'Esperanto',
+    'es': 'Spanish_(Latin_America)',
+    'et': 'Estonian',
+    'eu': 'Basque',
+    'fa': 'Persian',
+    'fa-Latn': 'Persian_(Pinglish)',
+    'fi': 'Finnish',
+    'fr': 'French_(France)',
+    'ga': 'Gaelic_(Irish)',
+    'gd': 'Gaelic_(Scottish)',
+    'gn': 'Guarani',
+    'grc': 'Greek_(Ancient)',
+    'gu': 'Gujarati',
+    'hi': 'Hindi',
+    'hr': 'Croatian',
+    'hu': 'Hungarian',
+    'hy': 'Armenian_(East_Armenia)',
+    'hy-arevmda': 'Armenian_(West_Armenia)',
+    'ia': 'Interlingua',
+    'id': 'Indonesian',
+    'is': 'Icelandic',
+    'it': 'Italian',
+    'ja': 'Japanese',
+    'jbo': 'Lojban',
+    'ka': 'Georgian',
+    'kl': 'Greenlandic',
+    'kn': 'Kannada',
+    'ko': 'Korean',
+    'kok': 'Konkani',
+    'ku': 'Kurdish',
+    'ky': 'Kyrgyz',
+    'la': 'Latin',
+    'lfn': 'Lingua_Franca_Nova',
+    'lt': 'Lithuanian',
+    'lv': 'Latvian',
+    'mi': 'poz/mi',
+    'mk': 'Macedonian',
+    'ml': 'Malayalam',
+    'mr': 'Marathi',
+    'ms': 'Malay',
+    'mt': 'Maltese',
+    'my': 'Burmese',
+    'nci': 'Nahuatl_(Classical)',
+    'ne': 'Nepali',
+    'nl': 'Dutch',
+    'om': 'Oromo',
+    'or': 'Oriya',
+    'pa': 'Punjabi',
+    'pap': 'Papiamento',
+    'pl': 'Polish',
+    'pt': 'Portuguese_(Brazil)',
+    'ro': 'Romanian',
+    'ru': 'Russian',
+    'sd': 'Sindhi',
+    'si': 'Sinhala',
+    'sk': 'Slovak',
+    'sl': 'Slovenian',
+    'sq': 'Albanian',
+    'sr': 'Serbian',
+    'sv': 'Swedish',
+    'sw': 'Swahili',
+    'ta': 'Tamil',
+    'te': 'Telugu',
+    'tn': 'Setswana',
+    'tr': 'Turkish',
+    'tt': 'Tatar',
+    'ur': 'Urdu',
+    'vi': 'Vietnamese_(Northern)',
+    'vi-vn-x-central': 'Vietnamese_(Central)',
+    'vi-vn-x-south': 'Vietnamese_(Southern)',
+    'yue': 'Chinese_(Cantonese)'
+}
 
 
 def speak(text):
@@ -60,10 +138,11 @@ def speak(text):
         text = int(text)
     safetext = '{}'.format(quote(str(text)))
 
-    lang = os.environ['LANG'][0:2]
+    lang = os.environ['LANG'][0:2].lower()
     if lang in VOICES:
-        command = 'espeak -v %s "%s"' % (VOICES[lang], safetext)
+        command = 'espeak -v %s "%s"' % (lang, safetext)
     else:
         command = 'espeak "%s"' % (safetext)
 
-    os.system('%s --stdout | aplay' % command)
+    os.system('%s --stdout > /tmp/espeak.audio' % command)
+    os.system('gst-play-1.0 /tmp/espeak.audio')
-- 
2.19.1

