--- ./plugins/clipboard-history/ClipboardHistory.vala
+++ ./plugins/clipboard-history/ClipboardHistory.vala
@@ -90,7 +90,7 @@
         
         // Delete last item, if the size of the list store > MAX_SIZE
         if (list_store.get_iter_from_string (out iter, (MAX_SIZE - 1).to_string ()))
-            list_store.remove (iter);
+            list_store.remove (ref iter);
 
         // Delete dupplicates from list store, if exists
         delete_dupplicates (clipboard_content);
@@ -143,7 +143,7 @@
         });
 
         if (to_delete != null)
-            list_store.remove (to_delete);
+            list_store.remove (ref to_delete);
     }
 
     void build_plugin_ui () {
@@ -214,7 +214,7 @@
         if (!selection.get_selected(out model, out iter)) {
             return;
         }
-        list_store.remove (iter);
+        list_store.remove (ref iter);
 
         // Hiding PlugIn, if no more items exist in the list store.
         if (!list_store.get_iter_first (out iter))

