From af8a83279045fa5e675bdc237b7a4fb92ccb3b8d Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Wed, 11 Dec 2019 01:31:39 -0300
Subject: [PATCH 1/2] info

---
 activity/activity.info | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index 1a1cc72..83bdbb7 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,12 +1,18 @@
 [Activity]
 name = Story
 activity_version = 19
+release_date = 2018-08-31
 license = GPLv3+;GPLv2+;MIT
+metadata_license = CC0-1.0
 bundle_id = org.sugarlabs.StoryActivity
 exec = sugar-activity StoryActivity.StoryActivity
 icon = activity-story
 show_launcher = yes
-summary = Prompts learner to tell a story by displaying images at random.
-tags = Language;Game
+summary = Tell a story
+description = Story prompts the learner to tell a story by displaying images at random. The learner should try to tell a story that ties the images together into a comprehensive narrative.
+tags = Education
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
 repository = https://github.com/sugarlabs/story
 url = https://help.sugarlabs.org/en/story.html
+screenshots = https://raw.githubusercontent.com/sugarlabs/story/master/screenshots/en/1.png https://raw.githubusercontent.com/sugarlabs/story/master/screenshots/en/2.png
-- 
2.19.1

