From 36436c0a77e6ef31d0c880ead70d35a608f163fe Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Wed, 11 Dec 2019 01:11:01 -0300
Subject: [PATCH 2/4] fix-autoplay

---
 game.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/game.py b/game.py
index 9736b32..84d19db 100644
--- a/game.py
+++ b/game.py
@@ -365,6 +365,7 @@ class Game():
 
     def _poll_audio(self):
         if self._parent.audio_process is None:  # Already stopped?
+            self._next_image()
             return
 
         if self._parent.audio_process.poll() is None:
-- 
2.19.1

