--- ./subsurface/scripts/build.sh
+++ ./subsurface/scripts/build.sh
@@ -65,7 +65,7 @@
 fi
 
 mkdir -p install-root
-INSTALL_ROOT=$SRC/install-root
+INSTALL_ROOT=/app
 
 # make sure we find our own packages first (e.g., libgit2 only uses pkg_config to find libssh2)
 export PKG_CONFIG_PATH=$INSTALL_ROOT/lib/pkgconfig:$PKG_CONFIG_PATH
@@ -272,7 +272,7 @@
 		-DMARBLE_INCLUDE_DIR=$INSTALL_ROOT/include \
 		-DMARBLE_LIBRARIES=$INSTALL_ROOT/lib/libssrfmarblewidget.$SH_LIB_EXT \
 		-DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH \
-		-DNO_PRINTING=OFF
+		-DNO_PRINTING=ON -DNO_MARBLE=ON -DUSE_WEBENGINE=ON -DCMAKE_INSTALL_PREFIX\:PATH=/app
 
 	if [ $PLATFORM = Darwin ] ; then
 		rm -rf Subsurface.app

@@ -106,7 +106,7 @@
 	fi
 	mkdir -p build
 	cd build
-	cmake -DCMAKE_INSTALL_PREFIX=$INSTALL_ROOT -DCMAKE_BUILD_TYPE=Release -DBUILD_CLAR=OFF ..
+	cmake -DCMAKE_INSTALL_PREFIX=$INSTALL_ROOT -DCMAKE_BUILD_TYPE=Release -DBUILD_CLAR=OFF -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=true ..
 	make -j4
 	make install
 
