From 0727cece25aa95d7152330a0ac8e81cb22b069c7 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Tue, 10 Dec 2019 16:38:41 -0300
Subject: [PATCH] robust-widgets

---
 src/sugar3/activity/widgets.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/sugar3/activity/widgets.py b/src/sugar3/activity/widgets.py
index 57223f4b..0fb42774 100644
--- a/src/sugar3/activity/widgets.py
+++ b/src/sugar3/activity/widgets.py
@@ -336,7 +336,7 @@ class ActivityToolbar(Gtk.Toolbar):
 
         self._activity = activity
 
-        if activity.metadata:
+        if activity.metadata and 'title' in activity.metadata:
             title_button = TitleEntry(activity)
             title_button.connect('enter-key-press',
                                  lambda widget: self.emit('enter-key-press'))
@@ -351,7 +351,7 @@ class ActivityToolbar(Gtk.Toolbar):
             self.insert(separator, -1)
             separator.show()
 
-        if activity.metadata:
+        if activity.metadata and 'description' in activity.metadata:
             description_item = DescriptionItem(activity)
             description_item.show()
             self.insert(description_item, -1)
-- 
2.19.1

