diff --git a/src/sugar3/graphics/icon.py b/src/sugar3/graphics/icon.py
index 342e802a..64fea95e 100644
--- a/src/sugar3/graphics/icon.py
+++ b/src/sugar3/graphics/icon.py
@@ -137,7 +137,7 @@ class _SVGLoader(object):
                 logging.error(
                     'Icon %s, entity %s is invalid.', file_name, entity)
 
-        return Rsvg.Handle.new_from_data(icon.encode('utf-8'))
+        return Rsvg.Handle.new_from_data(icon)
 
 
 class _IconInfo(object):
