--- SuiteSparse-5.6.0/Makefile.orig	2019-11-05 22:37:26.086135974 +0000
+++ SuiteSparse-5.6.0/Makefile	2019-11-05 22:37:37.549632429 +0000
@@ -16,7 +16,7 @@
 # installs all libraries SuiteSparse/lib.
 go: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd Mongoose && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
+	( cd Mongoose && $(MAKE) library CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 	( cd AMD && $(MAKE) )
 	( cd BTF && $(MAKE) )
 	( cd CAMD && $(MAKE) )
