diff --git a/build-linux.sh b/build-linux.sh
index 7a5d133..7aace67 100755
--- a/build-linux.sh
+++ b/build-linux.sh
@@ -327,11 +327,11 @@ if [[ ! -z "$option_local" ]]; then
     headless_dest_path="$HOME/bin"
     data_path="$HOME/.local/share/Surge"
 else
-    vst2_dest_path="/usr/lib/vst"
-    vst3_dest_path="/usr/lib/vst3"
-    lv2_dest_path="/usr/lib/lv2"
-    headless_dest_path="/usr/bin"
-    data_path="/usr/share/Surge"
+    vst2_dest_path="/app/extensions/Lv2Plugins/Surge/vst"
+    vst3_dest_path="/app/extensions/Lv2Plugins/Surge/vst3"
+    lv2_dest_path="/app/extensions/Lv2Plugins/Surge/lv2"
+    headless_dest_path="/app/extensions/Lv2Plugins/Surge/bin"
+    data_path="/app/extensions/Lv2Plugins/Surge/share"
 fi
 
 case $1 in
diff --git a/premake5.lua b/premake5.lua
index 7004e74..a6054ed 100644
--- a/premake5.lua
+++ b/premake5.lua
@@ -307,7 +307,7 @@ function plugincommon()
         buildoptions {
             "-Wno-unused-variable",
             "`pkg-config --cflags cairo fontconfig freetype2 xkbcommon-x11 xcb-cursor xcb-keysyms xcb-xkb xcb-util x11`",
-            "-std=c++14"
+            "-std=c++17"
         }
 
         configuration { "Release*" }
@@ -345,7 +345,6 @@ function plugincommon()
 
         links {
             "pthread",
-            "stdc++fs",
             "gcc_s",
             "gcc",
             "dl",
diff --git a/src/common/SurgeStorage.cpp b/src/common/SurgeStorage.cpp
index 10ed0ad..0e22c1d 100644
--- a/src/common/SurgeStorage.cpp
+++ b/src/common/SurgeStorage.cpp
@@ -191,7 +191,7 @@ SurgeStorage::SurgeStorage(std::string suppliedDataPath)
        */
        if (! fs::is_directory(datapath))
        {
-           datapath = "/usr/share/Surge/";
+           datapath = "/app/extensions/Lv2Plugins/Surge/share/";
        }
    }
    else
diff --git a/src/common/SurgeStorage.h b/src/common/SurgeStorage.h
index 45684f5..fd43daa 100644
--- a/src/common/SurgeStorage.h
+++ b/src/common/SurgeStorage.h
@@ -16,9 +16,7 @@
 #endif
 #include <tinyxml.h>
 
-#if LINUX 
-#include <experimental/filesystem>
-#elif MAC || (WINDOWS && TARGET_RACK)
+#if MAC || (WINDOWS && TARGET_RACK)
 #include <filesystem.h>
 #else
 #include <filesystem>
@@ -31,7 +29,7 @@
 #include "Tunings.h"
 
 
-#if WINDOWS && ( _MSC_VER >= 1920 )
+#if LINUX || (WINDOWS && ( _MSC_VER >= 1920 ))
 // vs2019
 namespace fs = std::filesystem;
 #else
diff --git a/src/common/SurgeSynthesizerIO.cpp b/src/common/SurgeSynthesizerIO.cpp
index f3e4fb6..dbe9bc4 100644
--- a/src/common/SurgeSynthesizerIO.cpp
+++ b/src/common/SurgeSynthesizerIO.cpp
@@ -5,9 +5,7 @@
 #include "DspUtilities.h"
 #include <time.h>
 #include <vt_dsp/vt_dsp_endian.h>
-#if LINUX
-#include <experimental/filesystem>
-#elif MAC || TARGET_RACK
+#if MAC || TARGET_RACK
 #include <filesystem.h>
 #else
 #include <filesystem>
@@ -16,7 +14,7 @@
 #include <iterator>
 #include "UserInteractions.h"
 
-#if WINDOWS && ( _MSC_VER >= 1920 )
+#if LINUX || (WINDOWS && ( _MSC_VER >= 1920 ))
 // vs2019
 namespace fs = std::filesystem;
 #else
diff --git a/src/common/UserDefaults.cpp b/src/common/UserDefaults.cpp
index 34adaff..a69c791 100644
--- a/src/common/UserDefaults.cpp
+++ b/src/common/UserDefaults.cpp
@@ -9,15 +9,13 @@
 #include <sstream>
 #include <fstream>
 
-#if LINUX
-#include <experimental/filesystem>
-#elif MAC || TARGET_RACK
+#if MAC || TARGET_RACK
 #include <filesystem.h>
 #else
 #include <filesystem>
 #endif
 
-#if WINDOWS && ( _MSC_VER >= 1920 )
+#if LINUX || (WINDOWS && ( _MSC_VER >= 1920 ))
 // vs2019
 namespace fs = std::filesystem;
 #else
diff --git a/src/common/WavSupport.cpp b/src/common/WavSupport.cpp
index fa1b041..eb393c3 100644
--- a/src/common/WavSupport.cpp
+++ b/src/common/WavSupport.cpp
@@ -24,15 +24,13 @@
 #include "SurgeStorage.h"
 #include <sstream>
 
-#if LINUX
-#include <experimental/filesystem>
-#elif MAC || TARGET_RACK
+#if MAC || TARGET_RACK
 #include <filesystem.h>
 #else
 #include <filesystem>
 #endif
 
-#if WINDOWS && ( _MSC_VER >= 1920 )
+#if LINUX || (WINDOWS && ( _MSC_VER >= 1920 ))
 // vs2019
 namespace fs = std::filesystem;
 #else
diff --git a/src/common/gui/COscillatorDisplay.cpp b/src/common/gui/COscillatorDisplay.cpp
index 1ffbdee..faa456a 100644
--- a/src/common/gui/COscillatorDisplay.cpp
+++ b/src/common/gui/COscillatorDisplay.cpp
@@ -8,8 +8,6 @@
 #include "UserInteractions.h"
 #if MAC
 #include "filesystem.h"
-#elif LINUX
-#include "experimental/filesystem"
 #else
 #include "filesystem"
 #endif
@@ -17,7 +15,7 @@
 
 using namespace VSTGUI;
 
-#if WINDOWS && ( _MSC_VER >= 1920 )
+#if LINUX || (WINDOWS && ( _MSC_VER >= 1920 ))
 // vs2019
 namespace fs = std::filesystem;
 #else
diff --git a/src/common/gui/SurgeGUIEditor.cpp b/src/common/gui/SurgeGUIEditor.cpp
index 34af1e0..f7dee82 100644
--- a/src/common/gui/SurgeGUIEditor.cpp
+++ b/src/common/gui/SurgeGUIEditor.cpp
@@ -43,14 +43,14 @@
 #if LINUX
 #include "vstgui/lib/platform/platform_x11.h"
 #include "vstgui/lib/platform/linux/x11platform.h"
-#include <experimental/filesystem>
-#elif MAC || TARGET_RACK
+#endif
+#if MAC || TARGET_RACK
 #include <filesystem.h>
 #else
 #include <filesystem>
 #endif
 
-#if WINDOWS && ( _MSC_VER >= 1920 )
+#if LINUX || (WINDOWS && ( _MSC_VER >= 1920 ))
 // vs2019
 namespace fs = std::filesystem;
 #else
