From 826281fcf044773f0ed71e6abfb4985446577eb9 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sun, 16 Jan 2022 14:08:48 +0100
Subject: [PATCH 1/2] Remove comment from application name

---
 share/applications/sViewMP.desktop | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/share/applications/sViewMP.desktop b/share/applications/sViewMP.desktop
index a73620ac..90c56fe1 100644
--- a/share/applications/sViewMP.desktop
+++ b/share/applications/sViewMP.desktop
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Version=1.0
-Name=sView - Stereoscopic Movie Player
+Name=sView
 Comment=Stereoscopic Media Player
-Name[ru]=sView - Универсальный проигрыватель
+Name[ru]=sView
 Comment[ru]=Универсальный проигрыватель стереоскопического видео, аудио и изображений
 Exec=sView --in=video - %U
 StartupWMClass=sView
-- 
2.33.1


From a0ef7d3e059751d2e2409aa55a8cba01b649f74a Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sun, 16 Jan 2022 14:09:32 +0100
Subject: [PATCH 2/2] Add keywords to .desktop file

---
 share/applications/sViewMP.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/share/applications/sViewMP.desktop b/share/applications/sViewMP.desktop
index 90c56fe1..36de6169 100644
--- a/share/applications/sViewMP.desktop
+++ b/share/applications/sViewMP.desktop
@@ -10,4 +10,5 @@ Icon=sview
 Terminal=false
 Type=Application
 Categories=AudioVideo;Player;
+Keywords=stereoscopic;3d;vr;360;180;anaglyph;shutter;hmd;glasses;
 MimeType=application/x-matroska;application/x-extension-mp4;audio/mp4;x-content/video-dvd;video/x-matroska;video/x-matroska-3d;video/dv;video/mpeg;video/x-mpeg;video/msvideo;video/webm;video/quicktime;video/x-avi;video/x-ms-asf;video/x-ms-wmv;video/x-msvideo;video/x-flv;video/mp4;video/mp4v-es;video/x-bik;application/ogg;application/x-ogg;audio/ogg;audio/x-mp3;audio/aac;audio/x-mpeg;audio/mpeg;audio/x-wav;audio/x-mpegurl;audio/x-scpls;audio/x-m4a;audio/x-ms-asf;audio/x-ms-asx;audio/x-ms-wax;audio/x-ape;application/vnd.rn-realmedia;audio/x-real-audio;audio/x-pn-realaudio;application/x-flac;audio/x-flac;audio/x-matroska;application/x-shockwave-flash;audio/vnd.rn-realaudio;audio/x-pn-aiff;audio/x-pn-au;audio/x-pn-wav;audio/x-pn-windows-acm;image/vnd.rn-realpix;video/vnd.rn-realvideo;audio/x-pn-realaudio-plugin;audio/x-ac3;
-- 
2.33.1

