diff --git a/activity/activity.info b/activity/activity.info
index 652beb4..7b2ffcd 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,7 +1,17 @@
 [Activity]
 name = Swift Feet
 activity_version = 1
-bundle_id = ca.tbsf.SwiftFeet
+release_date = 2020-08-22
+bundle_id = org.sugarlabs.SwiftFeet
 exec = sugar-activity3 activity.SwiftFeetActivity
 icon = activity-swiftfeet
 license = GPLv2+
+metadata_license = CC0-1.0
+summary = Get on your feet and dance!
+description = A high energy and up-beat action program, Swift Feet encourages children to be active. 20 different exercises and 10 different dances set to music take users through high, moderate and low impact movements. This activity was designed for OLPC Canada in collaboration with ParticipACTION Canada and Ophea.
+repository = https://github.com/sugarlabs/swiftfeet-activity.git
+url = https://github.com/sugarlabs/swiftfeet-activity.git
+tags = Education
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = http://activities.sugarlabs.org/en-US/sugar/images/p/995/1363971396
