diff --git a/Package.swift b/Package.swift
index 089b554c6..37f74cf89 100644
--- a/Package.swift
+++ b/Package.swift
@@ -23,7 +23,7 @@ let package = Package(
             name: "SwiftLintPlugin",
             capability: .buildTool(),
             dependencies: [
-                .target(name: "SwiftLintBinary", condition: .when(platforms: [.macOS])),
+                // .target(name: "SwiftLintBinary", condition: .when(platforms: [.macOS])),
                 .target(name: "swiftlint", condition: .when(platforms: [.linux]))
             ]
         ),
@@ -134,10 +134,10 @@ let package = Package(
                 "SwiftLintTestHelpers"
             ]
         ),
-        .binaryTarget(
-            name: "SwiftLintBinary",
-            url: "https://github.com/realm/SwiftLint/releases/download/0.54.0/SwiftLintBinary-macos.artifactbundle.zip",
-            checksum: "963121d6babf2bf5fd66a21ac9297e86d855cbc9d28322790646b88dceca00f1"
-        ),
+        //.binaryTarget(
+        //    name: "SwiftLintBinary",
+        //    url: "https://github.com/realm/SwiftLint/releases/download/0.54.0/SwiftLintBinary-macos.artifactbundle.zip",
+        //    checksum: "963121d6babf2bf5fd66a21ac9297e86d855cbc9d28322790646b88dceca00f1"
+        //),
         .macro(
             name: "SwiftLintCoreMacros",
