diff --git a/activity.py b/activity.py
index 312f4b1..fa82bf5 100644
--- a/activity.py
+++ b/activity.py
@@ -54,6 +54,8 @@ from sugar3.activity.widgets import StopButton
 
 from videos import EXERCISES, DANCES
 
+from sugarapp.widgets import SugarCompatibleActivity
+
 Gst.init(None)
 
 class PaddedVBox(Gtk.VBox):
@@ -242,9 +244,9 @@ class VideoButton(Gtk.EventBox):
         # self.window.set_cursor(Gdk.Cursor.new(Gdk.HAND2))
         self.get_window().set_cursor(Gdk.Cursor(Gdk.CursorType.HAND2))
 
-class SwiftFeetActivity(activity.Activity):
+class SwiftFeetActivity(SugarCompatibleActivity):
     def __init__(self, handle):
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
         self._current_video_idx = None
         self.max_participants = 1
 
