diff -rupN --no-dereference tesseract-4.1.1/src/ccutil/Makefile.am tesseract-4.1.1-new/src/ccutil/Makefile.am
--- tesseract-4.1.1/src/ccutil/Makefile.am	2019-12-26 15:21:51.000000000 +0100
+++ tesseract-4.1.1-new/src/ccutil/Makefile.am	2019-12-27 21:19:53.950010557 +0100
@@ -2,7 +2,7 @@ AM_CXXFLAGS =
 AM_CPPFLAGS =
 
 if !NO_TESSDATA_PREFIX
-AM_CXXFLAGS += -DTESSDATA_PREFIX=@datadir@
+AM_CXXFLAGS += -DTESSDATA_PREFIX=@datadir@/tesseract
 endif
 
 if VISIBILITY
diff -rupN --no-dereference tesseract-4.1.1/src/lstm/Makefile.am tesseract-4.1.1-new/src/lstm/Makefile.am
--- tesseract-4.1.1/src/lstm/Makefile.am	2019-12-26 15:21:51.000000000 +0100
+++ tesseract-4.1.1-new/src/lstm/Makefile.am	2019-12-27 21:19:53.950010557 +0100
@@ -16,7 +16,7 @@ AM_CPPFLAGS += -I/usr/include/tensorflow
 endif
 
 if !NO_TESSDATA_PREFIX
-AM_CXXFLAGS += -DTESSDATA_PREFIX=@datadir@
+AM_CXXFLAGS += -DTESSDATA_PREFIX=@datadir@/tesseract
 endif
 
 if DISABLED_LEGACY_ENGINE
diff -rupN --no-dereference tesseract-4.1.1/tessdata/configs/Makefile.am tesseract-4.1.1-new/tessdata/configs/Makefile.am
--- tesseract-4.1.1/tessdata/configs/Makefile.am	2019-12-26 15:21:51.000000000 +0100
+++ tesseract-4.1.1-new/tessdata/configs/Makefile.am	2019-12-27 21:19:53.950010557 +0100
@@ -1,4 +1,4 @@
-datadir = @datadir@/tessdata/configs
+datadir = @datadir@/tesseract/tessdata/configs
 data_DATA = inter makebox box.train unlv ambigs.train lstm.train lstmdebug
 data_DATA += api_config kannada box.train.stderr quiet logfile digits get.images
 data_DATA += lstmbox wordstrbox
diff -rupN --no-dereference tesseract-4.1.1/tessdata/Makefile.am tesseract-4.1.1-new/tessdata/Makefile.am
--- tesseract-4.1.1/tessdata/Makefile.am	2019-12-26 15:21:51.000000000 +0100
+++ tesseract-4.1.1-new/tessdata/Makefile.am	2019-12-27 21:19:53.950010557 +0100
@@ -1,4 +1,4 @@
-datadir = @datadir@/tessdata
+datadir = @datadir@/tesseract/tessdata
 
 data_DATA = pdf.ttf
 EXTRA_DIST = $(data_DATA)
diff -rupN --no-dereference tesseract-4.1.1/tessdata/tessconfigs/Makefile.am tesseract-4.1.1-new/tessdata/tessconfigs/Makefile.am
--- tesseract-4.1.1/tessdata/tessconfigs/Makefile.am	2019-12-26 15:21:51.000000000 +0100
+++ tesseract-4.1.1-new/tessdata/tessconfigs/Makefile.am	2019-12-27 21:19:53.951010558 +0100
@@ -1,3 +1,3 @@
-datadir = @datadir@/tessdata/tessconfigs
+datadir = @datadir@/tesseract/tessdata/tessconfigs
 data_DATA = batch batch.nochop nobatch matdemo segdemo msdemo
 EXTRA_DIST = batch batch.nochop nobatch matdemo segdemo msdemo