diff --git a/src/TWUtils.cpp b/src/TWUtils.cpp
index 793f004..2b09c3a 100644
--- a/src/TWUtils.cpp
+++ b/src/TWUtils.cpp
@@ -42,6 +42,7 @@
 #include <QCryptographicHash>
 #include <QTextStream>
 #include <QDateTime>
+#include <QStandardPaths>
 
 #include <hunspell.h>
 
@@ -91,9 +92,7 @@ const QString TWUtils::getLibraryPath(const QString& subdir, const bool updateOn
 	
 	libRootPath = TWApp::instance()->getPortableLibPath();
 	if (libRootPath.isEmpty()) {
-#if defined(Q_OS_DARWIN)
-		libRootPath = QDir::homePath() + QLatin1String("/Library/" TEXWORKS_NAME "/");
-#elif defined(Q_OS_UNIX) // && !defined(Q_OS_DARWIN)
+#if defined(Q_OS_UNIX) // && !defined(Q_OS_DARWIN)
 		if (subdir == QLatin1String("dictionaries")) {
 			libPath = QString::fromLatin1(TW_DICPATH);
 			QString dicPath = QString::fromLocal8Bit(getenv("TW_DICPATH"));
@@ -101,11 +100,16 @@ const QString TWUtils::getLibraryPath(const QString& subdir, const bool updateOn
 				libPath = dicPath;
 			return libPath; // don't try to create/update the system dicts directory
 		}
-		else
-			libRootPath = QDir::homePath() + QLatin1String("/." TEXWORKS_NAME "/");
-#else // defined(Q_OS_WIN)
-		libRootPath = QDir::homePath() + QLatin1String("/" TEXWORKS_NAME "/");
 #endif
+#if QT_VERSION < 0x050400
+	libRootPath = QStandardPaths::writableLocation(QStandardPaths::DataLocation);
+#else
+	libRootPath = QStandardPaths::writableLocation(QStandardPaths::AppDataLocation);
+#endif
+
+	// Paranoia: this should not happen
+	if (libRootPath.isEmpty())
+		libRootPath = QDir::homePath() + QLatin1String("/." TEXWORKS_NAME "/");
 	}
 	libPath = QDir(libRootPath).absolutePath() + QDir::separator() + subdir;
 
diff --git a/texworks.appdata.xml b/texworks.appdata.xml
index 87e9de2..2e90062 100644
--- a/texworks.appdata.xml
+++ b/texworks.appdata.xml
@@ -245,6 +245,7 @@
 	</provides>
 	<launchable type="desktop-id">texworks.desktop</launchable>
 	<translation type="qt" />
+	<content_rating type="oars-1.0" />
 
 	<releases>
 		<release version="0.6.3" date="2019-03-16" />
