--- ./src/settings.cpp	2019-10-07 16:46:07.243532605 +0300
+++ ./src/settings-new.cpp	2019-10-08 02:19:24.113977761 +0300
@@ -11,7 +11,7 @@
 #ifdef Q_OS_WIN
 #  define THUNDERBIRD_EXE_PATH "\"%ProgramFiles(x86)%\\Mozilla Thunderbird\\thunderbird.exe\""
 #else
-#  define THUNDERBIRD_EXE_PATH "/usr/bin/thunderbird"
+#  define THUNDERBIRD_EXE_PATH "/app/bin/thunderbird"
 #endif
 
 Settings * pSettings;
