From c0349769acc1289189110f0a2a7d24401fbfae08 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sat, 20 Feb 2016 04:24:57 +0100
Subject: [PATCH] log-store-xml: Allow overriding basedir

---
 telepathy-logger/log-store-xml.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/telepathy-logger/log-store-xml.c b/telepathy-logger/log-store-xml.c
index be881d0..01676fa 100644
--- a/telepathy-logger/log-store-xml.c
+++ b/telepathy-logger/log-store-xml.c
@@ -1744,6 +1744,10 @@ log_store_xml_get_basedir (TplLogStoreXml *self)
         {
           user_data_dir = g_getenv ("TPL_TEST_LOG_DIR");
         }
+      else if (g_getenv ("TPL_LOG_DIR") != NULL)
+        {
+          user_data_dir = g_getenv ("TPL_LOG_DIR");
+        }
       else
         {
           user_data_dir = g_get_user_data_dir ();
-- 
2.7.1

