diff --git a/external/sol/sol.hpp b/external/sol/sol.hpp
index d30d9697..b33e4128 100644
--- a/external/sol/sol.hpp
+++ b/external/sol/sol.hpp
@@ -6512,7 +6512,8 @@ namespace sol {
 			}
 
 			static int push(lua_State* L, const wchar_t(&str)[N], std::size_t sz) {
-				return stack::push<const wchar_t*>(L, str, str + sz);
+				const wchar_t* str_ptr = static_cast<const wchar_t*>(str);
+				return stack::push<const wchar_t*>(L, str_ptr, str_ptr + sz);
 			}
 		};
 
@@ -6523,7 +6524,8 @@ namespace sol {
 			}
 
 			static int push(lua_State* L, const char16_t(&str)[N], std::size_t sz) {
-				return stack::push<const char16_t*>(L, str, str + sz);
+				const char16_t* str_ptr = static_cast<const char16_t*>(str);
+				return stack::push<const char16_t*>(L, str_ptr, str_ptr + sz);
 			}
 		};
 
@@ -6534,7 +6536,8 @@ namespace sol {
 			}
 
 			static int push(lua_State* L, const char32_t(&str)[N], std::size_t sz) {
-				return stack::push<const char32_t*>(L, str, str + sz);
+				const char32_t* str_ptr = static_cast<const char32_t*>(str);
+				return stack::push<const char32_t*>(L, str_ptr, str_ptr + sz);
 			}
 		};
 
