diff --git a/Makefile.orig b/Makefile
index 27e56d0..738a546 100644
--- a/Makefile.orig
+++ b/Makefile
@@ -246,7 +246,7 @@ endif
 #############################################################################
 
 BD=$(BUILD_DIR)/debug-$(PLATFORM)-$(ARCH)
-BR=$(BUILD_DIR)/release-$(PLATFORM)-$(ARCH)
+BR=$(BUILD_DIR)/release
 
 CDIR=$(MOUNT_DIR)/client
 SDIR=$(MOUNT_DIR)/server
@@ -1760,16 +1760,16 @@ TARGETS += $(B)/granger$(FULLBINEXT)
 endif
 
 $(B)/scripts:
-	rsync -rupE --exclude=".*" scripts $(B)
+	cp -a scripts $(B)
 
 TARGETS += $(B)/scripts
 
 $(B)/GPL:
-	rsync GPL $(B)
+	cp GPL $(B)
 $(B)/COPYING:
-	rsync COPYING $(B)
+	cp COPYING $(B)
 $(B)/CC:
-	rsync CC $(B)
+	cp CC $(B)
 
 TARGETS += $(B)/GPL $(B)/COPYING $(B)/CC
 
