diff --git a/TurtleArt/turtleblocks.py b/TurtleArt/turtleblocks.py
index f15f1bae..75db765b 100755
--- a/TurtleArt/turtleblocks.py
+++ b/TurtleArt/turtleblocks.py
@@ -437,6 +437,7 @@ return %s(self)" % (p, P, P)
         self.fixed.set_size_request(width, height)
 
         self.vbox = Gtk.VBox(False, 0)
+        self.vbox.set_size_request(width, height)
         self.vbox.show()
 
         self.menu_bar = self._get_menu_bar()
