diff --git a/setup.py b/setup.py
index 68c0c29a..4e9b9009 100755
--- a/setup.py
+++ b/setup.py
@@ -39,6 +39,8 @@ def generate_appdata(prefix, bundle_id):
         info.get('Activity', 'description')))
     root.append(desc)
 
+    ET.SubElement(root, 'content_rating', type='oars-1.1')
+
     copy_pairs = [('metadata_license', 'metadata_license'),
                   ('license', 'project_license'),
                   ('summary', 'summary'),
