diff --git a/setup.py b/setup.py
index 4e9b9009..36a889ea 100755
--- a/setup.py
+++ b/setup.py
@@ -41,6 +41,13 @@ def generate_appdata(prefix, bundle_id):
 
     ET.SubElement(root, 'content_rating', type='oars-1.1')
 
+    release_pairs = [('218', '2018-05-22'),
+                     ('216', '2017-06-24'),
+                     ('215', '2017-04-19')]
+    releases_root = ET.SubElement(root, 'releases')
+    for version, date in release_pairs:
+        ET.SubElement(releases_root, 'release', date=date, version=version)
+
     copy_pairs = [('metadata_license', 'metadata_license'),
                   ('license', 'project_license'),
                   ('summary', 'summary'),
