From 304db393445db4c67bcf8bc4fd4f310f2675ddff Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Sat, 7 Dec 2019 02:15:41 -0300
Subject: [PATCH 2/4] fix-turtle-color

---
 game.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/game.py b/game.py
index 7d83d65..1cfb85c 100644
--- a/game.py
+++ b/game.py
@@ -399,7 +399,7 @@ class Game():
         self._svg_width = self._dot_size * 2
         self._svg_height = self._dot_size * 2
         self._stroke = '#101010'
-        self._fill = '#404040'
+        self._fill = '#00FF11'
         return svg_str_to_pixbuf(
             self._header() + \
             self._turtle() + \
-- 
2.19.1

