From f9375f466319da57dfe608c2af3253423efa818b Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Sat, 7 Dec 2019 02:20:26 -0300
Subject: [PATCH 3/4] monitors

---
 TurtlePondActivity.py | 5 +++--
 game.py               | 7 +++++--
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/TurtlePondActivity.py b/TurtlePondActivity.py
index dd6d96f..fbdb52d 100644
--- a/TurtlePondActivity.py
+++ b/TurtlePondActivity.py
@@ -43,6 +43,7 @@ from utils import svg_str_to_pixbuf
 import logging
 _logger = logging.getLogger('turtle-in-a-pond-activity')
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import DesktopOpenChooser
 from sugarapp.widgets import SugarCompatibleActivity
 
@@ -71,8 +72,8 @@ class TurtlePondActivity(SugarCompatibleActivity):
 
         # Create a canvas
         canvas = Gtk.DrawingArea()
-        canvas.set_size_request(Gdk.Screen.width(), \
-                                Gdk.Screen.height())
+        canvas.set_size_request(PrimaryMonitor.width(), \
+                                PrimaryMonitor.height())
         self.set_canvas(canvas)
         canvas.show()
         self.show_all()
diff --git a/game.py b/game.py
index 1cfb85c..fc2ad8c 100644
--- a/game.py
+++ b/game.py
@@ -33,6 +33,9 @@ except ImportError:
 
 from sprites import Sprites, Sprite
 
+from sugarapp.helpers import PrimaryMonitor
+
+
 FILL = 1
 STROKE = 0
 THIRTEEN = 13
@@ -97,8 +100,8 @@ class Game():
         self._canvas.connect("draw", self.__draw_cb)
         self._canvas.connect("button-press-event", self._button_press_cb)
 
-        self._width = Gdk.Screen.width()
-        self._height = Gdk.Screen.height() - (GRID_CELL_SIZE * 1.5)
+        self._width = PrimaryMonitor.width()
+        self._height = PrimaryMonitor.height() - (GRID_CELL_SIZE * 1.5)
         self._scale = self._height / (14.0 * DOT_SIZE * 1.2)
         self._dot_size = int(DOT_SIZE * self._scale)
         self._turtle_offset = 0
-- 
2.19.1

