diff --git a/turtleblocks.desktop b/turtleblocks.desktop
index f22df19b..2ac34640 100644
--- a/turtleblocks.desktop
+++ b/turtleblocks.desktop
@@ -5,8 +5,8 @@ Name[es]=TortugArte
 GenericName=Sugar TurtleBlocks activity
 GenericName[nb]=Sugar Skilpaddekunst-aktivitet
 GenericName[es]=Actividad Sugar TortugArte
-TryExec=/usr/share/sugar/activities/TurtleArt.activity/turtleblocks.py
-Exec=/usr/share/sugar/activities/TurtleArt.activity/turtleblocks.py
+TryExec=turtleblocks
+Exec=turtleblocks
 Icon=turtleart
 Type=Application
 Comment=A Logo programming environment
