diff --git a/activity/activity.info b/activity/activity.info
index 012c401e..ae98c346 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,6 +2,7 @@
 name = TurtleBlocks
 activity_version = 218
 license = MIT
+metadata_license = CC0-1.0
 bundle_id = org.laptop.TurtleArtActivity
 exec = sugar-activity TurtleArtActivity.TurtleArtActivity
 icon = activity-turtleart
@@ -10,5 +11,6 @@ website = http://wiki.sugarlabs.org/go/Activities/Turtle_Art
 mime_types = application/x-turtle-art;application/vnd.turtleblocks
 categories = programming art
 summary = A Logo-inspired turtle that draws colorful pictures with snap-together visual programming blocks
+description = Turtle Art, also known as Turtle Blocks, is an activity with a Logo-inspired graphical "turtle" that draws colorful art based on snap-together visual programming elements. Its "low floor" provides an easy entry point for beginners. It also has "high ceiling" programming, graphics, mathematics, and Computer Science features which will challenge the more adventurous student.
 repository = https://github.com/sugarlabs/turtleart-activity
 
