From f239aec3d77a7f10280efe67152befc706da9615 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Sat, 7 Dec 2019 02:30:00 -0300
Subject: [PATCH 4/4] info

---
 activity/activity.info | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index e8ab945..8f128f3 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,12 +1,18 @@
 [Activity]
 name = Turtle in a Pond
 activity_version = 9
+release_date = 2019-11-19
 license = GPLv3+;MIT
+metadata_license = CC0-1.0
 bundle_id = org.sugarlabs.TurtlePondActivity
 exec = sugar-activity TurtlePondActivity.TurtlePondActivity
 icon = activity-turtlepond
 show_launcher = yes
-summary = A strategy game with the objective to surround the turtle before it runs of the screen.  
-tags = Maths;Game;Programming
+summary = Surround the turtle before it runs of the screen
+description = A strategy game. The goal is to surround the turtle before it runs off the screen. You can load your own strategy for the turtle by importing Python code.
+tags = Game
 repository = https://github.com/sugarlabs/turtlepond
 url = https://help.sugarlabs.org/en/turtle_in_a_pond.html
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/turtlepond/master/screenshots/en/1.png
-- 
2.19.1

