diff --git a/setup.cfg b/setup.cfg
index 891cdbd8..0eaae453 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,6 @@
 [install]
-install_data=/usr/share/TurtleBlocks
-install_scripts=/usr/bin
+install_data=/app/share/TurtleBlocks
+install_scripts=/app/bin
 
 [pycodestyle]
 # E402 module level import not at top of file
diff --git a/setup.py b/setup.py
index 042675d4..c1b67221 100755
--- a/setup.py
+++ b/setup.py
@@ -75,7 +75,7 @@ if len(sys.argv) > 1 and '--no-sugar' == sys.argv[1]:
         ('activity', get_files('activity/')),
         ('icons', get_files('icons/')),
         ('images', get_files('images/')),
-        ('/usr/share/applications', ['turtleblocks.desktop']),
+        ('/app/share/applications', ['turtleblocks.desktop']),
         ('org.laptop.TurtleArtActivity.gschema.xml')
     ]
 
