diff --git a/Makefile b/Makefile
index 911aa5c..b9e4932 100644
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,7 @@ dist :
 	rm -Rf $(NAME)-$(VERSION)
 
 install : $(NAME)
-	install -d $(DESTDIR)/usr/bin
-	install -d $(DESTDIR)/usr/man/man6
-	install -m755 $(NAME) $(DESTDIR)/usr/bin
-	install -m644 man/$(NAME).6.gz $(DESTDIR)/usr/man/man6
+	install -d $(DESTDIR)/app/bin
+	install -d $(DESTDIR)/app/man/man6
+	install -m755 $(NAME) $(DESTDIR)/app/bin
+	install -m644 man/$(NAME).6.gz $(DESTDIR)/app/man/man6
