From e215494d3d6a4b7f1290673deb6bf2aa329fa5e5 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Wed, 1 Mar 2017 17:01:29 -0800
Subject: [PATCH] Use XDG_CONFIG_HOME when set

---
 tuxpuck.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/tuxpuck.c b/tuxpuck.c
index 4c0d6d7..07f7b1d 100644
--- a/tuxpuck.c
+++ b/tuxpuck.c
@@ -235,7 +235,7 @@ static void _save_settings(void)
 static void _tuxpuck_init(void)
 {
 #ifndef windows
-  char *homeDir = NULL;
+  char *settingsDir = NULL;
 #endif
   srand(time(NULL));
   SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO);
@@ -249,8 +249,13 @@ static void _tuxpuck_init(void)
   _settings->fullscreen = 0;
   _settings->mouse_speed = 5;
 #ifndef windows
-  homeDir = getenv("HOME");
-  sprintf(_settings_file, "%s/.tuxpuckrc", homeDir);
+  settingsDir = getenv("XDG_CONFIG_HOME");
+  if (settingsDir) {
+    sprintf(_settings_file, "%s/tuxpuckrc", settingsDir);
+  } else {
+    settingsDir = getenv("HOME");
+    sprintf(_settings_file, "%s/.tuxpuckrc", settingsDir);
+  }
 #endif
   _read_settings();
   audio_set_mute(!_settings->sound);
-- 
2.9.3

