diff -up tuxtype_w_fonts-1.8.1/Makefile.am.patch tuxtype_w_fonts-1.8.1/Makefile.am
--- tuxtype_w_fonts-1.8.1/Makefile.am.patch	2010-05-01 11:40:44.000000000 +0200
+++ tuxtype_w_fonts-1.8.1/Makefile.am	2010-05-01 11:41:08.000000000 +0200
@@ -48,8 +48,6 @@ install-data-hook:
 # Global sites for game state data (e.g. /var/games/tuxtype):
 	$(MKDIR_P) -m 2755 $(DESTDIR)$(pkglocalstatedir)
 	$(MKDIR_P) -m 2755 $(DESTDIR)$(pkglocalstatedir)/words
-#	chown root:games $(DESTDIR)$(pkglocalstatedir)
-	chown root:games $(DESTDIR)$(pkglocalstatedir)/words
 
 
 uninstall-hook:
