diff --git a/activity/activity.info b/activity/activity.info
index 7e9a26f..4f689a8 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,10 +2,16 @@
 name = Typing Turtle
 activity_version = 32
 license = GPLv3+
-bundle_id = org.laptop.community.TypingTurtle
+release_date = 2024-06-05
+bundle_id = org.sugarlabs.TypingTurtle
 icon = Activity-typingturtle
 exec = sugar-activity3 typingturtle.TypingTurtle
-summary = Need some help typing? Here you will learn the best way to hold your hands in order for you to become a faster typist!
+summary = Typing Turtle is an interactive touch typing program.
+description = Need some help typing? Here you will learn the best way to hold your hands in order for you to become a faster typist! It gradually introduces the keys on the keyboard through a series of lessons, until the student has learned the entire keyboard.
 repository = https://github.com/sugarlabs/typing-turtle-activity
-tags = Language;System;Game
+tags = Education
+metadata_license = CC0-1.0
 url = https://help.sugarlabs.org/en/typing_turtle.html
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+developer_id = org.sugarlabs
