diff --git a/mainscreen.py b/mainscreen.py
index 250b0ee..3ef4cfd 100644
--- a/mainscreen.py
+++ b/mainscreen.py
@@ -29,6 +29,8 @@ from gi.repository import GObject
 import sugar3.activity.activity
 from sugar3.graphics import *
 
+from sugarapp.helpers import PrimaryMonitor
+
 # Import activity modules.
 import lessonscreen, medalscreen
 import balloongame
@@ -107,8 +109,8 @@ class MainScreen(Gtk.VBox):
         self.lessons = sorted(self.lessons, key=lambda x: x['order']) # replace sort with sorted for faster exec
 
         # Load all the keyboard images.
-        width = int(Gdk.Screen.width())
-        height = int(Gdk.Screen.height()*0.3)
+        width = int(PrimaryMonitor.width())
+        height = int(PrimaryMonitor.height()*0.3)
 
         self.keyboard_images = keyboard.KeyboardImages(width, height)
         
diff --git a/titlescene.py b/titlescene.py
index b28d8b3..8a6b51e 100644
--- a/titlescene.py
+++ b/titlescene.py
@@ -25,6 +25,7 @@ from gi.repository import Pango
 from gi.repository import PangoCairo
 from gi.repository import GdkPixbuf
 
+from sugarapp.helpers import PrimaryMonitor
 
 class TitleScene(Gtk.DrawingArea):
     # Maximum portion of the screen the background can fill vertically.
@@ -41,8 +42,8 @@ class TitleScene(Gtk.DrawingArea):
 
         pbuf = GdkPixbuf.Pixbuf.new_from_file('images/main-background.jpg')
 
-        width_ratio = float(Gdk.Screen.width()) / pbuf.get_width()
-        height_ratio = float(Gdk.Screen.height()*TitleScene.BACKGROUND_HEIGHT_RATIO) / pbuf.get_height()
+        width_ratio = float(PrimaryMonitor.width()) / pbuf.get_width()
+        height_ratio = float(PrimaryMonitor.height()*TitleScene.BACKGROUND_HEIGHT_RATIO) / pbuf.get_height()
 
         ratio = min(width_ratio, height_ratio)
         self.backgroundpixbuf = pbuf.scale_simple(int(pbuf.get_width()*ratio), int(pbuf.get_height()*ratio), GdkPixbuf.InterpType.BILINEAR)
