diff --git a/mainscreen.py b/mainscreen.py
index 250b0ee..3ef4cfd 100644
--- a/mainscreen.py
+++ b/mainscreen.py
@@ -29,6 +29,8 @@ from gi.repository import GObject
 import sugar3.activity.activity
 from sugar3.graphics import *
 
+from sugarapp.helpers import PrimaryMonitor
+
 # Import activity modules.
 import lessonscreen, medalscreen
 import balloongame
@@ -107,8 +109,8 @@ class MainScreen(Gtk.VBox):
         self.lessons = sorted(self.lessons, key=lambda x: x['order']) # replace sort with sorted for faster exec
 
         # Load all the keyboard images.
-        width = int(Gdk.Screen.width())
-        height = int(Gdk.Screen.height()*0.3)
+        width = int(PrimaryMonitor.width())
+        height = int(PrimaryMonitor.height()*0.3)
 
         self.keyboard_images = keyboard.KeyboardImages(width, height)
         
