diff --git a/keyboard.py b/keyboard.py
index 621f71c..aaec3d1 100644
--- a/keyboard.py
+++ b/keyboard.py
@@ -104,14 +104,14 @@ def _is_olpcm_model():
     keyboard, and 'olpc' for membrane keyboard.
 
     """
-    code = None
-    p = subprocess.Popen(["setxkbmap", "-query"], stdout=subprocess.PIPE)
-    out, err = p.communicate()
+    # code = None
+    # p = subprocess.Popen(["setxkbmap", "-query"], stdout=subprocess.PIPE)
+    # out, err = p.communicate()
     
-    for line in out.decode('utf-8').splitlines():
-        if line.startswith('model:'):
-            code = line.split()[1]
-    return code == 'olpcm'
+    # for line in out.decode('utf-8').splitlines():
+    #     if line.startswith('model:'):
+    #         code = line.split()[1]
+    return None
 
 def get_layout():
     if _is_olpcm_model():
