diff --git a/titlescene.py b/titlescene.py
index b28d8b3..8a6b51e 100644
--- a/titlescene.py
+++ b/titlescene.py
@@ -25,6 +25,7 @@ from gi.repository import Pango
 from gi.repository import PangoCairo
 from gi.repository import GdkPixbuf
 
+from sugarapp.helpers import PrimaryMonitor
 
 class TitleScene(Gtk.DrawingArea):
     # Maximum portion of the screen the background can fill vertically.
@@ -41,8 +42,8 @@ class TitleScene(Gtk.DrawingArea):
 
         pbuf = GdkPixbuf.Pixbuf.new_from_file('images/main-background.jpg')
 
-        width_ratio = float(Gdk.Screen.width()) / pbuf.get_width()
-        height_ratio = float(Gdk.Screen.height()*TitleScene.BACKGROUND_HEIGHT_RATIO) / pbuf.get_height()
+        width_ratio = float(PrimaryMonitor.width()) / pbuf.get_width()
+        height_ratio = float(PrimaryMonitor.height()*TitleScene.BACKGROUND_HEIGHT_RATIO) / pbuf.get_height()
 
         ratio = min(width_ratio, height_ratio)
         self.backgroundpixbuf = pbuf.scale_simple(int(pbuf.get_width()*ratio), int(pbuf.get_height()*ratio), GdkPixbuf.InterpType.BILINEAR)
