index 298ccea..12382d6 100755
--- a/typingturtle.py
+++ b/typingturtle.py
@@ -42,7 +42,9 @@ from sugar3 import profile
 
 from sugar3.graphics.toolbarbox import ToolbarBox
 from sugar3.activity.widgets import StopButton
-from sugar3.activity.widgets import ActivityToolbarButton
+
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
 
 # Initialize logging.
 log = logging.getLogger('Typing Turtle')
@@ -78,9 +80,9 @@ import mainscreen, editlessonlistscreen
 # 
 # It owns the main application window, and all the various toolbars and options.
 # Activity Screens are stored in a stack, with the currently active screen on top.
-class TypingTurtle(sugar3.activity.activity.Activity):
+class TypingTurtle(SugarCompatibleActivity):
     def __init__ (self, handle):
-        sugar3.activity.activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
         self.set_title(_("Typing Turtle"))
         self.max_participants = 1
         
@@ -121,7 +123,7 @@ class TypingTurtle(sugar3.activity.activity.Activity):
     def build_toolbox(self):
         self.toolbar_box = ToolbarBox()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         self.toolbar_box.toolbar.insert(activity_button, 0)
         activity_button.show()
 
