From 75410cbc2f20b22b17f58119c1d0d6726d847610 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 3 Sep 2024 22:10:03 +0200
Subject: [PATCH] build: Don't install daemon files when building libs only
 
---
 data/meson.build | 46 ++++++++++++++++++++++++----------------------
 1 file changed, 24 insertions(+), 22 deletions(-)
 
diff --git a/data/meson.build b/data/meson.build
index c34dee3fa715..5b2db9d22517 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -30,28 +30,30 @@ if enable_pam
   )
 endif
 
-configure_file(
-  input: 'polkit.service.in',
-  output: '@BASENAME@',
-  configuration: service_conf,
-  install: true,
-  install_dir: systemdsystemunitdir,
-)
+if not get_option('libs-only')
+  configure_file(
+    input: 'polkit.service.in',
+    output: '@BASENAME@',
+    configuration: service_conf,
+    install: true,
+    install_dir: systemdsystemunitdir,
+  )
 
-configure_file(
-  input: 'polkit.conf.in',
-  output: '@BASENAME@',
-  configuration: service_conf,
-  install: true,
-  install_dir: sysusers_dir,
-)
+  configure_file(
+    input: 'polkit.conf.in',
+    output: '@BASENAME@',
+    configuration: service_conf,
+    install: true,
+    install_dir: sysusers_dir,
+  )
 
-install_data(
-  'policyconfig-1.dtd',
-  install_dir: pk_datadir / 'polkit-1'
-)
+  install_data(
+    'policyconfig-1.dtd',
+    install_dir: pk_datadir / 'polkit-1'
+  )
 
-install_data(
-  'polkit-tmpfiles.conf',
-  install_dir: tmpfiles_dir
-)
+  install_data(
+    'polkit-tmpfiles.conf',
+    install_dir: tmpfiles_dir
+  )
+endif
-- 
2.46.0
 