--- a/configure.ac
+++ b/configure.ac
@@ -153,11 +153,11 @@
 AC_SUBST(POLKIT_GOBJECT_1_CFLAGS)
 AC_SUBST(POLKIT_GOBJECT_1_LIBS)

-PKG_CHECK_MODULES(POLKIT_AGENT_1, [polkit-agent-1 >= 0.102])
-AC_SUBST(POLKIT_AGENT_1_CFLAGS)
-AC_SUBST(POLKIT_AGENT_1_LIBS)
-
 if test "x$enable_daemon" = "xyes"; then
+  PKG_CHECK_MODULES(POLKIT_AGENT_1, [polkit-agent-1 >= 0.102])
+  AC_SUBST(POLKIT_AGENT_1_CFLAGS)
+  AC_SUBST(POLKIT_AGENT_1_LIBS)
+
   PKG_CHECK_MODULES(GUDEV, [gudev-1.0 >= 165])
   AC_SUBST(GUDEV_CFLAGS)
   AC_SUBST(GUDEV_LIBS)
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -15,6 +15,7 @@
 	$(WARN_CFLAGS)                                                         \
 	$(NULL)

+if ENABLE_DAEMON
 bin_PROGRAMS = udisksctl

 udisksctl_SOURCES =                                                            \
@@ -36,6 +37,7 @@
 	$(top_builddir)/udisks/libudisks2.la                                   \
 	$(NULL)

+endif
 # ------------------------------------------------------------------------------

 if ENABLE_DAEMON
