diff --git a/Makefile.in b/Makefile.in
--- a/Makefile.in
+++ b/Makefile.in
@@ -167,7 +167,7 @@ install-info: units.info
	if test -f $(DESTDIR)$(infodir)/dir; then \
	  if $(SHELL) -c 'install-info --version' >/dev/null 2>&1; then \
	    install-info --dir-file=$(DESTDIR)$(infodir)/dir \
-	               $(DESTDIR)$(infodir)/units.info; \
+	               $(DESTDIR)$(infodir)/units.info || :; \
	  else true; fi \
	else true; fi
 
