--- a/osu.Game/Configuration/OsuConfigManager.cs
+++ b/osu.Game/Configuration/OsuConfigManager.cs
@@ -116,7 +116,7 @@ protected override void InitialiseDefaults()
             SetDefault(OsuSetting.GameplayDisableWinKey, true);
 
             // Update
-            SetDefault(OsuSetting.ReleaseStream, ReleaseStream.Lazer);
+            SetDefault(OsuSetting.ReleaseStream, ReleaseStream.Flathub);
 
             SetDefault(OsuSetting.Version, string.Empty);
 
--- a/osu.Game/Configuration/ReleaseStream.cs
+++ b/osu.Game/Configuration/ReleaseStream.cs
@@ -5,7 +5,7 @@ namespace osu.Game.Configuration
 {
     public enum ReleaseStream
     {
-        Lazer,
+        Flathub,
         //Stable40,
         //Beta40,
         //Stable
--- a/osu.Game/OsuGameBase.cs
+++ b/osu.Game/OsuGameBase.cs
@@ -52,7 +52,7 @@ namespace osu.Game
     /// </summary>
     public partial class OsuGameBase : Framework.Game, ICanAcceptFiles
     {
-        public const string CLIENT_STREAM_NAME = @"lazer";
+        public const string CLIENT_STREAM_NAME = @"flathub";
 
         public const int SAMPLE_CONCURRENCY = 6;
 
--- a/osu.Game/Updater/SimpleUpdateManager.cs
+++ b/osu.Game/Updater/SimpleUpdateManager.cs
@@ -49,6 +49,7 @@ protected override async Task<bool> PerformUpdateCheck()
 
                 if (latestTagName != version)
                 {
+		/*
                     Notifications.Post(new SimpleNotification
                     {
                         Text = $"A newer release of osu! has been found ({version} → {latestTagName}).\n\n"
@@ -60,6 +61,7 @@ protected override async Task<bool> PerformUpdateCheck()
                             return true;
                         }
                     });
+		*/
 
                     return true;
                 }
@@ -88,7 +90,7 @@ private string getBestUrl(GitHubRelease release)
                     break;
 
                 case RuntimeInfo.Platform.Linux:
-                    bestAsset = release.Assets?.Find(f => f.Name.EndsWith(".AppImage", StringComparison.Ordinal));
+                    // bestAsset = release.Assets?.Find(f => f.Name.EndsWith(".AppImage", StringComparison.Ordinal));
                     break;
 
                 case RuntimeInfo.Platform.iOS:
--- a/osu.Game/Updater/UpdateManager.cs
+++ b/osu.Game/Updater/UpdateManager.cs
@@ -20,7 +20,7 @@ public class UpdateManager : CompositeDrawable
         /// <summary>
         /// Whether this UpdateManager should be or is capable of checking for updates.
         /// </summary>
-        public bool CanCheckForUpdate => game.IsDeployedBuild &&
+        public bool CanCheckForUpdate => false &&
                                          // only implementations will actually check for updates.
                                          GetType() != typeof(UpdateManager);
 
