diff --git a/misc/dist/linux/org.godotengine.Godot.desktop b/misc/dist/linux/org.godotengine.Godot.desktop
--- a/misc/dist/linux/org.godotengine.Godot.desktop	2023-03-25 19:06:57.017826108 +0000
+++ b/misc/dist/linux/org.godotengine.Godot.desktop	2023-03-25 19:07:17.970046036 +0000
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=Godot Engine
+Name=Godot Engine 3
 GenericName=Libre game engine
 GenericName[el]=Ελεύθερη μηχανή παιχνιδιού
 GenericName[fr]=Moteur de jeu libre
