diff --git a/torbrowser_launcher/common.py b/torbrowser_launcher/common.py
index 5b387e7..1ed94f9 100644
--- a/torbrowser_launcher/common.py
+++ b/torbrowser_launcher/common.py
@@ -37,7 +37,7 @@ import re
 import gettext
 import gnupg
 
-SHARE = os.getenv('TBL_SHARE', sys.prefix + '/share') + '/torbrowser-launcher'
+SHARE = os.getenv('TBL_SHARE', '/app' + '/share') + '/torbrowser-launcher'
 
 gettext.install('torbrowser-launcher')
 
