diff --git a/src/main.cpp b/src/main.cpp
index 6346ab9..293e0d1 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -93,6 +93,7 @@ bool initialiseQv2ray()
     QStringList configFilePaths;
     configFilePaths << QDir::homePath() + "/.qv2ray" QV2RAY_CONFIG_DIR_SUFFIX;
     configFilePaths << QDir::homePath() + "/.config/qv2ray" QV2RAY_CONFIG_DIR_SUFFIX;
+    configFilePaths << QDir::homePath() + "/.var/app/com.github.Qv2ray/config/qv2ray" QV2RAY_CONFIG_DIR_SUFFIX;
     configFilePaths << QApplication::applicationDirPath() + "/config" QV2RAY_CONFIG_DIR_SUFFIX;
     QString configPath = "";
     //
