--- ./robocode-setup/robocode.sh	2019-08-31 19:41:10.000000000 +0100
+++ ./robocode.sh	2019-12-22 22:24:39.300829773 +0000
@@ -9,5 +9,6 @@

 pwd=`pwd`
 cd "${0%/*}"
-java -Xmx512M -cp libs/robocode.jar -XX:+IgnoreUnrecognizedVMOptions "--add-opens=java.base/sun.net.www.protocol.jar=ALL-UNNAMED" "--add-opens=java.base/java.lang.reflect=ALL-UNNAMED" "--add-opens=java.desktop/javax.swing.text=ALL-UNNAMED" "--add-opens=java.desktop/sun.awt=ALL-UNNAMED" robocode.Robocode $*
+java -DWORKINGDIRECTORY=$XDG_DATA_HOME \
+     -Xmx512M -cp libs/robocode.jar -XX:+IgnoreUnrecognizedVMOptions "--add-opens=java.base/sun.net.www.protocol.jar=ALL-UNNAMED" "--add-opens=java.base/java.lang.reflect=ALL-UNNAMED" "--add-opens=java.desktop/javax.swing.text=ALL-UNNAMED" "--add-opens=java.desktop/sun.awt=ALL-UNNAMED" robocode.Robocode $*
 cd "${pwd}"
