diff -u ./Project/QtCreator/old/brew.pri ./Project/QtCreator/brew.pri
--- ./Project/QtCreator/old/brew.pri	2017-08-15 13:05:18.000000000 +0200
+++ ./Project/QtCreator/brew.pri	2018-01-21 02:53:01.121327699 +0100
@@ -1,27 +1,49 @@
 USE_BREW = $$(QCTOOLS_USE_BREW)
+USE_SYSTEM = $${QCTOOLS_USE_SYSTEM}
 
 QCTOOLS_USE_BREW_NOT_EMPTY = false
+QCTOOLS_USE_SYSTEM_NOT_EMPTY = false
 
 !isEmpty(USE_BREW) {
     message("USE_BREW not empty")
     QCTOOLS_USE_BREW_NOT_EMPTY = true
 }
 
+!isEmpty(USE_SYSTEM) {
+    message("USE_SYSTEM not empty")
+    QCTOOLS_USE_SYSTEM_NOT_EMPTY = true
+}
+
 QCTOOLS_USE_BREW_EQUALS_TRUE = false
+QCTOOLS_USE_SYSTEM_EQUALS_TRUE = false
 
 equals(USE_BREW, true) {
     message("USE_BREW equals true")
     QCTOOLS_USE_BREW_EQUALS_TRUE = true
 }
 
+
+equals(USE_SYSTEM, true) {
+    message("USE_SYSTEM equals true")
+    QCTOOLS_USE_SYSTEM_EQUALS_TRUE = true
+}
+
 message("QCTOOLS_USE_BREW_NOT_EMPTY = " $$QCTOOLS_USE_BREW_NOT_EMPTY )
 message("QCTOOLS_USE_BREW_EQUALS_TRUE = " $$QCTOOLS_USE_BREW_EQUALS_TRUE )
 
+message("QCTOOLS_USE_SYSTEM_NOT_EMPTY = " $$QCTOOLS_USE_SYSTEM_NOT_EMPTY )
+message("QCTOOLS_USE_SYSTEM_EQUALS_TRUE = " $$QCTOOLS_USE_SYSTEM_EQUALS_TRUE )
+
 !isEmpty(USE_BREW):equals(USE_BREW, true) {
     message("DEFINES += USE_BREW")
     DEFINES += USE_BREW
 }
 
+!isEmpty(USE_SYSTEM):equals(USE_SYSTEM, true) {
+    message("DEFINES += USE_SYSTEM")
+    DEFINES += USE_SYSTEM
+}
+
 macx:contains(DEFINES, USE_BREW) {
 
     message("use brew")
@@ -34,6 +56,13 @@
 
     CONFIG += link_pkgconfig
 
+} else:unix:contains(DEFINES, USE_SYSTEM) {
+
+    message("use system libraries")
+
+    PKGCONFIG += Qt5Qwt6
+
+    CONFIG += link_pkgconfig
 } else {
     CONFIG += debug_and_release
 }
diff -u ./Project/QtCreator/old/ffmpeg.pri ./Project/QtCreator/ffmpeg.pri
--- ./Project/QtCreator/old/ffmpeg.pri	2017-08-15 13:05:19.000000000 +0200
+++ ./Project/QtCreator/ffmpeg.pri	2018-01-21 02:45:55.037541100 +0100
@@ -1,13 +1,12 @@
 DEFINES += __STDC_CONSTANT_MACROS
 
-macx:contains(DEFINES, USE_BREW) {
-    message("use ffmpeg from brew")
+macx:contains(DEFINES, USE_BREW)|unix:contains(DEFINES, USE_SYSTEM) {
+    message("use ffmpeg from brew/pkg-config")
 
-    PKGCONFIG += libavdevice libavcodec libavfilter libavformat libpostproc
+    PKGCONFIG += libavcodec libavfilter libavformat libpostproc
     PKGCONFIG += libswresample libswscale libavcodec libavutil
 
     CONFIG += link_pkgconfig
-
 } else {
     FFMPEG_PATH = $$absolute_path($$PWD/../../../ffmpeg)
     message("add external ffmpeg " $$FFMPEG_PATH )

