diff -ruN a/Rendering/RayTracing/CMakeLists.txt b/Rendering/RayTracing/CMakeLists.txt
--- a/Rendering/RayTracing/CMakeLists.txt	2020-06-26 20:24:40.000000000 +0700
+++ b/Rendering/RayTracing/CMakeLists.txt	2021-04-20 01:44:46.681054846 +0700
@@ -57,7 +57,7 @@
 if (VTK_ENABLE_OSPRAY)
   vtk_module_find_package(
     PACKAGE ospray
-    VERSION 1.8)
+    VERSION 2.5)
 
   vtk_module_link(VTK::RenderingRayTracing
     PUBLIC
