diff --git a/src/paths.c b/src/paths.c
index 7548115..e22cd51 100644
--- a/src/paths.c
+++ b/src/paths.c
@@ -240,44 +240,10 @@ static void Paths_InitHomeDirs(void)
 	{
 		/* $HOME not set, so let's use current working dir as home */
 		sUserHomeDir = Str_Dup(sWorkingDir);
-		sHatariHomeDir = Str_Dup(sWorkingDir);
-		return;
-	}
-
-	sHatariHomeDir = Str_Alloc(strlen(sUserHomeDir) + 1 + strlen(HATARI_HOME_DIR));
-
-	/* Try to use a private hatari directory in the users home directory */
-	sprintf(sHatariHomeDir, "%s%c%s", sUserHomeDir, PATHSEP, HATARI_HOME_DIR);
-	if (File_DirExists(sHatariHomeDir))
-	{
-		return;
-	}
-	/* Try legacy location ~/.hatari */
-	sprintf(sHatariHomeDir, "%s%c.hatari", sUserHomeDir, PATHSEP);
-	if (File_DirExists(sHatariHomeDir))
-	{
-		return;
 	}
 
-	/* Hatari home directory does not exists yet...
-	 * ... so let's try to create it: */
-#if !defined(__MACOSX__) && !defined(WIN32)
-	sprintf(sHatariHomeDir, "%s%c.config", sUserHomeDir, PATHSEP);
-	if (!File_DirExists(sHatariHomeDir))
-	{
-		/* ~/.config does not exist yet, create it first */
-		if (mkdir(sHatariHomeDir, 0700) != 0)
-		{
-			perror("Failed to create ~/.config directory");
-		}
-	}
-#endif
-	sprintf(sHatariHomeDir, "%s%c%s", sUserHomeDir, PATHSEP, HATARI_HOME_DIR);
-	if (mkdir(sHatariHomeDir, 0750) != 0)
-	{
-		/* Failed to create, so use user's home dir instead */
-		strcpy(sHatariHomeDir, sUserHomeDir);
-	}
+	sHatariHomeDir = Str_Dup(getenv("XDG_CONFIG_HOME"));
+	printf("home dir: %s\n", sHatariHomeDir);
 }
 
 
