Subject: [PATCH] Use download dir as default
---
Index: ui-gtk/UgtkApp.c
<+>UTF-8
===================================================================
diff --git a/ui-gtk/UgtkApp.c b/ui-gtk/UgtkApp.c
--- a/ui-gtk/UgtkApp.c	(revision 55cd9e831dc30a4a40d4006535af1238d568e911)
+++ b/ui-gtk/UgtkApp.c	(date 1705314900574)
@@ -1775,7 +1775,7 @@
 	cnode = uget_node_new (NULL);
 	common = ug_info_realloc (cnode->info, UgetCommonInfo);
 	common->name = ug_strdup_printf ("%s %d", _("New"), counts++);
-	common->folder = ug_strdup (g_get_home_dir ());
+	common->folder = ug_strdup (g_get_user_special_dir (G_USER_DIRECTORY_DOWNLOAD));
 	category = ug_info_realloc (cnode->info, UgetCategoryInfo);
 	*(char**)ug_array_alloc (&category->schemes, 1) = ug_strdup ("ftps");
 	*(char**)ug_array_alloc (&category->schemes, 1) = ug_strdup ("magnet");
