diff --git a/Quake/sys_sdl_unix.c b/Quake/sys_sdl_unix.c
index 58f7316b..0ddbc562 100644
--- a/Quake/sys_sdl_unix.c
+++ b/Quake/sys_sdl_unix.c
@@ -272,17 +272,7 @@ static void Sys_GetUserdir (char *dst, size_t dstsize)
 {
 	size_t		n;
 	const char	*home_dir = NULL;
-	struct passwd	*pwent;
-
-	pwent = getpwuid( getuid() );
-	if (pwent == NULL)
-		perror("getpwuid");
-	else
-		home_dir = pwent->pw_dir;
-	if (home_dir == NULL)
-		home_dir = getenv("HOME");
-	if (home_dir == NULL)
-		Sys_Error ("Couldn't determine userspace directory");
+	home_dir = getenv("XDG_DATA_HOME");
 
 /* what would be a maximum path for a file in the user's directory...
  * $HOME/SYS_USERDIR/game_dir/dirname1/dirname2/dirname3/filename.ext
@@ -293,6 +283,7 @@ static void Sys_GetUserdir (char *dst, size_t dstsize)
 		Sys_Error ("Insufficient array size for userspace directory");
 
-	q_snprintf (dst, dstsize, "%s/%s", home_dir, SYS_USERDIR);
+	q_snprintf (dst, dstsize, home_dir);
+	Sys_Printf("User directory is %s\n", dst);
 }
 #endif	/* PLATFORM_HAIKU */
 #endif	/* DO_USERDIRS */
