--- a/capbattleship/__main__.py	2020-11-24 00:25:52.153810719 +0100
+++ b/capbattleship/__main__.py	2020-11-24 00:27:33.274223963 +0100
@@ -17,7 +17,7 @@
 def main():
     # system path for assets and locales, used if equivalent directories are not
     # present with this file
-    sysprefix = "/usr"
+    sysprefix = "/app"
     xdgprefix = os.getenv('XDG_DATA_HOME', os.path.expanduser('~/.local/share'))
     binprefix = os.path.split(os.path.abspath(sys.argv[0]))[0]
 
