--- dvd+rw-booktype.cpp
+++ dvd+rw-booktype.cpp	2006/11/15 11:44:45
@@ -398,7 +398,7 @@
 		break;
 	case OPT_MEDIA:
 		if (profile!=0x1A && profile!=0x14 && profile!=13)
-		{   fprintf (stderr,":-( action is applicable to DVD±RW only\n");
+		{   fprintf (stderr,":-( action is applicable to DVDÂ±RW only\n");
 		    break;
 		}
 
--- rpl8.cpp
+++ rpl8.cpp	2006/11/15 11:44:45
@@ -288,7 +288,7 @@
 	    break;
 	}
 
-	fprintf (stderr,"\"%s\": ¡POINT OF NO RETURN!\n",argv[1]);
+	fprintf (stderr,"\"%s\": Â¡POINT OF NO RETURN!\n",argv[1]);
 	poll(NULL,0,3000);
 	signal (SIGINT,SIG_IGN);
 	signal (SIGTERM,SIG_IGN);
@@ -324,7 +324,7 @@
 	    {	fprintf (stderr,"\"%s\": firmware upgrade succeeded:-)\n",
 				argv[1]),
 		fprintf (stderr,"\"%s\": "
-				"¡CYCLE THE POWER BEFORE PROCEEDING!\n",
+				"Â¡CYCLE THE POWER BEFORE PROCEEDING!\n",
 				argv[1]);
 		break;
 	    }
