From 8823b470303fac6ac66d27b7943f4ead717e1584 Mon Sep 17 00:00:00 2001
From: tinywrkb <tinywrkb@gmail.com>
Date: Sun, 18 Apr 2021 02:17:45 +0300
Subject: [PATCH] systray icon: use reverse dns naming convention

---
 virtManager/systray.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/virtManager/systray.py b/virtManager/systray.py
index 4c4e7d29..8e5231b6 100644
--- a/virtManager/systray.py
+++ b/virtManager/systray.py
@@ -98,7 +98,7 @@ class _SystrayIndicator(_Systray):  # pragma: no cover
     """
     def __init__(self):
         self._icon = AppIndicator3.Indicator.new(
-                "virt-manager", "virt-manager",
+                "virt-manager", "org.virt_manager.virt-manager",
                 AppIndicator3.IndicatorCategory.APPLICATION_STATUS)
 
     def set_menu(self, menu):
@@ -129,7 +129,7 @@ class _SystrayStatusIcon(_Systray):  # pragma: no cover
     """
     def __init__(self):
         self._icon = Gtk.StatusIcon()
-        self._icon.set_property("icon-name", "virt-manager")
+        self._icon.set_property("icon-name", "org.virt_manager.virt-manager")
         self._icon.connect("activate", _toggle_manager)
         self._icon.connect("popup-menu", self._popup_cb)
         self._icon.set_tooltip_text(_("Virtual Machine Manager"))
-- 
2.31.1

