diff --git a/src/modules/bank.c b/src/modules/bank.c
index 2e67a0d..1e106e9 100644
--- a/src/modules/bank.c
+++ b/src/modules/bank.c
@@ -273,16 +273,6 @@ static int AllocatePluginFile (module_bank_t *bank, const char *abspath,
     if (bank->mode & CACHE_READ_FILE)
     {
         plugin = vlc_cache_lookup(&bank->cache, relpath);
-
-        if (plugin != NULL
-         && (plugin->mtime != (int64_t)st->st_mtime
-          || plugin->size != (uint64_t)st->st_size))
-        {
-            msg_Err(bank->obj, "stale plugins cache: modified %s",
-                    plugin->abspath);
-            vlc_plugin_destroy(plugin);
-            plugin = NULL;
-        }
     }
 
     if (plugin == NULL)