--- a/share/Makefile.in
+++ b/share/Makefile.in
@@ -997,7 +997,7 @@
 	lua/sd/icecast.lua lua/sd/icast.lua lua/sd/jamendo.lua \
 	lua/sd/metachannels.lua
 appdatadir = $(datadir)/applications
-@HAVE_WIN32_FALSE@appdata_DATA = vlc.desktop
+@HAVE_WIN32_FALSE@appdata_DATA = org.videolan.VLC.desktop
 iconsdatadir = $(datadir)/icons/hicolor
 icons16dir = $(iconsdatadir)/16x16/apps
 icons32dir = $(iconsdatadir)/32x32/apps
@@ -1011,7 +1011,7 @@
 
 @HAVE_WIN32_FALSE@dist_icons48_DATA = icons/48x48/vlc.png icons/48x48/vlc-xmas.png
 @HAVE_WIN32_FALSE@dist_icons128_DATA = icons/128x128/vlc.png icons/128x128/vlc-xmas.png
-@HAVE_WIN32_FALSE@dist_icons256_DATA = icons/256x256/vlc.png
+@HAVE_WIN32_FALSE@dist_icons256_DATA = icons/256x256/vlc.png org.videolan.VLC.png
 @HAVE_WIN32_FALSE@nobase_dist_vlcdata_SCRIPTS = \
 @HAVE_WIN32_FALSE@	utils/gnome-vlc-default.sh \
 @HAVE_WIN32_FALSE@	utils/audio-vlc-default.sh \
@@ -1678,11 +1678,14 @@
 	uninstall-soliddataDATA
 
 
-vlc.desktop: vlc.desktop.in $(top_builddir)/config.status
+org.videolan.VLC.desktop: vlc.desktop.in $(top_builddir)/config.status
 	$(AM_V_GEN)sed -e 's,\@bindir\@,$(bindir),g' < "$<" > tmp-$@
 	$(AM_V_at)$(DESKTOP_FILE_VALIDATE) tmp-$@
 	$(AM_V_at)mv -f tmp-$@ $@
 
+org.videolan.VLC.png: icons/256x256/vlc.png
+	$(AM_V_at)cp $< $@
+
 skins2/default.vlt: $(skins2_default_vlt_FILES)
 	mkdir -p skins2
 	(cd "$(srcdir)/skins2"; find default -print0 | \
--- a/share/vlc.desktop.in
+++ b/share/vlc.desktop.in
@@ -84,7 +84,7 @@
 Comment[zh_CN]=为您读取、捕获或发送多媒体流
 Exec=@bindir@/vlc --started-from-file %U
 TryExec=@bindir@/vlc
-Icon=vlc
+Icon=org.videolan.VLC
 Terminal=false
 Type=Application
 Categories=AudioVideo;Player;Recorder;
