diff --git a/source/vst/hosting/module_linux.cpp b/source/vst/hosting/module_linux.cpp
index df085c7..ec375c8 100644
--- a/public.sdk/source/vst/hosting/module_linux.cpp
+++ b/public.sdk/source/vst/hosting/module_linux.cpp
@@ -43,7 +43,7 @@
 #include <sys/utsname.h>
 #include <unistd.h>
 
-#if (__cplusplus >= 201707L)
+#if (__cplusplus >= 201703L)
 #if __has_include(<filesystem>)
 #define USE_EXPERIMENTAL_FS 0
 #elif __has_include(<experimental/filesystem>)
diff --git a/modules/SMTG_PlatformToolset.cmake b/modules/SMTG_PlatformToolset.cmake
index 49ff310..c4239bf 100644
--- a/cmake/modules/SMTG_PlatformToolset.cmake
+++ b/cmake/modules/SMTG_PlatformToolset.cmake
@@ -60,7 +60,7 @@ macro(smtg_setup_platform_toolset)
                     set(CMAKE_ANDROID_STL_TYPE c++_static)
                     link_libraries(dl)
                  else()
-                    link_libraries(stdc++fs pthread dl)
+                    link_libraries(pthread dl)
                 endif(ANDROID)
             endif(SMTG_MAC)
         endif(XCODE)
