--- o/src/vte.c 2018-05-03 20:53:16.000000000 +0200
+++ n/src/vte.c 2018-05-03 20:57:48.764010197 +0200
@@ -7513,17 +7513,10 @@ 
 				event->button, x, y);
 		switch (event->button) {
 		case 1:
-			/* If Shift is held down, or we're not in events mode,
-			 * copy the selected text. */
-			if ((terminal->pvt->modifiers & GDK_SHIFT_MASK) ||
-			    !terminal->pvt->mouse_tracking_mode)
-				handled = _vte_terminal_maybe_end_selection (terminal);
+			handled = _vte_terminal_maybe_end_selection (terminal);
 			break;
 		case 2:
-			if ((terminal->pvt->modifiers & GDK_SHIFT_MASK) ||
-			    !terminal->pvt->mouse_tracking_mode) {
-				handled = TRUE;
-			}
+			handled = TRUE;
 			break;
 		case 3:
 		default:
