commit 8cff105a2b280ed738b296955724590b9cd348ce
commit ef34977b6765be1036ea8cd016861199033e2231
commit 49e3433dea9a32d7ab9972b719484b49ca7f6b81
commit 94b6ca416d4e5b54be084a057ec1341bcfddabe1
commit ecb9fcc8cfb73207bcda93889e3e2564d41f79c9

--- origsrc/vte-0.28.2/src/pty.c	2011-08-16 16:52:48.000000000 -0500
+++ src/vte-0.28.2/src/pty.c	2015-08-03 14:01:55.151226800 -0500
@@ -616,7 +616,7 @@ __vte_pty_fork(VtePty *pty,
  * @pty: a #VtePty
  * @rows: the desired number of rows
  * @columns: the desired number of columns
- * @error: (allow-none); return location to store a #GError, or %NULL
+ * @error: (allow-none): return location to store a #GError, or %NULL
  *
  * Attempts to resize the pseudo terminal's window size.  If successful, the
  * OS kernel will send #SIGWINCH to the child process group.
--- origsrc/vte-0.28.2/src/vte.c	2015-08-03 13:56:24.348720200 -0500
+++ src/vte-0.28.2/src/vte.c	2015-08-03 16:15:52.185800500 -0500
@@ -2474,7 +2474,7 @@ _vte_terminal_set_pointer_visible(VteTer
  *
  * Creates a new terminal widget.
  *
- * Returns: (transfer full) (type Vte.Terminal): a new #VteTerminal object
+ * Returns: (transfer none) (type Vte.Terminal): a new #VteTerminal object
  */
 GtkWidget *
 vte_terminal_new(void)
@@ -3680,7 +3680,7 @@ vte_terminal_watch_child (VteTerminal *t
  * Gets the user's shell, or %NULL. In the latter case, the
  * system default (usually "/bin/sh") should be used.
  *
- * Returns: (tranfer full) (type filename): a newly allocated string with the
+ * Returns: (transfer full) (type filename): a newly allocated string with the
  *   user's shell, or %NULL
  *
  * Since: 0.28
@@ -4615,8 +4615,8 @@ out:
 /**
  * vte_terminal_feed:
  * @terminal: a #VteTerminal
- * @data: a string in the terminal's current encoding
- * @length: the length of the string
+ * @data: (array length=length) (element-type guint8): a string in the terminal's current encoding
+ * @length the length of the string
  *
  * Interprets @data as if it were data received from a child process.  This
  * can either be used to drive the terminal without a child process, or just
@@ -6233,7 +6233,7 @@ vte_terminal_copy_cb(GtkClipboard *clipb
  * @start_col: first column to search for data
  * @end_row: last row to search for data
  * @end_col: last column to search for data
- * @is_selected: a #VteSelectionFunc callback
+ * @is_selected: (scope call): a #VteSelectionFunc callback
  * @user_data: (closure): user data to be passed to the callback
  * @attributes: (out) (transfer full) (array) (element-type Vte.CharAttributes): location for storing text attributes
  *
@@ -6432,7 +6432,7 @@ vte_terminal_get_text_maybe_wrapped(VteT
 /**
  * vte_terminal_get_text:
  * @terminal: a #VteTerminal
- * @is_selected: a #VteSelectionFunc callback
+ * @is_selected: (scope call): a #VteSelectionFunc callback
  * @user_data: (closure): user data to be passed to the callback
  * @attributes: (out) (transfer full) (array) (element-type Vte.CharAttributes): location for storing text attributes
  *
@@ -6462,7 +6462,7 @@ vte_terminal_get_text(VteTerminal *termi
 /**
  * vte_terminal_get_text_include_trailing_spaces:
  * @terminal: a #VteTerminal
- * @is_selected: a #VteSelectionFunc callback
+ * @is_selected: (scope call): a #VteSelectionFunc callback
  * @user_data: (closure): user data to be passed to the callback
  * @attributes: (out) (transfer full) (array) (element-type Vte.CharAttributes): location for storing text attributes
  *
@@ -6497,7 +6497,7 @@ vte_terminal_get_text_include_trailing_s
  * vte_terminal_get_cursor_position:
  * @terminal: a #VteTerminal
  * @column: (out) (allow-none): a location to store the column, or %NULL
- * @row : (out) (allow-none): a location to store the row, or %NULL
+ * @row: (out) (allow-none): a location to store the row, or %NULL
  *
  * Reads the location of the insertion cursor and returns it.  The row
  * coordinate is absolute.
