--- orig/CMakeLists.txt
+++ CMakeLists.txt
@@ -59,10 +59,6 @@
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-# change these path to yours
-include_directories("/home/nihui/osd/ncnn/build/install/include")
-link_directories("/home/nihui/osd/ncnn/build/install/lib")
-
 # look for vulkan compute shader and compile
 set(SHADER_SPV_HEX_FILES)
 
@@ -76,3 +72,7 @@
 add_dependencies(waifu2x-ncnn-vulkan generate-spirv)
 
 target_link_libraries(waifu2x-ncnn-vulkan ncnn ${Vulkan_LIBRARY} ${OpenMP_CXX_LIBRARIES})
+
+install(TARGETS waifu2x-ncnn-vulkan DESTINATION bin)
+install(DIRECTORY models-cunet DESTINATION share)
+install(DIRECTORY models-upconv_7_anime_style_art_rgb DESTINATION share)
