--- orig/CMakeLists.txt
+++ CMakeLists.txt
@@ -63,10 +63,6 @@
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
 
-# change these path to yours
-include_directories("/home/nihui/osd/ncnn/build/install/include")
-link_directories("/home/nihui/osd/ncnn/build/install/lib")
-
 # look for vulkan compute shader and compile
 set(SHADER_SPV_HEX_FILES)
 
@@ -80,3 +76,7 @@
 add_dependencies(waifu2x-ncnn-vulkan generate-spirv)
 
 target_link_libraries(waifu2x-ncnn-vulkan ncnn ${Vulkan_LIBRARY} ${OpenMP_CXX_LIBRARIES})
+
+install(TARGETS waifu2x-ncnn-vulkan DESTINATION bin)
+install(DIRECTORY models-cunet DESTINATION share)
+install(DIRECTORY models-upconv_7_anime_style_art_rgb DESTINATION share)
