diff -up warmux-11.04/lib/warmux/action/action.cpp.gcc47 warmux-11.04/lib/warmux/action/action.cpp
--- warmux-11.04/lib/warmux/action/action.cpp.gcc47	2011-04-28 21:03:14.000000000 +0200
+++ warmux-11.04/lib/warmux/action/action.cpp	2012-02-27 11:20:35.571676046 +0100
@@ -85,7 +85,7 @@ Action::Action(const char *buffer, Dista
 {
   m_creator = _creator;
 
-  m_header.len = SDLNet_Read32(buffer);
+  m_header.len = SDLNet_Read32((void *)buffer);
   ASSERT(m_header.len >= sizeof(Header));
   buffer += 4;
   // All of the following could be skipped for the actions we now,
@@ -225,7 +225,7 @@ int Action::PopInt()
     return 0;
   }
 
-  uint32_t tmp = SDLNet_Read32(m_read); m_read += 4;
+  uint32_t tmp = SDLNet_Read32((void *)m_read); m_read += 4;
 
   return *((int32_t*)&tmp);
 }
@@ -247,9 +247,9 @@ Double Action::PopDouble()
   static const uint len = sizeof(fp::fint_t)/4;
   for (uint i=0; i<len; i++) {
 #  if SDL_BYTEORDER == SDL_LIL_ENDIAN
-    tmp.u32[i] = SDLNet_Read32(m_read);
+    tmp.u32[i] = SDLNet_Read32((void *)m_read);
 #  else
-    tmp.u32[len-1-i] = SDLNet_Read32(m_read);
+    tmp.u32[len-1-i] = SDLNet_Read32((void* )m_read);
 #  endif
     m_read += 4;
   }
diff -up warmux-11.04/tools/list_games/main.cpp.gcc47 warmux-11.04/tools/list_games/main.cpp
--- warmux-11.04/tools/list_games/main.cpp.gcc47	2011-04-28 21:03:12.000000000 +0200
+++ warmux-11.04/tools/list_games/main.cpp	2012-02-27 11:31:33.764447500 +0100
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <unistd.h>
 #include <WARMUX_types.h>
 #include <WARMUX_network.h>
 #include <WARMUX_index_server.h>
