diff -up warmux-11.04/src/graphic/surface.cpp.zlib warmux-11.04/src/graphic/surface.cpp
--- warmux-11.04/src/graphic/surface.cpp.zlib	2011-04-28 21:03:06.000000000 +0200
+++ warmux-11.04/src/graphic/surface.cpp	2012-01-16 10:34:37.371319039 +0100
@@ -25,6 +25,7 @@
 #include <SDL_image.h>
 #include <SDL_rotozoom.h>
 #include <png.h>
+#include <zlib.h>
 
 #include "graphic/surface.h"
 #include "tool/math_tools.h"
