--- src/interface/weapon_menu.cpp.orig	2016-03-10 09:50:29.415699594 -0600
+++ src/interface/weapon_menu.cpp	2016-03-10 09:51:02.384699238 -0600
@@ -391,7 +391,7 @@
 Weapon * WeaponsMenu::UpdateCurrentOverflyItem(const Polygon * poly)
 {
   if (!show)
-    return false;
+    return 0;
   const std::vector<PolygonItem *>& items = poly->GetItem();
   WeaponMenuItem * tmp;
   Interface::GetInstance()->SetCurrentOverflyWeapon(NULL);
