--- sameboy/Makefile-orig       2022-08-28 11:13:04.326979227 -0400
+++ sameboy/Makefile    2022-08-28 11:13:15.514826393 -0400
@@ -112,7 +112,7 @@
 endif

 # These must come before the -Wno- flags
-WARNINGS += -Werror -Wall -Wno-unknown-warning -Wno-unknown-warning-option -Wno-missing-braces
+WARNINGS += -Wall -Wno-unknown-warning -Wno-unknown-warning-option -Wno-missing-braces
 WARNINGS += -Wno-nonnull -Wno-unused-result -Wno-strict-aliasing -Wno-multichar -Wno-int-in-bool-context -Wno-format-truncation

 # Only add this flag if the compiler supports it
