From 1805861274019beb996391828ed800abd794bef0 Mon Sep 17 00:00:00 2001
From: Roeland Jansen <roeland.jansen69@gmail.com>
Date: Sun, 13 Aug 2023 21:09:00 +0200
Subject: [PATCH] DERP! the channel spacing 6.125 is not half of 12.5 kHz.
 Fixed it.

---
 wfmain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wfmain.cpp b/wfmain.cpp
index 3af1e62..e480e57 100644
--- a/wfmain.cpp
+++ b/wfmain.cpp
@@ -1040,7 +1040,7 @@ void wfmain::setupMainUI()
     ui->tuningStepCombo->addItem("1 kHz",     (unsigned int)    1000);
     ui->tuningStepCombo->addItem("2.5 kHz",   (unsigned int)    2500);
     ui->tuningStepCombo->addItem("5 kHz",     (unsigned int)    5000);
-    ui->tuningStepCombo->addItem("6.125 kHz", (unsigned int)    6125);	// PMR
+    ui->tuningStepCombo->addItem("6.25 kHz",  (unsigned int)    6250);	// PMR
     ui->tuningStepCombo->addItem("8.333 kHz", (unsigned int)    8333);	// airband stepsize
     ui->tuningStepCombo->addItem("9 kHz",     (unsigned int)    9000);	// European medium wave stepsize
     ui->tuningStepCombo->addItem("10 kHz",    (unsigned int)   10000);
-- 
2.41.0

