diff -up widelands-build19-src-gcc7/CMakeLists.txt.ppc64le widelands-build19-src-gcc7/CMakeLists.txt
--- widelands-build19-src-gcc7/CMakeLists.txt.ppc64le	2017-06-20 10:33:11.000000000 +0200
+++ widelands-build19-src-gcc7/CMakeLists.txt	2018-02-20 23:01:39.680715037 +0100
@@ -102,7 +102,7 @@ else()
   message(FATAL_ERROR "Unknown CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 endif()
 
-wl_add_flag(WL_GENERIC_CXX_FLAGS "-std=c++11")
+wl_add_flag(WL_GENERIC_CXX_FLAGS "-std=gnu++11")
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Weverything")
