diff -Naur a/src/graphic/text/bidi.h b/src/graphic/text/bidi.h
--- a/src/graphic/text/bidi.h	2018-05-16 04:41:22.888150519 +0200
+++ b/src/graphic/text/bidi.h	2018-05-16 04:41:48.123805851 +0200
@@ -40,7 +40,7 @@
 // True if a string contains a character from the script's code blocks
 bool has_script_character(const char* input, UI::FontSets::Selector script);
 UI::FontSet const* find_fontset(const char* word, const UI::FontSets& fontsets);
-std::string icustring2string(const UnicodeString& convertme);
+std::string icustring2string(const icu::UnicodeString& convertme);
 std::string icuchar2string(const UChar& convertme);
 bool cannot_start_line(const UChar& c);
 bool cannot_end_line(const UChar& c);
