diff --git a/src/windstille_main.cxx b/src/windstille_main.cxx
index 7f6b5208..3080c936 100644
--- a/src/windstille_main.cxx
+++ b/src/windstille_main.cxx
@@ -172,7 +172,7 @@ WindstilleMain::main(int argc, char** argv)
   bindir  = CL_System::get_exe_path();
 
   if (datadir.empty())
-    datadir = bindir + "../data/";
+    datadir = bindir + "../share/windstille-0.2/";
 
 #ifndef WIN32
   char* home_c = getenv("HOME");
