diff --git a/gdk/glue/windowmanager.c b/gdk/glue/windowmanager.c
index 484d713b7..854d84031 100644
--- a/gdk/glue/windowmanager.c
+++ b/gdk/glue/windowmanager.c
@@ -110,7 +110,7 @@ gtksharp_get_gdk_net_client_list (int *count)
 	list = g_malloc (*count * sizeof (gpointer));
 	/* Put all of the windows into a GList to return */
 	for (i = 0; i < *count; i ++) {
-		list [i] = data [i];
+		list [i] = (gpointer*) data [i];
 		g_message ("WinID: %d", list [i]);
 	}
 
