From a42ae370b6e4f2e0824987c9161ce359588c443f Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Sun, 29 Dec 2019 17:05:27 -0300
Subject: [PATCH 1/2] Update to espeak-ng 1.49.2

---
 wordsactivity.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wordsactivity.py b/wordsactivity.py
index 6bcc408..38093f6 100644
--- a/wordsactivity.py
+++ b/wordsactivity.py
@@ -66,7 +66,7 @@ _ESPEAK_TO_NEW_LANG_CODE = {
     'hindi': 'hi',
     'italian': 'it',
     'brazil': 'pt_br',
-    'spanish-latin-am': 'es_419'
+    'spanish-latin-am': 'es'
 }
 
 
@@ -545,7 +545,7 @@ class WordsActivity(SugarCompatibleActivity):
                         dictdmodel.espeak_voices[lang]]
                 )
             except KeyError:
-                speech_manager.say_text(text=text, lang_code='en')
+                speech_manager.say_text(text=text, lang_code='en_us')
 
     def __suggestion_selected_cb(self, treeview):
         selection = treeview.get_selection()
-- 
2.23.0

