From 10e52292b4f89d2c00be2989cb391b9660558e78 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Sun, 29 Dec 2019 17:01:31 -0300
Subject: [PATCH 2/3] info

---
 activity/activity.info | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/activity/activity.info b/activity/activity.info
index f1ea876..fd5349d 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,11 +1,19 @@
 [Activity]
 name = Words
-bundle_id = org.laptop.Words
+bundle_id = org.sugarlabs.Words
 exec = sugar-activity3 wordsactivity.WordsActivity
 icon = activity-icon
-activity_version = 23
+activity_version = 23.1
+release_date = 2019-12-12
 show_launcher = yes
 license = GPLv3+
-summary = Decode the world and help the world decode you. Use this translator for access and learn about many different languages!
+metadata_license = CC0-1.0
+summary = A multilingual offline dictionary
+description = Decode the world and help the world decode you. Use this translator for access and learn about many different languages!
 repository = https://github.com/sugarlabs/words-activity
+url = https://github.com/sugarlabs/words-activity
+tags = Education
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+screenshots = https://raw.githubusercontent.com/sugarlabs/words-activity/master/screenshots/1.png https://raw.githubusercontent.com/sugarlabs/words-activity/master/screenshots/2.png
 max_participants = 1
-- 
2.23.0

