From af82741730b93ba5e05d9eebb4f7ab5291bf6877 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Sun, 29 Dec 2019 17:01:04 -0300
Subject: [PATCH 1/3] port

---
 wordsactivity.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/wordsactivity.py b/wordsactivity.py
index ecedd61..6bcc408 100644
--- a/wordsactivity.py
+++ b/wordsactivity.py
@@ -53,6 +53,8 @@ import dictdmodel
 from roundbox import RoundBox
 from sugar3.speech import SpeechManager
 
+from sugarapp.widgets import SugarCompatibleActivity
+
 EMPTY_HTML = '<body bgcolor="#E5E5E5"></body>'
 _AUTOSEARCH_TIMEOUT = 1000
 _ESPEAK_TO_NEW_LANG_CODE = {
@@ -235,13 +237,13 @@ class FilterToolItem(Gtk.ToolButton):
         self.emit('changed', key)
 
 
-class WordsActivity(activity.Activity):
+class WordsActivity(SugarCompatibleActivity):
 
     """Words Activity as specified in activity.info"""
 
     def __init__(self, handle):
         """Set up the Words activity."""
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self._dictd_data_dir = './dictd/'
         self._dictionaries = dictdmodel.Dictionaries(self._dictd_data_dir)
-- 
2.23.0

