From 5733cad886e359701950e2e2a0f08787de68c517 Mon Sep 17 00:00:00 2001
From: Filippe LeMarchand <gasinvein@gmail.com>
Date: Mon, 26 Feb 2018 11:30:31 +0300
Subject: [PATCH] add missing <message> tag

---
 en_US/ts/ksomisc.ts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/en_US/ts/ksomisc.ts b/en_US/ts/ksomisc.ts
index cd2e75cf..91cb1c7c 100644
--- a/en_US/ts/ksomisc.ts
+++ b/en_US/ts/ksomisc.ts
@@ -882,10 +882,11 @@ The button icon, command group and task window position of each scheme are diffe
         <source>Microsoft PowerPoint 97-2003 Presentation</source>
         <translation type="unfinished"></translation>
     </message>
+    <message>
 	<source>Microsoft PowerPoint 97-2003 SlideShow</source>
         <translation type="unfinished"></translation>
     </message>
-	<message>
+    <message>
         <source>Microsoft PowerPoint 97-2003 Template</source>
         <translation type="unfinished"></translation>
     </message>
