From bba071753a2ba11fbd1fff0af4aed5a4b67c07b2 Mon Sep 17 00:00:00 2001
From: Filippe LeMarchand <gasinvein@gmail.com>
Date: Mon, 26 Feb 2018 11:44:16 +0300
Subject: [PATCH 1/2] use externally set $DEST var

---
 dev/make_install.sh | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/dev/make_install.sh b/dev/make_install.sh
index ff0541ee..4782c4bc 100755
--- a/dev/make_install.sh
+++ b/dev/make_install.sh
@@ -5,10 +5,12 @@ ICO=`cat lang.conf | sed -n '/Icon=/ s/Icon=//p'`
 
 source "$(dirname $0)/funcs.sh"
 
-if [ "x$USER" == "xroot" ]; then
-	DEST="$(root_install_path)/$LNG"
-else
-	DEST="$(user_install_path)/$LNG"
+if [ -z "$DEST" ]; then
+	if [ "x$USER" == "xroot" ]; then
+		DEST="$(root_install_path)/$LNG"
+	else
+		DEST="$(user_install_path)/$LNG"
+	fi
 fi
 
 function install_file

From 75a455debc181ec87d4352f62f21055e265cd36d Mon Sep 17 00:00:00 2001
From: Filippe LeMarchand <gasinvein@gmail.com>
Date: Mon, 26 Feb 2018 11:53:53 +0300
Subject: [PATCH 2/2] Update make_install.sh

---
 dev/make_install.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/dev/make_install.sh b/dev/make_install.sh
index 4782c4bc..5f7d926f 100755
--- a/dev/make_install.sh
+++ b/dev/make_install.sh
@@ -5,12 +5,14 @@ ICO=`cat lang.conf | sed -n '/Icon=/ s/Icon=//p'`
 
 source "$(dirname $0)/funcs.sh"
 
-if [ -z "$DEST" ]; then
+if [ -z "$DESTDIR" ]; then
 	if [ "x$USER" == "xroot" ]; then
 		DEST="$(root_install_path)/$LNG"
 	else
 		DEST="$(user_install_path)/$LNG"
 	fi
+else
+	DEST="$DESTDIR/mui/$LNG"
 fi
 
 function install_file
