--- wxGTK-2.8.12.orig/tests/test.cpp	2011-03-22 13:18:05.000000000 +0100
+++ wxGTK-2.8.12/tests/test.cpp	2011-04-14 10:35:44.000000000 +0200
@@ -86,11 +86,11 @@ void TestApp::OnInitCmdLine(wxCmdLinePar
         { wxCMD_LINE_SWITCH, _T("L"), _T("longlist"),
             _T("list the test cases, do not run them"),
             wxCMD_LINE_VAL_NONE, 0 },
-        { wxCMD_LINE_SWITCH, "d", "detail",
-            "print the test case names, run them (not implemented)",
+        { wxCMD_LINE_SWITCH, _T("d"), _T("detail"),
+            _T("print the test case names, run them (not implemented)"),
             wxCMD_LINE_VAL_NONE, 0 },
-        { wxCMD_LINE_SWITCH, "t", "timing",
-            "print names and mesure running time of individual test, run them (not implemented)",
+        { wxCMD_LINE_SWITCH, _T("t"), _T("timing"),
+            _T("print names and mesure running time of individual test, run them (not implemented)"),
             wxCMD_LINE_VAL_NONE, 0 },
         { wxCMD_LINE_PARAM, 0, 0, _T("REGISTRY"), wxCMD_LINE_VAL_STRING,
             wxCMD_LINE_PARAM_OPTIONAL | wxCMD_LINE_PARAM_MULTIPLE },
