diff -Naur wxmaxima-Version-18.12.0.orig/CMakeLists.txt wxmaxima-Version-18.12.0/CMakeLists.txt
--- wxmaxima-Version-18.12.0.orig/CMakeLists.txt	2018-12-25 10:03:08.000000000 +0100
+++ wxmaxima-Version-18.12.0/CMakeLists.txt	2018-12-29 10:17:48.438759968 +0100
@@ -1,5 +1,5 @@
 project(wxmaxima)
-set(VERSION 18.12.0-unofficial)
+set(VERSION 18.12.0)
 set(PACKAGE_VERSION ${VERSION})
 set(GITVERSION ${VERSION})
 set(PACKAGE wxMaxima)
