From 160959c0b79e5ca66ba2b98095ed77b75b97544a Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Sat, 4 May 2019 20:48:57 +0200
Subject: [PATCH] Another attempt to fix #1113

---
 src/Worksheet.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Worksheet.cpp b/src/Worksheet.cpp
index 7c63e4b9..1fb61585 100644
--- a/src/Worksheet.cpp
+++ b/src/Worksheet.cpp
@@ -405,6 +405,10 @@ void Worksheet::OnPaint(wxPaintEvent &WXUNUSED(event))
 #ifdef __WXGTK__
 #ifndef __WXGTK3__
   PrepareDC(dc);
+#else
+#if wxCHECK_VERSION(3, 1, 0)
+  PrepareDC(dc);
+#endif
 #endif
 #else
   PrepareDC(dc);
