From b7dfe83e7c4c37783049e0a1d293ab7d093ffcdc Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Sat, 4 May 2019 05:37:35 +0200
Subject: [PATCH] Call wxAutoBufferedPaintDC::SetPen() correctly

Closes #1121
---
 src/Worksheet.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Worksheet.cpp b/src/Worksheet.cpp
index 37fccf68..7c63e4b9 100644
--- a/src/Worksheet.cpp
+++ b/src/Worksheet.cpp
@@ -430,7 +430,7 @@ void Worksheet::OnPaint(wxPaintEvent &WXUNUSED(event))
   dc.Clear();
 #else
 #ifdef __WXGTK3__
-  dc.SetPen(wxTRANSPARENT_PEN);
+  dc.SetPen(*wxTRANSPARENT_PEN);
   dc.DrawRectangle(updateRegion);
 #else
   dc.Clear();
