diff -Naur wxmaxima-Version-19.01.2.orig/CMakeLists.txt wxmaxima-Version-19.01.2/CMakeLists.txt
--- wxmaxima-Version-19.01.2.orig/CMakeLists.txt	2019-01-22 10:48:06.000000000 +0100
+++ wxmaxima-Version-19.01.2/CMakeLists.txt	2019-01-24 07:07:35.477076979 +0100
@@ -1,5 +1,5 @@
 project(wxmaxima)
-set(VERSION 19.01.2x)
+set(VERSION 19.01.2)
 set(PACKAGE_VERSION ${VERSION})
 set(GITVERSION ${VERSION})
 set(PACKAGE wxMaxima)
