--- wxOil/camelot.cpp.orig	2007-11-07 03:48:25.000000000 +0300
+++ wxOil/camelot.cpp	2014-04-10 03:06:01.000000000 +0400
@@ -600,8 +600,10 @@
 bool CCamApp::OnInit()
 {
 	InInitOrDeInit = TRUE; // Don't allow the user to try carrying on working
+	#if wxUSE_ON_FATAL_EXCEPTION
 	::wxHandleFatalExceptions(TRUE);
 
+	#endif
 	//
 	// Parse command line. We do this early so we get flags which
 	// are useful for init, such as -u
@@ -1253,7 +1255,9 @@
 	// zap this after we know all windows have gone
 	wxWindowDeletionWatcher::DeInit();
 
+	#if wxUSE_ON_FATAL_EXCEPTION
 	::wxHandleFatalExceptions(FALSE);
+	#endif
 	return wxApp::OnExit();
 }
 
@@ -2569,7 +2573,9 @@
 		}
 
 		// switch fatal exception handling off
+		#if wxUSE_ON_FATAL_EXCEPTION
 		::wxHandleFatalExceptions(FALSE);
+		#endif
 		TRACE(_T("CCamApp::OnFatalException() called recursively - dying now"));
 		return; // this quits the app
 	}
@@ -2580,8 +2586,10 @@
 	do
 	{
 		// Ensure we are reinstated as the signal handler
+		#if wxUSE_ON_FATAL_EXCEPTION
 		::wxHandleFatalExceptions(FALSE);
 		::wxHandleFatalExceptions(TRUE);
+		#endif
 
 // This bit of code is currently not operative - looking into wxDebugReport instead
 #ifdef EXCEPTION_LOCATION
