From: Adam Sampson <unknown@bitbucket>
Date: Sun, 14 Oct 2018 14:04:18 +0200
Subject: detect512 is needed on all architectures

---
 source/common/cpu.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/source/common/cpu.cpp b/source/common/cpu.cpp
index 0681ff5..fa687da 100644
--- a/source/common/cpu.cpp
+++ b/source/common/cpu.cpp
@@ -110,6 +110,11 @@ const cpu_name_t cpu_names[] =
     { "", 0 },
 };
 
+bool detect512()
+{
+    return(enable512);
+}
+
 #if X265_ARCH_X86
 
 extern "C" {
@@ -123,10 +128,6 @@ uint64_t PFX(cpu_xgetbv)(int xcr);
 #pragma warning(disable: 4309) // truncation of constant value
 #endif
 
-bool detect512()
-{
-    return(enable512);
-}
 uint32_t cpu_detect(bool benableavx512 )
 {
 
