--- ./source/common/param.cpp.orig	2018-05-21 02:33:10.000000000 -0600
+++ ./source/common/param.cpp	2018-05-28 19:28:11.751321637 -0600
@@ -633,7 +633,7 @@ int x265_param_parse(x265_param* p, cons
         if (bValueWasNull)
             p->cpuid = atobool(value);
         else
-            p->cpuid = parseCpuName(value, bError);
+            p->cpuid = parseCpuName(value, bError, false);
 #endif
     }
     OPT("fps")
