Description: Add various linker flags
Author: Joachim Breitner <nomeata@debian.org>
Index: xaralx-0.7r1785/Makefile.am
===================================================================
--- xaralx-0.7r1785.orig/Makefile.am	2013-11-11 09:35:51.399145759 +0100
+++ xaralx-0.7r1785/Makefile.am	2013-11-11 09:36:59.767148675 +0100
@@ -46,7 +46,7 @@
     XaraLX_SOURCES	= Kernel/libKernel.a wxOil/libwxOil.a tools/libTools.a wxXtra/libwxXtra.a
     XaraLX_LDSOURCE = -Wl,--start-group $(XaraLX_SOURCES) -Wl,--end-group
 endif
-XaraLX_LDFLAGS		= --debug -L$(srcdir)/$(CDRAW_LIB_DIR) $(WX_LIBS) $(LIBS) $(LIBXML2_LIBS) -lCDraw
+XaraLX_LDFLAGS		= --debug -L$(srcdir)/$(CDRAW_LIB_DIR) $(WX_LIBS) $(GTK_LIBS) $(LIBS) $(LIBXML2_LIBS) -ljpeg -lpng -lCDraw -no-pie
 XaraLX_LD			?= $(CXX)
 
 XaraLX$(EXEEXT) : $(XaraLX_SOURCES)
