diff -Naur XaraLX-0.7r1785.orig/configure.in XaraLX-0.7r1785/configure.in
--- XaraLX-0.7r1785.orig/configure.in	2007-11-07 01:48:25.000000000 +0100
+++ XaraLX-0.7r1785/configure.in	2018-11-27 04:37:14.029990000 +0100
@@ -1,6 +1,7 @@
-AC_INIT(wxOil/camelot.cpp)
+AC_INIT([XaraLX], [0.7])
+AC_CONFIG_SRCDIR([wxOil/camelot.cpp])
 AC_CANONICAL_TARGET
-AM_INIT_AUTOMAKE(XaraLX, 0.7)
+AM_INIT_AUTOMAKE([subdir-objects])
 
 AC_PREFIX_PROGRAM( pkg-config )
 
diff -Naur XaraLX-0.7r1785.orig/filters/SVGFilter/configure.in XaraLX-0.7r1785/filters/SVGFilter/configure.in
--- XaraLX-0.7r1785.orig/filters/SVGFilter/configure.in	2007-11-07 01:48:04.000000000 +0100
+++ XaraLX-0.7r1785/filters/SVGFilter/configure.in	2018-11-27 16:03:35.319444866 +0100
@@ -1,5 +1,6 @@
-AC_INIT(svgfilter.cpp)
-AM_INIT_AUTOMAKE(SVGFilter, 0.4)
+AC_INIT([SVGFilter], [0.4])
+AC_CONFIG_SRCDIR([svgfilter.cpp])
+AM_INIT_AUTOMAKE()
 
 DEBUG_FLAGS=""
 OPT_FLAGS="-O3"
