Index: XaraLX-0.7r1785/wxOil/ftfonts.cpp
===================================================================
--- XaraLX-0.7r1785.orig/wxOil/ftfonts.cpp
+++ XaraLX-0.7r1785/wxOil/ftfonts.cpp
@@ -113,6 +113,7 @@ DECLARE_SOURCE( "$Revision: 1535 $" );
 #include "textfuns.h"
 #include "oilpanse.h"
 
+#define GSocket glib_GSocket
 #include <pango/pango.h>
 #include <pango/pangofc-font.h>
 #include <ft2build.h>
