--- wxOil/basebar2.cpp.orig	2014-04-10 03:16:53.000000000 +0400
+++ wxOil/basebar2.cpp	2014-04-10 03:15:56.000000000 +0400
@@ -4564,8 +4564,8 @@
 // -----------------------------------------------------------------------------------------
 // Message map - maintained by Class Wizard.
 
-BEGIN_EVENT_TABLE( BaseBar, wxWindow )
 //#pragma message( __LOCMSG__ "Removed BaseBar message map" )
+BEGIN_EVENT_TABLE( BaseBar, wxWindow )
 /*	ON_WM_NCCALCSIZE()
 	ON_WM_NCPAINT()
 	ON_WM_NCACTIVATE()
