Index: XaraLX-0.7r1785/wxOil/bitmapgriddropdown.cpp
===================================================================
--- XaraLX-0.7r1785.orig/wxOil/bitmapgriddropdown.cpp
+++ XaraLX-0.7r1785/wxOil/bitmapgriddropdown.cpp
@@ -105,6 +105,7 @@ service marks of Xara Group Ltd. All rig
 //-----------------------------------------------------------------------------------------
 // Include files
 
+#include <algorithm>
 #include "camtypes.h"
 #include "camelot.h"
 #include "errors.h"
