diff --git a/config.cpp b/config.cpp
index e65ed14..c95d83a 100644
--- a/config.cpp
+++ b/config.cpp
@@ -1,5 +1,6 @@
 #include <QFile>
 #include <QDir>
+#include <QStandardPaths>
 #include <QString>
 #include <QTextStream>
 #include <QtGlobal>
@@ -11,7 +12,7 @@ CConfig::CConfig()
 {
  init_vars();
 
- QFile file(QDir::homePath() + CONFIG_FILENAME);
+ QFile file(QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation) + CONFIG_FILENAME);
 
  if (!file.exists())
    save_config_file();      // create the file by saving the default values.
@@ -67,7 +68,7 @@ void CConfig::init_vars()
 }
 
 int CConfig::load_config_file() {
-  QFile file(QDir::homePath() + CONFIG_FILENAME);
+  QFile file(QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation) + CONFIG_FILENAME);
 
   int cpt = 0;
 
@@ -408,7 +409,7 @@ int CConfig::set_config_file(int param, bool enable)
 
 int CConfig::save_config_file()
 {
-  QFile file(QDir::homePath() + CONFIG_FILENAME);
+  QFile file(QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation) + CONFIG_FILENAME);
 
   if (!file.open(QIODevice::WriteOnly | QIODevice::Truncate | QIODevice::Text)) {
      return ERROPENWO;
