--- rayfowler/GameUI.java	2022-02-07 14:33:32.248028000 +0530
+++ Aman9das/GameUI.java	2022-02-07 21:56:14.375328000 +0530
@@ -712,20 +712,25 @@
     private void openRedditPage() {
         try {
             buttonClick();
-            Desktop.getDesktop().browse(new URL("http://www.reddit.com/r/rotp").toURI());
-        } catch (IOException | URISyntaxException e) {}
+            // Desktop.getDesktop().browse(new URL("http://www.reddit.com/r/rotp").toURI());
+            String[] command = {"/bin/xdg-open", "http://www.reddit.com/r/rotp"};
+            Runtime.getRuntime().exec(command);
+        } catch (IOException e) {}
     }
     private void openManual() {
         try {
             buttonClick();
             String filename = manualFilePath();
-            InputStream manualAsStream = fileInputStream(filename);
-            Path tempOutput = Files.createTempFile("ROTP_Manual", ".pdf");
-            tempOutput.toFile().deleteOnExit();
-            Files.copy(manualAsStream, tempOutput, StandardCopyOption.REPLACE_EXISTING);
-            File userManual = new File (tempOutput.toFile().getPath());
-            if (userManual.exists()) 
-                Desktop.getDesktop().open(userManual);
+            String fileuri = Rotp.class.getResource(filename).toString();
+            String[] command = {"/bin/xdg-open", fileuri};
+            Runtime.getRuntime().exec(command);
+            // InputStream manualAsStream = fileInputStream(filename);
+            // Path tempOutput = Files.createTempFile("ROTP_Manual", ".pdf");
+            // tempOutput.toFile().deleteOnExit();
+            // Files.copy(manualAsStream, tempOutput, StandardCopyOption.REPLACE_EXISTING);
+            // File userManual = new File (tempOutput.toFile().getPath());
+            // if (userManual.exists()) 
+            //     Desktop.getDesktop().open(userManual);
         } catch (IOException e) {}
     }
     public void continueGame() {
