diff -ruN original/zpath.c zsnes/zpath.c
--- original/zpath.c    2023-04-12 10:19:44.435685652 +0200
+++ zsnes/zpath.c       2023-04-12 10:19:03.652352332 +0200
@@ -76,7 +76,7 @@
     const char* const zpath = ZCONF;
 #else
 #ifndef __MACOSX__
-    const char* const zpath = ".zsnes";
+    const char* const zpath = "zsnes";
 #else
     const char* const zpath = "Library/Application Support/ZSNES";
 #endif
@@ -90,7 +90,7 @@

     if (ZCfgPath) {
         ZCfgAlloc = true;
-        strcpy(ZCfgPath, userinfo->pw_dir);
+        strcpy(ZCfgPath, getenv("XDG_CONFIG_HOME"));
         strcatslash(ZCfgPath);
         strcat(ZCfgPath, zpath);
