diff --git a/src/celestia/celestiacore.cpp b/src/celestia/celestiacore.cpp
index 800ac0a2..c7343059 100644
--- a/src/celestia/celestiacore.cpp
+++ b/src/celestia/celestiacore.cpp
@@ -2331,6 +2331,10 @@ bool CelestiaCore::initSimulation(const fs::path& configFileName,
         localConfigFile = PathExp("~/.celestia/celestia.cfg");
         if (!localConfigFile.empty())
             hasConfig |= ReadCelestiaConfig(localConfigFile, *config);
+
+        localConfigFile = WriteableDataPath() / "celestia.cfg";
+        if (!localConfigFile.empty())
+            hasConfig |= ReadCelestiaConfig(localConfigFile, *config);
     }
 
     if (!hasConfig)
