diff --git a/source/duke3d/src/common.cpp b/source/duke3d/src/common.cpp
index db1dab4..e947cd4 100644
--- a/source/duke3d/src/common.cpp	
+++ b/source/duke3d/src/common.cpp
@@ -789,6 +789,7 @@ void G_AddSearchPaths(void)
 #if defined __linux__ || defined EDUKE32_BSD
     char buf[BMAX_PATH];
     char *homepath = Bgethomedir();
+    char *xdg_docs_path = getenv("XDG_DOCUMENTS_DIR");
 
     Bsnprintf(buf, sizeof(buf), "%s/.steam/steam", homepath);
     G_AddSteamPaths(buf);
@@ -796,7 +797,17 @@ void G_AddSearchPaths(void)
     Bsnprintf(buf, sizeof(buf), "%s/.steam/steam/steamapps/libraryfolders.vdf", homepath);
     G_ParseSteamKeyValuesForPaths(buf);
 
+    if (xdg_docs_path) {
+        Bsnprintf(buf, sizeof(buf), "%s/Eduke32", xdg_docs_path);
+        addsearchpath(buf);
+    }
+    else {
+        Bsnprintf(buf, sizeof(buf), "%s/Documents/EDuke32", homepath);
+        addsearchpath(buf);
+    }
+
     Xfree(homepath);
+    Xfree(xdg_docs_path);
 
     addsearchpath("/usr/share/games/jfduke3d");
     addsearchpath("/usr/local/share/games/jfduke3d");
