diff --git a/src/celestia/celestiacore.cpp b/src/celestia/celestiacore.cpp
index e443160d..e39cf42c 100644
--- a/src/celestia/celestiacore.cpp
+++ b/src/celestia/celestiacore.cpp
@@ -2304,6 +2304,10 @@ bool CelestiaCore::initSimulation(const fs::path& configFileName,
         localConfigFile = PathExp("~/.celestia-1.7.cfg");
         if (!localConfigFile.empty())
             hasConfig |= ReadCelestiaConfig(localConfigFile, *config);
+
+        localConfigFile = WriteableDataPath() / "celestia.cfg";
+        if (!localConfigFile.empty())
+            hasConfig |= ReadCelestiaConfig(localConfigFile, *config);
     }
 
     if (!hasConfig)
