diff -Naur old/configure.ac new/configure.ac
--- old/configure.ac	2018-06-30 00:47:30.493300206 +0200
+++ new/configure.ac	2018-06-30 00:50:37.592578027 +0200
@@ -44,7 +44,7 @@
 AC_PROG_SED
 AC_PROG_AWK
 
-XSTAR_LIB_CHECK([LIBXML], [xml2-config])
+XSTAR_LIB_CHECK([LIBXML], [pkg-config libxml-2.0])
 
 AS_IF([test "x$LIBXML_SRCDIR" != x],
       [LIBXML_INCDIR="$LIBXML_SRCDIR/include"])
@@ -60,7 +60,7 @@
 [LIBXML_LDFLAGS="-L$LIBXML_LIBDIR"]
 
 
-XSTAR_LIB_CHECK([LIBXSLT], [xslt-config])
+XSTAR_LIB_CHECK([LIBXSLT], [pkg-config libxslt])
 
 AS_IF([test "x$LIBXSLT_SRCDIR" != x],
       [XSLTPROC_PATH="$LIBXSLT_SRCDIR/xsltproc:$PATH"
