diff -Naur XnConvert.orig/xnconvert.sh XnConvert/xnconvert.sh
--- XnConvert.orig/xnconvert.sh	2012-07-20 16:27:42.000000000 +0200
+++ XnConvert/xnconvert.sh	2019-03-09 12:37:22.400972533 +0100
@@ -1,6 +1,14 @@
 #!/bin/sh
 dirname="$(dirname "$(readlink -e "$0")")"
-export LD_LIBRARY_PATH="$dirname"/lib
-export QT_PLUGIN_PATH="$dirname"/lib
+if [[ -z "$LD_LIBRARY_PATH" ]]; then
+  export LD_LIBRARY_PATH="$dirname"/lib
+else
+  export LD_LIBRARY_PATH="$dirname"/lib:"$dirname"/Plugins:"$LD_LIBRARY_PATH"
+fi
+if [[ -z "$QT_PLUGIN_PATH" ]]; then
+  export QT_PLUGIN_PATH="$dirname"/lib
+else
+  export QT_PLUGIN_PATH="$dirname"/lib:"$QT_PLUGIN_PATH"
+fi
 "$dirname"/XnConvert "$@"
 
