diff -Naur XnConvert.orig/xnconvert.sh XnConvert/xnconvert.sh
--- XnConvert.orig/xnconvert.sh	2012-07-20 16:27:42.000000000 +0200
+++ XnConvert/xnconvert.sh	2019-03-03 20:06:28.511030070 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 dirname="$(dirname "$(readlink -e "$0")")"
-export LD_LIBRARY_PATH="$dirname"/lib
+if [[ -z "${LD_LIBRARY_PATH}" ]]; then
+  export LD_LIBRARY_PATH="$dirname"/lib
+else
+  export LD_LIBRARY_PATH="$dirname"/lib:"${LD_LIBRARY_PATH}"
+fi
 export QT_PLUGIN_PATH="$dirname"/lib
 "$dirname"/XnConvert "$@"
 
