diff --git a/shell_main.cc.orig b/shell_main.cc
index 7bbb4ad..7846852 100644
--- a/gtk/shell_main.cc.orig
+++ b/gtk/shell_main.cc
@@ -37,8 +37,8 @@
 #include "shell_spool.h"
 #include "core_main.h"
 #include "core_display.h"
-#include "icon-128x128.xpm"
-#include "icon-48x48.xpm"
+#include "icon-128x128.h"
+#include "icon-48x48.h"
 
 #ifndef _POSIX_HOST_NAME_MAX
 #define _POSIX_HOST_NAME_MAX 255
@@ -543,8 +543,8 @@ static void activate(GtkApplication *theApp, gpointer userData) {
     mainwindow = GTK_WIDGET(obj);
     gtk_window_set_application(GTK_WINDOW(mainwindow), app);
 
-    icon_128 = gdk_pixbuf_new_from_xpm_data((const char **) icon_128_xpm);
-    icon_48 = gdk_pixbuf_new_from_xpm_data((const char **) icon_48_xpm);
+    icon_128 = gdk_pixbuf_new_from_inline(-1, icon_128_pixbuf, FALSE, NULL);
+    icon_48 = gdk_pixbuf_new_from_inline(-1, icon_48_pixbuf, FALSE, NULL);
 
     gtk_window_set_icon(GTK_WINDOW(mainwindow), icon_128);
     gtk_window_set_title(GTK_WINDOW(mainwindow), TITLE);
