diff --git a/src/Misc/Bank.cpp b/src/Misc/Bank.cpp
index d81131ba..c9cfc66b 100644
--- a/src/Misc/Bank.cpp
+++ b/src/Misc/Bank.cpp
@@ -1444,9 +1444,9 @@ bool Bank::parseBanksFile(XMLwrapper *xml)
     string bankdirs[] = {
         localDir + "yoshimi/banks",
         "/usr/share/yoshimi/banks",
-        "/usr/local/share/yoshimi/banks",
+        "/app/extensions/Plugins/Yoshimi/share/yoshimi/banks",
         "/usr/share/zynaddsubfx/banks",
-        "/usr/local/share/zynaddsubfx/banks",
+        "/app/extensions/Plugins/ZynFusion/share/zynaddsubfx/banks",
         localDir + "zynaddsubfx/banks",
         extendLocalPath("/banks"),
         "end"
diff --git a/src/Misc/Config.cpp b/src/Misc/Config.cpp
index 59bc3251..39b78eb6 100644
--- a/src/Misc/Config.cpp
+++ b/src/Misc/Config.cpp
@@ -513,7 +513,7 @@ void Config::defaultPresets(void)
         // The following is not a default one.
         //string(getenv("HOME")) + "/" + string(EXTEN::config) + "/yoshimi/presets",
         "/usr/share/yoshimi/presets",
-        "/usr/local/share/yoshimi/presets",
+        "/app/extensions/Plugins/Yoshimi/share/yoshimi/presets",
         /*
          * We no longer include zyn presets as they changed the filenames.
         "/usr/share/zynaddsubfx/presets",
