diff -ur yubioath-desktop.old/yubioath-desktop.pro yubioath-desktop/yubioath-desktop.pro
--- yubioath-desktop.old/yubioath-desktop.pro	2019-05-29 08:42:23.000000000 +0200
+++ yubioath-desktop/yubioath-desktop.pro	2019-08-19 10:49:40.725980595 +0200
@@ -4,6 +4,8 @@
 SOURCES += main.cpp systemtray.cpp
 HEADERS += screenshot.h systemtray.h
 
+target.path = $$PREFIX/bin
+
 # This is the internal verson number, Windows requires 4 digits.
 win32|win64 {
     VERSION = 4.3.6.0
@@ -39,7 +41,7 @@
     pip.commands = python3 -m venv pymodules && source pymodules/bin/activate && pip3 install -r requirements.txt && deactivate
 }
 !macx {
-    pip.commands = pip3 install -r requirements.txt --target pymodules
+    pip.commands = pip3 install --no-deps --prefix $$PREFIX --target pymodules
 }
 
 # Default rules for deployment.
Only in yubioath-desktop: yubioath-desktop.pro.orig
