--- ZeroBraneStudio-linux/app/zbstudio.sh	2023-02-05 07:59:43.843974050 +0200
+++ ZeroBraneStudio-linux/app/zbstudio.sh	2023-02-05 08:01:10.465008789 +0200
@@ -3,4 +3,4 @@
 if [[ "$(uname -m)" == "x86_64" ]]; then ARCH="x64"; else ARCH="x86"; fi
 CWD="$PWD" # save the current directory, as it's going to change

-(cd "/opt/zbstudio"; bin/linux/$ARCH/lua src/main.lua zbstudio -cwd "$CWD" "$@" 2>/dev/null) &
+(cd "/app/opt/zbstudio"; bin/linux/$ARCH/lua src/main.lua zbstudio -cwd "$CWD" "$@" 2>/dev/null) &
