--- zita-convolver/source/Makefile.old	2019-02-01 22:42:34.465055892 -0500
+++ zita-convolver/source/Makefile	2019-02-01 22:42:50.694205967 -0500
@@ -56,7 +56,7 @@
 	install -d $(DESTDIR)$(LIBDIR)
 	install -m 644 $(ZITA-CONVOLVER_H) $(DESTDIR)$(INCDIR)
 	install -m 755 $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(LIBDIR)
-	ldconfig
+	#ldconfig
 	ln -sf $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-CONVOLVER_SO)
 
 uninstall:
