--- zita-resampler/source/Makefile.old	2019-02-01 22:44:57.863381983 -0500
+++ zita-resampler/source/Makefile	2019-02-01 22:45:08.306478558 -0500
@@ -58,7 +58,7 @@
 	install -d $(DESTDIR)$(LIBDIR)
 	install -m 644 $(ZITA-RESAMPLER_H)   $(DESTDIR)$(INCDIR)/zita-resampler
 	install -m 755 $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)
-	ldconfig
+	#ldconfig
 	ln -sf $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-RESAMPLER_SO)
 
 uninstall:
