From 169d911ab575c7efab203e91db03c325b20d14e4 Mon Sep 17 00:00:00 2001
From: Stone Tickle <lattis@mochiro.moe>
Date: Fri, 5 Jun 2020 12:51:25 +0900
Subject: [PATCH 2/4] set O_NONBLOCK on repeat timerfd

---
 src/wl_init.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/wl_init.c b/src/wl_init.c
index b6cdb39d..f092aef4 100644
--- a/src/wl_init.c
+++ b/src/wl_init.c
@@ -1198,8 +1198,9 @@ int _glfwPlatformInit(void)
     _glfwInitTimerPOSIX();
 
     _glfw.wl.timerfd = -1;
-    if (_glfw.wl.seatVersion >= 4)
-        _glfw.wl.timerfd = timerfd_create(CLOCK_MONOTONIC, TFD_CLOEXEC);
+    if (_glfw.wl.seatVersion >= 4) {
+        _glfw.wl.timerfd = timerfd_create(CLOCK_MONOTONIC, TFD_CLOEXEC | TFD_NONBLOCK);
+    }
 
 #ifndef WITH_DECORATION
     if (!_glfw.wl.wmBase)
-- 
2.32.0

