From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ecmel=20Berk=20Canl=C4=B1er?= <me@ecmelberk.com>
Date: Tue, 28 Dec 2021 20:30:44 +0300
Subject: Don't crash getting scancode name

---
 src/wl_window.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/wl_window.c b/src/wl_window.c
index c36b11cd..bf188e27 100644
--- a/src/wl_window.c
+++ b/src/wl_window.c
@@ -1205,8 +1205,7 @@ void _glfwPlatformSetCursorMode(_GLFWwindow* window, int mode)
 const char* _glfwPlatformGetScancodeName(int scancode)
 {
     // TODO
-    _glfwInputError(GLFW_FEATURE_UNIMPLEMENTED,
-                    "Wayland: Key names not yet implemented");
+    fprintf(stderr, "!!! Ignoring Error: Wayland: Key names not yet implemented\n");
     return NULL;
 }
 
