From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ecmel=20Berk=20Canl=C4=B1er?= <me@ecmelberk.com>
Date: Tue, 28 Dec 2021 22:01:03 +0300
Subject: Add libdecoration marker to stderr warning

---
 src/init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/init.c b/src/init.c
index c0ba6d31..803915bb 100644
--- a/src/init.c
+++ b/src/init.c
@@ -262,7 +262,7 @@ GLFWAPI int glfwInit(void)
 
     glfwDefaultWindowHints();
 
-    fprintf(stderr, "!!! Patched GLFW from https://github.com/Admicos/minecraft-wayland\n"
+    fprintf(stderr, "!!! Patched GLFW from https://github.com/Admicos/minecraft-wayland (libdecoration)\n"
          "!!! If any issues with the window, or some issues with rendering, occur, "
          "first try with the built-in GLFW, and if that solves the issue, report there first.\n"
          "!!! Use outside Minecraft is untested, and things might break.\n");
