--- a/src/libcec/env.h.in	2018-10-16 20:41:56.619945752 -0700
+++ b/src/libcec/env.h.in	2018-10-16 20:43:55.994936588 -0700
@@ -89,4 +89,4 @@
 #cmakedefine HAVE_PYTHON @HAVE_PYTHON@
 
 /* information about how libCEC was compiled */
-#define LIB_INFO ("@LIB_INFO@")
+#define LIB_INFO ("Flatpak")
--- a/src/cec-client/CMakeLists.txt	2018-10-16 21:02:09.099852669 -0700
+++ b/src/cec-client/CMakeLists.txt	2018-10-16 21:02:31.651850938 -0700
@@ -24,7 +24,7 @@
 set(cecclient_SOURCES cec-client.cpp)
 
 # curses
-check_library_exists(curses initscr "" HAVE_CURSES_API)
+check_library_exists(nocurses noinitscr "" HAVE_CURSES_API)
 if (HAVE_CURSES_API)
   list(APPEND cecclient_SOURCES curses/CursesControl.cpp)
 endif()
