diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2c5eea1a7..208f6c165 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,8 +98,8 @@ add_subdirectory(src/xalanc/Utils/XalanMsgLib)
 add_subdirectory(src/xalanc)
 add_subdirectory(src/xalanc/TestXSLT)
 add_subdirectory(src/xalanc/TestXPath)
-add_subdirectory(samples)
-add_subdirectory(Tests)
+add_subdirectory(samples EXCLUDE_FROM_ALL)
+add_subdirectory(Tests EXCLUDE_FROM_ALL)
 add_subdirectory(docs/doxygen)
 
 # Display configuration summary
