diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-09-26 19:43:48.000000000 +0200
+++ b/CMakeLists.txt	2025-09-03 17:35:15.362613583 +0200
@@ -2,7 +2,7 @@
 #        you can define SDL2_INCLUDE_DIR on the cmdline.  For example:
 #        cmake -DSDL2_INCLUDE_DIR=/opt/SDL2/include/SDL2 [other stuff]
 
-cmake_minimum_required(VERSION 3.0.0...3.5)
+cmake_minimum_required(VERSION 3.0.0...4.0)
 project(sdl12_compat
         VERSION 1.2.68
         LANGUAGES C)
