--- a/src/nvim/version.c
+++ b/src/nvim/version.c
@@ -2733,9 +2733,9 @@ void intro_message(bool colon)
   static char *(lines[]) = {
     N_(NVIM_VERSION_LONG),
     "",
-    N_("Nvim is open source and freely distributable"),
-    "https://neovim.io/#chat",
+    N_("You are running an unofficial Flatpak version of Neovide"),
     "",
+    N_("type  :help flatpak<Enter>    for flatpak info"),
     N_("type  :help nvim<Enter>       if you are new! "),
     N_("type  :checkhealth<Enter>     to optimize Nvim"),
     N_("type  :q<Enter>               to exit         "),
