--- a/linux/CMakeLists.txt      2025-05-25 16:58:31.000000000 +0200
+++ b/linux/CMakeLists.txt      2025-05-25 16:59:03.013600538 +0200
@@ -14,12 +14,14 @@
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2026/sqlite-autoconf-3520000.tar.gz
+    URL_HASH SHA256=f6b50b0c103392af32a8be15b2b9d25959de9a00a70c3979128aafeaa5338b3f
     DOWNLOAD_EXTRACT_TIMESTAMP NEW
   )
 else()
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2026/sqlite-autoconf-3520000.tar.gz
+    URL_HASH SHA256=f6b50b0c103392af32a8be15b2b9d25959de9a00a70c3979128aafeaa5338b3f
   )
 endif()
 FetchContent_MakeAvailable(sqlite3)