diff -ur mapnik-v3.0.13/Makefile mapnik-v3.0.13-patched/Makefile
--- mapnik-v3.0.13/Makefile	2017-02-08 15:13:57.000000000 +0200
+++ mapnik-v3.0.13-patched/Makefile	2017-04-02 19:20:26.058358808 +0300
@@ -51,7 +51,8 @@
 		src/svg/svg_transform_parser.os \
 
 
-mapnik: src/json/libmapnik-json.a
+#mapnik: src/json/libmapnik-json.a
+mapnik: 
 	# then install the rest with -j$(JOBS)
 	$(PYTHON) scons/scons.py -j$(JOBS) --config=cache --implicit-cache --max-drift=1
