--- a/hook/build.dart	2025-11-30 17:58:52.000000000 +0100
+++ b/hook/build.dart	2025-12-21 16:26:08.818660070 +0100
@@ -28,13 +28,15 @@
         }
 
         final target = File(p.join(dir.path, library.filename));
-        final tmp = File('${target.path}.tmp');
+        if (!await File(target.path).exists()) {
+          final tmp = File('${target.path}.tmp');
 
-        await sqlite
-            .fetch(input, output, library)
-            .cast<List<int>>()
-            .pipe(tmp.openWrite());
-        tmp.renameSync(target.path);
+          await sqlite
+              .fetch(input, output, library)
+              .cast<List<int>>()
+              .pipe(tmp.openWrite());
+          tmp.renameSync(target.path);
+        }
 
         output.assets.code.add(
           CodeAsset(
