diff --git a/BloomDesktop/Bloom.sln b/BloomDesktop/Bloom.sln
index fa8c612d9..2ffe81734 100644
--- a/BloomDesktop/Bloom.sln
+++ b/BloomDesktop/Bloom.sln
@@ -5,10 +5,6 @@ VisualStudioVersion = 15.0.26430.14
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BloomExe", "src\BloomExe\BloomExe.csproj", "{304D5612-167C-4725-AF27-B9F2BB788B57}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BloomTests", "src\BloomTests\BloomTests.csproj", "{57E7AC72-D901-4A80-A50B-71319AC96CDC}"
-EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BloomChorusPlugin", "src\Bloom-ChorusPlugin\BloomChorusPlugin.csproj", "{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}"
-EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution", "Solution", "{04AF3A87-C943-4731-BE3D-659EA705171D}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "DistFiles", "DistFiles", "{D38FC737-8C27-46CB-B490-3CB2E7AB3C99}"
@@ -49,28 +45,6 @@ Global
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|Mixed Platforms.Build.0 = Release|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|x86.ActiveCfg = Release|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|x86.Build.0 = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Any CPU.ActiveCfg = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Any CPU.Build.0 = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Mixed Platforms.ActiveCfg = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Mixed Platforms.Build.0 = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|x86.ActiveCfg = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|x86.Build.0 = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Any CPU.ActiveCfg = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Any CPU.Build.0 = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Mixed Platforms.ActiveCfg = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Mixed Platforms.Build.0 = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|x86.ActiveCfg = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|x86.Build.0 = Release|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Debug|Any CPU.ActiveCfg = Debug|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Debug|Any CPU.Build.0 = Debug|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Debug|Mixed Platforms.ActiveCfg = Debug|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Debug|x86.ActiveCfg = Debug|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Debug|x86.Build.0 = Debug|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Release|Any CPU.ActiveCfg = Release|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Release|Any CPU.Build.0 = Release|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Release|Mixed Platforms.ActiveCfg = Release|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Release|x86.ActiveCfg = Release|x86
-		{5B04DA06-782F-41B8-84FB-EA1AB82E3FB0}.Release|x86.Build.0 = Release|x86
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
diff --git a/BloomDesktop/debian/bloom b/BloomDesktop/debian/bloom
index 3e6be8764..8d2dc621e 100755
--- a/BloomDesktop/debian/bloom
+++ b/BloomDesktop/debian/bloom
@@ -1,11 +1,16 @@
 #!/bin/sh
 
-LIB=/usr/lib/bloom-desktop
-SHARE=/usr/share/bloom-desktop
+LIB=/app/lib/bloom-desktop
+SHARE=/app/share/bloom-desktop
 
 cd "$SHARE"
 RUNMODE=INSTALLED
 . ./environ
 cd "$OLDPWD"
 
-exec "$LIB"/BloomLauncher "$@"
+# needed for Sldr.Initialize() in flatpak
+mkdir -p /var/lock
+
+# Without this sync, communication with BloomLibrary (or at least S3) will fail.
+[ ! -d ~/.var/app/org.sil.Bloom/config/.mono/certs ] && /app/bin/mono /app/lib/mono/4.5/cert-sync.exe --user /app/share/ca-certificates/ca-certificates.crt
+exec /app/bin/mono --debug "$LIB"/Bloom.exe "$@"
diff --git a/BloomDesktop/environ b/BloomDesktop/environ
index 44b1f73b2..1d560153b 100644
--- a/BloomDesktop/environ
+++ b/BloomDesktop/environ
@@ -6,9 +6,9 @@
 # Note that this file is intended to be "sourced", not "executed".
 
 # the sourcing script should cd/pushd to the directory containing this script
-BASE="$(pwd)"
+BASE="$(pwd | sed s=/app/share/=/app/lib/=)"
 [ -z "$BUILD" ] && BUILD=Debug
-[ -z "$MONO_PREFIX" ] && MONO_PREFIX=/opt/mono5-sil
+[ -z "$MONO_PREFIX" ] && MONO_PREFIX=/app
 
 # Dependency locations
 # Search for xulrunner and geckofx, select the best, and add its location to LD_LIBRARY_PATH.
@@ -26,18 +26,18 @@ MONO_ENV_OPTIONS="-O=-gshared"
 # chmsee was phased out by Ubuntu in bionic (18.04), and supporting libraries by disco (19.04)
 
 if [ -z "$MONO_HELP_VIEWER" ]; then
-	if [ -f /usr/bin/chmsee ]; then
+	if [ -f /app/bin/chmsee ]; then
 		export MONO_HELP_VIEWER=chmsee
-	elif [ -f /usr/bin/xchm ]; then
+	elif [ -f /app/bin/xchm ]; then
 		export MONO_HELP_VIEWER=xchm
-	elif [ -f /usr/bin/kchmviewer ]; then
+	elif [ -f /app/bin/kchmviewer ]; then
 		export MONO_HELP_VIEWER=kchmviewer
 	fi
 fi
 
 ################################################################################################
 
-if [ "$RUNMODE" = "INSTALLED" ]
+if [ "$RUNMODE" = "INSTALLED" -o "$FLATPAK_ID" = "org.sil.BloomBeta" ]
 then
 	# Add the installation directory to the library paths
 	LD_LIBRARY_PATH="${BASE}:${LD_LIBRARY_PATH}"
@@ -55,7 +55,7 @@ fi
 
 ################################################################################################
 
-if [ "$RUNMODE" = "PACKAGE" -o "$RUNMODE" = "INSTALLED" ]
+if [ "$RUNMODE" = "PACKAGE" -o "$RUNMODE" = "INSTALLED" -o "$FLATPAK_ID" = "org.sil.BloomBeta" ]
 then
 	# Add packaged mono items to paths
 	PATH="${MONO_PREFIX}/bin:${PATH}"
@@ -73,9 +73,6 @@ else
 	MONO_GAC_PREFIX="${MONO_PREFIX}:/usr"
 fi
 
-# point to our private version of libtidy
-LD_LIBRARY_PATH="/opt/tidy-sil/lib:${LD_LIBRARY_PATH}"
-
 ################################################################################################
 
 MONO_TRACE_LISTENER="Console.Out"
@@ -101,3 +98,7 @@ export HGRCPATH=
 
 #sets keyboard input method to none
 unset XMODIFIERS
+
+if [ "$DEBUG_BLOOM" = "true" ]; then
+	env | sort
+fi
diff --git a/BloomDesktop/environ-xulrunner b/BloomDesktop/environ-xulrunner
index 91b1fa261..808a03c23 100644
--- a/BloomDesktop/environ-xulrunner
+++ b/BloomDesktop/environ-xulrunner
@@ -2,8 +2,8 @@
 # for them.  Note that this script lives in /usr/share/bloom-desktop* on user machines,
 # but on the root of the repository on developer machines.  This requires a bit of
 # hackery to reliably find the proper location of the Firefox/xulrunner libraries.
-tmpbasedir=$(pwd | sed s=usr/share/=usr/lib/=)
-if (echo $tmpbasedir | grep /usr/lib/bloom-desktop >/dev/null); then
+tmpbasedir=$(pwd | sed s=app/share/=app/lib/=)
+if (echo $tmpbasedir | grep /app/lib/bloom-desktop >/dev/null); then
   export XULRUNNER=$(dirname $(find $tmpbasedir -name libxul.so))
 else
   export XULRUNNER=$(dirname $(find ${tmpbasedir}/output/${BUILD} -name libxul.so))
diff --git a/BloomDesktop/src/BloomBrowserUI/gulpfile.js b/BloomDesktop/src/BloomBrowserUI/gulpfile.js
index 7112cb7a4..7c6afb748 100644
--- a/BloomDesktop/src/BloomBrowserUI/gulpfile.js
+++ b/BloomDesktop/src/BloomBrowserUI/gulpfile.js
@@ -94,7 +94,9 @@ var paths = {
 var allXliffFiles = globule.find(paths.xliff);
 // Check for the existence of the SIL mono, which flags we're on Linux
 // and need to use it to execute HtmlXliff.exe
-var IsLinux = globule.find(["/opt/mono5-sil/**/mono"]).length > 0;
+var IsLinuxWithMono5 = globule.find(["/opt/mono5-sil/**/mono"]).length > 0;
+// Check for files in /etc which flag we're on Linux, possibly with the system mono.
+var IsLinux = globule.find(["/etc/os-release","/etc/passwd","/etc/group"]).length === 3;
 
 gulp.task("less", function() {
     var less = require("gulp-less");
@@ -288,9 +290,11 @@ gulp.task("translateHtmlFiles", function() {
                     var xliffFile = xliffFiles[i]; // needed for error message to work
                     var outfile = getOutputFilename(file.path, xliffFile);
                     var cmd = "";
-                    if (IsLinux)
+                    if (IsLinuxWithMono5)
                         cmd =
                             "/opt/mono5-sil/bin/mono --debug ../../lib/dotnet/HtmlXliff.exe --inject";
+		    else if (IsLinux)
+		        cmd = "mono --debug  ../../lib/dotnet/HtmlXliff.exe --inject";
                     else cmd = "..\\..\\lib\\dotnet\\HtmlXliff.exe --inject";
                     cmd = cmd + ' -x "' + xliffFile + '"';
                     cmd = cmd + ' -o "' + outfile + '"';
@@ -318,9 +322,11 @@ gulp.task("createXliffFiles", function() {
             tap(function(file) {
                 var xliffFile = getXliffFilename(file.path);
                 var cmd = "";
-                if (IsLinux)
+                if (IsLinuxWithMono5)
                     cmd =
                         "/opt/mono5-sil/bin/mono --debug ../../lib/dotnet/HtmlXliff.exe --extract --preserve";
+		else if (IsLinux)
+		    cmd = "mono --debug  ../../lib/dotnet/HtmlXliff.exe --extract --preserve";
                 else
                     cmd =
                         "..\\..\\lib\\dotnet\\HtmlXliff.exe --extract --preserve";
diff --git a/BloomDesktop/src/BloomBrowserUI/package.json b/BloomDesktop/src/BloomBrowserUI/package.json
index 68e8a4f66..648ed5eac 100644
--- a/BloomDesktop/src/BloomBrowserUI/package.json
+++ b/BloomDesktop/src/BloomBrowserUI/package.json
@@ -8,7 +8,7 @@
         "test": "test"
     },
     "engines": {
-        "node": "~16.13.1"
+        "node": "^16.13.1"
     },
     "scripts": {
         "test": "karma start --browsers Firefox  --single-run",
diff --git a/BloomDesktop/src/BloomExe/BloomExe.csproj b/BloomDesktop/src/BloomExe/BloomExe.csproj
index 29f60f3d0..21fb46a4e 100644
--- a/BloomDesktop/src/BloomExe/BloomExe.csproj
+++ b/BloomDesktop/src/BloomExe/BloomExe.csproj
@@ -8,7 +8,7 @@
     <TargetFrameworkProfile>
     </TargetFrameworkProfile>
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
-    <RestorePackages>true</RestorePackages>
+    <RestorePackages>false</RestorePackages>
     <PackagesConfig Condition="'$(OS)'=='Windows_NT'">$(MSBuildProjectDirectory)\packages.config</PackagesConfig>
     <PackagesConfig Condition="'$(OS)'!='Windows_NT'">$(MSBuildProjectDirectory)/Linux/packages.config</PackagesConfig>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
@@ -173,7 +173,7 @@
     </Reference>
     <Reference Include="MarkdownDeep, Version=1.5.4615.26275, Culture=neutral, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\packages\MarkdownDeep.NET.1.5\lib\.NetFramework 3.5\MarkdownDeep.dll</HintPath>
+      <HintPath>..\..\packages\MarkdownDeep.NET.1.5.0\lib\.NetFramework 3.5\MarkdownDeep.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Bcl.AsyncInterfaces, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Microsoft.Bcl.AsyncInterfaces.5.0.0\lib\net461\Microsoft.Bcl.AsyncInterfaces.dll</HintPath>
@@ -216,7 +216,7 @@
     <Reference Include="PdfSharp">
       <HintPath>..\..\lib\dotnet\PdfSharp.dll</HintPath>
     </Reference>
-    <Reference Include="PodcastUtilities.PortableDevices">
+    <Reference Include="PodcastUtilities.PortableDevices" Condition="'$(OS)'=='Windows_NT'">
       <HintPath>..\..\lib\dotnet\PodcastUtilities.PortableDevices.dll</HintPath>
     </Reference>
     <Reference Include="PresentationCore" />
@@ -262,7 +262,7 @@
     <Reference Include="Spart, Version=1.0.0.0, Culture=neutral, PublicKeyToken=14d24e064e474331, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Spart.1.0.0\lib\net461\Spart.dll</HintPath>
     </Reference>
-    <Reference Include="Splat">
+    <Reference Include="Splat" Condition="'$(OS)'=='Windows_NT'">
       <HintPath>..\..\lib\dotnet\Splat.dll</HintPath>
     </Reference>
     <Reference Include="SQLite-net, Version=1.7.335.0, Culture=neutral, processorArchitecture=MSIL">
@@ -284,6 +284,9 @@
       <HintPath>..\..\lib\dotnet\Squirrel.dll</HintPath>
     </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Buffers, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.Buffers.4.5.1\lib\net461\System.Buffers.dll</HintPath>
+    </Reference>
     <Reference Include="System.Collections.Immutable, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\..\packages\System.Collections.Immutable.5.0.0\lib\net461\System.Collections.Immutable.dll</HintPath>
     </Reference>
@@ -295,6 +298,9 @@
       <HintPath>..\..\packages\System.Diagnostics.DiagnosticSource.4.7.1\lib\net46\System.Diagnostics.DiagnosticSource.dll</HintPath>
     </Reference>
     <Reference Include="System.IO.Compression.FileSystem" />
+    <Reference Include="System.IO.FileSystem, Version=4.0.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.IO.FileSystem.4.3.0\lib\net46\System.IO.FileSystem.dll</HintPath>
+    </Reference>
     <Reference Include="System.IO.FileSystem.AccessControl, Version=4.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\..\packages\System.IO.FileSystem.AccessControl.4.7.0\lib\net461\System.IO.FileSystem.AccessControl.dll</HintPath>
     </Reference>
@@ -310,6 +316,9 @@
     <Reference Include="System.Runtime.CompilerServices.Unsafe, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\..\packages\System.Runtime.CompilerServices.Unsafe.5.0.0\lib\net45\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
     </Reference>
+    <Reference Include="System.Runtime.InteropServices.RuntimeInformation, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.Runtime.InteropServices.RuntimeInformation.4.3.0\lib\net45\System.Runtime.InteropServices.RuntimeInformation.dll</HintPath>
+    </Reference>
     <Reference Include="System.Runtime.Serialization" />
     <Reference Include="System.Numerics.Vectors">
       <HintPath>..\..\packages\System.Numerics.Vectors.4.5.0\lib\net46\System.Numerics.Vectors.dll</HintPath>
@@ -329,9 +338,10 @@
       <HintPath>..\..\packages\System.Text.Json.5.0.2\lib\net461\System.Text.Json.dll</HintPath>
     </Reference>
     <Reference Include="System.Threading.Tasks, Version=4.0.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />
-    <Reference Include="System.ValueTuple, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\System.ValueTuple.4.5.0\lib\net461\System.ValueTuple.dll</HintPath>
+    <Reference Include="System.Threading.Tasks.Extensions, Version=4.2.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.Threading.Tasks.Extensions.4.5.4\lib\net461\System.Threading.Tasks.Extensions.dll</HintPath>
     </Reference>
+    <Reference Include="System.ValueTuple" />
     <Reference Include="System.Web" />
     <Reference Include="System.Web.Extensions" />
     <Reference Include="System.Xml.Linq" />
diff --git a/BloomDesktop/src/BloomExe/Linux/packages.config b/BloomDesktop/src/BloomExe/Linux/packages.config
index ee57c4e20..86fe597a9 100644
--- a/BloomDesktop/src/BloomExe/Linux/packages.config
+++ b/BloomDesktop/src/BloomExe/Linux/packages.config
@@ -23,7 +23,7 @@
   <package id="JsonFx" version="2.0.1209.2802" targetFramework="net461" />
   <package id="L10NSharp" version="4.2.0-beta0006" targetFramework="net461" />
   <package id="Markdig.Signed" version="0.22.0" targetFramework="net461" />
-  <package id="MarkdownDeep.NET" version="1.5" targetFramework="net40-Client" />
+  <package id="MarkdownDeep.NET" version="1.5.0" targetFramework="net40-Client" />
   <package id="Microsoft.Bcl.AsyncInterfaces" version="5.0.0" targetFramework="net461" />
   <package id="Microsoft.DotNet.PlatformAbstractions" version="3.1.6" targetFramework="net461" />
   <package id="Microsoft.Extensions.DependencyModel" version="3.1.6" targetFramework="net461" />
@@ -73,4 +73,7 @@
   <!-- Linux specific packages.  Changes above this line must be copied to ../packages.config -->
   <package id="Geckofx60.64.Linux" version="60.0.51" targetFramework="net461" />
   <package id="SharpFont" version="3.1.0" targetFramework="net461" />
+  <!-- copied from BloomTests -->
+  <package id="System.IO.FileSystem" version="4.3.0" targetFramework="net461" />
+  <package id="System.Threading.Tasks.Extensions" version="4.5.4" targetFramework="net461" />
 </packages>
