diff -up nvidia-settings-435.17/src/libXNVCtrl/Makefile.shared nvidia-settings-435.17/src/libXNVCtrl/Makefile
--- nvidia-settings-435.17/src/libXNVCtrl/Makefile.shared	2019-08-07 06:12:32.000000000 +0200
+++ nvidia-settings-435.17/src/libXNVCtrl/Makefile	2019-08-26 16:23:41.921778088 +0200
@@ -50,8 +50,9 @@ LDFLAGS += $(XNVCTRL_LDFLAGS)
 
 .PHONY: clean
 
-all: $(LIBXNVCTRL)
+all: $(LIBXNVCTRL) libXNVCtrl.so
 
 clean:
 	rm -rf $(LIBXNVCTRL) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d
+	rm -f libXNVCtrl.so libXNVCtrl.so.*
diff -up nvidia-settings-435.17/src/Makefile.shared nvidia-settings-435.17/src/Makefile
--- nvidia-settings-435.17/src/Makefile.shared	2019-08-07 06:12:31.000000000 +0200
+++ nvidia-settings-435.17/src/Makefile	2019-08-26 16:17:31.133406921 +0200
@@ -106,6 +106,9 @@ endif
 XNVCTRL_DIR             ?= libXNVCtrl
 XCONFIG_PARSER_DIR      ?= XF86Config-parser
 COMMON_UTILS_DIR        ?= common-utils
+XNVCTRL_SHARED          ?= $(XNVCTRL_DIR)/libXNVCtrl.so.0
+#XNVCTRL_LIB            ?= $(XNVCTRL_ARCHIVE)
+XNVCTRL_LIB             ?= $(XNVCTRL_SHARED)
 COMMON_UNIX_DIR         ?= common-unix
 VIRTUAL_RESOLUTIONS_DIR ?= $(COMMON_UNIX_DIR)/virtual-resolutions
 
diff -up nvidia-settings-435.17/src/libXNVCtrl/xnvctrl.mk.shared nvidia-settings-435.17/src/libXNVCtrl/xnvctrl.mk
--- nvidia-settings-435.17/src/libXNVCtrl/xnvctrl.mk.shared	2019-08-07 06:12:32.000000000 +0200
+++ nvidia-settings-435.17/src/libXNVCtrl/xnvctrl.mk	2019-08-26 16:54:44.518016474 +0200
@@ -47,3 +47,8 @@ $(eval $(call DEFINE_OBJECT_RULE,TARGET,
 
 $(LIBXNVCTRL) : $(LIBXNVCTRL_OBJ)
 	$(call quiet_cmd,AR) ru $@ $(LIBXNVCTRL_OBJ)
+
+libXNVCtrl.so: $(LIBXNVCTRL_OBJ)
+	$(CC) -shared -Wl,-soname=$@.0 -o $@.0.0.0 $(LDFLAGS) $^ -lXext -lX11
+	ln -s $@.0.0.0 $@.0
+	ln -s $@.0 $@
