diff --git a/po/meson.build b/po/meson.build
index 2ad4f42..8a62b1f 100644
--- a/po/meson.build
+++ b/po/meson.build
@@ -7,4 +7,4 @@
-if python_po.found() and spa_json_dump_po.found()
+if get_option('tools') and python_po.found() and spa_json_dump_po.found()
   conf_pot = custom_target('conf.pot',
     input : i18n_conf,
     output : 'conf.pot',
diff --git a/meson.build b/meson.build
index f118b37..d1811e8 100644
--- a/meson.build
+++ b/meson.build
@@ -182,5 +182,7 @@ if build_modules
   subdir('modules')
 endif
 subdir('src')
-subdir('po')
+if build_daemon or build_tools
+  subdir('po')
+endif
 subdir('docs')
