From 4dcedb79eeccd9e5a2c523ff9b8de023c4f45a98 Mon Sep 17 00:00:00 2001
From: Jagadeesh Kotra <jagadeesh@stdin.top>
Date: Sun, 31 Jul 2022 22:26:20 +0530
Subject: [PATCH] disable ownership check for plugins

---
 src/libnm-core-impl/nm-utils.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libnm-core-impl/nm-utils.c b/src/libnm-core-impl/nm-utils.c
index d5d884f2e4..bbe621a97b 100644
--- a/src/libnm-core-impl/nm-utils.c
+++ b/src/libnm-core-impl/nm-utils.c
@@ -3119,6 +3119,7 @@ _nm_utils_check_file(const char               *filename,
 
     /* with check_owner enabled, check that the file belongs to the
      * owner or root. */
+    /*
     if (check_owner >= 0 && (out_st->st_uid != 0 && (gint64) out_st->st_uid != check_owner)) {
         g_set_error(error,
                     NM_VPN_PLUGIN_ERROR,
@@ -3128,6 +3129,7 @@ _nm_utils_check_file(const char               *filename,
                     filename);
         return FALSE;
     }
+    */
 
     /* with check_owner enabled, check that the file cannot be modified
      * by other users (except root). */
-- 
2.37.1
