--- ring-client-gnome/daemon/contrib/src/opendht/rules.mak.orig	2019-03-19 22:26:52.000000000 +0100
+++ ring-client-gnome/daemon/contrib/src/opendht/rules.mak	2019-04-04 19:20:01.626841937 +0200
@@ -20,9 +20,6 @@
 ifneq ($(call need_pkg,"jsoncpp"),)
 DEPS_opendht += jsoncpp
 endif
-ifneq ($(call need_pkg,"gnutls >= 3.3.0"),)
-DEPS_opendht += gnutls
-endif
 
 $(TARBALLS)/opendht-$(OPENDHT_VERSION).tar.gz:
 	$(call download,$(OPENDHT_URL))
--- ring-client-gnome/daemon/contrib/src/pjproject/rules.mak.orig	2019-03-19 22:26:52.000000000 +0100
+++ ring-client-gnome/daemon/contrib/src/pjproject/rules.mak	2019-04-04 19:21:38.327316029 +0200
@@ -37,7 +37,6 @@
 PKGS_FOUND += pjproject
 endif
 
-DEPS_pjproject += gnutls
 ifndef HAVE_WIN32
 ifndef HAVE_MACOSX
 DEPS_pjproject += uuid
--- ring-client-gnome/daemon/contrib/src/secp256k1/rules.mak.orig	2019-03-19 22:26:52.000000000 +0100
+++ ring-client-gnome/daemon/contrib/src/secp256k1/rules.mak	2019-04-04 19:23:00.075712812 +0200
@@ -7,7 +7,6 @@
 ifeq ($(call need_pkg,"libsecp256k1"),)
 PKGS_FOUND += secp256k1
 endif
-DEPS_secp256k1 = gmp
 
 $(TARBALLS)/secp256k1-$(SECP256K1_VERSION).tar.gz:
 	$(call download,$(SECP256K1_URL))
