diff -urN ring-project-orig/daemon/contrib/src/dbus-cpp/rules.mak ring-project/daemon/contrib/src/dbus-cpp/rules.mak
--- ring-project-orig/daemon/contrib/src/dbus-cpp/rules.mak	2017-10-24 11:20:14.000000000 -0400
+++ ring-project/daemon/contrib/src/dbus-cpp/rules.mak	2017-11-20 19:31:28.194081738 -0500
@@ -24,7 +24,9 @@
 	$(APPLY) $(SRC)/dbus-cpp/dbus-c++-threading.patch
 	$(APPLY) $(SRC)/dbus-cpp/dbus-c++-writechar.patch
 	$(APPLY) $(SRC)/dbus-cpp/dbus-c++-gcc4.7.patch
+	cp -p /usr/share/automake-*/config.{sub,guess} $(UNPACK_DIR)
-	$(UPDATE_AUTOCONFIG)
+	#$(UPDATE_AUTOCONFIG)
+	#autoconf -f -B build/autoconf_prepend-include
 	$(MOVE)
 
 .dbus-cpp: dbus-cpp
