diff -urN ring-project-orig/make-ring.py ring-project/make-ring.py
--- ring-project-orig/make-ring.py	2017-10-24 11:20:01.000000000 -0400
+++ ring-project/make-ring.py	2017-11-19 14:04:56.812515000 -0500
@@ -248,6 +248,10 @@
         print("The Android version does not need more dependencies.\nPlease continue with the --install instruction.")
         sys.exit(1)

+    elif args.distribution == "GNOME":
+        print("Flatpak dependancinies are handled in top level json.\nPlease continue with the --install instruction.")
+        sys.exit(1)
+
     else:
         print("Not yet implemented for current distribution (%s)" % args.distribution)
         sys.exit(1)
@@ -398,7 +402,7 @@
     """Validate the args values, exit if error is found"""

     # Check arg values
-    supported_distros = ['Android', 'Ubuntu', 'Debian', 'OSX', IOS_DISTRIBUTION_NAME, 'Fedora', 'Arch Linux', 'openSUSE', 'Automatic', 'mingw32', 'mingw64']
+    supported_distros = ['Android', 'Ubuntu', 'Debian', 'OSX', IOS_DISTRIBUTION_NAME, 'Fedora', 'Arch Linux', 'openSUSE', 'Automatic', 'mingw32', 'mingw64', 'GNOME']

     if parsed_args.distribution not in supported_distros:
         print('Distribution \''+parsed_args.distribution+'\' not supported.\nChoose one of: %s' \
