diff -ruN ring-project/daemon/contrib/src/jsoncpp/rules.mak ring-project-fixed/daemon/contrib/src/jsoncpp/rules.mak
--- ring-project/daemon/contrib/src/jsoncpp/rules.mak	2018-02-09 18:10:00.000000000 -0500
+++ ring-project-fixed/daemon/contrib/src/jsoncpp/rules.mak	2018-02-10 14:26:17.202660730 -0500
@@ -1,5 +1,5 @@
 # JSONCPP
-JSONCPP_VERSION := 1.7.2
+JSONCPP_VERSION := 1.7.5
 JSONCPP_URL := https://github.com/open-source-parsers/jsoncpp/archive/$(JSONCPP_VERSION).tar.gz
 
 PKGS += jsoncpp
diff -ruN ring-project/daemon/contrib/src/jsoncpp/SHA512SUMS ring-project-fixed/daemon/contrib/src/jsoncpp/SHA512SUMS
--- ring-project/daemon/contrib/src/jsoncpp/SHA512SUMS	2018-02-09 18:10:00.000000000 -0500
+++ ring-project-fixed/daemon/contrib/src/jsoncpp/SHA512SUMS	2018-02-10 14:44:07.267122848 -0500
@@ -1 +1,2 @@
 32702147229ea7a3679654325572c38f4188f258ab6ac21f9e04059d53ef2a7cd0542ec4ec3b0e7b9089acd2b7bce389f16b9ff24b2e63e0ba2a5bcd46bab766  jsoncpp-1.7.2.tar.gz
+c8217e390d4b15e046a6f14ad54257ac8ecc13b70073a15c502b451df25c6d8bbc645ee50bb12e67433bf2c9053e2a39544d465c19124c7b882b69dd80b70ab2  jsoncpp-1.7.5.tar.gz
