diff -urN ring-project-orig/daemon/contrib/src/restbed/rules.mak ring-project/daemon/contrib/src/restbed/rules.mak
--- ring-project-orig/daemon/contrib/src/restbed/rules.mak	2017-10-24 11:20:14.000000000 -0400
+++ ring-project/daemon/contrib/src/restbed/rules.mak	2017-11-19 20:06:50.728976003 -0500
@@ -40,13 +40,10 @@
 			-DCMAKE_INSTALL_PREFIX=$(PREFIX) \
 			-DCMAKE_INSTALL_LIBDIR=lib
 
-$(TARBALLS)/kashmir.tar.gz:
-	$(call download,https://github.com/Corvusoft/kashmir-dependency/archive/master.tar.gz)
-
-restbed: restbed-$(RESTBED_VERSION).tar.gz kashmir.tar.gz .sum-restbed
+restbed: restbed-$(RESTBED_VERSION).tar.gz .sum-restbed
 	$(UNPACK)
 	rm -rf $(UNPACK_DIR)/dependency/kashmir
-	mv kashmir-dependency-master $(UNPACK_DIR)/dependency/kashmir
+	cp -a /app/include/kashmir $(UNPACK_DIR)/dependency/kashmir
 	$(APPLY) $(SRC)/restbed/findkashmir.patch
 	$(APPLY) $(SRC)/restbed/strand.patch
 	$(APPLY) $(SRC)/restbed/locale-fix.patch
