diff --git a/src/Application.vala b/src/Application.vala
index d1550a4..7b5e7ab 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -37,6 +37,7 @@ public class Harvey : Gtk.Application {
         }

         var app_window = new MainWindow (this);
+        Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");

         var window_x = settings.get_int ("window-x");
         var window_y = settings.get_int ("window-y");
