diff -ur murrine-0.98.2.orig/src/murrine_rc_style.h murrine-0.98.2/src/murrine_rc_style.h
--- murrine-0.98.2.orig/src/murrine_rc_style.h	2012-01-17 18:46:01.000000000 +0100
+++ murrine-0.98.2/src/murrine_rc_style.h	2022-12-14 21:33:03.242067429 +0100
@@ -154,5 +154,6 @@
 };
 
 GType murrine_rc_style_get_type	(void);
+void murrine_rc_style_register_types (GTypeModule *module);
 
 #endif /* MURRINE_RC_STYLE_H */
diff -ur murrine-0.98.2.orig/src/murrine_style.h murrine-0.98.2/src/murrine_style.h
--- murrine-0.98.2.orig/src/murrine_style.h	2012-01-17 18:46:01.000000000 +0100
+++ murrine-0.98.2/src/murrine_style.h	2022-12-14 21:33:41.604671684 +0100
@@ -102,5 +102,6 @@
 };
 
 GType murrine_style_get_type (void);
+void murrine_style_register_types (GTypeModule *module);
 
 #endif /* MURRINE_STYLE_H */
diff -ur murrine-0.98.2.orig/src/support.h murrine-0.98.2/src/support.h
--- murrine-0.98.2.orig/src/support.h	2012-01-17 18:46:01.000000000 +0100
+++ murrine-0.98.2/src/support.h	2022-12-14 21:27:27.288533100 +0100
@@ -118,6 +118,10 @@
 
 #define MRN_WIDGET_HAS_DEFAULT(object) ((object) && MRN_IS_WIDGET(object) && GTK_WIDGET_HAS_DEFAULT(object))
 
+gboolean murrine_object_is_a (const GObject * object, const gchar * type_name);
+gboolean murrine_widget_is_ltr (GtkWidget *widget);
+
+
 G_GNUC_INTERNAL GtkTextDirection murrine_get_direction (GtkWidget *widget);
 G_GNUC_INTERNAL GtkWidget *murrine_special_get_ancestor (GtkWidget *widget, GType widget_type);
 G_GNUC_INTERNAL GdkColor* murrine_get_parent_bgcolor (GtkWidget *widget);
