diff --git a/src/ibusenginesimple.c b/src/ibusenginesimple.c
index 43bd5283..8896207e 100644
--- a/src/ibusenginesimple.c
+++ b/src/ibusenginesimple.c
@@ -542,7 +542,7 @@ check_emoji_table (IBusEngineSimple       *simple,
         g_object_ref_sink (priv->lookup_table);
     }
     if (emoji_dict == NULL)
-        emoji_dict = priv->emoji_dict = load_emoji_dict (simple);
+        emoji_dict = priv->emoji_dict = load_emoji_dict ();
 
     if (emoji_dict == NULL || emoji_dict->dict == NULL)
         return FALSE;
