Description: Don't call fclose() twice.
Author: Jonathan Neuschäfer <j.neuschaefer@gmx.net>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=632947
---
 vorbis.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/vorbis.c
+++ b/vorbis.c
@@ -813,11 +813,6 @@
         fclose(infile);
     }
 
-    if (infile != NULL)
-    {
-        fclose(infile);
-    }
-
     if (delete_temp)
     {
         if (remove(temp_name) != 0)
