Description: "recursivly" should be spelled "recursively". Thanks Adam.
From: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=661999
Forwarded: no
--- a/vorbisgain.c
+++ b/vorbisgain.c
@@ -500,7 +500,7 @@
     fprintf(stderr, _("  -q, --quiet          Don't print any output (except errors)\n"));
     fprintf(stderr, _("  -p, --preserve-mtime Don't change the timestamps\n"));
 #ifdef ENABLE_RECURSIVE
-    fprintf(stderr, _("  -r, --recursive      Search for files recursivly, each folder as an album\n"));
+    fprintf(stderr, _("  -r, --recursive      Search for files recursively, each folder as an album\n"));
 #endif
     fprintf(stderr, _("  -s, --skip           Skip non-Vorbis or faulty files\n"));
     fprintf(stderr, _("  -v, --version        Display version number and exit\n\n"));
