diff --git a/llvm/cmake/modules/AddLLVM.cmake b/llvm/cmake/modules/AddLLVM.cmake
index fd69786544a..3f0090d0305 100644
--- a/llvm/cmake/modules/AddLLVM.cmake
+++ b/llvm/cmake/modules/AddLLVM.cmake
@@ -591,6 +591,8 @@ function(llvm_add_library name)
        ${LLVM_LINK_COMPONENTS}
        )
     endif()
+  elseif (${name} STREQUAL "LLVMTableGenShared" AND LLVM_BUILD_LLVM_DYLIB)
+    target_link_libraries(${name} PRIVATE LLVM)
   else()
     # Components have not been defined explicitly in CMake, so add the
     # dependency information for this library as defined by LLVMBuild.
@@ -726,8 +728,10 @@ macro(add_llvm_library name)
       if(${name} IN_LIST LLVM_DISTRIBUTION_COMPONENTS OR
           (in_llvm_libs AND "llvm-libraries" IN_LIST LLVM_DISTRIBUTION_COMPONENTS) OR
           NOT LLVM_DISTRIBUTION_COMPONENTS)
-        set(export_to_llvmexports EXPORT LLVMExports)
-        set_property(GLOBAL PROPERTY LLVM_HAS_EXPORTS True)
+        if(ARG_MODULE OR ARG_SHARED OR BUILD_SHARED_LIBS)
+          set(export_to_llvmexports EXPORT LLVMExports)
+          set_property(GLOBAL PROPERTY LLVM_HAS_EXPORTS True)
+        endif()
       endif()
 
       install(TARGETS ${name}
@@ -742,7 +746,10 @@ macro(add_llvm_library name)
                                  COMPONENT ${name})
       endif()
     endif()
-    set_property(GLOBAL APPEND PROPERTY LLVM_EXPORTS ${name})
+    if(ARG_MODULE OR ARG_SHARED OR BUILD_SHARED_LIBS)
+      set_property(GLOBAL APPEND PROPERTY LLVM_EXPORTS ${name})
+    else()
+    endif()
   endif()
   if (ARG_MODULE)
     set_target_properties(${name} PROPERTIES FOLDER "Loadable modules")
diff --git a/llvm/lib/TableGen/CMakeLists.txt b/llvm/lib/TableGen/CMakeLists.txt
index 1ece8145952..0fd727a9f35 100644
--- a/llvm/lib/TableGen/CMakeLists.txt
+++ b/llvm/lib/TableGen/CMakeLists.txt
@@ -1,3 +1,24 @@
+if (LLVM_BUILD_LLVM_DYLIB)
+  add_llvm_library(LLVMTableGenShared
+    Error.cpp
+    JSONBackend.cpp
+    Main.cpp
+    Record.cpp
+    SetTheory.cpp
+    StringMatcher.cpp
+    TableGenBackend.cpp
+    TGLexer.cpp
+    TGParser.cpp
+
+    SHARED
+
+    ADDITIONAL_HEADER_DIRS
+    ${LLVM_MAIN_INCLUDE_DIR}/llvm/TableGen
+
+    OUTPUT_NAME LLVMTableGen
+  )
+endif()
+
 add_llvm_component_library(LLVMTableGen
   Error.cpp
   JSONBackend.cpp
diff --git a/llvm/tools/llvm-config/llvm-config.cpp b/llvm/tools/llvm-config/llvm-config.cpp
index 6c31df3e173..edd5ae0da57 100644
--- a/llvm/tools/llvm-config/llvm-config.cpp
+++ b/llvm/tools/llvm-config/llvm-config.cpp
@@ -688,6 +688,13 @@ int main(int argc, char **argv) {
       };
 
       if (LinkMode == LinkModeShared && LinkDyLib) {
+        for (unsigned i = 0, e = RequiredLibs.size(); i != e; ++i) {
+	  if (RequiredLibs[i] == "LLVMTableGen") {
+	    PrintForLib(RequiredLibs[i]);
+	    OS << ' ';
+	    break ;
+	  }
+	}
         PrintForLib(DyLibName);
       } else {
         for (unsigned i = 0, e = RequiredLibs.size(); i != e; ++i) {
diff --git a/llvm/utils/llvm-build/llvmbuild/main.py b/llvm/utils/llvm-build/llvmbuild/main.py
index 99b82ad5e20..a1cbf8f796d 100644
--- a/llvm/utils/llvm-build/llvmbuild/main.py
+++ b/llvm/utils/llvm-build/llvmbuild/main.py
@@ -359,6 +359,7 @@ subdirectories = %s
         root_entries = set(e[0] for e in entries)
         for _,_,deps,_ in entries:
             root_entries -= set(deps)
+        root_entries.remove('tablegen')
         entries.append(('all', None, sorted(root_entries), True))
 
         entries.sort()
