diff --git a/protonvpn_gui/main.py b/protonvpn_gui/main.py
index 96f1a11..312de6a 100644
--- a/protonvpn_gui/main.py
+++ b/protonvpn_gui/main.py
@@ -36,7 +36,7 @@ class ProtonVPNGUI(Gtk.Application):
 
     def __init__(self):
         super().__init__(
-            application_id='com.protonvpn.www',
+            application_id='com.protonvpn.ProtonVPN',
             flags=Gio.ApplicationFlags.FLAGS_NONE
         )
         self.indicator = None