diff --git a/debian/control b/debian/control
index 3cd343e..354989f 100644
--- a/debian/control
+++ b/debian/control
@@ -10,7 +10,7 @@ Package: protonvpn-gui
 Architecture: all
 Homepage: https://github.com/ProtonVPN
 Section: net
-Depends: ${python3:Depends}, ${misc:Depends}, python3-gi, python3-gi-cairo, python3-psutil, python3-protonvpn-nm-lib, gir1.2-webkit2-4.0
+Depends: ${python3:Depends}, ${misc:Depends}, python3-gi, python3-gi-cairo, python3-psutil, python3-protonvpn-nm-lib, gir1.2-webkit2-4.1
 Suggests: gir1.2-appindicator3-0.1, gnome-shell-extension-appindicator
 Description: Proton VPN GUI (Python 3)
  Package installs official Proton VPN Graphical User Interface.
diff --git a/protonvpn_gui/view/dialog.py b/protonvpn_gui/view/dialog.py
index e26f815..eb507ae 100644
--- a/protonvpn_gui/view/dialog.py
+++ b/protonvpn_gui/view/dialog.py
@@ -3,7 +3,10 @@ import os
 import gi
 
 gi.require_version("Gtk", "3.0")
-gi.require_version("WebKit2", "4.0")
+try:
+    gi.require_version("WebKit2", "4.1")
+except (ImportError, ValueError):
+    gi.require_version("WebKit2", "4.0")
 
 from gi.repository import Gdk, Gio, Gtk
 from gi.repository import WebKit2 as Webkit