diff --git a/protonvpn_gui/view/dialog.py b/protonvpn_gui/view/dialog.py
index e26f815..f8adf3a 100644
--- a/protonvpn_gui/view/dialog.py
+++ b/protonvpn_gui/view/dialog.py
@@ -3,7 +3,7 @@ import os
 import gi
 
 gi.require_version("Gtk", "3.0")
-gi.require_version("WebKit2", "4.0")
+gi.require_version("WebKit2", "4.1")
 
 from gi.repository import Gdk, Gio, Gtk
 from gi.repository import WebKit2 as Webkit