diff --git a/protonvpn_nm_lib/core/dbus/dbus_reconnect.py b/protonvpn_nm_lib/core/dbus/dbus_reconnect.py
index 3db50e5..0fc336e 100644
--- a/protonvpn_nm_lib/core/dbus/dbus_reconnect.py
+++ b/protonvpn_nm_lib/core/dbus/dbus_reconnect.py
@@ -119,6 +119,7 @@ class DbusReconnect:
         Returns:
             int: indicates the status of the daemon process
         """
+        return 1
         logger.info("Checking daemon reconnector status")
         check_daemon = subprocess.run(
             ["systemctl", "status", "--user", "protonvpn_reconnect"],
@@ -156,6 +157,9 @@ class DbusReconnect:
             command (string): to either start or stop the process
         """
         logger.info("Calling daemon reconnector")
+
+        return
+
         if command not in self.DAEMON_COMMANDS:
             raise Exception("Invalid daemon command \"{}\"".format(command))
 
diff --git a/protonvpn_nm_lib/core/subprocess_wrapper.py b/protonvpn_nm_lib/core/subprocess_wrapper.py
index 7fac7bb..02b458a 100644
--- a/protonvpn_nm_lib/core/subprocess_wrapper.py
+++ b/protonvpn_nm_lib/core/subprocess_wrapper.py
@@ -18,7 +18,7 @@ class SubprocessWrapper:
         is effectivtly a layer on top of subprocess.
     """
 
-    _acceptable_binaries = {"nmcli", "systemctl", "clear", "xdg-open"}
+    _acceptable_binaries = {"nmcli", "clear", "xdg-open"}
     PIPE = _subprocess.PIPE
     STDOUT = _subprocess.STDOUT
     DEVNULL = _subprocess.DEVNULL