diff --git a/protonvpn_nm_lib/core/subprocess_wrapper.py b/protonvpn_nm_lib/core/subprocess_wrapper.py
index 96634f6..7fac7bb 100644
--- a/protonvpn_nm_lib/core/subprocess_wrapper.py
+++ b/protonvpn_nm_lib/core/subprocess_wrapper.py
@@ -23,9 +23,17 @@ class SubprocessWrapper:
     STDOUT = _subprocess.STDOUT
     DEVNULL = _subprocess.DEVNULL
 
+    @staticmethod
+    def is_flatpak():
+        """Checks if it's in the Flatpak sandbox environment.
+        """
+        return os.path.isfile("/.flatpak-info")
+
     @staticmethod
     def is_root_owned(path):
         stat_info = os.stat(path)
+        if SubprocessWrapper.is_flatpak():
+            return stat_info.st_uid == 1000 and stat_info.st_gid == 1000
         return stat_info.st_uid == 0 and stat_info.st_gid == 0
 
     def __init__(self):