diff --git a/protonvpn_nm_lib/core/dbus/dbus_reconnect.py b/protonvpn_nm_lib/core/dbus/dbus_reconnect.py
index 3db50e5..bba5b5d 100644
--- a/protonvpn_nm_lib/core/dbus/dbus_reconnect.py
+++ b/protonvpn_nm_lib/core/dbus/dbus_reconnect.py
@@ -47,7 +47,7 @@ class DbusReconnect:
         python_service_path = os.path.join(
             daemon_folder, "dbus_daemon_reconnector.py"
         )
-        python_interpreter_path = sys.executable
+        python_interpreter_path = f"flatpak run --command={sys.executable} com.protonvpn.www"
         exec_start = python_interpreter_path + " " + python_service_path
         filled_template = SERVICE_TEMPLATE.replace("EXEC_START", exec_start)