Index: rrootage-0.23a/src/rr.c
===================================================================
--- rrootage-0.23a.orig/src/rr.c	2007-08-29 21:01:36.000000000 +0000
+++ rrootage-0.23a/src/rr.c	2007-08-29 21:02:30.000000000 +0000
@@ -195,7 +195,7 @@
 static int accframe = 0;
 
 static void usage(char *argv0) {
-  fprintf(stderr, "Usage: %s [-lowres|-mediumres|-highres] [-nosound] [-window] [-reverse] [-nowait] [-accframe]\n", argv0);
+  fprintf(stderr, "Usage: %s [-lowres|-mediumres|-highres] [-nosound] [-window] [-fullscreen] [-reverse] [-nowait] [-accframe]\n", argv0);
 }
 
 static void parseArgs(int argc, char *argv[]) {
@@ -211,6 +211,8 @@
       noSound = 1;
     } else if ( strcmp(argv[i], "-window") == 0 ) {
       windowMode = 1;
+    } else if ( strcmp(argv[i], "-fullscreen") == 0 ) {
+      windowMode = 0;
     } else if ( strcmp(argv[i], "-reverse") == 0 ) {
       buttonReversed = 1;
     }
Index: rrootage-0.23a/src/screen.c
===================================================================
--- rrootage-0.23a.orig/src/screen.c	2007-08-29 21:01:43.000000000 +0000
+++ rrootage-0.23a/src/screen.c	2007-08-29 21:02:41.000000000 +0000
@@ -129,7 +129,7 @@
 #define TITLE_BMP "title.bmp"
 
 int resolution = MEDIUM_RESOLUTION;
-int windowMode = 0;
+int windowMode = 1;
 int brightness = DEFAULT_BRIGHTNESS;
 Uint8 *keys;
 SDL_Joystick *stick = NULL;
