From: Markus Koschany <apo@debian.org>
Date: Sat, 12 Dec 2015 17:24:29 +0100
Subject: 09_rootage_make_highres_default

Forwarded: no
---
 src/screen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/screen.c b/src/screen.c
index 1c83309..303ae49 100644
--- a/src/screen.c
+++ b/src/screen.c
@@ -128,7 +128,7 @@ static GLuint smokeTexture;
 static GLuint titleTexture;
 #define TITLE_BMP "title.bmp"
 
-int resolution = MEDIUM_RESOLUTION;
+int resolution = HIGH_RESOLUTION;
 int windowMode = 1;
 int brightness = DEFAULT_BRIGHTNESS;
 Uint8 *keys;
