--- a/src/compositor/extensions/extensions.pri	2021-10-04 16:31:30.000000000 -0400
+++ b/src/compositor/extensions/extensions.pri	2022-02-18 21:53:42.016755907 -0400
@@ -8,6 +8,7 @@
     ../extensions/touch-extension.xml \
     ../extensions/qt-key-unstable-v1.xml \
     ../extensions/qt-windowmanager.xml \
+    ../extensions/qt-texture-sharing-unstable-v1.xml \
     ../3rdparty/protocol/text-input-unstable-v2.xml \
     ../3rdparty/protocol/xdg-shell-unstable-v6.xml \
     ../3rdparty/protocol/xdg-shell.xml \
@@ -78,7 +79,6 @@
     qtConfig(wayland-compositor-texture-sharing-experimental) {
         HEADERS += extensions/qwltexturesharingextension_p.h
         SOURCES += extensions/qwltexturesharingextension.cpp
-        WAYLANDSERVERSOURCES += ../extensions/qt-texture-sharing-unstable-v1.xml
     }
 }
 
