Description: Use constants::version instead of frobby_version.
Author: Doug Torrance <dtorrance@piedmont.edu>
Bug: https://github.com/Macaulay2/frobby/issues/2
Bug-Debian: https://bugs.debian.org/992432
Last-Update: 2021-08-20

--- a/src/frobby.cpp
+++ b/src/frobby.cpp
@@ -27,7 +27,9 @@
 #include "IdealFacade.h"
 #include "SliceParams.h"
 
-const char* const frobby_version = "0.9.5";
+namespace constants {
+  const char * const version = "0.9.5";
+}
 
 class ConsumerWrapper {
 protected:
--- a/src/frobby.h
+++ b/src/frobby.h
@@ -21,7 +21,9 @@
 
 #include "gmp.h"
 
-extern "C" const char* const frobby_version;
+namespace constants {
+  extern const char* const version;
+}
 
 /** The namespace FrobbyImpl is for internal use inside Frobby only.
  It is not part of the public interface of Frobby.
--- a/src/HelpAction.cpp
+++ b/src/HelpAction.cpp
@@ -156,7 +156,7 @@
 
   FrobbyStringStream out;
 
-  out << "Frobby version " << frobby_version
+  out << "Frobby version " << constants::version
       << " Copyright (C) 2007 Bjarke Hammersholt Roune\n";
   out <<
     "Frobby performs a number of computations related to monomial "
