diff --git a/src/Makefile b/src/Makefile
index 95fe549..2c182db 100755
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,8 +1,8 @@
 # remove -fno-exceptions  if it causes problems with stdio library on some platforms (ARM)
 
-DATADIR=/usr/share/espeak-data
+DATADIR=/app/share/espeak-data
 
-PREFIX=/usr
+PREFIX=/app
 BINDIR=$(PREFIX)/bin
 INCDIR=$(PREFIX)/include/espeak
 LIBDIR=$(PREFIX)/lib
@@ -27,10 +27,10 @@ MKDIR = mkdir -p
 
 # 'runtime' uses pulseaudio if it is running, else uses portaudio
 #AUDIO = runtime
-AUDIO = portaudio
+#AUDIO = portaudio
 #AUDIO = portaudio0
 #AUDIO = portaudio2
-#AUDIO = pulseaudio
+AUDIO = pulseaudio
 #AUDIO = sada
 
 ifeq ($(AUDIO),runtime)
@@ -93,7 +93,7 @@ SRCS3 = espeak.cpp
 OBJS3=$(patsubst %.cpp,%.o,$(SRCS3))
 LIBS3=-lstdc++ -L . -lespeak
 
-CXXFLAGS=-O2
+CXXFLAGS=-O2 -Wno-narrowing
 
 
 all: $(BIN_NAME) $(LIB_NAME) $(STATIC_LIB_NAME) $(BIN2_NAME)
