Index: pymol-open-source-2.3.0/modules/pmg_tk/PMGApp.py
===================================================================
--- pymol-open-source-2.3.0.orig/modules/pmg_tk/PMGApp.py	(revision 4149)
+++ pymol-open-source-2.3.0/modules/pmg_tk/PMGApp.py	(working copy)
@@ -320,7 +320,7 @@
             
             # initialize Tcl/Tk
 
-            self.root = Tk() # creates the root window for the application
+            self.root = Tk(className='pymol') # creates the root window for the application
 
             # color scheme
 
