--- a/src/MACLib/Assembly/x86_64_mmx.asm
+++ b/src/MACLib/Assembly/x86_64_mmx.asm
@@ -112,3 +112,5 @@ loopDot:
         emms
 
         ret
+
+section .note.GNU-stack noalloc noexec nowrite progbits
--- a/src/MACLib/Assembly/x86_mmx.asm
+++ b/src/MACLib/Assembly/x86_mmx.asm
@@ -121,3 +121,5 @@ loopDot:
         mov     eax, [esp + 4]
 
         ret
+
+section .note.GNU-stack noalloc noexec nowrite progbits
