--- a/linux/CMakeLists.txt      2025-05-25 16:58:31.000000000 +0200
+++ b/linux/CMakeLists.txt      2025-05-25 16:59:03.013600538 +0200
@@ -14,12 +14,14 @@
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2025/sqlite-autoconf-3510100.tar.gz
+    URL_HASH SHA256=4f2445cd70479724d32ad015ec7fd37fbb6f6130013bd4bfbc80c32beb42b7e0
     DOWNLOAD_EXTRACT_TIMESTAMP NEW
   )
 else()
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2025/sqlite-autoconf-3510100.tar.gz
+    URL_HASH SHA256=4f2445cd70479724d32ad015ec7fd37fbb6f6130013bd4bfbc80c32beb42b7e0
   )
 endif()
 FetchContent_MakeAvailable(sqlite3)