--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -32,13 +32,9 @@ else()
     )
 endif()

-# Download pdfium
-include(../windows/DownloadProject.cmake)
-download_project(PROJ
-                 pdfium
-                 URL
-                 ${PDFIUM_URL})
+set(pdfium_SOURCE_DIR "${CMAKE_BINARY_DIR}/pdfium-src")
+set(pdfium_BINARY_DIR "${CMAKE_BINARY_DIR}/pdfium-build")

 # This value is used when generating builds using this plugin, so it must not be
 # changed
 set(PLUGIN_NAME "printing_plugin")
