diff --git a/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py b/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py
index 5bf1f8a..8ed89cb 100644
--- a/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py
+++ b/proton/vpn/backend/networkmanager/killswitch/wireguard/killswitch_connection_handler.py
@@ -196,7 +196,7 @@ class KillSwitchConnectionHandler:
     async def _run_ip_route_command():
         def run():
             return subprocess.run(  # nosec subprocess_without_shell_equals_true
-                ["/usr/sbin/ip", "route"], capture_output=True, encoding="utf-8", check=True
+                ["/app/bin/ip", "route"], capture_output=True, encoding="utf-8", check=True
             )
 
         loop = asyncio.get_running_loop()
