diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
index 1b79b71b252..14714f676f8 100644
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -281,8 +281,6 @@ static void set_dll_path(void)
     dll_paths = malloc( (count + 2) * sizeof(*dll_paths) );
     count = 0;
 
-    if (!build_dir) dll_paths[count++] = dll_dir;
-
     if (path)
     {
         path = strdup(path);
@@ -290,6 +288,8 @@ static void set_dll_path(void)
         free( path );
     }
 
+    if (!build_dir) dll_paths[count++] = dll_dir;
+
     for (i = 0; i < count; i++) dll_path_maxlen = max( dll_path_maxlen, strlen(dll_paths[i]) );
     dll_paths[count] = NULL;
 }
