diff --git a/dlls/user32/msgbox.c b/dlls/user32/msgbox.c
index 4d34577..ec7257c 100644
--- a/dlls/user32/msgbox.c
+++ b/dlls/user32/msgbox.c
@@ -388,7 +388,7 @@ static INT_PTR CALLBACK MSGBOX_DlgProc( HWND hwnd, UINT message,
  */
 INT WINAPI MessageBoxA(HWND hWnd, LPCSTR text, LPCSTR title, UINT type)
 {
-    return MessageBoxExA(hWnd, text, title, type, LANG_NEUTRAL);
+    return MessageBoxExA(hWnd, text, "NOT A VINEGAR ERROR!", type, LANG_NEUTRAL);
 }
 
 
