diff --git a/dlls/ntdll/unix/signal_x86_64.c b/dlls/ntdll/unix/signal_x86_64.c
index dabe436bfe7..3dc45b92bf7 100644
--- a/dlls/ntdll/unix/signal_x86_64.c
+++ b/dlls/ntdll/unix/signal_x86_64.c
@@ -2177,8 +2177,12 @@ static void install_bpf(struct sigaction *sig_act)
         }
     }
 
-    if (!(ret = prctl(PR_GET_SECCOMP, 0, NULL, 0, 0)))
+    ret = prctl(PR_GET_SECCOMP, 0, NULL, 0, 0);
+    if (ret >= 0)
     {
+        if (ret == 2)
+            TRACE("Seccomp filters are installed, but applying anyway.\n");
+
         if (prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0))
         {
             perror("prctl(PR_SET_NO_NEW_PRIVS, ...)");
@@ -2196,10 +2200,7 @@ static void install_bpf(struct sigaction *sig_act)
     }
     else
     {
-        if (ret == 2)
-            TRACE("Seccomp filters already installed.\n");
-        else
-            ERR("Seccomp filters cannot be installed, ret %d, error %s.\n", ret, strerror(errno));
+        ERR("Seccomp filters cannot be installed, ret %d, error %s.\n", ret, strerror(errno));
     }
 
     sigaction(SIGSYS, sig_act, NULL);
