diff --git a/run_build_tool.sh b/run_build_tool.sh
index 0e2c973..200b520 100755
--- a/run_build_tool.sh
+++ b/run_build_tool.sh
@@ -66,7 +66,7 @@ fi

 # Run pub get if needed.
 if [ ! -f "$PACKAGE_HASH_FILE" ]; then
-    "$DART" pub get --no-precompile
+    "$DART" pub get --no-precompile --offline
     "$DART" compile kernel bin/build_tool_runner.dart
     echo "$PACKAGE_HASH" > "$PACKAGE_HASH_FILE"
 fi
@@ -79,7 +79,7 @@ exit_code=$?

 # 253 means invalid snapshot version.
 if [ $exit_code == 253 ]; then
-  "$DART" pub get --no-precompile
+  "$DART" pub get --no-precompile --offline
   "$DART" compile kernel bin/build_tool_runner.dart
   "$DART" bin/build_tool_runner.dill "$@"
   exit_code=$?
