From ff04ba4bfbe65929534f61a9f66a39d03c973b78 Mon Sep 17 00:00:00 2001
From: bbhtt <bbhtt.zn0i8@slmail.me>
Date: Sat, 9 Sep 2023 15:15:00 +0530
Subject: [PATCH 1/2] Cython fixes

---
 src_c/pypm.pyx | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src_c/pypm.pyx b/src_c/pypm.pyx
index d66e7a10..706726c6 100644
--- a/src_c/pypm.pyx
+++ b/src_c/pypm.pyx
@@ -5,6 +5,8 @@
 # harrison@media.mit.edu
 # written in Pyrex
 
+# cython: language_level=2
+
 __version__ = "0.0.6"
 
 import array
@@ -277,7 +279,7 @@ cdef class Output:
             PmPtr = <PmTimeProcPtr>&Pt_Time
 
         if self.debug:
-            print "Opening Midi Output"
+            print "Opening Midi Output", output_device
 
         # Why is buffer size 0 here?
         err = Pm_OpenOutput(&(self.midi), output_device, NULL, 0, PmPtr, NULL,
-- 
2.41.0

