diff -rupN fluidsynth-1.1.6.orig/src/drivers/fluid_alsa.c fluidsynth-1.1.6/src/drivers/fluid_alsa.c
--- fluidsynth-1.1.6.orig/src/drivers/fluid_alsa.c	2012-08-16 05:01:13.000000000 +0100
+++ fluidsynth-1.1.6/src/drivers/fluid_alsa.c	2017-02-28 21:26:57.033244239 +0000
@@ -96,7 +96,7 @@ struct fluid_alsa_formats_t fluid_alsa_f
 };
 
 
-
+#if 0
 /*
  * fluid_alsa_rawmidi_driver_t
  *
@@ -119,7 +119,7 @@ fluid_midi_driver_t* new_fluid_alsa_rawm
 
 int delete_fluid_alsa_rawmidi_driver(fluid_midi_driver_t* p);
 static void fluid_alsa_midi_run(void* d);
-
+#endif
 
 /*
  * fluid_alsa_seq_driver_t
@@ -535,7 +535,7 @@ static void fluid_alsa_audio_run_s16 (vo
  *
  */
 
-
+#if 0
 void fluid_alsa_rawmidi_driver_settings(fluid_settings_t* settings)
 {
   fluid_settings_register_str(settings, "midi.alsa.device", "default", 0, NULL, NULL);
@@ -698,7 +698,7 @@ fluid_alsa_midi_run(void* d)
     }
   }
 }
-
+#endif
 /**************************************************************
  *
  *        Alsa sequencer
diff -rupN fluidsynth-1.1.6.orig/src/drivers/fluid_mdriver.c fluidsynth-1.1.6/src/drivers/fluid_mdriver.c
--- fluidsynth-1.1.6.orig/src/drivers/fluid_mdriver.c	2012-08-16 05:01:13.000000000 +0100
+++ fluidsynth-1.1.6/src/drivers/fluid_mdriver.c	2017-02-28 21:24:43.887833321 +0000
@@ -24,11 +24,13 @@
 
 /* ALSA */
 #if ALSA_SUPPORT
+#if 0
 fluid_midi_driver_t* new_fluid_alsa_rawmidi_driver(fluid_settings_t* settings,
 						 handle_midi_event_func_t handler,
 						 void* event_handler_data);
 int delete_fluid_alsa_rawmidi_driver(fluid_midi_driver_t* p);
 void fluid_alsa_rawmidi_driver_settings(fluid_settings_t* settings);
+#endif
 
 fluid_midi_driver_t* new_fluid_alsa_seq_driver(fluid_settings_t* settings,
 					     handle_midi_event_func_t handler,
@@ -109,10 +111,12 @@ struct fluid_mdriver_definition_t fluid_
     fluid_oss_midi_driver_settings },
 #endif
 #if ALSA_SUPPORT
+#if 0
   { "alsa_raw",
     new_fluid_alsa_rawmidi_driver,
     delete_fluid_alsa_rawmidi_driver,
     fluid_alsa_rawmidi_driver_settings },
+#endif
   { "alsa_seq",
     new_fluid_alsa_seq_driver,
     delete_fluid_alsa_seq_driver,
