From 24d1e9c28518cb4b01344eaf8377f2d6d489ad50 Mon Sep 17 00:00:00 2001
From: bbhtt <bbhtt.zn0i8@slmail.me>
Date: Sat, 9 Sep 2023 12:04:16 +0530
Subject: [PATCH] Fix deprecated GStaticMutex init with glib-2.70

---
 src/drivers/fluid_jack.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/fluid_jack.c b/src/drivers/fluid_jack.c
index fe3cd752..613a9820 100644
--- a/src/drivers/fluid_jack.c
+++ b/src/drivers/fluid_jack.c
@@ -100,7 +100,7 @@ int fluid_jack_driver_process(jack_nframes_t nframes, void *arg);
 int delete_fluid_jack_midi_driver(fluid_midi_driver_t *p);
 
 
-static fluid_mutex_t last_client_mutex = G_STATIC_MUTEX_INIT;     /* Probably not necessary, but just in case drivers are created by multiple threads */
+static fluid_mutex_t last_client_mutex;     /* Probably not necessary, but just in case drivers are created by multiple threads */
 static fluid_jack_client_t *last_client = NULL;       /* Last unpaired client. For audio/MIDI driver pairing. */
 
 
-- 
2.41.0

