diff -r 407caa445ee0 config_unix.py
--- a/config_unix.py	Mon Jan 16 21:12:10 2017 +0000
+++ b/config_unix.py	Fri Mar 03 15:10:07 2017 +0000
@@ -110,6 +110,12 @@
             self.found = 1
         else:
             print (self.name + '        '[len(self.name):] + ': not found')
+            print('incname',  incname)
+            print('incdirs', incdirs)
+            print('self.inc_dir', self.inc_dir)
+            print('libnames', libnames)
+            print('libdirs', libdirs)
+            print('self.lib_dir', self.lib_dir)
 
 
 class DependencyPython:
@@ -195,8 +201,8 @@
     incdirs += ["/usr/local"+d for d in origincdirs]
     libdirs += ["/usr/local"+d for d in origlibdirs]
     if localbase:
-        incdirs = [localbase+d for d in origincdirs]
-        libdirs = [localbase+d for d in origlibdirs]
+        incdirs += [localbase+d for d in origincdirs]
+        libdirs += [localbase+d for d in origlibdirs]
 
     for arg in DEPS[0].cflags.split():
         if arg[:2] == '-I':
