diff -upr ladspa_sdk.orig/src/load.c ladspa_sdk/src/load.c
--- ladspa_sdk.orig/src/load.c	2012-09-26 07:37:24.000000000 +0300
+++ ladspa_sdk/src/load.c	2012-09-26 07:38:58.000000000 +0300
@@ -54,6 +54,8 @@ dlopenLADSPA(const char * pcFilename, in
        to search. */
 
     pcLADSPAPath = getenv("LADSPA_PATH");
+    if (! pcLADSPAPath)
+      pcLADSPAPath = "/usr/lib/ladspa";
     
     if (pcLADSPAPath) {
 
diff -upr ladspa_sdk.orig/src/search.c ladspa_sdk/src/search.c
--- ladspa_sdk.orig/src/search.c	2012-09-26 07:37:24.000000000 +0300
+++ ladspa_sdk/src/search.c	2012-09-26 07:38:46.000000000 +0300
@@ -99,10 +99,7 @@ LADSPAPluginSearch(LADSPAPluginSearchCal
 
   pcLADSPAPath = getenv("LADSPA_PATH");
   if (!pcLADSPAPath) {
-    fprintf(stderr,
-	    "Warning: You do not have a LADSPA_PATH "
-	    "environment variable set.\n");
-    return;
+    pcLADSPAPath = "/usr/lib/ladspa";
   }
   
   pcStart = pcLADSPAPath;

