From af5c800431182d26ece93b21794753823701467c Mon Sep 17 00:00:00 2001
From: Joaquim Rocha <me@joaquimrocha.com>
Date: Fri, 21 Sep 2018 22:34:09 +0200
Subject: [PATCH 2/3] Remove FoeCommand class namespace from method in class

---
 src/foecommand.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/foecommand.h b/src/foecommand.h
index 5869b54..34a533a 100644
--- a/src/foecommand.h
+++ b/src/foecommand.h
@@ -36,10 +36,10 @@ class FoeCommand : public BulletMLRunner {
   
   virtual void doChangeDirection(double d);
   virtual void doChangeSpeed(double s);
-  virtual void FoeCommand::doAccelX(double ax);
-  virtual void FoeCommand::doAccelY(double ay);
-  virtual double FoeCommand::getBulletSpeedX();
-  virtual double FoeCommand::getBulletSpeedY();
+  virtual void doAccelX(double ax);
+  virtual void doAccelY(double ay);
+  virtual double getBulletSpeedX();
+  virtual double getBulletSpeedY();
   
  private:
   struct foe *foe;
-- 
2.17.1

