From e13744781b0e0e25f0b08f0824f537ee2b4eaa09 Mon Sep 17 00:00:00 2001
From: Joaquim Rocha <me@joaquimrocha.com>
Date: Fri, 21 Sep 2018 22:36:24 +0200
Subject: [PATCH 3/3] Fix makefile

---
 src/makefile.lin | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/makefile.lin b/src/makefile.lin
index 2c4a467..a35efd1 100644
--- a/src/makefile.lin
+++ b/src/makefile.lin
@@ -10,12 +10,12 @@ CXX    = g++
 PROG        = $(NAME)
 
 DEFAULT_CFLAGS = `sdl-config --cflags`
-LDFLAGS        = `sdl-config --libs` -L. -lglut -lbulletml -lSDL_mixer -mwindows -lstdc++
+LDFLAGS        = `sdl-config --libs` -L. -lGLU -lGL -lbulletml -lSDL_mixer -lm
 
-MORE_CFLAGS = -O3
+MORE_CFLAGS = -O2 -Wall
 
 CFLAGS  = $(DEFAULT_CFLAGS) $(MORE_CFLAGS)
-CPPFLAGS  = $(DEFAULT_CFLAGS) $(MORE_CFLAGS) -I./bulletml/
+CPPFLAGS  = $(DEFAULT_CFLAGS) $(MORE_CFLAGS)
 
 OBJS =	$(NAME).$(O) ship.$(O) shot.$(O) frag.$(O) bonus.$(O) \
 	foe.$(O) foecommand.$(O) barragemanager.$(O) attractmanager.$(O) \
@@ -23,7 +23,7 @@ OBJS =	$(NAME).$(O) ship.$(O) shot.$(O) frag.$(O) bonus.$(O) \
 	screen.$(O) clrtbl.$(O) vector.$(O) degutil.$(O) rand.$(O) soundmanager.$(O)
 
 $(PROG): $(OBJS)
-	$(CC) $(CFLAGS) -o $(PROG) $(OBJS) $(LDFLAGS)
+	$(CXX) $(CFLAGS) -o $(PROG) $(OBJS) $(LDFLAGS)
 
 clean:
 	$(RM) $(PROG) *.$(O)
-- 
2.17.1

