From 793d3ef8db0f27fbd3eacac91e8b4bffbf1ce091 Mon Sep 17 00:00:00 2001
From: d2weber <29163905+d2weber@users.noreply.github.com>
Date: Wed, 18 Jun 2025 19:19:35 +0200
Subject: [PATCH 2/3] Fix install to prefix

---
 libs/quirc/Makefile | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/libs/quirc/Makefile b/libs/quirc/Makefile
index 235e9c9..5d8c0d6 100644
--- a/libs/quirc/Makefile
+++ b/libs/quirc/Makefile
@@ -86,9 +86,11 @@ libquirc.so.$(LIB_VERSION): $(LIB_OBJ)
 	$(CXX) $(QUIRC_CXXFLAGS) -o $@ -c $<
 
 install: libquirc.a libquirc.so.$(LIB_VERSION)
-	install -o root -g root -m 0644 lib/quirc.h $(DESTDIR)$(PREFIX)/include
-	install -o root -g root -m 0644 libquirc.a $(DESTDIR)$(PREFIX)/lib
-	install -o root -g root -m 0755 libquirc.so.$(LIB_VERSION) \
+	mkdir -p $(DESTDIR)$(PREFIX)/include
+	mkdir -p $(DESTDIR)$(PREFIX)/lib
+	cp lib/quirc.h $(DESTDIR)$(PREFIX)/include
+	cp libquirc.a $(DESTDIR)$(PREFIX)/lib
+	cp libquirc.so.$(LIB_VERSION) \
 		$(DESTDIR)$(PREFIX)/lib
 
 uninstall:
-- 
2.43.0

