From 0b6b115a17180ec454ba3da4efd91606a111389f Mon Sep 17 00:00:00 2001
From: d2weber <29163905+d2weber@users.noreply.github.com>
Date: Wed, 18 Jun 2025 22:49:32 +0200
Subject: [PATCH 3/3] Add .so symlink without version suffix

---
 libs/quirc/Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libs/quirc/Makefile b/libs/quirc/Makefile
index 5d8c0d6..d26cbba 100644
--- a/libs/quirc/Makefile
+++ b/libs/quirc/Makefile
@@ -72,8 +72,8 @@ libquirc.a: $(LIB_OBJ)
 	ar cru $@ $(LIB_OBJ)
 	ranlib $@
 
-.PHONY: libquirc.so
 libquirc.so: libquirc.so.$(LIB_VERSION)
+	ln -s $< $@
 
 libquirc.so.$(LIB_VERSION): $(LIB_OBJ)
 	$(CC) -shared -o $@ $(LIB_OBJ) $(LDFLAGS) -lm
@@ -85,13 +85,14 @@ libquirc.so.$(LIB_VERSION): $(LIB_OBJ)
 .cxx.o:
 	$(CXX) $(QUIRC_CXXFLAGS) -o $@ -c $<
 
-install: libquirc.a libquirc.so.$(LIB_VERSION)
+install: libquirc.a libquirc.so
 	mkdir -p $(DESTDIR)$(PREFIX)/include
 	mkdir -p $(DESTDIR)$(PREFIX)/lib
 	cp lib/quirc.h $(DESTDIR)$(PREFIX)/include
 	cp libquirc.a $(DESTDIR)$(PREFIX)/lib
 	cp libquirc.so.$(LIB_VERSION) \
 		$(DESTDIR)$(PREFIX)/lib
+	cp -d libquirc.so $(DESTDIR)$(PREFIX)/lib
 
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/include/quirc.h
-- 
2.43.0

