diff --git a/config.py b/config.py
index 1ff26e32..69057340 100644
--- a/config.py
+++ b/config.py
@@ -218,12 +218,11 @@ class Config:
     def SetupEnvironment( self, env, config, useGtk = False, useGtkGL = False, useJPEG = False, useZ = False, usePNG = False ):
         env['CC'] = self.cc
         env['CXX'] = self.cxx
-        ( ret, xml2 ) = commands.getstatusoutput( 'xml2-config --cflags' )
+        ( ret, xml2 ) = commands.getstatusoutput( 'pkg-config --cflags libxml-2.0' )
         if ( ret != 0 ):
-            print 'xml2-config failed'
+            print 'pkg-config could not find libxml-2.0'
             assert( False )
-        xml2libs = commands.getoutput( 'xml2-config --libs' )
-        env.ParseConfig( 'xml2-config --libs' )
+        env.ParseConfig( 'pkg-config --libs libxml-2.0' )
         baseflags = [ '-pipe', '-Wall', '-fmessage-length=0', '-fvisibility=hidden', xml2.split( ' ' ) ]
 
         if ( useGtk ):
