diff --git a/tools/quake3/common/threads.c b/tools/quake3/common/threads.c
index 89f24088..eefa3484 100644
--- a/tools/quake3/common/threads.c
+++ b/tools/quake3/common/threads.c
@@ -552,7 +552,7 @@ void RunThreadsOn( int workcnt, qboolean showpacifier, void ( *func )( int ) ){
 		for ( i = 0 ; i < numthreads ; i++ )
 		{
 			/* Default pthread attributes: joinable & non-realtime scheduling */
-			if ( pthread_create( &work_threads[i], NULL, (void*)func, (void*)(uintptr_t)i ) != 0 ) {
+			if ( pthread_create( &work_threads[i], NULL, (void*)func, (void*)(size_t)i ) != 0 ) {
 				Error( "pthread_create failed" );
 			}
 		}
