From 62cda978596a323cd7042722f906df906007e5b5 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Sat, 15 Apr 2017 13:24:39 -0700
Subject: [PATCH] Comment out GTK detection

---
 configure.in | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/configure.in b/configure.in
index 20b1355..2cb4753 100644
--- a/configure.in
+++ b/configure.in
@@ -208,13 +208,13 @@ AC_ARG_ENABLE(gtk_player,
 [  --enable-gtk-player     build a GTk sample SMPEG player [default=yes]],
               , enable_gtk_player=yes)
 have_gtk=no
-if test x$enable_gtk_player = xyes; then
-    AM_PATH_GTK(1.2.1, have_gtk=yes)
-    if test x$have_gtk = xyes; then
-        CFLAGS="$CFLAGS $GTK_CFLAGS"
-    fi
-    AC_SUBST(GTK_LIBS)
-fi
+dnl if test x$enable_gtk_player = xyes; then
+dnl     AM_PATH_GTK(1.2.1, have_gtk=yes)
+dnl     if test x$have_gtk = xyes; then
+dnl         CFLAGS="$CFLAGS $GTK_CFLAGS"
+dnl     fi
+dnl     AC_SUBST(GTK_LIBS)
+dnl fi
 AM_CONDITIONAL(HAVE_GTK, test x$have_gtk = xyes)
 
 dnl See if we can build the Mesa player
-- 
2.12.2

