--- a/linux/CMakeLists.txt	2025-05-25 16:58:31.000000000 +0200
+++ b/linux/CMakeLists.txt	2025-05-25 16:59:03.013600538 +0200
@@ -14,12 +14,14 @@
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2025/sqlite-autoconf-3500100.tar.gz
+    URL_HASH SHA256=00a65114d697cfaa8fe0630281d76fd1b77afcd95cd5e40ec6a02cbbadbfea71
     DOWNLOAD_EXTRACT_TIMESTAMP NEW
   )
 else()
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2025/sqlite-autoconf-3500100.tar.gz
+    URL_HASH SHA256=00a65114d697cfaa8fe0630281d76fd1b77afcd95cd5e40ec6a02cbbadbfea71
   )
 endif()
 FetchContent_MakeAvailable(sqlite3)
