--- a/linux/CMakeLists.txt      2025-05-25 16:58:31.000000000 +0200
+++ b/linux/CMakeLists.txt      2025-05-25 16:59:03.013600538 +0200
@@ -14,12 +14,14 @@
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2025/sqlite-autoconf-3500400.tar.gz
+    URL_HASH SHA256=a3db587a1b92ee5ddac2f66b3edb41b26f9c867275782d46c3a088977d6a5b18
     DOWNLOAD_EXTRACT_TIMESTAMP NEW
   )
 else()
   FetchContent_Declare(
     sqlite3
     URL https://sqlite.org/2025/sqlite-autoconf-3500400.tar.gz
+    URL_HASH SHA256=a3db587a1b92ee5ddac2f66b3edb41b26f9c867275782d46c3a088977d6a5b18
   )
 endif()
 FetchContent_MakeAvailable(sqlite3)