From 10e4ebff3f47fd60356a756c4d93e492afd292cd Mon Sep 17 00:00:00 2001
From: Dylan McCall <dylan@endlessos.org>
Date: Fri, 8 Apr 2022 14:12:35 -0700
Subject: [PATCH] Remove needless ifcfg warning

---
 src/ifcfg/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ifcfg/__init__.py b/src/ifcfg/__init__.py
index 5f4ebdb..fe4bea5 100644
--- a/src/ifcfg/__init__.py
+++ b/src/ifcfg/__init__.py
@@ -25,7 +25,7 @@ def get_parser_class():
         if not os.path.exists(Parser.get_command().split(' ')[0]):
             Parser = parser.UnixIPParser
             if not os.path.exists(Parser.get_command().split(' ')[0]):
-                Log.warning("Neither `ifconfig` (`%s`) nor `ip` (`%s`) commands are available, listing network interfaces is likely to fail",
+                Log.debug("Neither `ifconfig` (`%s`) nor `ip` (`%s`) commands are available, listing network interfaces is likely to fail",
                             parser.LinuxParser.get_command(),
                             parser.UnixIPParser.get_command())
     elif distro in ['Darwin', 'MacOSX']:
-- 
2.35.1

