From bba34e509c2e144000becdc89e18bf6e34fe5517 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hubert=20Figui=C3=A8re?= <hub@figuiere.net>
Date: Tue, 18 Nov 2025 20:33:54 -0500
Subject: [PATCH 1/5] Build xmm_quantize_sub.c with -msse
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

From: Sebastian Ramacher <sramacher@debian.org>
Bug: http://sourceforge.net/p/lame/bugs/443/
Bug-Debian: https://bugs.debian.org/760047
Forwarded: http://sourceforge.net/p/lame/bugs/443/
Last-Update: 2014-08-31

Signed-off-by: Hubert Figuière <hub@figuiere.net>
---
 libmp3lame/vector/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libmp3lame/vector/Makefile.am b/libmp3lame/vector/Makefile.am
index ac595b1..6b05380 100644
--- a/libmp3lame/vector/Makefile.am
+++ b/libmp3lame/vector/Makefile.am
@@ -20,6 +20,7 @@ xmm_sources = xmm_quantize_sub.c
 
 if WITH_XMM
 liblamevectorroutines_la_SOURCES = $(xmm_sources)
+liblamevectorroutines_la_CFLAGS = -msse
 endif
 
 noinst_HEADERS = lame_intrin.h
-- 
2.51.0

