From 850ca75cc47fef9ed3eb6dfe4e95b6ac715aa48f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hubert=20Figui=C3=A8re?= <hub@figuiere.net>
Date: Tue, 18 Nov 2025 21:41:31 -0500
Subject: [PATCH 5/5] Update autoconf
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Hubert Figuière <hub@figuiere.net>
---
 configure.in | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/configure.in b/configure.in
index 05b2fe7..ecafb45 100644
--- a/configure.in
+++ b/configure.in
@@ -17,9 +17,10 @@ dnl FRONTEND_LDFLAGS
 dnl FRONTEND_CFLAGS
 dnl FRONTEND_LDADD
 
-AC_PREREQ(2.68)
+AC_PREREQ([2.72])
 AC_INIT([lame],[3.99.5],[lame-dev@lists.sf.net])
 AC_CONFIG_SRCDIR([libmp3lame/lame.c])
+AC_CONFIG_MACRO_DIRS([m4])
 AC_LANG([C])
 
 dnl check system
@@ -37,9 +38,9 @@ AM_MAINTAINER_MODE
 AM_MAKE_INCLUDE
 
 dnl check environment
-AC_AIX
-AC_ISC_POSIX
-AC_MINIX
+AC_USE_SYSTEM_EXTENSIONS
+AC_SEARCH_LIBS([strerror],[cposix])
+AC_USE_SYSTEM_EXTENSIONS
 case $host_os in
   *cygwin* ) CYGWIN=yes;;
          * ) CYGWIN=no;;
@@ -47,7 +48,7 @@ esac
 
 dnl libtool
 # AC_DISABLE_SHARED
-AC_PROG_LIBTOOL
+LT_INIT
 AC_SUBST(LIBTOOL_DEPS)
 CFLAGS="${ac_save_CFLAGS}"
 
@@ -85,7 +86,6 @@ if test "${ac_cv_header_dmalloc_h}" = "yes"; then
 fi
 
 dnl Checks for header files.
-AC_HEADER_STDC
 AC_CHECK_HEADERS( \
 		 errno.h \
 		 fcntl.h \
@@ -135,8 +135,8 @@ if test $ac_cv_sizeof_short -eq 0 \
 	exit 1
 fi
 
-AC_C_LONG_DOUBLE
-if test "${ac_cv_c_have_long_double}" = "yes" ; then
+AC_TYPE_LONG_DOUBLE_WIDER
+if test "${ac_cv_type_long_double_wider}" = "yes" ; then
 	AC_CHECK_SIZEOF(long double)
 fi
 
@@ -324,7 +324,6 @@ if test ${cross_compiling} = "yes"; then
 fi
 
 AC_TYPE_SIZE_T
-AC_HEADER_TIME
 
 dnl Checks for library functions.
 AC_FUNC_ALLOCA
-- 
2.51.0

