From b3abdb95be2737222bb228e6e5b807661ebcaf84 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Wed, 31 Aug 2022 10:09:01 -0500
Subject: [PATCH] Clang build script: Don't build against the sysroot

---
 tools/clang/scripts/build.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/tools/clang/scripts/build.py b/tools/clang/scripts/build.py
index 44f2a2e6af5be..7e290bafa8faa 100755
--- a/tools/clang/scripts/build.py
+++ b/tools/clang/scripts/build.py
@@ -903,7 +903,7 @@ def main():
     if sys.platform.startswith('linux'):
       base_cmake_args += [ '-DLLVM_STATIC_LINK_CXX_STDLIB=ON' ]
 
-  if sys.platform.startswith('linux'):
+  if sys.platform.startswith('linux') and False:
     sysroot_amd64 = DownloadDebianSysroot('amd64', args.skip_checkout)
     sysroot_i386 = DownloadDebianSysroot('i386', args.skip_checkout)
     sysroot_arm = DownloadDebianSysroot('arm', args.skip_checkout)
@@ -1194,7 +1194,7 @@ def main():
   if sys.platform.startswith('linux'):
     runtimes_triples_args['i386-unknown-linux-gnu'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_i386,
+            # 'CMAKE_SYSROOT=%s' % sysroot_i386,
             # TODO(https://crbug.com/1374690): pass proper flags to i386 tests so they compile correctly
             'LLVM_INCLUDE_TESTS=OFF',
         ],
@@ -1205,7 +1205,7 @@ def main():
     }
     runtimes_triples_args['x86_64-unknown-linux-gnu'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_amd64,
+            # 'CMAKE_SYSROOT=%s' % sysroot_amd64,
         ],
         "profile": True,
         "sanitizers": True,
@@ -1219,7 +1219,7 @@ def main():
     # Maybe it should work for builtins too?
     runtimes_triples_args['armv7-unknown-linux-gnueabihf'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_arm,
+            # 'CMAKE_SYSROOT=%s' % sysroot_arm,
             # Can't run tests on x86 host.
             'LLVM_INCLUDE_TESTS=OFF',
         ],
@@ -1230,7 +1230,7 @@ def main():
     }
     runtimes_triples_args['aarch64-unknown-linux-gnu'] = {
         "args": [
-            'CMAKE_SYSROOT=%s' % sysroot_arm64,
+            # 'CMAKE_SYSROOT=%s' % sysroot_arm64,
             # Can't run tests on x86 host.
             'LLVM_INCLUDE_TESTS=OFF',
         ],
-- 
2.43.2

