From 468a41ee4a45c8fab34d40b741e2896868435045 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <git@refi64.dev>
Date: Wed, 13 Dec 2023 14:13:55 -0600
Subject: [PATCH] media: Add libfdk_aac to the audio codec allow list

---
 media/filters/ffmpeg_glue.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/filters/ffmpeg_glue.cc b/media/filters/ffmpeg_glue.cc
index af52aeb5173af..58e4703684de1 100644
--- a/media/filters/ffmpeg_glue.cc
+++ b/media/filters/ffmpeg_glue.cc
@@ -145,7 +145,7 @@ const char* FFmpegGlue::GetAllowedAudioDecoders() {
         "vorbis,libopus,flac,pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,"
         "mp3,pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw");
 #if BUILDFLAG(USE_PROPRIETARY_CODECS)
-    allowed_decoders += ",aac";
+    allowed_decoders += ",libfdk_aac";
 #endif
     return allowed_decoders;
   }());
-- 
2.43.2

