diff --git a/data/org.gnome.NetworkDisplays.appdata.xml.in b/data/org.gnome.NetworkDisplays.appdata.xml.in
index 18785a3..f731c92 100644
--- a/data/org.gnome.NetworkDisplays.appdata.xml.in
+++ b/data/org.gnome.NetworkDisplays.appdata.xml.in
@@ -17,6 +17,9 @@
 		GNOME Network Displays allows you to cast your desktop to a remote display. Currently
 		implemented is support for casting to Wi-Fi Display capable devices (a.k.a. Miracast).
 		</p>
+		<p>
+		This software requires PipeWire version 0.3.8 or higher on the host.
+		</p>
 	</description>
 	<update_contact>bberg@gnome.org</update_contact>
 	<content_rating type="oars-1.1"/>
