Index: gtkmathview/src/backend/ps/T1_FontDataBase.hh
===================================================================
--- gtkmathview.orig/src/backend/ps/T1_FontDataBase.hh	2014-03-05 18:48:33.237410987 +0100
+++ gtkmathview/src/backend/ps/T1_FontDataBase.hh	2014-03-05 18:48:54.161411801 +0100
@@ -19,7 +19,6 @@
 #ifndef __T1_FontDataBase_hh__
 #define __T1_FontDataBase_hh__
 
-#include <t1lib.h>
 #include <config.h>
 #include <map>
 #include <list>
