From 3f5cf59824a37ea34d1aeac897f6405cb9f76c0f Mon Sep 17 00:00:00 2001
From: casept <davids.paskevics@gmail.com>
Date: Mon, 17 Apr 2017 15:01:55 +0200
Subject: [PATCH] Apply patch_include_il.h from NetBSD

---
 include/IL/il.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/IL/il.h b/include/IL/il.h
index 540a56e..21fd6b2 100644
--- a/include/IL/il.h
+++ b/include/IL/il.h
@@ -63,7 +63,7 @@ extern "C" {
 	#endif
 #endif
 
-#ifdef RESTRICT_KEYWORD
+#if defined(RESTRICT_KEYWORD) && !defined(__cplusplus)
 #define RESTRICT restrict
 #define CONST_RESTRICT const restrict
 #else
-- 
2.7.4

