diff -ru giac-1.9.0.orig/src/Graph.cc giac-1.9.0/src/Graph.cc
--- giac-1.9.0.orig/src/Graph.cc	2022-11-04 19:00:52.059786792 +0100
+++ giac-1.9.0/src/Graph.cc	2022-11-04 19:05:34.765503717 +0100
@@ -5640,7 +5640,7 @@
   }
 
   bool inputdouble(const char * msg,double & d,GIAC_CONTEXT){
-    const char * ch=fl_input(msg,print_DOUBLE_(d).c_str());
+    const char * ch=fl_input("%s",msg,print_DOUBLE_(d).c_str());
     if (!ch) return false;
     gen g(ch,contextptr);
     g=evalf_double(g,1,contextptr);
