diff --git a/Palaso.sln b/Palaso.sln
index 284353c1..89068322 100755
--- a/Palaso.sln
+++ b/Palaso.sln
@@ -3,101 +3,45 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 16
 VisualStudioVersion = 16.0.29409.12
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Core.Tests", "SIL.Core.Tests\SIL.Core.Tests.csproj", "{5DE33CD7-60CB-4B9F-A123-A83C1C686E47}"
-	ProjectSection(ProjectDependencies) = postProject
-		{24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4} = {24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4}
-	EndProjectSection
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms", "SIL.Windows.Forms\SIL.Windows.Forms.csproj", "{DB44F49C-D8C6-434F-81ED-28EA5C9E8195}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AddSortKey", "AddSortKey\AddSortKey.csproj", "{3D4F73A8-DB05-4580-98A0-DEE34C010636}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.Tests", "SIL.Windows.Forms.Tests\SIL.Windows.Forms.Tests.csproj", "{6E16866F-FF8D-4136-AD67-36AFE3626432}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Media", "SIL.Media\SIL.Media.csproj", "{66B3504A-3B4E-45CB-903A-A9A75B22EF68}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Media.Tests", "SIL.Media.Tests\SIL.Media.Tests.csproj", "{CD4AF11C-C214-4D90-8F12-6DA97836B800}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.TestUtilities", "SIL.TestUtilities\SIL.TestUtilities.csproj", "{D74CE910-D44A-44F5-8C0F-F5E50B26F85D}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Reporting.TestApp", "TestApps\ReportingTest\Reporting.TestApp.csproj", "{13B5E313-1B26-4DAE-9E96-133768B62FD2}"
-	ProjectSection(ProjectDependencies) = postProject
-		{DB44F49C-D8C6-434F-81ED-28EA5C9E8195} = {DB44F49C-D8C6-434F-81ED-28EA5C9E8195}
-	EndProjectSection
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Archiving", "SIL.Archiving\SIL.Archiving.csproj", "{BCE1F124-5479-4B23-90B1-B7A4EBE44FA3}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Archiving.Tests", "SIL.Archiving.Tests\SIL.Archiving.Tests.csproj", "{892C7F20-FBBB-4AB3-BAC2-E40A135567B6}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Scripture", "SIL.Scripture\SIL.Scripture.csproj", "{F71BA7B9-D9DC-4F8C-A307-87B503D0E05B}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Scripture.Tests", "SIL.Scripture.Tests\SIL.Scripture.Tests.csproj", "{4528287F-BA86-49D8-AFBB-1250A60CE7EA}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TestAppKeyboard", "TestApps\TestAppKeyboard\TestAppKeyboard.csproj", "{776BECCD-102A-4C69-A478-4A4A6F695019}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.GeckoBrowserAdapter", "SIL.Windows.Forms.GeckoBrowserAdapter\SIL.Windows.Forms.GeckoBrowserAdapter.csproj", "{D000879E-9AA6-4811-A6F0-80DF6119FB08}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.Scripture", "SIL.Windows.Forms.Scripture\SIL.Windows.Forms.Scripture.csproj", "{6CA85A3E-4AB3-4BBC-861F-0C68F018868B}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.Scripture.Tests", "SIL.Windows.Forms.Scripture.Tests\SIL.Windows.Forms.Scripture.Tests.csproj", "{CCF34218-52D9-4092-ADB8-1893BBE5ED31}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.WritingSystems.Tests", "SIL.WritingSystems.Tests\SIL.WritingSystems.Tests.csproj", "{641EB241-BF16-4A5F-89BD-54411F9AC973}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.WritingSystems", "SIL.Windows.Forms.WritingSystems\SIL.Windows.Forms.WritingSystems.csproj", "{F7DF58E2-5B67-498A-9F64-B5F1584A5AB9}"
 	ProjectSection(ProjectDependencies) = postProject
 		{0580ED5F-6274-4928-83EE-175F01AD052D} = {0580ED5F-6274-4928-83EE-175F01AD052D}
 	EndProjectSection
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.WritingSystems.Tests", "SIL.Windows.Forms.WritingSystems.Tests\SIL.Windows.Forms.WritingSystems.Tests.csproj", "{24E0B3A0-EE71-48CE-82F9-549F1F47322F}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.Keyboarding", "SIL.Windows.Forms.Keyboarding\SIL.Windows.Forms.Keyboarding.csproj", "{0580ED5F-6274-4928-83EE-175F01AD052D}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.Keyboarding.Tests", "SIL.Windows.Forms.Keyboarding.Tests\SIL.Windows.Forms.Keyboarding.Tests.csproj", "{B732A1A3-5CAF-4BC3-B127-A154F2C7E4FA}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.DblBundle", "SIL.DblBundle\SIL.DblBundle.csproj", "{D4F0F320-FAF6-4775-8411-4D03FAE99F84}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.DblBundle.Tests", "SIL.DblBundle.Tests\SIL.DblBundle.Tests.csproj", "{66685009-8F78-4FED-A106-E5BD6E746E1A}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Lexicon", "SIL.Lexicon\SIL.Lexicon.csproj", "{FAD20DE2-D5BB-4D9D-AEEF-61F02663105F}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Lexicon.Tests", "SIL.Lexicon.Tests\SIL.Lexicon.Tests.csproj", "{67C6AD7F-B1AC-46E1-90C7-E4E2B416AEBD}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.DictionaryServices.Tests", "SIL.DictionaryServices.Tests\SIL.DictionaryServices.Tests.csproj", "{CC84BDED-CECD-466A-9397-F5D83527B218}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.DictionaryServices", "SIL.DictionaryServices\SIL.DictionaryServices.csproj", "{5FF5CBD8-B75D-4F79-A39D-2D898FBFADBD}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Lift", "SIL.Lift\SIL.Lift.csproj", "{60B4A08C-54F3-4BC7-83ED-F84274FABFAC}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Lift.Tests", "SIL.Lift.Tests\SIL.Lift.Tests.csproj", "{FE61F837-8EAA-4FD7-824A-9949194430DC}"
-EndProject
-Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "TestApps", "TestApps", "{3C695301-9C39-4715-8EDD-817C63EF81E7}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Email.TestApp", "TestApps\SIL.Email.TestApp\SIL.Email.TestApp.csproj", "{2A206833-C713-4ACB-ACD3-EB971415119E}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.TestApp", "TestApps\SIL.Windows.Forms.TestApp\SIL.Windows.Forms.TestApp.csproj", "{24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4}"
-	ProjectSection(ProjectDependencies) = postProject
-		{0580ED5F-6274-4928-83EE-175F01AD052D} = {0580ED5F-6274-4928-83EE-175F01AD052D}
-		{D000879E-9AA6-4811-A6F0-80DF6119FB08} = {D000879E-9AA6-4811-A6F0-80DF6119FB08}
-		{FAD20DE2-D5BB-4D9D-AEEF-61F02663105F} = {FAD20DE2-D5BB-4D9D-AEEF-61F02663105F}
-		{F7DF58E2-5B67-498A-9F64-B5F1584A5AB9} = {F7DF58E2-5B67-498A-9F64-B5F1584A5AB9}
-	EndProjectSection
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "FastSplitterTest", "TestApps\FastSplitterTest\FastSplitterTest.csproj", "{CA8F3657-3D85-4C97-81A6-5C212627B2F6}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Windows.Forms.DblBundle", "SIL.Windows.Forms.DblBundle\SIL.Windows.Forms.DblBundle.csproj", "{A5260B7F-A48B-495C-86C4-73B1419B17F2}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Linux.Logging", "SIL.Linux.Logging\SIL.Linux.Logging.csproj", "{D098D062-0B47-4A68-BE0A-5FB597B6896C}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Linux.Logging.Tests", "SIL.Linux.Logging.Tests\SIL.Linux.Logging.Tests.csproj", "{D7AD97B3-88DB-400D-B748-0E06914F17FC}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Core.Desktop", "SIL.Core.Desktop\SIL.Core.Desktop.csproj", "{C48DE000-2086-4CEB-A5A1-3171C272725E}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Core.Desktop.Tests", "SIL.Core.Desktop.Tests\SIL.Core.Desktop.Tests.csproj", "{4C21E03C-8A79-4D28-A4A5-A14BA1342BCF}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ExtractCopyright", "ExtractCopyright\ExtractCopyright.csproj", "{EB720294-B74B-47F4-BE43-9B3B4C33C86D}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ExtractCopyright.Tests", "ExtractCopyright.Tests\ExtractCopyright.Tests.csproj", "{085A7785-C407-4623-80C0-BFFD2B0D2475}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.Core", "SIL.Core\SIL.Core.csproj", "{53CB36D9-2F66-4823-B1F0-4523E8565200}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.WritingSystems", "SIL.WritingSystems\SIL.WritingSystems.csproj", "{A61CC666-1FAA-4A92-BB3B-AA36BD66BDDA}"
@@ -114,20 +58,12 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		Directory.Build.targets = Directory.Build.targets
 	EndProjectSection
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SIL.TestUtilities.Tests", "SIL.TestUtilities.Tests\SIL.TestUtilities.Tests.csproj", "{CCB5E6CC-8CC1-4248-924E-D4F36CBEF37D}"
-EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ClipboardTestApp", "TestApps\ClipboardTestApp\ClipboardTestApp.csproj", "{B1E0B522-FEDF-497E-BFCD-A572F67C03C1}"
-EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
 		Release|Any CPU = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{5DE33CD7-60CB-4B9F-A123-A83C1C686E47}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{5DE33CD7-60CB-4B9F-A123-A83C1C686E47}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{5DE33CD7-60CB-4B9F-A123-A83C1C686E47}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{5DE33CD7-60CB-4B9F-A123-A83C1C686E47}.Release|Any CPU.Build.0 = Release|Any CPU
 		{DB44F49C-D8C6-434F-81ED-28EA5C9E8195}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{DB44F49C-D8C6-434F-81ED-28EA5C9E8195}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{DB44F49C-D8C6-434F-81ED-28EA5C9E8195}.Release|Any CPU.ActiveCfg = Release|Any CPU
@@ -136,46 +72,22 @@ Global
 		{3D4F73A8-DB05-4580-98A0-DEE34C010636}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{3D4F73A8-DB05-4580-98A0-DEE34C010636}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{3D4F73A8-DB05-4580-98A0-DEE34C010636}.Release|Any CPU.Build.0 = Release|Any CPU
-		{6E16866F-FF8D-4136-AD67-36AFE3626432}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{6E16866F-FF8D-4136-AD67-36AFE3626432}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{6E16866F-FF8D-4136-AD67-36AFE3626432}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{6E16866F-FF8D-4136-AD67-36AFE3626432}.Release|Any CPU.Build.0 = Release|Any CPU
 		{66B3504A-3B4E-45CB-903A-A9A75B22EF68}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{66B3504A-3B4E-45CB-903A-A9A75B22EF68}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{66B3504A-3B4E-45CB-903A-A9A75B22EF68}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{66B3504A-3B4E-45CB-903A-A9A75B22EF68}.Release|Any CPU.Build.0 = Release|Any CPU
-		{CD4AF11C-C214-4D90-8F12-6DA97836B800}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{CD4AF11C-C214-4D90-8F12-6DA97836B800}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{CD4AF11C-C214-4D90-8F12-6DA97836B800}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{CD4AF11C-C214-4D90-8F12-6DA97836B800}.Release|Any CPU.Build.0 = Release|Any CPU
 		{D74CE910-D44A-44F5-8C0F-F5E50B26F85D}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{D74CE910-D44A-44F5-8C0F-F5E50B26F85D}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{D74CE910-D44A-44F5-8C0F-F5E50B26F85D}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{D74CE910-D44A-44F5-8C0F-F5E50B26F85D}.Release|Any CPU.Build.0 = Release|Any CPU
-		{13B5E313-1B26-4DAE-9E96-133768B62FD2}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{13B5E313-1B26-4DAE-9E96-133768B62FD2}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{13B5E313-1B26-4DAE-9E96-133768B62FD2}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{13B5E313-1B26-4DAE-9E96-133768B62FD2}.Release|Any CPU.Build.0 = Release|Any CPU
 		{BCE1F124-5479-4B23-90B1-B7A4EBE44FA3}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{BCE1F124-5479-4B23-90B1-B7A4EBE44FA3}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{BCE1F124-5479-4B23-90B1-B7A4EBE44FA3}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{BCE1F124-5479-4B23-90B1-B7A4EBE44FA3}.Release|Any CPU.Build.0 = Release|Any CPU
-		{892C7F20-FBBB-4AB3-BAC2-E40A135567B6}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{892C7F20-FBBB-4AB3-BAC2-E40A135567B6}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{892C7F20-FBBB-4AB3-BAC2-E40A135567B6}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{892C7F20-FBBB-4AB3-BAC2-E40A135567B6}.Release|Any CPU.Build.0 = Release|Any CPU
 		{F71BA7B9-D9DC-4F8C-A307-87B503D0E05B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{F71BA7B9-D9DC-4F8C-A307-87B503D0E05B}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{F71BA7B9-D9DC-4F8C-A307-87B503D0E05B}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{F71BA7B9-D9DC-4F8C-A307-87B503D0E05B}.Release|Any CPU.Build.0 = Release|Any CPU
-		{4528287F-BA86-49D8-AFBB-1250A60CE7EA}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{4528287F-BA86-49D8-AFBB-1250A60CE7EA}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{4528287F-BA86-49D8-AFBB-1250A60CE7EA}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{4528287F-BA86-49D8-AFBB-1250A60CE7EA}.Release|Any CPU.Build.0 = Release|Any CPU
-		{776BECCD-102A-4C69-A478-4A4A6F695019}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{776BECCD-102A-4C69-A478-4A4A6F695019}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{776BECCD-102A-4C69-A478-4A4A6F695019}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{776BECCD-102A-4C69-A478-4A4A6F695019}.Release|Any CPU.Build.0 = Release|Any CPU
 		{D000879E-9AA6-4811-A6F0-80DF6119FB08}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{D000879E-9AA6-4811-A6F0-80DF6119FB08}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{D000879E-9AA6-4811-A6F0-80DF6119FB08}.Release|Any CPU.ActiveCfg = Release|Any CPU
@@ -184,50 +96,22 @@ Global
 		{6CA85A3E-4AB3-4BBC-861F-0C68F018868B}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{6CA85A3E-4AB3-4BBC-861F-0C68F018868B}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{6CA85A3E-4AB3-4BBC-861F-0C68F018868B}.Release|Any CPU.Build.0 = Release|Any CPU
-		{CCF34218-52D9-4092-ADB8-1893BBE5ED31}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{CCF34218-52D9-4092-ADB8-1893BBE5ED31}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{CCF34218-52D9-4092-ADB8-1893BBE5ED31}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{CCF34218-52D9-4092-ADB8-1893BBE5ED31}.Release|Any CPU.Build.0 = Release|Any CPU
-		{641EB241-BF16-4A5F-89BD-54411F9AC973}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{641EB241-BF16-4A5F-89BD-54411F9AC973}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{641EB241-BF16-4A5F-89BD-54411F9AC973}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{641EB241-BF16-4A5F-89BD-54411F9AC973}.Release|Any CPU.Build.0 = Release|Any CPU
 		{F7DF58E2-5B67-498A-9F64-B5F1584A5AB9}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{F7DF58E2-5B67-498A-9F64-B5F1584A5AB9}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{F7DF58E2-5B67-498A-9F64-B5F1584A5AB9}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{F7DF58E2-5B67-498A-9F64-B5F1584A5AB9}.Release|Any CPU.Build.0 = Release|Any CPU
-		{24E0B3A0-EE71-48CE-82F9-549F1F47322F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{24E0B3A0-EE71-48CE-82F9-549F1F47322F}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{24E0B3A0-EE71-48CE-82F9-549F1F47322F}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{24E0B3A0-EE71-48CE-82F9-549F1F47322F}.Release|Any CPU.Build.0 = Release|Any CPU
 		{0580ED5F-6274-4928-83EE-175F01AD052D}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{0580ED5F-6274-4928-83EE-175F01AD052D}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{0580ED5F-6274-4928-83EE-175F01AD052D}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{0580ED5F-6274-4928-83EE-175F01AD052D}.Release|Any CPU.Build.0 = Release|Any CPU
-		{B732A1A3-5CAF-4BC3-B127-A154F2C7E4FA}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{B732A1A3-5CAF-4BC3-B127-A154F2C7E4FA}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{B732A1A3-5CAF-4BC3-B127-A154F2C7E4FA}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{B732A1A3-5CAF-4BC3-B127-A154F2C7E4FA}.Release|Any CPU.Build.0 = Release|Any CPU
 		{D4F0F320-FAF6-4775-8411-4D03FAE99F84}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{D4F0F320-FAF6-4775-8411-4D03FAE99F84}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{D4F0F320-FAF6-4775-8411-4D03FAE99F84}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{D4F0F320-FAF6-4775-8411-4D03FAE99F84}.Release|Any CPU.Build.0 = Release|Any CPU
-		{66685009-8F78-4FED-A106-E5BD6E746E1A}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{66685009-8F78-4FED-A106-E5BD6E746E1A}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{66685009-8F78-4FED-A106-E5BD6E746E1A}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{66685009-8F78-4FED-A106-E5BD6E746E1A}.Release|Any CPU.Build.0 = Release|Any CPU
 		{FAD20DE2-D5BB-4D9D-AEEF-61F02663105F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{FAD20DE2-D5BB-4D9D-AEEF-61F02663105F}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{FAD20DE2-D5BB-4D9D-AEEF-61F02663105F}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{FAD20DE2-D5BB-4D9D-AEEF-61F02663105F}.Release|Any CPU.Build.0 = Release|Any CPU
-		{67C6AD7F-B1AC-46E1-90C7-E4E2B416AEBD}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{67C6AD7F-B1AC-46E1-90C7-E4E2B416AEBD}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{67C6AD7F-B1AC-46E1-90C7-E4E2B416AEBD}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{67C6AD7F-B1AC-46E1-90C7-E4E2B416AEBD}.Release|Any CPU.Build.0 = Release|Any CPU
-		{CC84BDED-CECD-466A-9397-F5D83527B218}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{CC84BDED-CECD-466A-9397-F5D83527B218}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{CC84BDED-CECD-466A-9397-F5D83527B218}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{CC84BDED-CECD-466A-9397-F5D83527B218}.Release|Any CPU.Build.0 = Release|Any CPU
 		{5FF5CBD8-B75D-4F79-A39D-2D898FBFADBD}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{5FF5CBD8-B75D-4F79-A39D-2D898FBFADBD}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{5FF5CBD8-B75D-4F79-A39D-2D898FBFADBD}.Release|Any CPU.ActiveCfg = Release|Any CPU
@@ -236,22 +120,6 @@ Global
 		{60B4A08C-54F3-4BC7-83ED-F84274FABFAC}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{60B4A08C-54F3-4BC7-83ED-F84274FABFAC}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{60B4A08C-54F3-4BC7-83ED-F84274FABFAC}.Release|Any CPU.Build.0 = Release|Any CPU
-		{FE61F837-8EAA-4FD7-824A-9949194430DC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{FE61F837-8EAA-4FD7-824A-9949194430DC}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{FE61F837-8EAA-4FD7-824A-9949194430DC}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{FE61F837-8EAA-4FD7-824A-9949194430DC}.Release|Any CPU.Build.0 = Release|Any CPU
-		{2A206833-C713-4ACB-ACD3-EB971415119E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{2A206833-C713-4ACB-ACD3-EB971415119E}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{2A206833-C713-4ACB-ACD3-EB971415119E}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{2A206833-C713-4ACB-ACD3-EB971415119E}.Release|Any CPU.Build.0 = Release|Any CPU
-		{24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4}.Release|Any CPU.Build.0 = Release|Any CPU
-		{CA8F3657-3D85-4C97-81A6-5C212627B2F6}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{CA8F3657-3D85-4C97-81A6-5C212627B2F6}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{CA8F3657-3D85-4C97-81A6-5C212627B2F6}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{CA8F3657-3D85-4C97-81A6-5C212627B2F6}.Release|Any CPU.Build.0 = Release|Any CPU
 		{A5260B7F-A48B-495C-86C4-73B1419B17F2}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{A5260B7F-A48B-495C-86C4-73B1419B17F2}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{A5260B7F-A48B-495C-86C4-73B1419B17F2}.Release|Any CPU.ActiveCfg = Release|Any CPU
@@ -260,26 +128,14 @@ Global
 		{D098D062-0B47-4A68-BE0A-5FB597B6896C}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{D098D062-0B47-4A68-BE0A-5FB597B6896C}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{D098D062-0B47-4A68-BE0A-5FB597B6896C}.Release|Any CPU.Build.0 = Release|Any CPU
-		{D7AD97B3-88DB-400D-B748-0E06914F17FC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{D7AD97B3-88DB-400D-B748-0E06914F17FC}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{D7AD97B3-88DB-400D-B748-0E06914F17FC}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{D7AD97B3-88DB-400D-B748-0E06914F17FC}.Release|Any CPU.Build.0 = Release|Any CPU
 		{C48DE000-2086-4CEB-A5A1-3171C272725E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{C48DE000-2086-4CEB-A5A1-3171C272725E}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{C48DE000-2086-4CEB-A5A1-3171C272725E}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{C48DE000-2086-4CEB-A5A1-3171C272725E}.Release|Any CPU.Build.0 = Release|Any CPU
-		{4C21E03C-8A79-4D28-A4A5-A14BA1342BCF}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{4C21E03C-8A79-4D28-A4A5-A14BA1342BCF}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{4C21E03C-8A79-4D28-A4A5-A14BA1342BCF}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{4C21E03C-8A79-4D28-A4A5-A14BA1342BCF}.Release|Any CPU.Build.0 = Release|Any CPU
 		{EB720294-B74B-47F4-BE43-9B3B4C33C86D}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{EB720294-B74B-47F4-BE43-9B3B4C33C86D}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{EB720294-B74B-47F4-BE43-9B3B4C33C86D}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{EB720294-B74B-47F4-BE43-9B3B4C33C86D}.Release|Any CPU.Build.0 = Release|Any CPU
-		{085A7785-C407-4623-80C0-BFFD2B0D2475}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{085A7785-C407-4623-80C0-BFFD2B0D2475}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{085A7785-C407-4623-80C0-BFFD2B0D2475}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{085A7785-C407-4623-80C0-BFFD2B0D2475}.Release|Any CPU.Build.0 = Release|Any CPU
 		{53CB36D9-2F66-4823-B1F0-4523E8565200}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{53CB36D9-2F66-4823-B1F0-4523E8565200}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{53CB36D9-2F66-4823-B1F0-4523E8565200}.Release|Any CPU.ActiveCfg = Release|Any CPU
@@ -288,26 +144,10 @@ Global
 		{A61CC666-1FAA-4A92-BB3B-AA36BD66BDDA}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{A61CC666-1FAA-4A92-BB3B-AA36BD66BDDA}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{A61CC666-1FAA-4A92-BB3B-AA36BD66BDDA}.Release|Any CPU.Build.0 = Release|Any CPU
-		{CCB5E6CC-8CC1-4248-924E-D4F36CBEF37D}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{CCB5E6CC-8CC1-4248-924E-D4F36CBEF37D}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{CCB5E6CC-8CC1-4248-924E-D4F36CBEF37D}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{CCB5E6CC-8CC1-4248-924E-D4F36CBEF37D}.Release|Any CPU.Build.0 = Release|Any CPU
-		{B1E0B522-FEDF-497E-BFCD-A572F67C03C1}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{B1E0B522-FEDF-497E-BFCD-A572F67C03C1}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{B1E0B522-FEDF-497E-BFCD-A572F67C03C1}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{B1E0B522-FEDF-497E-BFCD-A572F67C03C1}.Release|Any CPU.Build.0 = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
 	EndGlobalSection
-	GlobalSection(NestedProjects) = preSolution
-		{13B5E313-1B26-4DAE-9E96-133768B62FD2} = {3C695301-9C39-4715-8EDD-817C63EF81E7}
-		{776BECCD-102A-4C69-A478-4A4A6F695019} = {3C695301-9C39-4715-8EDD-817C63EF81E7}
-		{2A206833-C713-4ACB-ACD3-EB971415119E} = {3C695301-9C39-4715-8EDD-817C63EF81E7}
-		{24FAE6A1-3B6E-4E5A-BAB2-78DCAA1566C4} = {3C695301-9C39-4715-8EDD-817C63EF81E7}
-		{CA8F3657-3D85-4C97-81A6-5C212627B2F6} = {3C695301-9C39-4715-8EDD-817C63EF81E7}
-		{B1E0B522-FEDF-497E-BFCD-A572F67C03C1} = {3C695301-9C39-4715-8EDD-817C63EF81E7}
-	EndGlobalSection
 	GlobalSection(ExtensibilityGlobals) = postSolution
 		SolutionGuid = {0915B762-D4AE-4892-BF4F-AC428A687B02}
 	EndGlobalSection
diff --git a/SIL.Core/IO/PathUtilities.cs b/SIL.Core/IO/PathUtilities.cs
index 3b61d832..29c1f26d 100644
--- a/SIL.Core/IO/PathUtilities.cs
+++ b/SIL.Core/IO/PathUtilities.cs
@@ -224,8 +224,11 @@ public static bool DeleteToRecycleBin(string filePath, FileOperationFlags flags)
 			// move file or directory
 			if (Directory.Exists(filePath) || File.Exists(filePath))
 			{
-				var trashPath = Path.Combine(Environment.GetFolderPath(
-					Environment.SpecialFolder.LocalApplicationData), "Trash");
+				var localDataPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+				// We want to use the standard Trash location for flatpak, not the sandboxed one.
+				if (Platform.IsFlatpak)
+					localDataPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), ".local/share");
+				var trashPath = Path.Combine(localDataPath, "Trash");
 				var trashedFileName = Path.GetRandomFileName();
 				if (!Directory.Exists(trashPath))
 				{
@@ -239,6 +242,13 @@ public static bool DeleteToRecycleBin(string filePath, FileOperationFlags flags)
 
 				WriteTrashInfoFile(trashPath, filePath, trashedFileName);
 				// Directory.Move works for directories and files
+				if (Platform.IsFlatpak)
+				{
+					// flatpak mounts pieces of the file system differently for sandboxing, so Move won't work.
+					DirectoryHelper.Copy(filePath, recyclePath);
+					Directory.Delete(filePath, true);
+					return true;
+				}
 				DirectoryHelper.Move(filePath, recyclePath);
 				return true;
 			}
@@ -380,6 +390,11 @@ private static string GetDefaultFileManager()
 				var desktopFilename = Path.Combine(
 					Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData),
 					"applications", desktopFile);
+				if (Platform.IsFlatpak && !File.Exists(desktopFilename))
+				{
+					// We didn't find the .desktop file yet, so check in /app/share/applications
+					desktopFilename = Path.Combine("/app/share/applications", desktopFile);
+				}
 				if (!File.Exists(desktopFilename))
 				{
 					// We didn't find the .desktop file yet, so check in ~/.local/share/applications
diff --git a/SIL.Core/PlatformUtilities/Platform.cs b/SIL.Core/PlatformUtilities/Platform.cs
index 41b2e022..19d1b453 100644
--- a/SIL.Core/PlatformUtilities/Platform.cs
+++ b/SIL.Core/PlatformUtilities/Platform.cs
@@ -19,6 +19,7 @@ public static class Platform
 
 		public static bool IsUnix => Environment.OSVersion.Platform == PlatformID.Unix;
 		public static bool IsWasta => IsUnix && System.IO.File.Exists("/etc/wasta-release");
+		public static bool IsFlatpak => IsLinux && !String.IsNullOrEmpty(Environment.GetEnvironmentVariable("FLATPAK_ID"));
 
 		public static bool IsCinnamon => IsUnix &&
 			(SessionManager.StartsWith("/usr/bin/cinnamon-session") ||
diff --git a/SIL.Windows.Forms/ImageToolbox/ImageGallery/ImageCollectionManager.cs b/SIL.Windows.Forms/ImageToolbox/ImageGallery/ImageCollectionManager.cs
index 5561bacb..7d70b09c 100644
--- a/SIL.Windows.Forms/ImageToolbox/ImageGallery/ImageCollectionManager.cs
+++ b/SIL.Windows.Forms/ImageToolbox/ImageGallery/ImageCollectionManager.cs
@@ -58,6 +58,8 @@ internal ImageCollectionManager(string searchLanguageId)
 			ImageCollectionsFolder =
 				Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData)
 					.CombineForPath("SIL", "ImageCollections");
+			if (SIL.PlatformUtilities.Platform.IsFlatpak && ImageCollectionsFolder.StartsWith("/usr/"))
+				ImageCollectionsFolder = ImageCollectionsFolder.Replace("/usr/", "/app/");
 			_searchLanguage = searchLanguageId;
 			_collections = new List<ImageCollection>();
 		}
diff --git a/SIL.Windows.Forms/ImageToolbox/ImageToolboxDialog.cs b/SIL.Windows.Forms/ImageToolbox/ImageToolboxDialog.cs
index 5cafc0d6..7f5cfe04 100644
--- a/SIL.Windows.Forms/ImageToolbox/ImageToolboxDialog.cs
+++ b/SIL.Windows.Forms/ImageToolbox/ImageToolboxDialog.cs
@@ -45,14 +45,14 @@ private void _okButton_Click(object sender, EventArgs e)
 			//enhance: doesn't tell us all that much.
 			UsageReporter.SendNavigationNotice("ImageToolboxDialog/Ok");
 			DialogResult = (ImageInfo==null || ImageInfo.Image==null)? DialogResult.Cancel : DialogResult.OK;
-			_imageToolboxControl.Closing();
+			//_imageToolboxControl.Closing();
 			Close();
 		}
 
 		private void _cancelButton_Click(object sender, EventArgs e)
 		{
 			DialogResult = DialogResult.Cancel;
-			_imageToolboxControl.Closing();
+			//_imageToolboxControl.Closing();
 			Close();
 		}
 
