diff --git a/SIL.Core/SIL.Core.csproj b/SIL.Core/SIL.Core.csproj
index be5068b1..ecf243fb 100644
--- a/SIL.Core/SIL.Core.csproj
+++ b/SIL.Core/SIL.Core.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>net461</TargetFrameworks>
     <Description>SIL.Core provides general utilities for language software. It is the base library for all Palaso libraries.</Description>
   </PropertyGroup>
 
diff --git a/SIL.DblBundle/SIL.DblBundle.csproj b/SIL.DblBundle/SIL.DblBundle.csproj
index e1eff2b6..8c7b50cc 100644
--- a/SIL.DblBundle/SIL.DblBundle.csproj
+++ b/SIL.DblBundle/SIL.DblBundle.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>net461</TargetFrameworks>
     <RootNamespace>SIL.DblBundle</RootNamespace>
     <AssemblyTitle>SIL.DblBundle</AssemblyTitle>
     <Description>SIL.DblBundle provides classes for building a Digital Bible Library (DBL) bundle.</Description>
diff --git a/SIL.Scripture/SIL.Scripture.csproj b/SIL.Scripture/SIL.Scripture.csproj
index 5c7f6888..dbb99182 100644
--- a/SIL.Scripture/SIL.Scripture.csproj
+++ b/SIL.Scripture/SIL.Scripture.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>net461</TargetFrameworks>
     <RootNamespace>SIL.Scripture</RootNamespace>
     <AssemblyTitle>SIL.Scripture</AssemblyTitle>
     <Description>SIL.Scripture provides classes for working with Scripture data such as references and versifications.</Description>
diff --git a/SIL.WritingSystems/SIL.WritingSystems.csproj b/SIL.WritingSystems/SIL.WritingSystems.csproj
index cc23dbe9..dad1042f 100644
--- a/SIL.WritingSystems/SIL.WritingSystems.csproj
+++ b/SIL.WritingSystems/SIL.WritingSystems.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>net461</TargetFrameworks>
     <Description>SIL.WritingSystems contains classes for managing and persisting writing systems using the Locale Data Markup Language (LDML) format. This library also contains classes for processing IETF (BCP-47) language tags and accessing the SIL Locale Data Repository (SLDR).</Description>
   </PropertyGroup>
 
