diff --git a/Geckofx-Winforms/GeckoWebBrowser.cs b/Geckofx-Winforms/GeckoWebBrowser.cs
--- a/Geckofx-Winforms/GeckoWebBrowser.cs
+++ b/Geckofx-Winforms/GeckoWebBrowser.cs
@@ -207,6 +207,7 @@ namespace Gecko
                 if (_Document != null)
                     _Document.Dispose();
                 _Document = null;
+                _topLevelWindow = null;
             }
             base.Dispose(disposing);
         }
diff --git a/Geckofx-Winforms/Geckofx-Winforms.csproj b/Geckofx-Winforms/Geckofx-Winforms.csproj
--- a/Geckofx-Winforms/Geckofx-Winforms.csproj
+++ b/Geckofx-Winforms/Geckofx-Winforms.csproj
@@ -294,14 +294,6 @@
     </PropertyGroup>
   </Target>
   <Target Name="AfterBuildLinux" Condition=" '$(OS)' == 'Unix' ">
-    <!-- using explicit version rather than 'latest' as nuget.exe 4.6.2 and newer doesn't work with mono 4.2) -->
-    <Exec Command="wget -nc https://dist.nuget.org/win-x86-commandline/v4.5.1/nuget.exe" />
-    <Exec Command="mkdir -p Firefox-Linux64" />
-    <Exec Command="cp -ra ../PutXulRunnerFolderHere/64/Firefox/* Firefox-Linux64/" />
-    <Exec Command="mono nuget.exe pack Geckofx-Winforms.64.Linux.nuspec -Prop Configuration=$(Configuration) -Prop Version=`monodis --assembly $(TargetDir)\$(TargetName).dll | grep Version | awk '{print $2}'` -Prop FirefoxDir=Firefox-Linux64 -MinClientVersion 2.5" />
-    <Exec Command="mkdir -p Firefox-Linux32" />
-    <Exec Command="cp -ra ../PutXulRunnerFolderHere/32/Firefox/* Firefox-Linux32/" />
-    <Exec Command="mono nuget.exe pack Geckofx-Winforms.32.Linux.nuspec -Prop Configuration=$(Configuration) -Prop Version=`monodis --assembly $(TargetDir)\$(TargetName).dll | grep Version | awk '{print $2}'` -Prop FirefoxDir=Firefox-Linux32 -MinClientVersion 2.5" />
   </Target>
   <Target Name="AfterBuildWindows" Condition=" '$(OS)' == 'Windows_NT' AND '$(Platform)' == 'x86'" DependsOnTargets="GetAssemblyVersionNumbers">
     <CallTarget Targets="CopyFirefoxFiles" />
@@ -315,13 +307,13 @@
     <WebDownload Condition="!Exists('nuget.exe')" Filename="nuget.exe" FileUri="https://dist.nuget.org/win-x86-commandline/latest/nuget.exe" />
     <Exec Command="nuget pack Geckofx-Winforms.64.nuspec -Prop Configuration=$(Configuration) -Prop Platform=x64 -Prop Version=$(AssemblyVersionMajor).$(AssemblyVersionMinor).$(AssemblyVersionPatch)$(AssemblyVersionPreRelease) -MinClientVersion 2.5" />
   </Target>
-  <Import Project="..\packages\Geckofx60.32.Windows.0.18.0\build\Geckofx60.32.Windows.targets" Condition="Exists('..\packages\Geckofx60.32.Windows.0.18.0\build\Geckofx60.32.Windows.targets')" />
-  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+  <!--Import Project="..\packages\Geckofx60.32.Windows.0.18.0\build\Geckofx60.32.Windows.targets" Condition="Exists('..\packages\Geckofx60.32.Windows.0.18.0\build\Geckofx60.32.Windows.targets')" /-->
+  <!--Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
       <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
     </PropertyGroup>
     <Error Condition="!Exists('..\packages\Geckofx60.32.Windows.0.18.0\build\Geckofx60.32.Windows.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Geckofx60.32.Windows.0.18.0\build\Geckofx60.32.Windows.targets'))" />
     <Error Condition="!Exists('..\packages\Geckofx60.64.Windows.0.18.0\build\Geckofx60.64.Windows.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Geckofx60.64.Windows.0.18.0\build\Geckofx60.64.Windows.targets'))" />
-  </Target>
-  <Import Project="..\packages\Geckofx60.64.Windows.0.18.0\build\Geckofx60.64.Windows.targets" Condition="Exists('..\packages\Geckofx60.64.Windows.0.18.0\build\Geckofx60.64.Windows.targets')" />
-</Project>
\ No newline at end of file
+  </Target-->
+  <!--Import Project="..\packages\Geckofx60.64.Windows.0.18.0\build\Geckofx60.64.Windows.targets" Condition="Exists('..\packages\Geckofx60.64.Windows.0.18.0\build\Geckofx60.64.Windows.targets')" /-->
+</Project>
diff --git a/Geckofx-Winforms/Linux/GtkReparentingWrapperNoThread.cs b/Geckofx-Winforms/Linux/GtkReparentingWrapperNoThread.cs
--- a/Geckofx-Winforms/Linux/GtkReparentingWrapperNoThread.cs
+++ b/Geckofx-Winforms/Linux/GtkReparentingWrapperNoThread.cs
@@ -262,12 +262,12 @@ namespace GtkDotNet
                 if (_filterAdded)
                     _popupWindow.Window.RemoveFilter(FilterFunc);
                 _filterAdded = false;
-                _popupWindow.Window.Destroy();
+//                _popupWindow.Window.Destroy();
             }
             _parent.HandleCreated -= HandleParentCreated;
 
-            _popupWindow?.Destroy();
-            _popupWindow?.Dispose();
+//            _popupWindow?.Destroy();
+//            _popupWindow?.Dispose();
             _popupWindow = null;
             _xDisplayPointer = IntPtr.Zero;
 
diff --git a/Geckofx-Winforms/Linux/GtkWrapperNoThread.cs b/Geckofx-Winforms/Linux/GtkWrapperNoThread.cs
--- a/Geckofx-Winforms/Linux/GtkWrapperNoThread.cs
+++ b/Geckofx-Winforms/Linux/GtkWrapperNoThread.cs
@@ -72,8 +72,8 @@ namespace GtkDotNet
 			System.Windows.Forms.Application.Idle -= HandleSystemWindowsFormsApplicationIdle;
 			if (_popupWindow != null)
 			{
-				_popupWindow.Destroy();
-				_popupWindow.Dispose();
+//				_popupWindow.Destroy();
+//				_popupWindow.Dispose();
 				_popupWindow = null;
 			}
 		}
diff --git a/Geckofx.sln b/Geckofx.sln
--- a/Geckofx.sln
+++ b/Geckofx.sln
@@ -3,19 +3,8 @@ Microsoft Visual Studio Solution File, F
 # Visual Studio 2012
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Geckofx-Core", "Geckofx-Core\Geckofx-Core.csproj", "{6AB4EEE6-1889-43AB-83F7-A4F083230BE7}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "GeckoFxTest", "GeckoFxTest\GeckoFxTest.csproj", "{9F904D01-0295-4866-8F11-34482E2E2715}"
-EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "GeckofxUnitTests", "GeckofxUnitTests\GeckofxUnitTests.csproj", "{F11A8E22-9540-472A-8D08-8262060A33AD}"
-EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Geckofx-Winforms", "Geckofx-Winforms\Geckofx-Winforms.csproj", "{D7ADC901-E1C4-480E-8F6D-B1FFD5CF87E8}"
 EndProject
-Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".build", ".build", "{E17F5B67-C261-4E89-95CE-7B3CCC68F81C}"
-	ProjectSection(SolutionItems) = preProject
-		Build.proj = Build.proj
-		.build\MSBuild.Community.Tasks.dll = .build\MSBuild.Community.Tasks.dll
-		.build\MSBuild.Community.Tasks.targets = .build\MSBuild.Community.Tasks.targets
-	EndProjectSection
-EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug_Linux|Any CPU = Debug_Linux|Any CPU
@@ -74,75 +63,6 @@ Global
 		{6AB4EEE6-1889-43AB-83F7-A4F083230BE7}.Release|Mixed Platforms.Build.0 = Release_Windows|x86
 		{6AB4EEE6-1889-43AB-83F7-A4F083230BE7}.Release|x86.ActiveCfg = Release_Windows|x86
 		{6AB4EEE6-1889-43AB-83F7-A4F083230BE7}.Release|x86.Build.0 = Release_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Linux|Any CPU.ActiveCfg = Debug_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Linux|Any CPU.Build.0 = Debug_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Linux|Mixed Platforms.ActiveCfg = Debug_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Linux|Mixed Platforms.Build.0 = Debug_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Linux|x86.ActiveCfg = Debug_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Linux|x86.Build.0 = Debug_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Windows|Any CPU.ActiveCfg = Debug_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Windows|Any CPU.Build.0 = Debug_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Windows|Mixed Platforms.ActiveCfg = Debug_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Windows|Mixed Platforms.Build.0 = Debug_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Windows|x86.ActiveCfg = Debug_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug_Windows|x86.Build.0 = Debug_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug|Any CPU.ActiveCfg = Debug|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug|Any CPU.Build.0 = Debug|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug|Mixed Platforms.ActiveCfg = Debug|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug|Mixed Platforms.Build.0 = Debug|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug|x86.ActiveCfg = Debug|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Debug|x86.Build.0 = Debug|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Linux|Any CPU.ActiveCfg = Release_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Linux|Any CPU.Build.0 = Release_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Linux|Mixed Platforms.ActiveCfg = Release_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Linux|Mixed Platforms.Build.0 = Release_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Linux|x86.ActiveCfg = Release_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Linux|x86.Build.0 = Release_Linux|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Windows|Any CPU.ActiveCfg = Release_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Windows|Any CPU.Build.0 = Release_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Windows|Mixed Platforms.ActiveCfg = Release_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Windows|Mixed Platforms.Build.0 = Release_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Windows|x86.ActiveCfg = Release_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release_Windows|x86.Build.0 = Release_Windows|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release|Any CPU.ActiveCfg = Release|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release|Any CPU.Build.0 = Release|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release|Mixed Platforms.ActiveCfg = Release|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release|Mixed Platforms.Build.0 = Release|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release|x86.ActiveCfg = Release|x86
-		{9F904D01-0295-4866-8F11-34482E2E2715}.Release|x86.Build.0 = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Linux|Any CPU.ActiveCfg = Debug|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Linux|Any CPU.Build.0 = Debug|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Linux|Mixed Platforms.ActiveCfg = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Linux|Mixed Platforms.Build.0 = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Linux|x86.ActiveCfg = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Linux|x86.Build.0 = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Windows|Any CPU.ActiveCfg = Debug|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Windows|Any CPU.Build.0 = Debug|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Windows|Mixed Platforms.ActiveCfg = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Windows|Mixed Platforms.Build.0 = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Windows|x86.ActiveCfg = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug_Windows|x86.Build.0 = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug|Any CPU.ActiveCfg = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug|Any CPU.Build.0 = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug|Mixed Platforms.ActiveCfg = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug|Mixed Platforms.Build.0 = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug|x86.ActiveCfg = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Debug|x86.Build.0 = Debug|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Linux|Any CPU.ActiveCfg = Release|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Linux|Mixed Platforms.ActiveCfg = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Linux|Mixed Platforms.Build.0 = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Linux|x86.ActiveCfg = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Linux|x86.Build.0 = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Windows|Any CPU.ActiveCfg = Release|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Windows|Mixed Platforms.ActiveCfg = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Windows|Mixed Platforms.Build.0 = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release_Windows|x86.ActiveCfg = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release|Any CPU.Build.0 = Release|Any CPU
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release|Mixed Platforms.ActiveCfg = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release|Mixed Platforms.Build.0 = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release|x86.ActiveCfg = Release|x86
-		{F11A8E22-9540-472A-8D08-8262060A33AD}.Release|x86.Build.0 = Release|x86
 		{D7ADC901-E1C4-480E-8F6D-B1FFD5CF87E8}.Debug_Linux|Any CPU.ActiveCfg = Debug_Linux|Any CPU
 		{D7ADC901-E1C4-480E-8F6D-B1FFD5CF87E8}.Debug_Linux|Any CPU.Build.0 = Debug_Linux|Any CPU
 		{D7ADC901-E1C4-480E-8F6D-B1FFD5CF87E8}.Debug_Linux|Mixed Platforms.ActiveCfg = Debug_Linux|x86
