--- gtkglext-1.2.0.orig/gdk/Makefile.am
+++ gtkglext-1.2.0/gdk/Makefile.am
@@ -180,7 +180,7 @@ $(srcdir)/gdkglenumtypes.c: @REBUILD@ $(
 #
 # Rule to install gdkglext-config.h header file
 #
-configexecincludedir = $(libdir)/gtkglext-@GTKGLEXT_API_VERSION@/include
+configexecincludedir = $(includedir)/gtkglext-@GTKGLEXT_API_VERSION@
 #configexecinclude_DATA = gdkglext-config.h
 
 install-exec-local: gdkglext-config.h
