--- ../orig/gtkglext-1.2.0/gdk/x11/gdkglquery-x11.c	2004-02-20 09:38:36.000000000 +0000
+++ ./gdk/x11/gdkglquery-x11.c	2012-09-02 07:43:29.000000000 +0000
@@ -356,7 +356,7 @@
 {
   typedef GdkGLProc (*__glXGetProcAddressProc) (const GLubyte *);
   static __glXGetProcAddressProc glx_get_proc_address = (__glXGetProcAddressProc) -1;
-  gchar *file_name;
+  gchar *file_name, *file_name_2;
   GModule *module;
   GdkGLProc proc_address = NULL;
 
@@ -371,12 +371,28 @@
            */
 
           file_name = g_module_build_path (NULL, "GL");
+          file_name_2 = NULL;
+          if( file_name )
+            file_name_2 = g_malloc( strlen( file_name ) + 3 );
+          module = NULL;
+          if( file_name_2 )
+          {
+            strcpy( file_name_2, file_name );
+            strcat( file_name_2, ".1" );
+            GDK_GL_NOTE (MISC, g_message (" - Open %s", file_name_2));
+            module = g_module_open (file_name_2, G_MODULE_BIND_LAZY);
+            g_free (file_name_2);
+            file_name_2 = NULL;
+          }
+          if( ! module )
+          {
           GDK_GL_NOTE (MISC, g_message (" - Open %s", file_name));
           module = g_module_open (file_name, G_MODULE_BIND_LAZY);
-          g_free (file_name);
+          }
 
           if (module != NULL)
             {
+              g_free (file_name);
               g_module_symbol (module, "glXGetProcAddress",
                                (gpointer) &glx_get_proc_address);
               if (glx_get_proc_address == NULL)
@@ -396,6 +412,7 @@
           else
             {
               g_warning ("Cannot open %s", file_name);
+              g_free (file_name);
               glx_get_proc_address = NULL;
               return NULL;
             }
@@ -416,12 +433,28 @@
 
       /* libGL */
       file_name = g_module_build_path (NULL, "GL");
+      file_name_2 = NULL;
+      if( file_name )
+        file_name_2 = g_malloc( strlen( file_name ) + 3 );
+      module = NULL;
+      if( file_name_2 )
+      {
+        strcpy( file_name_2, file_name );
+        strcat( file_name_2, ".1" );
+        GDK_GL_NOTE (MISC, g_message (" - Open %s", file_name_2));
+        module = g_module_open (file_name_2, G_MODULE_BIND_LAZY);
+        g_free (file_name_2);
+        file_name_2 = NULL;
+      }
+      if( ! module )
+      {
       GDK_GL_NOTE (MISC, g_message (" - Open %s", file_name));
       module = g_module_open (file_name, G_MODULE_BIND_LAZY);
-      g_free (file_name);
+      }
 
       if (module != NULL)
         {
+          g_free (file_name);
           g_module_symbol (module, proc_name, (gpointer) &proc_address);
           GDK_GL_NOTE (MISC, g_message (" - g_module_symbol () - %s",
                                         proc_address ? "succeeded" : "failed"));
@@ -430,6 +463,7 @@
       else
         {
           g_warning ("Cannot open %s", file_name);
+          g_free (file_name);
         }
 
       if (proc_address == NULL)
