diff --git a/libsecret/secret-file-collection.c b/libsecret/secret-file-collection.c
index 2e90e3c..a27cf04 100644
--- a/libsecret/secret-file-collection.c
+++ b/libsecret/secret-file-collection.c
@@ -16,6 +16,7 @@
 
 #include "secret-file-collection.h"
 
+#include "egg/egg-libgcrypt.h"
 #include "egg/egg-secure-memory.h"
 
 EGG_SECURE_DECLARE (secret_file_collection);
@@ -271,6 +272,8 @@ secret_file_collection_class_init (SecretFileCollectionClass *klass)
 		   g_param_spec_boxed ("password", "password", "Password",
 				       SECRET_TYPE_VALUE,
 				       G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY));
+
+	egg_libgcrypt_initialize ();
 }
 
 static void
