diff --git a/src/luaconf.h b/src/luaconf.h
index 137103e..fe255c4 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -11,6 +11,9 @@
 #include <limits.h>
 #include <stddef.h>
 
+#define LUA_COMPAT_5_1
+#define LUA_COMPAT_5_2
+
 
 /*
 ** ===================================================================
@@ -223,7 +226,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"$FLATPAK_DEST/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
@@ -756,7 +759,7 @@
 ** of a function in debug information.
 ** CHANGE it if you want a different size.
 */
-#define LUA_IDSIZE	60
+#define LUA_IDSIZE	512
 
 
 /*
