diff --git a/client/CMakeLists.txt b/client/CMakeLists.txt
index 6a48248..31f6136 100644
--- a/client/CMakeLists.txt
+++ b/client/CMakeLists.txt
@@ -1,13 +1,5 @@
-add_executable( TSLDownload TSLDownload.cpp )
-target_link_libraries(TSLDownload Qt${QT_VERSION_MAJOR}::Network)
 get_target_property(qtCore_install_prefix Qt${QT_VERSION_MAJOR}::qmake IMPORTED_LOCATION)
 get_filename_component(qtCore_install_prefix ${qtCore_install_prefix} DIRECTORY)
-add_custom_command(
-	OUTPUT TSL.qrc
-	DEPENDS TSLDownload
-	COMMAND $<TARGET_FILE:TSLDownload> "${CMAKE_CURRENT_BINARY_DIR}" ${TSL_URL} ${TSL_INCLUDE}
-	WORKING_DIRECTORY ${qtCore_install_prefix}
-)
 
 configure_file( translations/tr.qrc tr.qrc COPYONLY )
 qt_add_translation(SOURCES translations/en.ts translations/et.ts translations/ru.ts)
@@ -20,7 +12,7 @@
 	images/images.qrc
 	fonts/fonts.qrc
 	${CMAKE_CURRENT_BINARY_DIR}/tr.qrc
-	${CMAKE_CURRENT_BINARY_DIR}/TSL.qrc
+	TSL.qrc
 	main.cpp
 	Application.cpp
 	CheckConnection.cpp
