--- DigiDoc4-Client/client/QPKCS11.cpp	2021-06-03 10:22:23.486463791 +0300
+++ DigiDoc4-Client/client/QPKCS11-fix.cpp	2021-06-03 16:43:00.722224476 +0300
@@ -364,7 +364,7 @@
 #elif defined(Q_OS_WIN)
 		{ "opensc-pkcs11.dll", {} },
 #else
-		{ "opensc-pkcs11.so", {} },
+		{ "/app/lib/opensc-pkcs11.so", {} },
 #if defined(Q_OS_LINUX)
 		{ "/opt/latvia-eid/lib/eidlv-pkcs11.so", "3BDD18008131FE45904C41545649412D65494490008C" }, // LV-G1
 		{ "/opt/latvia-eid/lib/eidlv-pkcs11.so", "3BDB960080B1FE451F830012428F536549440F900020" }, // LV-G2

