diff -up a/nss/build.sh.orig b/nss/build.sh
--- a/nss/build.sh.orig	2017-10-13 11:24:47.436647729 +0200
+++ b/nss/build.sh	2017-10-13 11:25:25.141116333 +0200
@@ -196,7 +196,7 @@ if [ "$rebuild_gyp" = 1 ]; then
         set_nspr_path "$obj_dir/include/nspr:$obj_dir/lib"
     fi
 
-    run_verbose run_scanbuild ${GYP} -f ninja "${gyp_params[@]}" "$cwd"/nss.gyp
+    run_verbose run_scanbuild ${GYP} --no-parallel -f ninja "${gyp_params[@]}" "$cwd"/nss.gyp
 
     mv -f "$gyp_config".new "$gyp_config"
 fi
