diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index cc433d0..9a824be 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -8,7 +8,7 @@ add_custom_target(python ALL
     COMMAND python3 setup.py build
     DEPENDS opendht opendht_cpp.pxd opendht.pyx)
 
-install(CODE "execute_process(COMMAND python3 setup.py install --root=\$ENV{DESTDIR}/ WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})")
+install(CODE "execute_process(COMMAND python3 setup.py install --skip-build --prefix=\$ENV{FLATPAK_DEST}/ --root=/  --optimize=1 WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})")
 if (OPENDHT_TOOLS)
 	install(PROGRAMS tools/dhtcluster.py DESTINATION ${CMAKE_INSTALL_BINDIR} RENAME dhtcluster)
 endif()
